---
title: "DESeq2 and PCA Analysis"
subtitle: "For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird "
date: Last Knit "`r Sys.Date()`"
author: "Last Substantive change November 2023"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
    number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE, fig.align="center", comment=FALSE)
```

```{r libraries used, echo=FALSE}
library(plyr)
library(reshape2)
library(ggplot2)
library(gridExtra)
library(ggpubr)
library(PCAtools)
library(fdrtool)
library(stringr)
library(lubridate)
library(DESeq2)
library(kableExtra)
library(VennDiagram)
library(wesanderson)
library(WGCNA)
library(clusterProfiler)
library(flashClust)
library(limma)
library(UpSetR)
library(viridis)
  
source("config.R") #key colour and other variables go in here
source("functions.R") #repeated functions go in here.
```

Our primary aim is to explore whether among-individual variation in T modulated cooperative behaviour is associated with differences in gene expression.  We have 5 main traits that were measured.

*  Status: Territorial or Floater
* Testosterone concentration
  + *mean_T* is best estimate of among individual variation in T across all birds ('r summary(behav$mean_T)`). This measure is repeatable (Ryder et al 2020) within individuals and could be considered a hormonal phenotype. 
* Strength
 + Mean number of social interactions per 10 hr day as detected by telemetry.
 
 
Here, I take the filtered dataset from the QC process, and run a whole brain analysis, as well as individual tissue analysis. I also explore the relationships between our explanatory variables.  

```{r reading & splitting data, echo=TRUE}

key<- read.csv("../data_filtered/data_key_Parsed_ReplicatesRemoved.csv")
behav<- read.csv("../data_unfiltered/PIPFIL_T_and_behav_data.csv")
rownames(key)<- key$X

key$Color_ID<- sub("/","", key$Color_ID)
key<- plyr::rename(key, replace=c("Color_ID"="colorID"))
behav$status<- plyr::revalue(behav$status, c("floa"="floater", "terr"="territorial"))
key_behav<- merge(key, behav, by="colorID")
key_behav<- key_behav[!is.na(key_behav$last_behav),]


#create a data.frame with all of the observations. 
not_in_behav<- key[!key$X %in% key_behav$X,]
cols_not_key<- colnames(key_behav)[!colnames(key_behav) %in% colnames(key)]
cols_not_key_df<- data.frame(matrix(NA, nrow = nrow(not_in_behav), ncol = length(cols_not_key)))
colnames(cols_not_key_df)<- cols_not_key
not_in_behav<- cbind(not_in_behav, cols_not_key_df)

key_behav<- rbind(key_behav, not_in_behav)

rownames(key_behav)<- key_behav$X


key_behav$Tissue<- factor(key_behav$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
key_behav<- key_behav[order(rownames(key_behav)),]
key_behav$Class<- as.factor(key_behav$Class)
key_behav$Class<- revalue(key_behav$Class, replace=c("SCB floater"="Predefinitive floater", "DCB floater "="Definitive floater", "DCB territorial"="Territorial"))
key_behav$Class<- factor(key_behav$Class, levels=c("Predefinitive floater", "Definitive floater", "Territorial"))
rownames(key_behav)<- key_behav$sampleID
key_behav<- key_behav[order(key_behav$sampleID),]
key_behav$Year<- as.factor(key_behav$Year)
key_behav$Batch<- as.factor(key_behav$Batch)
key_behav$Status<- as.factor(key_behav$Status)

key_behav_unique<- key_behav[!duplicated(key_behav$colorID),]

#read the raw count data
data<- read.csv("../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv")
data$X[data$X=="LOC113993669"] <- "CYP19A1"
data$X[data$X=="LOC113983511"] <- "OXT"
data$X[data$X=="LOC113983498"] <- "AVP"
data$X[data$X=="LOC113982601"] <- "AVPR2"
rownames(data)<- data$X
data$X<- NULL

### Gene Ontology


go_terms<- read.csv("../GO_annotations/pfil_GO_key_raw.csv")
go_terms<- plyr::rename(go_terms, replace=c("GeneID"="gene"))
go2gene_bp<- go_terms[which(go_terms$Aspect=="P"),c("GO_ID", "gene")]
go_obo<- read.csv("../GO_annotations/ontology_obo_out.csv")
go_obo<- plyr::rename(go_obo, replace=c("id"="GO_ID"))
go2name_bp<- go_obo[which(go_obo$namespace=="biological_process"),c("GO_ID", "name")]


## annotating the LOC no
genes_key<- read.csv("../GO_annotations/Maggies_annotations_modifiedR.csv")
genes_key<- plyr::rename(genes_key, replace=c("GeneID"="gene"))
genes_key$gene[genes_key$gene=="LOC113993669"] <- "CYP19A1"
genes_key$gene[genes_key$gene=="LOC113983511"] <- "OXT"
genes_key$gene[genes_key$gene=="LOC113983498"] <- "AVP"
genes_key$gene[genes_key$gene=="LOC113982601"] <- "AVPR2"
data_genes<- data.frame(gene=rownames(data))
genes_key<- merge(data_genes, genes_key, by="gene", all.x=TRUE)
#genes_key<- genes_key[which(grepl("LOC[0-9]+",genes_key$gene)),]

genes_key$display_gene_ID<- ifelse(grepl("LOC[0-9]+", genes_key$gene) & !grepl("LOC[0-9]+",genes_key$best_anno) & !is.na(genes_key$best_anno), paste0(genes_key$gene," (",genes_key$best_anno,")"), as.character(genes_key$gene))
genes_key<- genes_key[,c("gene","best_anno","display_gene_ID")]
  
candidates2<- c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B", "AVPR2")
```

```{r keep}
keep<- c("data","key","behav","key_behav", "dodge", "peri_theme", "keep", "tissues_status", "tissues_mean_T", "tissues_final_T", "tissues_strength", "rm_continuous_outliers", "status_cols", "interactions","go_key","go2gene_bp","go2name_bp", "peri_figure","peri_figure_supp","dj_col", "res_go_bp","gene_plots","status_cols2", "peri_geneplots","res_gsea_bp", "candidates2","genes_key", "geom.text.size", "tissue_cols","volcano_plots")
```


I have to run each tissue separately in part because of linear combinations of batch variables and tissues (some tissues were sequenced only on a single flow cell) that means DESeq2 throws a "model not full rank" error. But also we are interested in the within-tissue gene expression.

<br><br>


# Gonads (GON)

I will use the first two tissues to demonstrate the processes applied across all tissues, but code echoing will be suppressed in future tissues so as to save space.

```{r GON filtering, echo=TRUE}
tissue="GON"
gon_key<- subset(key_behav, Tissue==tissue)
gon_key<- droplevels(gon_key)
gon_data<- data[,colnames(data) %in% rownames(gon_key)]


start<- nrow(gon_data)
#remove genes with less than 5 reads
gon_data$avg_count<- apply(gon_data, 1, mean)
gon_data<- gon_data[gon_data$avg_count>5,]
gon_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gon_data$percent_0<- apply(gon_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gon_data)/2
gon_data<- gon_data[gon_data$percent_0<=thresh,]
gon_data$percent_0<-NULL

```

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(gon_data)`

## Checking the sampling

Before I go into the analyses, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gonad samples for `r nrow(gon_key)`, with respect to T and the tracking data we have `r nrow(gon_key[gon_key$Batch!="pilot",])` samples.

Here, we will run all the data to check for DEGs against status, then subset the data to only include those individuals with T and behavioural data. 


```{r GON sampling status, echo=TRUE, fig.align='center'}

fsb<- fisher.test(table(gon_key$Status, gon_key$Batch))
knitr::kable(table(gon_key$Status, gon_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value)) %>% kable_styling()

fsy<- fisher.test(table(gon_key$Status, gon_key$Year))
knitr::kable(table(gon_key$Status, gon_key$Year), caption=paste0("Status across sampling years, fisher test p=", fsy$p.value)) %>% kable_styling()

```



## Status GON

First, I will run through a data checking process. I will check for outliers, and then find out how our potential nuisance variables effect our gene expression data. If they have effects they will be incorporated into the final DESeq2 models. 

<br>

### Expression data exploration

```{r GON status set up, echo=TRUE}
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 


plotDispEsts(dd)

#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables
vsd_data<- getVarianceStabilizedData(dd)

p <- pca(vsd_data, metadata = gon_key)
a<- biplot(p, lab=gon_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Gonads all samples PCA") + peri_figure
a
## check to see how our interest and nuisance variables are correlated to the PC axes. This will help decide on model form for DESeq.
b<- eigencorplot(p, metavars = c("time_kill","Batch","Year", "Status"))
b

#use WGCNA connectivity measure to identify outliers.

datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

A=adjacency(t(datExpr0),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)
#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```



Based on the eigencor plot, I will include Batch as a variable in my analyses, as it is associated with PC4.

<br>

#### Corrections applied

Also let's see how the data look when we account for Batch in our model by using limma::RemoveBatchEffect()

```{r GON removeBatchEffect, echo=TRUE}
#remove batch effect
mat<- limma::removeBatchEffect(vsd_data, gon_key$Batch)
p <- pca(mat, metadata = gon_key)
biplot(p, lab=gon_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")

#use this if Batch and Year are both pinged in the previous section. 
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))


traits<- gon_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

### Analysis

Now run DEseq with the updated model form and extract DEGs.

Formula ~ Batch + Status.

```{r GON Status2, fig.height=4, fig.width=10, echo=TRUE}
varname="Status"
design(dd)<- formula(~ Batch + Status)



dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",design(dd)))
hist(res$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
dev.off()


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$pvalue),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_GON_Status.csv", row.names=TRUE)

```

```{r GON status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r GON status geneplotting, fig.height=8, fig.width=14, echo=TRUE}
gene_plots(out_res,c("Status","Class"),6, status_cols)
g

```

<br>

### Gene Ontology

Conduct Gene Set enrichment analysis (GSEA) when there are few significant differentially expressed genes. This will leverage the whole list. Then, I will also conduct foreground background testing to get an explicit view of either the top genes after FDR correction (q<0.1), or the top genes without FDR correction (p<0.05)


```{r GON Status GSEA, echo=TRUE, fig.height=10, fig.width=18}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GON Status enrichment, echo=TRUE}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))%>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))



write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```


Below is the code for making the figures in the supplement.

```{r, echo=TRUE}

g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename="../DE_results/supp_figure_GON_status.png", plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```




<br>

## Testosterone GON

Given that the behaviors are best correlated with mean T, then this is our variable of interest

<br>

### Expression data exploration

But first, we need to look at these new data in the gonads with the new subset with respect to our interest variables in the PCA. 

```{r GON gene filtering 2, echo=TRUE}
rm(list= ls()[!(ls() %in% c(keep,"gon_data","gon_behav"))])
tissue="GON"
gon_behav<- subset(key_behav, Tissue==tissue)
gon_behav<- subset(gon_behav, Batch!="pilot")
gon_behav<- droplevels(gon_behav)
gon_data<- data[,colnames(data) %in% rownames(gon_behav)]

start<- nrow(gon_data)
#remove genes with less than 5 reads
gon_data$avg_count<- apply(gon_data, 1, mean)
gon_data<- gon_data[gon_data$avg_count>5,]
gon_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gon_data$percent_0<- apply(gon_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gon_data)/2
gon_data<- gon_data[gon_data$percent_0<=thresh,]
gon_data$percent_0<-NULL

```

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(gon_data)`...

```{r GON continuous variable setup, echo=TRUE}

dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)

write.csv(counts(dd, normalized=TRUE), "../DE_results/data_GON_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = gon_behav)
biplot(p, lab=gon_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T in GON")



eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

A=adjacency(t(datExpr0),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)
#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, gon_behav$Year)
p <- pca(mat, metadata = gon_behav)
biplot(p, lab=gon_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="PCA Gonads removeBatchEffect(Year)")
eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))
```

<br>

### Analysis

DESeq model specification ~ Year +  mean_T

This chunk also includes a function I wrote to remove highly influential observations (as measured by Cook's Distance) from the continuous data. I know from previous iterations of this analysis that there are observations that are highly influential and also [DESeq2 does not exclude these from the results in continuous data](https://support.bioconductor.org/p/126040/).

```{r GON mean T DEseq, fig.height=4, fig.width=10, echo=TRUE}
varname="mean_T"
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Year + mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)
summary(res)

#DEseq2 doesn't do outlier removal for continuous variables. I'm doing that here with a bespoke function
res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$pvalue),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_GON_mean_T.csv", row.names=TRUE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r GON mean_T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```



```{r GON mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("mean_T","Class"),6, status_cols)
g
```

```{r hub plot, eval=FALSE, echo=FALSE}

hub<-"AVPR1A"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$mean_T, dat$count)^2


p<- ggplot(dat, aes(x=mean_T, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", padj=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_GON_AVPR1A.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")
```

<br>

### Gene Ontology


```{r GON mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GON mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename="../DE_results/supp_figure_GON_mean_T.png", plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")

```

<br>

## Status x mean T GON

For pulling out candidate genes with interactions

```{r GON mean T x Status, echo=TRUE, , fig.height=4, fig.width=10, echo=TRUE}
rm(list= ls()[!(ls() %in% c(keep,"gon_data","gon_behav","tissue"))])
varname="Status x mean_T"
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~Year + Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]


#summary(res)
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/interactions_GON_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

How sensitive is the candidate gene results to model form?

```{r}
#### batch LRT ####
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(gon_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in GON",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength GON

### Analysis

formula ~ Year + scale(Strength)

Without scaling strength, DESeq2 would throw a warning, recommending the variable be scaled.

```{r GON Strength, fig.height=4, fig.width=10, echo=TRUE}
rm(list= ls()[!(ls() %in% c(keep,"gon_data","gon_behav","tissue","dd"))])
varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Year + strength.all_study)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
summary(res)
res<- res[order(res$padj),]

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"


write.csv(out_res, file="../DE_results/results_GON_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r GON Strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r GON strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```


<br>

### Gene Ontology


```{r GON strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GON strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))


write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)



#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")

```


<br><br>

# Pituitary (PIT)


```{r PIT data set up}
rm(list= ls()[!(ls() %in% keep)])

tissue="PIT"
#pit_key<- subset(key, Tissue=="PIT")
#pit_key<- droplevels(pit_key)
pit_behav<- subset(key_behav, Tissue=="PIT")
pit_behav<- droplevels(pit_behav)


#Filtering
pit_data<- data[,colnames(data) %in% rownames(pit_behav)]
start<- nrow(pit_data)
#remove genes with less than 5 reads
pit_data$avg_count<- apply(pit_data, 1, mean)
pit_data<- pit_data[pit_data$avg_count>5,]
pit_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pit_data$percent_0<- apply(pit_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pit_data)/2
pit_data<- pit_data[pit_data$percent_0<=thresh,]
pit_data$percent_0<-NULL
```

## Checking the sampling

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(pit_data)`

Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pit samples for `r nrow(pit_behav)`, with respect to T and the tracking data.

```{r PIT sampling, fig.height=5, fig.width=10, fig.align='center'}

fsb<- fisher.test(table(pit_behav$Status, pit_behav$Batch))
knitr::kable(table(pit_behav$Status, pit_behav$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value)) %>% kable_styling()

fsyb<- fisher.test(table(pit_behav$Batch, pit_behav$Year))
knitr::kable(table(pit_behav$Batch, pit_behav$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", round(fsyb$p.value,2))) %>% kable_styling()


mean_t_wtest<- wilcox.test(pit_behav$mean_T ~ pit_behav$Batch, alternative="two.sided")
a<- ggplot(pit_behav, aes(x=Batch, y=mean_T)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(pit_behav$strength.all_study ~ pit_behav$Batch, alternative="two.sided")
d<- ggplot(pit_behav, aes(x=Batch, y=strength.all_study)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme
grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE)

```

<br>

### Expression data exploration

```{r PIT DEseq1}

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
##vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_PIT_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_PIT_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = pit_behav)
biplot(p, lab=pit_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PIT PCA not corrected for batch")


eigencorplot(p, metavars = c("time_kill","Year","Batch","Status", "mean_T","strength.all_study"))

```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, pit_behav$Batch)
p <- pca(mat, metadata = pit_behav)
biplot(p, lab=pit_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="PCA PIT removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("time_kill","Batch","Year","Status", "mean_T","strength.all_study"))




wgcnadata<- as.data.frame(t(mat))


traits<- pit_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

Looks controlling for Batch also removes the association with Year. Let's just include Batch.

<br>

## Status PIT

### Analysis

Formula = ~ Batch + Status


In this section, I find results with different p-value distributions. I was following [THIS](https://www.huber.embl.de/users/klaus/Teaching/DESeq2-Analysis.pdf) tutorial by Bernd Klaus and it recommended using `fdrtool` to correct these p-values.

```{r PIT Status, fig.height=8, fig.width=10, echo=TRUE}
varname="Status"
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_PIT_status.csv", row.names=FALSE)

par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected PIT Status")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()
```

```{r PIT status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PIT status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

```{r, eval=FALSE}
hub<-"GH1"
dat<- plotCounts(dd, gene=row.names(res1[hub,]), intgroup=c("Class","Status"), returnData=TRUE)
p<- ggplot(dat, aes(x=Status, y=count, color=Status)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="social status", y="Normalized Counts", title=hub, subtitle=paste0("LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],1), ", p=",signif(res1$pvalue[rownames(res1)==hub],1), ", q=", signif(res1$padj[rownames(res1)==hub],1))) + geom_boxplot(fill=NA, outlier.shape = NA, lwd=0.3) + theme(legend.position = "none", aspect.ratio = 1)

ggsave("../DE_results/figure_PIT_GH1_status.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")

```

### Gene Ontology

<br>

```{r PIT status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PIT status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)

  
#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=205, units="mm", bg="white")
```


## Testosterone PIT

### Analysis

```{r PIT mean T, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pit_data","pit_behav","dd", "tissue"))])
varname="mean_T"
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_PIT_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```

```{r PIT mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PIT mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r PIT mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PIT mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>


## mean T x Status


```{r PIT mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Batch + Status + mean_T + Status:mean_T)
dd<- DESeq(dd, test="LRT",reduced=~ Batch + Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/interactions_PIT_mean_T_Status.csv", row.names=FALSE)




par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```


```{r}
#### batch LRT ####
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$fdrtool<- "no"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)
int_red_obs_lrt$fdrtool<- "no"

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)
int_min_obs_lrt$fdrtool<- "no"

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
#candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(pit_behav)

write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in PIT",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength PIT

### Analysis
Formula: ~ Batch + scale(Strength)


```{r PIT Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pit_data","pit_behav","dd","tissue"))])
varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_PIT_Strength_withPFT3.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r PIT strength geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- cor(dat$strength.all_study, dat$count)^2
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class, label=Harvest_ID)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," LFC=",signif(res$log2FoldChange[i],2) ," padj=", formatC(res$padj[i],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + geom_text(nudge_x=-0.08)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by social network strength in PIT",gp=gpar(fontsize=14)), bottom=textGrob("social network strength",gp=gpar(fontsize=14)))
g


cooks<- as.data.frame(assays(dd)[["cooks"]])
 p=ncol(model.matrix(design(dd), colData(dd)))
  n=nrow(colData(dd))
 
par(mar=c(8,5,2,2))
boxplot(log10(cooks), range=0, las=2, main="Cook's Distances for Strength across ALL genes")
abline(h=log10(qf(0.98,p,n-p)))
```

```{r pft3hub, eval=FALSE}
hub<-"PRL"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2


p<- ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", padj=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_PIT_PRL_str_noerrorPFT3.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")
```


As you can see above, PFT3 has consistently high leverage on the social network analysis. Even if I play around with the outlier theshold this I am always getting non-linear plots like above. So, let's see what happens when I remove this sample. 

<br>

### Re-analysis - outlier removed

```{r PIT Strength2, fig.height=4, fig.width=10}

outlier<- "PFT3_PIT_run2"

pit_data<- pit_data[,!colnames(pit_data) %in% outlier]
pit_behav<- pit_behav[!rownames(pit_behav) %in% outlier,]

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status)


dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_PIT_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r PIT strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PIT strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

```{r, eval=FALSE}
hub<-"PRL"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2


p<- ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", padj=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_PIT_PRL_str_noerror.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")
```

```{r plotting PIT, eval=FALSE}
PIT_modules<- read.csv("../WGCNA/ByTissue_2021/PIT_results_ModuleMembership.csv")


topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


i="PIT"
 tissue<- strength_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>2, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""    
tissue<- merge(tissue, PIT_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","pink4","lightgreen","plum1","violet"          ,"lightcyan1","darkolivegreen4","plum2","floralwhite","coral1","blue","brown4","darkgreen","darkolivegreen2","firebrick3","blue2"))

modulecols<- c("grey30","grey60","pink4","lightgreen","plum1","violet"          ,"lightcyan1","darkolivegreen4","plum2","floralwhite","coral1","blue","brown4","darkgreen","darkolivegreen2","firebrick3","blue2")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")



#ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=0.0)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(limits=c(-5,5.5), breaks=c(-5,-3,0,3,5))

```


<br>

### Gene Ontology

```{r PIT strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PIT strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Ventromedial Hypothalamus (VMH)


```{r VMH data set up}
rm(list= ls()[!(ls() %in% keep)])
tissue="VMH"
vmh_key<- subset(key_behav, Tissue==tissue)

vmh_key<- droplevels(vmh_key)

vmh_behav<- vmh_key[vmh_key$Batch!="pilot",]

#Filtering
vmh_data<- data[,colnames(data) %in% rownames(vmh_key)]
start<- nrow(vmh_data)
#remove genes with less than 5 reads
vmh_data$avg_count<- apply(vmh_data, 1, mean)
vmh_data<- vmh_data[vmh_data$avg_count>5,]
vmh_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
vmh_data$percent_0<- apply(vmh_data, 1, function(x)length(x[x==0]))
thresh<- ncol(vmh_data)/2
vmh_data<- vmh_data[vmh_data$percent_0<=thresh,]
vmh_data$percent_0<-NULL
```

## Checking the sampling

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(vmh_data)`

Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have vmh samples for `r nrow(vmh_key)`, with respect to T and the tracking data.

```{r VMH sampling, fig.height=5, fig.width=10, fig.align='center'}

fsb<- fisher.test(table(vmh_key$Status, vmh_key$Batch))
knitr::kable(table(vmh_key$Status, vmh_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()


fsyb<- fisher.test(table(vmh_key$Batch, vmh_key$Year))
knitr::kable(table(vmh_key$Batch, vmh_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", signif(fsyb$p.value,2))) %>% kable_styling()


mean_t_wtest<- wilcox.test(vmh_behav$mean_T ~ vmh_behav$Batch, alternative="two.sided")
a<- ggplot(vmh_behav, aes(x=Batch, y=mean_T))  + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,2)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(vmh_behav$strength.all_study ~ vmh_behav$Year, alternative="two.sided")
d<- ggplot(vmh_behav, aes(x=Batch, y=strength.all_study)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme
grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables VMH",gp=gpar(fontsize=14)))

```


The majority of our data in VMH is in run 1.

### Expression data exploration

```{r VMH DEseq1}

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = vmh_key)
biplot(p, lab=vmh_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in VMH")


eigencorplot(p, metavars = c("time_kill","Year","Batch","Status"))


wgcnadata<- as.data.frame(t(vsd_data))


traits<- vmh_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, vmh_key$Batch)
p <- pca(mat, metadata = vmh_key)
#biplot(p, lab=vmh_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Year","Status"))


```

Looks like controlling for Batch is sufficient here, as Batch is highly correlated with Year.

## Status VMH

### Analysis

Formula = ~ Batch + Status

```{r VMH Status, fig.height=4, fig.width=10}
 varname="Status"
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_VMH_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r VMH status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r VMH status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r VMH status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r VMH status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=190, width=210, units="mm", bg="white")
```
<br>

## Testosterone VMH

<br>

### Expression data exploration

```{r VMH DESeq continuous variable subset}
rm(list= ls()[!(ls() %in% c(keep,"vmh_data","vmh_behav","tissue"))])

vmh_behav<- subset(key_behav, Tissue==tissue)
vmh_behav<- subset(vmh_behav, Batch!="pilot")
vmh_behav<- droplevels(vmh_behav)
vmh_data<- data[,colnames(data) %in% rownames(vmh_behav)]

#Filtering
#remove genes with less than 5 reads
vmh_data$avg_count<- apply(vmh_data, 1, mean)
vmh_data<- vmh_data[vmh_data$avg_count>5,]
vmh_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
vmh_data$percent_0<- apply(vmh_data, 1, function(x)length(x[x==0]))
thresh<- ncol(vmh_data)/2
vmh_data<- vmh_data[vmh_data$percent_0<=thresh,]
vmh_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_VMH_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_VMH_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = vmh_behav)
biplot(p, lab=vmh_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="mean T")


eigencorplot(p, metavars = c("time_kill","Year","Batch","Status", "mean_T","strength.all_study"))
```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, vmh_behav$Batch)
p <- pca(mat, metadata = vmh_behav)
biplot(p, lab=vmh_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA VMH removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("time_kill","Year","Batch","Status", "mean_T","strength.all_study"))


wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```


Batch does not account for year here... so have to include both... 

<br>

### Analysis

Formula: ~ Batch + Year + mean_T

```{r VMH mean T, fig.height=8, fig.width=10}
varname="mean_T"
design(dd)<- formula(~Batch + Year + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"
write.csv(out_res, file="../DE_results/results_VMH_mean_T.csv", row.names=TRUE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```

```{r VMH mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r VMH mean T  geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```


<br>

### Gene Ontology

```{r VMH mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r VMH mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## Status x meanT


```{r VMH mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ Batch + Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~Batch + Year + Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/interactions_VMH_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="VMH Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r}
#### batch LRT ####
#testing the effect of both batch AND year
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]

candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(vmh_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```


```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


#hub<- "GAL"
#out<-  plotCounts(dd, gene=row.names(res1[hub,]), intgroup=c("Status","mean_T", "Year","Class"), #returnData=TRUE)
#ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + #labs(title=paste0(row.names(res1[hub,])," #LFC=",signif(interaction_res$log2FoldChange[interaction_res$gene==hub],2) ," padj=", #signif(interaction_res$padj[interaction_res$gene==hub],2)),x="", y="Normalised Counts") + peri_theme + #scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none") + facet_wrap(~ #Status)



g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in VMH",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength VMH

### Analysis

Formula ~ Batch + Year + scale(Strength)

```{r vmh Strength, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"vmh_data","vmh_behav","dd","tissue"))])
varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + Year + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_VMH_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")

dev.off()
```


```{r VMH strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r VMH strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r VMH strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r VMH strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Anterior Hypothalamus (AH)

<br>

```{r AH data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="AH"
#ah_key<- subset(key, Tissue=="AH")
ah_behav<- subset(key_behav, Tissue==tissue)
ah_behav<- droplevels(ah_behav)

#Filtering
ah_data<- data[,colnames(data) %in% rownames(ah_behav)]
start<- nrow(ah_data)
#remove genes with less than 5 reads
ah_data$avg_count<- apply(ah_data, 1, mean)
ah_data<- ah_data[ah_data$avg_count>5,]
ah_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ah_data$percent_0<- apply(ah_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ah_data)/2
ah_data<- ah_data[ah_data$percent_0<=thresh,]
ah_data$percent_0<-NULL

```
## Checking the sampling

For the AH, we started with `r start` genes but after filering we have `r nrow(ah_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ah samples for `r nrow(ah_behav)`, with respect to T and the tracking data.


```{r AH sampling, fig.align='center'}

#fsb<- fisher.test(table(ah_behav$Status, ah_behav$Batch))
knitr::kable(table(ah_behav$Status, ah_behav$Batch), caption=paste0("Status across sequencing runs")) %>% kable_styling()



#fsyb<- fisher.test(table(ah_behav$Batch, ah_behav$Year))
knitr::kable(table(ah_behav$Batch, ah_behav$Year), caption=paste0("Sampling years distributed across batches")) %>% kable_styling()

```

<br>

### Expression data exploration

```{r AH DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_AH_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_AH_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ah_behav)

biplot(p, lab=ah_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in AH")

eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ah_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")


```

<br>



PFT1 and PFT11 cluster with the BSTm as previously identified by the brain tissue PCA plot and the all tissue WGCNA dendrogram. We know that there were issues with the sectioning plane in individual PFT1 that could affect this tissue, but not PFT11. 

The effects of removing these samples will be explored for each analysis vs removing a randomly selected pair of samples. These results (not shown) showed similar effects, so these samples were not excluded.

<br>

## Status AH

### Analysis

Formula = ~ Status

```{r AH Status, fig.height=4, fig.width=10}
varname="Status"
design(dd)<- formula(~ Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]




des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_AH_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r AH status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AH status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```


<br>

### Gene Ontology

```{r AH Status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AH status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## Testosterone AH

### Analysis

Formula ~ mean_T

```{r AH mean T, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"ah_data","ah_behav","tissue","dd"))])
varname="mean_T"
dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_AH_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```


```{r AH mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AH mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```
<br>

### Gene Ontology

```{r AH mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AH mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## Status x mean T


```{r AH mean T x Status, echo=FALSE, fig.height=8, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~ Status+mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/interactions_AH_mean_T_Status.csv", row.names=FALSE)




par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```

```{r}

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- results(int_min_obs_lrt, alpha=0.1)
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
res1<- res[!is.na(int_min_obs_lrt$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- int_min_obs_lrt[is.na(int_min_obs_lrt$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
int_min_obs_lrt<- as.data.frame(res1)
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)
int_min_obs_lrt$fdrtool<- "yes"

out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(ah_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```


```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in AH",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



<br>

## Strength AH

### Analysis

Formula ~ scale(Strength)

```{r AH Strength, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"ah_data","ah_behav","tissue","dd"))])
varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~ strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_AH_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")

dev.off()
```


```{r AH strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AH strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

```{r, eval=FALSE}
hub<- "TSHR"
out<-  plotCounts(dd, gene=hub, intgroup=c("strength.all_study", "Class"), returnData=TRUE)
Rsq<- cor(out$strength.all_study, out$count)^2
ggplot(out, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=4) + labs(title=paste0(row.names(res[hub,])," LFC=",signif(strength_res$log2FoldChange[strength_res$gene==hub],2) ," padj=", formatC(strength_res$padj[strength_res$gene==hub],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none")
```

<br>

### Gene Ontology

```{r AH strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AH strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br><br>

# Paraventricular Nucleus (PVN)

<br>

```{r PVN data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="PVN"
#pvn_key<- subset(key, Tissue=="PVN")
pvn_behav<- subset(key_behav, Tissue==tissue)
pvn_behav<- droplevels(pvn_behav)

#Filtering
pvn_data<- data[,colnames(data) %in% rownames(pvn_behav)]
start<- nrow(pvn_data)
#remove genes with less than 5 reads
pvn_data$avg_count<- apply(pvn_data, 1, mean)
pvn_data<- pvn_data[pvn_data$avg_count>5,]
pvn_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pvn_data$percent_0<- apply(pvn_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pvn_data)/2
pvn_data<- pvn_data[pvn_data$percent_0<=thresh,]
pvn_data$percent_0<-NULL

```

## Checking the sampling

For the PVN, we started with `r start` genes but after filering we have `r nrow(pvn_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pvn samples for `r nrow(pvn_behav)`, with respect to T and the tracking data.

```{r PVN sampling, fig.align='center'}

#fsb<- fisher.test(table(pvn_behav$Status, pvn_behav$Batch))
knitr::kable(table(pvn_behav$Status, pvn_behav$Batch), caption=paste0("Status across sequencing runs")) %>% kable_styling()
```

### Expression data exploration

```{r PVN DEseq1}
dd<- DESeqDataSetFromMatrix(countData=pvn_data, colData=pvn_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_PVN_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_PVN_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = pvn_behav)

biplot(p, lab=pvn_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")


eigencorplot(p, metavars = c("time_kill", "Year","Status", "mean_T","strength.all_study"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- pvn_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Looks like we don't have any outliers, nor do we need to include Year (or Batch) as a covariate.

<br>

## Status PVN

### Analysis

Formula = ~ Status

```{r PVN Status, fig.height=8, fig.width=10}
#design(dd)<- formula(~ Status)
#dd<- DESeq(dd)
varname="Status"
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_PVN_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()
```

```{r PVN status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PVN status T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

```{r plotting PVN, eval=FALSE}
PVN_modules<- read.csv("../WGCNA/ByTissue_2021/PVN_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B")


 tissue<- status_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  topgenes<- 1:6
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")


tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, PVN_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]
tissue$color<-factor(tissue$color,levels=c("grey60", "grey30","midnightblue","turquoise",      "darkolivegreen","blue","darkred","greenyellow","black","darkmagenta","white","saddlebrown","green"))
modulecols<- c("grey30", "grey60","midnightblue","turquoise",      "darkolivegreen","blue","darkred","greenyellow","black","darkmagenta","white","saddlebrown","green")


tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


#ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none")

```
```{r, eval=FALSE}
hub="ESR2"
dat<- plotCounts(dd, gene=row.names(res[hub,]), intgroup=c("Class","Status"), returnData=TRUE)
p<- ggplot(dat, aes(x=Status, y=count, color=Status)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="social status", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],1), ", p=",signif(res1$pvalue[rownames(res1)==hub],1), ", q=", signif(res1$padj[rownames(res1)==hub],1))) + geom_boxplot(fill=NA, outlier.shape = NA, lwd=0.3) + theme(legend.position = "none", aspect.ratio = 1)
p
ggsave("../DE_results/figure_PVN_ESR2_status.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")
```

<br>

### Gene Ontology

```{r PVN status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PVN status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## Testosterone PVN

### Analysis

Formula = ~ mean_T

```{r PVN mean T, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pvn_data","pvn_behav","dd","tissue"))])
varname="mean_T"
design(dd)<- formula(~ mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_PVN_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()

```

```{r PVN mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PVN mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r PVN mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PVN mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## Status x mean T


```{r PVN mean T x Status, echo=FALSE, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=pvn_data, colData=pvn_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd, test="LRT",reduced=~ Status + mean_T
           )
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$fdrtool<- "no"
write.csv(out_res, file="../DE_results/interactions_PVN_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="PVN Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r}
#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=pvn_data, colData=pvn_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)
int_min_obs_lrt$fdrtool<- "no"

out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(pvn_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in PVN",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength PVN

### Analysis

Formula ~ scale(Strength)

```{r PVN Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pvn_data","pvn_behav","dd","tissue"))])
varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_PVN_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r PVN strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r PVN strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```


```{r, eval=FALSE}
hub<-"LOC113983511"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
p<- ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_LS_OXT_str_noerror.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")

```

<br>

### Gene Ontology

```{r PVN strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r PVN strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Medial Preoptic Area (POM)



```{r POM data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="POM"
pom_key<- subset(key_behav, Tissue==tissue)
pom_key<- droplevels(pom_key)

pom_behav<- pom_key[pom_key$Batch!="pilot",]

#Filtering
pom_data<- data[,colnames(data) %in% rownames(pom_key)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL

```
## Checking the sampling

For the POM, we started with `r start` genes but after filering we have `r nrow(pom_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pom samples for `r nrow(pom_behav)`, with respect to T and the tracking data.

```{r POM sampling, fig.align='center'}

fsb<- fisher.test(table(pom_key$Status, pom_key$Batch))
knitr::kable(table(pom_key$Status, pom_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value)) %>% kable_styling()


fsyb<- fisher.test(table(pom_key$Batch, pom_key$Year))
knitr::kable(table(pom_key$Batch, pom_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", signif(fsyb$p.value,2))) %>% kable_styling()
```

<br>

### Expression data exploration

```{r POM DEseq1}
dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)



p <- pca(vsd_data, metadata = pom_key)

biplot(p, lab=pom_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in POM")

eigencorplot(p, metavars = c("Year","Status", "Batch"))
```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, pom_key$Batch)
p <- pca(mat, metadata = pom_key)
biplot(p, lab=pom_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- pom_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

Don't need to control for year because accounting for Batch seems to fix that.

<br>

### Outlier removed

PFT2 was previously identified as a potential outlier, and it's coming out quite separately here, even after all the corrections for batch effects. So, this one I am going to REMOVE. 

```{r remove PFT2_POM}

outliers=c("PFT2_POM_run1")

pom_key<- pom_key[!rownames(pom_key) %in% outliers,]
pom_data<- data[,colnames(data) %in% rownames(pom_key)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL


```

<br>

##Status POM

###Analysis

Formula = ~ Batch + Status

```{r POM Status, fig.height=4, fig.width=10}
varname="Status"

dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_key, design= ~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"


write.csv(out_res, file="../DE_results/results_POM_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r POM status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r POM status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r POM status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r POM status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## Testosterone POM

### Expression data exploration
```{r POM continuous data set up}

rm(list= ls()[!(ls() %in% c(keep,"pom_data","pom_behav","dd","tissue","outliers"))])

#Filtering
pom_behav<- subset(key_behav, Tissue=="POM")
pom_behav<- subset(pom_behav, Batch!="pilot")
pom_behav<- droplevels(pom_behav)
pom_behav<- pom_behav[!rownames(pom_behav) %in% outliers,]
pom_data<- data[,colnames(data) %in% rownames(pom_behav)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_POM_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = pom_behav)
biplot(p, lab=pom_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")



eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))
```


<br>

### Analysis
Formula = ~ mean_T

```{r POM mean T, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pom_data","pom_behav","dd","tissue"))])
varname="mean_T"
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"


write.csv(out_res, file="../DE_results/results_POM_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```

```{r POM mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r POM mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```



```{r, eval=FALSE}
hub<-"AR"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$mean_T, dat$count)^2

p<- ggplot(dat, aes(x=mean_T, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="mean testosterone", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", padj=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19) + scale_y_continuous(breaks=c(500,1000,1500), limits=c(200,1500))
p
ggsave("../DE_results/figure_POM_AR_meanT_noerror.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")


```

```{r plotting POM, eval=FALSE}
POM_modules<- read.csv("../WGCNA/ByTissue_2021/POM_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B")

topgenes<- 1:6
 tissue<- mean_T_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, POM_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]
tissue$color<-factor(tissue$color,levels=c("grey60", "grey30","floralwhite","lightpink4","black","brown","blue4","lightsteelblue","bisque4","purple","mediumpurple4","coral3","lightcoral","navajowhite1"))
modulecols<- c("grey30", "grey60","floralwhite","lightpink4","black","brown","blue4","lightsteelblue","bisque4","purple","mediumpurple4","coral3","lightcoral","navajowhite1")


tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


#ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title="POM",subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-3,-2,-1,0,1,2,3), limits=c(-3,3))

```


<br>

### Gene Ontology


```{r POM mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r POM mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## mean T x Status


```{r POM mean T x Status, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd, test="LRT",reduced=~ Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/interactions_POM_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="POM Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```
```{r}
#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)
int_min_obs_lrt$fdrtool<- "no"

out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(pom_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in POM",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength POM

### Analysis

Formula ~ scale(Strength)

```{r POM Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"pom_data","pom_behav","dd","tissue"))])
varname="Strength"

dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"



write.csv(out_res, file="../DE_results/results_POM_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r POM strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r POM strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```
<br>

### Gene Ontology

```{r POM strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r POM strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Dorsomedial Intercollicular nucleus (ICo)

<br>

```{r ICO data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="ICO"


ico_key<- subset(key_behav, Tissue==tissue)
ico_key<- droplevels(ico_key)
ico_behav<- ico_key[ico_key$Batch!="pilot",]
#Filtering
ico_data<- data[,colnames(data) %in% rownames(ico_key)]
start<- nrow(ico_data)
#remove genes with less than 5 reads
ico_data$avg_count<- apply(ico_data, 1, mean)
ico_data<- ico_data[ico_data$avg_count>5,]
ico_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression



ico_data$percent_0<- apply(ico_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ico_data)/2
ico_data<- ico_data[ico_data$percent_0<=thresh,]
ico_data$percent_0<-NULL

#for some reason the the above code stalls when I try to knit the document so I am #just going to manually remove the sites


#low_genes<- c("LOC113990771","LOC114004308") 
#ico_data<- ico_data[!rownames(ico_data) %in% low_genes,]

```


## Checking the sampling

For the ICO, we started with `r start` genes but after filering we have `r nrow(ico_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ico samples for `r nrow(ico_behav)`, with respect to T and the tracking data.

```{r ICO sampling, fig.height=5, fig.width=10}

fsb<- fisher.test(table(ico_key$Status, ico_key$Batch))
knitr::kable(table(ico_key$Status, ico_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value, 2))) %>% kable_styling()


fsyb<- fisher.test(table(ico_key$Batch, ico_key$Year))
knitr::kable(table(ico_key$Batch, ico_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", signif(fsyb$p.value,2))) %>% kable_styling()


mean_t_wtest<- wilcox.test(ico_behav$mean_T ~ ico_behav$Batch, alternative="two.sided")
a<- ggplot(ico_behav, aes(x=Batch, y=mean_T))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme


strength_wtest<- wilcox.test(ico_behav$strength.all_study ~ ico_behav$Batch, alternative="two.sided")
d<- ggplot(ico_behav, aes(x=Batch, y=strength.all_study))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables ICO",gp=gpar(fontsize=14)))

```

### Expression data exploration

```{r ICO DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)


p <- pca(vsd_data, metadata = ico_key)

biplot(p, lab=ico_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in ICO")

eigencorplot(p, metavars = c("time_kill","Year","Status", "Batch"))


```

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, ico_key$Batch)
p <- pca(mat, metadata = ico_key)
biplot(p, lab=ico_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ico_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

<br>

## Status ICO

### Analysis
Formula = ~ Batch + Status

```{r ICO Status, fig.height=4, fig.width=10}
varname="Status"
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_ICO_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r ICO status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r ICO status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r ICO status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r ICO status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## Testosterone ICO

### Expression data exploration

```{r ICO continuous data set up}
rm(list= ls()[!(ls() %in% c(keep,"ico_data","ico_behav","dd","tissue"))])


#Filtering
ico_behav<- subset(key_behav, Tissue==tissue)
ico_behav<- subset(ico_behav, Batch!="pilot")
ico_behav<- droplevels(ico_behav)
ico_data<- data[,colnames(data) %in% rownames(ico_behav)]
start<- nrow(ico_data)
#remove genes with less than 5 reads
ico_data$avg_count<- apply(ico_data, 1, mean)
ico_data<- ico_data[ico_data$avg_count>5,]
ico_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ico_data$percent_0<- apply(ico_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ico_data)/2
ico_data<- ico_data[ico_data$percent_0<=thresh,]
ico_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)

write.csv(counts(dd, normalized=TRUE), "../DE_results/data_ICO_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = ico_behav)

biplot(p, lab=ico_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("time_kill","Year","Status", "Batch", "mean_T","strength.all_study"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, ico_behav$Batch)
p <- pca(mat, metadata = ico_behav)
biplot(p, lab=ico_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))


```

Still need to control for Year here...

<br>

### Analysis


Formula = ~ Batch + Year + mean_T

```{r ICO mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~Batch + Year + mean_T)
dd<- DESeq(dd)
varname="mean_T"
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_ICO_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```


```{r ICO mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r ICO mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

```{r, eval=FALSE}
hub<- "MMP2"
out<-  plotCounts(dd, gene=hub, intgroup=c("mean_T", "Class"), returnData=TRUE)
Rsq<- cor(out$mean_T, out$count)^2
i=3
ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + labs(title=paste0(row.names(res1[hub,])," LFC=",signif(mean_T_res$log2FoldChange[mean_T_res$gene==hub],2) ," padj=", formatC(mean_T_res$padj[mean_T_res$gene==hub],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none")
```


<br>

### Gene Ontology

```{r ICO mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r ICO mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## mean T x Status


```{r ICO mean T x Status, echo=FALSE, fig.height=8, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ Batch + Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd, test="LRT",reduced= ~ Batch + Year + Status + mean_T )
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/interactions_ICO_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="ICO mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```

```{r}
#### batch LRT ####
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(ico_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in ICO",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength ICO

### Analysis
Formula ~ Batch + Year + scale(Strength)

```{r ICO Strength, fig.height=8, fig.width=10}

rm(list= ls()[!(ls() %in% c(keep,"ico_data","ico_behav","dd","tissue"))])

varname="Strength"
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + Year + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"
write.csv(out_res, file="../DE_results/results_ICO_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```


```{r ICO strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r ICO strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

```{r, eval=FALSE}

hub<-"VIPR2"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
ggplot(dat, aes(x=strength.all_study, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="strength", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],2), ", p=",signif(res1$pvalue[rownames(res1)==hub],2), ", padj=", signif(res1$padj[rownames(res1)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")


```

<br>

### Gene Ontology

```{r ICO strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r ICO strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br><br>

# Midbrain Central Grey (GCt)

<br>

```{r GCT data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="GCT"
gct_key<- subset(key_behav, Tissue==tissue)
gct_key<- droplevels(gct_key)


gct_behav<- gct_key[gct_key$Batch!="pilot",]

#Filtering
gct_data<- data[,colnames(data) %in% rownames(gct_key)]
start<- nrow(gct_data)
#remove genes with less than 5 reads
gct_data$avg_count<- apply(gct_data, 1, mean)
gct_data<- gct_data[gct_data$avg_count>5,]
gct_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gct_data$percent_0<- apply(gct_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gct_data)/2
gct_data<- gct_data[gct_data$percent_0<=thresh,]
gct_data$percent_0<-NULL


```
## Checking the sampling

For the GCT, we started with `r start` genes but after filering we have `r nrow(gct_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gct samples for `r nrow(gct_behav)`, with respect to T and the tracking data.

```{r GCT sampling, fig.align='center', fig.height=5, fig.width=10}

fsb<- fisher.test(table(gct_key$Status, gct_key$Batch))
knitr::kable(table(gct_key$Status, gct_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()

fsyb<- fisher.test(table(gct_key$Batch, gct_key$Year))
knitr::kable(table(gct_key$Batch, gct_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", signif(fsyb$p.value,2))) %>% kable_styling()


mean_t_wtest<- wilcox.test(gct_behav$mean_T ~ gct_behav$Batch, alternative="two.sided")
a<- ggplot(gct_behav, aes(x=Batch, y=mean_T))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme




strength_wtest<- wilcox.test(gct_behav$strength.all_study ~ gct_behav$Batch, alternative="two.sided")
d<- ggplot(gct_behav, aes(x=Batch, y=strength.all_study))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables GCT",gp=gpar(fontsize=14)))

```

given there is only 1 GCT sample from the pilot, I will just exclude the pilot data.

<br>

### Expression data exploration

```{r GCT DEseq1}
dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)


p <- pca(vsd_data, metadata = gct_key)
biplot(p, lab=gct_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status")
#biplot(p, lab=gct_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "Batch"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK
```

<br>

### Corrections applied 

```{r}
mat<- limma::removeBatchEffect(vsd_data, gct_key$Batch)
p <- pca(mat, metadata = gct_key)
biplot(p, lab=gct_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- gct_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```



<br>

## Status GCT

### Analysis

Formula = ~ Batch + Status

```{r GCT Status, fig.height=8, fig.width=10}
varname="Status"

design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]

summary(res1)




des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"
write.csv(out_res, file="../DE_results/results_GCT_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()
```

```{r GCT status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r GCT status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```
<br>

### Gene Ontology


```{r GCT status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GCT status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```



<br>

## Testosterone GCT

### Expression data exploration

```{r GCT Continuous data setup}

rm(list= ls()[!(ls() %in% c(keep,"gct_data","gct_behav","dd","tissue"))])


#Filtering
gct_behav<- subset(key_behav, Tissue==tissue)
gct_behav<- subset(gct_behav, Batch!="pilot")
gct_behav<- droplevels(gct_behav)

gct_data<- data[,colnames(data) %in% rownames(gct_behav)]
start<- nrow(gct_data)
#remove genes with less than 5 reads
gct_data$avg_count<- apply(gct_data, 1, mean)
gct_data<- gct_data[gct_data$avg_count>5,]
gct_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gct_data$percent_0<- apply(gct_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gct_data)/2
gct_data<- gct_data[gct_data$percent_0<=thresh,]
gct_data$percent_0<-NULL



dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_GCT_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = gct_behav)

biplot(p, lab=gct_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("time_kill","Year","Status", "Batch", "mean_T","strength.all_study"))



datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

sampleTree = hclust(dist(datExpr0), method = "average")

par(mar = c(0, 4, 2, 0))
plot(sampleTree, main = "Sample clustering on all genes in GCT", xlab="", sub="", cex = 1)
dev.off()



```

<br>

### Analysis

Formula = ~ Batch + mean_T

```{r GCT mean T, fig.height=8, fig.width=10}
varname="mean_T"
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)
#resLRT<- DESeq(dd, test="LRT", reduced=~Batch)
#res<- results(resLRT, alpha=0.1)
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"
write.csv(out_res, file="../DE_results/results_GCT_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```

```{r GCT mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r GCT geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology


```{r GCT mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GCT mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## mean T x Status


```{r GCT mean T x Status, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ Batch + Status +  mean_T + Status:mean_T)
dd<- DESeq(dd, test="LRT",reduced=~Batch + Status +  mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)



res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/interactions_GCT_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GCT mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r}
#### batch LRT ####
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(gct_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in GCT",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g

#geneid<-"NAAA"
#blah<- plotCounts(dd, gene=row.names(res[geneid,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
#ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point()
```


<br>

## Strength GCT

### Analysis

Formula ~ Batch + scale(Strength)

```{r GCT Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"gct_data","gct_behav","dd","tissue"))])
varname="Strength"

dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch +  strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"


write.csv(out_res, file="../DE_results/results_GCT_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r GCT strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r GCT strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```


<br>

### Gene Ontology


```{r GCT strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r GCT strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br><br>

# Nucleus Taenia (TnA)

<br>

```{r TNA data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="TNA"
tna_key<- subset(key_behav, Tissue==tissue)
tna_key<- droplevels(tna_key)

tna_behav<- tna_key[tna_key$Batch!="pilot",]

#Filtering
tna_data<- data[,colnames(data) %in% rownames(tna_key)]
start<- nrow(tna_data)
#remove genes with less than 5 reads
tna_data$avg_count<- apply(tna_data, 1, mean)
tna_data<- tna_data[tna_data$avg_count>5,]
tna_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
tna_data$percent_0<- apply(tna_data, 1, function(x)length(x[x==0]))
thresh<- ncol(tna_data)/2
tna_data<- tna_data[tna_data$percent_0<=thresh,]
tna_data$percent_0<-NULL

```
## Checking the sampling

For the TNA, we started with `r start` genes but after filering we have `r nrow(tna_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have tna samples for `r nrow(tna_behav)`, with respect to T and the tracking data.

```{r TNA sampling, fig.align='center', fig.width=10, fig.height=5}

fsb<- fisher.test(table(tna_key$Status, tna_key$Batch))
knitr::kable(table(tna_key$Status, tna_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()


fsyb<- fisher.test(table(tna_key$Batch, tna_key$Year))
knitr::kable(table(tna_key$Batch, tna_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",signif(fsyb$p.value,2))) %>% kable_styling()


mean_t_wtest<- wilcox.test(tna_behav$mean_T ~ tna_behav$Batch, alternative="two.sided")
a<- ggplot(tna_behav, aes(x=Batch, y=mean_T)) + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(tna_behav$strength.all_study ~ tna_behav$Batch, alternative="two.sided")
d<- ggplot(tna_behav, aes(x=Batch, y=strength.all_study)) + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables TNA",gp=gpar(fontsize=14)))

```

<br>

### Expression data exploration

```{r TNA DEseq1}
dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = tna_key)

biplot(p, lab=tna_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in TNA")

eigencorplot(p, metavars = c("Year","Status", "Batch"))




```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, tna_key$Batch)
p <- pca(mat, metadata = tna_key)
biplot(p, lab=tna_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- tna_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

PFT1 was previously identified as an outlier based on WGCNA dendrograms only. But here it's not consistently dentified as an outlier.Also previous analysis shows that it doesn't have a strong influence on the results, so I am going to keep it in this data. Looks like we only need to control for Batch, as it removes the effects of year on our data.

<br>

## Status TNA


### Analysis
Formula = ~ Batch + Status

```{r TNA Status, fig.height=4, fig.width=10}
varname="Status"
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)




des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_TNA_Status.csv", row.names=FALSE)




par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r TNA status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r TNA status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r TNA status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r TNA status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```



<br>

## Testosterone TNA

### Expression data exploration
```{r TNA continuous data set up}
rm(list= ls()[!(ls() %in% c(keep,"tna_data","tna_behav","dd","tissue"))])

#Filtering
tna_behav<- subset(key_behav, Tissue==tissue)
tna_behav<- subset(tna_behav, Batch!="pilot")
tna_behav<- droplevels(tna_behav)


tna_data<- data[,colnames(data) %in% rownames(tna_behav)]
start<- nrow(tna_data)
#remove genes with less than 5 reads
tna_data$avg_count<- apply(tna_data, 1, mean)
tna_data<- tna_data[tna_data$avg_count>5,]
tna_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
tna_data$percent_0<- apply(tna_data, 1, function(x)length(x[x==0]))
thresh<- ncol(tna_data)/2
tna_data<- tna_data[tna_data$percent_0<=thresh,]
tna_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_TNA_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_TNA_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = tna_behav)
#biplot(p, lab=tna_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")
biplot(p, lab=tna_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("time_kill","Year","Status", "Batch", "mean_T","strength.all_study"))

#plotloadings(p, labSize = 3, components = getComponents(p, c(1,2,3)), rangeRetain=0.1)
```

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, tna_behav$Batch)
p <- pca(mat, metadata = tna_behav)
biplot(p, lab=tna_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="PCA TNA removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Taking the batch effect into account seems to remove the association with Year too. Only need to include Batch in the model. 

### Analysis
Formula = ~ Batch + mean_T

```{r TNA mean T, fig.height=4, fig.width=10}
varname="mean_T"
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_TNA_mean_T.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r plotting TNA, eval=FALSE}
TNA_modules<- read.csv("../WGCNA/ByTissue_2021/TNA_results_ModuleMembership.csv")

topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


#i="TNA"
 tissue<- mean_T_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, TNA_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","coral1","lightyellow","blue","coral3","tan","black","blue2","honeydew","skyblue2","darkorange","skyblue4","antiquewhite4","grey","darkorange2","lightsteelblue"))
modulecols<- c("grey30","grey60","coral1","lightyellow","blue","coral3","tan","black","blue2","honeydew","skyblue2","darkorange","skyblue4","antiquewhite4","grey","darkorange2","lightsteelblue")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


  

ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title="TnA",subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-4,-3,-2,-1,0,1,2,3,4))
```


```{r TNA mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r TNA mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```


```{r, eval=FALSE}
hub<-"ESR2"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$mean_T, dat$count)^2
p<- ggplot(dat, aes(x=mean_T, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="mean testosterone", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_TNA_ESR2_meanT.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")
```

<br>

### Gene Ontology


```{r TNA mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r TNA mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## mean T x Status


```{r TNA mean T x Status, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Batch + Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~Batch + Status + mean_T)

#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$design<- des
out_res$fdrtool<- "no"
write.csv(out_res, file="../DE_results/interactions_TNA_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="TNA Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r}
#### batch LRT ####
batch_lrt<- DESeq(dd,test="LRT",reduced=~Status + mean_T + Status:mean_T)
batch_obs_lrt<-  as.data.frame(results(batch_lrt, alpha=0.1))
hist(batch_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
batch_obs_lrt$n_p005<- nrow(batch_obs_lrt[which(batch_obs_lrt$pvalue<0.05),])
batch_obs_lrt$n_q01<- nrow(batch_obs_lrt[which(batch_obs_lrt$padj<0.1),])
batch_obs_lrt<- batch_obs_lrt[rownames(batch_obs_lrt) %in% candidates2,]
batch_obs_lrt$Tissue<- tissue
batch_obs_lrt$Test<- "LRT_batch"
batch_obs_lrt$design<- paste0(as.character(design(dd)), collapse=" ")
batch_obs_lrt$gene<- rownames(batch_obs_lrt)

#reduced LRT
dd_red <- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Status + mean_T + Status:mean_T)
int_red_obs_lrt<- DESeq(dd_red,test="LRT",reduced=~Status + mean_T)

int_red_obs_lrt<- as.data.frame(results(int_red_obs_lrt, alpha=0.1))
hist(int_red_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_red_obs_lrt$n_p005<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$pvalue<0.05),])
int_red_obs_lrt$n_q01<- nrow(int_red_obs_lrt[which(int_red_obs_lrt$padj<0.1),])
int_red_obs_lrt<- int_red_obs_lrt[rownames(int_red_obs_lrt) %in% candidates2,]
int_red_obs_lrt$Tissue<- tissue
int_red_obs_lrt$Test<- "LRT_reduced_interaction"
int_red_obs_lrt$design<- paste0(as.character(design(dd_red)), collapse=" ")
int_red_obs_lrt$gene<- rownames(int_red_obs_lrt)

#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

candidates_tests<- rbind(batch_obs_lrt, int_red_obs_lrt, int_min_obs_lrt)
candidates_tests$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(candidates_tests, out_res)
candidates_tests$n<- nrow(tna_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in TNA",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength TNA


### Analysis
Formula ~ Batch + scale(Strength)

```{r TNA Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"tna_data","tna_behav","dd","tissue"))])
varname="Strength"

dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Batch + strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]

summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_TNA_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r TNA strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r TNA strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

<br>

### Gene Ontology


```{r TNA strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r TNA strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Arcopallium intermedium (Ai)

<br>

```{r AI data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="AI"
ai_key<- subset(key_behav, Tissue==tissue)
ai_key<- droplevels(ai_key)

ai_behav<- ai_key[ai_key$Batch!="pilot",]
#Filtering
ai_data<- data[,colnames(data) %in% rownames(ai_key)]
start<- nrow(ai_data)
#remove genes with less than 5 reads
ai_data$avg_count<- apply(ai_data, 1, mean)
ai_data<- ai_data[ai_data$avg_count>5,]
ai_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ai_data$percent_0<- apply(ai_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ai_data)/2
ai_data<- ai_data[ai_data$percent_0<=thresh,]
ai_data$percent_0<-NULL

```

## Checking the sampling

For the AI, we started with `r start` genes but after filering we have `r nrow(ai_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ai samples for `r nrow(ai_behav)`, with respect to T and the tracking data.

```{r AI sampling, fig.align='center'}

fsb<- fisher.test(table(ai_key$Status, ai_key$Batch))
knitr::kable(table(ai_key$Status, ai_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()



fsyb<- fisher.test(table(ai_key$Batch, ai_key$Year))
knitr::kable(table(ai_key$Batch, ai_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",signif(fsyb$p.value,2))) %>% kable_styling
```

<br>

### Expression data exploration

```{r AI DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = ai_key)

biplot(p, lab=ai_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in AI")

eigencorplot(p, metavars = c("Year","Status", "Batch"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

sampleTree = hclust(dist(datExpr0), method = "average")

par(mar = c(0, 4, 2, 0))
plot(sampleTree, main = "Sample clustering on all genes in TNA", xlab="", sub="", cex = 1)
dev.off()
```

<br>

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, ai_key$Batch)
p <- pca(mat, metadata = ai_key)
biplot(p, lab=ai_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))


wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ai_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Removing the batch effect effectively controls for Year.

<br>

## Status AI

### Analysis

Formula = ~ Batch + Status

```{r AI Status, fig.height=8, fig.width=10}
varname="Status"
design(dd)<- formula(~ Batch+ Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_AI_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```

```{r AI status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AI status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```


<br>

### Gene Ontology


```{r AI status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AI status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```



<br>

## Testosterone AI

### Expression data exploration

```{r AI continuous data set up}
rm(list= ls()[!(ls() %in% c(keep,"ai_data","ai_behav","dd","tissue"))])

#Filtering
ai_behav<- subset(key_behav, Tissue==tissue)
ai_behav<- subset(ai_behav, Batch!="pilot")
ai_behav<- droplevels(ai_behav)

ai_data<- data[,colnames(data) %in% rownames(ai_behav)]
start<- nrow(ai_data)
#remove genes with less than 5 reads
ai_data$avg_count<- apply(ai_data, 1, mean)
ai_data<- ai_data[ai_data$avg_count>5,]
ai_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ai_data$percent_0<- apply(ai_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ai_data)/2
ai_data<- ai_data[ai_data$percent_0<=thresh,]
ai_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_AI_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_AI_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ai_behav)
#biplot(p, lab=ai_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")
biplot(p, lab=ai_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

No association with year.  

<br>

### Analysis

Formula = ~ mean_T

```{r AI mean T, fig.height=8, fig.width=10}
varname="mean_T"
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"
write.csv(out_res, file="../DE_results/results_AI_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```

```{r AI mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AI mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```

<br>

### Gene Ontology

```{r AI mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AI mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```



<br>

## mean T x Status


```{r AI mean T x Status, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~ Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$design<- des
out_res$fdrtool<- "no"
write.csv(out_res, file="../DE_results/interactions_AI_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```
```{r}
#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)


int_min_obs_lrt$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(ai_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in AI",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

<br>

## Strength AI

### Analysis
Formula ~ scale(Strength)

```{r AI Strength, fig.height=8, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"ai_data","ai_behav","dd","tissue"))])
varname="Strength"

dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_AI_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```



```{r AI Strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r AI Strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

<br>

### Gene Ontology


```{r AI Strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r AI Strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br><br>

# Lateral Septum (Ls)

<br>

```{r LS data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="LS"
ls_key<- subset(key_behav, Tissue==tissue)
ls_key<- droplevels(ls_key)


ls_behav<- ls_key[ls_key$Batch!="pilot",]

#Filtering
ls_data<- data[,colnames(data) %in% rownames(ls_key)]
start<- nrow(ls_data)
#remove genes with less than 5 reads
ls_data$avg_count<- apply(ls_data, 1, mean)
ls_data<- ls_data[ls_data$avg_count>5,]
ls_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ls_data$percent_0<- apply(ls_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ls_data)/2
ls_data<- ls_data[ls_data$percent_0<=thresh,]
ls_data$percent_0<-NULL

```

## Checking the sampling

For the LS, we started with `r start` genes but after filering we have `r nrow(ls_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ls samples for `r nrow(ls_behav)`, with respect to T and the tracking data.

```{r LS sampling, fig.align='center'}

fsb<- fisher.test(table(ls_key$Status, ls_key$Batch))
knitr::kable(table(ls_key$Status, ls_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()



fsyb<- fisher.test(table(ls_key$Batch, ls_key$Year))
knitr::kable(table(ls_key$Batch, ls_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",signif(fsyb$p.value,2))) %>% kable_styling()


```

### Expression data exploration

```{r LS DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = ls_key)

biplot(p, lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in LS")
#biplot(p, x="PC3", y="PC4", lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in LS")
eigencorplot(p, metavars = c("Year","Status", "Batch"))

```

### Corrections applied

```{r}
mat<- limma::removeBatchEffect(vsd_data, ls_key$Batch)
p <- pca(mat, metadata = ls_key)
biplot(p, lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ls_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

PFT2 was previously identified as an outlier, but using the quantitative threshold and controlling for batch it doesn't come out as different. Controlling for Batch adequately controls for Year.

## Status LS

### Analysis
Formula = ~ Batch + Status

```{r LS Status, fig.height=4, fig.width=10}
varname="Status"
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_LS_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r LS status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r LS status geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```
<br>

### Gene Ontology

```{r LS Status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r LS Status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))


write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## Testosterone LS

### Expression data exploration


```{r LS continuous data set up}
rm(list= ls()[!(ls() %in% c(keep,"ls_data","ls_behav","dd","tissue"))])

tissue="LS"
#Filtering
ls_behav<- subset(key_behav, Tissue==tissue)
ls_behav<- subset(ls_behav, Batch!="pilot")
ls_behav<- droplevels(ls_behav)
ls_data<- data[,colnames(data) %in% rownames(ls_behav)]
start<- nrow(ls_data)
#remove genes with less than 5 reads
ls_data$avg_count<- apply(ls_data, 1, mean)
ls_data<- ls_data[ls_data$avg_count>5,]
ls_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ls_data$percent_0<- apply(ls_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ls_data)/2
ls_data<- ls_data[ls_data$percent_0<=thresh,]
ls_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_LS_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ls_behav)

biplot(p, lab=ls_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK




A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

<br>

### Analysis
Formula = ~ mean_T

```{r LS mean T, fig.height=8, fig.width=10}
varname="mean_T"
design(dd)<- formula(~mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]

summary(res1)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res1)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "yes"

write.csv(out_res, file="../DE_results/results_LS_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()

```


```{r LS mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r LS mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```


<br>

### Gene Ontology

```{r LS mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r LS mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## mean T x Status


```{r LS mean T x Status, echo=FALSE, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$design<- des
out_res$fdrtool<- "no"
write.csv(out_res, file="../DE_results/interactions_LS_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r}
#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

int_min_obs_lrt$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(ls_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in LS",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```

```{r, eval=FALSE}
hub<-"DRD3"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
p<- ggplot(dat, aes(x=strength.all_study, y=count, colour=Status)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_LS_NPY1R_str.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")


p<- ggplot(dat, aes(x=mean_T, y=count, color=Status)) + geom_point(pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(title="AR", subtitle="blah",x="mean T", y="normalized counts") + geom_smooth(method="lm", alpha=0.15, size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio=1)

```



<br>

## Strength LS


### Analysis
Formula ~ scale(Strength)

```{r LS Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"ls_data","ls_behav","dd","tissue"))])
varname="Strength"
  
dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_LS_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r LS Strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r LS Strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```
```{r, eval=FALSE}
hub<-"NPY1R"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
p<- ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_LS_NPY1R_str.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")

```


```{r plotting LS, eval=FALSE}
LS_modules<- read.csv("../WGCNA/ByTissue_2021/LS_results_ModuleMembership.csv")


i="LS"
topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


tissue<- strength_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>0.3, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, LS_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","brown4","black","green","navajowhite2",  "darkorange","lightsteelblue1","orangered4","darkmagenta","maroon","blue","darkgrey","paleturquoise"))
modulecols<- c("grey30","grey60","brown4","black","green","navajowhite2",  "darkorange","lightsteelblue1","orangered4","darkmagenta","maroon","blue","darkgrey","paleturquoise")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")



#ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=0.0)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-0.6,-0.4,-0.2,0,0.2,0.4,0.6))

```

<br>

### Gene Ontology


```{r LS Strength BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r LS strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br><br>

# Bed Nucleus of the Stria Terminalis (BSTm)

<br>

```{r BSTm data set up}

rm(list= ls()[!(ls() %in% keep)])
tissue="BSTm"
bstm_key<- subset(key_behav, Tissue==tissue)
bstm_key<- droplevels(bstm_key)

bstm_behav<- bstm_key[bstm_key$Batch!="pilot",]

#Filtering
bstm_data<- data[,colnames(data) %in% rownames(bstm_key)]
start<- nrow(bstm_data)
#remove genes with less than 5 reads
bstm_data$avg_count<- apply(bstm_data, 1, mean)
bstm_data<- bstm_data[bstm_data$avg_count>5,]
bstm_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
bstm_data$percent_0<- apply(bstm_data, 1, function(x)length(x[x==0]))
thresh<- ncol(bstm_data)/2
bstm_data<- bstm_data[bstm_data$percent_0<=thresh,]
bstm_data$percent_0<-NULL

```
## Checking the sampling

For the BSTm, we started with `r start` genes but after filering we have `r nrow(bstm_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have bstm samples for `r nrow(bstm_behav)`, with respect to T and the tracking data.

```{r BSTm sampling, fig.align='center'}

fsb<- fisher.test(table(bstm_key$Status, bstm_key$Batch))
knitr::kable(table(bstm_key$Status, bstm_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", signif(fsb$p.value,2))) %>% kable_styling()



fsyb<- fisher.test(table(bstm_key$Batch, bstm_key$Year))
knitr::kable(table(bstm_key$Batch, bstm_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",signif(fsyb$p.value,2))) %>% kable_styling()


```

<br>

### Expression data exploration

```{r BSTm DEseq1}
dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = bstm_key)

biplot(p, lab=bstm_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in BSTm")

eigencorplot(p, metavars = c("Year","Status", "Batch"))




```

<br>

### Corrections applied

```{r}

mat<- limma::removeBatchEffect(vsd_data, bstm_key$Batch)
p <- pca(mat, metadata = bstm_key)
biplot(p, lab=bstm_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- bstm_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")


```

Batch controls effectively for Year.

<br>

## Status BSTm

### Analysis

Formula = ~ Batch + Status

```{r BSTm Status, fig.height=4, fig.width=10}
varname="Status"

design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_BSTm_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

#EnhancedVolcano(res,
#    lab = rownames(res),
#    x = 'log2FoldChange',
#    y = 'padj',
#    ylab = bquote(~Log[10]~ 'padj'),
#    pCutoff=0.1,
#    FCcutoff = 0,
#    pointSize = 3.0,
#    labSize = 4.0,
#    col=c('grey20', 'grey20', 'grey20', 'red3'),
#    colAlpha = 1,
#    legendLabels=c("",'Not sig.',"",'significant padj<0.1'),
#    drawConnectors = TRUE)
```

```{r BSTm status volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r BSTm geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```
```{r,eval=FALSE}
hub="AR"
dat<- plotCounts(dd, gene=row.names(res[hub,]), intgroup=c("Class","Status"), returnData=TRUE)
p<- ggplot(dat, aes(x=Status, y=count, color=Status)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="social status", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1))) + geom_boxplot(fill=NA, outlier.shape = NA, lwd=0.3) + theme(legend.position = "none", aspect.ratio = 1)
p
ggsave("../DE_results/figure_BSTm_AR_status.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")




```

<br>

### Gene Ontology


```{r BSTm Status BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r BSTm status Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


<br>

## Testosterone BSTm

### Expression data exploration
```{r BSTm continuous data set up}
rm(list= ls()[!(ls() %in% c(keep,"bstm_data","bstm_behav","dd","tissue"))])

#Filtering
bstm_behav<- subset(key_behav, Tissue=="BSTm")
bstm_behav<- subset(bstm_behav, Batch!="pilot")
bstm_behav<- droplevels(bstm_behav)
bstm_data<- data[,colnames(data) %in% rownames(bstm_behav)]
start<- nrow(bstm_data)
#remove genes with less than 5 reads
bstm_data$avg_count<- apply(bstm_data, 1, mean)
bstm_data<- bstm_data[bstm_data$avg_count>5,]
bstm_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
bstm_data$percent_0<- apply(bstm_data, 1, function(x)length(x[x==0]))
thresh<- ncol(bstm_data)/2
bstm_data<- bstm_data[bstm_data$percent_0<=thresh,]
bstm_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../DE_results/data_BSTm_norm.csv", row.names=TRUE, quote=FALSE)



p <- pca(vsd_data, metadata = bstm_behav)
biplot(p, lab=bstm_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")
#grid.arrange(a,b,c,d,e,f, ncol=2, nrow=3, newpage=TRUE, top=textGrob("PCA plots for each Variable in BSTm", gp=gpar(fontsize=14)))
eigencorplot(p, metavars = c("time_kill","Year","Status", "mean_T","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

<br>

### Analysis
Formula = ~ mean_T

```{r BSTm mean T, fig.height=4, fig.width=10}
varname="mean_T"
design(dd)<- formula(~mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res, 0.95)
res<- res[order(res$padj),]

summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_BSTm_mean_T.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r BSTm mean T volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r BSTm mean T geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c(varname,"Class"),6, status_cols)
g
```
<br>

### Gene Ontology

```{r BSTm mean T BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r BSTm mean T Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```

<br>

## mean T x Status


```{r BSTm mean T x Status, echo=FALSE, fig.height=4, fig.width=10}

dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd,test="LRT",reduced=~ + Status + mean_T)
#dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$n_p005<- nrow(out_res[which(out_res$pvalue<0.05),])
out_res$n_q01<- nrow(out_res[which(out_res$padj<0.1),])
out_res$Test<- "LRT_full_interaction"
out_res$fdrtool<- "no"
write.csv(out_res, file="../DE_results/interactions_BSTm_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="BSTm mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```

```{r}
#minimal LRT.
dd_min <- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_behav, design= ~ Status:mean_T)
int_min_obs_lrt<- DESeq(dd_min,test="LRT",reduced=~1)
int_min_obs_lrt<- as.data.frame(results(int_min_obs_lrt, alpha=0.1))
hist(int_min_obs_lrt$pvalue, breaks=100, col="grey", main="", xlab="uncorrected P-value")
int_min_obs_lrt$n_p005<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$pvalue<0.05),])
int_min_obs_lrt$n_q01<- nrow(int_min_obs_lrt[which(int_min_obs_lrt$padj<0.1),])
int_min_obs_lrt<- int_min_obs_lrt[rownames(int_min_obs_lrt) %in% candidates2,]
int_min_obs_lrt$Tissue<- tissue
int_min_obs_lrt$Test<- "LRT_minimal_interaction"
int_min_obs_lrt$design<- paste0(as.character(design(dd_min)), collapse=" ")
int_min_obs_lrt$gene<- rownames(int_min_obs_lrt)

int_min_obs_lrt$fdrtool<- "no"
out_res<- out_res[which(out_res$gene %in% candidates2),]
out_res<- out_res[,-which(names(out_res) %in% c("best_anno","display_gene_ID"))]
candidates_tests<- rbind(int_min_obs_lrt, out_res)
candidates_tests$n<- nrow(bstm_behav)
write.csv(candidates_tests,file=paste0("../Candidate_gene_results/results_DE_",tissue,"int_models.csv"),row.names = FALSE )

```

```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in BSTm",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


<br>

## Strength BSTm

### Analysis
Formula ~ scale(Strength)

```{r BSTm Strength, fig.height=4, fig.width=10}
rm(list= ls()[!(ls() %in% c(keep,"bstm_data","bstm_behav","dd","tissue"))])
varname="Strength"

dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


des<- as.character(design(dd))
des<- paste0(des, collapse=" ")
out_res<- data.frame(res)
out_res$gene<- row.names(out_res)
out_res<- merge(out_res, genes_key,by="gene")
out_res<- out_res[order(out_res$padj),]
out_res$Tissue<- tissue
out_res$design<- des
out_res$fdrtool<- "no"

write.csv(out_res, file="../DE_results/results_BSTm_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main=paste("MA-plot",tissue,"exp",des))

hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()


```




```{r BSTm strength volcano plot}
volcano_plots(out_res, tissue_cols[[tissue]])
vp
```


```{r BSTm strength geneplots, fig.height=8, fig.width=14}
gene_plots(out_res,c("strength.all_study","Class"),6, status_cols)
g
```

```{r,  eval=FALSE}

hub<-"PGR"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
p<- ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=0.6, pch=19) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="average number interactions/day (scaled)", y="normalized counts", title=hub, subtitle=paste0("LFC=", signif(res$log2FoldChange[rownames(res)==hub],1), ", p=",signif(res$pvalue[rownames(res)==hub],1), ", q=", signif(res$padj[rownames(res)==hub],1), " R^2=", signif(Rsq,1))) + geom_smooth(method="lm", alpha=0.15,, color="grey50", size=0.4, se=FALSE) + theme(legend.position = "none", aspect.ratio = 1) + geom_point(aes(color=Status), size=0.6, pch=19)
p
ggsave("../DE_results/figure_BSTm_PGR_str_noerror.pdf", plot=p, device="pdf", height=1.60, width=1.60, units="in")

```


<br>

### Gene Ontology


```{r BSTm BP GSEA}
res_gsea_bp(out_res)

gseaplot

write.csv(gsearesult, file=paste0("../DE_results/GSEA_BP_",tissue,"_",varname,".csv"), row.names=FALSE)
```

```{r BSTm strength Enrich}
res_go_bp(out_res)

go_plot_result

knitr::kable(enrich_results[1:10,-1], caption=paste0("Top 10 enriched GO BP in ",tissue, " according to ",varname)) %>% kable_styling()

go_table<- enrich_results[1:10,c("ID","Description","GeneRatio","pvalue","p.adjust")]
go_table<- go_table[!is.na(go_table$ID),]
go_table$pvalue<- signif(go_table$pvalue,2)
go_table$p.adjust<- signif(go_table$p.adjust,2)

go_table<- ggtexttable(as.data.frame(go_table),theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")),rows=NULL) %>% tab_add_title(text = main.title, size=6, padding = unit(0.1, "line"))
 

write.csv(enrich_results, file=paste0("../DE_results/GO_BP_",tissue,"_",varname,".csv"), row.names=FALSE)

```

```{r}
g2<- ggarrange(ggarrange(vp,g,widths=c(0.7,1), labels=c("A","B"),ncol=2,nrow=1),ggarrange(gseaplot,go_table, ncol=2,nrow=1, labels=c("C","D")), nrow=2)


#g2
ggsave(filename=paste0("../DE_results/supp_figure_",tissue,"_",varname,".png"), plot=g2, device="png" ,height=180, width=200, units="mm", bg="white")
```


Volcano plots and similarity of gene expression between tissues is in another Rmarkdown on this site.

```{r}
sessionInfo()
```


