---
title: "Candidate Gene Analysis"
subtitle: "For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird "
date: Last Knit "`r Sys.Date()`"
author: "Last Substantive Change December 2023"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE, fig.align="center", comment=FALSE)
```

```{r libraries used, echo=FALSE}
library(plyr)
library(reshape2)
library(ggplot2)
library(ggpubr)
library(stringr)
library(DESeq2)
library(kableExtra)
library(VennDiagram)
library(MASS)
library(pheatmap)
library(RColorBrewer)
library(reshape2)
library(WGCNA)

save_pheatmap_png <- function(x, filename, width=1200, height=1000, res = 150) {
  png(filename, width = width, height = height, res = res)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

source("config.R")


```

```{r results data}
key<- read.csv("../data_filtered/data_key_Parsed_ReplicatesRemoved.csv")
               
behav<- read.csv("../data_unfiltered/PIPFIL_T_and_behav_data.csv")
rownames(key)<- key$X

key$Color_ID<- sub("/","", key$Color_ID)
key<- plyr::rename(key, replace=c("Color_ID"="colorID"))
behav$status<- plyr::revalue(behav$status, c("floa"="floater", "terr"="territorial"))
key_behav<- merge(key, behav, by="colorID")
key_behav<- key_behav[!is.na(key_behav$last_behav),]


#create a data.frame with all of the observations. 
not_in_behav<- key[!key$X %in% key_behav$X,]
cols_not_key<- colnames(key_behav)[!colnames(key_behav) %in% colnames(key)]
cols_not_key_df<- data.frame(matrix(NA, nrow = nrow(not_in_behav), ncol = length(cols_not_key)))
colnames(cols_not_key_df)<- cols_not_key
not_in_behav<- cbind(not_in_behav, cols_not_key_df)

key_behav<- rbind(key_behav, not_in_behav)

rownames(key_behav)<- key_behav$X
key_behav<- subset(key_behav, select=c("sampleID", "Year", "Tissue", "Batch", "Status", "Class", "final_corrected_T", "mean_T", "effort.all_study", "degree.all_study", "strength.all_study"))
#key_behav$Class<- word(key_behav$Class,1)

key_behav$Tissue<- factor(key_behav$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
key_behav<- key_behav[order(rownames(key_behav)),]
key_behav$Class<- plyr::revalue(key_behav$Class, replace=c("SCB floater"="Predefinitive floater", "DCB floater "="Definitive floater", "DCB territorial"="Territorial"))
key_behav$Class<- factor(key_behav$Class, levels=c("Predefinitive floater", "Definitive floater", "Territorial"))


data<- read.csv("../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv")
data$X[data$X=="LOC113993669"] <- "CYP19A1"
data$X[data$X=="LOC113983511"] <- "OXT"
data$X[data$X=="LOC113983498"] <- "AVP"
data$X[data$X=="LOC113982601"] <- "AVPR2"
rownames(data)<- data$X
data$X<- NULL



setwd("G:/My Drive/Manakins/Pipra Brain/PIFI_brain_transcriptome/DE_results")

norm_data<- list.files(pattern="*norm.csv")
norm_data<- lapply(norm_data, read.csv)
for (i in seq_along(norm_data)){

  norm_data[[i]]$X[norm_data[[i]]$X=="LOC113993669"] <- "CYP19A1"
  norm_data[[i]]$X[norm_data[[i]]$X=="LOC113983511"] <- "OXT"
  norm_data[[i]]$X[norm_data[[i]]$X=="LOC113983498"] <- "AVP"
  norm_data[[i]]$X[norm_data[[i]]$X=="LOC113982601"] <- "AVPR2"
}


results_names<- list.files(pattern="results*")
results_names<- results_names[!grepl("brain",results_names)]
results_data<- lapply(results_names, read.csv)
results_names<- sub(".csv","", results_names)
results_names<- sub("results_", "",results_names)


names(results_data)<- results_names
results_data<- mapply(`[<-`, results_data, 'result', value = results_names, SIMPLIFY = FALSE)
results_data<- lapply(results_data, function(x) { x["X"] <- NULL; x })
results_data<- do.call("rbind", results_data)
results_data$Tissue<- word(results_data$result, 1,1, sep="_")
results_data$Test<- word(results_data$result, 2,2, sep="_")
results_data$Test<- as.factor(results_data$Test)
setwd("G:/My Drive/Manakins/Pipra Brain/PIFI_brain_transcriptome/analysis")

results_data$Test<- revalue(results_data$Test, c("status"="Status", "mean"="mean T"))
#results_sig<- results_data[results_data$pvalue<0.05,]
results_data$Tissue<- factor(results_data$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS", "BSTm"))


results_data<- droplevels(results_data)
## List of candidate genes

candidates<- c("AR", "SRD5A2", "LOC113993669", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","LOC113983511", "OXTR", "LOC113983498", "AVPR1A", "AVPR1B", "LOC113982601")
candidates2<- c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B", "AVPR2")


results_data<- results_data[which(results_data$gene %in% candidates2),]


results_data<- results_data[,c("Tissue","gene","Test","design", "fdrtool","result","baseMean","log2FoldChange","pvalue","padj")]
#results_data$gene[results_data$gene=="LOC113993669"]<- "CYP19A1"
#results_data$gene[results_data$gene=="LOC113983511"]<- "OXT"
#results_data$gene[results_data$gene=="LOC113983498"]<- "AVP"
#results_data$gene[results_data$gene=="LOC113982601"]<- "AVPR2"
```


```{r interaction results}
## reading in the interaction data
setwd("../Candidate_gene_results/")

results_names<- list.files(pattern="int_models.csv")
results_int<- lapply(results_names, read.csv)
results_int[[8]]$n<- 12
results_int<- do.call("rbind", results_int)

setwd("G:/My Drive/Manakins/Pipra Brain/PIFI_brain_transcriptome/analysis")

results_int$Tissue<- factor(results_int$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS", "BSTm"))


#results_int<- results_int[which(results_int$gene %in% candidates2),]

results_int<- results_int[,c("Tissue","gene","Test","design", "fdrtool","n","baseMean","log2FoldChange","pvalue","padj")]

#results_int$gene[results_int$gene=="LOC113993669"]<- "CYP19A1"
#results_int$gene[results_int$gene=="LOC113983511"]<- "OXT"
#results_int$gene[results_int$gene=="LOC113983498"]<- "AVP"
#results_int$gene[results_int$gene=="LOC113982601"]<- "AVPR2"


```


```{r keep}
final_res<- list()
keep<- c("candidates", "candidates2","key_behav", "results_data","results_int","norm_data", "peri_theme","key","dodge", "status_cols", "dj_col", "colours", "keep", "save_pheatmap_png", "data", "status_cols2", "peri_figure", "peri_geneplots","final_res","peri_figure_supp")

```

This document contains the differential expression results from DESeq2 for our candidate genes. These candidates are: ```r candidates```.  

* LOC113993669 is CYP19A1 (Aromatase)

* LOC113983511 is OXT (Mesotocin)

* LOC113983498 is vasotocin-neurophysin (AVP precursor) 

* LOC113982601 is AVPR2. 

All p-values (raw and corrected) presented in this document are derived from the outputs of DESeq2 without further modification. 


# Gonads (GON)

The next sections are scatter plots showing the expression profiles of candidate genes in each tissue. The candidate genes were first selected from the DESeq2 results based on a raw p-value of <0.05, and then plotted for validation. We excluded genes which had low R^2, and/or highly influential observations from further analysis. 

```{r GON prep, echo=TRUE}
rm(list= ls()[!(ls() %in% keep)])
tissue="GON"

### Create subset for Status ####
gon_key<- subset(key_behav, Tissue=="GON")
gon_key<- droplevels(gon_key)


gon_data<- data[,colnames(data) %in% rownames(gon_key)]

#remove genes with less than 5 reads
gon_data$avg_count<- apply(gon_data, 1, mean)
gon_data<- gon_data[gon_data$avg_count>5,]
gon_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gon_data$percent_0<- apply(gon_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gon_data)/2
gon_data<- gon_data[gon_data$percent_0<=thresh,]
gon_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_key, design= ~ Batch)
dd<- DESeq(dd)

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(gon_key, status_results, by="sampleID")


## data for continuous variables. 
gon_norm<- norm_data[[5]]
rownames(gon_norm)<- gon_norm$X
gon_norm$X<- NULL
gon_norm<- as.data.frame(t(gon_norm))
gon_norm<- gon_norm[,colnames(gon_norm) %in% candidates2]
gon_norm$sampleID<- rownames(gon_norm)
gon_norm<- merge(gon_norm, gon_key, by="sampleID")
```

```{r GON initial results}
## Extract GON results
#gon_cand<- candidates2[candidates2 %in% rownames(gon_data)]

results_gon<- results_data[results_data$Tissue==tissue,]
#results_gon<- results_gon[results_gon$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_gon2<- results_gon[which(results_gon$pvalue<0.05),]
results_gon2<- results_gon2[order(results_gon2$gene),]
results_gon2<- rbind(subset(results_int_sub2,select=-n), subset(results_gon2, select=-result))
#results_gon2$padj_cand<- p.adjust(results_gon2$pvalue, method="BH", n=length(gon_cand))

knitr::kable(results_gon2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()
```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r GON batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone! 

### Alternate models

```{r GON alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in",tissue)) %>% kable_styling()

```

Therefore not only is the Year variable not important for expression in any of our candidate genes, but status specific expression of VIPR2 is not sensitive to inclusion of this parameter in the model. Therefore I am going to use the model without YEAR included. 

### AIC

#### VIPR2

```{r GON AIC vIPR2}
gene="VIPR2"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



a<- ggplot(gon_norm, aes(x=mean_T, y=VIPR2, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
gon_lm_Status<- lm(VIPR2 ~ Status, data=gon_norm)
gon_lm_meanT<- lm(VIPR2 ~ mean_T, data=gon_norm)
gon_lm_int<- lm(VIPR2 ~ Status*mean_T, data=gon_norm)
#against the NULL model
gon_lm_null<- lm(VIPR2 ~ 1, data=gon_norm)

model=c("VIPR2 ~ 1", "VIPR2 ~ Status", "VIPR2 ~ mean T","VIPR2 ~ Status + mean_T + Status:mean_T")
AIC=c(AIC(gon_lm_null), AIC(gon_lm_Status),AIC(gon_lm_meanT),AIC(gon_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(gon_lm_Status)$'Pr(>F)'[1],anova(gon_lm_meanT)$'Pr(>F)'[1], anova(gon_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_gon$pvalue[results_gon$gene==gene & results_gon$Test=="Status"], results_gon$pvalue[results_gon$gene==gene & results_gon$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b, labels=c("A","B"), nrow=2)
figure

ggsave(filename="../Candidate_gene_results/GON_interaction.png", figure,device="png", width=80, height=100, units="mm", dpi=300, bg="white")
```

Not going to accept VIPR2 because it's not better than the null. 

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

```{r}
results_gon2$keep<- c("no","yes","yes","no")

```



## Gene Plots

```{r GON lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_gon2)){
  #g=2
  gene<- results_gon2$gene[g]
  gene<- as.character(gene)
  test<- word(results_gon2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(gon_norm[,gene] ~ Status * mean_T, data=gon_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(gon_norm[,gene] ~ gon_norm[,cont_batch], data=gon_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(gon_norm[,gene] ~ gon_norm[,cont_batch] + gon_norm[,test], data=gon_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(gon_norm[,test], gon_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(gon_norm[,gene] ~ gon_norm[,test], data=gon_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(gon_norm[,test], gon_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(gon_cand))

results_gon3<- cbind(results_gon2, lm_out[,-4])
final_res[[tissue]]<- results_gon3
```


```{r GON plotting}
plots<- list()

for(g in 1:nrow(results_gon3)){
  #g=1
  gene<- results_gon3$gene[g]
  gene<- as.character(gene)
  test<- word(results_gon3$design[g],-1)
  kp<- ifelse(results_gon3$keep[g]=="yes","*","")
  des<- results_gon3$design[g]
  
  lfc<- signif(results_gon3$log2FoldChange[g],2)
  praw<- signif(results_gon3$pvalue[g],2)
  padj=signif(results_gon3$padj[g],2)
  Rsq=signif(results_gon3$rsq[g],2)
  #lm_model<- results_gon3$lm_model[g]
  #lm_p<-signif(results_gon3$lm_p[g],2)
  #lm_q<- signif(results_gon3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(gon_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(gon_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(gon_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-1], ncol=2, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/GON_significant.png", figure,device="png",height=100, width=100, units="mm", bg="white")

```



# Pituitary (PIT)

Hereafter, I exclude repeated code from presentation, but it is available on the github repository. Any variation on the code presented previously is shown.

```{r PIT prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="PIT"

pit_key<- subset(key_behav, Tissue=="PIT")
pit_key<- droplevels(pit_key)

#there were no pituitary samples in the pilot study so can just use the same data for the whole lot... 
pit_norm<- norm_data[[8]]
rownames(pit_norm)<- pit_norm$X
pit_norm$X<- NULL
pit_norm<- as.data.frame(t(pit_norm))
pit_norm<- pit_norm[,colnames(pit_norm) %in% candidates2]
pit_norm$sampleID<- rownames(pit_norm)
pit_norm<- merge(pit_norm,pit_key, by="sampleID")

```

This one is a little more complicated because PFT3 was excluded from the overall strength DE analysis owing to some genes with strong outlier effects that I couldn't remove otherwise. So we're also going to compare results with and without PFT3.

```{r PIT initial results}
## Extract PIT results
pit_cand<- candidates2[candidates2 %in% rownames(pit_norm)]


results_pit<- results_data[results_data$Tissue==tissue & !grepl("Strength",results_data$result),]
results_pit<- results_pit[results_pit$gene %in% candidates2,]

#strength results without PFT3
results_strength<- results_data[results_data$Tissue==tissue & !grepl("PFT3|mean|Status",results_data$result) & results_data$gene %in% candidates2,]
results_strength<- results_strength[order(results_strength$gene),]
colnames(results_strength)<- paste0("woPFT3_",colnames(results_strength))

#Strength results WITH PFT3
results_pft3<- results_data[results_data$Tissue==tissue & grepl("PFT3",results_data$result) & results_data$gene %in% candidates2,]
results_pft3<- results_pft3[order(results_pft3$gene),]

all.equal(results_pft3$woPFT3_gene, results_strength$gene)

results_strength<- cbind(results_pft3,results_strength[,c(7:10)])
  
results_pit[,c("woPFT3_baseMean","woPFT3_log2FoldChange","woPFT3_pvalue","woPFT3_padj")]<- NA
results_pit<- rbind(results_strength, results_pit)


results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]
results_int_sub2[,c("woPFT3_baseMean","woPFT3_log2FoldChange","woPFT3_pvalue","woPFT3_padj")]<- NA


results_pit2<- results_pit[which(results_pit$pvalue<0.05 | results_pit$woPFT3_pvalue<0.05),]
results_pit2<- results_pit2[order(results_pit2$gene),]
results_pit2<- results_pit2[,c(1:5,7:14)]

results_pit2<- rbind(subset(results_int_sub2,select=-n), results_pit2)
#results_pit2$padj_cand<- p.adjust(results_pit2$pvalue, method="BH", n=length(pit_cand))

knitr::kable(results_pit2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()
```


## Exploring status specific expression

### Batch

```{r PIT batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()


```

Batch is important for AVP. 

### Alternate models

```{r PIT alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in",tissue)) %>% kable_styling()
```

Looks like AVPR1A is not stable to model form, and while AVP is stable to model form, the gene plots for this gene are rubbish (see below). I'm still going to plot it but likely going to exclude it. 

### AIC

#### AVPR1A

```{r PIT AIC AVPR1A}
gene="AVPR1A"
design="~ Batch + Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(pit_norm, aes(x=mean_T, y=AVPR1A, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pit_lm_Status<- lm(AVPR1A ~ Batch +  Status, data=pit_norm)
pit_lm_meanT<- lm(AVPR1A  ~ Batch + mean_T, data=pit_norm)
pit_lm_int<- lm(AVPR1A ~ Batch + Status*mean_T, data=pit_norm)
#against the NULL model
pit_lm_null<- lm(AVPR1A ~ Batch, data=pit_norm)

model=c("AVPR1A ~ Batch", "AVPR1A ~ Batch + Status", "AVPR1A ~ Batch +  mean T","AVPR1A ~ Batch +  Status * mean T")
AIC=c(AIC(pit_lm_null), AIC(pit_lm_Status),AIC(pit_lm_meanT),AIC(pit_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])

lm_p<- c(anova(pit_lm_null)$'Pr(>F)'[1],anova(pit_lm_Status)$'Pr(>F)'[2],anova(pit_lm_meanT)$'Pr(>F)'[2],anova(pit_lm_int)$'Pr(>F)'[4])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pit$pvalue[results_pit$gene==gene & results_pit$Test=="Status"], results_pit$pvalue[results_pit$gene==gene & results_pit$Test=="mean T"], results_int_sub2$pvalue[results_int_sub2$gene==gene])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the pitads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


#figure<- ggarrange(a,b, labels=c("A","B"), nrow=2)
#figure

#ggsave(filename=paste0("../Candidate_gene_results/",tissue,"_interaction.png"), figure,device="png", width=20, height=15, units="cm", dpi=300, bg="white")
```

Not going to accept AVPR1A, doesn't beat AIC test and is sensitive to inclusion of Batch.

#### VIPR2

VIPR2 was found to be associated with both mean_T and strength.

```{r PIT AIC VIPR}
gene="VIPR2"
design="~ strength.all_study"


  
pit_lm_strength<- lm(VIPR2 ~ strength.all_study, data=pit_norm)
pit_lm_meanT<- lm(VIPR2  ~  mean_T, data=pit_norm)
#pit_lm_int<- lm(VIPR2 ~ Batch + Status*mean_T, data=pit_norm)
#against the NULL model
pit_lm_null<- lm(VIPR2 ~ 1, data=pit_norm)

model=c("VIPR2 ~ 1", "*VIPR2 ~ Strength", "VIPR2 ~ mean T")
AIC=c(AIC(pit_lm_null), AIC(pit_lm_strength),AIC(pit_lm_meanT))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3])

lm_p<- c(anova(pit_lm_null)$'Pr(>F)'[1],anova(pit_lm_strength)$'Pr(>F)'[1],anova(pit_lm_meanT)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pit$pvalue[results_pit$gene==gene & results_pit$Test=="Strength"], results_pit$pvalue[results_pit$gene==gene & results_pit$Test=="mean T"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the pitads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

c<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)), rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

blank <- grid.rect(gp=gpar(col="white"))
figure<- ggarrange(ggarrange(a,b, nrow=2, ncol=1, labels=c("A","B"), widths=c(1,1.5)),c,labels=c("","C"), ncol=2)
figure

ggsave(filename=paste0("../Candidate_gene_results/",tissue,"_interaction.png"), figure,device="png", width=180, height=120, units="mm", dpi=300, bg="white")
```


## Final Res

I'm excluding AVP and OXT owing to errant expression patterns, I'm excluding ESR1 because of an influential observation, VIPR1 and AVPR1B because results don't hold after removal of PFT3, and AVPR1A because it failed the AIC and model parameter tests.

```{r}
results_pit2$keep<- ifelse(!grepl("AVP|OXT$|ESR1|VIPR1|AVPR1A|AVPR1B",results_pit2$gene),"yes","no")
results_pit2$keep<- ifelse((results_pit2$gene=="VIPR2" & results_pit2$Test=="mean T"),"no", as.character(results_pit2$keep))

```



## Gene Plots
```{r PIT lm}

lm_out<- list()
cont_batch<- "Batch"
status_batch<- "Batch"
status_data<- pit_norm
pit_sub<- pit_norm[which(!grepl("PFT3",pit_norm$sampleID)),]

for(g in 1:nrow(results_pit2)){
  #g=2
  gene<- results_pit2$gene[g]
  gene<- as.character(gene)
  test<- word(results_pit2$design[g],-1)
  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(pit_norm[,gene] ~ Batch + Status * mean_T, data=pit_norm)

  out<- data.frame(lm_model= " ~ Batch + Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  out$lm_p_woPFT3<- NA
  out$rsq_woPFT3<- NA
  
  lm_out[[g]]<- out
  }else if(test=="mean_T" ){
    batch_p<-  summary(lm(pit_norm[,gene] ~ pit_norm[,cont_batch], data=pit_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(pit_norm[,gene] ~ pit_norm[,cont_batch] + pit_norm[,test], data=pit_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<- signif(cor(pit_norm[,test], pit_norm[,gene])^2,2)
      out$gene<- gene
      out$lm_p_woPFT3<- NA
      out$rsq_woPFT3<- NA
      lm_out[[g]]<- out
      
    }else{
      lm_cont<- lm(pit_norm[,gene] ~ pit_norm[,test], data=pit_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<- signif(cor(pit_norm[,test], pit_norm[,gene])^2,2)
      out$gene<- gene
      out$lm_p_woPFT3<- NA
      out$rsq_woPFT3<- NA
      lm_out[[g]]<- out
    }
  }else if(test=="strength.all_study"){
        batch_p<-  summary(lm(pit_norm[,gene] ~ pit_norm[,cont_batch], data=pit_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(pit_norm[,gene] ~ pit_norm[,cont_batch] + pit_norm[,test], data=pit_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<- signif(cor(pit_norm[,test], pit_norm[,gene])^2,2)
      out$gene<- gene
      
      #remove PFT3 and run again.
      lm_cont<- lm(pit_sub[,gene] ~ pit_sub[,cont_batch] + pit_sub[,test], data=pit_sub)
      out$lm_p_woPFT3<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq_woPFT3<- signif(cor(pit_sub[,test], pit_sub[,gene])^2,2)
      lm_out[[g]]<- out
      
    }else{
      lm_cont<- lm(pit_norm[,gene] ~ pit_norm[,test], data=pit_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<- signif(cor(pit_norm[,test], pit_norm[,gene])^2,2)
      out$gene<- gene
      
      ##remove PFT3 
      lm_cont<- lm(pit_sub[,gene] ~ pit_sub[,test], data=pit_sub)
      out$lm_p_woPFT3<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq_woPFT3<- signif(cor(pit_sub[,test], pit_sub[,gene])^2,2)
      lm_out[[g]]<- out
    }
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        out$lm_p_woPFT3<- NA
        out$rsq_woPFT3<- NA
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        out$lm_p_woPFT3<- NA
        out$rsq_woPFT3<- NA
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(pit_cand))

results_pit3<- cbind(results_pit2, lm_out)
```

```{r PIT plotting}
plots<- list()
#results_pit_most<- results_pit3[!grepl("strength", results_pit3$design),]
results_pit3<- results_pit3[order(results_pit3$Test),]

for(g in 1:nrow(results_pit3)){
  #g=1
  gene<- results_pit3$gene[g]
  test<- word(results_pit3$design[g],-1)
  kp<- ifelse(results_pit3$keep[g]=="yes","*","")
  des<- results_pit3$design[g]
  
  lfc<- signif(results_pit3$log2FoldChange[g],2)
  praw<- signif(results_pit3$pvalue[g],2)
  padj=signif(results_pit3$padj[g],2)
  Rsq=signif(results_pit3$rsq[g],2)
  #lm_model<- results_pit3$lm_model[g]
  #lm_p<-signif(results_pit3$lm_p[g],2)
  #lm_q<- signif(results_pit3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj))) + theme(aspect.ratio = 1)
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    #test<- gsub(":"," x ",test)
    
  #plots[[g]]<-ggplot(pit_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(gene, " ~ ", test),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 

    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(pit_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(aspect.ratio = 1)
  }else if(test=="strength.all_study"){
    stra<- ggplot(pit_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="", y="", title=paste0(kp,gene,  des, "\nw pft3"),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status") + theme(legend.position = "none")
    
    lfc<- signif(results_pit3$woPFT3_log2FoldChange[g],2)
    praw<- signif(results_pit3$woPFT3_pvalue[g],2)
    padj=signif(results_pit3$woPFT3_padj[g],2)
    Rsq=signif(results_pit3$rsq_woPFT3[g],2)
   
   strb<- ggplot(pit_sub, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des, "\nw/o pft3"),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status") + theme(legend.position = "none")
   p<- ggarrange(stra,strb, nrow=2,ncol=1)
   plots[[g]]<- p
  }
}

figure<- ggarrange(plotlist = plots[-c(1:2)], ncol=6, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/PIT_significant.png", figure,device="png",height=200, width=280, units="mm", bg="white")

write.csv(results_pit3, file="../Candidate_gene_results/final_list_PIT.csv", row.names=FALSE)
```




# Ventromedial Hypothalamus (VMH)

This tissue used additional samples for the Status comparison - thus the code used to plot these figures is presented.

```{r VMH prep, echo=TRUE}
rm(list= ls()[!(ls() %in% keep)])
tissue="VMH"
vmh_key<- subset(key_behav, Tissue=="VMH")
vmh_key<- droplevels(vmh_key)


vmh_data<- data[,colnames(data) %in% rownames(vmh_key)]

#remove genes with less than 5 reads
vmh_data$avg_count<- apply(vmh_data, 1, mean)
vmh_data<- vmh_data[vmh_data$avg_count>5,]
vmh_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
vmh_data$percent_0<- apply(vmh_data, 1, function(x)length(x[x==0]))
thresh<- ncol(vmh_data)/2
vmh_data<- vmh_data[vmh_data$percent_0<=thresh,]
vmh_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_key, design= ~ Batch)
dd<- DESeq(dd)


vmh_norm<- norm_data[[12]]
rownames(vmh_norm)<- vmh_norm$X
vmh_norm$X<- NULL
vmh_norm<- as.data.frame(t(vmh_norm))
vmh_norm<- vmh_norm[,colnames(vmh_norm) %in% candidates2]
vmh_norm$sampleID<- rownames(vmh_norm)
vmh_norm<- merge(vmh_norm, vmh_key, by="sampleID")

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(vmh_key, status_results, by="sampleID")

```

```{r VMH initial results}
results_vmh<- results_data[results_data$Tissue=="VMH",]


results_vmh<- results_data[results_data$Tissue==tissue,]
#results_vmh<- results_vmh[results_vmh$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_vmh2<- results_vmh[which(results_vmh$pvalue<0.05),]
results_vmh2<- results_vmh2[order(results_vmh2$gene),]
results_vmh2<- rbind(subset(results_int_sub2,select=-n), subset(results_vmh2, select=-result))
#results_vmh2$padj_cand<- p.adjust(results_vmh2$pvalue, method="BH", n=length(vmh_cand))

knitr::kable(results_vmh2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()

```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r VMH batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```



### Alternate models

```{r VMH alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in",tissue)) %>% kable_styling()
```

Batch + Year seems to be important 

### AIC

#### GNRH1

```{r VMH AIC GNRH1}
gene="GNRH1"
design="~ Batch + Year + Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(vmh_norm, aes(x=mean_T, y=GNRH1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  

a

vmh_lm_Status<- lm(GNRH1 ~ Batch + Year + Status, data=vmh_norm)
vmh_lm_meanT<- lm(GNRH1 ~ Batch + Year + mean_T, data=vmh_norm)
vmh_lm_int<- lm(GNRH1 ~ Batch + Year + Status*mean_T, data=vmh_norm)

#against the NULL model
vmh_lm_null<- lm(GNRH1 ~ Batch + Year, data=vmh_norm)

model=c("GNRH1 ~ 1", "GNRH1 ~ Batch + Year + Status", "GNRH1 ~ Batch + Year + mean T","GNRH1 ~ Batch + Year +  Status + mean T + Status:meanT")
AIC=c(AIC(vmh_lm_null), AIC(vmh_lm_Status),AIC(vmh_lm_meanT),AIC(vmh_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(vmh_lm_Status)$'Pr(>F)'[1],anova(vmh_lm_meanT)$'Pr(>F)'[1], anova(vmh_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_vmh$pvalue[results_vmh$gene==gene & results_vmh$Test=="Status"], results_vmh$pvalue[results_vmh$gene==gene & results_vmh$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()
```

Rubbish

#### AVPR1A

```{r VMH AIC AVPR1A}
gene="AVPR1A"
design="~ Batch + Year + Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(vmh_norm, aes(x=mean_T, y=AVPR1A, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
a
```

Rubbish plot, and requires batch + year parameters to be significant. OVERPARAMETERIZATION. 

#### PGR

```{r VMH AIC PGR}
gene="PGR"
design="~ Batch + Year + Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(vmh_norm, aes(x=mean_T, y=PGR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
a
```

Rubbish plot, and requires batch + year parameters to be significant. OVERPARAMETERIZATION.


## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

I am excluding AR on account of low Rsq,GNRH1, AVP and OXT on account of influential observations and overparameterization.

```{r}
results_vmh2$keep<- ifelse(grepl("GNRH1|OXT$|AVP$|AR|PGR|AVPR1A", results_vmh2$gene),"no","yes")

```

## Gene Plots

```{r VMH lm}

lm_out<- list()
cont_batch<- "Batch"
cont_batch2<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_vmh2)){
  #g=5
  gene<- results_vmh2$gene[g]
  gene<- as.character(gene)
  test<- word(results_vmh2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(vmh_norm[,gene] ~ Batch + Year + Status * mean_T, data=vmh_norm)

  out<- data.frame(lm_model= " ~ Batch + Year + Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[5]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(vmh_norm[,gene] ~ vmh_norm[,cont_batch] + vmh_norm[,cont_batch2], data=vmh_norm))$coefficients[11]
    year_p<- summary(lm(vmh_norm[,gene] ~ vmh_norm[,cont_batch] + vmh_norm[,cont_batch2], data=vmh_norm))$coefficients[12]
    if(batch_p<0.05 | year_p<0.05){
      lm_cont<- lm(vmh_norm[,gene] ~ vmh_norm[,cont_batch] + vmh_norm[,cont_batch2] + vmh_norm[,test], data=vmh_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",cont_batch2," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[3]
      out$rsq<-  signif(cor(vmh_norm[,test], vmh_norm[,gene])^2,2)
      
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(vmh_norm[,gene] ~ vmh_norm[,test], data=vmh_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(vmh_norm[,test], vmh_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(vmh_cand))

results_vmh3<- cbind(results_vmh2, lm_out[,-4])
final_res[[tissue]]<- results_vmh3
```



```{r VMH plotting}
plots<- list()

for(g in 1:nrow(results_vmh3)){
  #g=1
  gene<- results_vmh3$gene[g]
  gene<- as.character(gene)
  test<- word(results_vmh3$design[g],-1)
  kp<- ifelse(results_vmh3$keep[g]=="yes","*","")
  des<- results_vmh3$design[g]
  
  lfc<- signif(results_vmh3$log2FoldChange[g],2)
  praw<- signif(results_vmh3$pvalue[g],2)
  padj=signif(results_vmh3$padj[g],2)
  Rsq=signif(results_vmh3$rsq[g],2)
  #lm_model<- results_vmh3$lm_model[g]
  #lm_p<-signif(results_vmh3$lm_p[g],2)
  #lm_q<- signif(results_vmh3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(vmh_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(vmh_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(vmh_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:3)], ncol=2, nrow=3, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/VMH_significant.png", figure,device="png",height=150, width=100, units="mm", bg="white")

final_res[[tissue]]<- results_vmh3
```


# Anterior Hypothalamus (AH)

```{r AH prep}
rm(list= ls()[!(ls() %in% keep)])

tissue="AH"
ah_key<- subset(key_behav, Tissue=="AH")
ah_key<- droplevels(ah_key)


ah_norm<- norm_data[[1]]
rownames(ah_norm)<- ah_norm$X
ah_norm$X<- NULL
ah_norm<- as.data.frame(t(ah_norm))
ah_norm<- ah_norm[,colnames(ah_norm) %in% candidates2]
ah_norm$sampleID<- rownames(ah_norm)
ah_norm<- merge(ah_norm, ah_key, by="sampleID")
```


```{r AH initial results}

results_ah<- results_data[which(results_data$Tissue==tissue),]
#results_ah<- results_ah[results_ah$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_ah2<- results_ah[which(results_ah$pvalue<0.05),]
results_ah2<- results_ah2[order(results_ah2$gene),]
results_ah2<- rbind(subset(results_int_sub2,select=-n), subset(results_ah2, select=-result))
#results_ah2$padj_cand<- p.adjust(results_ah2$pvalue, method="BH", n=length(ah_cand))

knitr::kable(results_ah2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()


results_ah2$keep<- "no"
```

In the latest revision, I realized there was a mistake in the code for AH so the results are slightly different for this nucleus and then this means we have a significant gene! But still no interaction effects.


## Gene Plots


```{r AH lm}


lm_out<- list()
#cont_batch<- "Batch"

status_data<- ah_norm
gene="GNRH1"
test="Status"

        lm_status<- lm(status_data[,gene] ~  status_data[,test], data=status_data)
        lm_out<- data.frame(lm_model=paste0(" ~ ",test))
        lm_out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        lm_out$rsq<- NA
        lm_out$gene<- gene
    

#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(ah_cand))

results_ah3<- cbind(results_ah2, lm_out[,-4])
final_res[[tissue]]<- results_ah3
```



```{r AH plotting}
plots<- list()


  g=1
  gene<- results_ah3$gene[g]
  gene<- as.character(gene)
  test<- word(results_ah3$design[g],-1)
  kp<- ifelse(results_ah3$keep[g]=="yes","*","")
  des<- results_ah3$design[g]
  
  lfc<- signif(results_ah3$log2FoldChange[g],2)
  praw<- signif(results_ah3$pvalue[g],2)
  padj=signif(results_ah3$padj[g],2)
  Rsq=signif(results_ah3$rsq[g],2)
  #lm_model<- results_ah3$lm_model[g]
  #lm_p<-signif(results_ah3$lm_p[g],2)
  #lm_q<- signif(results_ah3$lm_q[g],2)
  
  
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="Normalized Counts", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
 
ggsave(filename="../Candidate_gene_results/AH_significant.png", p,device="png",height=50, width=70, units="mm", bg="white")

```


# Paraventricular Nucleus (PVN) 


```{r PVN prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="PVN"
pvn_key<- subset(key_behav, Tissue=="PVN")
pvn_key<- droplevels(pvn_key)


pvn_norm<- norm_data[[10]]
rownames(pvn_norm)<- pvn_norm$X
pvn_norm$X<- NULL
pvn_norm<- as.data.frame(t(pvn_norm))
pvn_norm<- pvn_norm[,colnames(pvn_norm) %in% candidates2]
pvn_norm$sampleID<- rownames(pvn_norm)
pvn_norm<- merge(pvn_norm, pvn_key, by="sampleID")
```

```{r pvn initial results}
## Extract pvn results

results_pvn<- results_data[results_data$Tissue==tissue,]
#results_pvn<- results_pvn[results_pvn$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_pvn2<- results_pvn[which(results_pvn$pvalue<0.05),]
results_pvn2<- results_pvn2[order(results_pvn2$gene),]
results_pvn2<- rbind(subset(results_int_sub2,select=-n), subset(results_pvn2, select=-result))
#results_pvn2$padj_cand<- p.adjust(results_pvn2$pvalue, method="BH", n=length(pvn_cand))

knitr::kable(results_pvn2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()
```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

There were no relevant batch variables for PVN.

### Alternate models

```{r PVN alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in",tissue)) %>% kable_styling()

```




### AIC

#### AR

```{r PVN AIC AR}
gene="AR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(pvn_norm, aes(x=mean_T, y=AR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pvn_lm_Status<- lm(AR ~ Status, data=pvn_norm)
pvn_lm_meanT<- lm(AR ~ mean_T, data=pvn_norm)
pvn_lm_int<- lm(AR ~ Status*mean_T, data=pvn_norm)

#against the NULL model
pvn_lm_null<- lm(AR ~ 1, data=pvn_norm)
pvn_lm_strength<- lm(AR ~ strength.all_study, data=pvn_norm)

model=c("AR ~ 1", "AR ~ Status", "AR ~ mean T","*AR ~ Status + mean T + Status:meanT", "AR ~ Strength")
AIC=c(AIC(pvn_lm_null), AIC(pvn_lm_Status),AIC(pvn_lm_meanT),AIC(pvn_lm_int), AIC(pvn_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4], AIC[1]-AIC[5])
lm_p<- c(NA,anova(pvn_lm_Status)$'Pr(>F)'[1],anova(pvn_lm_meanT)$'Pr(>F)'[1], anova(pvn_lm_int)$'Pr(>F)'[3],anova(pvn_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="Status"], results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"], results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="Strength"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the pvnads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


```

#### OXT

```{r PVN AIC OXT}
gene="OXT"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



c<- ggplot(pvn_norm, aes(x=mean_T, y=OXT, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pvn_lm_Status<- lm(OXT ~ Status, data=pvn_norm)
pvn_lm_meanT<- lm(OXT ~ mean_T, data=pvn_norm)
pvn_lm_int<- lm(OXT ~ Status*mean_T, data=pvn_norm)
#against the NULL model
pvn_lm_null<- lm(OXT ~ 1, data=pvn_norm)
pvn_lm_strength<- lm(OXT ~ strength.all_study, data=pvn_norm)

model=c("OXT ~ 1", "OXT ~ Status", "OXT ~ mean T","OXT ~ Status + mean T + Status:meanT", "OXT ~ Strength")
AIC=c(AIC(pvn_lm_null), AIC(pvn_lm_Status),AIC(pvn_lm_meanT),AIC(pvn_lm_int), AIC(pvn_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4],AIC[1]-AIC[5])
lm_p<- c(NA,anova(pvn_lm_Status)$'Pr(>F)'[1],anova(pvn_lm_meanT)$'Pr(>F)'[1], anova(pvn_lm_int)$'Pr(>F)'[3],anova(pvn_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="Status"], results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"], results_pvn$pvalue[results_pvn$gene==gene & results_pvn$Test=="Strength"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d, labels=c("A","B","C","D"), nrow=2,ncol=2)
figure

ggsave(filename="../Candidate_gene_results/PVN_interaction.png", figure,device="png",width=160, height=120, units="mm", dpi=300, bg="white")
```



## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

I am excluding OXT, AVP, 

```{r}
results_pvn2$keep<- ifelse(grepl("OXT$|AVP$|AR", results_pvn2$gene),"no","yes")
results_pvn2$keep<- ifelse((results_pvn2$gene=="AR" & results_pvn2$Test=="LRT_full_interaction"),"yes",as.character(results_pvn2$keep))

```

## Gene Plots


```{r PVN lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Year"
status_data<- pvn_norm

for(g in 1:nrow(results_pvn2)){
  #g=2
  gene<- results_pvn2$gene[g]
  gene<- as.character(gene)
  test<- word(results_pvn2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(pvn_norm[,gene] ~ Status * mean_T, data=pvn_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(pvn_norm[,gene] ~ pvn_norm[,cont_batch], data=pvn_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(pvn_norm[,gene] ~ pvn_norm[,cont_batch] + pvn_norm[,test], data=pvn_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(pvn_norm[,test], pvn_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(pvn_norm[,gene] ~ pvn_norm[,test], data=pvn_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(pvn_norm[,test], pvn_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(pvn_cand))

results_pvn3<- cbind(results_pvn2, lm_out[,-4])
final_res[[tissue]]<- results_pvn3
```



```{r PVN plotting}
plots<- list()

for(g in 1:nrow(results_pvn3)){
  #g=1
  gene<- results_pvn3$gene[g]
  gene<- as.character(gene)
  test<- word(results_pvn3$design[g],-1)
  kp<- ifelse(results_pvn3$keep[g]=="yes","*","")
  des<- results_pvn3$design[g]
  
  lfc<- signif(results_pvn3$log2FoldChange[g],2)
  praw<- signif(results_pvn3$pvalue[g],2)
  padj=signif(results_pvn3$padj[g],2)
  Rsq=signif(results_pvn3$rsq[g],2)
  #lm_model<- results_pvn3$lm_model[g]
  #lm_p<-signif(results_pvn3$lm_p[g],2)
  #lm_q<- signif(results_pvn3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(pvn_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(pvn_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(pvn_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:2)], ncol=2, nrow=4, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/PVN_significant.png", figure,device="png",height=200, width=100, units="mm", bg="white")

```



# Medial Preoptic Area (POM) 


```{r POM prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="POM"
pom_key<- subset(key_behav, Tissue=="POM")
pom_key<- droplevels(pom_key)

outlier<- "PFT2_POM_run1"

pom_key<- pom_key[!rownames(pom_key) %in% outlier,]


pom_data<- data[,colnames(data) %in% rownames(pom_key)]

#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_key, design= ~ Batch + Status)
dd<- DESeq(dd)
gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(pom_key, status_results, by="sampleID")

## data for continuous variables. 
pom_norm<- norm_data[[9]]
rownames(pom_norm)<- pom_norm$X
pom_norm$X<- NULL
pom_norm<- as.data.frame(t(pom_norm))
pom_norm<- pom_norm[,colnames(pom_norm) %in% candidates2]
pom_norm$sampleID<- rownames(pom_norm)
pom_norm<- merge(pom_norm, pom_key, by="sampleID")
```

```{r pom initial results}
## Extract pom results
pom_cand<- candidates2[candidates2 %in% rownames(pom_data)]

results_pom<- results_data[results_data$Tissue==tissue,]
#results_pom<- results_pom[results_pom$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_pom2<- results_pom[which(results_pom$pvalue<0.05),]
results_pom2<- results_pom2[order(results_pom2$gene),]
results_pom2<- rbind(subset(results_int_sub2,select=-n), subset(results_pom2, select=-result))
#results_pom2$padj_cand<- p.adjust(results_pom2$pvalue, method="BH", n=length(pom_cand))

knitr::kable(results_pom2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()
```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r POM batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone! 

### Alternate models

```{r POM alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

VIP still looks good in the reduced model.

### AIC

#### VIP

```{r pom AIC VIP}
gene="VIP"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



a<- ggplot(pom_norm, aes(x=mean_T, y=VIP, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pom_lm_Status<- lm(VIP ~ Status, data=pom_norm)
pom_lm_meanT<- lm(VIP ~ mean_T, data=pom_norm)
pom_lm_int<- lm(VIP ~ Status*mean_T, data=pom_norm)
#against the NULL model
pom_lm_null<- lm(VIP ~ 1, data=pom_norm)

model=c("VIP ~ 1", "VIP ~ Status", "VIP ~ mean T","VIP ~ Status + mean T + Status:mean T")
AIC=c(AIC(pom_lm_null), AIC(pom_lm_Status),AIC(pom_lm_meanT),AIC(pom_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(pom_lm_Status)$'Pr(>F)'[1],anova(pom_lm_meanT)$'Pr(>F)'[1], anova(pom_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="Status"], results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_full_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the pomads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


#figure<- ggarrange(a,b, labels=c("A","B"), nrow=2)
#figure

#ggsave(filename="../Candidate_gene_results/pom_interaction.png", figure,device="png",width=80, height=120, units="mm", dpi=300, bg="white")
```

#### AR

```{r pom AIC AR}
gene="AR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



c<- ggplot(pom_norm, aes(x=mean_T, y=AR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pom_lm_Status<- lm(AR ~ Status, data=pom_norm)
pom_lm_meanT<- lm(AR ~ mean_T, data=pom_norm)
pom_lm_int<- lm(AR ~ Status*mean_T, data=pom_norm)
#against the NULL model
pom_lm_null<- lm(AR ~ 1, data=pom_norm)

model=c("AR ~ 1", "AR ~ Status", "*AR ~ mean T","AR ~ Status * mean T")
AIC=c(AIC(pom_lm_null), AIC(pom_lm_Status),AIC(pom_lm_meanT),AIC(pom_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(pom_lm_Status)$'Pr(>F)'[1],anova(pom_lm_meanT)$'Pr(>F)'[1], anova(pom_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="Status"], results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the pomads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

picking mean T

#### PRLR

```{r pom AIC PRLR}
gene="PRLR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_full_interaction"],2)



e<- ggplot(pom_norm, aes(x=mean_T, y=PRLR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
pom_lm_Status<- lm(PRLR ~ Status, data=pom_norm)
pom_lm_meanT<- lm(PRLR ~ mean_T, data=pom_norm)
pom_lm_int<- lm(PRLR ~ Status*mean_T, data=pom_norm)
#against the NULL model
pom_lm_null<- lm(PRLR ~ 1, data=pom_norm)

model=c("PRLR ~ 1", "PRLR ~ Status", "*PRLR ~ mean T","PRLR ~ Status * mean T")
AIC=c(AIC(pom_lm_null), AIC(pom_lm_Status),AIC(pom_lm_meanT),AIC(pom_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(pom_lm_Status)$'Pr(>F)'[1],anova(pom_lm_meanT)$'Pr(>F)'[1], anova(pom_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ YePRLR
DESeq2_raw_p=c(NA,results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="Status"], results_pom$pvalue[results_pom$gene==gene & results_pom$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

f<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d,e,f, labels=c("A","B","C","D","E","F"), nrow=3,ncol=2)
#figure

ggsave(filename="../Candidate_gene_results/POM_interaction.png", figure,device="png",width=160, height=150, units="mm", dpi=300, bg="white")


```

picking meanT

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.
I'm dropping VIP, AR (Status) + PRLR (Status)

```{r}
results_pom2$keep<- ifelse(grepl("VIP$|AR|PRLR", results_pom2$gene),"no","yes")
results_pom2$keep<- ifelse((results_pom2$gene=="PRLR" & results_pom2$Test=="mean T"),"yes",as.character(results_pom2$keep))
results_pom2$keep<- ifelse((results_pom2$gene=="AR" & results_pom2$Test=="mean T"),"yes",as.character(results_pom2$keep))

```

## Gene Plots


```{r POM lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_pom2)){
  #g=2
  gene<- results_pom2$gene[g]
  gene<- as.character(gene)
  test<- word(results_pom2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(pom_norm[,gene] ~ Status * mean_T, data=pom_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(pom_norm[,gene] ~ pom_norm[,cont_batch], data=pom_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(pom_norm[,gene] ~ pom_norm[,cont_batch] + pom_norm[,test], data=pom_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(pom_norm[,test], pom_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(pom_norm[,gene] ~ pom_norm[,test], data=pom_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(pom_norm[,test], pom_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ", status_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(pom_cand))

results_pom3<- cbind(results_pom2, lm_out[,-4])
final_res[[tissue]]<- results_pom3
```



```{r POM plotting}
plots<- list()

for(g in 1:nrow(results_pom3)){
  #g=1
  gene<- results_pom3$gene[g]
  gene<- as.character(gene)
  test<- word(results_pom3$design[g],-1)
  kp<- ifelse(results_pom3$keep[g]=="yes","*","")
  des<- results_pom3$design[g]
  
  lfc<- signif(results_pom3$log2FoldChange[g],2)
  praw<- signif(results_pom3$pvalue[g],2)
  padj=signif(results_pom3$padj[g],2)
  Rsq=signif(results_pom3$rsq[g],2)
  #lm_model<- results_pom3$lm_model[g]
  #lm_p<-signif(results_pom3$lm_p[g],2)
  #lm_q<- signif(results_pom3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(pom_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(pom_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(pom_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-1], ncol=4, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/POM_significant.png", figure,device="png",height=100, width=200, units="mm", bg="white")

```



# Dorsomedial Intercollicular Nucleus (ICo)


```{r ICO prep}

rm(list= ls()[!(ls() %in% keep)])
tissue="ICO"
ico_key<- subset(key_behav, Tissue=="ICO")
ico_key<- droplevels(ico_key)


ico_data<- data[,colnames(data) %in% rownames(ico_key)]

#remove genes with less than 5 reads
ico_data$avg_count<- apply(ico_data, 1, mean)
ico_data<- ico_data[ico_data$avg_count>5,]
ico_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ico_data$percent_0<- apply(ico_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ico_data)/2
ico_data<- ico_data[ico_data$percent_0<=thresh,]
ico_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_key, design= ~ Batch)
dd<- DESeq(dd)
gd<- counts(dd,normalized=TRUE)

status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(ico_key, status_results, by="sampleID")


## data for continuous variables. 
ico_norm<- norm_data[[6]]
rownames(ico_norm)<- ico_norm$X
ico_norm$X<- NULL
ico_norm<- as.data.frame(t(ico_norm))
ico_norm<- ico_norm[,colnames(ico_norm) %in% candidates2]
ico_norm$sampleID<- rownames(ico_norm)
ico_norm<- merge(ico_norm, ico_key, by="sampleID")
```



```{r ico results initial}
results_ico<- results_data[results_data$Tissue==tissue,]
#results_ico<- results_ico[results_ico$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_ico2<- results_ico[which(results_ico$pvalue<0.05),]
results_ico2<- results_ico2[order(results_ico2$gene),]
results_ico2<- rbind(subset(results_int_sub2,select=-n), subset(results_ico2, select=-result))
#results_ico2$padj_cand<- p.adjust(results_ico2$pvalue, method="BH", n=length(ico_cand))

knitr::kable(results_ico2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()



```


## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r ICO batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone identified as being significant for our interest variables.

### Alternate models

```{r ICO alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

OXTR and AVPR1A are robust to model choice.

### AIC

#### OXTR

```{r ico AIC OXTR}
gene="OXTR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)


a<- ggplot(ico_norm, aes(x=mean_T, y=OXTR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
ico_lm_Status<- lm(OXTR ~ Status, data=ico_norm)
ico_lm_meanT<- lm(OXTR ~ mean_T, data=ico_norm)
ico_lm_int<- lm(OXTR ~ Status*mean_T, data=ico_norm)

#against the NULL model
ico_lm_null<- lm(OXTR ~ 1, data=ico_norm)

model=c("OXTR ~ 1", "OXTR ~ Status", "OXTR ~ mean T","OXTR ~ Status + meanT + Status:meanT")
AIC=c(AIC(ico_lm_null), AIC(ico_lm_Status),AIC(ico_lm_meanT),AIC(ico_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(ico_lm_Status)$'Pr(>F)'[1],anova(ico_lm_meanT)$'Pr(>F)'[1], anova(ico_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ico$pvalue[results_ico$gene==gene & results_ico$Test=="Status"], results_ico$pvalue[results_ico$gene==gene & results_ico$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_reduced_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


#figure<- ggarrange(a,b, labels=c("A","B"), nrow=2)
#figure

#ggsave(filename="../Candidate_gene_results/ico_interaction.png", figure,device="png",width=80, height=120, units="mm", dpi=300, bg="white")
```

Influential observations

#### AVPR1A

```{r ico AIC AVPR1A}
gene="AVPR1A"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



c<- ggplot(ico_norm, aes(x=mean_T, y=AVPR1A, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
ico_lm_Status<- lm(AVPR1A ~ Status, data=ico_norm)
ico_lm_meanT<- lm(AVPR1A ~ mean_T, data=ico_norm)
ico_lm_int<- lm(AVPR1A ~ Status*mean_T, data=ico_norm)

#against the NULL model
ico_lm_null<- lm(AVPR1A ~ 1, data=ico_norm)

model=c("AVPR1A ~ 1", "AVPR1A ~ Status", "AVPR1A ~ mean T","AVPR1A ~ Status + meanT + Status:meanT")
AIC=c(AIC(ico_lm_null), AIC(ico_lm_Status),AIC(ico_lm_meanT),AIC(ico_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(ico_lm_Status)$'Pr(>F)'[1],anova(ico_lm_meanT)$'Pr(>F)'[1], anova(ico_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ico$pvalue[results_ico$gene==gene & results_ico$Test=="Status"], results_ico$pvalue[results_ico$gene==gene & results_ico$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_reduced_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the icoads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d, labels=c("A","B","C","D"), nrow=2,ncol=2)
figure

ggsave(filename="../Candidate_gene_results/ICO_interaction.png", figure,device="png",width=160, height=100, units="mm", dpi=300, bg="white")
```

## Final Res

Reject ESR2 because it was sensitive to batch parameters. OXTR because of the influential observations and AVPR1A because it didn't make the AIC test.

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

```{r}
results_ico2$keep<- ifelse(grepl("OXTR|ESR2|AVPR1A", results_ico2$gene), "no","yes")

```

## Gene Plots


```{r ICO lm}

lm_out<- list()
cont_batch<- "Batch"
cont_batch2<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_ico2)){
  #g=2
  gene<- results_ico2$gene[g]
  gene<- as.character(gene)
  test<- word(results_ico2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(ico_norm[,gene] ~ Status * mean_T, data=ico_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(ico_norm[,gene] ~ ico_norm[,cont_batch] + ico_norm[,cont_batch2], data=ico_norm))$coefficients[11]
    year_p<- summary(lm(ico_norm[,gene] ~ ico_norm[,cont_batch] + ico_norm[,cont_batch2], data=ico_norm))$coefficients[12]
    if(batch_p<0.05 | year_p<0.05){
      lm_cont<- lm(ico_norm[,gene] ~ ico_norm[,cont_batch] + ico_norm[,cont_batch2] + ico_norm[,test], data=ico_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[3]
      out$rsq<-  signif(cor(ico_norm[,test], ico_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(ico_norm[,gene] ~ ico_norm[,test], data=ico_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(ico_norm[,test], ico_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(ico_cand))

results_ico3<- cbind(results_ico2, lm_out[,-4])
final_res[[tissue]]<- results_ico3
```


# Midbrain Central Gray (GCT)


```{r GCT prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="GCT"
gct_key<- subset(key_behav, Tissue=="GCT")
gct_key<- droplevels(gct_key)


gct_data<- data[,colnames(data) %in% rownames(gct_key)]

#remove genes with less than 5 reads
gct_data$avg_count<- apply(gct_data, 1, mean)
gct_data<- gct_data[gct_data$avg_count>5,]
gct_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gct_data$percent_0<- apply(gct_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gct_data)/2
gct_data<- gct_data[gct_data$percent_0<=thresh,]
gct_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_key, design= ~ Batch)
dd<- DESeq(dd)

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(gct_key, status_results, by="sampleID")

## data for continuous variables. 
gct_norm<- norm_data[[4]]
rownames(gct_norm)<- gct_norm$X
gct_norm$X<- NULL
gct_norm<- as.data.frame(t(gct_norm))
gct_norm<- gct_norm[,colnames(gct_norm) %in% candidates2]
gct_norm$sampleID<- rownames(gct_norm)
gct_norm<- merge(gct_norm, gct_key, by="sampleID")
```


```{r GCT results initial}
results_gct<- results_data[results_data$Tissue==tissue,]
#results_gct<- results_gct[results_gct$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_gct2<- results_gct[which(results_gct$pvalue<0.05),]
results_gct2<- results_gct2[order(results_gct2$gene),]
results_gct2<- rbind(subset(results_int_sub2,select=-n), subset(results_gct2, select=-result))
#results_gct2$padj_cand<- p.adjust(results_gct2$pvalue, method="BH", n=length(gct_cand))

knitr::kable(results_gct2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()


```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r GCT batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch isn't important for VIP, ESR1, PGR, PRLR. 

### Alternate models

```{r GCT alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

All are robust to batch parameter.

### AIC

#### VIP

```{r GCT AIC VIP}
gene="VIP"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



a<- ggplot(gct_norm, aes(x=mean_T, y=VIP, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
a 
gct_lm_Status<- lm(VIP ~ Status, data=gct_norm)
gct_lm_meanT<- lm(VIP ~ mean_T, data=gct_norm)
gct_lm_int<- lm(VIP ~ Status*mean_T, data=gct_norm)
#against the NULL model
gct_lm_null<- lm(VIP ~ 1, data=gct_norm)
gct_lm_strength<- lm(VIP ~ strength.all_study, data=gct_norm)

model=c("VIP ~ 1", "VIP ~ Status", "VIP ~ mean T","*VIP ~ Status + mean T + Status:meanT", "VIP ~ Strength")
AIC=c(AIC(gct_lm_null), AIC(gct_lm_Status),AIC(gct_lm_meanT),AIC(gct_lm_int), AIC(gct_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4],AIC[1]-AIC[5])
lm_p<- c(NA,anova(gct_lm_Status)$'Pr(>F)'[1],anova(gct_lm_meanT)$'Pr(>F)'[1], anova(gct_lm_int)$'Pr(>F)'[3],anova(gct_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Status"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_reduced_interaction"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Strength"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the gctads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


```

Keep VIP

#### PGR


```{r GCT AIC PGR}
gene="PGR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



c<- ggplot(gct_norm, aes(x=mean_T, y=PGR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
 c 
gct_lm_Status<- lm(PGR ~ Status, data=gct_norm)
gct_lm_meanT<- lm(PGR ~ mean_T, data=gct_norm)
gct_lm_int<- lm(PGR ~ Status*mean_T, data=gct_norm)
#against the NULL model
gct_lm_null<- lm(PGR ~ 1, data=gct_norm)
gct_lm_strength<- lm(PGR ~ strength.all_study, data=gct_norm)

model=c("PGR ~ 1", "PGR ~ Status", "PGR ~ mean T","*PGR ~ Status + mean T + Status:meanT", "PGR ~ Strength")
AIC=c(AIC(gct_lm_null), AIC(gct_lm_Status),AIC(gct_lm_meanT),AIC(gct_lm_int), AIC(gct_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4],AIC[1]-AIC[5])
lm_p<- c(NA,anova(gct_lm_Status)$'Pr(>F)'[1],anova(gct_lm_meanT)$'Pr(>F)'[1], anova(gct_lm_int)$'Pr(>F)'[3],anova(gct_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Status"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_reduced_interaction"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Strength"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

Keep PGR.

#### ESR1

```{r GCT AIC ESR1}
gene="ESR1"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



e<- ggplot(gct_norm, aes(x=mean_T, y=ESR1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  e
gct_lm_Status<- lm(ESR1 ~ Status, data=gct_norm)
gct_lm_meanT<- lm(ESR1 ~ mean_T, data=gct_norm)
gct_lm_int<- lm(ESR1 ~ Status*mean_T, data=gct_norm)
#against the NULL model

gct_lm_null<- lm(ESR1 ~ 1, data=gct_norm)


model=c("ESR1 ~ 1", "ESR1 ~ Status", "ESR1 ~ mean T","*ESR1 ~ Status + mean T + Status:mean T")
AIC=c(AIC(gct_lm_null), AIC(gct_lm_Status),AIC(gct_lm_meanT),AIC(gct_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(gct_lm_Status)$'Pr(>F)'[1],anova(gct_lm_meanT)$'Pr(>F)'[1], anova(gct_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Status"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the gctads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

f<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))



```

Keep ESR1

#### PRLR


```{r GCT AIC PRLR}
gene="PRLR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



g<- ggplot(gct_norm, aes(x=mean_T, y=PRLR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
gct_lm_Status<- lm(PRLR ~ Status, data=gct_norm)
gct_lm_meanT<- lm(PRLR ~ mean_T, data=gct_norm)
gct_lm_int<- lm(PRLR ~ Status*mean_T, data=gct_norm)


#against the NULL model

gct_lm_null<- lm(PRLR ~ 1, data=gct_norm)


model=c("PRLR ~ 1", "PRLR ~ Status", "PRLR ~ mean T","PRLR ~ Status + mean T + Status:mean T")
AIC=c(AIC(gct_lm_null), AIC(gct_lm_Status),AIC(gct_lm_meanT),AIC(gct_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(gct_lm_Status)$'Pr(>F)'[1],anova(gct_lm_meanT)$'Pr(>F)'[1], anova(gct_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="Status"], results_gct$pvalue[results_gct$gene==gene & results_gct$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the gctads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

h<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d,e,f,g,h, labels=c("A","B","C","D","E","F","G","H"), nrow=4,ncol=2, common.legend = TRUE)
#figure

ggsave(filename="../Candidate_gene_results/GCT_interaction.png", figure,device="png",width=160, height=250, units="mm", dpi=300, bg="white")
```


Pass all except PRLR

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

Exclude PRLR, AR strength, and PGR in mean T and strength and VIP in status, mean T and Strength, OXT due to outliers

```{r}
results_gct2$keep<- ifelse(grepl("PRLR|AR|VIP$|PGR|OXT$", results_gct2$gene),"no","yes")
results_gct2$keep<- ifelse((results_gct2$gene=="VIP" & grepl("Status:mean_T", results_gct2$design)), "yes",as.character(results_gct2$keep))
results_gct2$keep<- ifelse((results_gct2$gene=="PGR" & grepl("Status:mean_T", results_gct2$design)), "yes",as.character(results_gct2$keep))

```


## Gene Plots


```{r GCT lm}

lm_out<- list()
cont_batch<- "Batch"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_gct2)){
  #g=2
  gene<- results_gct2$gene[g]
  gene<- as.character(gene)
  test<- word(results_gct2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(gct_norm[,gene] ~ Status * mean_T, data=gct_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(gct_norm[,gene] ~ gct_norm[,cont_batch], data=gct_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(gct_norm[,gene] ~ gct_norm[,cont_batch] + gct_norm[,test], data=gct_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(gct_norm[,test], gct_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(gct_norm[,gene] ~ gct_norm[,test], data=gct_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(gct_norm[,test], gct_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(gct_cand))

results_gct3<- cbind(results_gct2, lm_out[,-4])
final_res[[tissue]]<- results_gct3
```




```{r GCT plotting}
plots<- list()

for(g in 1:nrow(results_gct3)){
  #g=1
  gene<- results_gct3$gene[g]
  gene<- as.character(gene)
  test<- word(results_gct3$design[g],-1)
  kp<- ifelse(results_gct3$kp[g]=="yes","*","")
  des<- results_gct3$design[g]
  
  lfc<- signif(results_gct3$log2FoldChange[g],2)
  praw<- signif(results_gct3$pvalue[g],2)
  padj=signif(results_gct3$padj[g],2)
  Rsq=signif(results_gct3$rsq[g],2)
  #lm_model<- results_gct3$lm_model[g]
  #lm_p<-signif(results_gct3$lm_p[g],2)
  #lm_q<- signif(results_gct3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(gct_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(gct_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(gct_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:4)], ncol=2, nrow=4, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/GCT_significant.png", figure,device="png",height=200, width=100, units="mm", bg="white")

```


# Nucleus Taenia (TnA)


```{r TNA prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="TNA"
tna_key<- subset(key_behav, Tissue=="TNA")
tna_key<- droplevels(tna_key)


tna_data<- data[,colnames(data) %in% rownames(tna_key)]

#remove genes with less than 5 reads
tna_data$avg_count<- apply(tna_data, 1, mean)
tna_data<- tna_data[tna_data$avg_count>5,]
tna_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
tna_data$percent_0<- apply(tna_data, 1, function(x)length(x[x==0]))
thresh<- ncol(tna_data)/2
tna_data<- tna_data[tna_data$percent_0<=thresh,]
tna_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_key, design= ~ Batch)
dd<- DESeq(dd)

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(tna_key, status_results, by="sampleID")


## data for continuous variables. 
tna_norm<- norm_data[[11]]
rownames(tna_norm)<- tna_norm$X
tna_norm$X<- NULL
tna_norm<- as.data.frame(t(tna_norm))
tna_norm<- tna_norm[,colnames(tna_norm) %in% candidates2]
tna_norm$sampleID<- rownames(tna_norm)
tna_norm<- merge(tna_norm, tna_key, by="sampleID")
```

```{r TNA results intial}
results_tna<- results_data[results_data$Tissue==tissue,]
#results_tna<- results_tna[results_tna$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_tna2<- results_tna[which(results_tna$pvalue<0.05),]
results_tna2<- results_tna2[order(results_tna2$gene),]
results_tna2<- rbind(subset(results_int_sub2,select=-n), subset(results_tna2, select=-result))
#results_tna2$padj_cand<- p.adjust(results_tna2$pvalue, method="BH", n=length(tna_cand))
knitr::kable(results_tna2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()

```


## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r TNA batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for any of the significant interaction models

### Alternate models

```{r TNA alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

All the interaction genes were stable in the reduced model.

PRLR, VIP VIPR1, CYP19A1.

### AIC

#### CYP19A1

```{r TNA AIC CYP19A1}
gene="CYP19A1"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



a<- ggplot(tna_norm, aes(x=mean_T, y=CYP19A1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
tna_lm_Status<- lm(CYP19A1 ~ Status, data=tna_norm)
tna_lm_meanT<- lm(CYP19A1 ~ mean_T, data=tna_norm)
tna_lm_int<- lm(CYP19A1 ~ Status*mean_T, data=tna_norm)

#against the NULL model

tna_lm_null<- lm(CYP19A1 ~ 1, data=tna_norm)


model=c("CYP19A1 ~ 1", "CYP19A1 ~ Status", "CYP19A1 ~ mean T","CYP19A1 ~ Status + mean T + Status:mean T")
AIC=c(AIC(tna_lm_null), AIC(tna_lm_Status),AIC(tna_lm_meanT),AIC(tna_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(tna_lm_Status)$'Pr(>F)'[1],anova(tna_lm_meanT)$'Pr(>F)'[1], anova(tna_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="Status"], results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])





knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the tnaads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

Does not pass AIC test.

#### PRLR

```{r TNA AIC PRLR}
gene="PRLR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



c<- ggplot(tna_norm, aes(x=mean_T, y=PRLR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
tna_lm_Status<- lm(PRLR ~ Status, data=tna_norm)
tna_lm_meanT<- lm(PRLR ~ mean_T, data=tna_norm)
tna_lm_int<- lm(PRLR ~ Status*mean_T, data=tna_norm)


#against the NULL model

tna_lm_null<- lm(PRLR ~ 1, data=tna_norm)


model=c("PRLR ~ 1", "PRLR ~ Status", "PRLR ~ mean T","PRLR ~ Status + mean T + Status:mean T")
AIC=c(AIC(tna_lm_null), AIC(tna_lm_Status),AIC(tna_lm_meanT),AIC(tna_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(tna_lm_Status)$'Pr(>F)'[1],anova(tna_lm_meanT)$'Pr(>F)'[1], anova(tna_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="Status"], results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the tnaads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

Reject PRLR due to not sig in lm

#### VIP

```{r TNA AIC VIP}
gene="VIP"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



e<- ggplot(tna_norm, aes(x=mean_T, y=VIP, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
tna_lm_Status<- lm(VIP ~ Status, data=tna_norm)
tna_lm_meanT<- lm(VIP ~ mean_T, data=tna_norm)
tna_lm_int<- lm(VIP ~ Status*mean_T, data=tna_norm)

#against the NULL model

tna_lm_null<- lm(VIP ~ 1, data=tna_norm)


model=c("VIP ~ 1", "VIP ~ Status", "VIP ~ mean T","VIP ~ Status + mean T + Status:mean T")
AIC=c(AIC(tna_lm_null), AIC(tna_lm_Status),AIC(tna_lm_meanT),AIC(tna_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(tna_lm_Status)$'Pr(>F)'[1],anova(tna_lm_meanT)$'Pr(>F)'[1], anova(tna_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="Status"], results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])




knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in the tnaads for ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

f<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

Reject VIP

#### VIPR1

```{r TNA AIC VIPR1}
gene="VIPR1"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design & results_int_sub$Test=="LRT_reduced_interaction"],2)



g<- ggplot(tna_norm, aes(x=mean_T, y=VIPR1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
tna_lm_Status<- lm(VIPR1 ~ Status, data=tna_norm)
tna_lm_meanT<- lm(VIPR1 ~ mean_T, data=tna_norm)
tna_lm_int<- lm(VIPR1 ~ Status*mean_T, data=tna_norm)

#against the NULL model

tna_lm_null<- lm(VIPR1 ~ 1, data=tna_norm)


model=c("VIPR1 ~ 1", "VIPR1 ~ Status", "VIPR1 ~ mean T","VIPR1 ~ Status + mean T + Status:mean T")
AIC=c(AIC(tna_lm_null), AIC(tna_lm_Status),AIC(tna_lm_meanT),AIC(tna_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(tna_lm_Status)$'Pr(>F)'[1],anova(tna_lm_meanT)$'Pr(>F)'[1], anova(tna_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="Status"], results_tna$pvalue[results_tna$gene==gene & results_tna$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene  & results_int_sub$Test=="LRT_reduced_interaction"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

h<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d,e,f,g,h, labels=c("A","B","C","D","E","F","G","H"), nrow=4,ncol=2, common.legend = TRUE)
figure

ggsave(filename="../Candidate_gene_results/TNA_interaction.png", figure,device="png",width=160, height=250, units="mm", dpi=300, bg="white")
```

Reject VIPR1.

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

Reject VIP, CYP19A1 due to not passing the AIC tests, AR & AVP due to influential observations and PGR due to low R^2.

```{r}
results_tna2$keep<- ifelse(grepl("VIP|CYP19A1|AR|AVP|PGR|PRLR", results_tna2$gene),"no","yes")
```



## Gene Plots


```{r TNA lm}

lm_out<- list()
cont_batch<- "Batch"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_tna2)){
  #g=2
  gene<- results_tna2$gene[g]
  gene<- as.character(gene)
  test<- word(results_tna2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(tna_norm[,gene] ~ Status * mean_T, data=tna_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(tna_norm[,gene] ~ tna_norm[,cont_batch], data=tna_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(tna_norm[,gene] ~ tna_norm[,cont_batch] + tna_norm[,test], data=tna_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(tna_norm[,test], tna_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(tna_norm[,gene] ~ tna_norm[,test], data=tna_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(tna_norm[,test], tna_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(tna_cand))

results_tna3<- cbind(results_tna2, lm_out[,-4])
final_res[[tissue]]<- results_tna3
```


```{r TNA plotting}
plots<- list()

for(g in 1:nrow(results_tna3)){
  #g=1
  gene<- results_tna3$gene[g]
  gene<- as.character(gene)
  test<- word(results_tna3$design[g],-1)
  kp<- ifelse(results_tna3$keep[g]=="yes","*","")
  des<- results_tna3$design[g]
  
  lfc<- signif(results_tna3$log2FoldChange[g],2)
  praw<- signif(results_tna3$pvalue[g],2)
  padj=signif(results_tna3$padj[g],2)
  Rsq=signif(results_tna3$rsq[g],2)
  #lm_model<- results_tna3$lm_model[g]
  #lm_p<-signif(results_tna3$lm_p[g],2)
  #lm_q<- signif(results_tna3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(tna_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(tna_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(tna_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:4)], ncol=2, nrow=3, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/TNA_significant.png", figure,device="png",height=150, width=100, units="mm", bg="white")

```

# Arcopallium Intermedium (Ai)

```{r AI prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="AI"
ai_key<- subset(key_behav, Tissue=="AI")
ai_key<- droplevels(ai_key)


ai_data<- data[,colnames(data) %in% rownames(ai_key)]

#remove genes with less than 5 reads
ai_data$avg_count<- apply(ai_data, 1, mean)
ai_data<- ai_data[ai_data$avg_count>5,]
ai_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ai_data$percent_0<- apply(ai_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ai_data)/2
ai_data<- ai_data[ai_data$percent_0<=thresh,]
ai_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_key, design= ~ Batch)
dd<- DESeq(dd)

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(ai_key, status_results, by="sampleID")


## data for continuous variables. 
ai_norm<- norm_data[[2]]
rownames(ai_norm)<- ai_norm$X
ai_norm$X<- NULL
ai_norm<- as.data.frame(t(ai_norm))
ai_norm<- ai_norm[,colnames(ai_norm) %in% candidates2]
ai_norm$sampleID<- rownames(ai_norm)
ai_norm<- merge(ai_norm, ai_key, by="sampleID")
```

```{r results initial}

results_ai<- results_data[results_data$Tissue==tissue,]
#results_ai<- results_ai[results_ai$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_ai2<- results_ai[which(results_ai$pvalue<0.05),]
results_ai2<- results_ai2[order(results_ai2$gene),]
results_ai2<- rbind(subset(results_int_sub2,select=-n), subset(results_ai2, select=-result))
#results_ai2$padj_cand<- p.adjust(results_ai2$pvalue, method="BH", n=length(ai_cand))

knitr::kable(results_ai2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()


```

## Exploring status specific expression


### Batch

```{r AI batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone! 

### Alternate models

```{r AI alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

VIPR is stable to model parameterization.  


### AIC

#### VIPR1

```{r AI AIC VIPR1}
gene="VIPR1"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



a<- ggplot(ai_norm, aes(x=mean_T, y=VIPR1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
ai_lm_Status<- lm(VIPR1 ~ Status, data=ai_norm)
ai_lm_meanT<- lm(VIPR1 ~ mean_T, data=ai_norm)
ai_lm_int<- lm(VIPR1 ~ Status*mean_T, data=ai_norm)

#against the NULL model
ai_lm_null<- lm(VIPR1 ~ 1, data=ai_norm)

model=c("VIPR1 ~ 1", "VIPR1 ~ Status", "VIPR1 ~ mean T","VIPR1 ~ Status + meanT + Status:meanT")
AIC=c(AIC(ai_lm_null), AIC(ai_lm_Status),AIC(ai_lm_meanT),AIC(ai_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(ai_lm_Status)$'Pr(>F)'[1],anova(ai_lm_meanT)$'Pr(>F)'[1], anova(ai_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="Status"], results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


```

Passed AIC, and lm, but really hinges on a single observation due to reduced sample size in Ai.

#### AVP

```{r AI AIC AVP}
gene="AVP"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



c<- ggplot(ai_norm, aes(x=mean_T, y=AVP, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
ai_lm_Status<- lm(AVP ~ Status, data=ai_norm)
ai_lm_meanT<- lm(AVP ~ mean_T, data=ai_norm)
ai_lm_int<- lm(AVP ~ Status*mean_T, data=ai_norm)

#against the NULL model
ai_lm_null<- lm(AVP ~ 1, data=ai_norm)

model=c("AVP ~ 1", "AVP ~ Status", "AVP ~ mean T","AVP ~ Status + meanT + Status:meanT")
AIC=c(AIC(ai_lm_null), AIC(ai_lm_Status),AIC(ai_lm_meanT),AIC(ai_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(ai_lm_Status)$'Pr(>F)'[1],anova(ai_lm_meanT)$'Pr(>F)'[1], anova(ai_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="Status"], results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


```

FAIL

#### VIP

distinguish between mean_T and strength

```{r AI AIC vIP}
gene="VIP"
design="~ strength.all_study"

ai_lm_Status<- lm(VIP ~ Status, data=ai_norm)
ai_lm_meanT<- lm(VIP ~ mean_T, data=ai_norm)
ai_lm_int<- lm(VIP ~ Status*mean_T, data=ai_norm)
ai_lm_strength<- lm(VIP ~ strength.all_study, data=ai_norm)
#against the NULL model
ai_lm_null<- lm(VIP ~ 1, data=ai_norm)

model=c("VIP ~ 1", "VIP ~ Status", "?VIP ~ mean T","VIP ~ Status + meanT + Status:meanT", "?VIP ~ Strength")
AIC=c(AIC(ai_lm_null), AIC(ai_lm_Status),AIC(ai_lm_meanT),AIC(ai_lm_int), AIC(ai_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4], AIC[1]-AIC[5])
lm_p<- c(NA,anova(ai_lm_Status)$'Pr(>F)'[1],anova(ai_lm_meanT)$'Pr(>F)'[1], anova(ai_lm_int)$'Pr(>F)'[3],anova(ai_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="Status"], results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$Test=="LRT_full_interaction"], results_ai$pvalue[results_ai$gene==gene & results_ai$Test=="Strength"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

e<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d,e, common.legend = TRUE, labels=c("A","B","C","D","E"), ncol=2, nrow=3)
figure
ggsave(filename="../Candidate_gene_results/AI_interaction.png", figure,device="png",width=160, height=200, units="mm", dpi=300, bg="white")
```

cannot distinguish between them. keep both.

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.
Exclude VIPR1,AVP, and PRLR

```{r}

results_ai2$keep<- ifelse(grepl("AVP$|VIPR1|PRLR|AVPR1A",results_ai2$gene),"no","yes")


```

## Gene Plots


```{r AI lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_ai2)){
  #g=2
  gene<- results_ai2$gene[g]
  gene<- as.character(gene)
  test<- word(results_ai2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(ai_norm[,gene] ~ Status * mean_T, data=ai_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(ai_norm[,gene] ~ ai_norm[,cont_batch], data=ai_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(ai_norm[,gene] ~ ai_norm[,cont_batch] + ai_norm[,test], data=ai_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(ai_norm[,test], ai_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(ai_norm[,gene] ~ ai_norm[,test], data=ai_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(ai_norm[,test], ai_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(ai_cand))

results_ai3<- cbind(results_ai2, lm_out[,-4])
final_res[[tissue]]<- results_ai3
```

```{r AI plotting}
plots<- list()

for(g in 1:nrow(results_ai3)){
  #g=1
  gene<- results_ai3$gene[g]
  gene<- as.character(gene)
  test<- word(results_ai3$design[g],-1)
  kp<- ifelse(results_ai3$keep[g]=="yes","*","")
  des<- results_ai3$design[g]
  
  lfc<- signif(results_ai3$log2FoldChange[g],2)
  praw<- signif(results_ai3$pvalue[g],2)
  padj=signif(results_ai3$padj[g],2)
  Rsq=signif(results_ai3$rsq[g],2)
  #lm_model<- results_ai3$lm_model[g]
  #lm_p<-signif(results_ai3$lm_p[g],2)
  #lm_q<- signif(results_ai3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(ai_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(ai_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(ai_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:2)], ncol=2, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/AI_significant.png", figure,device="png",height=100, width=100, units="mm", bg="white")

```




# Lateral Septum (LS)



```{r LS prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="LS"
ls_key<- subset(key_behav, Tissue=="LS")
ls_key<- droplevels(ls_key)


ls_data<- data[,colnames(data) %in% rownames(ls_key)]

#remove genes with less than 5 reads
ls_data$avg_count<- apply(ls_data, 1, mean)
ls_data<- ls_data[ls_data$avg_count>5,]
ls_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ls_data$percent_0<- apply(ls_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ls_data)/2
ls_data<- ls_data[ls_data$percent_0<=thresh,]
ls_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_key, design= ~ Batch)
dd<- DESeq(dd)
gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(ls_key, status_results, by="sampleID")


## data for continuous variables. 
ls_norm<- norm_data[[7]]
rownames(ls_norm)<- ls_norm$X
ls_norm$X<- NULL
ls_norm<- as.data.frame(t(ls_norm))
ls_norm<- ls_norm[,colnames(ls_norm) %in% candidates2]
ls_norm$sampleID<- rownames(ls_norm)
ls_norm<- merge(ls_norm, ls_key, by="sampleID")
```



```{r LS results init}
results_ls<- results_data[results_data$Tissue==tissue,]
#results_ls<- results_ls[results_ls$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_ls2<- results_ls[which(results_ls$pvalue<0.05),]
results_ls2<- results_ls2[order(results_ls2$gene),]
results_ls2<- rbind(subset(results_int_sub2,select=-n), subset(results_ls2, select=-result))
#results_ls2$padj_cand<- p.adjust(results_ls2$pvalue, method="BH", n=length(ls_cand))

knitr::kable(results_ls2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()

```

## Exploring status specific expression


### Batch

```{r LS batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone! 

### Alternate models

```{r LS alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()
```

AR was stable under a 1 parameter reduced model. 

### AIC

#### AR

```{r LS AIC AR}
gene="AR"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



a<- ggplot(ls_norm, aes(x=mean_T, y=AR, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
ls_lm_Status<- lm(AR ~ Status, data=ls_norm)
ls_lm_meanT<- lm(AR ~ mean_T, data=ls_norm)
ls_lm_int<- lm(AR ~ Status*mean_T, data=ls_norm)

#against the NULL model
ls_lm_null<- lm(AR ~ 1, data=ls_norm)

model=c("AR ~ 1", "AR ~ Status", "AR ~ mean T","AR ~ Status + meanT + Status:meanT")
AIC=c(AIC(ls_lm_null), AIC(ls_lm_Status),AIC(ls_lm_meanT),AIC(ls_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(ls_lm_Status)$'Pr(>F)'[1],anova(ls_lm_meanT)$'Pr(>F)'[1], anova(ls_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_ls$pvalue[results_ls$gene==gene & results_ls$Test=="Status"], results_ls$pvalue[results_ls$gene==gene & results_ls$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

figure<- ggarrange(a,b, labels=c("A","B"), nrow=2)
figure

ggsave(filename="../Candidate_gene_results/LS_interaction.png", figure,device="png",width=80, height=120, units="mm", dpi=300, bg="white")
```

fail

## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.

AR failed tests above, GNRH1 and VIP have influential observations. 

```{r}
results_ls2$keep<- ifelse(grepl("GNRH1|VIP$|AR", results_ls2$gene),"no","yes")

```

## Gene Plots


```{r LS lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_ls2)){
  #g=2
  gene<- results_ls2$gene[g]
  gene<- as.character(gene)
  test<- word(results_ls2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(ls_norm[,gene] ~ Status * mean_T, data=ls_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(ls_norm[,gene] ~ ls_norm[,cont_batch], data=ls_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(ls_norm[,gene] ~ ls_norm[,cont_batch] + ls_norm[,test], data=ls_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(ls_norm[,test], ls_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(ls_norm[,gene] ~ ls_norm[,test], data=ls_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(ls_norm[,test], ls_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(ls_cand))

results_ls3<- cbind(results_ls2, lm_out[,-4])
final_res[[tissue]]<- results_ls3
```


```{r LS plotting}
plots<- list()

for(g in 1:nrow(results_ls3)){
  #g=1
  gene<- results_ls3$gene[g]
  gene<- as.character(gene)
  test<- word(results_ls3$design[g],-1)
  kp<- ifelse(results_ls3$keep[g]=="yes","*","")
  des<- results_ls3$design[g]
  
  lfc<- signif(results_ls3$log2FoldChange[g],2)
  praw<- signif(results_ls3$pvalue[g],2)
  padj=signif(results_ls3$padj[g],2)
  Rsq=signif(results_ls3$rsq[g],2)
  #lm_model<- results_ls3$lm_model[g]
  #lm_p<-signif(results_ls3$lm_p[g],2)
  #lm_q<- signif(results_ls3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(ls_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(ls_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(ls_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-1], ncol=2, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/LS_significant.png", figure,device="png",height=100, width=100, units="mm", bg="white")

```


# Bed Nucleus of the Stria Terminalis (BSTm)

```{r BSTm prep}
rm(list= ls()[!(ls() %in% keep)])
tissue="BSTm"
bstm_key<- subset(key_behav, Tissue=="BSTm")
bstm_key<- droplevels(bstm_key)


bstm_data<- data[,colnames(data) %in% rownames(bstm_key)]

#remove genes with less than 5 reads
bstm_data$avg_count<- apply(bstm_data, 1, mean)
bstm_data<- bstm_data[bstm_data$avg_count>5,]
bstm_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
bstm_data$percent_0<- apply(bstm_data, 1, function(x)length(x[x==0]))
thresh<- ncol(bstm_data)/2
bstm_data<- bstm_data[bstm_data$percent_0<=thresh,]
bstm_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_key, design= ~ Batch)
dd<- DESeq(dd)

gd<- counts(dd,normalized=TRUE)
status_results<- gd[rownames(gd) %in% candidates2,]
status_results<- as.data.frame(t(status_results))
status_results$sampleID<- rownames(status_results)
status_results<- merge(bstm_key, status_results, by="sampleID")


## data for continuous variables. 
bstm_norm<- norm_data[[3]]
rownames(bstm_norm)<- bstm_norm$X
bstm_norm$X<- NULL
bstm_norm<- as.data.frame(t(bstm_norm))
bstm_norm<- bstm_norm[,colnames(bstm_norm) %in% candidates2]
bstm_norm$sampleID<- rownames(bstm_norm)
bstm_norm<- merge(bstm_norm, bstm_key, by="sampleID")
```


```{r bstm res init}
results_bstm<- results_data[results_data$Tissue==tissue,]
#results_bstm<- results_bstm[results_bstm$gene %in% candidates,]
results_int_sub<- results_int[which(results_int$Tissue==tissue),]
results_int_sub2<- results_int_sub[which(results_int_sub$pvalue<0.05 & results_int_sub$Test=="LRT_full_interaction"),]


results_bstm2<- results_bstm[which(results_bstm$pvalue<0.05),]
results_bstm2<- results_bstm2[order(results_bstm2$gene),]
results_bstm2<- rbind(subset(results_int_sub2,select=-n), subset(results_bstm2, select=-result))
#results_bstm2$padj_cand<- p.adjust(results_bstm2$pvalue, method="BH", n=length(bstm_cand))

knitr::kable(results_bstm2[,-c(1,3,5)], row.names=FALSE, digits=3, padding=20, caption=paste0("Significant results from DESeq2 before FDR correction in the ",tissue,". Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)")) %>% kable_styling()

```

## Exploring status specific expression

A reviewer had a concern about overparameterizing the interaction models. Therefore, for the interaction models we ran an alternate set of models to test whether number of model parameters influenced whether we got a significant result on that gene. So we ran reduced models to test whether we still those effects in reduced models, and whether batch had a significant effect on any candidate genes. The latter will decide whether I need to also include the batch variables in follow up lms and AIC. 

### Batch

```{r BSTM batch}
knitr::kable(results_int_sub[results_int_sub$Test=="LRT_batch" & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Batch variable results for ",tissue)) %>% kable_styling()
```

Batch wasn't important for anyone! 

### Alternate models

```{r BSTm alternate models}
knitr::kable(results_int_sub[!grepl("batch|minimal", results_int_sub$Test) & results_int_sub$pvalue<0.05 ,-c(1,3,5,6)], row.names=FALSE, digits=3, padding=20, caption=paste0("Results from alternate models for interaction in ",tissue)) %>% kable_styling()

```

Both ESR1 and ESR2 pass the model stability test. 

### AIC

#### ESR1

```{r BSTm AIC ESR1}
gene="ESR1"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



a<- ggplot(bstm_norm, aes(x=mean_T, y=ESR1, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
bstm_lm_Status<- lm(ESR1 ~ Status, data=bstm_norm)
bstm_lm_meanT<- lm(ESR1 ~ mean_T, data=bstm_norm)
bstm_lm_int<- lm(ESR1 ~ Status*mean_T, data=bstm_norm)

#against the NULL model
bstm_lm_null<- lm(ESR1 ~ 1, data=bstm_norm)

model=c("ESR1 ~ 1", "ESR1 ~ Status", "ESR1 ~ mean T","ESR1 ~ Status + meanT + Status:meanT")
AIC=c(AIC(bstm_lm_null), AIC(bstm_lm_Status),AIC(bstm_lm_meanT),AIC(bstm_lm_int))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4])
lm_p<- c(NA,anova(bstm_lm_Status)$'Pr(>F)'[1],anova(bstm_lm_meanT)$'Pr(>F)'[1], anova(bstm_lm_int)$'Pr(>F)'[3])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_bstm$pvalue[results_bstm$gene==gene & results_bstm$Test=="Status"], results_bstm$pvalue[results_bstm$gene==gene & results_bstm$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

b<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))

```

Fail AIC test

#### ESR2

```{r BSTm AIC ESR2}
gene="ESR2"
design="~ Status + mean_T + Status:mean_T"

lfc<- signif(results_int_sub$log2FoldChange[results_int_sub$gene==gene & results_int_sub$design==design],2)
praw<- signif(results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],2)
padj=signif(results_int_sub$padj[results_int_sub$gene==gene & results_int_sub$design==design],2)



c<- ggplot(bstm_norm, aes(x=mean_T, y=ESR2, color=Status)) + geom_point(size=2,pch=19) + geom_smooth(method="lm", alpha=0.15) + labs(title=paste0(gene," ",design),
subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)), x="mean Testosterone", y="Normalized Counts") + peri_figure_supp + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15)  
  
bstm_lm_Status<- lm(ESR2 ~ Status, data=bstm_norm)
bstm_lm_meanT<- lm(ESR2 ~ mean_T, data=bstm_norm)
bstm_lm_int<- lm(ESR2 ~ Status*mean_T, data=bstm_norm)

bstm_lm_strength<- lm(ESR2 ~ strength.all_study, data=bstm_norm)
#against the NULL model
bstm_lm_null<- lm(ESR2 ~ 1, data=bstm_norm)

model=c("ESR2 ~ 1", "ESR2 ~ Status", "ESR2 ~ mean T","*ESR2 ~ Status + meanT + Status:meanT","ESR2 ~ Strength")
AIC=c(AIC(bstm_lm_null), AIC(bstm_lm_Status),AIC(bstm_lm_meanT),AIC(bstm_lm_int), AIC(bstm_lm_strength))

deltaAIC<- c(NA, AIC[1]-AIC[2], AIC[1]-AIC[3], AIC[1]-AIC[4], AIC[1]-AIC[5])
lm_p<- c(NA,anova(bstm_lm_Status)$'Pr(>F)'[1],anova(bstm_lm_meanT)$'Pr(>F)'[1], anova(bstm_lm_int)$'Pr(>F)'[3],anova(bstm_lm_strength)$'Pr(>F)'[1])

## from the DESeq full models that include ~ Year
DESeq2_raw_p=c(NA,results_bstm$pvalue[results_bstm$gene==gene & results_bstm$Test=="Status"], results_bstm$pvalue[results_bstm$gene==gene & results_bstm$Test=="mean T"], results_int_sub$pvalue[results_int_sub$gene==gene & results_int_sub$design==design],results_bstm$pvalue[results_bstm$gene==gene & results_bstm$Test=="Strength"])



knitr::kable(data.frame(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),row.names=FALSE, padding=20, caption=paste0("Model comparisons in ",tissue," where full lm model is ", design,", but DESeq p is from full models tested in DESeq2")) %>% kable_styling()

d<- ggtexttable(data.frame(model=model, AIC=signif(AIC,4), deltaAIC=signif(deltaAIC,2), lm_p=signif(lm_p,1), DESeq_raw_p=signif(DESeq2_raw_p,1)),rows=NULL, theme = ttheme(base_size = 6,padding=unit(c(4,10),"pt")))


figure<- ggarrange(a,b,c,d, labels=c("A","B","C","D"), nrow=2,ncol=2, common.legend=TRUE)
figure

ggsave(filename="../Candidate_gene_results/BSTm_interaction.png", figure,device="png",width=150, height=120, units="mm", dpi=300, bg="white")

```


## Final Res

Note these decisions are made after the plotting below, but I am including a decision element in the plotting so that's why it's above.
Excluding strength ESR2 due to low R^2

```{r}
results_bstm2$keep<- ifelse((results_bstm2$gene=="ESR2" & results_bstm2$Test=="Strength"),"no","yes")
results_bstm2$keep<- ifelse(results_bstm2$gene=="ESR1","no",as.character(results_bstm2$keep))

```


## Gene Plots


```{r BSTm lm}

lm_out<- list()
cont_batch<- "Year"
status_batch<- "Batch"
status_data<- status_results

for(g in 1:nrow(results_bstm2)){
  #g=2
  gene<- results_bstm2$gene[g]
  gene<- as.character(gene)
  test<- word(results_bstm2$design[g],-1)

  
  if(test=="Status:mean_T"){
  ## we've already decided the model form for interaction
  lm_int<- lm(bstm_norm[,gene] ~ Status * mean_T, data=bstm_norm)

  out<- data.frame(lm_model= " ~ Status + mean_T + Status:mean_T")
  out$lm_p<- anova(lm_int)$'Pr(>F)'[3]
  out$rsq<- summary(lm_int)$r.squared
  out$gene<- gene
  lm_out[[g]]<- out
  }else if(test=="mean_T" | test=="strength.all_study"){
    batch_p<-  summary(lm(bstm_norm[,gene] ~ bstm_norm[,cont_batch], data=bstm_norm))$coefficients[8]
    if(batch_p<0.05){
      lm_cont<- lm(bstm_norm[,gene] ~ bstm_norm[,cont_batch] + bstm_norm[,test], data=bstm_norm)
      out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[2]
      out$rsq<-  signif(cor(bstm_norm[,test], bstm_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }else{
      lm_cont<- lm(bstm_norm[,gene] ~ bstm_norm[,test], data=bstm_norm)
      out<- data.frame(lm_model=paste0(" ~ ",test))
      out$lm_p<- anova(lm_cont)$'Pr(>F)'[1]
      out$rsq<-  signif(cor(bstm_norm[,test], bstm_norm[,gene])^2,2)
      out$gene<- gene
      lm_out[[g]]<- out
    }
 
  }else if(test=="Status"){
    ### STATUS
    
     batch_p<-  summary(lm(status_data[,gene] ~ status_data[,status_batch], data=status_data))$coefficients[8]
     
     if(batch_p<0.05){
        lm_status<- lm(status_data[,gene] ~ status_data[,status_batch] + status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",cont_batch," + ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[2]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }else{
        lm_status<- lm(status_data[,gene] ~ status_data[,test], data=status_data)
        out<- data.frame(lm_model=paste0(" ~ ",test))
        out$lm_p<- anova(lm_status)$'Pr(>F)'[1]
        out$rsq<- NA
        out$gene<- gene
        lm_out[[g]]<- out  
     }
  }
}

lm_out<- do.call("rbind", lm_out)
#lm_out$lm_q<- p.adjust(lm_out$lm_p, method="BH", n=length(bstm_cand))

results_bstm3<- cbind(results_bstm2, lm_out[,-4])
final_res[[tissue]]<- results_bstm3
```




```{r BSTm plotting}
plots<- list()

for(g in 1:nrow(results_bstm3)){
  #g=1
  gene<- results_bstm3$gene[g]
  gene<- as.character(gene)
  test<- word(results_bstm3$design[g],-1)
  kp<- ifelse(results_bstm3$keep[g]=="yes","*","")
  des<- results_bstm3$design[g]
  
  lfc<- signif(results_bstm3$log2FoldChange[g],2)
  praw<- signif(results_bstm3$pvalue[g],2)
  padj=signif(results_bstm3$padj[g],2)
  Rsq=signif(results_bstm3$rsq[g],2)
  #lm_model<- results_bstm3$lm_model[g]
  #lm_p<-signif(results_bstm3$lm_p[g],2)
  #lm_q<- signif(results_bstm3$lm_q[g],2)
  
  if(test=="Status"){
        p<- ggplot(status_data, aes_string(x="Status", y=gene, color="Class")) + geom_point(size=2)
      p<- p+ peri_geneplots + scale_colour_manual(values=status_cols, name="Age x Status")
      p<- p + labs(x="Status", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c),list(a=lfc, b=praw, c=padj)))
      plots[[g]]<- p
      
  }else if(test=="Status:mean_T"){
    test<- gsub(":"," x ",test)
    
  plots[[g]]<-ggplot(bstm_norm, aes_string(x="mean_T", y=gene, color="Status")) + geom_point(size=2) + geom_smooth(method="lm", alpha=0.15,se=FALSE) + labs(title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq)), x="mean Testosterone", y="") + peri_geneplots + scale_colour_manual(values=status_cols2) + geom_smooth(method="lm", alpha=0.15, se=FALSE) 
  
  }else if(test=="strength.all_study"){
    plots[[g]]<- ggplot(bstm_norm, aes_string(x="strength.all_study", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Strength", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
    
  }else if(test=="mean_T"){
    
  plots[[g]]<- ggplot(bstm_norm, aes_string(x="mean_T", y=gene, color="Class")) + geom_point(size=2) + peri_geneplots + labs(x="Mean Testosterone", y="", title=paste0(kp,gene,  des),subtitle=substitute(paste("LFC=",a," p=",b," padj=", c,  " R"^{2},"=",d),list(a=lfc, b=praw, c=padj, d=Rsq))) + scale_colour_manual(values=status_cols, name="Age x Status")
  }
}

figure<- ggarrange(plotlist = plots[-c(1:2)], ncol=2, nrow=2, common.legend = TRUE)
figure<- annotate_figure(figure, top = text_grob(paste0("Candidate genes in ",tissue), face = "bold", size = 7), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=7)))
figure

ggsave(filename="../Candidate_gene_results/BSTm_significant.png", figure,device="png",height=100, width=100, units="mm", bg="white")

```


# Heat map of all candidate genes

Scaled according to individual gene's expression.

```{r, fig.width=30, fig.height=12}
data_vst<- read.csv("../data_filtered/data_VST_all_ReplicatesRemoved_antisense_V2.csv")
#data_vst<- data

data_vst$X<- plyr::revalue(data_vst$X, replace = c("LOC113993669"="CYP19A1","LOC113983511"="OXT","LOC113983498"="AVP", "LOC113982601"="AVPR2"))
#rownames(data_vst)<- plyr::revalue(rownames(data_vst), replace = c("LOC113993669"="CYP19A1","LOC113983511"="OXT","LOC113983498"="AVP"))

candidates<- c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B","AVPR2")

rownames(data_vst)<- data_vst$X
data_vst$X<- NULL
all_candidates <- data_vst[rownames(data_vst) %in% candidates,]
outlier<- "PFT2_POM_run1"
all_candidates<- all_candidates[,!colnames(all_candidates) %in% outlier,]
cand2<- candidates[candidates %in% rownames(all_candidates)]
theorder<- match(cand2, rownames(all_candidates))
all_candidates<- all_candidates[theorder,]
#terr<- key_behav[key_behav$Status=="territorial",]
#all_candidates_terr<- all_candidates[,colnames(all_candidates) %in% rownames(terr)]
colnames(all_candidates)<- paste0(word(colnames(all_candidates), 2, sep="_"),"_",word(colnames(all_candidates), 1, sep="_"))
all_candidates<- all_candidates[,order(colnames(all_candidates))]


#columns<- colnames(all_candidates)
#columns<- word(columns,1,sep="_")
#theorder<- match(columns, levels(key_behav$Tissue))
#all_candidates<- all_candidates[,theorder]

#df <- as.data.frame(key_behav[,c("Status","Batch", "mean_T")])
#df<- df[!rownames(df) %in% outlier,]
#rownames(df)<- paste0(word(rownames(df), 2, sep="_"),"_",word(rownames(df), 1, sep="_"))

color=colorRampPalette(brewer.pal(n = 7, name =
  "YlOrRd"))(100)


my_heatmap<- pheatmap(all_candidates, cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=FALSE, scale="row",color=blueWhiteRed(200))


save_pheatmap_png(my_heatmap, "../Candidate_gene_results/candidate_heatmap.png",width=3800, height=1400, res = 150)
```

## Average expression plot

```{r}
dd<- DESeqDataSetFromMatrix(countData = data, colData=key_behav, design= ~ Tissue)
dd<- DESeq(dd)
data2<- counts(dd,normalized=TRUE)
candidates<- candidates2

data2<- as.data.frame(data2[rownames(data2) %in% candidates,])
data2<- data2[c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B","AVPR2"),]
out<- list()
tissues<- c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS", "BSTm")

for(t in tissues){
    #t="PIT"
    sub2<- data2[,grepl(t,colnames(data2))]
    avg<- rowMeans(sub2)
    genes<- rownames(sub2)
    df<- data.frame(tissue=t,gene=genes,exp=avg)
    out[[t]]<- df
}
  
exp<- do.call("rbind",out)
exp$gene<- factor(exp$gene, levels=c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B","AVPR2"))
exp$gene<- factor(exp$gene, levels=rev(levels(exp$gene)))
exp$tissue<- factor(exp$tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS", "BSTm"))
exp$exp<- ifelse(exp$exp<5,0, as.numeric(exp$exp))
library(viridis)
p<- ggplot(exp, aes(x=tissue, y=gene,fill=log10(exp))) + geom_tile(colour="grey50") + scale_fill_viridis(option="A", na.value="white", direction=-1, name="log10(exp)") + theme(aspect.ratio = 1, axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())
p
ggsave("../Candidate_gene_results/figure_expression.pdf", plot=p, device="pdf", height=2.5, width=2.5, units="in")
ggsave("../Candidate_gene_results/figure_expression.png", plot=p, device="png", height=2.5, width=2.5, units="in",bg="white")
```

# Final Results Heatmaps

```{r}
final_res2<- do.call("rbind",final_res)
write.csv(final_res2, "../Candidate_gene_results/final_list_rest.csv")
final_pit<- read.csv("../Candidate_gene_results/final_list_PIT.csv")
final_pit<- final_pit[,colnames(final_pit) %in% colnames(final_res2)]

final_res2<- rbind(final_res2,final_pit)

strength<- final_res2[which(final_res2$Test=="Strength" & final_res2$keep=="yes"),]


```


```{r}
sessionInfo()
```










