---
title: "Volcano Plots and Overlap Analyses"
subtitle: "For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird "
date: Last Knit "`r Sys.Date()`"
author: "Last Substantive Change April 2021"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE, fig.align="center", comment=FALSE)
```

This code is summarizing the results from the DESeq2 within tissues experiment, and calculating genes that have similar expression patterns across tissue types. The document is broken up by interest variable, as the data read in for the volcano plots forms a scaffold for the rest.

```{r libraries and presets}
library(stringr)
library(RColorBrewer)
library(pheatmap)
library(ggplot2)
library(ggpubr)
library(DESeq2)
library(ggrepel)
library(pvclust)
library(grid)
library(gridExtra)
library(RRHO2)
library(reshape2)
library(viridis)
library(gridExtra)
library(clusterProfiler)
library(rrvgo)
library(kableExtra)
library(dplyr)

peri_theme <- theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
    axis.title.x = element_text(size = 16), axis.text.x = element_text(size = 14, 
        colour = "black"), axis.title.y = element_text(size = 16), 
    axis.text.y = element_text(size = 14, colour = "black"), 
    legend.text = element_text(size = 14), legend.title = element_text(size = 16), 
    axis.line.y = element_line(colour = "black"), axis.line.x = element_line(colour = "black"),
    plot.title = element_text(hjust = 0.5, size=22),
    plot.subtitle = element_text(hjust = 0.5, size=18, color="grey40"))  + theme(legend.key=element_blank())
dodge=position_dodge(0.8)

peri_figure <- theme(panel.background = element_blank(), 
                    panel.grid.major = element_blank(), 
                    panel.grid.minor = element_blank(),
                    plot.background = element_blank(), 
                    axis.title.x = element_text(size = 6), 
                    axis.text.x = element_text(size =5, colour = "black"), 
                    axis.title.y = element_text(size = 6),
                    axis.text.y = element_text(size = 5, colour = "black"),
                    axis.line.y = element_line(colour = "black", size = 0.4), 
                    axis.line.x = element_line(colour = "black", size = 0.4),
                    axis.ticks = element_line(colour = "black", size = 0.3),
                    plot.title = element_text(hjust = 0.5, size=6),
                    plot.subtitle = element_text(hjust = 0.5, size=5),
                    legend.text = element_text(size = 5), legend.title = element_text(size =5),
                    legend.key=element_blank(),
                    plot.margin = margin(t = 0,  # Top margin
                                         r = 0,  # Right margin
                                         b = 0,  # Bottom margin
                                         l = 0))


save_pheatmap_png <- function(x, filename, width=1200, height=1000, res = 150) {
  png(filename, width = width, height = height, res = res)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

```

```{r, read in data}

go_terms<- read.csv("../GO_annotations/pfil_GO_key_raw.csv")
go_terms<- plyr::rename(go_terms, replace=c("GeneID"="gene"))
go2gene_bp<- go_terms[which(go_terms$Aspect=="P"),c("GO_ID", "gene")]
go2gene_mf<- go_terms[which(go_terms$Aspect=="F"),c("GO_ID", "gene")]
go_obo<- read.csv("../GO_annotations/ontology_obo_out.csv")
go_obo<- plyr::rename(go_obo, replace=c("id"="GO_ID"))
go2name_bp<- go_obo[which(go_obo$namespace=="biological_process"),c("GO_ID", "name")]
go2name_mf<- go_obo[which(go_obo$namespace=="molecular_function"),c("GO_ID", "name")]

### more presets:

candidates<- c("AR", "SRD5A2", "CYP19A1", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "OXTR", "AVP", "AVPR1A", "AVPR1B","AVPR2")

tissue_order<- c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS", "BSTm")
tissue_cols<- list(GON=c("#2A9D8F","#69CCBE","#A1E5D9"),PIT=c("#2A9D8F","#69CCBE","#A1E5D9"),VMH=c("#E76F51", "#EF8E7B","#F4ABA2"),AH=c("#E76F51", "#EF8E7B","#F4ABA2"),PVN=c("#E76F51", "#EF8E7B","#F4ABA2"), POM=c("#E76F51", "#EF8E7B","#F4ABA2"), ICO=c("#E76F51", "#EF8E7B","#F4ABA2"),GCT=c("#E76F51", "#EF8E7B","#F4ABA2"), TNA=c("#E76F51", "#EF8E7B","#F4ABA2"), AI=c("#E9C46A","#EFD497","#F7E3BA"), LS=c("#F4A261","#F4B184","#F4C6A6"), BSTm=c("#F4A261","#F4B184","#F4C6A6"))

#this palette is for the pairwise heatmaps.
tissue_colors<- c(GON="#2A9D8F", PIT="#2A9D8F", VMH="#E76F51",AH="#E76F51",PVN="#E76F51",POM="#E76F51", ICO="#E76F51", GCT="#E76F51", TNA="#E76F51",AI="#E9C46A", LS="#F4A261", BSTm="#F4A261")
status_cols<- c("#86AD44","#E54849","#414042")
topgenes<- 1:6

key<- read.csv("../data_filtered/data_key_Parsed_ReplicatesRemoved.csv")
               
behav<- read.csv("../data_unfiltered/PIPFIL_T_and_behav_data.csv")
rownames(key)<- key$X

key$Color_ID<- sub("/","", key$Color_ID)
key<- plyr::rename(key, replace=c("Color_ID"="colorID"))
behav$status<- plyr::revalue(behav$status, c("floa"="floater", "terr"="territorial"))
key_behav<- merge(key, behav, by="colorID")
key_behav<- key_behav[!is.na(key_behav$last_behav),]


#create a data.frame with all of the observations. 
not_in_behav<- key[!key$X %in% key_behav$X,]
cols_not_key<- colnames(key_behav)[!colnames(key_behav) %in% colnames(key)]
cols_not_key_df<- data.frame(matrix(NA, nrow = nrow(not_in_behav), ncol = length(cols_not_key)))
colnames(cols_not_key_df)<- cols_not_key
not_in_behav<- cbind(not_in_behav, cols_not_key_df)

key_behav<- rbind(key_behav, not_in_behav)

rownames(key_behav)<- key_behav$X
key_behav<- subset(key_behav, select=c("sampleID", "Year", "Tissue", "Batch", "Status", "Class", "final_corrected_T", "mean_T", "effort.all_study", "degree.all_study", "strength.all_study"))
#key_behav$Class<- word(key_behav$Class,1)

key_behav$Tissue<- factor(key_behav$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
key_behav<- key_behav[order(rownames(key_behav)),]
key_behav$Class<- plyr::revalue(key_behav$Class, replace=c("SCB floater"="Predefinitive floater", "DCB floater "="Definitive floater", "DCB territorial"="Territorial"))
key_behav$Class<- factor(key_behav$Class, levels=c("Predefinitive floater", "Definitive floater", "Territorial"))

#read the raw count data
data<- read.csv("../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv")

rownames(data)<- data$X
data$X<- NULL


## annotating the LOC no
genes_key<- read.csv("../GO_annotations/Maggies_annotations_modifiedR.csv")
genes_key<- plyr::rename(genes_key, replace=c("GeneID"="gene"))
genes_key$gene[genes_key$gene=="LOC113993669"] <- "CYP19A1"
genes_key$gene[genes_key$gene=="LOC113983511"] <- "OXT"
genes_key$gene[genes_key$gene=="LOC113983498"] <- "AVP"
genes_key$gene[genes_key$gene=="LOC113982601"] <- "AVPR2"
data_genes<- data.frame(gene=rownames(data))
genes_key<- merge(data_genes, genes_key, by="gene", all.x=TRUE)
#genes_key<- genes_key[which(grepl("LOC[0-9]+",genes_key$gene)),]

genes_key$display_gene_ID<- ifelse(grepl("LOC[0-9]+", genes_key$gene) & !grepl("LOC[0-9]+",genes_key$best_anno) & !is.na(genes_key$best_anno), paste0(genes_key$gene," (",genes_key$best_anno,")"), as.character(genes_key$gene))
genes_key<- genes_key[,c("gene","best_anno","display_gene_ID")]


keep<- c("candidates","tissue_order", "tissue_cols","tissue_colors", "topgenes", "go_terms","peri_theme","peri_figure", "go2gene_bp","go2gene_mf", "go_obo","go2name_bp", "go2name_mf","keep", "key_behav", "data", "dodge","status_cols", "save_pheatmap_png", "genes_key", "geom_text.size")

  
```


# Social Status

## Volcano Plots

We do not show the volcano plots here, but they are in the supplementary material Section 3. 

```{r status data,  fig.height= 20, fig.width=15, echo=TRUE}
rm(list= ls()[!(ls() %in% keep)])
setwd("../DE_results/")
files<- list.files(pattern="results_[a-zA-Z]+_Status.csv", ignore.case=TRUE)
files<- files[-3]
status_results<- lapply(files, read.csv)
names(status_results)<- word(files, 2, sep="_")

output<- list()
plots<- list()
directionalp<- list()
pos_neg<- list()

for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  
  ## counts of positive and negative genes
  pos_out<- tissue
  pos_out$neg_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$neg_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$pos_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange>0,1,0)
  pos_out$pos_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange>0,1,0)

  pos_neg[[i]]<- data.frame(tissue=i, res=c(-sum(pos_out$neg_sig, na.rm=TRUE), -sum(pos_out$neg_marg, na.rm=TRUE), sum(pos_out$pos_sig, na.rm=TRUE), sum(pos_out$pos_marg, na.rm=TRUE)), type=c("negative significant", "negative marginal", "positive significant","positive marginal"))
  
  ##plots
  tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP", "LOC113982601"="AVPR2"))
  
  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
  
#tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
#tissue$is_interesting<- tissue$interesting_genes==""

#plots[[i]]<- ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + scale_x_continuous(limits=c(-4,4), breaks=c(-4,-3,-2,-1,0,1,2,3,4)) + scale_y_continuous(limits=c(0,5))


tissue$X<- NULL
tissue$pvalue<- ifelse(is.na(tissue$padj), NA, as.numeric(tissue$pvalue))
sub<- subset(tissue, select=c("gene","log2FoldChange","pvalue"))
tissue<- subset(tissue, select=c("gene","pvalue"))
colnames(tissue)[colnames(tissue)=="pvalue"] <- i
output[[i]]<- tissue

sub$logP<- -log(sub$pvalue, 10)
sub$logP<- ifelse(sub$log2FoldChange<0, -(sub$logP), sub$logP)
sub<- subset(sub, select=c("gene", "logP"))
colnames(sub)[colnames(sub)=="logP"] <- i
directionalp[[i]]<- sub
}
setwd("../analysis/")

```


```{r status volcano plots, fig.height= 20, fig.width=15, eval=FALSE}
g<- ggarrange(plotlist=plots,ncol=3, nrow=4, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Volcano plots by Status per tissue",gp=gpar(fontsize=22)), bottom=textGrob("LogFoldChange",gp=gpar(fontsize=20)), left=textGrob("-log10(p-adjusted)",gp=gpar(fontsize=20), rot = 90))
g
ggexport(g, filename = "../DE_results/figure_volcanoes_status.png", width = 1500,height = 1500, pointsize=32)

```

```{r Status summary plot for Figure}
pos_neg_out<- do.call("rbind", pos_neg)
pos_neg_out$type<- factor(pos_neg_out$type, levels=c("negative significant", "negative marginal","positive significant","positive marginal"))
pos_neg_out$tissue<- factor(pos_neg_out$tissue, levels=tissue_order)
pos_neg_out$abs_val<- ifelse(grepl("significant",pos_neg_out$type), abs(pos_neg_out$res), "")
pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")
geom_text.size = (5/14)*(5) 

p<- ggplot(pos_neg_out, aes(x=tissue, y=res, fill=type)) + geom_bar(stat="identity") + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(-1200, 1200)) + labs(title="Status", x="", y="") + scale_fill_manual(values=pos_cols) + geom_hline(yintercept=0)+ theme(legend.position = "none", aspect.ratio = 1) + geom_text(aes(label=abs_val),vjust=pos_neg_out$res < 0, size=geom_text.size) + theme(axis.text.x = element_text(angle=90, vjust=0.5, hjust=1)) + labs(title="social status", subtitle=" ", y="number of genes")
p
ggsave("../Overlap_results/figure_Status_bars.pdf", plot=p, device="pdf", height=1.75, width=1.75, units="in")

de_genes<- pos_neg_out %>% group_by(tissue) %>% summarize(status_genes=sum(abs(res)))
keep<- c(keep, "de_genes")
```

```{r testing something COX, eval=FALSE}
all_data<- data

start<- nrow(all_data)
#remove genes with avg read counts <20
all_data$avg_count<- apply(all_data, 1, mean)
all_data<- all_data[all_data$avg_count>20,]
all_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
all_data$percent_0<- apply(all_data, 1, function(x)length(x[x==0]))
thresh<- ncol(all_data)/2
all_data<- all_data[all_data$percent_0<=thresh,]
all_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=all_data, colData=key_behav, design= ~ Tissue)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

#NCBI cox genes for Pipra
cox<- c("COX1-homolog"="LOC113984744", "COX6-homolog"="LOC114003485","COX14-homolog"="LOC113993743", "COX15-homolog"="LOC113982573")
iegs<- c(Fos="FOS", jun="JUN", EGR1="EGR1")
cox<- iegs
a<- plotCounts(dd, gene=cox[1], intgroup=c("Class","Status", "Tissue","strength.all_study","mean_T"), returnData=TRUE)
a$sample<- rownames(a)
ap<- ggplot(a, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title=names(cox[1]),x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])
b<- plotCounts(dd, gene=cox[2], intgroup=c("Class","Status", "Tissue","strength.all_study","mean_T"), returnData=TRUE)
b$sample<- rownames(b)
bp<- ggplot(b, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title=names(cox[2]),x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])
c<- plotCounts(dd, gene=cox[3], intgroup=c("Class","Status", "Tissue","strength.all_study","mean_T"), returnData=TRUE)
c$sample<- rownames(c)
cp<- ggplot(c, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title=names(cox[3]),x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])
#d<- plotCounts(dd, gene=cox[4], intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
#dp<- ggplot(d, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title=names(cox[4]),x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

ggarrange(ap,bp,cp)

a$ind<- word(a$sample,1,1, sep="_")
a<- dcast(a, ind + mean_T + Status + strength.all_study ~ Tissue, value.var="count")
colnames(a)[5:16]<- paste0("FOS_",colnames(a)[5:16])
b$ind<- word(b$sample,1,1,sep="_")
b<- dcast(b,ind + mean_T + Status + strength.all_study ~ Tissue, value.var="count")
colnames(b)[5:16]<- paste0("JUN_",colnames(b)[5:16])
c$ind<- word(c$sample,1,1,sep="_")
c<- dcast(c, ind + mean_T + Status + strength.all_study ~ Tissue, value.var="count")
colnames(c)[5:16]<- paste0("EGR1_",colnames(c)[5:16])

iegs<- merge(a,b, by="ind")
iegs<- merge(iegs,c, by="ind")
iegs<- iegs[,!grepl("\\.x", colnames(iegs))]
iegs<- iegs[,!grepl("\\.y", colnames(iegs))]
iegs2<- iegs[,-c(1,27)]
pvalue_cor<- cor(iegs2, use= "pairwise.complete.obs", method="pearson")
pheatmap(pvalue_cor)


candidates<- c("AR", "SRD5A2", "LOC113993669", "ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","LOC113983511", "OXTR", "LOC113983498", "AVPR1A")
out<- list()
for(i in candidates){
  a<- plotCounts(dd, gene=i, intgroup=c("Class","Status", "Tissue","strength.all_study","mean_T"), returnData=TRUE)
a$sample<- rownames(a)
a$ind<- word(a$sample,1,1, sep="_")
a<- dcast(a, ind  ~ Tissue, value.var="count")
out[[i]]<- a
}
candidate_genes<- do.call("cbind", out)
inds<- candidate_genes$AR.ind
candidate_genes<- candidate_genes[,!grepl("ind",colnames(candidate_genes))]
candidate_genes$ind<- inds

key_behav$ind<- word(key_behav$sampleID,1,1, sep="_")

cg<- merge(key_behav[,c(5,8,11,12)], candidate_genes, by="ind")
cg$Status<- as.numeric(ifelse(cg$Status=="territorial",1,0))

cgcor<- cor(cg[,-1], use= "pairwise.complete.obs", method="pearson")
pheatmap(cgcor, clustering_method="complete")

xy <- t(combn(colnames(cgcor[,-1]), 2))
xy<- data.frame(xy, dist=cgcor[xy,-1])
cor_cgclust<-
ggplot(xy, aes(x=dist, fill=..x..)) + geom_histogram(color="grey80", size=0.3) + scale_fill_gradient2(low=color[42], high = color[74]) + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(0,15)) + theme(legend.position="none", aspect.ratio = 0.2) + labs(x="r") + scale_x_continuous(limits=c(-1,1)) + geom_vline(aes(xintercept=mean(dist)),
            color="grey70", linetype="dashed", size=0.3) 
keep<- c(keep, "xy_status")

result <- pvclust(cgcor, method.dist="cor", method.hclust="complete", nboot=1000, parallel=TRUE)

plot(result)
pvrect(result, alpha=0.90)
```


## Pairwise Tissue Similarities

Pulling out the p-value data.

```{r status tissue similarity, echo=TRUE}

temp<- merge(output[[1]],output[[2]],by="gene", all=TRUE)
temp<- merge(temp, output[[3]], by="gene", all=TRUE)
temp<- merge(temp, output[[4]], by="gene", all=TRUE)
temp<- merge(temp, output[[5]], by="gene", all=TRUE)
temp<- merge(temp, output[[6]], by="gene", all=TRUE)
temp<- merge(temp, output[[7]], by="gene", all=TRUE)
temp<- merge(temp, output[[8]], by="gene", all=TRUE)
temp<- merge(temp, output[[9]], by="gene", all=TRUE)
temp<- merge(temp, output[[10]], by="gene", all=TRUE)
temp<- merge(temp, output[[11]], by="gene", all=TRUE)
pvalue_data<- merge(temp, output[[12]], by="gene", all=TRUE)
rownames(pvalue_data)<- pvalue_data$gene

pvalue_data$gene<- NULL

```


and the directional log-transformed p-values: 

```{r status tissue similarity2, echo=TRUE}
temp<- merge(directionalp[[1]],directionalp[[2]],by="gene", all=TRUE)
temp<- merge(temp, directionalp[[3]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[4]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[5]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[6]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[7]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[8]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[9]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[10]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[11]], by="gene", all=TRUE)
logpvalue_data<- merge(temp, directionalp[[12]], by="gene", all=TRUE)
rownames(logpvalue_data)<- logpvalue_data$gene

logpvalue_data$gene<- NULL
logpvalue_status<- logpvalue_data
keep<- c(keep, "logpvalue_status")

pvalue_cor<- cor(logpvalue_data, use= "pairwise.complete.obs", method="pearson")
xy <- t(combn(colnames(pvalue_cor), 2))
xy<- data.frame(xy, dist=pvalue_cor[xy])
#hist(xy$dist, breaks=20) 
#min(xy$dist)
mean(xy$dist)
ps<-colorRampPalette(brewer.pal(n = 7, name =
  "Purples"))(100)

#color=colorRampPalette(brewer.pal(n = 7, name ="YlOrRd"))(100)
color=rev(colorRampPalette(brewer.pal(n = 7, name =
  "RdBu"))(100))


exp<- data.frame(row.names=colnames(logpvalue_data),mean_logP=apply(-log10(pvalue_data),2, mean,na.rm=TRUE)) #using the raw p-value data here because taking the average of directional p-values would be silly.
exp$tissue<- tissue_order
ann_cols<- list(tissue=tissue_colors, mean_logP=ps[1:52]) #Max mean_log pvalue is 0.52



p<- pheatmap(pvalue_cor, color=color[42:100], annotation_row=exp, border_color = NA, display_numbers = signif(pvalue_cor,1), annotation_colors = ann_cols, clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method="complete")
p
save_pheatmap_png(p, filename="../Overlap_results/heatmap_status", width=800, height=800)


xy_status<-
ggplot(xy, aes(x=dist, fill=..x..)) + geom_histogram(color="grey80", size=0.3) + scale_fill_gradient2(low=color[42], high = color[74]) + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(0,15)) + theme(legend.position="none", aspect.ratio = 0.2) + labs(x="r") + scale_x_continuous(limits=c(-1,1)) + geom_vline(aes(xintercept=mean(dist)),
            color="grey70", linetype="dashed", size=0.3) 
keep<- c(keep, "xy_status")

result <- pvclust(pvalue_cor, method.dist="euclidean", method.hclust="complete", nboot=10000, parallel=TRUE)

plot(result)
pvrect(result, alpha=0.90)
```

## RRHO2
This approach uses package 'RRHO2' and compares rank ordered gene lists against each-other and uses a hypergeometric test to compare the similarity of gene names in shorter segments along the length gene list. 

The results reported in the paper are scaled to the largest log10 P-value in all of the pairwise comparisons, as a visual aid to help scale for multiple comparisons across tissues. However, we also present the unscaled versions too. 

```{r RRHO2,  eval=FALSE, echo=TRUE}
RRHO_objects<- list()

comparisons<- combn(tissue_order, 2)
RRHO_objects<- list()
for(i in 1:ncol(comparisons)){
  comparison<- comparisons[,i]
  p1_name<- comparison[1]
  p2_name<- comparison[2]
  p1<- data.frame(gene=row.names(logpvalue_data), logP_1=logpvalue_data[,p1_name])
  p2<- data.frame(gene=row.names(logpvalue_data), logP_2=logpvalue_data[,p2_name])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  p11<- comb[, c("gene","logP_1")]
  p11<- p11[order(p11$logP_1, decreasing=TRUE),]
  p22<- comb[,c("gene","logP_2")]
  p22<- p22[order(p22$logP_2, decreasing=TRUE),]
  p1_p2<- paste(p1_name,p2_name, sep="_")
  RRHO_objects[[p1_p2]]<- RRHO2_initialize(p11, p22, labels = c(p1_name, p2_name), log10.ind=TRUE, boundary=0.01)
  #jpeg(filename=paste0("RRHO2_status",p1_p2,".jpg"))
  #RRHO2_heatmap(RRHO_objects[[p1_p2]])
  #dev.off()
}
save(RRHO_objects,file="../Overlap_results/RRHO2/RRHO2_Status.RData")

```


```{r, eval=FALSE, echo=TRUE}

#load("../Overlap_results/RRHO2/RRHO2_Status.RData")



maxp<- vector(mode="numeric")
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  maxp[i]<- max(sub, na.rm=TRUE)
}
maxp<- max(maxp)

plots_scaled<- list() # for the plots on the same scale
plots<- list()
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  sub<- melt(sub, na.rm=TRUE)
  x<- word(i,1,1,sep="_")
  y<- word(i,2,2,sep="_")
  plots_scaled[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank(), legend.position="none") + labs(x=x, y=y)
  plots[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis() + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
}


m <- matrix(NA, 11, 11)
m[lower.tri(m, diag = T)] <- 1:66

## This plot is just to farm the legend. 
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
leg<- get_legend(p)

m1<- m
m1[1,11]<- 67
plots_scaled[["legend"]]<- leg

```

```{r, eval=FALSE, echo=TRUE}
g<-arrangeGrob(grobs= plots_scaled, layout_matrix = m1)

ggsave(file="../Overlap_results/RRHO2/figure_RRHO_status_scaled.pdf", g, units="in", width=20, height=20)


g<-grid.arrange(grobs= plots, layout_matrix = m)
ggsave(file="../Overlap_results/RRHO2/figure_RRHO_status_unscaled.pdf", g, units="in", width=25, height=15)
```


```{r, eval=FALSE}
## getting the BSTm/Ls figure for the manuscript.
sub<- RRHO_objects[["LS_BSTm"]]$hypermat
sub<- melt(sub, na.rm=TRUE)
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_figure + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(name="log(10)P-value") + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
p
ggsave("../Overlap_results/RRHO2/figure_Status_results.png", plot=p, device="png", height=1.75, width=2.5, units="in", bg="white")

```

I don't know how to get these figures to not look crappy on the markdown so they are available to view as a raw pdf. The scaled PDF is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_status_scaled.pdf), and the unscaled is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_status_unscaled.pdf).



## Overall brain patterns

Here I am taking just the brain tissues and taking the median p-value of each gene in the rank-ordered gene list. This will give an idea for similar gene expression patterns across the entire brain.

```{r Status brain, echo=TRUE}

brain<- logpvalue_data[,-c(1:2)]

brain$p_combo<- apply(brain,1, median)
brain<- brain[complete.cases(brain),]
brain$gene<- rownames(brain)
brain_go<- merge(brain, go_terms, by="gene", all.x=TRUE)
brain_go<- brain_go[order(brain_go$p_combo, decreasing=TRUE),]
write.csv(brain,"../Overlap_results/results_brain_status.csv", row.names=FALSE)
geneList<- brain_go$p_combo
names(geneList)<- brain_go$gene

#subsets<- brain[brain$p_combo>=1.3,]

y<- enricher(brain$gene[which(abs(brain$p_combo)>1.3)], universe=brain$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)
ego2<- ego2[ego2$GeneRatio2>1,]

p<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=pvalue)) + geom_bar(stat="identity", size=0.3, color="#2A1E59") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1.2)) + coord_flip() + peri_figure + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched") + theme(aspect.ratio = 1.2) + scale_fill_distiller(palette="Purples")
p
ggsave("../Overlap_results/figure_brain_GO_status.pdf", plot=p, device="pdf", height=1.60, width=4.60, units="in")


knitr::kable(y@result[1:10,], caption="BP enriched GO terms in the median p in the brain", row.names = FALSE) %>% kable_styling()

write.csv(y@result,"../Overlap_results/GO_status_brain.csv", row.names=FALSE)



#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
#write.csv(y@result,"GSEA_status_brain.csv", row.names=FALSE)

                   
ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                               scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


#heatmapPlot(simMatrix,
#            reducedTerms,
#            annotateParent=TRUE,
#            annotationLabel="parentTerm",
#            fontsize=6)
```



```{r}
#pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")
genes<- brain[which(abs(brain$p_combo)>1.3),]
genes_GO<- go2gene_bp[which(go2gene_bp$gene %in% genes$gene),]
genes_GO<- genes_GO[which(genes_GO$GO_ID %in% ego2$ID[1:20]),]
genes_GO<- merge(genes_GO, go2name_bp, by="GO_ID")
genes_GO<- merge(genes_GO, genes, by="gene")
genes_GO$gene_GO_combo<- paste0(genes_GO$gene, genes_GO$GO_ID)
genes_GO<- genes_GO[!duplicated(genes_GO$gene_GO_combo),]
genes_GO$pos_neg<- ifelse(genes_GO$p_combo<0, "neg","pos")
genes_GO<- merge(genes_GO, genes_key, by="gene")
colnames(genes_GO)[colnames(genes_GO)=="GO_ID"]<- "ID"
genes_GO<- merge(genes_GO, ego2, by="ID")
#genes_GO2<- genes_GO %>% group_by(name) %>% summarise()


a<- ggplot(data=genes_GO, aes(x=name, y=p_combo, color=pos_neg, alpha=))  + geom_point(size=1) + scale_color_manual(values=c("#0D8CFF","#FF3300")) + coord_flip() + geom_text_repel(aes(label=display_gene_ID), size=geom_text.size) + peri_figure + labs(x="",y="expression direction x log10(p)") + theme(legend.position = "none")
a
ggsave(filename="../Overlap_results/figure_GO_status_top.png",a,height=2.60, width=4.60, units="in", device="png", bg="white")
```


```{r status brain gene plotting, echo=TRUE}
tophits<- brain
tophits$p_combo<- abs(tophits$p_combo)
tophits<- tophits[order(tophits$p_combo, decreasing=TRUE),]
tophits<- tophits$gene[1:10]
out<- list()
#i="POM"
for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  sub<- tissue[tissue$gene %in% tophits,]
  out[[i]]<- data.frame(gene=sub$gene, lfc=sub$log2FoldChange, p=sub$pvalue, padj=sub$padj, Tissue=i)
  }
status_results_tophits<- do.call("rbind", out)
status_results_tophits<- status_results_tophits[status_results_tophits$Tissue!="GON" & status_results_tophits$Tissue!="PIT",]


outliers=c("PFT2_POM_run1")

brain_behav<- subset(key_behav, Tissue!="GON")
brain_behav<- subset(brain_behav, Tissue!="PIT")
brain_behav<- brain_behav[!rownames(brain_behav) %in% outliers,]
brain_behav<- droplevels(brain_behav)


brain_data<- data[,colnames(data) %in% rownames(brain_behav)]



start<- nrow(brain_data)
#remove genes with avg read counts <20
brain_data$avg_count<- apply(brain_data, 1, mean)
brain_data<- brain_data[brain_data$avg_count>20,]
brain_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
brain_data$percent_0<- apply(brain_data, 1, function(x)length(x[x==0]))
thresh<- ncol(brain_data)/2
brain_data<- brain_data[brain_data$percent_0<=thresh,]
brain_data$percent_0<-NULL
#13652 genes.

dd<- DESeqDataSetFromMatrix(countData=brain_data, colData=brain_behav, design= ~ Tissue +  Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 

res<- results(dd, alpha=0.1)


gone<- plotCounts(dd, gene=row.names(res[tophits[1],]), intgroup=c("Tissue","Status"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[tophits[2],]), intgroup=c("Tissue","Status"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[tophits[3],]), intgroup=c("Tissue","Status"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[tophits[4],]), intgroup=c("Tissue","Status"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[tophits[5],]), intgroup=c("Tissue","Status"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[tophits[6],]), intgroup=c("Tissue","Status"), returnData=TRUE)
vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)
#
plots<- list()
#i=2
for(i in 1:6){
  dat<- vars[[i]]
  name<- tophits[i]
  result<- status_results_tophits[status_results_tophits$gene==name,]
  result$label<- paste0("LFC=",signif(result$lfc,2), "\np(raw)=", formatC(result$p,2,format="e"))
  result$x<- 1.5
  result$y<- max(dat$count)-30
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count)) + geom_point(size=2, position=dodge, aes(color=Status)) + geom_boxplot(fill=NA, position=dodge, aes(color=Status)) + labs(title=paste0(name, " p(median)=", formatC(10^-abs(brain_go$p_combo[brain_go$gene==name]),2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols[2:3]) + facet_wrap(. ~Tissue, ncol=5) + geom_text(data=result, aes(x=x, y=y, label=label))
}


g<- ggarrange(plotlist=plots[1:2],ncol=1, nrow=2, common.legend = TRUE, labels=c("A","B"))
g<- annotate_figure(g, top=textGrob("Top two genes consistently DE in brain with social status",gp=gpar(fontsize=22)), bottom=textGrob("social status",gp=gpar(fontsize=18)))
g
ggexport(g, filename = "../Overlap_results/figure_braintop_status.png", width = 700,height = 900)

hub=rownames(res[tophits[1],])
gone$Tissue<- factor(gone$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
result<- status_results_tophits[status_results_tophits$gene==hub,]

result$Tissue<- factor(result$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
result<- result[order(result$Tissue),]

result$label<- paste0("LFC=",signif(result$lfc,1), ", p=", signif(result$p,1), ", q=", signif(result$padj,1))
result$x<- 1.5
result$y<- rep(c(max(gone$count)-30,-2),5)
geom_text.size = (5/14)*(5) 


a<- ggplot(gone, aes(x=Tissue, y=count)) + geom_point(size=0.6, pch=19, aes(color=Status), position=position_dodge(0.5)) + geom_boxplot(outlier.shape = NA, fill=NA, aes(color=Status),lwd=0.3, width=0.5, position=position_dodge(0.5)) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="", y="normalized counts", title=hub) + theme(legend.position = "none", strip.text = element_text(size=5)) + geom_text(data=result, aes(x=Tissue, y=y, label=label), size=geom_text.size) 


a




hub=rownames(res[tophits[2],])
gtwo$Tissue<- factor(gtwo$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
result<- status_results_tophits[status_results_tophits$gene==hub,]

result$Tissue<- factor(result$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
result<- result[order(result$Tissue),]
result$label<- paste0("LFC=",signif(result$lfc,1), ", p=", signif(result$p,1), ", q=", signif(result$padj,1))
result$x<- 1.50
result$y<- rep(c(max(gtwo$count)-30,-2),5)
geom_text.size = (5/14)*(5) 
b<- ggplot(gtwo, aes(x=Tissue, y=count)) + geom_point(size=0.6, pch=19, aes(color=Status), position=position_dodge(0.5)) + geom_boxplot(outlier.shape = NA, fill=NA, aes(color=Status),lwd=0.3, width=0.5, position=position_dodge(0.5)) + peri_figure + scale_colour_manual(values=status_cols[2:3]) + labs(x="", y="normalized counts", title=hub) + theme(legend.position = "none", strip.text = element_text(size=5)) + geom_text(data=result, aes(x=Tissue, y=y, label=label), size=geom_text.size) 
b

p<- ggarrange(a,b, nrow=2)

ggsave("../Overlap_results/figure_braintop_status.pdf", plot=p, device="pdf", height=3, width=5.5, units="in")
```

```{r}

status_results_tophits<- merge(status_results_tophits, genes_key, by="gene")
#status_results_tophits<- #status_results_tophits[grepl("ROBO4|POLD1|GRINA|LOC113988442|GAMT|FGFR4|LRRC25",status_results_tophits$display_gene_ID),]
status_results_tophits$Tissue<- factor(status_results_tophits$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))

#ggplot(status_results_tophits, aes(x=Tissue,y=lfc, colour=display_gene_ID)) + geom_point(size=0.5) + peri_figure
status_results_tophits$padj2<- ifelse(status_results_tophits$padj<0.1,"yes","no")
#geom_text(aes(label=padj2),size=geom_text.size,color="black") + 
g<- ggplot(status_results_tophits, aes(x=Tissue,y=display_gene_ID, fill=lfc)) + geom_tile(aes(color=padj2)) +  scale_color_manual(values=c("grey50","black")) + scale_fill_gradient2(low="#0D8CFF",high="#FF3300",mid="white") + theme(aspect.ratio = 1, axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())

ggsave("../Overlap_results/figure_braintop10grid_status.pdf", plot=g, device="pdf", height=3, width=3, units="in")
```



## GO Overlap

```{r status GO overlap}
gos<- list.files(path="../DE_results/", pattern="GO_BP")
gos<- gos[grepl("status",gos, ignore.case=TRUE)]
setwd("../DE_results/")
gos2<- lapply(gos, read.csv)
setwd("G:/My Drive/Manakins/Pipra Brain/PIFI_brain_transcriptome/analysis")

names(gos2)<- gos
names(gos2)<- sub(".csv","", names(gos2))
names(gos2)<- sub("GO_BP_", "",names(gos2))
names(gos2)<- sub("_Status", "",names(gos2))
gos2<- mapply(`[<-`, gos2, 'tissue', value = names(gos2), SIMPLIFY = FALSE)
gos3<- do.call("rbind",gos2)
gos3<- gos3[which(gos3$pvalue<0.05),]
terms<- as.data.frame(table(gos3$Description))
colnames(terms)<- c("Description","Freq")
terms<- merge(terms, gos3[,c("ID","Description")], by="Description")
terms<- terms[!duplicated(terms$Description),]
terms<- terms[order(terms$Freq, decreasing=TRUE),]

terms$Description<- factor(terms$Description, levels=terms$Description[order(terms$Freq, decreasing=TRUE)])

ggplot(data=terms[1:20,], aes(x=Description, y=Freq)) + geom_bar(stat="identity", fill="darkturquoise") + theme(axis.text.x=element_text(angle=90)) + coord_flip() + peri_figure


simMatrix <- calculateSimMatrix(gos3$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(terms$Freq, terms$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                               scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
#scatterPlot(simMatrix , reducedTerms)

red<- reducedTerms
colnames(red)[colnames(red)=="go"]<- "ID"
test<- merge(red, terms, by="ID")

parents<- test %>% group_by(parentTerm) %>% summarize(parent_size=sum(score))
test<- merge(parents, test, by="parentTerm")
test<- test[order(test$parent_size, decreasing=TRUE),]
write.csv(test,file="../Overlap_results/results_GO_overlap_rrvgo_status.csv", row.names=FALSE)
tissues<- merge(gos3, test, by="ID")
tissues2<- tissues %>% group_by(parentTerm,tissue) %>% summarise(n_term=(length(unique(ID))))

tissues<- tissues %>% group_by(parentTerm) %>% summarize(n_tissue=length(unique(tissue)), n_terms=length(unique(ID)))

tissues<- tissues[order(tissues$n_tissue, decreasing = TRUE),]
tissues$parentTerm<- factor(tissues$parentTerm, levels=tissues$parentTerm[order(tissues$n_tissue, decreasing = TRUE)])



p<- ggplot(data=tissues[1:20,], aes(x=parentTerm, y=n_tissue, fill=n_terms)) + geom_bar(stat="identity", colour="grey20", linewidth=0.2) + theme(axis.text.x=element_text(angle=90), legend.key.size=unit(3,"mm")) + coord_flip() + peri_figure + scale_fill_distiller(palette="Purples", direction=1, name="number GO terms") + scale_y_continuous(expand=c(0,0), limit=c(0,14), breaks=c(2,4,6,8,10,12)) + labs(x="parent GO term", y="Number of Tissues")
p

ggsave("../Overlap_results/figure_GOparent_overlap_status.pdf", plot=p, device="pdf", height=1.60, width=4.60, units="in")
```



```{r Status geom_tile}


tissues3<- tissues2[tissues2$parentTerm %in% tissues$parentTerm[which(tissues$n_tissue>6)],]

out<- list()
for(i in tissues3$parentTerm){
  #i="response to hypoxia"
  term<- tissues3[tissues3$parentTerm==i,]
  term<-droplevels(term)
  term$tissue<- as.character(term$tissue)
  if(any(!tissue_order %in% unique(term$tissue))){
    tiss<- tissue_order[!tissue_order %in% unique(term$tissue)]
    out[[i]]<- data.frame(parentTerm=i,tissue=tiss,n_term=NA)
    
  }
  
}
out<- do.call("rbind",out)
tissues3<- rbind(tissues3, out)
tissues3<- merge(tissues3, tissues[,c("parentTerm","n_tissue")],by="parentTerm")

tissues3$tissue<- factor(tissues3$tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS","BSTm"))
tissues3<- tissues3[which(tissues3$n_tissue>8),]

excl<- c("viral life cycle","SRP-dependent cotranslational protein targeting to membrane","nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","negative regulation of Ras protein signal transduction","protein homotrimerization","positive regulation of cell-substrate adhesion")

tissues3<- tissues3[!tissues3$parentTerm %in% excl,]

p2<- ggplot(data=tissues3, aes(x=tissue, y=parentTerm, fill=n_term)) + geom_tile(colour="grey50") + scale_fill_viridis(option="A", na.value="white", direction=-1, name="Number Terms p<0.05") + theme(axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())
p2
ggsave("../Overlap_results/figure_GOparent_overlap_status_heatmap.pdf", plot=p2, device="pdf", height=2.0, width=5, units="in")
```

# Mean testosterone phenotype

## Volcano Plots 

```{r mean T data, fig.height= 20, fig.width=15}
rm(list= ls()[!(ls() %in% keep)])
setwd("../DE_results/")
files<- list.files(pattern="results_[a-zA-Z]+_mean_T.csv", ignore.case=TRUE)
status_results<- lapply(files, read.csv)
names(status_results)<- word(files, 2, sep="_")


output<- list()
plots<- list()
directionalp<- list()
pos_neg<- list()
for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  
  ## counts of positive and negative genes
  pos_out<- tissue
  pos_out$neg_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$neg_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$pos_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange>0,1,0)
  pos_out$pos_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange>0,1,0)

  pos_neg[[i]]<- data.frame(tissue=i, res=c(-sum(pos_out$neg_sig, na.rm=TRUE), -sum(pos_out$neg_marg, na.rm=TRUE), sum(pos_out$pos_sig, na.rm=TRUE), sum(pos_out$pos_marg, na.rm=TRUE)), type=c("negative significant", "negative marginal", "positive significant","positive marginal"))
  
  ##plots
  tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP", "LOC113982601"="AVPR2"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
  
#tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
#tissue$is_interesting<- tissue$interesting_genes==""
#plots[[i]]<- ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + scale_x_continuous(limits=c(-5,6), breaks=c(-5,-4,-3,-2,-1,0,1,2,3,4,5,6)) + scale_y_continuous(limits=c(0,10)) + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)
tissue$X<- NULL
tissue$pvalue<- ifelse(is.na(tissue$padj), NA, as.numeric(tissue$pvalue))
sub<- subset(tissue, select=c("gene","log2FoldChange","pvalue"))
tissue<- subset(tissue, select=c("gene","pvalue"))
colnames(tissue)[colnames(tissue)=="pvalue"] <- i
output[[i]]<- tissue

sub$logP<- -log(sub$pvalue, 10)
sub$logP<- ifelse(sub$log2FoldChange<0, -(sub$logP), sub$logP)
sub<- subset(sub, select=c("gene", "logP"))
colnames(sub)[colnames(sub)=="logP"] <- i
directionalp[[i]]<- sub

}
setwd("../analysis/")
```
```{r meanT volcano plotting, fig.height= 20, fig.width=15, eval=FALSE}
g<- ggarrange(plotlist=plots,ncol=3, nrow=4, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Volcano plots by mean T per tissue",gp=gpar(fontsize=22)), bottom=textGrob("LogFoldChange",gp=gpar(fontsize=20)), left=textGrob("-log10(p-adjusted)",gp=gpar(fontsize=20), rot = 90))
g
ggexport(g, filename = "../DE_results/figure_volcanoes_mean_T.png", width = 1500,height = 1500, pointsize=32)

```


```{r mean T summary}
pos_neg_mean_T<- do.call("rbind", pos_neg)
pos_neg_mean_T$type<- factor(pos_neg_mean_T$type, levels=c("negative significant", "negative marginal","positive significant","positive marginal"))
pos_neg_mean_T$tissue<- factor(pos_neg_mean_T$tissue, levels=tissue_order)
pos_neg_mean_T$abs_val<- ifelse(grepl("significant",pos_neg_mean_T$type), abs(pos_neg_mean_T$res), "")
pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")

geom_text.size = (5/14)*(5) 

p<- ggplot(pos_neg_mean_T, aes(x=tissue, y=res, fill=type)) + geom_bar(stat="identity") + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(-1200, 1200)) + labs(title="Status", x="", y="") + scale_fill_manual(values=pos_cols) + geom_hline(yintercept=0)+ theme(legend.position = "none", aspect.ratio = 1) + geom_text(aes(label=abs_val),vjust=pos_neg_mean_T$res < 0, size=geom_text.size) + theme(axis.text.x = element_text(angle=90, vjust=0.5, hjust=1)) + labs(title="mean testosterone", subtitle=" ", y="number of genes")
p
ggsave("../Overlap_results/figure_meanT_bars.pdf", plot=p, device="pdf", height=1.75, width=1.75, units="in")


de_genes_T<- pos_neg_mean_T %>% group_by(tissue) %>% summarize(meanT_genes=sum(abs(res)))
de_genes<- merge(de_genes, de_genes_T, by="tissue")
keep<- c(keep, "de_genes")

```

```{r mean T plotting GON, eval=FALSE}
GON_modules<- read.csv("../WGCNA/ByTissue_2021/GON_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B", "TH", "NR4A1","CHRNA3")

i="GON"
 tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, GON_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","turquoise","pink","green","salmon","blue","black","brown"))
modulecols<- c("grey30","grey60","turquoise","pink","green","salmon","blue","black","brown")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$interesting_genes[tissue$interesting_genes=="LOC423605"]<- "LOC113988442"
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")

ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none")
```


```{r mean T plotting ICO, eval=FALSE}
ICO_modules<- read.csv("../WGCNA/ByTissue_2021/ICO_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B","MMP2")

i="ICO"
 tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, ICO_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","darkturquoise","turquoise","coral1","maroon","darkviolet" ,"thistle1","bisque4","orangered4","navajowhite1","brown2","firebrick4","darkseagreen3","honeydew1" ,"palevioletred3","darkmagenta","blue2", "greenyellow"))
modulecols<- c("grey30","grey60","darkturquoise","turquoise","coral1","maroon","darkviolet" ,"thistle1","bisque4","orangered4","navajowhite1","brown2","firebrick4","darkseagreen3","honeydew1" ,"palevioletred3","darkmagenta","blue2", "greenyellow")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$interesting_genes[is.na(tissue$interesting_genes)]<- "LOC113988355"
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")

ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none")
```

## Pairwise Tissue Similarity

```{r mean T similarity pvalue}
temp<- merge(output[[1]],output[[2]],by="gene", all.x=TRUE)
temp<- merge(temp, output[[3]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[4]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[5]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[6]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[7]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[8]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[9]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[10]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[11]], by="gene", all.x=TRUE)
pvalue_data<- merge(temp, output[[12]], by="gene", all.x=TRUE)
rownames(pvalue_data)<- pvalue_data$gene
pvalue_data$gene<- NULL

```

```{r mean T similarity directional p}
temp<- merge(directionalp[[1]],directionalp[[2]],by="gene", all=TRUE)
temp<- merge(temp, directionalp[[3]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[4]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[5]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[6]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[7]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[8]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[9]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[10]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[11]], by="gene", all=TRUE)
logpvalue_data<- merge(temp, directionalp[[12]], by="gene", all=TRUE)
rownames(logpvalue_data)<- logpvalue_data$gene

logpvalue_data$gene<- NULL
logpvalue_meanT<- logpvalue_data
keep<- c(keep, "logpvalue_meanT")

pvalue_cor<- cor(logpvalue_data, use= "pairwise.complete.obs", method="pearson")


xy <- t(combn(colnames(pvalue_cor), 2))
xy<- data.frame(xy, dist=pvalue_cor[xy])
#hist(xy$dist, breaks=20) 
min(xy$dist)
mean(xy$dist)
ps<-colorRampPalette(brewer.pal(n = 7, name =
  "Purples"))(100)

#color=colorRampPalette(brewer.pal(n = 7, name ="YlOrRd"))(100)
color=rev(colorRampPalette(brewer.pal(n = 7, name =
  "RdBu"))(100))


exp<- data.frame(row.names=colnames(pvalue_data),mean_logP=apply(-log10(pvalue_data),2, mean,na.rm=TRUE))
exp$tissue<- tissue_order
ann_cols<- list(tissue=tissue_colors, mean_logP=ps[1:56])#Max mean_log pvalue is 0.56



pheatmap(pvalue_cor, color=color[44:100], annotation_row=exp, border_color = NA, display_numbers = signif(pvalue_cor,1), annotation_colors = ann_cols, clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method="complete")+theme(aspect.ratio=1)
#ggsave("figure_heatmap_mean_T.png", device="png", width=9.5, height=9.5*1.25)

xy_t<-
ggplot(xy, aes(x=dist, fill=..x..)) + geom_histogram(color="grey80", size=0.3) + scale_fill_gradient2(low=color[44], high = color[76]) + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(0,15)) + theme(legend.position="none", aspect.ratio = 0.2) + labs(x="r") + scale_x_continuous(limits=c(-1,1)) + geom_vline(aes(xintercept=mean(dist)),
            color="grey70", linetype="dashed", size=0.3)
keep<- c(keep, "xy_t")

result <- pvclust(pvalue_cor, method.dist="euclidean", method.hclust="complete", nboot=10000, parallel=TRUE)


plot(result)
pvrect(result, alpha=0.95)
```

## RRHO2

This approach uses package 'RRHO2' and compares rank ordered gene lists against each-other and uses a hypergeometric test to compare the similarity of gene names in shorter segments along the length gene list. 
```{r RRHO2 mean T, eval=FALSE}
RRHO_objects<- list()

comparisons<- combn(tissue_order, 2)
RRHO_objects<- list()
for(i in 1:ncol(comparisons)){
  comparison<- comparisons[,i]
  p1_name<- comparison[1]
  p2_name<- comparison[2]
  p1<- data.frame(gene=row.names(logpvalue_data), logP_1=logpvalue_data[,p1_name])
  p2<- data.frame(gene=row.names(logpvalue_data), logP_2=logpvalue_data[,p2_name])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  p11<- comb[, c("gene","logP_1")]
  p11<- p11[order(p11$logP_1, decreasing=TRUE),]
  p22<- comb[,c("gene","logP_2")]
  p22<- p22[order(p22$logP_2, decreasing=TRUE),]
  p1_p2<- paste(p1_name,p2_name, sep="_")
  RRHO_objects[[p1_p2]]<- RRHO2_initialize(p11, p22, labels = c(p1_name, p2_name), log10.ind=TRUE, boundary=0.01)
  #jpeg(filename=paste0("RRHO2_meanT",p1_p2,".jpg"))
  RRHO2_heatmap(RRHO_objects[[p1_p2]])
  #dev.off()
}
save(RRHO_objects,file="../Overlap_results/RRHO2/RRHO2_meanT.RData")

```


```{r, eval=FALSE}

load("../Overlap_results/RRHO2/RRHO2_meanT.RData")



maxp<- vector(mode="numeric")
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  maxp[i]<- max(sub, na.rm=TRUE)
}
maxp<- max(maxp)

plots_scaled<- list() # for the plots on the same scale
plots<- list()
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  sub<- melt(sub, na.rm=TRUE)
  x<- word(i,1,1,sep="_")
  y<- word(i,2,2,sep="_")
  plots_scaled[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank(), legend.position="none") + labs(x=x, y=y)
  plots[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis() + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
}


#m = diag(11)
#v=1:66
#upperm = upper.tri(m, diag = T)
#m[upperm] = v
#m<- t(m)
#for some reasion gridExtra puts pane #1 right in the middle of the plot, not at the top... So I'll have to live with filling by column, not by row. 


m <- matrix(NA, 11, 11)
m[lower.tri(m, diag = T)] <- 1:66

## This plot is just to farm the legend. 
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
leg<- get_legend(p)

m1<- m
m1[1,11]<- 67
plots_scaled[["legend"]]<- leg

```

```{r, eval=FALSE}
g<-arrangeGrob(grobs= plots_scaled, layout_matrix = m1)


ggsave(file="../Overlap_results/RRHO2/figure_RRHO_meanT_scaled.pdf", g, units="in", width=20, height=20)


g<-grid.arrange(grobs= plots, layout_matrix = m)
ggsave(file="../Overlap_results/RRHO2/figure_RRHO_meanT_unscaled.pdf", g, units="in", width=25, height=15)
```

I don't know how to get these figures to not look crappy on the markdown so they are available to view as a raw pdf. The scaled PDF is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_meanT_scaled.pdf), and the unscaled is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_meanT_unscaled.pdf).



```{r, eval=FALSE}
## getting the BSTm/Ls figure for the manuscript.
sub<- RRHO_objects[["LS_BSTm"]]$hypermat
sub<- melt(sub, na.rm=TRUE)
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_figure + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(name="log(10)P-value") + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
p
ggsave("../Overlap_results/RRHO2/figure_meanT_results.png", plot=p, device="png", height=1.75, width=2.5, units="in", bg="white")

```


To view pdf versions of the p-value scaled (featured here), and raw unscaled data please visit the main branch of the github repository under "Overlap_results/RRHO2".

## Overall brain patterns

Here I am taking just the brain tissues and taking the median p-value of each gene in the rank-ordered gene list. This will give an idea for similar gene expression patterns across the entire brain.

```{r mean T brain}

brain<- logpvalue_data[,-c(1:2)]

brain$p_combo<- apply(brain,1, median)
brain<- brain[complete.cases(brain),]
brain$gene<- rownames(brain)
brain_go<- merge(brain, go_terms, by="gene", all.x=TRUE)
brain_go<- brain_go[order(brain_go$p_combo, decreasing=TRUE),]
write.csv(brain,"../Overlap_results/results_brain_meanT.csv", row.names=FALSE)
geneList<- brain_go$p_combo
names(geneList)<- brain_go$gene

#subsets<- brain_go[-1.3>brain_go$p_combo,]
#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
#go_mean_T<- y@result
#write.csv(y@result, file="../DE_results/GSEA_BP_brain_mean_T.csv", row.names=FALSE)

#nrow(brain_go[brain_go$p_combo>=1.30103 | brain_go$p_combo<= -1.30103,])

y<- enricher(brain$gene[which(abs(brain$p_combo)>1.3)], universe=brain$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

ggplot(ego2[1:10,], aes(x=Description, y=GeneRatio2, fill=-qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + xlab("") + ylab("No genes enriched") + scale_fill_viridis()

knitr::kable(y@result[1:10,],caption="BP enriched GO terms in the median p in the brain for mean T", row.names = FALSE) %>% kable_styling()

write.csv(y@result,"../Overlap_results/GO_meanT_brain.csv", row.names=FALSE)

ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

```

```{r}
#pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")
genes<- brain[which(abs(brain$p_combo)>1.3),]
genes_GO<- go2gene_bp[which(go2gene_bp$gene %in% genes$gene),]
genes_GO<- genes_GO[which(genes_GO$GO_ID %in% ego2$ID[1:20]),]
genes_GO<- merge(genes_GO, go2name_bp, by="GO_ID")
genes_GO<- merge(genes_GO, genes, by="gene")
genes_GO$gene_GO_combo<- paste0(genes_GO$gene, genes_GO$GO_ID)
genes_GO<- genes_GO[!duplicated(genes_GO$gene_GO_combo),]
genes_GO$pos_neg<- ifelse(genes_GO$p_combo<0, "neg","pos")
genes_GO<- merge(genes_GO, genes_key, by="gene")
colnames(genes_GO)[colnames(genes_GO)=="GO_ID"]<- "ID"
genes_GO<- merge(genes_GO, ego2, by="ID")
#genes_GO2<- genes_GO %>% group_by(name) %>% summarise()


a<- ggplot(data=genes_GO, aes(x=name, y=p_combo, color=pos_neg, alpha=))  + geom_point(size=1) + scale_color_manual(values=c("#0D8CFF","#FF3300")) + coord_flip() + geom_text_repel(aes(label=display_gene_ID), size=geom_text.size) + peri_figure + labs(x="",y="expression direction x log10(p)") + theme(legend.position = "none")
a
ggsave(filename="../Overlap_results/figure_GO_meanT_top.png",a,height=2.60, width=4.60, units="in", device="png", bg="white")
```



```{r}

tophits<- brain
tophits$p_combo<- abs(tophits$p_combo)
tophits<- tophits[order(tophits$p_combo, decreasing=TRUE),]
tophits<- tophits$gene[1:10]
out<- list()
#i="POM"
for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  sub<- tissue[tissue$gene %in% tophits,]
  out[[i]]<- data.frame(gene=sub$gene, lfc=sub$log2FoldChange, p=sub$pvalue, padj=sub$padj, Tissue=i)
  }
status_results_tophits<- do.call("rbind", out)
status_results_tophits<- status_results_tophits[status_results_tophits$Tissue!="GON" & status_results_tophits$Tissue!="PIT",]


outliers=c("PFT2_POM_run1")

brain_behav<- subset(key_behav, Tissue!="GON")
brain_behav<- subset(brain_behav, Tissue!="PIT")
brain_behav<- brain_behav[!rownames(brain_behav) %in% outliers,]
brain_behav<- droplevels(brain_behav)
brain_behav<- brain_behav[which(!is.na(brain_behav$mean_T)),]

brain_data<- data[,colnames(data) %in% rownames(brain_behav)]



start<- nrow(brain_data)
#remove genes with avg read counts <20
brain_data$avg_count<- apply(brain_data, 1, mean)
brain_data<- brain_data[brain_data$avg_count>20,]
brain_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
brain_data$percent_0<- apply(brain_data, 1, function(x)length(x[x==0]))
thresh<- ncol(brain_data)/2
brain_data<- brain_data[brain_data$percent_0<=thresh,]
brain_data$percent_0<-NULL




dd<- DESeqDataSetFromMatrix(countData=brain_data, colData=brain_behav, design= ~ Tissue + mean_T)
dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 

res<- results(dd, alpha=0.1)


gone<- plotCounts(dd, gene=row.names(res[tophits[1],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[tophits[2],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[tophits[3],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[tophits[4],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE)
#gfive<-  plotCounts(dd, gene=row.names(res[tophits[5],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE) #PLIN1 must have met one of the filtering criteria when filtering on all tissues. That's okay I don't look at more than the first two anyway
gsix<-  plotCounts(dd, gene=row.names(res[tophits[6],]), intgroup=c("Tissue","mean_T","Status"), returnData=TRUE) 
vars<- list(gone,gtwo,gthree,gfour,gsix)

plots<- list()
#i=1
for(i in 1:5){
  dat<- vars[[i]]
  name<- tophits[i]
  result<- status_results_tophits[status_results_tophits$gene==name,]
  result$label<- paste0("LFC=",signif(result$lfc,2), "\np(raw)=", formatC(result$p,2,format="e"))
  result$x<- 0.5
  result$y<- max(dat$count)-70
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count)) + geom_point(size=2, aes(color=Status)) + labs(title=paste0(name, " p(median)=", formatC(10^-abs(brain_go$p_combo[brain_go$gene==name]),2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols[2:3]) + facet_wrap(. ~Tissue, ncol=5) + geom_text(data=result, aes(x=x, y=y, label=label))
}

```

```{r, fig.width=7, fig.height=14}
g<- ggarrange(plotlist=plots[1:2],ncol=1, nrow=2, common.legend = TRUE, labels=c("A","B"))
g<- annotate_figure(g, top=textGrob("Top two genes consistently DE in brain with mean T",gp=gpar(fontsize=22)), bottom=textGrob("individual mean testosterone",gp=gpar(fontsize=18)))

g

```

```{r}
ggexport(g, filename = "../Overlap_results/figure_braintop_meanT.png", width = 700,height = 900)
```

```{r}

status_results_tophits<- merge(status_results_tophits, genes_key, by="gene")
#status_results_tophits<- #status_results_tophits[grepl("ROBO4|POLD1|GRINA|LOC113988442|GAMT|FGFR4|LRRC25",status_results_tophits$display_gene_ID),]
status_results_tophits$Tissue<- factor(status_results_tophits$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))

#ggplot(status_results_tophits, aes(x=Tissue,y=lfc, colour=display_gene_ID)) + geom_point(size=0.5) + peri_figure
status_results_tophits$padj2<- ifelse(status_results_tophits$padj<0.1,"yes","no")
#geom_text(aes(label=padj2),size=geom_text.size,color="black") + 
g<- ggplot(status_results_tophits, aes(x=Tissue,y=display_gene_ID, fill=lfc)) + geom_tile(aes(color=padj2)) +  scale_color_manual(values=c("grey50","black")) + scale_fill_gradient2(low="#0D8CFF",high="#FF3300",mid="white") + theme(aspect.ratio = 1, axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())

ggsave("../Overlap_results/figure_braintop10grid_meanT.pdf", plot=g, device="pdf", height=3, width=3, units="in")
```

## GO Overlap

```{r mean TGO overlap}
gos<- list.files(path="../DE_results/", pattern="GO_BP")
gos<- gos[grepl("mean",gos, ignore.case=TRUE)]
setwd("../DE_results/")
gos2<- lapply(gos, read.csv)
setwd("../analysis")

names(gos2)<- gos
names(gos2)<- sub(".csv","", names(gos2))
names(gos2)<- sub("GO_BP_", "",names(gos2))
names(gos2)<- sub("_mean_T", "",names(gos2))
gos2<- mapply(`[<-`, gos2, 'tissue', value = names(gos2), SIMPLIFY = FALSE)
gos3<- do.call("rbind",gos2)
gos3<- gos3[which(gos3$pvalue<0.05),]
terms<- as.data.frame(table(gos3$Description))
colnames(terms)<- c("Description","Freq")
terms<- merge(terms, gos3[,c("ID","Description")], by="Description")
terms<- terms[!duplicated(terms$Description),]
terms<- terms[order(terms$Freq, decreasing=TRUE),]

terms$Description<- factor(terms$Description, levels=terms$Description[order(terms$Freq, decreasing=TRUE)])

ggplot(data=terms[1:20,], aes(x=Description, y=Freq)) + geom_bar(stat="identity", fill="darkturquoise") + theme(axis.text.x=element_text(angle=90)) + coord_flip() + peri_figure


simMatrix <- calculateSimMatrix(gos3$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(terms$Freq, terms$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                               scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
#scatterPlot(simMatrix , reducedTerms)

red<- reducedTerms
colnames(red)[colnames(red)=="go"]<- "ID"
test<- merge(red, terms, by="ID")
write.csv(test,file="../Overlap_results/results_GO_overlap_rrvgo_mean_T.csv", row.names=FALSE)
parents<- test %>% group_by(parentTerm) %>% summarize(parent_size=sum(score))
test<- merge(parents, test, by="parentTerm")
test<- test[order(test$parent_size, decreasing=TRUE),]
tissues<- merge(gos3, test, by="ID")
tissues2<- tissues %>% group_by(parentTerm,tissue) %>% summarise(n_term=(length(unique(ID))))

tissues<- tissues %>% group_by(parentTerm) %>% summarize(n_tissue=length(unique(tissue)), n_terms=length(unique(ID)))

tissues<- tissues[order(tissues$n_tissue, decreasing = TRUE),]
tissues$parentTerm<- factor(tissues$parentTerm, levels=tissues$parentTerm[order(tissues$n_tissue, decreasing = TRUE)])



p<- ggplot(data=tissues[1:20,], aes(x=parentTerm, y=n_tissue, fill=n_terms)) + geom_bar(stat="identity", colour="grey20", linewidth=0.2) + theme(axis.text.x=element_text(angle=90), legend.key.size=unit(3,"mm")) + coord_flip() + peri_figure + scale_fill_distiller(palette="Purples", direction=1, name="number GO terms") + scale_y_continuous(expand=c(0,0), limit=c(0,14), breaks=c(2,4,6,8,10,12)) + labs(x="parent GO term", y="Number of Tissues")
p

ggsave("../Overlap_results/figure_GOparent_overlap_mean_T.pdf", plot=p, device="pdf", height=1.60, width=4.60, units="in")
```



```{r meanT geom_tile}


tissues3<- tissues2[tissues2$parentTerm %in% tissues$parentTerm[which(tissues$n_tissue>8)],]

out<- list()
for(i in tissues3$parentTerm){
  #i="response to hypoxia"
  term<- tissues3[tissues3$parentTerm==i,]
  term<-droplevels(term)
  term$tissue<- as.character(term$tissue)
  if(any(!tissue_order %in% unique(term$tissue))){
    tiss<- tissue_order[!tissue_order %in% unique(term$tissue)]
    out[[i]]<- data.frame(parentTerm=i,tissue=tiss,n_term=NA)
    
  }
  
}
out<- do.call("rbind",out)
tissues3<- rbind(tissues3, out)
tissues3<- merge(tissues3, tissues[,c("parentTerm","n_tissue")],by="parentTerm")

tissues3$tissue<- factor(tissues3$tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS","BSTm"))
tissues3<- tissues3[which(tissues3$n_tissue>9),]

excl<- c("negative regulation of smooth muscle cell migration","positive regulation of vascular smooth muscle cell proliferation","relaxation of cardiac muscle","response to ethanol")

tissues3<- tissues3[!tissues3$parentTerm %in% excl,]

p2<- ggplot(data=tissues3, aes(x=tissue, y=parentTerm, fill=n_term)) + geom_tile(colour="grey50") + scale_fill_viridis(option="A", na.value="white", direction=-1, name="Number Terms p<0.05") + theme(axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())
p2
ggsave("../Overlap_results/figure_GOparent_overlap_mean_T_heatmap.pdf", plot=p2, device="pdf", height=2.0, width=5, units="in")
```


# Social network strength (Cooperative tendency)

## Volcanoes

```{r strength data,  fig.height= 20, fig.width=15}
rm(list= ls()[!(ls() %in% keep)])
setwd("../DE_results/")
files<- list.files(pattern="results_[a-zA-Z]+_Strength.csv", ignore.case=TRUE)
files<- files[-3]
status_results<- lapply(files, read.csv)
names(status_results)<- word(files, 2, sep="_")


output<- list()
plots<- list()
directionalp<- list()
pos_neg<- list()
for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  ## counts of positive and negative genes
  pos_out<- tissue
  pos_out$neg_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$neg_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange<0,1,0)
  pos_out$pos_sig<- ifelse(pos_out$padj<0.1 & pos_out$log2FoldChange>0,1,0)
  pos_out$pos_marg<- ifelse(pos_out$pvalue<0.05 & pos_out$padj>0.1 & pos_out$log2FoldChange>0,1,0)

  pos_neg[[i]]<- data.frame(tissue=i, res=c(-sum(pos_out$neg_sig, na.rm=TRUE), -sum(pos_out$neg_marg, na.rm=TRUE), sum(pos_out$pos_sig, na.rm=TRUE), sum(pos_out$pos_marg, na.rm=TRUE)), type=c("negative significant", "negative marginal", "positive significant","positive marginal"))
  
  ##plots
  tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP","LOC113982601"="AVPR2"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
  
#tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
#tissue$is_interesting<- tissue$interesting_genes==""
#plots[[i]]<- ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + scale_x_continuous(limits=c(-4,4), breaks=c(-4,-3,-2,-1,0,1,2,3,4)) + scale_y_continuous(limits=c(0,6)) + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)
tissue$X<- NULL
tissue$pvalue<- ifelse(is.na(tissue$padj), NA, as.numeric(tissue$pvalue))
sub<- subset(tissue, select=c("gene","log2FoldChange","pvalue"))
tissue<- subset(tissue, select=c("gene","pvalue"))
colnames(tissue)[colnames(tissue)=="pvalue"] <- i
output[[i]]<- tissue

sub$logP<- -log(sub$pvalue, 10)
sub$logP<- ifelse(sub$log2FoldChange<0, -(sub$logP), sub$logP)
sub<- subset(sub, select=c("gene", "logP"))
colnames(sub)[colnames(sub)=="logP"] <- i
directionalp[[i]]<- sub
}
setwd("../analysis/")
```
```{r strength volcano plotting, fig.height= 20, fig.width=15,eval=FALSE}
g<- ggarrange(plotlist=plots,ncol=3, nrow=4, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Volcano plots by social network strength per tissue",gp=gpar(fontsize=22)), bottom=textGrob("LogFoldChange",gp=gpar(fontsize=20)), left=textGrob("-log10(p-adjusted)",gp=gpar(fontsize=20), rot = 90))
g
ggexport(g, filename = "../DE_results/figure_volcanoes_strength.png", width = 1500,height = 1500, pointsize=32)

```


```{r strength summary}
pos_neg_str<- do.call("rbind", pos_neg)
pos_neg_str$type<- factor(pos_neg_str$type, levels=c("negative significant", "negative marginal","positive significant","positive marginal"))
pos_neg_str$tissue<- factor(pos_neg_str$tissue, levels=tissue_order)
pos_neg_str$abs_val<- ifelse(grepl("significant",pos_neg_str$type), abs(pos_neg_str$res), "")
pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")

geom_text.size = (5/14)*(5) 

p<- ggplot(pos_neg_str, aes(x=tissue, y=res, fill=type)) + geom_bar(stat="identity") + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(-1200, 1200)) + labs(title="Status", x="", y="") + scale_fill_manual(values=pos_cols) + geom_hline(yintercept=0)+ theme(legend.position = "none", aspect.ratio = 1) + geom_text(aes(label=abs_val),vjust=pos_neg_str$res < 0, size=geom_text.size) + theme(axis.text.x = element_text(angle=90, vjust=0.5, hjust=1)) + labs(title="social network strength", subtitle=" ", y="number of genes")
p
ggsave("../Overlap_results/figure_strength_bars.pdf", plot=p, device="pdf", height=1.75, width=1.75, units="in")

de_genes_str<- pos_neg_str %>% group_by(tissue) %>% summarize(strength_genes=sum(abs(res)))
de_genes<- merge(de_genes, de_genes_str, by="tissue")
keep<- c(keep, "de_genes")
```

## Pairwise Tissue Similarity

```{r}
temp<- merge(output[[1]],output[[2]],by="gene", all.x=TRUE)
temp<- merge(temp, output[[3]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[4]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[5]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[6]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[7]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[8]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[9]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[10]], by="gene", all.x=TRUE)
temp<- merge(temp, output[[11]], by="gene", all.x=TRUE)
pvalue_data<- merge(temp, output[[12]], by="gene", all.x=TRUE)
rownames(pvalue_data)<- pvalue_data$gene
pvalue_data$gene<- NULL

```

```{r}
temp<- merge(directionalp[[1]],directionalp[[2]],by="gene", all=TRUE)
temp<- merge(temp, directionalp[[3]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[4]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[5]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[6]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[7]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[8]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[9]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[10]], by="gene", all=TRUE)
temp<- merge(temp, directionalp[[11]], by="gene", all=TRUE)
logpvalue_data<- merge(temp, directionalp[[12]], by="gene", all=TRUE)
rownames(logpvalue_data)<- logpvalue_data$gene

logpvalue_data$gene<- NULL
logpvalue_strength<- logpvalue_data
keep<- c(keep, "logpvalue_strength")

pvalue_cor<- cor(logpvalue_data, use= "pairwise.complete.obs", method="pearson")

xy <- t(combn(colnames(pvalue_cor), 2))
xy<- data.frame(xy, dist=pvalue_cor[xy])
#hist(xy$dist, breaks=20) 
#min(xy$dist)
#mean(xy$dist)
ps<-colorRampPalette(brewer.pal(n = 7, name =
  "Purples"))(100)

#color=colorRampPalette(brewer.pal(n = 7, name ="YlOrRd"))(100)
color=rev(colorRampPalette(brewer.pal(n = 7, name =
  "RdBu"))(100))

exp<- data.frame(row.names=colnames(pvalue_data),mean_logP=apply(-log10(pvalue_data),2, mean,na.rm=TRUE))
exp$tissue<- tissue_order
ann_cols<- list(tissue=tissue_colors, mean_logP=ps[1:67])





pheatmap(pvalue_cor, color=color[20:100], annotation_row=exp, border_color = NA, display_numbers = signif(pvalue_cor,1), annotation_colors = ann_cols, clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method="complete")


xy_str<-
ggplot(xy, aes(x=dist, fill=..x..)) + geom_histogram(color="grey80", size=0.3) + scale_fill_gradient2(low=color[20], high = color[82]) + peri_figure + scale_y_continuous(expand=c(0,0), limits=c(0,15)) + theme(legend.position="none", aspect.ratio = 0.2) + labs(x="r") + scale_x_continuous(limits=c(-1,1)) + geom_vline(aes(xintercept=mean(dist)),
            color="grey70", linetype="dashed", size=0.3)
keep<- c(keep, "xy_str")


result <- pvclust(pvalue_cor, method.dist="euclidean", method.hclust="complete", nboot=10000, parallel=TRUE)
plot(result)
pvrect(result, alpha=0.95)
```

## RRHO2

```{r RRHO strength, eval=FALSE}
RRHO_objects<- list()

comparisons<- combn(tissue_order, 2)
RRHO_objects<- list()
for(i in 1:ncol(comparisons)){
  comparison<- comparisons[,i]
  p1_name<- comparison[1]
  p2_name<- comparison[2]
  p1<- data.frame(gene=row.names(logpvalue_data), logP_1=logpvalue_data[,p1_name])
  p2<- data.frame(gene=row.names(logpvalue_data), logP_2=logpvalue_data[,p2_name])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  p11<- comb[, c("gene","logP_1")]
  p11<- p11[order(p11$logP_1, decreasing=TRUE),]
  p22<- comb[,c("gene","logP_2")]
  p22<- p22[order(p22$logP_2, decreasing=TRUE),]
  p1_p2<- paste(p1_name,p2_name, sep="_")
  RRHO_objects[[p1_p2]]<- RRHO2_initialize(p11, p22, labels = c(p1_name, p2_name), log10.ind=TRUE, boundary=0.01)
  jpeg(filename=paste0("RRHO2_strength",p1_p2,".jpg"))
  RRHO2_heatmap(RRHO_objects[[p1_p2]])
  dev.off()
}
save(RRHO_objects,file="../Overlap_results/RRHO2/RRHO2_strength.RData")

```

```{r}
load("../Overlap_results/RRHO2/RRHO2_strength.RData")
maxp<- vector(mode="numeric")
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  maxp[i]<- max(sub, na.rm=TRUE)
}
maxp<- max(maxp)

plots_scaled<- list() # for the plots on the same scale
plots<- list()
for(i in names(RRHO_objects)){
  sub<- RRHO_objects[[i]]$hypermat
  sub<- melt(sub, na.rm=TRUE)
  x<- word(i,1,1,sep="_")
  y<- word(i,2,2,sep="_")
  plots_scaled[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank(), legend.position="none") + labs(x=x, y=y)
  plots[[i]]<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis() + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
}


m <- matrix(NA, 11, 11)
m[lower.tri(m, diag = T)] <- 1:66

## This plot is just to farm the legend. 
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_theme + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(limits=c(0,maxp)) + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
leg<- get_legend(p)

m1<- m
m1[1,11]<- 67
plots_scaled[["legend"]]<- leg
```
```{r, echo=FALSE}
g<-arrangeGrob(grobs= plots_scaled, layout_matrix = m1)


ggsave(file="../Overlap_results/RRHO2/figure_RRHO_strength_scaled.pdf", g, units="in", width=25, height=20)


g<-grid.arrange(grobs= plots, layout_matrix = m)
ggsave(file="../Overlap_results/RRHO2/figure_RRHO_strength_unscaled.pdf", g, units="in", width=25, height=20)
```

The scaled PDF is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_strength_scaled.pdf), and the unscaled is [HERE](https://github.com/periperipatus/PIFI_brain_transcriptome/blob/main/Overlap_results/RRHO2/figure_RRHO_strength_unscaled.pdf).


```{r, eval=FALSE}
## getting the BSTm/Ls figure for the manuscript.
sub<- RRHO_objects[["LS_BSTm"]]$hypermat
sub<- melt(sub, na.rm=TRUE)
p<- ggplot(data=sub, aes(Var1, Var2, fill=value)) + geom_tile() + peri_figure + scale_y_continuous(expand=c(0,0)) + scale_x_continuous(expand=c(0,0)) + scale_fill_viridis(name="log(10)P-value") + theme(axis.text.x=element_blank(),axis.text.y=element_blank(), axis.ticks=element_blank()) + labs(x=x, y=y)
p
ggsave("../Overlap_results/RRHO2/figure_strength_results.png", plot=p, device="png", height=1.75, width=2.5, units="in", bg="white")

```




## Overall brain patterns

```{r}
brain<- logpvalue_data[,-c(1:2)]

brain$p_combo<- apply(brain,1, median)
brain<- brain[complete.cases(brain),]
brain$gene<- rownames(brain)
brain_go<- merge(brain, go_terms, by="gene", all.x=TRUE)
brain_go<- brain_go[order(brain_go$p_combo, decreasing=TRUE),]
write.csv(brain,"../Overlap_results/results_brain_strength.csv", row.names=FALSE)
geneList<- brain_go$p_combo
names(geneList)<- brain_go$gene

#subsets<- brain[-1.3>brain$p_combo,]

#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)

y<- enricher(brain$gene[which(abs(brain$p_combo)>1.3)], universe=brain$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

ggplot(ego2[1:10,], aes(x=Description, y=GeneRatio2, fill=-qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + xlab("") + ylab("No genes enriched") + scale_fill_viridis()

knitr::kable(y@result[1:10,], caption="BP enriched GO terms in the median p in the brain for strength", row.names = FALSE) %>% kable_styling()

write.csv(y@result,"../Overlap_results/GO_strength_brain.csv", row.names=FALSE)

ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
```

```{r}

tophits<- brain
tophits$p_combo<- abs(tophits$p_combo)
tophits<- tophits[order(tophits$p_combo, decreasing=TRUE),]
tophits<- tophits$gene[1:10]
out<- list()
#i="POM"
for(i in tissue_order){
  tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  sub<- tissue[tissue$gene %in% tophits,]
  out[[i]]<- data.frame(gene=sub$gene, lfc=sub$log2FoldChange, p=sub$pvalue, padj=sub$padj, Tissue=i)
  }
status_results_tophits<- do.call("rbind", out)
status_results_tophits<- status_results_tophits[status_results_tophits$Tissue!="GON" & status_results_tophits$Tissue!="PIT",]


outliers=c("PFT2_POM_run1")

brain_behav<- subset(key_behav, Tissue!="GON")
brain_behav<- subset(brain_behav, Tissue!="PIT")
brain_behav<- brain_behav[!rownames(brain_behav) %in% outliers,]
brain_behav<- droplevels(brain_behav)
brain_behav<- brain_behav[which(!is.na(brain_behav$strength.all_study)),]

brain_data<- data[,colnames(data) %in% rownames(brain_behav)]



start<- nrow(brain_data)
#remove genes with avg read counts <20
brain_data$avg_count<- apply(brain_data, 1, mean)
brain_data<- brain_data[brain_data$avg_count>20,]
brain_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
brain_data$percent_0<- apply(brain_data, 1, function(x)length(x[x==0]))
thresh<- ncol(brain_data)/2
brain_data<- brain_data[brain_data$percent_0<=thresh,]
brain_data$percent_0<-NULL
#13652 genes.
brain_behav$strength.all_study<- scale(brain_behav$strength.all_study)
dd<- DESeqDataSetFromMatrix(countData=brain_data, colData=brain_behav, design= ~ Tissue + strength.all_study)
dd<- DESeq(dd)
#dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 

res<- results(dd, alpha=0.1)


gone<- plotCounts(dd, gene=row.names(res[tophits[1],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[tophits[2],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[tophits[3],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[tophits[4],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[tophits[5],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[tophits[6],]), intgroup=c("Tissue","strength.all_study","Status"), returnData=TRUE)
vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()
#i=1
for(i in 1:6){
  dat<- vars[[i]]
  name<- tophits[i]
  result<- status_results_tophits[status_results_tophits$gene==name,]
  result$label<- paste0("LFC=",signif(result$lfc,2), "\np(raw)=", formatC(result$p,2,format="e"))
  result$x<- 0.5
  result$y<- max(dat$count)-100
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=2, position=dodge, aes(color=Status)) + labs(title=paste0(name, " p(median)=", formatC(10^-abs(brain_go$p_combo[brain_go$gene==name]),2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols[2:3]) + facet_wrap(. ~Tissue, ncol=5) + geom_text(data=result, aes(x=x, y=y, label=label))
}

```


```{r, fig.width=7, fig.height=14}
g<- ggarrange(plotlist=plots[1:2],ncol=1, nrow=2, common.legend = TRUE, labels=c("A","B"))
g<- annotate_figure(g, top=textGrob("Top two genes consistently DE in brain with strength",gp=gpar(fontsize=22)), bottom=textGrob("social network strength",gp=gpar(fontsize=18)))

g
ggexport(g, filename = "../Overlap_results/figure_braintop_strength.png", width = 700,height = 900)
```

```{r}
#pos_cols<- c("#0D8CFF", "#AED9FF","#FF3300","#FFBBAA")
genes<- brain[which(abs(brain$p_combo)>1.3),]
genes_GO<- go2gene_bp[which(go2gene_bp$gene %in% genes$gene),]
genes_GO<- genes_GO[which(genes_GO$GO_ID %in% ego2$ID[1:20]),]
genes_GO<- merge(genes_GO, go2name_bp, by="GO_ID")
genes_GO<- merge(genes_GO, genes, by="gene")
genes_GO$gene_GO_combo<- paste0(genes_GO$gene, genes_GO$GO_ID)
genes_GO<- genes_GO[!duplicated(genes_GO$gene_GO_combo),]
genes_GO$pos_neg<- ifelse(genes_GO$p_combo<0, "neg","pos")
genes_GO<- merge(genes_GO, genes_key, by="gene")
colnames(genes_GO)[colnames(genes_GO)=="GO_ID"]<- "ID"
genes_GO<- merge(genes_GO, ego2, by="ID")
#genes_GO2<- genes_GO %>% group_by(name) %>% summarise()


a<- ggplot(data=genes_GO, aes(x=name, y=p_combo, color=pos_neg, alpha=))  + geom_point(size=1) + scale_color_manual(values=c("#0D8CFF","#FF3300")) + coord_flip() + geom_text_repel(aes(label=display_gene_ID), size=geom_text.size) + peri_figure + labs(x="",y="expression direction x log10(p)") + theme(legend.position = "none")
a
ggsave(filename="../Overlap_results/figure_GO_strength_top.png",a,height=2.60, width=4.60, units="in", device="png", bg="white")
```

```{r}

status_results_tophits<- merge(status_results_tophits, genes_key, by="gene")
#status_results_tophits<- #status_results_tophits[grepl("ROBO4|POLD1|GRINA|LOC113988442|GAMT|FGFR4|LRRC25",status_results_tophits$display_gene_ID),]
status_results_tophits$Tissue<- factor(status_results_tophits$Tissue, levels=c("VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))

#ggplot(status_results_tophits, aes(x=Tissue,y=lfc, colour=display_gene_ID)) + geom_point(size=0.5) + peri_figure
status_results_tophits$padj2<- ifelse(status_results_tophits$padj<0.1,"yes","no")
#geom_text(aes(label=padj2),size=geom_text.size,color="black") + 
g<- ggplot(status_results_tophits, aes(x=Tissue,y=display_gene_ID, fill=lfc)) + geom_tile(aes(color=padj2)) +  scale_color_manual(values=c("grey50","black")) + scale_fill_gradient2(low="#0D8CFF",high="#FF3300",mid="white") + theme(aspect.ratio = 1, axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())

ggsave("../Overlap_results/figure_braintop10grid_strength.pdf", plot=g, device="pdf", height=3, width=3, units="in")
```

```{r plotting BSTm, eval=FALSE}
BSTm_modules<- read.csv("../WGCNA/ByTissue_2021/BSTm_results_ModuleMembership.csv")


i="BSTm"
 tissue<- status_results[[i]]
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes, as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, BSTm_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","darkmagenta","salmon4","blue",        "saddlebrown","brown","darkred","coral2" ))
modulecols<- c("grey30","grey60","darkmagenta","salmon4","blue",        "saddlebrown","brown","darkred","coral2")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")



#ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=0.0)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none")


```

## GO Overlap

```{r Strength GO overlap}
gos<- list.files(path="../DE_results/", pattern="GO_BP")
gos<- gos[grepl("strength",gos, ignore.case=TRUE)]
setwd("../DE_results/")
gos2<- lapply(gos, read.csv)
setwd("../analysis")

names(gos2)<- gos
names(gos2)<- sub("_strength.csv","", names(gos2),ignore.case=TRUE)
names(gos2)<- sub("GO_BP_", "",names(gos2))

gos2<- mapply(`[<-`, gos2, 'tissue', value = names(gos2), SIMPLIFY = FALSE)
gos3<- do.call("rbind",gos2)
gos3<- gos3[which(gos3$pvalue<0.05),]
terms<- as.data.frame(table(gos3$Description))
colnames(terms)<- c("Description","Freq")
terms<- merge(terms, gos3[,c("ID","Description")], by="Description")
terms<- terms[!duplicated(terms$Description),]
terms<- terms[order(terms$Freq, decreasing=TRUE),]

terms$Description<- factor(terms$Description, levels=terms$Description[order(terms$Freq, decreasing=TRUE)])

ggplot(data=terms[1:20,], aes(x=Description, y=Freq)) + geom_bar(stat="identity", fill="darkturquoise") + theme(axis.text.x=element_text(angle=90)) + coord_flip() + peri_figure


simMatrix <- calculateSimMatrix(gos3$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(terms$Freq, terms$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                               scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
#scatterPlot(simMatrix , reducedTerms)

red<- reducedTerms
colnames(red)[colnames(red)=="go"]<- "ID"
test<- merge(red, terms, by="ID")
write.csv(test,file="../Overlap_results/results_GO_overlap_rrvgo_strength.csv", row.names=FALSE)
parents<- test %>% group_by(parentTerm) %>% summarize(parent_size=sum(score))
test<- merge(parents, test, by="parentTerm")
test<- test[order(test$parent_size, decreasing=TRUE),]
tissues<- merge(gos3, test, by="ID")
tissues2<- tissues %>% group_by(parentTerm,tissue) %>% summarise(n_term=(length(unique(ID))))

tissues<- tissues %>% group_by(parentTerm) %>% summarize(n_tissue=length(unique(tissue)), n_terms=length(unique(ID)))

tissues<- tissues[order(tissues$n_tissue, decreasing = TRUE),]
tissues$parentTerm<- factor(tissues$parentTerm, levels=tissues$parentTerm[order(tissues$n_tissue, decreasing = TRUE)])



p<- ggplot(data=tissues[1:20,], aes(x=parentTerm, y=n_tissue, fill=n_terms)) + geom_bar(stat="identity", colour="grey20", linewidth=0.2) + theme(axis.text.x=element_text(angle=90), legend.key.size=unit(3,"mm")) + coord_flip() + peri_figure + scale_fill_distiller(palette="Purples", direction=1, name="number GO terms") + scale_y_continuous(expand=c(0,0), limit=c(0,14), breaks=c(2,4,6,8,10,12)) + labs(x="parent GO term", y="Number of Tissues")
p

ggsave("../Overlap_results/figure_GOparent_overlap_strength.pdf", plot=p, device="pdf", height=1.60, width=4.60, units="in")
```


```{r strength geom_tile}


tissues3<- tissues2[tissues2$parentTerm %in% tissues$parentTerm[which(tissues$n_tissue>7)],]

out<- list()
for(i in tissues3$parentTerm){
  #i="response to hypoxia"
  term<- tissues3[tissues3$parentTerm==i,]
  term<-droplevels(term)
  term$tissue<- as.character(term$tissue)
  if(any(!tissue_order %in% unique(term$tissue))){
    tiss<- tissue_order[!tissue_order %in% unique(term$tissue)]
    out[[i]]<- data.frame(parentTerm=i,tissue=tiss,n_term=NA)
    
  }
  
}
out<- do.call("rbind",out)
tissues3<- rbind(tissues3, out)
tissues3<- merge(tissues3, tissues[,c("parentTerm","n_tissue")],by="parentTerm")

tissues3$tissue<- factor(tissues3$tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","TNA","AI","LS","BSTm"))
#tissues3<- tissues3[which(tissues3$n_tissue>=8),]

excl<- c("protein glycosylation","protein homotetramerization","negative regulation of cell death","negative regulation of epithelial cell proliferation","phosphatidylinositol-mediated signaling","cardiac conduction","cellular amino acid metabolic process","intermembrane lipid transfer","cell adhesion","negative regulation of angiogenesis")

tissues3<- tissues3[!tissues3$parentTerm %in% excl,]

p2<- ggplot(data=tissues3, aes(x=tissue, y=parentTerm, fill=n_term)) + geom_tile(colour="grey50") + scale_fill_viridis(option="A", na.value="white", direction=-1, name="Number Terms p<0.05") + theme(axis.ticks=element_blank()) + labs(x="",y="") + theme(  axis.text.x = element_text(size =5, colour = "black", angle=90), axis.text.y = element_text(size = 5, colour = "black"),legend.text = element_text(size = 5), legend.title = element_text(size =5),legend.key.size=unit(3,"mm"),panel.background = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), plot.background = element_blank())
p2
ggsave("../Overlap_results/figure_GOparent_overlap_strength_heatmap.pdf", plot=p2, device="pdf",height=2.0, width=5, units="in")
```


# Comparing all analyses

```{r number of DE genes}
meanT_status<- cor.test(de_genes$status_genes, de_genes$meanT_genes)
meanT_strength<- cor.test(de_genes$meanT_genes, de_genes$strength_genes)
status_strength<- cor.test(de_genes$status_genes, de_genes$strength_genes)

#cors<- data.frame(comparison=c("meanT_status", "meanT_strength", "status_strength"), cor=c(meanT_status$estimate, meanT_strength$estimate, status_strength$estimate), p=c(meanT_status$p.value, meanT_strength$p.value, status_strength$p.value))

a<-ggplot(de_genes, aes(x=status_genes, y=meanT_genes, color=tissue)) + geom_point() + peri_figure + labs(title="Status x mean T",  subtitle=paste0("R=",signif(meanT_status$estimate,2), ", p=", signif(meanT_status$p.value,2))) + theme(aspect.ratio=1) + scale_y_continuous(limits=c(200,1500)) + labs(x="number DEGs by social status", y="number DEGs by mean testosterone")
b<-ggplot(de_genes, aes(x=meanT_genes, y=strength_genes, color=tissue)) + geom_point() + peri_figure + labs(title="mean T x Strength",  subtitle=paste0("R=",signif(meanT_strength$estimate,2), ", p=", signif(meanT_strength$p.value,2))) + theme(aspect.ratio = 1)+ scale_y_continuous(limits=c(200,1500))+ labs(x="number DEGs by mean testosterone", y="number DEGs by social network strength")
c<-ggplot(de_genes,aes(x=status_genes, y=strength_genes, color=tissue)) + geom_point() + peri_figure + labs(title="Status x Strength",  subtitle=paste0("R=",signif(status_strength$estimate,2), ", p=", signif(status_strength$p.value,2))) + theme(aspect.ratio = 1)+ scale_y_continuous(limits=c(200,1500)) + labs(x="number DEGs by social status", y="number DEGs by social network strength")

corrs<- ggarrange(a,b,c, ncol=3, common.legend = TRUE)
ggsave("../Overlap_results/figure_corr_degs.png", plot=corrs, device="png", bg="white",width=1500,height=800,units="px")
#ggsave("../Overlap_results/figure_corr_degs_sml.pdf", plot=corrs, device="pdf", height=1.15, width=2.445, units="in")
```

```{r landscape landscapes, eval=FALSE}

meanT_status_out<- list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_meanT), mean_T=logpvalue_meanT[,i])
  p2<- data.frame(gene=row.names(logpvalue_status), status=logpvalue_status[,i])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  meanT_status_out[[i]]<- cor(comb$mean_T, comb$status, method="pearson")
}
meanT_status_out<- as.data.frame(do.call("rbind", meanT_status_out))

meanT_strength_out<- list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_meanT), mean_T=logpvalue_meanT[,i])
  p2<- data.frame(gene=row.names(logpvalue_strength), strength=logpvalue_strength[,i])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  meanT_strength_out[[i]]<- cor(comb$mean_T, comb$strength, method="pearson")
}
meanT_strength_out<- as.data.frame(do.call("rbind", meanT_strength_out))

status_strength_out<- list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_status), status=logpvalue_status[,i])
  p2<- data.frame(gene=row.names(logpvalue_strength), strength=logpvalue_strength[,i])
  comb<- merge(p1,p2)
  comb<- comb[complete.cases(comb),]
  status_strength_out[[i]]<- cor(comb$status, comb$strength, method="pearson")
}
status_strength_out<- as.data.frame(do.call("rbind",status_strength_out))

gene_correlations<- data.frame(tissue=c(rep(tissue_order,3)), comparison=c(rep("meanT_status",12), rep("meanT_strength",12), rep("status_strength",12)), R=c(meanT_status_out$V1,meanT_strength_out$V1, status_strength_out$V1))
gene_correlations$R<- as.numeric(gene_correlations$R)                         

gene_correlations$tissue<- factor(gene_correlations$tissue,levels=tissue_order)

p<- ggplot(gene_correlations, aes(x=tissue, y=R, fill=comparison)) + geom_bar(stat="identity",position=position_dodge(width=0.9),colour="grey20",linewidth=0.2) + peri_figure + theme(legend.key.size = unit(3,"mm"), axis.text.x=element_text(angle=90)) +scale_fill_viridis(discrete=TRUE)+theme(panel.grid.major.x = element_line(color="grey80", linetype="dashed",linewidth=0.2))

ggsave("../Overlap_results/figure_corr_lists.png", plot=p, device="png", bg="white",width=900,height=800,units="px")
ggsave("../Overlap_results/figure_corr_lists.pdf", plot=p, device="pdf", bg="white",height=2.0, width=3.5, units="in")
```

```{r landscape landscapes1, eval=FALSE}


status_meanT_shared<-list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_meanT), mean_T=logpvalue_meanT[,i])
  p1<- p1[which(p1$mean_T>abs(1.3)),]
  p2<- data.frame(gene=row.names(logpvalue_status), status=logpvalue_status[,i])
  p2<- p2[which(p2$status>abs(1.3)),]

    status_meanT_shared[[i]]<- data.frame(tissue=i,n_shared=length(p1$gene[p1$gene %in% p2$gene]), n_p1=length(p1$gene),n_p2=length(p2$gene),comparison="meanT_status")
}

status_meanT_shared<- do.call("rbind",status_meanT_shared)

meanT_strength_out<- list()
meanT_strength_shared<- list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_meanT), mean_T=logpvalue_meanT[,i])
  p1<- p1[which(p1$mean_T>abs(1.3)),]
  p2<- data.frame(gene=row.names(logpvalue_strength), strength=logpvalue_strength[,i])
  p2<- p2[which(p2$strength>abs(1.3)),]
 
    meanT_strength_shared[[i]]<- data.frame(tissue=i,n_shared=length(p1$gene[p1$gene %in% p2$gene]), n_p1=length(p1$gene),n_p2=length(p2$gene),comparison="meanT_strength")
}
meanT_strength_shared<- do.call("rbind",meanT_strength_shared)

status_strength_out<- list()
status_strength_shared<- list()
for(i in tissue_order){
  #i="GON"
  comparison<- tissue_order[i]
  p1<- data.frame(gene=row.names(logpvalue_status), status=logpvalue_status[,i])
  p1<- p1[which(p1$status>abs(1.3)),]
  p2<- data.frame(gene=row.names(logpvalue_strength), strength=logpvalue_strength[,i])
  p2<- p2[which(p2$strength>abs(1.3)),]

  
  status_strength_shared[[i]]<- data.frame(tissue=i,n_shared=length(p1$gene[p1$gene %in% p2$gene]), n_p1=length(p1$gene),n_p2=length(p2$gene),comparison="status_strength")
}
status_strength_shared<- do.call("rbind",status_strength_shared)              
genes_shared<- rbind(status_meanT_shared, meanT_strength_shared,status_strength_shared)              

genes_shared$tissue<- factor(genes_shared$tissue,levels=tissue_order)

genes_shared$perc<- ((genes_shared$n_shared)/(genes_shared$n_p1+genes_shared$n_p2))*100

p<- ggplot(genes_shared, aes(x=tissue, y=perc, fill=comparison)) + geom_bar(stat="identity",position=position_dodge(width=0.9),colour="grey20",linewidth=0.2) + peri_figure + theme(legend.key.size = unit(3,"mm"), axis.text.x=element_text(angle=90)) +scale_fill_viridis(discrete=TRUE)+theme(panel.grid.major.x = element_line(color="grey80", linetype="dashed",linewidth=0.2)) + labs(y="percent p<0.05 genes shared",x="")

ggsave("../Overlap_results/figure_corr_lists_sig.png", plot=p, device="png", bg="white",width=900,height=800,units="px")
ggsave("../Overlap_results/figure_corr_lists_sig.pdf", plot=p, device="pdf", bg="white",height=2.0, width=3.5, units="in")
```


```{r}

str<- read.csv("../Overlap_results/results_GO_overlap_rrvgo_strength.csv")
str<- str[!duplicated(str$parentTerm),]
t<- read.csv("../Overlap_results/results_GO_overlap_rrvgo_mean_T.csv")
t<- t[!duplicated(t$parentTerm),]
st<- read.csv("../Overlap_results/results_GO_overlap_rrvgo_status.csv")
st<- st[!duplicated(st$parentTerm),]

gl<- list(strength=str$parentTerm,mean_T=t$parentTerm,status=st$parentTerm)
library(ggvenn)
library(viridis)

g<- ggvenn(gl, stroke_size=0.2, set_name_size=2,fill_color=c("#0073C2FF", "#EFC000FF", "#CD534CFF"),text_size=1) 
ggsave(filename="../Overlap_results/figure_GO_overlap_all.pdf", plot=g, device="pdf", height=1.60, width=1.60, units="in")


common<- str$parentTerm[which(str$parentTerm %in% t$parentTerm)]
common<- common[which(common %in% st$parentTerm)]
common
```


