---
title: "DESeq2 and PCA Analysis"
subtitle: "For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird "
date: Last Knit "`r Sys.Date()`"
author: "Peri Bolton"
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
    
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning=FALSE)
```

```{r libraries used, echo=FALSE}
library(plyr)
library(reshape2)
library(ggplot2)
library(gridExtra)
library(ggpubr)
library(PCAtools)
library(fdrtool)
library(stringr)
library(lubridate)
library(DESeq2)
library(kableExtra)
library(VennDiagram)
library(wesanderson)
library(WGCNA)
library(clusterProfiler)
library(org.Hs.eg.db)
library(flashClust)
library(limma)
library(UpSetR)
library(rrvgo)
library(EnhancedVolcano)
library(viridis)
  
peri_theme <- theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
    axis.title.x = element_text(size = 12), axis.text.x = element_text(size = 10, 
        colour = "black"), axis.title.y = element_text(size = 12), 
    axis.text.y = element_text(size = 10, colour = "black"), 
    legend.text = element_text(size = 10), legend.title = element_text(size = 12), 
    axis.line.y = element_line(colour = "black"), axis.line.x = element_line(colour = "black"),
    plot.title = element_text(hjust = 0.5, size=12), plot.subtitle = element_text(hjust = 0.5, size=12))  + theme(legend.key=element_blank())
dodge=position_dodge(0.8)


#remove outliers from continuous datasets in DESeq2
rm_continuous_outliers<- function(deseq=x, results=y, pct=0.98){
  cooks<- as.data.frame(assays(deseq)[["cooks"]])
  p=ncol(model.matrix(design(deseq), colData(deseq)))
  n=nrow(colData(deseq))
  cd_cutoff<- qf(pct,p,n-p)
  out<- as.data.frame(t(apply(cooks, 1, function(x){ x > cd_cutoff })))
  out$outlier<- apply(out,1, any)
  outliers<- rownames(out)[out$outlier==TRUE]
  results$padj[rownames(results) %in% outliers]<-NA
  results$pvalue[rownames(results) %in% outliers]<-NA
  return(results)
}

```


```{r reading & splitting data, echo=TRUE}

key<- read.csv("../data_filtered/data_key_Parsed_ReplicatesRemoved.csv")
behav<- read.csv("../data_unfiltered/PIPFIL_T_and_behav_data.csv")
rownames(key)<- key$X

key$Color_ID<- sub("/","", key$Color_ID)
key<- plyr::rename(key, replace=c("Color_ID"="colorID"))
behav$status<- plyr::revalue(behav$status, c("floa"="floater", "terr"="territorial"))
key_behav<- merge(key, behav, by="colorID")
key_behav<- key_behav[!is.na(key_behav$last_behav),]


#create a data.frame with all of the observations. 
not_in_behav<- key[!key$X %in% key_behav$X,]
cols_not_key<- colnames(key_behav)[!colnames(key_behav) %in% colnames(key)]
cols_not_key_df<- data.frame(matrix(NA, nrow = nrow(not_in_behav), ncol = length(cols_not_key)))
colnames(cols_not_key_df)<- cols_not_key
not_in_behav<- cbind(not_in_behav, cols_not_key_df)

key_behav<- rbind(key_behav, not_in_behav)

rownames(key_behav)<- key_behav$X


key_behav$Tissue<- factor(key_behav$Tissue, levels=c("GON","PIT","VMH","AH","PVN","POM","ICO","GCT","AI","TNA","LS", "BSTm"))
key_behav<- key_behav[order(rownames(key_behav)),]
key_behav$Class<- as.factor(key_behav$Class)
key_behav$Class<- revalue(key_behav$Class, replace=c("SCB floater"="Predefinitive floater", "DCB floater "="Definitive floater", "DCB territorial"="Territorial"))
key_behav$Class<- factor(key_behav$Class, levels=c("Predefinitive floater", "Definitive floater", "Territorial"))
rownames(key_behav)<- key_behav$sampleID
key_behav<- key_behav[order(key_behav$sampleID),]
key_behav$Year<- as.factor(key_behav$Year)
key_behav$Batch<- as.factor(key_behav$Batch)
key_behav$Status<- as.factor(key_behav$Status)

key_behav_unique<- key_behav[!duplicated(key_behav$colorID),]


#what individuals are missing? 
#key$in_behav<- key$Harvest_ID %in% behav$Harvest_ID
#key[key$in_behav==FALSE, c("colorID", "Harvest_ID")]
#only unbanded individuals are missing from the larger behavioural dataset. 

#check that status matches between datasets
#key_behav$status_match<- ifelse(key_behav$status==key_behav$Status, "same","not")
#nrow(key_behav[key_behav$status_match=="not",]) #0 GOOD!






#read the raw count data
data<- read.csv("../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv")

rownames(data)<- data$X
data$X<- NULL

### Gene Ontology

go_key<- read.csv("../GO_annotations/Maggies_annotations_modifiedR.csv")
go_key<- plyr::rename(go_key, replace=c("GeneID"="gene"))

go_terms<- read.csv("../GO_annotations/pfil_GO_key_raw.csv")
go_terms<- plyr::rename(go_terms, replace=c("GeneID"="gene"))
go2gene_bp<- go_terms[which(go_terms$Aspect=="P"),c("GO_ID", "gene")]
go2gene_mf<- go_terms[which(go_terms$Aspect=="F"),c("GO_ID", "gene")]
go_obo<- read.csv("../GO_annotations/ontology_obo_out.csv")
go_obo<- plyr::rename(go_obo, replace=c("id"="GO_ID"))
go2name_bp<- go_obo[which(go_obo$namespace=="biological_process"),c("GO_ID", "name")]
go2name_mf<- go_obo[which(go_obo$namespace=="molecular_function"),c("GO_ID", "name")]

```



# Exploration of the behavioural data

Our primary aim is to explore whether among-individual variation in T modulated cooperative behaviour is associated with differences in gene expression.  We have 5 main traits that were measured.

* Territorial or Floater
* Testosterone concentration
  + *mean_T* is best estimate of among individual var in T across all birds ('r summary(behav$mean_T)`), or 
* Strength
 + Mean number of social interactions per 10 hr day


ADJ metrics are those as per above but have been adjusted for seasonal and temperature variation.


we were a little concerned about the signal from the last few days outweighing the signal from their overall phenotype, so we will explore how different these measures are from eachother.

```{r variable correlations}
dj_col<- c("#00A08A", "#F2AD00")
status_cols<- c("#86AD44","#E54849","#414042")
status_cols<- c("#cfc451","#E54849","#414042")


s1<- ggplot(behav, aes(x=strength.all_study, y=strength.sac_year)) + geom_point(color="#00A08A", size=4) + labs(y="Strength: sacrifice year", x="Strength: all study") + peri_theme
s2<- ggplot(behav, aes(x=strength.all_study, y=strength.last_day)) + geom_point(color="#00A08A", size=4) + labs(y="Strength: last day before sacrifice", x="Strength: all study") + peri_theme


g<- ggarrange(s1, s2, ncol=1, nrow=2)
g<- annotate_figure(g, top = textGrob("Social Network Strength over measurement periods",gp=gpar(fontsize=14)))
g


```



# Associations between variables


```{r}
library(GGally)
 

ggpairs(key_behav_unique[,c("mean_T","effort.all_study","degree.all_study", "strength.all_study")], aes(alpha = 0.4)) + theme_bw() 


ggpairs(key_behav_unique[,c("Status","mean_T","strength.all_study")], aes(colour = Status, alpha = 0.4)) + theme_bw()

```

Given the extremely highly significant relationship between degree and strength, I will only keep one of these variables - the strength. This was the most repeatable of the two sociality variables in Ryder et al. (2020).

All significance tests are based on just territorial and floater, but colour-coded according to plumage*status.

```{r cor status, fig.height=10, fig.width=10}
mean_t_wtest<- wilcox.test(key_behav_unique$mean_T ~ key_behav_unique$Status, alternative="less")
a<- ggplot(key_behav_unique, aes(x=Status, y=mean_T, color=Class)) + geom_point(size=4) + labs(title=paste0("Status and mean T, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme + scale_color_manual(values=status_cols)


strength_wtest<- wilcox.test(key_behav_unique$strength.all_study ~ key_behav_unique$status, alternative="less")
d<- ggplot(key_behav_unique, aes(x=status, y=strength.all_study, color=Class)) + geom_point(size=4, position=dodge) + labs(title=paste0("Status and Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme + scale_color_manual(values=status_cols)

#tests<- ggarrange(a,d, ncol=2, common.legend=TRUE)
#tests<- annotate_figure(tests, top = textGrob("Testing relationships between Status and endocrine and behavioral traits",gp=gpar(fontsize=14)))
#tests


blah<- key[!duplicated(key$Harvest_ID),]
blah<- as.data.frame(table(key_behav_unique$Class))
blah$Status<- c("floater","floater", "territorial")
e<- ggplot(blah, aes(x=Status,y=Freq, fill=Var1)) + geom_bar(stat="identity", position="stack") + scale_fill_manual(values=status_cols) + scale_y_continuous(expand=c(0,0), breaks=c(0,2,4,6,8)) + peri_theme
tests<- ggarrange(e, a, d, ncol=3, common.legend=TRUE)
tests


```

```{r mean T and behav, fig.height=10,fig.width=10}


#summary(lm(key_behav_unique$strength.all_study ~ key_behav_unique$mean_T))
b<- ggplot(key_behav_unique, aes(x=mean_T, y=strength.all_study, color=Class)) + geom_point(size=4) + labs(title="lm slope=11.45,p=0.12", x="mean Testosterone", y="social network strength") + peri_theme + scale_color_manual(values=status_cols) + geom_text(aes(label=colorID), hjust=1.1)
b

```


In the previous QC exploration I found substantial DEGs according to collection Year and Sequencing Batch. Here we explore how the continuous variables are distributed across sampling years. 



```{r variables according to Year, fig.height=10, fig.width=10, echo=FALSE}

mean_t_wtest<- wilcox.test(key_behav$mean_T ~ key_behav$Year, alternative="two.sided")
a<- ggplot(droplevels(key_behav), aes(x=Year, y=mean_T))  + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme


strength_wtest<- wilcox.test(key_behav$strength.all_study ~ key_behav$Year, alternative="two.sided")
d<- ggplot(key_behav, aes(x=Year, y=strength.all_study)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme


g<- ggarrange(a,d,ncol=2, nrow=1, common.legend = TRUE)
g<- annotate_figure(g,top = textGrob("How interest variables are distributed across Year",gp=gpar(fontsize=14,font=3)))
g


```

# All tissues PCA

Plotting all the tissues

```{r all filtering, echo=TRUE} 
all_data<- data

start<- nrow(all_data)
#remove genes with avg read counts <20
all_data$avg_count<- apply(all_data, 1, mean)
all_data<- all_data[all_data$avg_count>20,]
all_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
all_data$percent_0<- apply(all_data, 1, function(x)length(x[x==0]))
thresh<- ncol(all_data)/2
all_data<- all_data[all_data$percent_0<=thresh,]
all_data$percent_0<-NULL
#13652 genes.
```



```{r}
dd<- DESeqDataSetFromMatrix(countData=all_data, colData=key_behav, design= ~ Tissue)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 

#plotDispEsts(dd)

#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables
vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
p <- pca(vsd_data, metadata = key_behav)

tissue_cols<- c("GON"="#2A9D8F","PIT"="#2A9D8F","VMH"="#E76F51","AH"="#E76F51","PVN"="#E76F51", "POM"="#E76F51", "ICO"="#E76F51","GCT"="#E76F51", "TNA"="#E76F51", "AI"="#f0d497", "LS"="#F4A261", "BSTm"="#F4A261")

biplot(p, lab=NULL, colby="Tissue", shape="Tissue", legendPosition="none", title="", colkey=tissue_cols, pointSize = 4.5) + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + scale_shape_manual(values=c(16,17,rep(15,10)))


eigencorplot(p, metavars = c("Tissue","time_kill","Batch","Year","mean_T", "strength.all_study","Status"), col=blueWhiteRed(50))

```


```{r}
a<- biplot(p, lab=NULL, colby="Tissue", shape="Batch", legendPosition="right", title="") + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + theme(plot.margin = unit(c(0,1,0,1), "lines"))


b<-biplot(p, x="PC1",y="PC3", lab=NULL, colby="Tissue", shape="Batch", legendPosition="right", title="") + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + theme(plot.margin = unit(c(0,1,0,1), "lines"))

c<- eigencorplot(p, metavars = c("Tissue","time_kill","Batch","Year","mean_T", "strength.all_study","Status"), col=blueWhiteRed(50))


multiplot<- ggarrange(ggarrange(a,b,ncol=2, labels=c("A","B"), common.legend = TRUE, legend="right")           
          ,c,
          nrow=2, 
          labels = c("","C")) 
multiplot
#ggsave(filename="figure_alltissues_PCA.pdf", device="pdf",width =8.5 ,height=8, units="in")
#ggsave(filename="figure_alltissues_PCA.png", device="png",width =8.5 ,height=8, units="in")
```

```{r all tissues pc7}
library(nlme)



test<- data.frame(PC7=p$rotated$PC7)
test<- cbind(test, key_behav)

#summary(lm(PC5 ~ Year + Tissue*Status,data=test ))
m1<- lme(PC7 ~ Year + Tissue*Status, random=~1|Harvest_ID, data=test)
pc5test<- anova(m1)


#pc5test<- summary(aov(test$PC5 ~ test$Year + test$Status))
pc5test_anova <- ggtexttable(signif(pc5test,2))
#,  theme = ttheme("mOrange"))




e<-biplot(p, x="PC1",y="PC7", lab=NULL, colby="Tissue", shape="Status", legendPosition="right", title="") + peri_theme
f<-ggplot(test, aes(x=Status, y=PC7)) + geom_boxplot() + geom_jitter()+ peri_theme  + theme(legend.position="none") 

ggarrange(e,ggarrange(pc5test_anova,f, nrow=2), ncol=2, nrow=1,labels=c(LETTERS[1:2]))
#ggsave(filename="figure_alltissues_PC7.pdf", device="pdf",height =4 ,width=8, units="in")
#ggsave(filename="figure_alltissues_PC7.png", device="png",height =4 ,width=8, units="in")
```


```{r loadings all tissues}

loadings<- p$loadings
loadings$gene<- rownames(loadings)
loadings<- merge(loadings, go_key, by="gene")


plots<- list()
pcs<- paste0("PC",1:10)
for(i in pcs){
  #i="PC2"
  sub<- loadings[,c(i, "best_anno","gene")]
  sub<- sub[order(sub[,i]),]
  top30<- rbind(head(sub,15), tail(sub,15))
  top30$best_anno2<- factor(top30$best_anno, levels=unique(top30$best_anno[order(top30[,i],top30$best_anno)]), ordered=TRUE)
  top30$pos<- ifelse(top30[,i]<0, "neg", "pos")
  plots[[i]]<-ggplot(top30, aes(x=best_anno2, y=top30[,i], fill=pos)) + geom_bar(stat="identity")+  peri_theme + coord_flip() + labs(x="gene", y=paste(i,"loading")) + scale_fill_manual(values=c("#0D8CFF","#FF3300")) + theme(legend.position="none")
} 
ggarrange(plotlist=plots, ncol=5, nrow=2)
#ggsave(filename="figure_alltissue_loadings.pdf", device="pdf",height =8.5 ,width=16, units="in")
#ggsave(filename="figure_alltissue_loadings.png", device="png",height =8.5 ,width=16, units="in")
```


```{r candidates sex, eval=FALSE}
a<- plotCounts(dd, gene="AR", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ap<- ggplot(a, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="AR",x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

b<- plotCounts(dd, gene="SRD5A2", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
bp<- ggplot(b, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="SRD5A2",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])


c<- plotCounts(dd, gene="LOC113993669", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
cp<- ggplot(c, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="CYP19A1",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

d<- plotCounts(dd, gene="ESR1", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
dp<- ggplot(d, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="ESR1",x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

e<- plotCounts(dd, gene="ESR2", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ep<- ggplot(e, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="ESR2",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

f<- plotCounts(dd, gene="PGR", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
fp<- ggplot(f, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="PGR",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])


g<- plotCounts(dd, gene="GNRH1", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
gp<- ggplot(g, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="GNRH1",x="Tissue", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

h<- plotCounts(dd, gene="PRL", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
hp<- ggplot(h, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="PRL",x="Tissue", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

i<- plotCounts(dd, gene="PRLR", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ip<- ggplot(i, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="PRLR",x="Tissue", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

ggarrange(ap,bp,cp, dp, ep, fp, gp,hp, ip, ncol=3, nrow=3, labels=LETTERS[seq(from=1,to=9)], common.legend=TRUE)

```

```{r candidates neuropep, eval=FALSE}
a<- plotCounts(dd, gene="VIP", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ap<- ggplot(a, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="VIP",x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

b<- plotCounts(dd, gene="VIPR1", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
bp<- ggplot(b, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="VIPR1",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])


c<- plotCounts(dd, gene="VIPR2", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
cp<- ggplot(c, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="VIPR2",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

d<- plotCounts(dd, gene="LOC113983511", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
dp<- ggplot(d, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="OXT",x="", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

e<- plotCounts(dd, gene="OXTR", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ep<- ggplot(e, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="OXTR",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

f<- plotCounts(dd, gene="LOC113983498", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
fp<- ggplot(f, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="AVP",x="", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])


g<- plotCounts(dd, gene="AVPR1A", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
gp<- ggplot(g, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="AVPR1A",x="Tissue", y="Normalised Counts") + peri_theme + scale_color_manual(values=status_cols[2:3])

h<- plotCounts(dd, gene="AVPR1B", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
hp<- ggplot(h, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="AVPR1B",x="Tissue", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

i<- plotCounts(dd, gene="LOC113982601", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ip<- ggplot(i, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point(aes(color=Status)) + labs(title="AVPR2",x="Tissue", y="") + peri_theme + scale_color_manual(values=status_cols[2:3])

ggarrange(ap,bp,cp, dp, ep, fp, gp,hp, ip, ncol=3, nrow=3, labels=LETTERS[seq(from=1,to=9)], common.legend=TRUE)
```

# Wholebrain

Looking for cohesive signatures across the whole brain.

```{r Wholebrain filtering, echo=TRUE} 
#brain_key<- subset(key_behav, Tissue!="GON")
#brain_key<- subset(brain_key, Tissue!="PIT")
outliers=c("PFT2_POM_run1")
#brain_behav<- brain_key[!rownames(brain_behav) %in% outliers,]
#brain_key<- droplevels(brain_key)
brain_behav<- subset(key_behav, Tissue!="GON")
brain_behav<- subset(brain_behav, Tissue!="PIT")
brain_behav<- brain_behav[!rownames(brain_behav) %in% outliers,]
brain_behav<- droplevels(brain_behav)


brain_data<- data[,colnames(data) %in% rownames(brain_behav)]



start<- nrow(brain_data)
#remove genes with avg read counts <20
brain_data$avg_count<- apply(brain_data, 1, mean)
brain_data<- brain_data[brain_data$avg_count>20,]
brain_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
brain_data$percent_0<- apply(brain_data, 1, function(x)length(x[x==0]))
thresh<- ncol(brain_data)/2
brain_data<- brain_data[brain_data$percent_0<=thresh,]
brain_data$percent_0<-NULL
#13652 genes.
```


Overall, before this round of filtering we have `r start` genes. After filtering we have `r nrow(brain_data)` genes. We are excluding `r outliers` because the analyses below indicated that it was an outlier within POM. 

```{r}
dd<- DESeqDataSetFromMatrix(countData=brain_data, colData=brain_behav, design= ~ Tissue +  Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 

#plotDispEsts(dd)

#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables
vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
p <- pca(vsd_data, metadata=brain_behav)

a<-biplot(p, lab=NULL, colby="Tissue", shape="Batch", legendPosition="right", title="") + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + theme(plot.margin = unit(c(0,1,0,1), "lines"))

## check to see how our interest and nuisance variables are correlated to the PC axes. This will help decide on model form for DESeq.

b<- eigencorplot(p, metavars = c("Tissue","time_kill","Batch","Year","mean_T", "strength.all_study","Status"), col=blueWhiteRed(50)) 

horn <- parallelPCA(vsd_data)
elbow <- findElbowPoint(p$variance)
c<- screeplot(p,
    components = getComponents(p, 1:20),
    vline = c(horn$n, elbow)) +

    geom_label(aes(x = horn$n + 1, y = 50,
      label = 'Horn\'s', vjust = -1, size = 8)) +
    geom_label(aes(x = elbow + 1, y = 50,
      label = 'Elbow method', vjust = -1, size = 8)) + labs(title="") + theme(plot.margin = unit(c(0,2,0,1), "lines")) + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) 




d<-biplot(p, x="PC1",y="PC3", lab=NULL, colby="Tissue", shape="Batch", legendPosition="right", title="") + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + theme(plot.margin = unit(c(0,1,0,1), "lines"))



multiplot<- ggarrange(ggarrange(a,d,ncol=2, labels=c("A","B"), common.legend = TRUE, legend="right")           
          ,b,c,
          nrow=3, 
          labels = c("","C","D")) 
multiplot
#ggsave(filename="figure_wholebrain_PCA.pdf", device="pdf",width =8.5 ,height=11, units="in")
#ggsave(filename="figure_wholebrain_PCA.png", device="png",width =8.5 ,height=11, units="in")
```


```{r wholebrain pc5 and 10}
library(nlme)



test<- data.frame(PC5=p$rotated$PC5, PC10=p$rotated$PC10)
test<- cbind(test, brain_behav)

#summary(lm(PC5 ~ Year + Tissue*Status,data=test ))
m1<- lme(PC5 ~ Batch + Tissue*Status, random=~1|Harvest_ID, data=test)
pc5test<- anova(m1)


#pc5test<- summary(aov(test$PC5 ~ test$Year + test$Status))
pc5test_anova <- ggtexttable(signif(pc5test,2))
#,  theme = ttheme("mOrange"))




e<-biplot(p, x="PC1",y="PC5", lab=NULL, colby="Tissue", shape="Status", legendPosition="right", title="") + peri_theme
f<-ggplot(test, aes(x=Status, y=PC5)) + geom_boxplot() + geom_jitter()+ peri_theme  + theme(legend.position="none") 


m1<- lme(PC10 ~ Year + Tissue*strength.all_study, random=~1|Harvest_ID, data=test[!is.na(test$strength.all_study),])
pc10test<- anova(m1)


#pc5test<- summary(aov(test$PC5 ~ test$Year + test$Status))
pc10test_anova <- ggtexttable(signif(pc10test,2))


##strength
g<-biplot(p, x="PC1",y="PC10", lab=NULL, colby="strength.all_study", shape="Status", legendPosition="right", title="") + peri_theme
h<-ggplot(test, aes(x=strength.all_study, y=PC10, color=Tissue)) + geom_point() + peri_theme  + theme(legend.position="none") 

ggarrange(e,ggarrange(pc5test_anova,f, nrow=2),g,ggarrange(pc10test_anova,h, nrow=2),ncol=2, nrow=2,labels=c(LETTERS[1:4]))
#ggsave(filename="figure_wholebrain_PC5_10.png", device="png",height =8.5 ,width=11, units="in")

#AVP<- plotCounts(dd, gene="LOC113983498", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
#ggplot(AVP, aes(x=Status, y=count)) + geom_point(size=2, pch=21 ,fill="navajowhite",color="black") #+ labs(title="AVP, navajowhite",x="Status", y="Normalised Counts") + facet_wrap( ~ Tissue) + #geom_boxplot() + peri_theme
#ggplot(AVP, aes(x=Status, y=count, color=Tissue)) + geom_point(size=2) + #labs(title="AVP",x="Status", y="Normalised Counts")  + peri_theme + geom_boxplot()
#
#MBP<- plotCounts(dd, gene="LOC114000755", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
#ggplot(MBP, aes(x=Status, y=count)) + geom_boxplot() + geom_point(size=2, pch=21 #,fill="navajowhite",color="black") + labs(title="MBP",x="Status", y="Normalised Counts") + #facet_wrap( ~ Tissue) + peri_theme
#
#ggplot(MBP, aes(x=Status, y=count)) + geom_boxplot() + geom_point(size=2) + #labs(title="MBP",x="Status", y="Normalised Counts")+ peri_theme

```

```{r ai, eval=FALSE}
a<- plotCounts(dd, gene="C1QL3", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ap<- ggplot(a, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="C1QL3",x="", y="Normalised Counts") + peri_theme

b<- plotCounts(dd, gene="ETV1", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
bp<- ggplot(b, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="ETV1",x="", y="") + peri_theme 


c<- plotCounts(dd, gene="KCTD12", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
cp<- ggplot(c, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="KCTD12",x="", y="") + peri_theme

d<- plotCounts(dd, gene="MGP", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
dp<- ggplot(d, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="MGP",x="", y="Normalised Counts") + peri_theme

e<- plotCounts(dd, gene="HTR1D", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ep<- ggplot(e, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="HTR1D",x="", y="") + peri_theme

f<- plotCounts(dd, gene="HTR2A", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
fp<- ggplot(f, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="HTR2A",x="", y="") + peri_theme


g<- plotCounts(dd, gene="PCP4", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
gp<- ggplot(g, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="PCP4",x="Tissue", y="Normalised Counts") + peri_theme

h<- plotCounts(dd, gene="PVALB", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
hp<- ggplot(h, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="PVALB",x="Tissue", y="") + peri_theme

i<- plotCounts(dd, gene="SCUBE1", intgroup=c("Class","Status", "Tissue"), returnData=TRUE)
ip<- ggplot(i, aes(x=Tissue, y=count)) + geom_boxplot(outlier.color = NA) + geom_point() + labs(title="SCUBE1",x="Tissue", y="") + peri_theme

ggarrange(ap,bp,cp, dp, ep, fp, gp,hp, ip, ncol=3, nrow=3, labels=LETTERS[seq(from=1,to=9)])

```


```{r loadings}
#plotloadings(p,
#    components = getComponents(p, 1:5),
#    rangeRetain = 0.01,
#    labSize = 4.0,
#    absolute = FALSE,
#    title = 'Loadings plot',
#    caption = 'Top 1% variables',
#    shape = 21, shapeSizeRange = c(3, 16),
#    col = c('white', 'grey60'),
#    drawConnectors = TRUE) + peri_theme

 
loadings<- p$loadings
loadings$gene<- rownames(loadings)
loadings<- merge(loadings, go_key, by="gene")


plots<- list()
pcs<- paste0("PC",1:10)
for(i in pcs){
  #i="PC2"
  sub<- loadings[,c(i, "best_anno","gene")]
  sub<- sub[order(sub[,i]),]
  top30<- rbind(head(sub,15), tail(sub,15))
  top30$best_anno2<- factor(top30$best_anno, levels=unique(top30$best_anno[order(top30[,i],top30$best_anno)]), ordered=TRUE)
  top30$pos<- ifelse(top30[,i]<0, "neg", "pos")
  plots[[i]]<-ggplot(top30, aes(x=best_anno2, y=top30[,i], fill=pos)) + geom_bar(stat="identity")+  peri_theme + coord_flip() + labs(x="gene", y=paste(i,"loading")) + scale_fill_manual(values=c("#0D8CFF","#FF3300")) + theme(legend.position="none")
} 
ggarrange(plotlist=plots, ncol=5, nrow=2)
#ggsave(filename="figure_wholebrain_loadings.pdf", device="pdf",height =8.5 ,width=16, units="in")
#ggsave(filename="figure_wholebrain_loadings.png", device="png",height =8.5 ,width=16, units="in")
```


```{r PC5 enrichment}
library(viridis)
res_go<- loadings[c("gene","PC5","best_anno")]
#res_go$logP<- -log(abs(res_go$PC5), 10)
#res_go$logP<- ifelse(res_go$PC5<0, -(res_go$logP), res_go$logP)
res_go<- res_go[order(res_go$PC5, decreasing=TRUE),]


geneList<- res_go$PC5
names(geneList)<- res_go$gene




y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
ridgeplot(y) + scale_fill_viridis(direction=-1)

write.csv(y@result, file="../DE_results/GSEA_BP_whole_brain_PC5.csv", row.names=FALSE)
ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
scatterPlot(simMatrix, reducedTerms)

x <- cmdscale(as.matrix(as.dist(1 - simMatrix)), eig = TRUE, 
        k = 2)
df <- cbind(as.data.frame(x$points), reducedTerms[match(rownames(x$points), 
        reducedTerms$go), c("term", "parent", "parentTerm", 
        "size")])
df$ID<-rownames(df)
df<- merge(df, y@result, by="ID")
df$pos<- ifelse(df$enrichmentScore>0,"pos","neg")
df$pos<- as.factor(df$pos)
#df$pos<- factor(df$pos, levels=c("neg","pos"))

labelSize=4
p <- ggplot2::ggplot(df, ggplot2::aes(x = V1, y = V2, color = pos)) + 
        ggplot2::geom_point(ggplot2::aes(size = size), alpha = 0.5) + 
        ggplot2::scale_color_manual(guide = FALSE, values=c("#00BFC4","#F8766D")) + ggplot2::scale_size_continuous(guide = FALSE, 
        range = c(0, 25)) + ggplot2::scale_x_continuous(name = "") + 
        ggplot2::scale_y_continuous(name = "") + ggplot2::theme_minimal() + 
        ggplot2::theme(axis.text.x = ggplot2::element_blank(), 
            axis.text.y = ggplot2::element_blank())
p<- p + ggrepel::geom_label_repel(aes(label = parentTerm), 
            data = subset(df, parent == ID), box.padding = grid::unit(1, "lines"), size = labelSize)+ theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank())

p 

```

```{r}
i="PC5"
  sub<- loadings[,c(i, "best_anno","gene")]
  sub<- sub[order(sub[,i]),]
  top30<- rbind(head(sub,20), tail(sub,20))
  top30$best_anno2<- factor(top30$best_anno, levels=unique(top30$best_anno[order(top30[,i],top30$best_anno)]), ordered=TRUE)
  top30$pos<- ifelse(top30[,i]<0, "neg", "pos")
  pc5loadings<-ggplot(top30, aes(x=best_anno2, y=top30[,i], fill=pos)) + geom_bar(stat="identity")+  peri_theme + coord_flip() + labs(x="gene", y=paste(i,"loading")) + scale_fill_manual(values=c("#0D8CFF","#FF3300")) + theme(legend.position="none")
  
  
ggarrange(ggarrange(f,pc5test_anova, nrow=2), pc5loadings, p, ncol=3, nrow=1)
  
```


```{r whole-brain pca figure, eval=FALSE}
tissue_cols<- c("VMH"="#E76F51","AH"="#E76F51","PVN"="#E76F51", "POM"="#E76F51", "ICO"="#E76F51","GCT"="#E76F51", "TNA"="#E76F51", "AI"="#f0d497", "LS"="#F4A261", "BSTm"="#F4A261")
#shape_key<- c("VMH"="0","AH"="1","PVN"="2", "POM"="3", "ICO"="4","GCT"="5", "TNA"="8", "AI"="7", "LS"="10", "BSTm"="14")

biplot(p, lab=NULL, colby="Tissue", shape="Tissue", legendPosition="right", colkey = tissue_cols, pointSize = 6) + scale_shape_manual(values=c(0,1,2,3,4,5,8,15,17,19)) + theme(panel.background = element_rect(fill="white"),panel.grid.major = element_blank(), panel.grid.minor = element_blank(),legend.position="none")

```
 

 
What happens if I use batch effect exclusion on the whole brain? 
 
```{r}  
mat<- limma::removeBatchEffect(vsd_data, brain_behav$Batch)
p <- pca(mat, metadata = brain_behav)
biplot(p, lab=NULL, colby="Tissue", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")

#use this if Batch and Year are both pinged in the previous section. 
eigencorplot(p, metavars = c("Tissue","Batch","Year","time_kill" ,"Status"))

```

It appears to make the differences between batches worse, even though the eigencorplot doesn't show it. So, not only does accounting for individual not work in these analyses owing to unequal sample sizes among tissues (DESeq2 issue), but also accounting for batch across the brain might actually **amplify** batch effects.



# Gonads GON

I will use the first two tissues to demonstrate the processes applied across all tissues, but code echoing will be supressed in future tissues.

```{r GON filtering, echo=TRUE} 
gon_key<- subset(key_behav, Tissue=="GON")
gon_key<- droplevels(gon_key)
gon_data<- data[,colnames(data) %in% rownames(gon_key)]


start<- nrow(gon_data)
#remove genes with less than 5 reads
gon_data$avg_count<- apply(gon_data, 1, mean)
gon_data<- gon_data[gon_data$avg_count>5,]
gon_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gon_data$percent_0<- apply(gon_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gon_data)/2
gon_data<- gon_data[gon_data$percent_0<=thresh,]
gon_data$percent_0<-NULL

```

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(gon_data)`

## Checking the sampling

Before I go into the analyses, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gonad samples for `r nrow(gon_key)`, with respect to T and the tracking data we have `r nrow(gon_key[gon_key$Batch!="pilot",])` samples.

Here, we will run all the data to check for DEGs against status, then subset the data to only include those individuals with T and behavioural data. 


```{r GON sampling status, echo=TRUE}

fsb<- fisher.test(table(gon_key$Status, gon_key$Batch))
knitr::kable(table(gon_key$Status, gon_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))

fsy<- fisher.test(table(gon_key$Status, gon_key$Year))
knitr::kable(table(gon_key$Status, gon_key$Year), caption=paste0("Status across sampling years, fisher test p=", fsy$p.value))

```


## Status GON

First, I will run through a data checking process. I will check for outliers, and then find out how our potential nuisance variables effect our gene expression data. If they have effects they will be incorporated into the final DESeq models. 

```{r GON status set up, echo=TRUE}
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 


#plotDispEsts(dd)

#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables
vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
p <- pca(vsd_data, metadata = gon_key)
biplot(p, lab=gon_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Gonads all samples PCA")

## check to see how our interest and nuisance variables are correlated to the PC axes. This will help decide on model form for DESeq.
eigencorplot(p, metavars = c("Batch","Year", "Status"))


```



Based on the eigencor plot, I will include Batch as a variable in my output. 

There isn't a good correspondence between the outlier samples on the PCA plot vs the outliers on the WGCNA dendrogram, so I am not convinced they are outliers. 
Also let's see how those "outliers" look when we account for Batch in our model by using limma::RemoveBatchEffect()

```{r GON removeBatchEffect}
#remove batch effect
mat<- limma::removeBatchEffect(vsd_data, gon_key$Batch)
p <- pca(mat, metadata = gon_key)
biplot(p, lab=gon_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")

#use this if Batch and Year are both pinged in the previous section. 
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))


traits<- gon_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Now run DEseq with the updated model form and extract DEGs.

Formula ~ Batch + Status.

```{r GON Status2, fig.height=4, fig.width=10, echo=TRUE}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="Gonads Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()


Status_res<- data.frame(res)
Status_res$gene<- row.names(res)
Status_res<- Status_res[order(Status_res$gene),]

write.csv(Status_res, file="../DE_results/results_GON_Status.csv", row.names=TRUE)

```


```{r GON status geneplotting, fig.height=8, fig.width=14, echo=TRUE}

status_cols<- c("#86AD44","#E54849","#414042")
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Class","Status"), returnData=TRUE)

gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Class","Status"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Class","Status"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Class","Status"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Class","Status"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Class","Status"), returnData=TRUE)
vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<- formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])),subtitle=paste0("LFC=",lfc," padj=", padj),x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in GON",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)))
g

```

#### Gene Ontology
Conduct Gene Set enrichment analysis when there are few significant differentially expressed genes.

```{r GON Status BP, echo=TRUE, fig.height=5, fig.width=15}
res_go<- Status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_status<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_GON_Status.csv", row.names=FALSE)
ego2<- as.data.frame(go_status$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_status$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```


Below is the code for making the figures in the supplement.

```{r, echo=TRUE, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_GON_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```





## Testosterone GON

Given that the behaviors are best correlated with mean T, then this is our variable of interest

But first, we need to look at these new data in the GONADS with the new subset with respect to our interest variables in the PCA. 

```{r GON gene filtering 2, echo=TRUE}
gon_behav<- subset(key_behav, Tissue=="GON")
gon_behav<- subset(gon_behav, Batch!="pilot")
gon_behav<- droplevels(gon_behav)
gon_data<- gon_data[,colnames(gon_data) %in% rownames(gon_behav)]

start<- nrow(gon_data)
#remove genes with less than 5 reads
gon_data$avg_count<- apply(gon_data, 1, mean)
gon_data<- gon_data[gon_data$avg_count>5,]
gon_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gon_data$percent_0<- apply(gon_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gon_data)/2
gon_data<- gon_data[gon_data$percent_0<=thresh,]
gon_data$percent_0<-NULL

```

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(gon_data)`...

```{r GON continuous variable setup, echo=TRUE}
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)

write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_GON_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = gon_behav)
biplot(p, lab=gon_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")



eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

A=adjacency(t(datExpr0),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)
#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

```{r, eval=FALSE}
mat<- limma::removeBatchEffect(vsd_data, gon_behav$Year)
p <- pca(mat, metadata = gon_behav)
biplot(p, lab=gon_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="PCA Gonads removeBatchEffect(Year)")
#eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

```



### mean T GON

DESeq model specification ~ Year +  mean_T

This chunk also includes a function I wrote to remove highly influential observations (as measured by Cook's Distance) from the continuous data. I know from previous iterations of this analysis that there are observations that are highly influential and also DESeq2 does not exclude these from the results in continuous data. 

```{r GON mean T DEseq, fig.height=4, fig.width=10, echo=TRUE}
dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Year + mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1)
summary(res)


rm_continuous_outliers<- function(deseq=x, results=y, pct=0.98){
  cooks<- as.data.frame(assays(deseq)[["cooks"]])
  p=ncol(model.matrix(design(deseq), colData(deseq)))
  n=nrow(colData(deseq))
  cd_cutoff<- qf(pct,p,n-p)
  out<- as.data.frame(t(apply(cooks, 1, function(x){ x > cd_cutoff })))
  out$outlier<- apply(out,1, any)
  outliers<- rownames(out)[out$outlier==TRUE]
  results$padj[rownames(results) %in% outliers]<-NA
  results$pvalue[rownames(results) %in% outliers]<-NA
  return(results)
}

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)


mean_T_res<- data.frame(res[order(rownames(res)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_GON_mean_T.csv", row.names=TRUE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="Gonads Mean T")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()


```




```{r GON mean T geneplots, fig.height=8, fig.width=14, echo=TRUE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()



for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}

                    


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by mean T in GON",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r hub plot, eval=FALSE, echo=FALSE}

hub<- "TH"
out<-  plotCounts(dd, gene=hub, intgroup=c("mean_T", "Class"), returnData=TRUE)
Rsq<- cor(out$mean_T, out$count)^2
hplot<- ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + labs(title=expaste0(row.names(res[hub,])," LFC=",signif(mean_T_res$log2FoldChange[mean_T_res$gene==hub],2) ," padj=", signif(mean_T_res$padj[mean_T_res$gene==hub],2), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none")
```

#### Gene Ontology


```{r GON mean T BP GSEA, eval=FALSE}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
ridgeplot(y)

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_GON_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(mean_T_res$gene[which(mean_T_res$padj<0.1)], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)

ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)
ego2$qval2<- -ego2$qvalue


goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms by mean_T in GON", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_GON_mean_T.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_GON_meanT.png", plot=g2, device="png" ,height=11, width=9, units="in")
```



### mean T x Status

For pulling out candidate genes with interactions

```{r GON mean T x Status, echo=TRUE, , fig.height=4, fig.width=10, echo=TRUE}

dd<- DESeqDataSetFromMatrix(countData=gon_data, colData=gon_behav, design= ~ Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")
res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "GON"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_GON_mean_T_Status.csv", row.names=FALSE)




par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GON mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r, echo=FALSE,  fig.height=8, fig.width=14,echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in GON",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


## Strength GON

formula ~ Year + scale(Strength)

```{r GON Strength, fig.height=4, fig.width=10, echo=TRUE}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Year + strength.all_study)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)

res<- rm_continuous_outliers(dd,res)
summary(res)
res<- res[order(res$padj),]





strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)
write.csv(strength_res, file="../DE_results/results_GON_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GON Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r GON strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]

  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")

}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by cooperative strength in GON",gp=gpar(fontsize=12)), bottom=textGrob("Social network strength",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r GON strength BP}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_strength<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_GON_strength.csv", row.names=FALSE)
ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p

```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_GON_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```



# Pituitary PIT


```{r PIT data set up}
keep<- c("data","key","behav","key_behav", "dodge", "peri_theme", "keep", "tissues_status", "tissues_mean_T", "tissues_final_T", "tissues_strength", "dj_col", "fromList", "get_intersect_members", "status_cols", "selFun","rm_continuous_outliers", "status_cols", "interactions","go_key","go2gene_bp", "go2gene_mf","go2name_bp","go2name_mf")
rm(list= ls()[!(ls() %in% keep)])

#pit_key<- subset(key, Tissue=="PIT")
#pit_key<- droplevels(pit_key)
pit_behav<- subset(key_behav, Tissue=="PIT")
pit_behav<- droplevels(pit_behav)


#Filtering
pit_data<- data[,colnames(data) %in% rownames(pit_behav)]
start<- nrow(pit_data)
#remove genes with less than 5 reads
pit_data$avg_count<- apply(pit_data, 1, mean)
pit_data<- pit_data[pit_data$avg_count>5,]
pit_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pit_data$percent_0<- apply(pit_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pit_data)/2
pit_data<- pit_data[pit_data$percent_0<=thresh,]
pit_data$percent_0<-NULL
```

### Checking the sampling

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(pit_data)`
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pit samples for `r nrow(pit_behav)`, with respect to T and the tracking data.
```{r PIT sampling, fig.height=8, fig.width=14}

fsb<- fisher.test(table(pit_behav$Status, pit_behav$Batch))
knitr::kable(table(pit_behav$Status, pit_behav$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))

fsyb<- fisher.test(table(pit_behav$Batch, pit_behav$Year))
knitr::kable(table(pit_behav$Batch, pit_behav$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", fsyb$p.value))


mean_t_wtest<- wilcox.test(pit_behav$mean_T ~ pit_behav$Batch, alternative="two.sided")
a<- ggplot(pit_behav, aes(x=Batch, y=mean_T)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(pit_behav$strength.all_study ~ pit_behav$Batch, alternative="two.sided")
d<- ggplot(pit_behav, aes(x=Batch, y=strength.all_study)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme
grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE)

```


```{r PIT DEseq1}

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
##vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_PIT_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_PIT_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = pit_behav)
biplot(p, lab=pit_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PIT PCA not corrected for batch")


eigencorplot(p, metavars = c("time_kill","Year","Batch","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

```

```{r}
mat<- limma::removeBatchEffect(vsd_data, pit_behav$Batch)
p <- pca(mat, metadata = pit_behav)
biplot(p, lab=pit_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="PCA PIT removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))




wgcnadata<- as.data.frame(t(mat))


traits<- pit_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

Looks controlling for Batch also removes the association with Year. Let's just include Batch.

##Status PIT

Formula = ~ Batch + Status

```{r PIT Status, fig.height=8, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)




status_res<- data.frame(res1)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]

write.csv(status_res, file="../DE_results/results_PIT_status.csv", row.names=FALSE)

par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="Pituitary Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected PIT Status")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()
```

```{r PIT Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Class","Status"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Class","Status"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Class","Status"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Class","Status"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Class","Status"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Class","Status"), returnData=TRUE)
vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)


plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b,),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}

g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in PIT",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g

```
```{r, eval=FALSE}
hub<-"GH1"
dat<- plotCounts(dd, gene=row.names(res1[hub,]), intgroup=c("Class","Status"), returnData=TRUE)
ggplot(dat, aes(x=Status, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="Status", y="GH1 Normalized Counts", title=paste0(hub,"\n LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],2), ", p=",signif(res1$pvalue[rownames(res1)==hub],3), ", padj=", signif(res1$padj[rownames(res1)==hub],2))) + geom_boxplot(fill=NA) + theme(legend.position = "none")


```

#### Gene Ontology

```{r PIT Status BP}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_status<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_PIT_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_status$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_status$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))


```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_PIT_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone PIT

### mean T PIT

```{r PIT mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)

write.csv(mean_T_res, file="../DE_results/results_PIT_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="Pituitary mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```


```{r PIT mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}

g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by mean T in PIT",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g


#hub<- "PRL"
#out<-  plotCounts(dd, gene=hub, intgroup=c("mean_T", "Class"), returnData=TRUE)
#Rsq<- cor(out$mean_T, out$count)^2
##i=3
#a<- ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + #labs(title=paste0(row.names(res1[hub,])," #LFC=",signif(mean_T_res$log2FoldChange[mean_T_res$gene==hub],2) ," padj=", #signif(mean_T_res$padj[mean_T_res$gene==hub],2), " R^2=", signif(Rsq,2)),x="", y="Normalised #Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ #theme(legend.position="none")
#
#hub<- "VIPR2"
#out<-  plotCounts(dd, gene=hub, intgroup=c("mean_T", "Class"), returnData=TRUE)
#Rsq<- cor(out$mean_T, out$count)^2
##i=3
#b<- ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + #labs(title=paste0(row.names(res1[hub,])," #LFC=",signif(mean_T_res$log2FoldChange[mean_T_res$gene==hub],2) ," padj=", #signif(mean_T_res$padj[mean_T_res$gene==hub],2), " R^2=", signif(Rsq,2)),x="", y="Normalised #Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ #theme(legend.position="none")
#ggarrange(a,b,ncol=2)
```


#### Gene Ontology

```{r PIT mean T BP}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_PIT_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_PIT_mean_T.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```






### mean T x Status


```{r PIT mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Batch + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

interaction_res<- as.data.frame(res1)
interaction_res$gene<- rownames(res1)
interaction_res$Tissue<- "PIT"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_PIT_mean_T_Status.csv", row.names=FALSE)




par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="Pituitary Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```


```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in PIT",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



## Strength PIT

Formula: ~ Batch + scale(Strength)


```{r PIT Strength, fig.height=4, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)





strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

#write.csv(strength_res, file="../DE_results/results_PIT_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="PIT Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r PIT strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- cor(dat$strength.all_study, dat$count)^2
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class, label=Harvest_ID)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," LFC=",signif(res$log2FoldChange[i],2) ," padj=", formatC(res$padj[i],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + geom_text(nudge_x=-0.08)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by social network strength in PIT",gp=gpar(fontsize=14)), bottom=textGrob("social network strength",gp=gpar(fontsize=14)))
g


cooks<- as.data.frame(assays(dd)[["cooks"]])
 p=ncol(model.matrix(design(dd), colData(dd)))
  n=nrow(colData(dd))
 
par(mar=c(8,5,2,2), mfrow=c(2,1))
boxplot(log10(cooks), range=0, las=2, main="Cook's Distances for Strength across ALL genes")
abline(h=log10(qf(0.98,p,n-p)))
```

As you can see above, PFT3 has consistently high leverage on the social network analysis. Even if I play around with the outlier theshold this I am always getting non-linear plots like above. So, let's see what happens when I remove this sample. 


```{r PIT Strength2, fig.height=4, fig.width=10}
outlier<- "PFT3_PIT_run2"

pit_data<- pit_data[,!colnames(pit_data) %in% outlier]
pit_behav<- pit_behav[!rownames(pit_behav) %in% outlier,]

dd<- DESeqDataSetFromMatrix(countData=pit_data, colData=pit_behav, design= ~ Status)


dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)





strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)
write.csv(strength_res, file="../DE_results/results_PIT_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="PIT Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r PIT strength geneplots2, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class", "Harvest_ID"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") 
}



g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by strength in PIT",gp=gpar(fontsize=12)), bottom=textGrob("Social network strength",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r, eval=FALSE}
hub<-"VIPR2"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
ggplot(dat, aes(x=strength.all_study, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="strength", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res$log2FoldChange[rownames(res)==hub],2), ", p=",signif(res$pvalue[rownames(res)==hub],2), ", padj=", signif(res$padj[rownames(res)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")
```

```{r plotting PIT, eval=FALSE}
PIT_modules<- read.csv("../WGCNA/ByTissue_2021/PIT_results_ModuleMembership.csv")


topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


i="PIT"
 tissue<- strength_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>2, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""    
tissue<- merge(tissue, PIT_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","pink4","lightgreen","plum1","violet"          ,"lightcyan1","darkolivegreen4","plum2","floralwhite","coral1","blue","brown4","darkgreen","darkolivegreen2","firebrick3","blue2"))

modulecols<- c("grey30","grey60","pink4","lightgreen","plum1","violet"          ,"lightcyan1","darkolivegreen4","plum2","floralwhite","coral1","blue","brown4","darkgreen","darkolivegreen2","firebrick3","blue2")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")



#ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=0.0)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(limits=c(-5,5.5), breaks=c(-5,-3,0,3,5))

```




#### Gene Ontology

```{r PIT strength BP}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- enricher(strength_res$gene[strength_res$padj<0.1], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot


knitr::kable(y@result[1:10,], caption="BP enriched GO terms in PIT", row.names = FALSE) %>% kable_styling()


write.csv(y@result, file="../DE_results/GO_BP_PIT_strength.csv", row.names=FALSE)
ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot,p, labels=c("A","B","C"), nrow=3)

g2
ggsave(filename="supp_figure_PIT_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```


# Ventromedial Hypothalamus VMH


```{r VMH data set up}
rm(list= ls()[!(ls() %in% keep)])

vmh_key<- subset(key_behav, Tissue=="VMH")

vmh_key<- droplevels(vmh_key)

vmh_behav<- vmh_key[vmh_key$Batch!="pilot",]

#Filtering
vmh_data<- data[,colnames(data) %in% rownames(vmh_key)]
start<- nrow(vmh_data)
#remove genes with less than 5 reads
vmh_data$avg_count<- apply(vmh_data, 1, mean)
vmh_data<- vmh_data[vmh_data$avg_count>5,]
vmh_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
vmh_data$percent_0<- apply(vmh_data, 1, function(x)length(x[x==0]))
thresh<- ncol(vmh_data)/2
vmh_data<- vmh_data[vmh_data$percent_0<=thresh,]
vmh_data$percent_0<-NULL
```

### Checking the sampling

Before filtering we had `r start` genes, after filtering for mean read count and excluding genes where >50% of samples had a count of 0 we are left with `r nrow(vmh_data)`

Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have vmh samples for `r nrow(vmh_key)`, with respect to T and the tracking data.

```{r VMH sampling}

fsb<- fisher.test(table(vmh_key$Status, vmh_key$Batch))
knitr::kable(table(vmh_key$Status, vmh_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))


fsyb<- fisher.test(table(vmh_key$Batch, vmh_key$Year))
knitr::kable(table(vmh_key$Batch, vmh_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", fsyb$p.value))


mean_t_wtest<- wilcox.test(vmh_behav$mean_T ~ vmh_behav$Batch, alternative="two.sided")
a<- ggplot(vmh_behav, aes(x=Batch, y=mean_T))  + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(vmh_behav$strength.all_study ~ vmh_behav$Year, alternative="two.sided")
d<- ggplot(vmh_behav, aes(x=Batch, y=strength.all_study)) + geom_point(size=4, color=dj_col[1]) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme
grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables VMH",gp=gpar(fontsize=14)))

```


The majority of our data in VMH is in run 1.


```{r VMH DEseq1}
dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
#plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = vmh_key)
biplot(p, lab=vmh_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in VMH")


eigencorplot(p, metavars = c("Year","Batch","Status"))


wgcnadata<- as.data.frame(t(vsd_data))


traits<- vmh_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```


```{r}
mat<- limma::removeBatchEffect(vsd_data, vmh_key$Batch)
p <- pca(mat, metadata = vmh_key)
#biplot(p, lab=vmh_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Year","Status"))


```

Looks like controlling for Batch is sufficient here, as Batch is highly correlated with Year.

##Status VMH

Formula = ~ Batch + Status

```{r VMH Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]


write.csv(status_res, file="../DE_results/results_VMH_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="VMH Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r VMH Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in VMH",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g

```

#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_VMH_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))

```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_VMH_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone VMH

### mean T VMH

```{r VMH DESeq continuous variable subset}
vmh_behav<- subset(key_behav, Tissue=="VMH")
vmh_behav<- subset(vmh_behav, Batch!="pilot")
vmh_behav<- droplevels(vmh_behav)
vmh_data<- data[,colnames(data) %in% rownames(vmh_behav)]

#Filtering
#remove genes with less than 5 reads
vmh_data$avg_count<- apply(vmh_data, 1, mean)
vmh_data<- vmh_data[vmh_data$avg_count>5,]
vmh_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
vmh_data$percent_0<- apply(vmh_data, 1, function(x)length(x[x==0]))
thresh<- ncol(vmh_data)/2
vmh_data<- vmh_data[vmh_data$percent_0<=thresh,]
vmh_data$percent_0<-NULL

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_VMH_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Canidate_gene_rsults/data_VMH_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = vmh_behav)
biplot(p, lab=vmh_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="mean T")


eigencorplot(p, metavars = c("Year","Batch","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))
```


```{r}
mat<- limma::removeBatchEffect(vsd_data, vmh_behav$Year)
p <- pca(mat, metadata = vmh_behav)
biplot(p, lab=vmh_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA VMH removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Year","Batch","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))


wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK

A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```


Batch does not account for year here... so have to include both... 


Formula: ~ Batch + Year + mean_T

```{r VMH mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~Batch + Year + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_VMH_mean_T.csv", row.names=TRUE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="VMH mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```


```{r vmh mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
    Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")

}

g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by mean T in VMH",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g

```


#### Gene Ontology
```{r VMH mean T BP}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

write.csv(y@result, file="../DE_results/GSEA_BP_VMH_meanT.csv", row.names=FALSE)
#ego2<- as.data.frame(y@result$ID)
#colnames(ego2)<- "ID"
#ego2$p.adjust<- y@result$p.adjust
#
#simMatrix <- calculateSimMatrix(ego2$ID,
#                                orgdb="org.Hs.eg.db",
#                                ont="BP",
#                                method="Rel")
#scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
#reducedTerms <- reduceSimMatrix(simMatrix,
#                                scores,
#                                threshold=0.7,
#                                orgdb="org.Hs.eg.db")
#treemapPlot(reducedTerms)
##scatterPlot(simMatrix, reducedTerms)
#
#x <- cmdscale(as.matrix(as.dist(1 - simMatrix)), eig = TRUE, 
#        k = 2)
#df <- cbind(as.data.frame(x$points), reducedTerms[match(rownames(x$points), 
#        reducedTerms$go), c("term", "parent", "parentTerm", 
#        "size")])
#df$ID<-rownames(df)
#df<- merge(df, y@result, by="ID")
#df$pos<- ifelse(df$enrichmentScore>0,"pos","neg")
#df$pos<- as.factor(df$pos)
##df$pos<- factor(df$pos, levels=c("neg","pos"))
#
#labelSize=3
#p <- ggplot2::ggplot(df, ggplot2::aes(x = V1, y = V2, color = pos)) + 
#        ggplot2::geom_point(ggplot2::aes(size = size), alpha = 0.5) + 
#        ggplot2::scale_color_manual(guide = FALSE, values=c("#00BFC4","#F8766D")) + #ggplot2::scale_size_continuous(guide = FALSE, 
#        range = c(0, 25)) + ggplot2::scale_x_continuous(name = "") + 
#        ggplot2::scale_y_continuous(name = "") + ggplot2::theme_minimal() + 
#        ggplot2::theme(axis.text.x = ggplot2::element_blank(), 
#            axis.text.y = ggplot2::element_blank())
#p<- p + ggrepel::geom_label_repel(aes(label = parentTerm), 
#            data = subset(df, parent == ID), box.padding = grid::unit(1, "lines"), size = #labelSize)
#
#p

```

```{r, eval=FALSE}
g2<-ggarrange(g, gseaplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_VMH_meanT.png", plot=g2, device="png" ,height=10, width=9, units="in")
```

### mean T x Status


```{r VMH mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=vmh_data, colData=vmh_behav, design= ~ Batch + Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=TRUE, verbose=TRUE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

interaction_res<- as.data.frame(res1)
interaction_res$gene<- rownames(res1)
interaction_res$Tissue<- "VMH"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_VMH_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="VMH Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```


```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


#hub<- "GAL"
#out<-  plotCounts(dd, gene=row.names(res1[hub,]), intgroup=c("Status","mean_T", "Year","Class"), #returnData=TRUE)
#ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + #labs(title=paste0(row.names(res1[hub,])," #LFC=",signif(interaction_res$log2FoldChange[interaction_res$gene==hub],2) ," padj=", #signif(interaction_res$padj[interaction_res$gene==hub],2)),x="", y="Normalised Counts") + peri_theme + #scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none") + facet_wrap(~ #Status)



g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in VMH",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



## Strength VMH

Formula ~ Batch + Year + scale(Strength)

```{r vmh Strength, fig.height=8, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + Year + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



strength_res<- data.frame(res1[order(rownames(res1)),])
strength_res$gene<- row.names(strength_res)


write.csv(strength_res, file="../DE_results/results_VMH_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="VMH Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")

dev.off()
```



```{r VMH strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by strength in VMH",gp=gpar(fontsize=12)), bottom=textGrob("Social network strength",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```


#### Gene Ontology

```{r VMH strength BP}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

write.csv(y@result, file="../DE_results/GSEA_BP_VMH_strength.csv", row.names=FALSE)
ego2<- as.data.frame(y@result$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- y@result$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_VMH_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```


# Anterior Hypothalamus (AH)


```{r AH data set up}

rm(list= ls()[!(ls() %in% keep)])

#ah_key<- subset(key, Tissue=="AH")
ah_behav<- subset(key_behav, Tissue=="AH")
ah_behav<- droplevels(ah_behav)

#Filtering
ah_data<- data[,colnames(data) %in% rownames(ah_behav)]
start<- nrow(ah_data)
#remove genes with less than 5 reads
ah_data$avg_count<- apply(ah_data, 1, mean)
ah_data<- ah_data[ah_data$avg_count>5,]
ah_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ah_data$percent_0<- apply(ah_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ah_data)/2
ah_data<- ah_data[ah_data$percent_0<=thresh,]
ah_data$percent_0<-NULL

```
### Checking the sampling

For the AH, we started with `r start` genes but after filering we have `r nrow(ah_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ah samples for `r nrow(ah_behav)`, with respect to T and the tracking data.

```{r AH sampling}

#fsb<- fisher.test(table(ah_behav$Status, ah_behav$Batch))
knitr::kable(table(ah_behav$Status, ah_behav$Batch), caption=paste0("Status across sequencing runs"))



#fsyb<- fisher.test(table(ah_behav$Batch, ah_behav$Year))
knitr::kable(table(ah_behav$Batch, ah_behav$Year), caption=paste0("Sampling years distributed across batches"))

```



```{r AH DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_AH_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_AH_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ah_behav)

biplot(p, lab=ah_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in AH")

eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

```

```{r}
mat<- limma::removeBatchEffect(vsd_data, ah_behav$Year)
p <- pca(mat, metadata = ah_behav)
biplot(p, lab=ah_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="PCA Gonads removeBatchEffect(Year)")


wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ah_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```


PFT1 and PFT11 cluster with the BSTm as previously identified by the brain tissue PCA plot and the all tissue WGCNA dendrogram. We know that there were issues with the sectioning plane in individual PFT1 that could affect this tissue, but not PFT11.

The effects of removing these samples will be explored for each analysis vs removing a randomly selected pair of samples. 

##Status AH

Formula = ~ Year + Status

```{r AH Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Year + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)





status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]


write.csv(status_res, file="../DE_results/results_AH_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AH Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r AH Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in AH",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_AH_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r. eval=FALSE}
g2<-ggarrange(g, gseaplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_AH_status.png", plot=g2, device="png" ,height=10, width=9, units="in")
```

## Testosterone AH

### mean T AH

```{r AH mean T, fig.height=8, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Year + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)




mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_AH_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AH Mean T")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```


```{r AH mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
   Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by mean T in AH",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g

```

#### Gene Ontology
```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_AH_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

#simMatrix <- calculateSimMatrix(ego2$ID,
#                               orgdb="org.Hs.eg.db",
#                               ont="BP",
#                               method="Rel")
#scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
#reducedTerms <- reduceSimMatrix(simMatrix,
#                                scores,
#                                threshold=0.7,
#                                orgdb="org.Hs.eg.db")
#treemapPlot(reducedTerms)
##scatterPlot(simMatrix, reducedTerms)
#
#x <- cmdscale(as.matrix(as.dist(1 - simMatrix)), eig = TRUE, 
#        k = 2)
#df <- cbind(as.data.frame(x$points), reducedTerms[match(rownames(x$points), 
#        reducedTerms$go), c("term", "parent", "parentTerm", 
#        "size")])
#df$ID<-rownames(df)
#df<- merge(df, y@result, by="ID")
#df$pos<- ifelse(df$enrichmentScore>0,"pos","neg")
#df$pos<- as.factor(df$pos)
##df$pos<- factor(df$pos, levels=c("neg","pos"))
#
#labelSize=3
#p <- ggplot2::ggplot(df, ggplot2::aes(x = V1, y = V2, color = pos)) + 
#        ggplot2::geom_point(ggplot2::aes(size = size), alpha = 0.5) + 
#        ggplot2::scale_color_manual(guide = FALSE, values=c("#00BFC4","#F8766D")) + ggplot2::scale_size_continuous(guide = FALSE, 
#        range = c(0, 25)) + ggplot2::scale_x_continuous(name = "") + 
#        ggplot2::scale_y_continuous(name = "") + ggplot2::theme_minimal() + 
#        ggplot2::theme(axis.text.x = ggplot2::element_blank(), 
#            axis.text.y = ggplot2::element_blank())
#p<- p + ggrepel::geom_label_repel(aes(label = parentTerm), 
#            data = subset(df, parent == ID), box.padding = grid::unit(1, "lines"), size = labelSize)
#
#p
```

```{r}
g2<-ggarrange(g, gseaplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_AH_meanT.png", plot=g2, device="png" ,height=10, width=9, units="in")
```

### mean T x Status


```{r AH mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=ah_data, colData=ah_behav, design= ~ Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)


res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

interaction_res<- as.data.frame(res1)
interaction_res$gene<- rownames(res1)
interaction_res$Tissue<- "AH"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_AH_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AH Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' P-value")
dev.off()

```


```{r, echo=FALSE,  fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in AH",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```




## Strength AH

Formula ~ scale(Strength)

```{r AH Strength, fig.height=8, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Year + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
summary(res1)




strength_res<- data.frame(res1[order(rownames(res1)),])
strength_res$gene<- row.names(strength_res)


write.csv(strength_res, file="../DE_results/results_AH_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="AH Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")

dev.off()
```



```{r AH Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by strengthin AH",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```
```{r, eval=FALSE}
hub<- "TSHR"
out<-  plotCounts(dd, gene=hub, intgroup=c("strength.all_study", "Class"), returnData=TRUE)
Rsq<- cor(out$strength.all_study, out$count)^2
ggplot(out, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=4) + labs(title=paste0(row.names(res[hub,])," LFC=",signif(strength_res$log2FoldChange[strength_res$gene==hub],2) ," padj=", formatC(strength_res$padj[strength_res$gene==hub],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none")
```

#### Gene Ontology


```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(strength_res$gene[strength_res$padj<0.1], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms in AH", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_AH_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

#simMatrix <- calculateSimMatrix(ego2$ID,
#                                orgdb="org.Hs.eg.db",
#                                ont="BP",
#                                method="Rel")
#scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
#reducedTerms <- reduceSimMatrix(simMatrix,
#                                scores,
#                                threshold=0.7,
#                                orgdb="org.Hs.eg.db")
#treemapPlot(reducedTerms)
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_AH_strength.png", plot=g2, device="png" ,height=11, width=9, units="in")
```

# Paraventricular Nucleus PVN



```{r PVN data set up}

rm(list= ls()[!(ls() %in% keep)])

#pvn_key<- subset(key, Tissue=="PVN")
pvn_behav<- subset(key_behav, Tissue=="PVN")
pvn_behav<- droplevels(pvn_behav)

#Filtering
pvn_data<- data[,colnames(data) %in% rownames(pvn_behav)]
start<- nrow(pvn_data)
#remove genes with less than 5 reads
pvn_data$avg_count<- apply(pvn_data, 1, mean)
pvn_data<- pvn_data[pvn_data$avg_count>5,]
pvn_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pvn_data$percent_0<- apply(pvn_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pvn_data)/2
pvn_data<- pvn_data[pvn_data$percent_0<=thresh,]
pvn_data$percent_0<-NULL

```
### Checking the sampling

For the PVN, we started with `r start` genes but after filering we have `r nrow(pvn_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pvn samples for `r nrow(pvn_behav)`, with respect to T and the tracking data.

```{r PVN sampling}

#fsb<- fisher.test(table(pvn_behav$Status, pvn_behav$Batch))
knitr::kable(table(pvn_behav$Status, pvn_behav$Batch), caption=paste0("Status across sequencing runs"))
```



```{r PVN DEseq1}
dd<- DESeqDataSetFromMatrix(countData=pvn_data, colData=pvn_behav, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_PVN_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_PVN_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = pvn_behav)

biplot(p, lab=pvn_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")


eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- pvn_behav[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Looks like we don't have any outliers, nor do we need to include Year (or Batch) as a covariate.

##Status PVN

Formula = ~ Status

```{r PVN Status, fig.height=8, fig.width=10}
#design(dd)<- formula(~ Status)
#dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


status_res<- data.frame(res1)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]



write.csv(status_res, file="../DE_results/results_PVN_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="PVN Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()
```

```{r PVN Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in PVN",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g


```

```{r plotting PVN, eval=FALSE}
PVN_modules<- read.csv("../WGCNA/ByTissue_2021/PVN_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B")


 tissue<- status_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  topgenes<- 1:6
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")


tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, PVN_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]
tissue$color<-factor(tissue$color,levels=c("grey60", "grey30","midnightblue","turquoise",      "darkolivegreen","blue","darkred","greenyellow","black","darkmagenta","white","saddlebrown","green"))
modulecols<- c("grey30", "grey60","midnightblue","turquoise",      "darkolivegreen","blue","darkred","greenyellow","black","darkmagenta","white","saddlebrown","green")


tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


#ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none")

```



```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(status_res$gene[which(status_res$padj<0.1)], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)



ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms by status in PVN", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_PVN_Status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p

```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot,p, labels=c("A","B","C"), nrow=3)

g2
ggsave(filename="supp_figure_PVN_Status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```


## Testosterone PVN

### mean T PVN
Formula = ~ mean_T

```{r PVN mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~ mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)




mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_PVN_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="PVN mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()

```


```{r PVN mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in PVN",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology
```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_PVN_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_PVN_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r PVN mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=pvn_data, colData=pvn_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "PVN"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_PVN_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="PVN Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in PVN",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


## Strength PVN

Formula ~ scale(Strength)

```{r PVN Strength, fig.height=4, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)




strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_PVN_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="PVN Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r PVN Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")

}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in PVN",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_PVN_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_PVN_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

# Medial Preoptic Area POM



```{r POM data set up}

rm(list= ls()[!(ls() %in% keep)])

pom_key<- subset(key_behav, Tissue=="POM")
pom_key<- droplevels(pom_key)

pom_behav<- pom_key[pom_key$Batch!="pilot",]

#Filtering
pom_data<- data[,colnames(data) %in% rownames(pom_key)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL

```
### Checking the sampling

For the POM, we started with `r start` genes but after filering we have `r nrow(pom_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pom samples for `r nrow(pom_behav)`, with respect to T and the tracking data.

```{r POM sampling}

fsb<- fisher.test(table(pom_key$Status, pom_key$Batch))
knitr::kable(table(pom_key$Status, pom_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))


fsyb<- fisher.test(table(pom_key$Batch, pom_key$Year))
knitr::kable(table(pom_key$Batch, pom_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=", fsyb$p.value))
```



```{r POM DEseq1}
dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)



p <- pca(vsd_data, metadata = pom_key)

biplot(p, lab=pom_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in POM")

eigencorplot(p, metavars = c("Year","Status", "Batch"))
```


```{r}
mat<- limma::removeBatchEffect(vsd_data, pom_key$Batch)
p <- pca(mat, metadata = pom_key)
biplot(p, lab=pom_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- pom_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

Don't need to control for year because accounting for Batch seems to fix that.

PFT2 was previously identified as a potential outlier, and it's coming out quite separately here, even after all the corrections for batch effects. So, this one I am going to REMOVE. 

```{r remove PFT2_POM}

outliers=c("PFT2_POM_run1")

pom_key<- pom_key[!rownames(pom_key) %in% outliers,]
pom_data<- data[,colnames(data) %in% rownames(pom_key)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL


```



##Status POM

Formula = ~ Batch + Status

```{r POM Status, fig.height=4, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_key, design= ~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]


write.csv(status_res, file="../DE_results/results_POM_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="POM Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r POM Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in POM",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12))
)
g
```


#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_POM_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_POM_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone POM
```{r POM continuous data set up}
#Filtering
pom_behav<- subset(key_behav, Tissue=="POM")
pom_behav<- subset(pom_behav, Batch!="pilot")
pom_behav<- droplevels(pom_behav)
pom_behav<- pom_behav[!rownames(pom_behav) %in% outliers,]
pom_data<- data[,colnames(data) %in% rownames(pom_behav)]
start<- nrow(pom_data)
#remove genes with less than 5 reads
pom_data$avg_count<- apply(pom_data, 1, mean)
pom_data<- pom_data[pom_data$avg_count>5,]
pom_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
pom_data$percent_0<- apply(pom_data, 1, function(x)length(x[x==0]))
thresh<- ncol(pom_data)/2
pom_data<- pom_data[pom_data$percent_0<=thresh,]
pom_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_POM_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = pom_behav)
biplot(p, lab=pom_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")
write.csv()


eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))
```


```{r}

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```



### mean T POM
Formula = ~ mean_T

```{r POM mean T, fig.height=8, fig.width=10}
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)





mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)

write.csv(mean_T_res, file="../DE_results/results_POM_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="POM mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```


```{r POM mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}

g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in POM",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g

```
```{r, eval=FALSE}
hub<-"PRLH"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$mean_T, dat$count)^2
ggplot(dat, aes(x=mean_T, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="mean Testosterone", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],2), ", p=",signif(res1$pvalue[rownames(res1)==hub],3), ", padj=", signif(res1$padj[rownames(res1)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")





```

```{r plotting POM, eval=FALSE}
POM_modules<- read.csv("../WGCNA/ByTissue_2021/POM_results_ModuleMembership.csv")


candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983511", "AVPR1A", "AVPR1B")

topgenes<- 1:6
 tissue<- mean_T_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983511"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, POM_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]
tissue$color<-factor(tissue$color,levels=c("grey60", "grey30","floralwhite","lightpink4","black","brown","blue4","lightsteelblue","bisque4","purple","mediumpurple4","coral3","lightcoral","navajowhite1"))
modulecols<- c("grey30", "grey60","floralwhite","lightpink4","black","brown","blue4","lightsteelblue","bisque4","purple","mediumpurple4","coral3","lightcoral","navajowhite1")


tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


#ggplot(tissue, aes(x=log2FoldChange, y=-log10(padj), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=1)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title="POM",subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-3,-2,-1,0,1,2,3), limits=c(-3,3))

```



#### Gene Ontology

```{r POM meanT GSEA, eval=FALSE}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
#write.csv(y@result, file="GSEA_BP_POM_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
#scatterPlot(simMatrix, reducedTerms)

x <- cmdscale(as.matrix(as.dist(1 - simMatrix)), eig = TRUE, 
        k = 2)
df <- cbind(as.data.frame(x$points), reducedTerms[match(rownames(x$points), 
        reducedTerms$go), c("term", "parent", "parentTerm", 
        "size")])
df$ID<-rownames(df)
df<- merge(df, y@result, by="ID")
df$pos<- ifelse(df$enrichmentScore>0,"pos","neg")
df$pos<- as.factor(df$pos)
#df$pos<- factor(df$pos, levels=c("neg","pos"))

labelSize=3
p <- ggplot(df, ggplot2::aes(x = V1, y = V2, color = pos)) + 
        geom_point(ggplot2::aes(size = size), alpha = 0.5) + 
        scale_color_manual(guide = FALSE, values=c("#00BFC4","#F8766D")) + scale_size_continuous(guide = FALSE, 
        range = c(0, 25)) + scale_x_continuous(name = "") + 
        scale_y_continuous(name = "") + theme_bw() + 
        theme(axis.text.x = element_blank(), 
            axis.text.y = element_blank())
p<- p + ggrepel::geom_label_repel(aes(label = parentTerm), 
            data = subset(df, parent == ID), box.padding = grid::unit(1, "lines"), size = labelSize) + labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```

```{r POM mean T GO}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(mean_T_res$gene[which(mean_T_res$padj<0.1)], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms by mean_T in POM", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_POM_mean_T.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,goplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_POM_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r POM mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "POM"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_POM_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="POM Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in POM",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



## Strength POM

Formula ~ scale(Strength)

```{r POM Strength, fig.height=4, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=pom_data, colData=pom_behav, design= ~ strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)




strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_POM_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="POM Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r POM Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in POM",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology


```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_POM_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_POM_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```


# Dorsomedial Intercollicular nucleus (ICO)



```{r ICO data set up}

rm(list= ls()[!(ls() %in% keep)])



ico_key<- subset(key_behav, Tissue=="ICO")
ico_key<- droplevels(ico_key)
ico_behav<- ico_key[ico_key$Batch!="pilot",]
#Filtering
ico_data<- data[,colnames(data) %in% rownames(ico_key)]
start<- nrow(ico_data)
#remove genes with less than 5 reads
ico_data$avg_count<- apply(ico_data, 1, mean)
ico_data<- ico_data[ico_data$avg_count>5,]
ico_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression



ico_data$percent_0<- apply(ico_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ico_data)/2
ico_data<- ico_data[ico_data$percent_0<=thresh,]
ico_data$percent_0<-NULL

#for some reason the the above code stalls when I try to knit the document so I am #just going to manually remove the sites


#low_genes<- c("LOC113990771","LOC114004308") 
#ico_data<- ico_data[!rownames(ico_data) %in% low_genes,]

```


### Checking the sampling

For the ICO, we started with `r start` genes but after filering we have `r nrow(ico_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ico samples for `r nrow(ico_behav)`, with respect to T and the tracking data.

```{r ICO sampling}

fsb<- fisher.test(table(ico_key$Status, ico_key$Batch))
knitr::kable(table(ico_key$Status, ico_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))


fsyb<- fisher.test(table(ico_key$Batch, ico_key$Year))
knitr::kable(table(ico_key$Batch, ico_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))


mean_t_wtest<- wilcox.test(ico_behav$mean_T ~ ico_behav$Batch, alternative="two.sided")
a<- ggplot(ico_behav, aes(x=Batch, y=mean_T))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme


strength_wtest<- wilcox.test(ico_behav$strength.all_study ~ ico_behav$Batch, alternative="two.sided")
d<- ggplot(ico_behav, aes(x=Batch, y=strength.all_study))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables ICO",gp=gpar(fontsize=14)))

```



```{r ICO DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)


p <- pca(vsd_data, metadata = ico_key)

biplot(p, lab=ico_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in ICO")

eigencorplot(p, metavars = c("Year","Status", "Batch"))


```



```{r}
mat<- limma::removeBatchEffect(vsd_data, ico_key$Batch)
p <- pca(mat, metadata = ico_key)
biplot(p, lab=ico_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ico_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```


## Status ICO

Formula = ~ Batch + Status

```{r ICO Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]

write.csv(status_res, file="../DE_results/results_ICO_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="LS Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r ICO Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in ICO",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology
```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
#write.csv(y@result, file="GSEA_BP_ICO_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_ICO_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone ICO
```{r ICO continuous data set up}
#Filtering
ico_behav<- subset(key_behav, Tissue=="ICO")
ico_behav<- subset(ico_behav, Batch!="pilot")
ico_behav<- droplevels(ico_behav)
ico_data<- data[,colnames(data) %in% rownames(ico_behav)]
start<- nrow(ico_data)
#remove genes with less than 5 reads
ico_data$avg_count<- apply(ico_data, 1, mean)
ico_data<- ico_data[ico_data$avg_count>5,]
ico_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ico_data$percent_0<- apply(ico_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ico_data)/2
ico_data<- ico_data[ico_data$percent_0<=thresh,]
ico_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)

write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_LS_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = ico_behav)

biplot(p, lab=ico_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "Batch", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

mat<- limma::removeBatchEffect(vsd_data, ico_behav$Batch)
p <- pca(mat, metadata = ico_behav)
biplot(p, lab=ico_behav$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))


```

Still need to control for Year here... 

### mean T ICO
Formula = ~ Batch + Year + mean_T

```{r ICO mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~Batch + Year + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)

write.csv(mean_T_res, file="../DE_results/results_ICO_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="ICO mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```


```{r ICO mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in ICO",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r, eval=FALSE}
hub<- "MMP2"
out<-  plotCounts(dd, gene=hub, intgroup=c("mean_T", "Class"), returnData=TRUE)
Rsq<- cor(out$mean_T, out$count)^2
i=3
ggplot(out, aes(x=mean_T, y=count, color=Class)) + geom_point(size=4) + labs(title=paste0(row.names(res1[hub,])," LFC=",signif(mean_T_res$log2FoldChange[mean_T_res$gene==hub],2) ," padj=", formatC(mean_T_res$padj[mean_T_res$gene==hub],2, format="e"), " R^2=", signif(Rsq,2)),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")+ theme(legend.position="none")
```



#### Gene Ontology

```{r, eval=FALSE}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
ridgeplot(y)
go_mean_T<- y@result
write.csv(y@result, file="GSEA_BP_ICO_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(mean_T_res$gene[which(mean_T_res$padj<0.1)], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms by mean_T in ICO", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_ICO_mean_T.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot,p, labels=c("A","B","C"), nrow=3)

g2
ggsave(filename="supp_figure_ICO_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r ICO mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=ico_data, colData=ico_behav, design= ~ Batch + Year + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)

interaction_res<- as.data.frame(res1)
interaction_res$gene<- rownames(res1)
interaction_res$Tissue<- "ICO"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_ICO_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="ICO mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res1[i,])," padj=", formatC(res1$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in ICO",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


## Strength ICO

Formula ~ Batch + Year + scale(Strength)

```{r ICO Strength, fig.height=8, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch + Year + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)


strength_res<- data.frame(res1[order(rownames(res1)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_ICO_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="ICO Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```



```{r ICO Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in ICO",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r, eval=FALSE}

hub<-"VIPR2"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
ggplot(dat, aes(x=strength.all_study, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="strength", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res1$log2FoldChange[rownames(res1)==hub],2), ", p=",signif(res1$pvalue[rownames(res1)==hub],2), ", padj=", signif(res1$padj[rownames(res1)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")


```

#### Gene Ontology

```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result

write.csv(y@result, file="../DE_results/GSEA_BP_ICO_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_ICO_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

# Midbrain Central Grey GCT


```{r GCT data set up}

rm(list= ls()[!(ls() %in% keep)])

gct_key<- subset(key_behav, Tissue=="GCT")
gct_key<- droplevels(gct_key)


gct_behav<- gct_key[gct_key$Batch!="pilot",]

#Filtering
gct_data<- data[,colnames(data) %in% rownames(gct_key)]
start<- nrow(gct_data)
#remove genes with less than 5 reads
gct_data$avg_count<- apply(gct_data, 1, mean)
gct_data<- gct_data[gct_data$avg_count>5,]
gct_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gct_data$percent_0<- apply(gct_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gct_data)/2
gct_data<- gct_data[gct_data$percent_0<=thresh,]
gct_data$percent_0<-NULL


```
### Checking the sampling

For the GCT, we started with `r start` genes but after filering we have `r nrow(gct_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gct samples for `r nrow(gct_behav)`, with respect to T and the tracking data.

```{r GCT sampling}

fsb<- fisher.test(table(gct_key$Status, gct_key$Batch))
knitr::kable(table(gct_key$Status, gct_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))

fsyb<- fisher.test(table(gct_key$Batch, gct_key$Year))
knitr::kable(table(gct_key$Batch, gct_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))


mean_t_wtest<- wilcox.test(gct_behav$mean_T ~ gct_behav$Batch, alternative="two.sided")
a<- ggplot(gct_behav, aes(x=Batch, y=mean_T))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme




strength_wtest<- wilcox.test(gct_behav$strength.all_study ~ gct_behav$Batch, alternative="two.sided")
d<- ggplot(gct_behav, aes(x=Batch, y=strength.all_study))  + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, nrow=1, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables GCT",gp=gpar(fontsize=14)))

```

given there is only 1 GCT sample from the pilot, I will just exclude the pilot data.

```{r GCT DEseq1}
dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)


p <- pca(vsd_data, metadata = gct_key)
biplot(p, lab=gct_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status")
#biplot(p, lab=gct_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "Batch"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK


mat<- limma::removeBatchEffect(vsd_data, gct_key$Batch)
p <- pca(mat, metadata = gct_key)
biplot(p, lab=gct_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- gct_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```




##Status GCT

Formula = ~ Batch + Status

```{r GCT Status, fig.height=8, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]

summary(res1)




status_res<- data.frame(res1)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]

write.csv(status_res, file="../DE_results/results_GCT_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GCT Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()
```

```{r GCT Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in GCT",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),
left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result

write.csv(y@result, file="../DE_results/GSEA_BP_GCT_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_GCT_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone GCT

```{r GCT Continuous data setup}
#Filtering
gct_behav<- subset(key_behav, Tissue=="GCT")
gct_behav<- subset(gct_behav, Batch!="pilot")
gct_behav<- droplevels(gct_behav)

gct_data<- data[,colnames(data) %in% rownames(gct_behav)]
start<- nrow(gct_data)
#remove genes with less than 5 reads
gct_data$avg_count<- apply(gct_data, 1, mean)
gct_data<- gct_data[gct_data$avg_count>5,]
gct_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
gct_data$percent_0<- apply(gct_data, 1, function(x)length(x[x==0]))
thresh<- ncol(gct_data)/2
gct_data<- gct_data[gct_data$percent_0<=thresh,]
gct_data$percent_0<-NULL



dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_GCT_norm.csv", row.names=TRUE, quote=FALSE)

p <- pca(vsd_data, metadata = gct_behav)

biplot(p, lab=gct_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "Batch", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))



datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

sampleTree = hclust(dist(datExpr0), method = "average")

par(mar = c(0, 4, 2, 0))
plot(sampleTree, main = "Sample clustering on all genes in GCT", xlab="", sub="", cex = 1)
dev.off()



```


### mean T GCT
Formula = ~ Batch + mean_T

```{r GCT mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)
resLRT<- DESeq(dd, test="LRT", reduced=~Batch)
resLRT2<- results(resLRT, alpha=0.1)
res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_GCT_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GCT mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")
dev.off()

```


```{r GCT mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in GCT",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```


#### Gene Ontology

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)

gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

go_mean_T<- y@result

write.csv(y@result, file="../DE_results/GSEA_BP_GCT_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))

```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_GCT_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r GCT mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=gct_data, colData=gct_behav, design= ~ Batch + Status +  mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")



res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "GCT"
write.csv(interaction_res, file="../DE_results/interactions_GCT_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="GCT mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(resLRT2[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(resLRT2[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(resLRT2[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(resLRT2[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(resLRT2[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(resLRT2[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(resLRT2[i,])," padj=", formatC(resLRT2$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in GCT",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g

geneid<-"NAAA"
blah<- plotCounts(dd, gene=row.names(resLRT2[geneid,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point()
```




## Strength GCT

Formula ~ Batch + scale(Strength)

```{r GCT Strength, fig.height=4, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~Batch +  strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)





strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)


write.csv(strength_res, file="../DE_results/results_GCT_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="GCT Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r GCT Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in GCT",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_GCT_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))

```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_GCT_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```



# Nucleus Taenia TNA


```{r TNA data set up}

rm(list= ls()[!(ls() %in% keep)])

tna_key<- subset(key_behav, Tissue=="TNA")
tna_key<- droplevels(tna_key)

tna_behav<- tna_key[tna_key$Batch!="pilot",]

#Filtering
tna_data<- data[,colnames(data) %in% rownames(tna_key)]
start<- nrow(tna_data)
#remove genes with less than 5 reads
tna_data$avg_count<- apply(tna_data, 1, mean)
tna_data<- tna_data[tna_data$avg_count>5,]
tna_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
tna_data$percent_0<- apply(tna_data, 1, function(x)length(x[x==0]))
thresh<- ncol(tna_data)/2
tna_data<- tna_data[tna_data$percent_0<=thresh,]
tna_data$percent_0<-NULL

```
### Checking the sampling

For the TNA, we started with `r start` genes but after filering we have `r nrow(tna_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have tna samples for `r nrow(tna_behav)`, with respect to T and the tracking data.

```{r TNA sampling}

fsb<- fisher.test(table(tna_key$Status, tna_key$Batch))
knitr::kable(table(tna_key$Status, tna_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))


fsyb<- fisher.test(table(tna_key$Batch, tna_key$Year))
knitr::kable(table(tna_key$Batch, tna_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))


mean_t_wtest<- wilcox.test(tna_behav$mean_T ~ tna_behav$Batch, alternative="two.sided")
a<- ggplot(tna_behav, aes(x=Batch, y=mean_T)) + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Mean Testosterone, W=", mean_t_wtest$statistic, " p=", signif(mean_t_wtest$p.value,3)), y="mean Testosterone") + peri_theme



strength_wtest<- wilcox.test(tna_behav$strength.all_study ~ tna_behav$Batch, alternative="two.sided")
d<- ggplot(tna_behav, aes(x=Batch, y=strength.all_study)) + geom_point(colour=dj_col[1], size=4) + labs(title=paste0("Social Network Strength, W=", strength_wtest$statistic, " p=", signif(strength_wtest$p.value,3)), y="social network strength") + peri_theme

grid.arrange(a,d, ncol=2, newpage=TRUE, top = textGrob("Interest Variables vs Nuisance Variables TNA",gp=gpar(fontsize=14)))

```



```{r TNA DEseq1, fig.height=8, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = tna_key)

biplot(p, lab=tna_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in TNA")

eigencorplot(p, metavars = c("Year","Status", "Batch"))




```

```{r}
mat<- limma::removeBatchEffect(vsd_data, tna_key$Batch)
p <- pca(mat, metadata = tna_key)
biplot(p, lab=tna_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- tna_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

PFT1 was previously identified as an outlier based on WGCNA dendrograms only. But here it's not consistently dentified as an outlier.Also previous analysis shows that it doesn't have a strong influence on the results, so I am going to keep it in this data. Looks like we only need to control for Batch, as it removes the effects of year on our data.

##Status TNA

Formula = ~ Batch + Status

```{r TNA Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]


write.csv(status_res, file="../DE_results/results_TNA_Status.csv", row.names=FALSE)




par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="TNA Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r TNA Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status" , "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in TNA",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_TNA_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)
p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r,eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_TNA_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone TNA
```{r TNA continuous data set up}
#Filtering
tna_behav<- subset(key_behav, Tissue=="TNA")
tna_behav<- subset(tna_behav, Batch!="pilot")
tna_behav<- droplevels(tna_behav)


tna_data<- data[,colnames(data) %in% rownames(tna_behav)]
start<- nrow(tna_data)
#remove genes with less than 5 reads
tna_data$avg_count<- apply(tna_data, 1, mean)
tna_data<- tna_data[tna_data$avg_count>5,]
tna_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
tna_data$percent_0<- apply(tna_data, 1, function(x)length(x[x==0]))
thresh<- ncol(tna_data)/2
tna_data<- tna_data[tna_data$percent_0<=thresh,]
tna_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_TNA_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_TNA_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = tna_behav)
#biplot(p, lab=tna_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")
biplot(p, lab=tna_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "Batch", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))

#plotloadings(p, labSize = 3, components = getComponents(p, c(1,2,3)), rangeRetain=0.1)



mat<- limma::removeBatchEffect(vsd_data, tna_behav$Batch)
p <- pca(mat, metadata = tna_behav)
biplot(p, lab=tna_behav$Harvest_ID, colby="mean_T", shape="Batch", legendPosition="right", title="PCA TNA removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Taking the batch effect into account seems to remove the association with Year too. Only need to include Batch in the model. 

### mean T TNA
Formula = ~ Batch + mean_T

```{r TNA mean T, fig.height=4, fig.width=10}
design(dd)<- formula(~Batch + mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


mean_T_res<- data.frame(res[order(rownames(res)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_TNA_mean_T.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="TNA mean Testosterone", )
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r plotting TNA, eval=FALSE}
TNA_modules<- read.csv("../WGCNA/ByTissue_2021/TNA_results_ModuleMembership.csv")

topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


#i="TNA"
 tissue<- mean_T_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>1, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, TNA_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","coral1","lightyellow","blue","coral3","tan","black","blue2","honeydew","skyblue2","darkorange","skyblue4","antiquewhite4","grey","darkorange2","lightsteelblue"))
modulecols<- c("grey30","grey60","coral1","lightyellow","blue","coral3","tan","black","blue2","honeydew","skyblue2","darkorange","skyblue4","antiquewhite4","grey","darkorange2","lightsteelblue")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")


  

ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title="TnA",subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-4,-3,-2,-1,0,1,2,3,4))
```



```{r TNA mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("mean_T" ,"Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("mean_T","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}





g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in TNA",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r, eval=FALSE}
hub<-"LOC113992168"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "mean_T"), returnData=TRUE)
Rsq<- cor(dat$mean_T, dat$count)^2
ggplot(dat, aes(x=mean_T, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="mean Testosterone", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res$log2FoldChange[rownames(res)==hub],2), ", p=",signif(res$pvalue[rownames(res)==hub],3), ", padj=", signif(res$padj[rownames(res)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")
```

#### Gene Ontology
```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
y<- enricher(mean_T_res$gene[mean_T_res$padj<0.1], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms in TnA", row.names = FALSE) %>% kable_styling()

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_TNA_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
p
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot, p,labels=c("A","B","C"), nrow=3)

g2
ggsave(filename="supp_figure_TNA_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r TNA mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Batch + Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "TNA"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_TNA_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="TNA Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in TNA",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



## Strength TNA

Formula ~ Batch + scale(Strength)

```{r TNA Strength, fig.height=4, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=tna_data, colData=tna_behav, design= ~ Batch + strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~Batch + strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]

summary(res)




strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_TNA_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="TNA Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r TNA Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in TNA",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12))
)
g


```

#### Gene Ontology

There are no enriched terms here. 

```{r}
#res_go<- strength_res
#res_go$logP<- -log(res_go$pvalue, 10)
#res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
#res_go$gene<- rownames(res_go)
#res_go<- as.data.frame(res_go)
#res_go<- merge(res_go, go_key, by="gene")
#res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
#res_go<- res_go[!is.na(res_go$logP),]
#res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))
#
#geneList<- res_go$logP
#names(geneList)<- res_go$gene
#
#
#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#
#gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + #scale_fill_viridis()
#gseaplot
#
#go_mean_T<- y@result
##write.csv(y@result, file="GSEA_BP_TNA_strength.csv", row.names=FALSE)
#ego2<- as.data.frame(go_mean_T$ID)
#colnames(ego2)<- "ID"
#ego2$p.adjust<- go_mean_T$p.adjust
#
#simMatrix <- calculateSimMatrix(ego2$ID,
#                                orgdb="org.Hs.eg.db",
#                                ont="BP",
#                                method="Rel")
#scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
#reducedTerms <- reduceSimMatrix(simMatrix,
#                                scores,
#                                threshold=0.7,
#                                orgdb="org.Hs.eg.db")
#treemapPlot(reducedTerms)
##scatterPlot(simMatrix, reducedTerms)
#
#p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g, gseaplot, p, labels=c("A","B","C"), nrow=3, ncol=1)
ggsave(filename="supp_figure_TNA_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")

```



# Arcopallium intermedium AI


```{r AI data set up}

rm(list= ls()[!(ls() %in% keep)])

ai_key<- subset(key_behav, Tissue=="AI")
ai_key<- droplevels(ai_key)

ai_behav<- ai_key[ai_key$Batch!="pilot",]
#Filtering
ai_data<- data[,colnames(data) %in% rownames(ai_key)]
start<- nrow(ai_data)
#remove genes with less than 5 reads
ai_data$avg_count<- apply(ai_data, 1, mean)
ai_data<- ai_data[ai_data$avg_count>5,]
ai_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ai_data$percent_0<- apply(ai_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ai_data)/2
ai_data<- ai_data[ai_data$percent_0<=thresh,]
ai_data$percent_0<-NULL

```
### Checking the sampling

For the AI, we started with `r start` genes but after filering we have `r nrow(ai_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ai samples for `r nrow(ai_behav)`, with respect to T and the tracking data.

```{r AI sampling}

fsb<- fisher.test(table(ai_key$Status, ai_key$Batch))
knitr::kable(table(ai_key$Status, ai_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))



fsyb<- fisher.test(table(ai_key$Batch, ai_key$Year))
knitr::kable(table(ai_key$Batch, ai_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))
```



```{r AI DEseq1}
dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = ai_key)

biplot(p, lab=ai_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in AI")

eigencorplot(p, metavars = c("Year","Status", "Batch"))


datExpr0<- as.data.frame(t(vsd_data))
#gsg<- goodSamplesGenes(datExpr0, verbose=3)
#gsg$allOK

sampleTree = hclust(dist(datExpr0), method = "average")

par(mar = c(0, 4, 2, 0))
plot(sampleTree, main = "Sample clustering on all genes in TNA", xlab="", sub="", cex = 1)
dev.off()

mat<- limma::removeBatchEffect(vsd_data, ai_key$Batch)
p <- pca(mat, metadata = ai_key)
biplot(p, lab=ai_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))


wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ai_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```

Removing the batch effect effectively controls for Year.

##Status AI

Formula = ~ Batch + Status

```{r AI Status, fig.height=8, fig.width=10}
design(dd)<- formula(~ Batch+ Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)



status_res<- data.frame(res1)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]

write.csv(status_res, file="../DE_results/results_AI_Status.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```

```{r AI Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("Status","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("Status","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("Status","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("Status","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("Status","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("Status","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in AI",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)), left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```


#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_AI_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_AI_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone AI
```{r AI continuous data set up}
#Filtering
ai_behav<- subset(key_behav, Tissue=="AI")
ai_behav<- subset(ai_behav, Batch!="pilot")
ai_behav<- droplevels(ai_behav)

ai_data<- data[,colnames(data) %in% rownames(ai_behav)]
start<- nrow(ai_data)
#remove genes with less than 5 reads
ai_data$avg_count<- apply(ai_data, 1, mean)
ai_data<- ai_data[ai_data$avg_count>5,]
ai_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ai_data$percent_0<- apply(ai_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ai_data)/2
ai_data<- ai_data[ai_data$percent_0<=thresh,]
ai_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)
#write.csv(vsd_data, "data_AI_vsd.csv", row.names=TRUE, quote=FALSE)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_AI_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ai_behav)
#biplot(p, lab=ai_behav$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status")
biplot(p, lab=ai_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

No association with year. There are 3 samples that fall out from the main group. 

### mean T AI
Formula = ~ mean_T

```{r AI mean T, fig.height=8, fig.width=10}

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)




mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)

write.csv(mean_T_res, file="../DE_results/results_AI_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()

```


```{r AI mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in AI",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_AI_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_AI_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

### mean T x Status


```{r AI mean T x Status}

dd<- DESeqDataSetFromMatrix(countData=ai_data, colData=ai_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "AI"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_AI_mean_T_Status.csv", row.names=FALSE)



par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in AI",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


## Strength AI

Formula ~ scale(Strength)

```{r AI Strength, fig.height=8, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)

res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")

outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]
summary(res1)





strength_res<- data.frame(res1[order(rownames(res1)),])
strength_res$gene<- row.names(strength_res)


write.csv(strength_res, file="../DE_results/results_AI_Strength.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="AI Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="FDR tool corrected")
hist(res1.fdr$pval, breaks=20, col="grey", main="P-value correction 'fdrtool'", xlab="'Corrected' 
P-value")
dev.off()
```



```{r AI Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in AI",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

#### Gene Ontology

```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_AI_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                              orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                              orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<-ggarrange(g, gseaplot,p , labels=c("A","B","C"), nrow=3)

g2
ggsave(filename="supp_figure_AI_strength.png", plot=g2, device="png" ,height=11, width=9, units="in")
```

# Lateral Septum LS


```{r LS data set up}

rm(list= ls()[!(ls() %in% keep)])

ls_key<- subset(key_behav, Tissue=="LS")
ls_key<- droplevels(ls_key)


ls_behav<- ls_key[ls_key$Batch!="pilot",]

#Filtering
ls_data<- data[,colnames(data) %in% rownames(ls_key)]
start<- nrow(ls_data)
#remove genes with less than 5 reads
ls_data$avg_count<- apply(ls_data, 1, mean)
ls_data<- ls_data[ls_data$avg_count>5,]
ls_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ls_data$percent_0<- apply(ls_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ls_data)/2
ls_data<- ls_data[ls_data$percent_0<=thresh,]
ls_data$percent_0<-NULL

```

### Checking the sampling

For the LS, we started with `r start` genes but after filering we have `r nrow(ls_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ls samples for `r nrow(ls_behav)`, with respect to T and the tracking data.

```{r LS sampling}

fsb<- fisher.test(table(ls_key$Status, ls_key$Batch))
knitr::kable(table(ls_key$Status, ls_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))



fsyb<- fisher.test(table(ls_key$Batch, ls_key$Year))
knitr::kable(table(ls_key$Batch, ls_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))


```



```{r LS DEseq1, fig.height=8, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = ls_key)

biplot(p, lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in LS")
#biplot(p, x="PC3", y="PC4", lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="Status in LS")
eigencorplot(p, metavars = c("Year","Status", "Batch"))

```

```{r}
mat<- limma::removeBatchEffect(vsd_data, ls_key$Batch)
p <- pca(mat, metadata = ls_key)
biplot(p, lab=ls_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- ls_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```

PFT2 was previously identified as an outlier, but using the quantitative threshold and controlling for batch it doesn't come out as different. Controlling for Batch adequately controls for Year.

##Status LS

Formula = ~ Batch + Status

```{r LS Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]

write.csv(status_res, file="../DE_results/results_LS_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="LS Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```

```{r LS Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in LS",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```


#### Gene Ontology

```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_LS_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))

```

```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_LS_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```

## Testosterone LS
```{r LS continuous data set up}
#Filtering
ls_behav<- subset(key_behav, Tissue=="LS")
ls_behav<- subset(ls_behav, Batch!="pilot")
ls_behav<- droplevels(ls_behav)
ls_data<- data[,colnames(data) %in% rownames(ls_behav)]
start<- nrow(ls_data)
#remove genes with less than 5 reads
ls_data$avg_count<- apply(ls_data, 1, mean)
ls_data<- ls_data[ls_data$avg_count>5,]
ls_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
ls_data$percent_0<- apply(ls_data, 1, function(x)length(x[x==0]))
thresh<- ncol(ls_data)/2
ls_data<- ls_data[ls_data$percent_0<=thresh,]
ls_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_LS_norm.csv", row.names=TRUE, quote=FALSE)


p <- pca(vsd_data, metadata = ls_behav)

biplot(p, lab=ls_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")

eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK




A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")
```



### mean T LS
Formula = ~ mean_T

```{r LS mean T, fig.height=8, fig.width=10}
design(dd)<- formula(~mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res1<- res[!is.na(res$padj),]
res1<- res1[!is.na(res1$pvalue),]
res1$padj<- NULL
res1.fdr<- fdrtool(res1$stat, statistic="normal", plot=FALSE, verbose=FALSE)
res1$padj<- p.adjust(res1.fdr$pval, method="BH")
outs<- res[is.na(res$padj),] # add those outliers back in... 
res1<- rbind(res1, outs)
res1<- res1[order(res1$padj),]

summary(res1)




mean_T_res<- data.frame(res1[order(rownames(res1)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_LS_mean_T.csv", row.names=FALSE)


par(mfrow=c(2,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="LS mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
DESeq2::plotMA(res1, ylim = c(-2, 2), main="fdr tools adjusted")
hist(res1.fdr$pval, breaks=20, col="grey", main="", xlab="Corrected P-value")

dev.off()

```


```{r LS mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res1[1,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res1[2,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res1[3,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res1[4,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res1[5,]), intgroup=c("mean_T", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res1[6,]), intgroup=c("mean_T", "Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res1$log2FoldChange[i],2)
  padj<-formatC(res1$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res1[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in LS",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```



#### Gene Ontology

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot


go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_LS_meanT.csv", row.names=FALSE)

ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_LS_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```
### mean T x Status


```{r LS mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd,res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "LS"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_LS_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="AI Status * mean Testosterone ")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in LS",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```



## Strength LS

Formula ~ scale(Strength)

```{r LS Strength, fig.height=4, fig.width=10}
dd<- DESeqDataSetFromMatrix(countData=ls_data, colData=ls_behav, design= ~ strength.all_study)
dd$strength.all_study<- scale(dd$strength.all_study) # the unscaled version of this says it should be scaled for best model performance
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)



strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_LS_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="LS Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```



```{r LS Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study", "Class"), returnData=TRUE)


vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in LS",gp=gpar(fontsize=12)), bottom=textGrob("Social Network Strength",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r, eval=FALSE}
hub<-"NPY1R"

dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
ggplot(dat, aes(x=strength.all_study, y=count, color=Status)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="strength", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res$log2FoldChange[rownames(res)==hub],2), ", p=",signif(res$pvalue[rownames(res)==hub],2), ", padj=", signif(res$padj[rownames(res)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none")
```


```{r plotting LS, eval=FALSE}
LS_modules<- read.csv("../WGCNA/ByTissue_2021/LS_results_ModuleMembership.csv")


i="LS"
topgenes<- 1:6
candidates<- c("AR", "SRD5A2", "CYP19A1", "LOC113993669","ESR1", "ESR2", "PGR","GNRH1", "PRL","PRLR","VIP", "VIPR1", "VIPR2","OXT", "LOC113983511","OXTR", "AVP","LOC113983498", "AVPR1A", "AVPR1B")


tissue<- strength_res
  tissue<- tissue[!is.na(tissue$pvalue),]
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  #tissue$gene<- plyr::revalue(tissue$gene, c("LOC113993669"="CYP19A1","LOC113983511"="OXT", "LOC113983498"="AVP"))

  tissue$sig<- ifelse(tissue$padj<0.1,"Q<0.1",ifelse(tissue$pvalue<0.05,"P<0.05","NS"))
  tissue$highlfc<- ifelse(tissue$padj<0.1 & abs(tissue$log2FoldChange)>0.3, "high", "NS")
  
  tissue<- tissue[order(tissue$padj),]
  rownames(tissue)<- 1:nrow(tissue)
  
tissue$interesting_genes<- ifelse((tissue$gene %in% candidates & tissue$pvalue<0.05) | rownames(tissue) %in% topgenes | tissue$highlfc=="high", as.character(tissue$gene), "")
tissue$is_interesting<- tissue$interesting_genes==""  
  
tissue<- merge(tissue, LS_modules[,c("gene","best_anno","moduleColor")], by="gene")
tissue$color<- ifelse(tissue$pvalue<0.05, "grey30", "grey60")
tissue$color<- ifelse(tissue$padj<0.1, as.character(tissue$moduleColor), as.character(tissue$color))
tissue<- tissue[!is.na(tissue$color),]

tissue$color<-factor(tissue$color,levels=c("grey60","grey30","brown4","black","green","navajowhite2",  "darkorange","lightsteelblue1","orangered4","darkmagenta","maroon","blue","darkgrey","paleturquoise"))
modulecols<- c("grey30","grey60","brown4","black","green","navajowhite2",  "darkorange","lightsteelblue1","orangered4","darkmagenta","maroon","blue","darkgrey","paleturquoise")

tissue$interesting_genes<- ifelse(tissue$interesting_genes!="", as.character(tissue$best_anno),"")
tissue$outline<- ifelse(tissue$padj<0.1 | tissue$is_interesting==FALSE, "yes", "no")



#ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=sig)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + geom_point(data=tissue[tissue$interesting_genes!="",], aes(fill=sig), pch=21,color="grey40", show.legend=FALSE, size=5) + scale_color_manual(values=tissue_cols[[i]]) + scale_fill_manual(values=tissue_cols[[i]][2:3]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + geom_hline(yintercept=-log10(0.1), linetype="dashed", color = "grey70", size=0.0)


ggplot(tissue, aes(x=log2FoldChange, y=-log10(pvalue), color=color)) + geom_point(size=4) + geom_text_repel(aes(label=interesting_genes), color="grey40",size=8, label.padding=1, force_pull=0, force=6) + peri_theme + scale_color_manual(values=modulecols) + geom_point(data=tissue[tissue$outline=="yes",],fill=NA, pch=21, stroke=0.01, color="grey40", show.legend=FALSE, size=5) + scale_fill_manual(values=modulecols[3:length(modulecols)]) + labs(title=i,subtitle=paste0(nrow(tissue),"/",nrow(tissue[which(tissue$pvalue<0.05),]),"/",nrow(tissue[which(tissue$padj<0.1),])),x="", y="") + theme(legend.position = "none") + scale_x_continuous(breaks=c(-0.6,-0.4,-0.2,0,0.2,0.4,0.6))

```


#### Gene Ontology
```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


#y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)

y<- enricher(strength_res$gene[strength_res$padj<0.1], universe=res_go$gene, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
#ridgeplot(y)
ego2<- y@result
ego2$GeneRatio2<- word(ego2$GeneRatio, 1,1, sep="/")
ego2$GeneRatio2<- as.numeric(ego2$GeneRatio2)

goplot<- ggplot(ego2[1:20,], aes(x=Description, y=GeneRatio2, fill=qvalue)) + geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1)) + coord_flip() + peri_theme + scale_y_continuous(expand=c(0,0)) + labs(x="", y="No. genes enriched", title="Top 20 GO BP") + scale_fill_viridis()
goplot

knitr::kable(y@result[1:10,], caption="BP enriched GO terms in LS", row.names = FALSE) %>% kable_styling()




go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GO_BP_LS_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)

p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```

```{r, eval=FALSE}
g2<-ggarrange(g, goplot, labels=c("A","B"), nrow=2)

g2
ggsave(filename="supp_figure_LS_strength.png", plot=g2, device="png" ,height=11, width=9, units="in")
```
# Bed Nucleus of the Stria Terminalis (BSTm)



```{r BSTm data set up}

rm(list= ls()[!(ls() %in% keep)])

bstm_key<- subset(key_behav, Tissue=="BSTm")
bstm_key<- droplevels(bstm_key)

bstm_behav<- bstm_key[bstm_key$Batch!="pilot",]

#Filtering
bstm_data<- data[,colnames(data) %in% rownames(bstm_key)]
start<- nrow(bstm_data)
#remove genes with less than 5 reads
bstm_data$avg_count<- apply(bstm_data, 1, mean)
bstm_data<- bstm_data[bstm_data$avg_count>5,]
bstm_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
bstm_data$percent_0<- apply(bstm_data, 1, function(x)length(x[x==0]))
thresh<- ncol(bstm_data)/2
bstm_data<- bstm_data[bstm_data$percent_0<=thresh,]
bstm_data$percent_0<-NULL

```
### Checking the sampling

For the BSTm, we started with `r start` genes but after filering we have `r nrow(bstm_data)` genes. 
Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have bstm samples for `r nrow(bstm_behav)`, with respect to T and the tracking data.

```{r BSTm sampling}

fsb<- fisher.test(table(bstm_key$Status, bstm_key$Batch))
knitr::kable(table(bstm_key$Status, bstm_key$Batch), caption=paste0("Status across sequencing runs, fisher test p=", fsb$p.value))



fsyb<- fisher.test(table(bstm_key$Batch, bstm_key$Year))
knitr::kable(table(bstm_key$Batch, bstm_key$Year), caption=paste0("Sampling years distributed across batches, fisher test p=",fsyb$p.value))


```



```{r BSTm DEseq1}
dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_key, design= ~ Status)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
#vsd_data<- assay(vsd)


p <- pca(vsd_data, metadata = bstm_key)

biplot(p, lab=bstm_key$Harvest_ID, colby="Status", shape="Year", legendPosition="right", title="Status in BSTm")

eigencorplot(p, metavars = c("Year","Status", "Batch"))




```


```{r}

mat<- limma::removeBatchEffect(vsd_data, bstm_key$Batch)
p <- pca(mat, metadata = bstm_key)
#biplot(p, lab=gct_key$Harvest_ID, colby="Status", shape="Batch", legendPosition="right", title="PCA Gonads removeBatchEffect(Batch)")
eigencorplot(p, metavars = c("Batch","Year", "Status"))

wgcnadata<- as.data.frame(t(mat))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


traits<- bstm_key[, c("Batch",  "Year")]
traits$pilot<- as.numeric(ifelse(traits$Batch=="pilot",1,0))
traits$Run1<- as.numeric(ifelse(traits$Batch=="run1",1,0))
traits$Run2<- as.numeric(ifelse(traits$Batch=="run2",1,0))
traits$yr2015<- as.numeric(ifelse(traits$Year=="2015",1,0))
traits$yr2017<- as.numeric(ifelse(traits$Year=="2017",1,0))
traits$yr2018<- as.numeric(ifelse(traits$Year=="2018",1,0))

traits<- traits[,-(1:2)] 



A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors
traitColors=data.frame(numbers2colors(traits,signed=FALSE))
dimnames(traitColors)[[2]]=paste(names(traits))
datColors=data.frame(outlier=outlierColor,traitColors)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")


```

Batch controls effectively for Year.

##Status BSTm

Formula = ~ Batch + Status

```{r BSTm Status, fig.height=4, fig.width=10}
design(dd)<- formula(~ Batch + Status)
dd<- DESeq(dd)
res<- results(dd, alpha=0.1)
res<- res[order(res$padj),]
summary(res)



status_res<- data.frame(res)
status_res$gene<- row.names(status_res)
status_res<- status_res[order(status_res$gene),]


write.csv(status_res, file="../DE_results/results_BSTm_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="BSTm Territorial vs Floater")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

#EnhancedVolcano(res,
#    lab = rownames(res),
#    x = 'log2FoldChange',
#    y = 'padj',
#    ylab = bquote(~Log[10]~ 'padj'),
#    pCutoff=0.1,
#    FCcutoff = 0,
#    pointSize = 3.0,
#    labSize = 4.0,
#    col=c('grey20', 'grey20', 'grey20', 'red3'),
#    colAlpha = 1,
#    legendLabels=c("",'Not sig.',"",'significant padj<0.1'),
#    drawConnectors = TRUE)
```

```{r BSTm Status Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=Status, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b),list(a=lfc, b=padj))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status in BSTm",gp=gpar(fontsize=12)), bottom=textGrob("Status",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g
```

```{r,eval=FALSE}

#AR<- plotCounts(dd, gene=row.names(res[row.names(res)=="AR",]), intgroup=c("Status"), returnData=TRUE)
#ggplot(AR, aes(x=Status, y=count))  + geom_point(size=2) + labs(title=paste0(row.names(res[row.names(res)=="AR",])," p=",signif(res$pvalue[row.names(res)=="AR"],4) ," padj=", signif(res$padj[row.names(res)=="AR"],2)), y="Normalised Counts") + peri_theme



```


#### Gene Ontology
```{r}
res_go<- status_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_BSTm_status.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_BSTm_status.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```
## Testosterone BSTm
```{r BSTm continuous data set up}
#Filtering
bstm_behav<- subset(key_behav, Tissue=="BSTm")
bstm_behav<- subset(bstm_behav, Batch!="pilot")
bstm_behav<- droplevels(bstm_behav)
bstm_data<- data[,colnames(data) %in% rownames(bstm_behav)]
start<- nrow(bstm_data)
#remove genes with less than 5 reads
bstm_data$avg_count<- apply(bstm_data, 1, mean)
bstm_data<- bstm_data[bstm_data$avg_count>5,]
bstm_data$avg_count<-NULL


#remove genes where >50% of samples have 0 gene expression
bstm_data$percent_0<- apply(bstm_data, 1, function(x)length(x[x==0]))
thresh<- ncol(bstm_data)/2
bstm_data<- bstm_data[bstm_data$percent_0<=thresh,]
bstm_data$percent_0<-NULL


dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_behav, design= ~ mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),] #remove any genes that didn't converge. 
plotDispEsts(dd)

vsd_data<- getVarianceStabilizedData(dd)
write.csv(counts(dd, normalized=TRUE), "../Candidate_gene_results/data_BSTm_norm.csv", row.names=TRUE, quote=FALSE)



p <- pca(vsd_data, metadata = bstm_behav)
biplot(p, lab=bstm_behav$Harvest_ID, colby="mean_T", shape="Year", legendPosition="right", title="mean T")
#grid.arrange(a,b,c,d,e,f, ncol=2, nrow=3, newpage=TRUE, top=textGrob("PCA plots for each Variable in BSTm", gp=gpar(fontsize=14)))
eigencorplot(p, metavars = c("Year","Status", "mean_T","final_corrected_T","effort.all_study","degree.all_study","strength.all_study"))


wgcnadata<- as.data.frame(t(vsd_data))

gsg = goodSamplesGenes(wgcnadata, verbose = 3)
gsg$allOK


A=adjacency(t(wgcnadata),type="signed")
#-----Calculate whole network connectivity
k=as.numeric(apply(A,2,sum))-1
#-----Standardized connectivity
Z.k=scale(k)
thresholdZ.k=-2.5 
outlierColor=ifelse(Z.k<thresholdZ.k,"red","black")
sampleTree = flashClust(as.dist(1-A), method = "average")
#-----Convert traits to colors

datColors=data.frame(outlier=outlierColor)

#-----Plot the sample dendrogram
plotDendroAndColors(sampleTree,groupLabels=names(datColors),
                    colors=datColors,main="Sample dendrogram and trait heatmap")

```



### mean T BSTm
Formula = ~ mean_T

```{r BSTm mean T, fig.height=4, fig.width=10}
design(dd)<- formula(~mean_T)
dd<- DESeq(dd)

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res, 0.95)
res<- res[order(res$padj),]

summary(res)


mean_T_res<- data.frame(res[order(rownames(res)),])
mean_T_res$gene<- row.names(mean_T_res)


write.csv(mean_T_res, file="../DE_results/results_BSTm_mean_T.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="BSTm mean Testosterone")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()
```


```{r BSTm mean_T Geneplotting, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("mean_T","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("mean_T","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  Rsq<- signif(cor(dat$mean_T, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Mean Testosterone in BSTm",gp=gpar(fontsize=12)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=12)),left=textGrob("Normalized counts", rot = 90, vjust = 1, gp=gpar(fontsize=12)))
g



```

#### Gene Ontology

```{r}
res_go<- mean_T_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot

go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_BSTm_meanT.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)



p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_BSTm_meanT.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```
### mean T x Status


```{r BSTm mean T x Status, echo=FALSE}

dd<- DESeqDataSetFromMatrix(countData=bstm_data, colData=bstm_behav, design= ~ Status + mean_T + Status:mean_T)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]
res<- results(dd, alpha=0.1, name = "Statusterritorial.mean_T")

res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)

interaction_res<- as.data.frame(res)
interaction_res$gene<- rownames(res)
interaction_res$Tissue<- "BSTm"
write.csv(interaction_res, file="../Candidate_gene_results/interactions_BSTm_mean_T_Status.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2, 2), main="BSTm mean Testosterone * Status")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()

```


```{r, fig.height=8, fig.width=14, echo=FALSE}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("Status","mean_T","Year","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("Status","mean_T", "Year","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
  plots[[i]]<-  ggplot(dat, aes(x=mean_T, y=count, color=Class)) + geom_point(size=2) + labs(title=paste0(row.names(res[i,])," padj=", formatC(res$padj[i],2, format="e")),x="", y="Normalised Counts") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status") + facet_wrap(~ Status)
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Status * Mean T in BSTm",gp=gpar(fontsize=14)), bottom=textGrob("Mean Testosterone",gp=gpar(fontsize=14)))
g
```


## Strength BSTm

Formula ~ scale(Strength)

```{r BSTm Strength, fig.height=4, fig.width=10}
dd$strength.all_study<- scale(dd$strength.all_study)
design(dd)<- formula(~strength.all_study)
dd<- DESeq(dd)
dd<- dd[which(mcols(dd)$betaConv),]

res<- results(dd, alpha=0.1)
res<- rm_continuous_outliers(dd, res)
res<- res[order(res$padj),]
summary(res)


strength_res<- data.frame(res[order(rownames(res)),])
strength_res$gene<- row.names(strength_res)

write.csv(strength_res, file="../DE_results/results_BSTm_Strength.csv", row.names=FALSE)


par(mfrow=c(1,2))
DESeq2::plotMA(res, ylim = c(-2,2),main="BSTm Social Network Strength")
hist(res$pvalue, breaks=20, col="grey", main="", xlab="uncorrected P-value")
dev.off()


```



```{r BSTm Strength geneplots, fig.height=8, fig.width=14}
gone<- plotCounts(dd, gene=row.names(res[1,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gtwo<- plotCounts(dd, gene=row.names(res[2,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gthree<-  plotCounts(dd, gene=row.names(res[3,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfour<-  plotCounts(dd, gene=row.names(res[4,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gfive<-  plotCounts(dd, gene=row.names(res[5,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)
gsix<-  plotCounts(dd, gene=row.names(res[6,]), intgroup=c("strength.all_study","Class"), returnData=TRUE)

vars<- list(gone,gtwo,gthree,gfour,gfive,gsix)

plots<- list()

for(i in 1:6){
  dat<- vars[[i]]
    Rsq<- signif(cor(dat$strength.all_study, dat$count)^2,2)
  lfc<- signif(res$log2FoldChange[i],2)
  padj<-formatC(res$padj[i],1, format="e")
  plots[[i]]<-  ggplot(dat, aes(x=strength.all_study, y=count, color=Class)) + geom_point(size=2) + labs(title=row.names(res[i,]), subtitle=substitute(paste("LFC=",a," padj=", b, " R"^{2},"=",c),list(a=lfc, b=padj, c=Rsq))
,x="", y="") + peri_theme + scale_colour_manual(values=status_cols, name="Age x Status")
}


g<- ggarrange(plotlist=plots,ncol=3, nrow=2, common.legend = TRUE)
g<- annotate_figure(g, top=textGrob("Top six DEGs by Social Network Strength in BSTm",gp=gpar(fontsize=14)), bottom=textGrob("BSTm",gp=gpar(fontsize=14)))
g

```
```{r,  eval=FALSE}
hub<-"PGR"
dat<- plotCounts(dd, gene=hub, intgroup=c("Class","Status", "strength.all_study"), returnData=TRUE)
Rsq<- cor(dat$strength.all_study, dat$count)^2
ggplot(dat, aes(x=strength.all_study, y=count)) + geom_point(size=4, pch=19) + peri_theme + scale_colour_manual(values=status_cols[2:3]) + labs(x="strength", y="Normalized Counts", title=paste0(hub,"\n LFC=", signif(res$log2FoldChange[rownames(res)==hub],2), ", p=",signif(res$pvalue[rownames(res)==hub],2), ", padj=", signif(res$padj[rownames(res)==hub],2), " R^2=", signif(Rsq,2))) + theme(legend.position = "none") + geom_smooth(method="lm", color="black") + geom_point(aes(color=Status), pch=19, size=4)

```

#### Gene Ontology

```{r}
res_go<- strength_res
res_go$logP<- -log(res_go$pvalue, 10)
res_go$logP<- ifelse(res_go$log2FoldChange<0, -(res_go$logP), res_go$logP)
res_go$gene<- rownames(res_go)
res_go<- as.data.frame(res_go)
res_go<- merge(res_go, go_key, by="gene")
res_go<- res_go[order(res_go$logP, decreasing=TRUE),]
res_go<- res_go[!is.na(res_go$logP),]
res_go$pvalue<- ifelse(is.na(res_go$padj), NA, as.numeric(res_go$pvalue))

geneList<- res_go$logP
names(geneList)<- res_go$gene


y<- GSEA(geneList, TERM2GENE = go2gene_bp, TERM2NAME = go2name_bp)
gseaplot<- ridgeplot(y) + peri_theme + labs(title="Top GO BP terms from GSEA") + scale_fill_viridis()
gseaplot
go_mean_T<- y@result
write.csv(y@result, file="../DE_results/GSEA_BP_BSTm_strength.csv", row.names=FALSE)
ego2<- as.data.frame(go_mean_T$ID)
colnames(ego2)<- "ID"
ego2$p.adjust<- go_mean_T$p.adjust

simMatrix <- calculateSimMatrix(ego2$ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")
scores <- setNames(-log10(ego2$p.adjust), ego2$ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Hs.eg.db")
treemapPlot(reducedTerms)


p<- scatterPlot(simMatrix, reducedTerms) + theme_bw() +  labs(title="Semantic similarity of GO terms") + theme(plot.title = element_text(hjust = 0.5, size=12))
```


```{r, eval=FALSE}
g2<- ggarrange(g,gseaplot,p, ncol=1, nrow=3, labels=c("A","B","C"))


g2
ggsave(filename="supp_figure_BSTm_strength.png", plot=g2, device="png" ,height=12, width=8.5, units="in")
```



Volcano plots and similarity of gene expression is in another Rmarkdown file on this github: volcanoes_similarity.Rmd

```{r}
sessionInfo()
```


