<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive Change October 2020" />


<title>Gene Expression QC</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_DESeq2.html">Differential Expression</a>
    </li>
    <li>
      <a href="./03_WGCNA_wholebrain.html">Whole-brain co-expression</a>
    </li>
    <li>
      <a href="./04_WGCNA_single_tissue.html">Co-expression in each tissue</a>
    </li>
    <li>
      <a href="./05_candidate_gene_analysis.html">Candidate Genes</a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene Expression QC</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive Change October 2020</h4>
<h4 class="date">Last Knit “2022-05-24”</h4>

</div>


<div id="sampling" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Sampling</h1>
<p>There are a total of 186 sequenced samples, across 16 individuals and 9-12 Tissues per individual. Each individual’s tissue has a unique library prep. Four tissues from the four individuals in the 2015 pilot study were repeated with fresh library preps and run alongside the new samples. Individuals/libraries were randomized across Batchs.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of individual libraries across sequencing Batchs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
PFT1
</th>
<th style="text-align:right;">
PFT10
</th>
<th style="text-align:right;">
PFT11
</th>
<th style="text-align:right;">
PFT12
</th>
<th style="text-align:right;">
PFT2
</th>
<th style="text-align:right;">
PFT3
</th>
<th style="text-align:right;">
PFT4
</th>
<th style="text-align:right;">
PFT5
</th>
<th style="text-align:right;">
PFT6
</th>
<th style="text-align:right;">
PFT7
</th>
<th style="text-align:right;">
PFT8
</th>
<th style="text-align:right;">
PFT9
</th>
<th style="text-align:right;">
TF1
</th>
<th style="text-align:right;">
TF2
</th>
<th style="text-align:right;">
TT1
</th>
<th style="text-align:right;">
TT2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of libraries in each tissue across sequencing Batchs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
AH
</th>
<th style="text-align:right;">
AI
</th>
<th style="text-align:right;">
BSTm
</th>
<th style="text-align:right;">
GCT
</th>
<th style="text-align:right;">
GON
</th>
<th style="text-align:right;">
ICO
</th>
<th style="text-align:right;">
LS
</th>
<th style="text-align:right;">
PIT
</th>
<th style="text-align:right;">
POM
</th>
<th style="text-align:right;">
PVN
</th>
<th style="text-align:right;">
TNA
</th>
<th style="text-align:right;">
VMH
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of libraries for each social status across sequencing Batchs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
floater
</th>
<th style="text-align:right;">
territorial
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
48
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
37
</td>
<td style="text-align:right;">
41
</td>
</tr>
</tbody>
</table>
<p>The plot below is just a quick check to see if any of the batches/samples stick out in terms of read depth. Gross differences in the number of reads in each batch might indicate a problem with comparing between them. <img src="01_reading_QC_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  files_key$Number.of.Reads[files_key$Batch == &quot;run1&quot;] and files_key$Number.of.Reads[files_key$Batch == &quot;run2&quot;]
## W = 3976, p-value = 0.0009372
## alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="mapping-results" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Mapping Results</h1>
<p>Briefly, I used STAR to map the reads to the <em>Pipra filicauda</em> Refseq Genome V1 GCF_003945595.1. See the mapping script on the github <a href="https://github.com/periperipatus/PIFI_brain_transcriptome/tree/analysis_code">repository</a></p>
<p>The figure below shows the results of the summary statistics for each library</p>
<p><img src="01_reading_QC_files/figure-html/read%20data-1.png" width="672" /></p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>The above plots identify 3 samples that mapped poorly, and these samples also had low RNA concentrations. Samples PFT3_AI_run2, PFT6_AH_run2, PFT9_ICO_run2 will be excluded from further analyses. Then genes with an average count of greater than or equal to 5 will be excluded from downstream analysis.</p>
</div>
<div id="exploration-of-the-expression-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Exploration of the expression data</h1>
<p>These data have been filtered a little bit already to exclude individuals with low read mapping, and genes with a mean read count of &lt;5.</p>
<p>Here I will document gene read depth variation across tissues.</p>
<p><img src="01_reading_QC_files/figure-html/Tissue%20read%20variation-1.png" width="1440" /></p>
<div id="batch-differences-and-outlier-detection" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Batch differences and outlier detection</h2>
<p>The primary purpose of these analyses is to determine:</p>
<ul>
<li><p>Do Year and Batch influence need to be accounted for?</p></li>
<li><p>Whether there is a significant library effect - as per the replicated samples.</p></li>
<li><p>Whether there are any significantly outlying samples.</p></li>
</ul>
<div id="all-samples-pca-and-wgcna" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> All Samples PCA and WGCNA</h3>
<p>Here I will use PCA outliers and WGCNA outliers to guide initial filtering.</p>
<p><img src="01_reading_QC_files/figure-html/pca-1.png" width="672" /></p>
<pre><code>## [1] &quot;Batch&quot;
## [1] &quot;Harvest_ID&quot;
## [1] &quot;Tissue&quot;
## [1] &quot;Year&quot;</code></pre>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>There appears to be some variation explained by Year and Batch variables.</p>
<pre><code>##  [1] &quot;Batch&quot;  &quot;Tissue&quot; &quot;Status&quot; &quot;Year&quot;   &quot;Run1&quot;   &quot;Run2&quot;   &quot;AH&quot;     &quot;AI&quot;    
##  [9] &quot;BSTm&quot;   &quot;GCT&quot;    &quot;ICO&quot;    &quot;LS&quot;     &quot;PIT&quot;    &quot;POM&quot;    &quot;PVN&quot;    &quot;TNA&quot;   
## [17] &quot;VMH&quot;    &quot;yr2017&quot; &quot;yr2018&quot;</code></pre>
<p><img src="01_reading_QC_files/figure-html/wgcna%20qc-1.png" width="480" /></p>
</div>
<div id="year-and-batch-effects" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Year and Batch Effects</h3>
<p>The DESeq design for the full dataset here is ~ Year + Batch + ind.n + Tissue. This includes the replicate samples separated out.</p>
<div id="batch-effects" class="section level4" number="3.1.2.1">
<h4><span class="header-section-number">3.1.2.1</span> Batch Effects</h4>
<p>There are 4577 genes with an adjusted p-val of &lt;0.05, and therefore differentially expressed with respect to sequencing run1 and run2 in 2019. Approximately equal numbers of genes were differentially expressed up or down.</p>
<p><img src="01_reading_QC_files/figure-html/Batch%20effects%20figs-1.png" width="768" /></p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-4-1.png" width="768" /></p>
</div>
<div id="year-effects" class="section level4" number="3.1.2.2">
<h4><span class="header-section-number">3.1.2.2</span> Year effects</h4>
<p><img src="01_reading_QC_files/figure-html/Year%20Effects%20figs-1.png" width="768" /></p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-5-1.png" width="768" /></p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-6-1.png" width="672" /><img src="01_reading_QC_files/figure-html/unnamed-chunk-6-2.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.509], polygon[GRID.polygon.510], polygon[GRID.polygon.511], polygon[GRID.polygon.512], text[GRID.text.513], text[GRID.text.514], text[GRID.text.515], text[GRID.text.516], text[GRID.text.517])</code></pre>
<div id="gonads-only" class="section level5" number="3.1.2.2.1">
<h5><span class="header-section-number">3.1.2.2.1</span> Gonads only</h5>
<p>Let’s look for Batch and year effects just in run 1 vs run 2, and only in the a given tissue. I’m just going to use the testes (GON) as a test.</p>
<p><img src="01_reading_QC_files/figure-html/Batch%20and%20year%20effects%20effects%20in%20gonads-1.png" width="672" /></p>
<pre><code>## [1] &quot;Batch&quot;
## [1] &quot;Year&quot;</code></pre>
<p><img src="01_reading_QC_files/figure-html/Batch%20and%20year%20effects%20effects%20in%20gonads-2.png" width="672" /></p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-7-1.png" width="768" /></p>
<p>Let’s have a look at the plots for the most significant gene.</p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>And now let’s explore Year in the Gonads.</p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-9-1.png" width="768" /></p>
<p>An example top gene <img src="01_reading_QC_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>And within the Gonads, which of the DEGs associated with Year and Batch are shared?</p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.863], polygon[GRID.polygon.864], polygon[GRID.polygon.865], polygon[GRID.polygon.866], text[GRID.text.867], text[GRID.text.868], text[GRID.text.869], text[GRID.text.870], text[GRID.text.871])</code></pre>
<p>There is colinearity between Batch and Year in both the full dataset and the single tissue test dataset. All subsequent models will include a composite variable for Year and Batch (Year_Batch) to account for these effects in subsequent analyses.</p>
</div>
<div id="random-variable" class="section level5" number="3.1.2.2.2">
<h5><span class="header-section-number">3.1.2.2.2</span> Random Variable</h5>
<p>Create a random variable that has the same proportions as Batch and ensure that the p-value distribution does not have an over-abundance of values &lt;0.05, which suggests that the samples were drawn from different distributions. This is essentially a test to ensure that what we are seeing with respect to Batch (and Year) above isn’t just random noise.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create a random variable and test against that: </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1283</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span> <span class="fu">sample</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="dv">183</span>, <span class="at">replace=</span><span class="cn">TRUE</span>, <span class="at">prob=</span><span class="fu">c</span>(<span class="fl">0.43</span>,<span class="fl">0.41</span>,<span class="fl">0.16</span>))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>dd<span class="sc">$</span>random_variable<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(x)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span>random_variable)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>testres<span class="ot">&lt;-</span> <span class="fu">results</span>(dd)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(testres, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(testres<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;p-value distribution from DE on random variable&quot;</span>)</span></code></pre></div>
<p><img src="01_reading_QC_files/figure-html/random%20variable-1.png" width="768" /></p>
<p>These results suggest that these few differentially expressed genes are drawn from the same distribution, so the results we saw previously are probably real.</p>
</div>
</div>
</div>
</div>
<div id="replicate-samples" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Replicate Samples</h2>
<p>Now, let’s see how different the samples from 2015 are from their replicates on the latest run. These were sequencing replicates from the 2015 libraries.</p>
<p><img src="01_reading_QC_files/figure-html/replicate%20samples-1.png" width="768" /></p>
<pre><code>## null device 
##           1</code></pre>
<p>Now let’s use ordination (1) and sample distance (2) to see how different they are from eachother.</p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>Let’s see how the expression differs between replicates for the top 20 most expressed genes. These show that expression is always in the same direction, and usually very close in value. <img src="01_reading_QC_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Now, take sample-sample distance and cluster. These show that replicates cluster together, and then tissues.</p>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>There are 64 differentially expressed genes between replicates. We appear to have broad batch effects (with respect to Flow Cell (Batch) and with respect to Year sampled), but also there is not really a strong replicate effect as shown.</p>
<p>In order to maximise sample size, and to maximise the number of individuals in the pilot batch group, I will take the first (pilot) instance of each replicate to include in further analyses. In addition, all subsequent analyses will include include the batch variables where necessary.</p>
</div>
</div>
<div id="finalizing-dataset" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Finalizing dataset</h1>
<p>Finally, using the replicate removed dataset we will do final PCA explorations of brain and pituitary data. Note the Gonads are above with the replicate samples.</p>
<div id="wgcna-outlier-dendrograms" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> WGCNA outlier dendrograms</h2>
<pre><code>## png 
##   2</code></pre>
</div>
<div id="brain" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Brain</h2>
<p>This PCA figure is annotated with potential outliers based on the above per-tissue WGCNA, and a draft version of the below PCA plot. Outliers were selected in the WGCNA plot based on branch length (sample distance) from main group and the number of samples. Similar logic applied to the PCA.</p>
<p><img src="01_reading_QC_files/figure-html/brain-1.png" width="672" /></p>
<pre><code>## [1] &quot;Tissue&quot;
## [1] &quot;Year_Batch&quot;</code></pre>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Still seeing Year and Batch effects in the whole brain dataset.</p>
</div>
<div id="pituitary" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Pituitary</h2>
<p><img src="01_reading_QC_files/figure-html/pituitary-1.png" width="672" /></p>
</div>
<div id="confound-between-tissue-and-batch" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Confound between tissue and batch</h2>
<p>On discussions with Chris, it is possible the number of genes we are seeing as differentially expressed in the main analysis is owing to a confound between tissue and batch, as a few tissues are sequenced exclusively on a single batch. To explore this, we can see whether the DEGs for batch are the same as those for tissue differences. Tissue will be AH, which only has run 2 and PIT, which was compared at Run 1 vs Run 2</p>
<table>
<caption>
Batches and Tissues
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
AH
</th>
<th style="text-align:right;">
PIT
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><img src="01_reading_QC_files/figure-html/batch%20effects%20contd-1.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.1475], polygon[GRID.polygon.1476], polygon[GRID.polygon.1477], polygon[GRID.polygon.1478], text[GRID.text.1479], text[GRID.text.1480], text[GRID.text.1481], text[GRID.text.1482], text[GRID.text.1483])</code></pre>
<p><img src="01_reading_QC_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.1484], polygon[GRID.polygon.1485], polygon[GRID.polygon.1486], polygon[GRID.polygon.1487], polygon[GRID.polygon.1488], polygon[GRID.polygon.1489], text[GRID.text.1490], text[GRID.text.1491], text[GRID.text.1492], text[GRID.text.1493], text[GRID.text.1494], text[GRID.text.1495], text[GRID.text.1496], text[GRID.text.1497], text[GRID.text.1498], text[GRID.text.1499])</code></pre>
<p>However, when you run PIT on its own with respect to batch there are 2441 differentially expressed genes. Seems like tissues need to be kept separate to remove this confound.</p>
<p>The next analysis I conduct will be the differential expression analyses in DESeq. See menus for these and more analyses.</p>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19044)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
## [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
## [5] LC_TIME=English_Australia.1252    
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] wesanderson_0.3.6           VennDiagram_1.6.20         
##  [3] futile.logger_1.4.3         kableExtra_1.3.1           
##  [5] WGCNA_1.69                  fastcluster_1.1.25         
##  [7] dynamicTreeCut_1.63-1       pheatmap_1.0.12            
##  [9] lubridate_1.8.0             DESeq2_1.28.1              
## [11] SummarizedExperiment_1.18.1 DelayedArray_0.14.0        
## [13] matrixStats_0.56.0          Biobase_2.48.0             
## [15] GenomicRanges_1.40.0        GenomeInfoDb_1.24.0        
## [17] IRanges_2.22.2              S4Vectors_0.26.1           
## [19] BiocGenerics_0.34.0         stringr_1.4.0              
## [21] PCAtools_2.0.0              cowplot_1.0.0              
## [23] lattice_0.20-41             ggrepel_0.8.2              
## [25] ggplot2_3.3.5               reshape2_1.4.4             
## [27] plyr_1.8.6                 
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1          ellipsis_0.3.2           
##   [3] htmlTable_2.0.1           XVector_0.28.0           
##   [5] base64enc_0.1-3           rstudioapi_0.13          
##   [7] farver_2.0.3              bit64_0.9-7              
##   [9] AnnotationDbi_1.50.0      fansi_0.4.1              
##  [11] xml2_1.3.2                codetools_0.2-16         
##  [13] splines_4.0.2             doParallel_1.0.15        
##  [15] impute_1.62.0             geneplotter_1.66.0       
##  [17] knitr_1.29                Formula_1.2-3            
##  [19] annotate_1.66.0           cluster_2.1.0            
##  [21] GO.db_3.11.4              png_0.1-7                
##  [23] compiler_4.0.2            httr_1.4.2               
##  [25] dqrng_0.2.1               backports_1.1.7          
##  [27] assertthat_0.2.1          Matrix_1.2-18            
##  [29] fastmap_1.1.0             limma_3.44.3             
##  [31] formatR_1.8               BiocSingular_1.4.0       
##  [33] acepack_1.4.1             htmltools_0.5.2          
##  [35] tools_4.0.2               rsvd_1.0.3               
##  [37] gtable_0.3.0              glue_1.5.1               
##  [39] GenomeInfoDbData_1.2.3    dplyr_1.0.7              
##  [41] Rcpp_1.0.7                jquerylib_0.1.4          
##  [43] vctrs_0.3.8               preprocessCore_1.50.0    
##  [45] iterators_1.0.12          DelayedMatrixStats_1.10.0
##  [47] xfun_0.29                 rvest_1.0.2              
##  [49] lifecycle_1.0.1           irlba_2.3.3              
##  [51] XML_3.99-0.5              zlibbioc_1.34.0          
##  [53] scales_1.1.1              ragg_1.2.1               
##  [55] lambda.r_1.2.4            RColorBrewer_1.1-2       
##  [57] yaml_2.2.1                memoise_1.1.0            
##  [59] gridExtra_2.3             rpart_4.1-15             
##  [61] latticeExtra_0.6-29       stringi_1.4.6            
##  [63] RSQLite_2.2.0             highr_0.8                
##  [65] genefilter_1.70.0         foreach_1.5.0            
##  [67] checkmate_2.0.0           BiocParallel_1.22.0      
##  [69] systemfonts_1.0.3         rlang_0.4.12             
##  [71] pkgconfig_2.0.3           bitops_1.0-6             
##  [73] evaluate_0.14             purrr_0.3.4              
##  [75] labeling_0.3              htmlwidgets_1.5.3        
##  [77] bit_4.0.4                 tidyselect_1.1.0         
##  [79] magrittr_2.0.1            R6_2.4.1                 
##  [81] generics_0.0.2            Hmisc_4.4-0              
##  [83] DBI_1.1.0                 pillar_1.6.4             
##  [85] foreign_0.8-80            withr_2.4.3              
##  [87] survival_3.2-3            RCurl_1.98-1.2           
##  [89] nnet_7.3-14               tibble_3.1.6             
##  [91] crayon_1.4.2              futile.options_1.0.1     
##  [93] utf8_1.1.4                rmarkdown_2.11           
##  [95] jpeg_0.1-8.1              locfit_1.5-9.4           
##  [97] data.table_1.13.0         blob_1.2.1               
##  [99] webshot_0.5.2             digest_0.6.25            
## [101] xtable_1.8-4              textshaping_0.3.6        
## [103] munsell_0.5.0             viridisLite_0.4.0</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4,h5",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
