<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive change December 2023" />


<title>Checks and PCA</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Supp 1: Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_1_Checks_PCA.html">Supp 1 &amp; 2: Checks and System-wide PCA</a>
    </li>
    <li>
      <a href="./02_2_WGCNA_wholebrain.html">Supp 2: Whole-brain WGCNA</a>
    </li>
    <li>
      <a href="./02_3_WGCNA_HPG.html">Supp 2: HPG WGCNA</a>
    </li>
    <li>
      <a href="./03_DESeq2.html">Supp 3: Per Tissue Differential Expression</a>
    </li>
    <li>
      <a href="./04_candidate_gene_analysis.html">Supp 4: Candidate Gene Analysis</a>
    </li>
    <li>
      <a href="./05_WGCNA_single_tissue.html">Supp 5: Within Tissue WGCNA </a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Supp 6: Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Checks and PCA</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male
cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive change December 2023</h4>
<h4 class="date">Last Knit “2024-01-05”</h4>

</div>


<p>Here, I take the filtered dataset from the QC process, and run a
whole brain analysis, as well as individual tissue analysis in the next
markdown. Here, I also explore the relationships between our explanatory
variables.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_key_Parsed_ReplicatesRemoved.csv&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>behav<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_unfiltered/PIPFIL_T_and_behav_data.csv&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">rownames</span>(key)<span class="ot">&lt;-</span> key<span class="sc">$</span>X</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>key<span class="sc">$</span>Color_ID<span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;/&quot;</span>,<span class="st">&quot;&quot;</span>, key<span class="sc">$</span>Color_ID)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;Color_ID&quot;</span><span class="ot">=</span><span class="st">&quot;colorID&quot;</span>))</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>behav<span class="sc">$</span>status<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(behav<span class="sc">$</span>status, <span class="fu">c</span>(<span class="st">&quot;floa&quot;</span><span class="ot">=</span><span class="st">&quot;floater&quot;</span>, <span class="st">&quot;terr&quot;</span><span class="ot">=</span><span class="st">&quot;territorial&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">merge</span>(key, behav, <span class="at">by=</span><span class="st">&quot;colorID&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">is.na</span>(key_behav<span class="sc">$</span>last_behav),]</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="co">#create a data.frame with all of the observations. </span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> key[<span class="sc">!</span>key<span class="sc">$</span>X <span class="sc">%in%</span> key_behav<span class="sc">$</span>X,]</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>cols_not_key<span class="ot">&lt;-</span> <span class="fu">colnames</span>(key_behav)[<span class="sc">!</span><span class="fu">colnames</span>(key_behav) <span class="sc">%in%</span> <span class="fu">colnames</span>(key)]</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>cols_not_key_df<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(not_in_behav), <span class="at">ncol =</span> <span class="fu">length</span>(cols_not_key)))</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="fu">colnames</span>(cols_not_key_df)<span class="ot">&lt;-</span> cols_not_key</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> <span class="fu">cbind</span>(not_in_behav, cols_not_key_df)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">rbind</span>(key_behav, not_in_behav)</span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>X</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>key_behav<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Tissue, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;GON&quot;</span>,<span class="st">&quot;PIT&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;AI&quot;</span>,<span class="st">&quot;TNA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(<span class="fu">rownames</span>(key_behav)),]</span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Class)</span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">revalue</span>(key_behav<span class="sc">$</span>Class, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;SCB floater&quot;</span><span class="ot">=</span><span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;DCB floater &quot;</span><span class="ot">=</span><span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;DCB territorial&quot;</span><span class="ot">=</span><span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Class, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>sampleID</span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(key_behav<span class="sc">$</span>sampleID),]</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a>key_behav<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Year)</span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a>key_behav<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Batch)</span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a>key_behav<span class="sc">$</span>Status<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Status)</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>key_behav_unique<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">duplicated</span>(key_behav<span class="sc">$</span>colorID),]</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a><span class="co">#read the raw count data</span></span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a>data<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv&quot;</span>)</span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a><span class="fu">rownames</span>(data)<span class="ot">&lt;-</span> data<span class="sc">$</span>X</span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a>data<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a><span class="do">### Gene Ontology</span></span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a>go_key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/Maggies_annotations_modifiedR.csv&quot;</span>)</span>
<span id="cb1-48"><a href="#cb1-48" tabindex="-1"></a>go_key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-49"><a href="#cb1-49" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/pfil_GO_key_raw.csv&quot;</span>)</span>
<span id="cb1-51"><a href="#cb1-51" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_terms, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-52"><a href="#cb1-52" tabindex="-1"></a>go2gene_bp<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;P&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb1-53"><a href="#cb1-53" tabindex="-1"></a>go2gene_mf<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;F&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb1-54"><a href="#cb1-54" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/ontology_obo_out.csv&quot;</span>)</span>
<span id="cb1-55"><a href="#cb1-55" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_obo, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;id&quot;</span><span class="ot">=</span><span class="st">&quot;GO_ID&quot;</span>))</span>
<span id="cb1-56"><a href="#cb1-56" tabindex="-1"></a>go2name_bp<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;biological_process&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span>
<span id="cb1-57"><a href="#cb1-57" tabindex="-1"></a>go2name_mf<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;molecular_function&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span>
<span id="cb1-58"><a href="#cb1-58" tabindex="-1"></a></span>
<span id="cb1-59"><a href="#cb1-59" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/Maggies_annotations_modifiedR.csv&quot;</span>)</span>
<span id="cb1-61"><a href="#cb1-61" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(genes_key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-62"><a href="#cb1-62" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113993669&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CYP19A1&quot;</span></span>
<span id="cb1-63"><a href="#cb1-63" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113983511&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;OXT&quot;</span></span>
<span id="cb1-64"><a href="#cb1-64" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113983498&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVP&quot;</span></span>
<span id="cb1-65"><a href="#cb1-65" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113982601&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVPR2&quot;</span></span>
<span id="cb1-66"><a href="#cb1-66" tabindex="-1"></a>data_genes<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span><span class="fu">rownames</span>(data))</span>
<span id="cb1-67"><a href="#cb1-67" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">merge</span>(data_genes, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-68"><a href="#cb1-68" tabindex="-1"></a><span class="co">#genes_key&lt;- genes_key[which(grepl(&quot;LOC[0-9]+&quot;,genes_key$gene)),]</span></span>
<span id="cb1-69"><a href="#cb1-69" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" tabindex="-1"></a>genes_key<span class="sc">$</span>display_gene_ID<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;LOC[0-9]+&quot;</span>, genes_key<span class="sc">$</span>gene) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;LOC[0-9]+&quot;</span>,genes_key<span class="sc">$</span>best_anno) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(genes_key<span class="sc">$</span>best_anno), <span class="fu">paste0</span>(genes_key<span class="sc">$</span>gene,<span class="st">&quot; (&quot;</span>,genes_key<span class="sc">$</span>best_anno,<span class="st">&quot;)&quot;</span>), <span class="fu">as.character</span>(genes_key<span class="sc">$</span>gene))</span>
<span id="cb1-71"><a href="#cb1-71" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> genes_key[,<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;best_anno&quot;</span>,<span class="st">&quot;display_gene_ID&quot;</span>)]</span></code></pre></div>
<p><br><br></p>
<div id="exploration-of-the-behavioural-data" class="section level1"
number="1">
<h1><span class="header-section-number">1</span> Exploration of the
behavioural data</h1>
<p>We were concerned about the signal from the last few days outweighing
the gene-expression signal from their overall phenotype, so we will
explore how different these measures are from eachother.</p>
<p><img src="02_1_Checks_PCA_files/figure-html/variable%20correlations-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="associations-between-variables" class="section level2"
number="1.1">
<h2><span class="header-section-number">1.1</span> Associations between
variables</h2>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Given the significant relationship between degree and strength, I
will only keep one of these variables - the strength. This was the most
repeatable of the two sociality variables in Ryder et al. (2020).</p>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="02_1_Checks_PCA_files/figure-html/cor%20status-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>All significance tests cited in the figure above are based on just
territorial and floater, but colour-coded according to
plumage*status.</p>
<p><img src="02_1_Checks_PCA_files/figure-html/mean%20T%20and%20behav-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="nuisance-variables" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Nuisance
variables</h2>
<p>In the previous QC exploration I found substantial DEGs according to
collection Year and Sequencing Batch. Here we explore how the continuous
variables are distributed across sampling years. The tests find there is
a difference in the medians between years for both our interest
traits.</p>
<p><img src="02_1_Checks_PCA_files/figure-html/variables%20according%20to%20Year-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
<div id="all-tissues-pca" class="section level1" number="2">
<h1><span class="header-section-number">2</span> All tissues PCA</h1>
<p>Plotting all the tissues</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> data</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(all_data)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#remove genes with avg read counts &lt;20</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>all_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(all_data, <span class="dv">1</span>, mean)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> all_data[all_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">20</span>,]</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>all_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>all_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(all_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(all_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> all_data[all_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>all_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="fu">nrow</span>(all_data)</span></code></pre></div>
<pre><code>FALSE [1] 14414</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co">#14414 genes.</span></span></code></pre></div>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Based on these results it seems like PC7 is associated with social
Status. Let’s explore this PC further.</p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co">#library(nlme)</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>test<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC7=</span>p<span class="sc">$</span>rotated<span class="sc">$</span>PC7)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>test<span class="ot">&lt;-</span> <span class="fu">cbind</span>(test, key_behav)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>m0<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7 <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data=</span>test)</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>m01<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7 <span class="sc">~</span> Year, <span class="at">data=</span>test)</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>m01.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m01, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>m02<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7 <span class="sc">~</span> Year <span class="sc">+</span> Tissue, <span class="at">data=</span>test)</span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>m02.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m02, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>m02.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC7
FALSE           Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year      2216.1   2  28.775 2.255e-11 ***
FALSE Tissue    5887.0  11  13.899 &lt; 2.2e-16 ***
FALSE Residuals 6045.6 157                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>m02.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m02)[,<span class="dv">1</span>])</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>m1<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7<span class="sc">~</span> Year <span class="sc">+</span> Tissue <span class="sc">+</span> Status,<span class="at">data=</span>test)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>m1.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m1, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>m1.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC7
FALSE           Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year      2103.1   2 28.3853 3.066e-11 ***
FALSE Tissue    5851.9  11 14.3607 &lt; 2.2e-16 ***
FALSE Status     266.5   1  7.1946    0.0081 ** 
FALSE Residuals 5779.0 156                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>m1.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m1)[,<span class="dv">1</span>])</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>m2<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7<span class="sc">~</span> Year <span class="sc">+</span> Tissue <span class="sc">*</span> Status,<span class="at">data=</span>test,<span class="at">contrasts=</span><span class="fu">list</span>(<span class="at">Tissue=</span>contr.sum, <span class="at">Status=</span>contr.sum))</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>m2.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m2, <span class="at">type=</span><span class="dv">3</span>)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>m2.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: PC7
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE (Intercept)   1778.0   1 47.8645 1.358e-10 ***
FALSE Year          2030.9   2 27.3355 8.440e-11 ***
FALSE Tissue        5792.6  11 14.1759 &lt; 2.2e-16 ***
FALSE Status         294.5   1  7.9272  0.005548 ** 
FALSE Tissue:Status  392.6  11  0.9609  0.484768    
FALSE Residuals     5386.4 145                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>m2.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m2)[,<span class="dv">1</span>])</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="fu">Anova</span>(m2, <span class="at">type=</span><span class="dv">2</span>) <span class="co"># this does not substantially change the interpretation</span></span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC7
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year          2030.9   2 27.3355  8.44e-11 ***
FALSE Tissue        5851.9  11 14.3211 &lt; 2.2e-16 ***
FALSE Status         266.5   1  7.1748  0.008247 ** 
FALSE Tissue:Status  392.6  11  0.9609  0.484768    
FALSE Residuals     5386.4 145                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>m3<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7<span class="sc">~</span> Tissue <span class="sc">+</span> Status <span class="sc">+</span> Tissue<span class="sc">:</span>Status,<span class="at">data=</span>test,<span class="at">contrasts=</span><span class="fu">list</span>(<span class="at">Tissue=</span>contr.sum, <span class="at">Status=</span>contr.sum))</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>m3.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m3, <span class="at">type=</span><span class="dv">3</span>)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>m3.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: PC7
FALSE               Sum Sq  Df F value   Pr(&gt;F)    
FALSE (Intercept)     14.4   1  0.2858 0.593714    
FALSE Tissue        5361.8  11  9.6603 5.32e-13 ***
FALSE Status         404.1   1  8.0096 0.005304 ** 
FALSE Tissue:Status  464.8  11  0.8375 0.602960    
FALSE Residuals     7417.3 147                     
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>m3.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m3)[,<span class="dv">1</span>])</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="fu">Anova</span>(m3, <span class="at">type=</span><span class="dv">2</span>) <span class="co"># this does not substantially change the interpretation</span></span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC7
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue        5421.3  11  9.7676 3.887e-13 ***
FALSE Status         379.5   1  7.5214  0.006855 ** 
FALSE Tissue:Status  464.8  11  0.8375  0.602960    
FALSE Residuals     7417.3 147                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>m4<span class="ot">&lt;-</span> <span class="fu">lm</span>(PC7<span class="sc">~</span> Tissue <span class="sc">+</span> Status,<span class="at">data=</span>test)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>m4.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m4, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>m4.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m4)[,<span class="dv">1</span>])</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>models<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;~ 1&quot;</span>, <span class="st">&quot;~ Year&quot;</span>, <span class="st">&quot;~ Year + Tissue&quot;</span>,<span class="st">&quot;~ Year + Tissue + Status&quot;</span> , <span class="st">&quot;~ Year + Tissue + Status + Tissue:Status&quot;</span>,<span class="st">&quot;~ Tissue + Status + Tissue:Status&quot;</span>, <span class="st">&quot;~ Tissue + Status&quot;</span>)</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>aic<span class="ot">=</span><span class="fu">signif</span>(<span class="fu">c</span>(<span class="fu">AIC</span>(m0),<span class="fu">AIC</span>(m01),<span class="fu">AIC</span>(m02),<span class="fu">AIC</span>(m1),<span class="fu">AIC</span>(m2),<span class="fu">AIC</span>(m3),<span class="fu">AIC</span>(m4)),<span class="dv">4</span>)</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>deltaAIC<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>,aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">2</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">3</span>], aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">4</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">5</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">6</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">7</span>]),<span class="dv">3</span>) </span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>batch_p<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>, m01.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m02.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m1.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m2.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>],<span class="cn">NA</span>,<span class="cn">NA</span>),<span class="dv">2</span>)</span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>e_p<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>, <span class="cn">NA</span>,<span class="cn">NA</span>,m1.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">3</span>],m2.aov<span class="sc">$</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">3</span>],m3.aov<span class="sc">$</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>],m4.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>]),<span class="dv">2</span>) </span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>sumVIF<span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">c</span>(<span class="cn">NA</span>,<span class="cn">NA</span>,m02.vif, m1.vif,m2.vif, m3.vif, m4.vif),<span class="dv">0</span>)</span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a>ms<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(models,aic, deltaAIC,batch_p,e_p, sumVIF)</span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a>ms<span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(ms,<span class="at">theme =</span> <span class="fu">ttheme</span>(<span class="at">base_size =</span> <span class="dv">6</span>,<span class="at">padding=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>),<span class="st">&quot;pt&quot;</span>)),<span class="at">rows=</span><span class="cn">NULL</span>)</span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a></span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a>pc7test<span class="ot">&lt;-</span> <span class="fu">cbind</span>(m1.aov, <span class="fu">c</span>(<span class="fu">eta_squared</span>(m1.aov)<span class="sc">$</span>Eta2_partial,<span class="cn">NA</span>))</span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a><span class="fu">colnames</span>(pc7test)[<span class="dv">5</span>]<span class="ot">&lt;-</span> <span class="st">&quot;eta^2&quot;</span> </span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a><span class="fu">summary</span>(m1)</span></code></pre></div>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = PC7 ~ Year + Tissue + Status, data = test)
FALSE 
FALSE Residuals:
FALSE     Min      1Q  Median      3Q     Max 
FALSE -33.893  -3.689   0.468   3.619  15.790 
FALSE 
FALSE Coefficients:
FALSE                   Estimate Std. Error t value Pr(&gt;|t|)    
FALSE (Intercept)        -8.7256     1.8534  -4.708 5.49e-06 ***
FALSE Year2017            9.9678     1.3420   7.427 6.79e-12 ***
FALSE Year2018            8.7427     1.4104   6.199 4.86e-09 ***
FALSE TissuePIT          -0.4707     2.3458  -0.201  0.84122    
FALSE TissueVMH           4.1733     2.1519   1.939  0.05426 .  
FALSE TissueAH           -6.4939     2.4057  -2.699  0.00771 ** 
FALSE TissuePVN          -5.9021     2.3458  -2.516  0.01288 *  
FALSE TissuePOM          -1.5015     2.1519  -0.698  0.48636    
FALSE TissueICO           4.2951     2.2295   1.926  0.05586 .  
FALSE TissueGCT          -5.1958     2.2836  -2.275  0.02425 *  
FALSE TissueAI           11.1688     2.2281   5.013 1.44e-06 ***
FALSE TissueTNA         -11.9698     2.1519  -5.562 1.13e-07 ***
FALSE TissueLS            1.5645     2.1886   0.715  0.47579    
FALSE TissueBSTm          1.9793     2.1519   0.920  0.35909    
FALSE Statusterritorial   2.5403     0.9471   2.682  0.00810 ** 
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 6.086 on 156 degrees of freedom
FALSE Multiple R-squared:  0.5786,  Adjusted R-squared:  0.5407 
FALSE F-statistic:  15.3 on 14 and 156 DF,  p-value: &lt; 2.2e-16</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co">#pc5test&lt;- summary(aov(test$PC5 ~ test$Year + test$Status))</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>pc5test_anova <span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(<span class="fu">signif</span>(pc7test,<span class="dv">2</span>),<span class="at">theme =</span> <span class="fu">ttheme</span>(<span class="at">base_size =</span> <span class="dv">6</span>,<span class="at">padding=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>),<span class="st">&quot;pt&quot;</span>)),<span class="at">rows=</span><span class="fu">rownames</span>(pc7test)) <span class="sc">%&gt;%</span> <span class="fu">tab_add_title</span>(<span class="at">text =</span> <span class="st">&quot;PC7 ~ Year + Tissue + Status &quot;</span>, <span class="at">size=</span><span class="dv">6</span>, <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;line&quot;</span>))</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>e<span class="ot">&lt;-</span><span class="fu">biplot</span>(p, <span class="at">x=</span><span class="st">&quot;PC1&quot;</span>,<span class="at">y=</span><span class="st">&quot;PC7&quot;</span>, <span class="at">lab=</span><span class="cn">NULL</span>, <span class="at">colby=</span><span class="st">&quot;Tissue&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Status&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;&quot;</span>) <span class="sc">+</span> peri_theme</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>f<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x=</span>Tissue, <span class="at">y=</span>PC7, <span class="at">colour=</span>Status)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="at">outlier.colour =</span> <span class="cn">NA</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position=</span>dodge) <span class="sc">+</span> peri_theme  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>status_cols[<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>])  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>))</span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>g<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(<span class="fu">ggarrange</span>(f,<span class="fu">ggarrange</span>(ms,pc5test_anova, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;B&quot;</span>,<span class="st">&quot;C&quot;</span>)), <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>)),c, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">2</span>,<span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;&quot;</span>,<span class="st">&quot;D&quot;</span>))</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>g</span></code></pre></div>
<p><img src="02_1_Checks_PCA_files/figure-html/all%20tissues%20pc7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_alltissues_PC7.pdf&quot;</span>,<span class="at">plot=</span>g, <span class="at">device=</span><span class="st">&quot;pdf&quot;</span>,<span class="at">height =</span><span class="dv">6</span> ,<span class="at">width=</span><span class="fl">7.5</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>,<span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_alltissues_PC7.png&quot;</span>, <span class="at">plot=</span>g,<span class="at">device=</span><span class="st">&quot;png&quot;</span>,<span class="at">height =</span><span class="dv">6</span> ,<span class="at">width=</span><span class="fl">7.5</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span></code></pre></div>
<p>Let’s see what genes are associated with each principal component.
See GitHub repository for larger version.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> p<span class="sc">$</span>loadings</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>loadings<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(loadings)</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> <span class="fu">merge</span>(loadings, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>plots<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>pcs<span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> pcs){</span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>  <span class="co">#i=&quot;PC2&quot;</span></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> loadings[,<span class="fu">c</span>(i, <span class="st">&quot;display_gene_ID&quot;</span>,<span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> sub[<span class="fu">order</span>(sub[,i]),]</span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a>  top30<span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">head</span>(sub,<span class="dv">15</span>), <span class="fu">tail</span>(sub,<span class="dv">15</span>))</span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a>  top30<span class="sc">$</span>best_anno2<span class="ot">&lt;-</span> <span class="fu">factor</span>(top30<span class="sc">$</span>display_gene_ID, <span class="at">levels=</span><span class="fu">unique</span>(top30<span class="sc">$</span>display_gene_ID[<span class="fu">order</span>(top30[,i],top30<span class="sc">$</span>display_gene_ID)]), <span class="at">ordered=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a>  top30<span class="sc">$</span>pos<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(top30[,i]<span class="sc">&lt;</span><span class="dv">0</span>, <span class="st">&quot;neg&quot;</span>, <span class="st">&quot;pos&quot;</span>)</span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a>  plots[[i]]<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top30, <span class="fu">aes</span>(<span class="at">x=</span>best_anno2, <span class="at">y=</span>top30[,i], <span class="at">fill=</span>pos)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span>  peri_figure_supp <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;gene&quot;</span>, <span class="at">y=</span><span class="fu">paste</span>(i,<span class="st">&quot;loading&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#0D8CFF&quot;</span>,<span class="st">&quot;#FF3300&quot;</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a>} </span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a>p<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(<span class="at">plotlist=</span>plots, <span class="at">ncol=</span><span class="dv">5</span>, <span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_alltissue_loadings.pdf&quot;</span>,p, <span class="at">device=</span><span class="st">&quot;pdf&quot;</span>,<span class="at">height =</span><span class="fl">8.5</span> ,<span class="at">width=</span><span class="dv">16</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_alltissue_loadings.png&quot;</span>,p, <span class="at">device=</span><span class="st">&quot;png&quot;</span>,<span class="at">height =</span><span class="fl">8.5</span> ,<span class="at">width=</span><span class="dv">16</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span></code></pre></div>
<div id="pc7-gene-ontology" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> PC7 Gene
Ontology</h2>
<p>What are the gene ontology categories enriched in PC7?</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> loadings[<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;PC7&quot;</span>,<span class="st">&quot;best_anno&quot;</span>)]</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a><span class="co">#res_go$logP&lt;- -log(abs(res_go$PC5), 10)</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co">#res_go$logP&lt;- ifelse(res_go$PC5&lt;0, -(res_go$logP), res_go$logP)</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> res_go[<span class="fu">order</span>(res_go<span class="sc">$</span>PC7, <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>geneList<span class="ot">&lt;-</span> res_go<span class="sc">$</span>PC7</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a><span class="fu">names</span>(geneList)<span class="ot">&lt;-</span> res_go<span class="sc">$</span>gene</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>y<span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> go2gene_bp, <span class="at">TERM2NAME =</span> go2name_bp)</span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a> y<span class="sc">@</span>result<span class="sc">$</span>Description<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">str_count</span>(y<span class="sc">@</span>result<span class="sc">$</span>Description, <span class="st">&#39;</span><span class="sc">\\</span><span class="st">w+&#39;</span>) <span class="sc">&gt;=</span> <span class="dv">10</span>,</span>
<span id="cb24-15"><a href="#cb24-15" tabindex="-1"></a>                                <span class="fu">gsub</span>(<span class="st">&quot;(^[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+)</span><span class="sc">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">1</span><span class="sc">\n</span><span class="st">&quot;</span>,y<span class="sc">@</span>result<span class="sc">$</span>Description, <span class="at">perl=</span><span class="cn">TRUE</span>),</span>
<span id="cb24-16"><a href="#cb24-16" tabindex="-1"></a>                                <span class="fu">ifelse</span>(<span class="fu">str_count</span>(y<span class="sc">@</span>result<span class="sc">$</span>Description, <span class="st">&#39;</span><span class="sc">\\</span><span class="st">w+&#39;</span>) <span class="sc">&gt;=</span> <span class="dv">6</span>,</span>
<span id="cb24-17"><a href="#cb24-17" tabindex="-1"></a>                                <span class="fu">gsub</span>(<span class="st">&quot;(^[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+</span><span class="sc">\\</span><span class="st">s[^</span><span class="sc">\\</span><span class="st">s]+)</span><span class="sc">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">1</span><span class="sc">\n</span><span class="st">&quot;</span>,y<span class="sc">@</span>result<span class="sc">$</span>Description, <span class="at">perl=</span><span class="cn">TRUE</span>),</span>
<span id="cb24-18"><a href="#cb24-18" tabindex="-1"></a>                                <span class="fu">as.character</span>(y<span class="sc">@</span>result<span class="sc">$</span>Description)))</span>
<span id="cb24-19"><a href="#cb24-19" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" tabindex="-1"></a><span class="fu">ridgeplot</span>(y, <span class="at">fill=</span><span class="st">&quot;pvalue&quot;</span>, <span class="at">showCategory=</span><span class="dv">20</span>) <span class="sc">+</span> peri_figure_supp <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="st">&quot;Top 20 GO BP terms from GSEA</span><span class="sc">\n</span><span class="st"> in PC7 according&quot;</span>), <span class="at">x=</span><span class="st">&quot;direction scaled log10(p-value)&quot;</span>) <span class="sc">+</span> <span class="fu">scale_fill_viridis</span>(<span class="at">option=</span><span class="st">&quot;G&quot;</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.key.size=</span><span class="fu">unit</span>(<span class="dv">3</span>,<span class="st">&quot;mm&quot;</span>))</span></code></pre></div>
<p><img src="02_1_Checks_PCA_files/figure-html/PC7%20enrichment-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
<div id="wholebrain-pca" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Wholebrain PCA</h1>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>outliers<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;PFT2_POM_run1&quot;</span>)</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">!=</span><span class="st">&quot;GON&quot;</span>)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(brain_behav, Tissue<span class="sc">!=</span><span class="st">&quot;PIT&quot;</span>)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> brain_behav[<span class="sc">!</span><span class="fu">rownames</span>(brain_behav) <span class="sc">%in%</span> outliers,]</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(brain_behav)</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(brain_behav)]</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(brain_data)</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a><span class="co">#remove genes with avg read counts &lt;20</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>brain_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(brain_data, <span class="dv">1</span>, mean)</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> brain_data[brain_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">20</span>,]</span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>brain_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>brain_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(brain_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(brain_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> brain_data[brain_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>brain_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a><span class="co">#13652 genes.</span></span></code></pre></div>
<p>Looking for cohesive signatures across the whole brain. Overall,
before this round of filtering for the brain dataset we have 16854
genes. After filtering we have 13653 genes. This final document is the
result of lots of work in single tissues to determine outlier status
that were identified in the QC process. We are excluding PFT2_POM_run1
because the analyses below indicated that it was an outlier within POM,
other potentials identified in the QC process were not deemed
outliers.</p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>PC5 seems to be associated with our interest variables.</p>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC5
FALSE           Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year      3471.5   2 46.7499 5.050e-16 ***
FALSE Tissue    3266.2   9  9.7745 2.494e-11 ***
FALSE Residuals 4826.7 130                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC5
FALSE           Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year      3271.1   2  48.888 &lt; 2.2e-16 ***
FALSE Tissue    3255.2   9  10.811 2.109e-12 ***
FALSE Status     511.0   1  15.273 0.0001494 ***
FALSE Residuals 4315.8 129                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: PC5
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE (Intercept)   2925.0   1 87.0463 6.787e-16 ***
FALSE Year          3197.9   2 47.5846 6.082e-16 ***
FALSE Tissue        3305.1   9 10.9288 2.709e-12 ***
FALSE Status         551.4   1 16.4107 9.084e-05 ***
FALSE Tissue:Status  283.5   9  0.9373    0.4957    
FALSE Residuals     4032.3 120                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: PC5
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE (Intercept)     44.2   1  0.7452 0.3897070    
FALSE Tissue        2626.3   9  4.9238 1.225e-05 ***
FALSE Status         748.9   1 12.6365 0.0005391 ***
FALSE Tissue:Status  356.7   9  0.6687 0.7358179    
FALSE Residuals     7230.2 122                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC5
FALSE               Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue        2565.2   9  4.8093 1.692e-05 ***
FALSE Status         711.4   1 12.0036 0.0007334 ***
FALSE Tissue:Status  356.7   9  0.6687 0.7358179    
FALSE Residuals     7230.2 122                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: PC5
FALSE           Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue    2565.2   9  4.9213 1.086e-05 ***
FALSE Status     711.4   1 12.2832 0.0006261 ***
FALSE Residuals 7586.9 131                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = PC5 ~ Year + Tissue + Status, data = test)
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -14.4015  -3.5505   0.1036   3.4221  12.6575 
FALSE 
FALSE Coefficients:
FALSE                   Estimate Std. Error t value Pr(&gt;|t|)    
FALSE (Intercept)        -7.1246     1.8083  -3.940 0.000133 ***
FALSE Year2017           13.0110     1.3751   9.462  &lt; 2e-16 ***
FALSE Year2018           12.5517     1.4441   8.692 1.39e-14 ***
FALSE TissueAH          -16.8285     2.2893  -7.351 2.01e-11 ***
FALSE TissuePVN          -7.8026     2.2324  -3.495 0.000650 ***
FALSE TissuePOM          -3.9960     2.0793  -1.922 0.056838 .  
FALSE TissueICO          -1.7180     2.1192  -0.811 0.419039    
FALSE TissueGCT          -8.7007     2.1717  -4.006 0.000104 ***
FALSE TissueAI            0.2132     2.1175   0.101 0.919963    
FALSE TissueTNA          -9.9407     2.0450  -4.861 3.33e-06 ***
FALSE TissueLS           -2.3957     2.0800  -1.152 0.251559    
FALSE TissueBSTm         -7.7393     2.0450  -3.785 0.000235 ***
FALSE Statusterritorial   3.8609     0.9879   3.908 0.000149 ***
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 5.784 on 129 degrees of freedom
FALSE Multiple R-squared:  0.6026,  Adjusted R-squared:  0.5656 
FALSE F-statistic:  16.3 on 12 and 129 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><img src="02_1_Checks_PCA_files/figure-html/wholebrain%20pc5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> p<span class="sc">$</span>loadings</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>loadings<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(loadings)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> <span class="fu">merge</span>(loadings, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>plots<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>pcs<span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> pcs){</span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a>  <span class="co">#i=&quot;PC2&quot;</span></span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> loadings[,<span class="fu">c</span>(i, <span class="st">&quot;display_gene_ID&quot;</span>,<span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> sub[<span class="fu">order</span>(sub[,i]),]</span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a>  top30<span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">head</span>(sub,<span class="dv">15</span>), <span class="fu">tail</span>(sub,<span class="dv">15</span>))</span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a>  top30<span class="sc">$</span>best_anno2<span class="ot">&lt;-</span> <span class="fu">factor</span>(top30<span class="sc">$</span>display_gene_ID, <span class="at">levels=</span><span class="fu">unique</span>(top30<span class="sc">$</span>display_gene_ID[<span class="fu">order</span>(top30[,i],top30<span class="sc">$</span>display_gene_ID)]), <span class="at">ordered=</span><span class="cn">TRUE</span>)</span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a>  top30<span class="sc">$</span>pos<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(top30[,i]<span class="sc">&lt;</span><span class="dv">0</span>, <span class="st">&quot;neg&quot;</span>, <span class="st">&quot;pos&quot;</span>)</span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>  plots[[i]]<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top30, <span class="fu">aes</span>(<span class="at">x=</span>best_anno2, <span class="at">y=</span>top30[,i], <span class="at">fill=</span>pos)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span>  peri_figure_supp <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;gene&quot;</span>, <span class="at">y=</span><span class="fu">paste</span>(i,<span class="st">&quot;loading&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#0D8CFF&quot;</span>,<span class="st">&quot;#FF3300&quot;</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>} </span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a><span class="fu">ggarrange</span>(<span class="at">plotlist=</span>plots, <span class="at">ncol=</span><span class="dv">5</span>, <span class="at">nrow=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="02_1_Checks_PCA_files/figure-html/loadings%20brain%20tissues-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_brain_loadings.pdf&quot;</span>, <span class="at">device=</span><span class="st">&quot;pdf&quot;</span>,<span class="at">height =</span><span class="fl">8.5</span> ,<span class="at">width=</span><span class="dv">16</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../PCA_results/figure_brain_loadings.png&quot;</span>, <span class="at">device=</span><span class="st">&quot;png&quot;</span>,<span class="at">height =</span><span class="fl">8.5</span> ,<span class="at">width=</span><span class="dv">16</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span></code></pre></div>
<div id="pc5-gene-ontology" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> PC5 Gene
Ontology</h2>
<p>What are the gene ontology categories enriched in PC5?</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> loadings[<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;PC5&quot;</span>,<span class="st">&quot;best_anno&quot;</span>)]</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="co">#res_go$logP&lt;- -log(abs(res_go$PC5), 10)</span></span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a><span class="co">#res_go$logP&lt;- ifelse(res_go$PC5&lt;0, -(res_go$logP), res_go$logP)</span></span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> res_go[<span class="fu">order</span>(res_go<span class="sc">$</span>PC5, <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a>geneList<span class="ot">&lt;-</span> res_go<span class="sc">$</span>PC5</span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a><span class="fu">names</span>(geneList)<span class="ot">&lt;-</span> res_go<span class="sc">$</span>gene</span>
<span id="cb36-9"><a href="#cb36-9" tabindex="-1"></a></span>
<span id="cb36-10"><a href="#cb36-10" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" tabindex="-1"></a></span>
<span id="cb36-12"><a href="#cb36-12" tabindex="-1"></a></span>
<span id="cb36-13"><a href="#cb36-13" tabindex="-1"></a>y<span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> go2gene_bp, <span class="at">TERM2NAME =</span> go2name_bp)</span>
<span id="cb36-14"><a href="#cb36-14" tabindex="-1"></a><span class="fu">ridgeplot</span>(y) <span class="sc">+</span> <span class="fu">scale_fill_viridis</span>(<span class="at">direction=</span><span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span> peri_theme</span></code></pre></div>
<p><img src="02_1_Checks_PCA_files/figure-html/PC5%20enrichment-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="batch-effects" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Batch Effects</h2>
<p>What happens if I use batch effect exclusion on the whole brain? This
way I wouldn’t have to use single tissue</p>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_1_Checks_PCA_files/figure-html/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The PCA plot shows that this makes the differences between batches
worse apparently stronger, even though the eigencorplot (correlation
between PC axis and variable) doesn’t show it. This is probably due to
linear combinations of the batch and tissue variables.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
