<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive Change December 2023" />


<title>Whole brain Co-expression</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_1_Checks_PCA.html">All Tissues and Whole-Brain PCA</a>
    </li>
    <li>
      <a href="./02_2_DESeq2.html">Differential Expression</a>
    </li>
    <li>
      <a href="./03_WGCNA_wholebrain.html">Whole brain co-expression</a>
    </li>
    <li>
      <a href="./04_WGCNA_single_tissue.html">Co-expression in each tissue</a>
    </li>
    <li>
      <a href="./05_candidate_gene_analysis.html">Candidate Genes</a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Whole brain Co-expression</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male
cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive Change December 2023</h4>
<h4 class="date">Last Knit “2023-12-19”</h4>

</div>


<pre><code>FALSE Allowing parallel execution with up to 7 working processes.</code></pre>
<p><br><br></p>
<p>In <a
href="https://onlinelibrary.wiley.com/doi/full/10.1111/gbb.12560">Horton
et al (2020)</a>, they identified bunches of genes with similar
expression patterns (co-expression modules) that aligned with
differences among brain nuclei. The co-expression analysis here (using
WGCNA), has two aims:</p>
<ul>
<li><p>Replicate the results of Horton et al (2020)</p></li>
<li><p>To identify if there are co-expression modules that are shared
amongst brain regions that are associated with our traits of
interest.</p></li>
</ul>
<p>This second aim is in part a complement to the whole brain PCA
analysis done in the <a
href="https://periperipatus.github.io/PIFI_brain_transcriptome/02_DESeq2_FINAL.html">DESeq2
document</a>.</p>
<div id="data-quality-control" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Data Quality
Control</h1>
<p>This is where I originally ran the test on the effect of batch
correction on the whole brain dataset (now also presented in the <a
href="https://periperipatus.github.io/PIFI_brain_transcriptome/02_DESeq2_FINAL.html">DESeq2
document</a>). As identified in that document and the QC process,
<code>PFT2_POM</code> sample was removed from this analysis.</p>
<div id="connectivity-test-for-outliers" class="section level2"
number="1.1">
<h2><span class="header-section-number">1.1</span> Connectivity Test for
Outliers</h2>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="effect-of-batch-correction" class="section level2"
number="1.2">
<h2><span class="header-section-number">1.2</span> Effect of Batch
Correction</h2>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/batch%20correction-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/batch%20correction-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The analysis above also suggests PFT1_TNA_run2 is an outlier, based
on the connectivity to the remainder of the analysis. PFT1_TNA_run2 is
still an outlier when I exclude it from the smaller dataset that exludes
the pilot samples (not shown). Therefore, I have excuded it below.</p>
<p>What do the data look like without the pilot samples (Horton et al
2020) and batch correction? Here I rerun the same analyses as above but
excluding the batch correction and without the pilot samples.</p>
<p><br></p>
</div>
<div id="removing-the-pilot-data" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Removing the pilot
data</h2>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/removing%20pilot%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="batch-effect-correction" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Batch Effect
Correction</h3>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The PCA plot without batch correction still shows that tissues (and
different batches) are clustered more closely. Perhaps in part this is
still because batches are not evenly distributed across tissues.</p>
<p>Going forward with the WGCNA analysis on all brain samples, I will
exclude pilot data and I will not conduct batch correction. However, I
will include a batch variable in the trait-module correlation
matrix.</p>
<p><br><br></p>
</div>
</div>
</div>
<div id="wgcna-analysis" class="section level1" number="2">
<h1><span class="header-section-number">2</span> WGCNA Analysis</h1>
<p>Using the dataset without the pilot data and the two outliers
<code>PFT2_POM_run1</code> and <code>PFT1_TNA_run2</code>, I will now
commence the main co-expression analysis.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#write.csv(datExpr0, file=&quot;wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>gsg <span class="ot">=</span> <span class="fu">goodSamplesGenes</span>(datExpr0, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<p><br></p>
<div id="soft-threshold-selection" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Soft Threshold
Selection</h2>
<pre><code>FALSE    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1    0.128 108.00          0.968 6810.00   6810.00 6870.0
FALSE 2      2    0.174  -9.19          0.893 3720.00   3710.00 4120.0
FALSE 3      3    0.111  -2.24          0.923 2170.00   2160.00 2750.0
FALSE 4      4    0.208  -1.89          0.945 1340.00   1320.00 2010.0
FALSE 5      5    0.343  -1.81          0.966  868.00    846.00 1540.0
FALSE 6      6    0.456  -1.74          0.978  584.00    559.00 1220.0
FALSE 7      7    0.558  -1.72          0.984  407.00    380.00  997.0
FALSE 8      8    0.641  -1.77          0.984  291.00    264.00  831.0
FALSE 9      9    0.695  -1.81          0.979  214.00    188.00  705.0
FALSE 10    10    0.734  -1.82          0.980  161.00    135.00  606.0
FALSE 11    11    0.757  -1.85          0.975  123.00     99.40  526.0
FALSE 12    12    0.784  -1.87          0.976   95.80     73.80  462.0
FALSE 13    14    0.821  -1.90          0.981   60.70     41.90  364.0
FALSE 14    16    0.845  -1.90          0.985   40.30     24.70  294.0
FALSE 15    18    0.859  -1.93          0.985   27.80     15.00  243.0
FALSE 16    20    0.873  -1.92          0.987   19.90      9.39  203.0
FALSE 17    22    0.870  -1.94          0.981   14.60      6.00  172.0
FALSE 18    24    0.876  -1.93          0.983   10.90      3.91  148.0
FALSE 19    26    0.883  -1.91          0.983    8.36      2.60  128.0
FALSE 20    28    0.884  -1.90          0.984    6.50      1.75  112.0
FALSE 21    30    0.885  -1.88          0.986    5.14      1.20   98.7</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/soft%20threshold-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Seems like there is no strong effect on Scale Free Topology The
soft-threshold power I have selected here is 20, after which there is
little improvement in the Scale free topology fit.</p>
<p>Going forward I will just use the pilot batch free data to be safe,
given this was the stronger of the batch effects.</p>
<p><br></p>
</div>
<div id="adjacency-and-topological-overlap-matrices"
class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Adjacency and
Topological Overlap matrices</h2>
<p>The adjacency matrix and Topological Overlap was made on an external
server to save my PC some RAMs.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">20</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#datExpr0&lt;- read.csv(&quot;../WGCNA_results/all_brain/wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#rownames(datExpr0)&lt;- datExpr0$X</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#datExpr0$X&lt;- NULL</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>adjacency<span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr0, <span class="at">power =</span> softPower, <span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>TOM<span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMType=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#dissTOM&lt;- 1-TOM</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="fu">save</span>(adjacency, TOM, <span class="at">file=</span><span class="st">&quot;../WGCNA_results/all_brain/wholebrain_network2023.RData&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="identify-co-expression-modules" class="section level2"
number="2.3">
<h2><span class="header-section-number">2.3</span> Identify
co-expression modules</h2>
<p>The first step is to identify modules of genes with similar gene
expression. Basically, the tool creates a hierarchical clustering of the
topological dissimilarity between genes.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co">#write.csv(datExpr0,file=&quot;../WGCNA_results/all_brain/wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">#for the rmarkdown knit this data is already loaded above.</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#datExpr0&lt;- read.csv(&quot;../WGCNA_results/all_brain/wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#rownames(datExpr0)&lt;- datExpr0$X</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="co">#datExpr0$X&lt;- NULL</span></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;../WGCNA_results/all_brain/wholebrain_network2023.RData&quot;</span>)</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>dissTOM<span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>TOM</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>geneTree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method=</span><span class="st">&quot;average&quot;</span>)</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#plot(geneTree, xlab=&quot;&quot;, sub=&quot;&quot;, main= &quot;Gene Clustering on TOM-based dissimilarity&quot;, labels= FALSE, hang=0.04)</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>minModuleSize<span class="ot">&lt;-</span><span class="dv">30</span> </span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>dynamicMods<span class="ot">&lt;-</span><span class="fu">cutreeDynamic</span>(<span class="at">dendro=</span> geneTree, <span class="at">distM=</span> dissTOM, <span class="at">deepSplit=</span><span class="dv">2</span>, <span class="at">pamRespectsDendro=</span> <span class="cn">FALSE</span>, <span class="at">minClusterSize=</span> minModuleSize)</span></code></pre></div>
<pre><code>FALSE  ..cutHeight not given, setting it to 0.998  ===&gt;  99% of the (truncated) height range in dendro.
FALSE  ..done.</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co">#table(dynamicMods)</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>dynamicColors<span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span> <span class="fl">0.05</span>, <span class="at">main=</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span></code></pre></div>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/calculate%20modules-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now we try to merge some of these modules that are particularly
similar in expression based on a similarity threshold.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co">#-----Merge modules whose expression profiles are very similar</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>MEList<span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, <span class="at">colors=</span> dynamicColors)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>MEs<span class="ot">=</span> MEList<span class="sc">$</span>eigengenes</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">#Calculate dissimilarity of module eigenegenes</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>MEDiss<span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(MEs)</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a><span class="co">#Cluster module eigengenes</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>METree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method=</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co">#plot(METree, main= &quot;Clustering of module eigengenes&quot;, xlab= &quot;&quot;, sub= &quot;&quot;)</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>MEDissThres<span class="ot">=</span> <span class="fl">0.30</span> <span class="co"># i.e. merge modules with an r2 &gt; 0.90. This is stringent, could relax to reduce number of modules and increase module size.</span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="co">#abline(h=MEDissThres, col=&quot;red&quot;)</span></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a>merge<span class="ot">=</span> <span class="fu">mergeCloseModules</span>(datExpr0, dynamicColors, <span class="at">cutHeight=</span> MEDissThres, <span class="at">verbose =</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>FALSE  mergeCloseModules: Merging modules whose distance is less than 0.3
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 61 module eigengenes in given set.
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 27 module eigengenes in given set.
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 22 module eigengenes in given set.
FALSE    Calculating new MEs...
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 22 module eigengenes in given set.</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>mergedColors<span class="ot">=</span>merge<span class="sc">$</span>colors</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>mergedMEs<span class="ot">=</span> merge<span class="sc">$</span>newMEs</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, <span class="fu">cbind</span>(dynamicColors, mergedColors), <span class="fu">c</span>(<span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="st">&quot;Merged dynamic&quot;</span>), <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>moduleColors<span class="ot">=</span> mergedColors</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>colorOrder<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="fu">standardColors</span>(<span class="dv">50</span>))</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>moduleLabels<span class="ot">=</span> <span class="fu">match</span>(moduleColors, colorOrder)<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>MEs<span class="ot">=</span>mergedMEs</span></code></pre></div>
<p><br></p>
</div>
<div id="module-trait-correlation" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Module-trait
correlation</h2>
<p>Correlate the eigenvector of each of the 22 co-expression modules
with brain regions, individual and batch, as well as our interest
variables (mean testosterone, status, and social network strength)</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> key_brain</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(datTraits))</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>datTraits<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;run1&quot;</span>,datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>datTraits<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>Year))</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>datTraits<span class="sc">$</span>Ai<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;AI&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>datTraits<span class="sc">$</span>TnA<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TNA&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>datTraits<span class="sc">$</span>BSTm<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;BST&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>datTraits<span class="sc">$</span>ICO<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;ICO&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>datTraits<span class="sc">$</span>GCT<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;GCT&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>datTraits<span class="sc">$</span>LS<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;LS&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>datTraits<span class="sc">$</span>POM<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;POM&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>datTraits<span class="sc">$</span>VMH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;VMH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a>datTraits<span class="sc">$</span>AH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;AH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>datTraits<span class="sc">$</span>PVN<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;PVN&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>datTraits<span class="sc">$</span>Bird<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>sampleID))</span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a>datTraits<span class="sc">$</span>Status2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(datTraits<span class="sc">$</span>Status<span class="sc">==</span><span class="st">&quot;territorial&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a></span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> <span class="fu">subset</span>(datTraits, <span class="at">select=</span><span class="fu">c</span>(<span class="st">&quot;Bird&quot;</span>,<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Year&quot;</span>,<span class="st">&quot;Status2&quot;</span>, <span class="st">&quot;mean_T&quot;</span>,<span class="st">&quot;strength.all_study&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;Ai&quot;</span>,<span class="st">&quot;TnA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a><span class="fu">names</span>(datTraits)[<span class="fu">names</span>(datTraits)<span class="sc">==</span><span class="st">&quot;Status2&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Status&quot;</span></span>
<span id="cb21-20"><a href="#cb21-20" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" tabindex="-1"></a></span>
<span id="cb21-22"><a href="#cb21-22" tabindex="-1"></a></span>
<span id="cb21-23"><a href="#cb21-23" tabindex="-1"></a><span class="co">#-----Define numbers of genes and samples</span></span>
<span id="cb21-24"><a href="#cb21-24" tabindex="-1"></a>nGenes <span class="ot">=</span> <span class="fu">ncol</span>(datExpr0);</span>
<span id="cb21-25"><a href="#cb21-25" tabindex="-1"></a>nSamples <span class="ot">=</span> <span class="fu">nrow</span>(datExpr0);</span>
<span id="cb21-26"><a href="#cb21-26" tabindex="-1"></a><span class="co">#-----Recalculate MEs with color labels</span></span>
<span id="cb21-27"><a href="#cb21-27" tabindex="-1"></a>MEs0 <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, moduleColors)<span class="sc">$</span>eigengenes</span>
<span id="cb21-28"><a href="#cb21-28" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb21-29"><a href="#cb21-29" tabindex="-1"></a></span>
<span id="cb21-30"><a href="#cb21-30" tabindex="-1"></a><span class="co">#-----Correlations of genes with eigengenes</span></span>
<span id="cb21-31"><a href="#cb21-31" tabindex="-1"></a>moduleGeneCor<span class="ot">=</span><span class="fu">cor</span>(MEs,datExpr0)</span>
<span id="cb21-32"><a href="#cb21-32" tabindex="-1"></a>moduleGenePvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleGeneCor, nSamples);</span>
<span id="cb21-33"><a href="#cb21-33" tabindex="-1"></a></span>
<span id="cb21-34"><a href="#cb21-34" tabindex="-1"></a>moduleTraitCor <span class="ot">=</span> <span class="fu">cor</span>(MEs, datTraits, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>);</span>
<span id="cb21-35"><a href="#cb21-35" tabindex="-1"></a>moduleTraitPvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleTraitCor, nSamples);</span>
<span id="cb21-36"><a href="#cb21-36" tabindex="-1"></a></span>
<span id="cb21-37"><a href="#cb21-37" tabindex="-1"></a><span class="co">#---------------------Module-trait heatmap</span></span>
<span id="cb21-38"><a href="#cb21-38" tabindex="-1"></a></span>
<span id="cb21-39"><a href="#cb21-39" tabindex="-1"></a></span>
<span id="cb21-40"><a href="#cb21-40" tabindex="-1"></a>textMatrix <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">signif</span>(moduleTraitCor, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>,</span>
<span id="cb21-41"><a href="#cb21-41" tabindex="-1"></a>                   <span class="fu">signif</span>(moduleTraitPvalue, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>);</span>
<span id="cb21-42"><a href="#cb21-42" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">=</span> <span class="fu">dim</span>(moduleTraitCor)</span>
<span id="cb21-43"><a href="#cb21-43" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">7</span>, <span class="dv">9</span>, <span class="dv">3</span>, <span class="dv">3</span>));</span>
<span id="cb21-44"><a href="#cb21-44" tabindex="-1"></a><span class="co"># Display the correlation values within a heatmap plot</span></span>
<span id="cb21-45"><a href="#cb21-45" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> moduleTraitCor,</span>
<span id="cb21-46"><a href="#cb21-46" tabindex="-1"></a>               <span class="at">xLabels =</span> <span class="fu">names</span>(datTraits),</span>
<span id="cb21-47"><a href="#cb21-47" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb21-48"><a href="#cb21-48" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb21-49"><a href="#cb21-49" tabindex="-1"></a>               <span class="at">colorLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb21-50"><a href="#cb21-50" tabindex="-1"></a>               <span class="at">colors =</span> <span class="fu">blueWhiteRed</span>(<span class="dv">50</span>),</span>
<span id="cb21-51"><a href="#cb21-51" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix,</span>
<span id="cb21-52"><a href="#cb21-52" tabindex="-1"></a>               <span class="at">setStdMargins =</span> <span class="cn">FALSE</span>,</span>
<span id="cb21-53"><a href="#cb21-53" tabindex="-1"></a>               <span class="at">cex.text =</span> <span class="fl">0.5</span>,</span>
<span id="cb21-54"><a href="#cb21-54" tabindex="-1"></a>               <span class="at">zlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb21-55"><a href="#cb21-55" tabindex="-1"></a>               <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module-trait relationships in whole brain&quot;</span>))</span></code></pre></div>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/trait%20correlation-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of genes per co-expression module
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
1143
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
891
</td>
</tr>
<tr>
<td style="text-align:left;">
cyan
</td>
<td style="text-align:right;">
680
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
1487
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
509
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen
</td>
<td style="text-align:right;">
136
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
91
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen4
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
216
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
275
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
50
</td>
</tr>
<tr>
<td style="text-align:left;">
ivory
</td>
<td style="text-align:right;">
164
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple3
</td>
<td style="text-align:right;">
333
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite2
</td>
<td style="text-align:right;">
808
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
662
</td>
</tr>
<tr>
<td style="text-align:left;">
plum1
</td>
<td style="text-align:right;">
488
</td>
</tr>
<tr>
<td style="text-align:left;">
purple
</td>
<td style="text-align:right;">
1592
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna3
</td>
<td style="text-align:right;">
1316
</td>
</tr>
<tr>
<td style="text-align:left;">
tan
</td>
<td style="text-align:right;">
1212
</td>
</tr>
<tr>
<td style="text-align:left;">
violet
</td>
<td style="text-align:right;">
350
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
yellowgreen
</td>
<td style="text-align:right;">
536
</td>
</tr>
</tbody>
</table>
<p>See the data portion of this repository to see the the module
membership and gene-significance results.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>datME<span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr0,mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>datKME<span class="ot">&lt;-</span> <span class="fu">signedKME</span>(datExpr0, datME, <span class="at">outputColumnName=</span><span class="st">&quot;MM.&quot;</span>) <span class="co">#use the &quot;signed eigennode connectivity&quot; or module membership</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>MMPvalue <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(datKME), nSamples)) <span class="co"># Calculate module membership P-values</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>datKME<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(datKME)</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>MMPvalue<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(MMPvalue)</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>genes<span class="ot">=</span><span class="fu">names</span>(datExpr0)</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>genes,<span class="at">moduleColor=</span>moduleColors)</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>color<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, datKME, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>) <span class="co">#these are from your original WGCNA analysis </span></span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a><span class="co">#head(color)</span></span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(color), <span class="at">file =</span> <span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/all_brain/wholebrain_results_ModuleMembership.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;_&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a>MMPvalue<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, MMPvalue, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a><span class="fu">write.csv</span>(MMPvalue, <span class="at">file=</span><span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/all_brain/wholebrain_results_ModuleMembership_P-value.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;_&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a><span class="do">#### gene-significance with traits of interest.</span></span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>Status) <span class="co">#change here for traits of interest</span></span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a><span class="fu">names</span>(trait) <span class="ot">=</span> <span class="st">&quot;status&quot;</span> <span class="co">#change here for traits of interest </span></span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a>modNames <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">names</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-24"><a href="#cb22-24" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-25"><a href="#cb22-25" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb22-26"><a href="#cb22-26" tabindex="-1"></a></span>
<span id="cb22-27"><a href="#cb22-27" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>mean_T)</span>
<span id="cb22-28"><a href="#cb22-28" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;mean_T&quot;</span></span>
<span id="cb22-29"><a href="#cb22-29" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb22-30"><a href="#cb22-30" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb22-31"><a href="#cb22-31" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-32"><a href="#cb22-32" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-33"><a href="#cb22-33" tabindex="-1"></a>GS2<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb22-34"><a href="#cb22-34" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>strength.all_study)</span>
<span id="cb22-37"><a href="#cb22-37" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;strength&quot;</span></span>
<span id="cb22-38"><a href="#cb22-38" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb22-39"><a href="#cb22-39" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb22-40"><a href="#cb22-40" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-41"><a href="#cb22-41" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb22-42"><a href="#cb22-42" tabindex="-1"></a>GS3<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb22-43"><a href="#cb22-43" tabindex="-1"></a></span>
<span id="cb22-44"><a href="#cb22-44" tabindex="-1"></a>GS<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(GS)</span>
<span id="cb22-45"><a href="#cb22-45" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(GS,GS2, GS3)</span>
<span id="cb22-46"><a href="#cb22-46" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0,GS, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb22-47"><a href="#cb22-47" tabindex="-1"></a></span>
<span id="cb22-48"><a href="#cb22-48" tabindex="-1"></a></span>
<span id="cb22-49"><a href="#cb22-49" tabindex="-1"></a><span class="fu">write.csv</span>(GS, <span class="at">file=</span><span class="st">&quot;../WGCNA_results/all_brain/wholebrain_results_GeneSignificance.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="intermodule-correlations" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> intermodule
correlations</h2>
<p>Given the network type used, all genes in a module can only go in the
same direction. This means that another module may represent genes that
are downregulated by the upregulation of genes in another module (or
vice-versa).</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Specify colors</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>colz<span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(MEs))</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="fu">names</span>(colz)<span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>ann_colors <span class="ot">=</span> <span class="fu">list</span>(<span class="at">module=</span>colz)</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>annotation_col<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">colnames</span>(MEs), <span class="at">module=</span><span class="fu">colnames</span>(MEs))</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a>correlation<span class="ot">&lt;-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a><span class="fu">pheatmap</span>(correlation, <span class="at">annotation_col =</span> annotation_col, <span class="at">annotation_colors =</span> ann_colors, <span class="at">annotation_legend=</span><span class="cn">FALSE</span>, <span class="at">legend_breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="fl">0.5</span>,<span class="dv">0</span>,<span class="fl">0.5</span>, <span class="dv">1</span>,<span class="dv">1</span>), </span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a><span class="at">main=</span><span class="st">&quot;module eigengene correlation in whole brain&quot;</span>, <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;-1&quot;</span>, <span class="st">&quot;-0.5&quot;</span>, <span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.5&quot;</span>,<span class="st">&quot;1&quot;</span>,<span class="st">&quot;correlation</span><span class="sc">\n\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/intermodule%20correlations-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="interesting-modules" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Interesting
Modules</h1>
<p>There are three modules that appear to be associated with our traits
of interest. Let’s explore these further. Because the module-trait
correlation procedure cannot account for tissues taken from the same
individual, I will test to see if these relationships still hold when
accounting for this in a linear mixed model.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co">#write.csv(MEs, file=&quot;../WGCNA_results/all_brain/ModuleEigengenes.csv&quot;)</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a><span class="co">#color&lt;- read.csv(&quot;../WGCNA_results/all_brain/wholebrain_results_ModuleMembership.csv&quot;)</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co">#MEs&lt;- read.csv(&quot;../WGCNA_results/all_brain/ModuleEigengenes.csv&quot;)</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="co">#rownames(MEs)&lt;- MEs$X</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="co">#MEs$X&lt;- NULL</span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a><span class="co">#merge module eigengenes with trait matrix.</span></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>mes<span class="ot">&lt;-</span> MEs</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>mes<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mes)</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>mes<span class="ot">&lt;-</span> <span class="fu">merge</span>(key_brain, mes, <span class="at">by=</span><span class="st">&quot;sampleID&quot;</span>)</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>mes<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(mes<span class="sc">$</span>Year)</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>mes<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(mes<span class="sc">$</span>Batch)</span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a>mes<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(mes<span class="sc">$</span>Tissue)</span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>mes<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">factor</span>(mes<span class="sc">$</span>Tissue, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;AI&quot;</span>,<span class="st">&quot;TNA&quot;</span>,<span class="st">&quot;LS&quot;</span>,<span class="st">&quot;BSTm&quot;</span>))</span></code></pre></div>
<div id="darkturquoise-module" class="section level3" number="3.0.1">
<h3><span class="header-section-number">3.0.1</span> darkturquoise
module</h3>
<p>This module is potentially associated with social status.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>whichModule<span class="ot">=</span><span class="st">&quot;darkturquoise&quot;</span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="co">#m1 &lt;- lme(MEdarkturquoise~Batch + Tissue + Status,random=~ 1|Harvest_ID,data=mes)</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="co">#anova(m1)</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="co">#Test 3 models </span></span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>m0<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data=</span>mes)</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>m01<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise <span class="sc">~</span> Batch, <span class="at">data=</span>mes)</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>m01.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m01, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>m02<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise <span class="sc">~</span> Batch <span class="sc">+</span> Tissue, <span class="at">data=</span>mes)</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>m02.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m02, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>m02.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m02)[,<span class="dv">1</span>])</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a><span class="co">#we&#39;re scaling Status because unscaled data strongly inflates VIF while the scale data doesn&#39;t change our P-values. </span></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a>m1<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise<span class="sc">~</span> Batch <span class="sc">+</span> Tissue <span class="sc">+</span> Status,<span class="at">data=</span>mes)</span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>m1.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m1, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>m1.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m1)[,<span class="dv">1</span>])</span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>m1.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEdarkturquoise
FALSE            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Batch     0.00315   1  0.5019 0.4802691    
FALSE Tissue    0.23429   9  4.1433 0.0001407 ***
FALSE Status    0.04323   1  6.8804 0.0100371 *  
FALSE Residuals 0.64714 103                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>m2<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise<span class="sc">~</span> Batch <span class="sc">+</span> Tissue <span class="sc">*</span> Status,<span class="at">data=</span>mes,<span class="at">contrasts=</span><span class="fu">list</span>(<span class="at">Tissue=</span>contr.sum, <span class="at">Status=</span>contr.sum))</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>m2.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m2, <span class="at">type=</span><span class="dv">3</span>)</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>m2.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m2)[,<span class="dv">1</span>])</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>m2.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEdarkturquoise
FALSE                Sum Sq Df F value    Pr(&gt;F)    
FALSE (Intercept)   0.00004  1  0.0062  0.937229    
FALSE Batch         0.00021  1  0.0342  0.853692    
FALSE Tissue        0.25965  9  4.6867 3.822e-05 ***
FALSE Status        0.04802  1  7.8013  0.006326 ** 
FALSE Tissue:Status 0.06850  9  1.2365  0.282528    
FALSE Residuals     0.57864 94                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">summary</span>(m2)</span></code></pre></div>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEdarkturquoise ~ Batch + Tissue * Status, data = mes, 
FALSE     contrasts = list(Tissue = contr.sum, Status = contr.sum))
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -0.15413 -0.05060 -0.01576  0.03812  0.23016 
FALSE 
FALSE Coefficients:
FALSE                  Estimate Std. Error t value Pr(&gt;|t|)    
FALSE (Intercept)     -0.001323   0.016756  -0.079  0.93723    
FALSE Batchrun2       -0.005392   0.029158  -0.185  0.85369    
FALSE Tissue1         -0.046006   0.024978  -1.842  0.06865 .  
FALSE Tissue2         -0.054524   0.027202  -2.004  0.04790 *  
FALSE Tissue3         -0.035985   0.026046  -1.382  0.17037    
FALSE Tissue4         -0.001935   0.027059  -0.072  0.94315    
FALSE Tissue5         -0.028952   0.023227  -1.246  0.21568    
FALSE Tissue6          0.039825   0.022426   1.776  0.07900 .  
FALSE Tissue7          0.043789   0.027648   1.584  0.11660    
FALSE Tissue8          0.112192   0.024503   4.579 1.43e-05 ***
FALSE Tissue9          0.030718   0.026046   1.179  0.24122    
FALSE Status1         -0.021125   0.007563  -2.793  0.00633 ** 
FALSE Tissue1:Status1 -0.028490   0.021930  -1.299  0.19708    
FALSE Tissue2:Status1 -0.016591   0.023257  -0.713  0.47738    
FALSE Tissue3:Status1  0.013550   0.021893   0.619  0.53747    
FALSE Tissue4:Status1  0.027446   0.022553   1.217  0.22666    
FALSE Tissue5:Status1 -0.034489   0.023889  -1.444  0.15214    
FALSE Tissue6:Status1  0.001332   0.021867   0.061  0.95155    
FALSE Tissue7:Status1 -0.009482   0.023257  -0.408  0.68442    
FALSE Tissue8:Status1  0.009210   0.022532   0.409  0.68364    
FALSE Tissue9:Status1  0.051759   0.021893   2.364  0.02013 *  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.07846 on 94 degrees of freedom
FALSE Multiple R-squared:  0.4214,  Adjusted R-squared:  0.2982 
FALSE F-statistic: 3.423 on 20 and 94 DF,  p-value: 2.87e-05</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>m3<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise<span class="sc">~</span> Tissue <span class="sc">+</span> Status <span class="sc">+</span> Tissue<span class="sc">:</span>Status,<span class="at">data=</span>mes,<span class="at">contrasts=</span><span class="fu">list</span>(<span class="at">Tissue=</span>contr.sum, <span class="at">Status=</span>contr.sum))</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>m3.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m3, <span class="at">type=</span><span class="dv">3</span>)</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>m3.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m3)[,<span class="dv">1</span>])</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>m3.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEdarkturquoise
FALSE                Sum Sq Df F value    Pr(&gt;F)    
FALSE (Intercept)   0.00185  1  0.3038  0.582781    
FALSE Tissue        0.31838  9  5.8058 2.067e-06 ***
FALSE Status        0.05025  1  8.2476  0.005032 ** 
FALSE Tissue:Status 0.07144  9  1.3028  0.245751    
FALSE Residuals     0.57885 95                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">summary</span>(m3)</span></code></pre></div>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEdarkturquoise ~ Tissue + Status + Tissue:Status, 
FALSE     data = mes, contrasts = list(Tissue = contr.sum, Status = contr.sum))
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -0.15798 -0.05001 -0.01422  0.03395  0.23016 
FALSE 
FALSE Coefficients:
FALSE                   Estimate Std. Error t value Pr(&gt;|t|)    
FALSE (Intercept)     -0.0040967  0.0074322  -0.551  0.58278    
FALSE Tissue1         -0.0437715  0.0217498  -2.013  0.04700 *  
FALSE Tissue2         -0.0571423  0.0231081  -2.473  0.01518 *  
FALSE Tissue3         -0.0386035  0.0217498  -1.775  0.07912 .  
FALSE Tissue4          0.0008387  0.0224069   0.037  0.97022    
FALSE Tissue5         -0.0289703  0.0231081  -1.254  0.21303    
FALSE Tissue6          0.0407500  0.0217498   1.874  0.06406 .  
FALSE Tissue7          0.0465629  0.0231081   2.015  0.04673 *  
FALSE Tissue8          0.1139775  0.0224069   5.087 1.83e-06 ***
FALSE Tissue9          0.0280999  0.0217498   1.292  0.19950    
FALSE Status1         -0.0213443  0.0074322  -2.872  0.00503 ** 
FALSE Tissue1:Status1 -0.0288100  0.0217498  -1.325  0.18848    
FALSE Tissue2:Status1 -0.0163718  0.0231081  -0.708  0.48038    
FALSE Tissue3:Status1  0.0137689  0.0217498   0.633  0.52822    
FALSE Tissue4:Status1  0.0276651  0.0224069   1.235  0.22000    
FALSE Tissue5:Status1 -0.0355218  0.0231081  -1.537  0.12757    
FALSE Tissue6:Status1  0.0012429  0.0217498   0.057  0.95455    
FALSE Tissue7:Status1 -0.0092628  0.0231081  -0.401  0.68943    
FALSE Tissue8:Status1  0.0093394  0.0224069   0.417  0.67776    
FALSE Tissue9:Status1  0.0519778  0.0217498   2.390  0.01883 *  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.07806 on 95 degrees of freedom
FALSE Multiple R-squared:  0.4212,  Adjusted R-squared:  0.3054 
FALSE F-statistic: 3.638 on 19 and 95 DF,  p-value: 1.469e-05</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>m4<span class="ot">&lt;-</span> <span class="fu">lm</span>(MEdarkturquoise<span class="sc">~</span> Tissue <span class="sc">+</span> Status,<span class="at">data=</span>mes)</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>m4.aov<span class="ot">&lt;-</span> <span class="fu">Anova</span>(m4, <span class="at">type=</span><span class="dv">2</span>)</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>m4.vif<span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">vif</span>(m4)[,<span class="dv">1</span>])</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>m4.aov</span></code></pre></div>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEdarkturquoise
FALSE            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue    0.30656   9  5.4475 4.117e-06 ***
FALSE Status    0.04779   1  7.6424  0.006745 ** 
FALSE Residuals 0.65029 104                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>models<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;~ 1&quot;</span>, <span class="st">&quot;~ Batch&quot;</span>, <span class="st">&quot;~ Batch + Tissue&quot;</span>,<span class="st">&quot;~ Batch + Tissue + Status&quot;</span> , <span class="st">&quot;~ Batch + Tissue + Status + Tissue:Status&quot;</span>,<span class="st">&quot;~ Tissue + Status + Tissue:Status&quot;</span>, <span class="st">&quot;~ Tissue + Status&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>aic<span class="ot">=</span><span class="fu">signif</span>(<span class="fu">c</span>(<span class="fu">AIC</span>(m0),<span class="fu">AIC</span>(m01),<span class="fu">AIC</span>(m02),<span class="fu">AIC</span>(m1),<span class="fu">AIC</span>(m2),<span class="fu">AIC</span>(m3),<span class="fu">AIC</span>(m4)),<span class="dv">4</span>)</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>deltaAIC<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>,aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">2</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">3</span>], aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">4</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">5</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">6</span>],aic[<span class="dv">1</span>]<span class="sc">-</span>aic[<span class="dv">7</span>]),<span class="dv">3</span>) </span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>batch_p<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>, m01.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m02.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m1.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],m2.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>],<span class="cn">NA</span>,<span class="cn">NA</span>),<span class="dv">2</span>)</span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>e_p<span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">c</span>(<span class="cn">NA</span>, <span class="cn">NA</span>,<span class="cn">NA</span>,m1.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">3</span>],m2.aov<span class="sc">$</span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">3</span>],m3.aov<span class="sc">$</span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>],m4.aov<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>]),<span class="dv">2</span>) </span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>sumVIF<span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">c</span>(<span class="cn">NA</span>,<span class="cn">NA</span>,m02.vif, m1.vif,m2.vif, m3.vif, m4.vif),<span class="dv">0</span>)</span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a>ms<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(models,aic, deltaAIC,batch_p,e_p, sumVIF)</span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a>ms<span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(ms,<span class="at">theme =</span> <span class="fu">ttheme</span>(<span class="at">base_size =</span> <span class="dv">6</span>,<span class="at">padding=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>),<span class="st">&quot;pt&quot;</span>)),<span class="at">rows=</span><span class="cn">NULL</span>)</span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a>m4.aov<span class="ot">&lt;-</span> <span class="fu">cbind</span>(m4.aov, <span class="fu">c</span>(<span class="fu">eta_squared</span>(m4.aov)<span class="sc">$</span>Eta2_partial,<span class="cn">NA</span>))</span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a><span class="fu">colnames</span>(m4.aov)[<span class="dv">5</span>]<span class="ot">&lt;-</span> <span class="st">&quot;eta^2&quot;</span></span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a></span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">signif</span>(m4.aov,<span class="dv">2</span>))</span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a></span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a></span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(res,<span class="at">theme =</span> <span class="fu">ttheme</span>(<span class="at">base_size =</span> <span class="dv">6</span>,<span class="at">padding=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>),<span class="st">&quot;pt&quot;</span>)),<span class="at">rows=</span><span class="fu">rownames</span>(res)) <span class="sc">%&gt;%</span> <span class="fu">tab_add_title</span>(<span class="at">text =</span> <span class="fu">paste0</span>(whichModule,models[<span class="dv">7</span>]), <span class="at">size=</span><span class="dv">6</span>, <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;line&quot;</span>))</span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a></span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a></span>
<span id="cb37-24"><a href="#cb37-24" tabindex="-1"></a><span class="co">#plot the module eigengene against our traits of interest</span></span>
<span id="cb37-25"><a href="#cb37-25" tabindex="-1"></a>a<span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data=</span>mes, <span class="fu">aes</span>(<span class="at">x=</span>Tissue, <span class="at">y=</span>MEdarkturquoise, <span class="at">color=</span>status), <span class="at">position=</span><span class="fu">position_dodge</span>(<span class="at">width=</span><span class="fl">0.7</span>)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="at">data=</span>mes, <span class="fu">aes</span>(<span class="at">x=</span>Tissue, <span class="at">y=</span>MEdarkturquoise, <span class="at">color=</span>status), <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">outlier.colour =</span> <span class="cn">NA</span>) <span class="sc">+</span> peri_figure <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;darkturquoise module eigengene vs status&quot;</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>status_cols2) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb37-26"><a href="#cb37-26" tabindex="-1"></a>a</span></code></pre></div>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/darkturquoise-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="co">#g&lt;- ggarrange(a,res, labels=LETTERS[1:2], ncol=2)</span></span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="co">#ggsave(filename=paste0(&quot;../WGCNA_results/all_brain/figure_&quot;,whichModule,&quot;_eigengene.png&quot;),plot = g, device=&quot;png&quot;, height=100, width=80, units=&quot;mm&quot;, bg=&quot;white&quot;)</span></span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a><span class="co">#a&lt;- ggplot() + geom_point(data=mes, aes(x=Tissue, y=MEdarkturquoise, color=status), position=position_dodge(width=0.7),size=0.5) + geom_boxplot(data=mes, aes(x=Tissue, y=MEdarkturquoise, color=status), fill=NA, outlier.colour = NA, linewidth=0.2) + peri_figure + labs(title=&quot;darkturquoise module eigengene vs status&quot;) + scale_colour_manual(values=status_cols2) + theme(legend.position=&quot;none&quot;, axis.text.x=element_text(angle=90))</span></span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a><span class="co">#ggsave(filename=paste0(&quot;../WGCNA_results/all_brain/figure_&quot;,whichModule,&quot;_ME.pdf&quot;),plot = a, device=&quot;pdf&quot;, height=38, width=45, units=&quot;mm&quot;, bg=&quot;white&quot;)</span></span></code></pre></div>
<p>Let’s now plot the top hub gene. This gene is the one with the
highest module membership score. UBA52 has a module membership score
~0.9</p>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkturquoise module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
61/200
</td>
<td style="text-align:left;">
81/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
LOC113987193/LOC113989206/LOC113990217/LOC113998898/RPL10A/RPL11/RPL12/RPL13/RPL14/RPL15/RPL17/RPL18A/RPL19/RPL21/RPL22/RPL23/RPL23A/RPL27/RPL27A/RPL29/RPL30/RPL31/RPL32/RPL34/RPL35/RPL35A/RPL36/RPL37/RPL38/RPL39/RPL4/RPL5/RPL6/RPL7/RPL7A/RPL9/RPLP0/RPLP1/RPLP2/RPS10/RPS12/RPS13/RPS14/RPS15/RPS15A/RPS20/RPS23/RPS25/RPS26/RPS27A/RPS28/RPS29/RPS3/RPS3A/RPS4X/RPS5/RPS6/RPS7/RPS8/RPSA/UBA52
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019083" class="uri">GO:0019083</a>
</td>
<td style="text-align:left;">
viral transcription
</td>
<td style="text-align:left;">
62/200
</td>
<td style="text-align:left;">
100/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
GTF2B/LOC113987193/LOC113989206/LOC113990217/LOC113998898/RPL10A/RPL11/RPL12/RPL13/RPL14/RPL15/RPL17/RPL18A/RPL19/RPL21/RPL22/RPL23/RPL23A/RPL27/RPL27A/RPL29/RPL30/RPL31/RPL32/RPL34/RPL35/RPL35A/RPL36/RPL37/RPL38/RPL39/RPL4/RPL5/RPL6/RPL7/RPL7A/RPL9/RPLP0/RPLP1/RPLP2/RPS10/RPS12/RPS13/RPS14/RPS15/RPS15A/RPS20/RPS23/RPS25/RPS26/RPS27A/RPS28/RPS29/RPS3/RPS3A/RPS4X/RPS5/RPS6/RPS7/RPS8/RPSA/UBA52
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006413" class="uri">GO:0006413</a>
</td>
<td style="text-align:left;">
translational initiation
</td>
<td style="text-align:left;">
65/200
</td>
<td style="text-align:left;">
117/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
EIF1/EIF2B3/EIF3I/EIF3L/LOC113987193/LOC113989206/LOC113990217/LOC113998898/RPL10A/RPL11/RPL12/RPL13/RPL14/RPL15/RPL17/RPL18A/RPL19/RPL21/RPL22/RPL23/RPL23A/RPL27/RPL27A/RPL29/RPL30/RPL31/RPL32/RPL34/RPL35/RPL35A/RPL36/RPL37/RPL38/RPL39/RPL4/RPL5/RPL6/RPL7/RPL7A/RPL9/RPLP0/RPLP1/RPLP2/RPS10/RPS12/RPS13/RPS14/RPS15/RPS15A/RPS20/RPS23/RPS25/RPS26/RPS27A/RPS28/RPS29/RPS3/RPS3A/RPS4X/RPS5/RPS6/RPS7/RPS8/RPSA/UBA52
</td>
<td style="text-align:right;">
65
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000184" class="uri">GO:0000184</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
</td>
<td style="text-align:left;">
63/200
</td>
<td style="text-align:left;">
111/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
LOC113987193/LOC113988372/LOC113989206/LOC113990217/LOC113998898/RBM8A/RPL10A/RPL11/RPL12/RPL13/RPL14/RPL15/RPL17/RPL18A/RPL19/RPL21/RPL22/RPL23/RPL23A/RPL27/RPL27A/RPL29/RPL30/RPL31/RPL32/RPL34/RPL35/RPL35A/RPL36/RPL37/RPL38/RPL39/RPL4/RPL5/RPL6/RPL7/RPL7A/RPL9/RPLP0/RPLP1/RPLP2/RPS10/RPS12/RPS13/RPS14/RPS15/RPS15A/RPS20/RPS23/RPS25/RPS26/RPS27A/RPS28/RPS29/RPS3/RPS3A/RPS4X/RPS5/RPS6/RPS7/RPS8/RPSA/UBA52
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:left;">
63/200
</td>
<td style="text-align:left;">
148/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
LOC113987193/LOC113989206/LOC113990217/LOC113998898/MRPL18/MRPL43/MRPS14/RPL10A/RPL11/RPL12/RPL13/RPL14/RPL15/RPL17/RPL18A/RPL19/RPL21/RPL22/RPL23/RPL23A/RPL27/RPL27A/RPL29/RPL30/RPL31/RPL32/RPL34/RPL35/RPL35A/RPL36/RPL37/RPL38/RPL39/RPL4/RPL5/RPL6/RPL7/RPL7A/RPL9/RPLP0/RPLP1/RPLP2/RPS10/RPS12/RPS13/RPS14/RPS15/RPS15A/RPS20/RPS23/RPS25/RPS26/RPS27A/RPS28/RPS29/RPS3/RPS3A/RPS4X/RPS5/RPS6/RPS7/RPS8/RPSA
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002181" class="uri">GO:0002181</a>
</td>
<td style="text-align:left;">
cytoplasmic translation
</td>
<td style="text-align:left;">
27/200
</td>
<td style="text-align:left;">
33/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
LOC113990217/LOC113998898/RPL10A/RPL11/RPL15/RPL17/RPL18A/RPL19/RPL22/RPL26L1/RPL29/RPL30/RPL31/RPL32/RPL35A/RPL36/RPL38/RPL39/RPL6/RPL9/RPLP0/RPLP1/RPS23/RPS26/RPS28/RPS29/RPSA
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006364" class="uri">GO:0006364</a>
</td>
<td style="text-align:left;">
rRNA processing
</td>
<td style="text-align:left;">
19/200
</td>
<td style="text-align:left;">
135/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
DDX54/EXOSC1/IMP3/LOC113987193/LOC113989206/MRM2/RPL11/RPL14/RPL27/RPL35A/RPL5/RPL7/RPS14/RPS15/RPS25/RPS28/RPS6/RPS7/UTP3
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000028" class="uri">GO:0000028</a>
</td>
<td style="text-align:left;">
ribosomal small subunit assembly
</td>
<td style="text-align:left;">
7/200
</td>
<td style="text-align:left;">
14/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:left;">
LOC113987193/RPS10/RPS14/RPS15/RPS28/RPS5/RPSA
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042273" class="uri">GO:0042273</a>
</td>
<td style="text-align:left;">
ribosomal large subunit biogenesis
</td>
<td style="text-align:left;">
8/200
</td>
<td style="text-align:left;">
26/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6.0e-07
</td>
<td style="text-align:right;">
5.0e-07
</td>
<td style="text-align:left;">
NIP7/NPM1/RPL11/RPL14/RPL26L1/RPL35A/RPL5/RPL7
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042274" class="uri">GO:0042274</a>
</td>
<td style="text-align:left;">
ribosomal small subunit biogenesis
</td>
<td style="text-align:left;">
7/200
</td>
<td style="text-align:left;">
20/11462
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.7e-06
</td>
<td style="text-align:right;">
1.6e-06
</td>
<td style="text-align:left;">
LOC113989206/NPM1/RPS15/RPS25/RPS28/RPS6/RPS7
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/darkturquoise%20go-1.png" width="672" style="display: block; margin: auto;" /><img src="02_2_WGCNA_wholebrain_files/figure-html/darkturquoise%20go-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="paleturquoise" class="section level3" number="3.0.2">
<h3><span class="header-section-number">3.0.2</span> paleturquoise</h3>
<p>This module is potentially associated with status.</p>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEpaleturquoise
FALSE            Sum Sq  Df F value   Pr(&gt;F)   
FALSE Batch     0.04473   1  8.8557 0.003641 **
FALSE Tissue    0.09618   9  2.1159 0.034552 * 
FALSE Status    0.03123   1  6.1828 0.014507 * 
FALSE Residuals 0.52023 103                    
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEpaleturquoise
FALSE                Sum Sq Df F value   Pr(&gt;F)   
FALSE (Intercept)   0.02458  1  4.8320 0.030391 * 
FALSE Batch         0.03703  1  7.2789 0.008271 **
FALSE Tissue        0.08799  9  1.9216 0.058017 . 
FALSE Status        0.03334  1  6.5533 0.012064 * 
FALSE Tissue:Status 0.04198  9  0.9168 0.514275   
FALSE Residuals     0.47825 94                    
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEpaleturquoise ~ Batch + Tissue * Status, data = mes, 
FALSE     contrasts = list(Tissue = contr.sum, Status = contr.sum))
FALSE 
FALSE Residuals:
FALSE       Min        1Q    Median        3Q       Max 
FALSE -0.189689 -0.031995  0.003197  0.044560  0.120097 
FALSE 
FALSE Coefficients:
FALSE                  Estimate Std. Error t value Pr(&gt;|t|)   
FALSE (Intercept)     -0.033486   0.015234  -2.198  0.03039 * 
FALSE Batchrun2        0.071517   0.026508   2.698  0.00827 **
FALSE Tissue1         -0.001447   0.022708  -0.064  0.94934   
FALSE Tissue2          0.039602   0.024730   1.601  0.11265   
FALSE Tissue3          0.015011   0.023679   0.634  0.52766   
FALSE Tissue4          0.008037   0.024600   0.327  0.74463   
FALSE Tissue5         -0.043206   0.021116  -2.046  0.04354 * 
FALSE Tissue6         -0.021655   0.020388  -1.062  0.29090   
FALSE Tissue7         -0.027142   0.025136  -1.080  0.28299   
FALSE Tissue8         -0.041810   0.022277  -1.877  0.06364 . 
FALSE Tissue9          0.067761   0.023679   2.862  0.00519 **
FALSE Status1          0.017602   0.006876   2.560  0.01206 * 
FALSE Tissue1:Status1  0.007276   0.019937   0.365  0.71595   
FALSE Tissue2:Status1  0.036683   0.021143   1.735  0.08603 . 
FALSE Tissue3:Status1 -0.007150   0.019904  -0.359  0.72023   
FALSE Tissue4:Status1 -0.013484   0.020503  -0.658  0.51237   
FALSE Tissue5:Status1  0.009704   0.021718   0.447  0.65604   
FALSE Tissue6:Status1  0.002213   0.019879   0.111  0.91160   
FALSE Tissue7:Status1 -0.002773   0.021143  -0.131  0.89593   
FALSE Tissue8:Status1 -0.006396   0.020485  -0.312  0.75555   
FALSE Tissue9:Status1 -0.042125   0.019904  -2.116  0.03695 * 
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.07133 on 94 degrees of freedom
FALSE Multiple R-squared:  0.5218,  Adjusted R-squared:   0.42 
FALSE F-statistic: 5.128 on 20 and 94 DF,  p-value: 2.291e-08</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEpaleturquoise
FALSE                Sum Sq Df F value    Pr(&gt;F)    
FALSE (Intercept)   0.00120  1  0.2218  0.638718    
FALSE Tissue        0.36672  9  7.5122 3.134e-08 ***
FALSE Status        0.04638  1  8.5514  0.004317 ** 
FALSE Tissue:Status 0.04967  9  1.0176  0.431771    
FALSE Residuals     0.51528 95                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEpaleturquoise ~ Tissue + Status + Tissue:Status, 
FALSE     data = mes, contrasts = list(Tissue = contr.sum, Status = contr.sum))
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -0.18969 -0.03528  0.00240  0.04270  0.17118 
FALSE 
FALSE Coefficients:
FALSE                  Estimate Std. Error t value Pr(&gt;|t|)    
FALSE (Intercept)      0.003303   0.007012   0.471 0.638718    
FALSE Tissue1         -0.031084   0.020521  -1.515 0.133158    
FALSE Tissue2          0.074331   0.021802   3.409 0.000957 ***
FALSE Tissue3          0.049740   0.020521   2.424 0.017249 *  
FALSE Tissue4         -0.028752   0.021141  -1.360 0.177036    
FALSE Tissue5         -0.042959   0.021802  -1.970 0.051708 .  
FALSE Tissue6         -0.033923   0.020521  -1.653 0.101606    
FALSE Tissue7         -0.063931   0.021802  -2.932 0.004216 ** 
FALSE Tissue8         -0.065488   0.021141  -3.098 0.002565 ** 
FALSE Tissue9          0.102489   0.020521   4.994 2.68e-06 ***
FALSE Status1          0.020506   0.007012   2.924 0.004317 ** 
FALSE Tissue1:Status1  0.011525   0.020521   0.562 0.575700    
FALSE Tissue2:Status1  0.033779   0.021802   1.549 0.124626    
FALSE Tissue3:Status1 -0.010053   0.020521  -0.490 0.625328    
FALSE Tissue4:Status1 -0.016387   0.021141  -0.775 0.440177    
FALSE Tissue5:Status1  0.023403   0.021802   1.073 0.285810    
FALSE Tissue6:Status1  0.003396   0.020521   0.166 0.868898    
FALSE Tissue7:Status1 -0.005676   0.021802  -0.260 0.795153    
FALSE Tissue8:Status1 -0.008108   0.021141  -0.384 0.702203    
FALSE Tissue9:Status1 -0.045028   0.020521  -2.194 0.030654 *  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.07365 on 95 degrees of freedom
FALSE Multiple R-squared:  0.4847,  Adjusted R-squared:  0.3817 
FALSE F-statistic: 4.703 on 19 and 95 DF,  p-value: 1.814e-07</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEpaleturquoise
FALSE            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue    0.39150   9  8.0078 6.295e-09 ***
FALSE Status    0.04420   1  8.1367  0.005234 ** 
FALSE Residuals 0.56496 104                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/pale%20turquoise-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The top hub gene for this module is KCTD5 with a MM score of ~0.95
<img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in paleturquoise module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0036444" class="uri">GO:0036444</a>
</td>
<td style="text-align:left;">
calcium import into the mitochondrion
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
10/11462
</td>
<td style="text-align:right;">
0.0010059
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
MAIP1/MCUR1/MICU2/MICU3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043484" class="uri">GO:0043484</a>
</td>
<td style="text-align:left;">
regulation of RNA splicing
</td>
<td style="text-align:left;">
7/571
</td>
<td style="text-align:left;">
36/11462
</td>
<td style="text-align:right;">
0.0017331
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
CELF1/LOC114001870/MBNL2/PTBP2/RBM12/RBM12B/TRA2B
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030148" class="uri">GO:0030148</a>
</td>
<td style="text-align:left;">
sphingolipid biosynthetic process
</td>
<td style="text-align:left;">
8/571
</td>
<td style="text-align:left;">
48/11462
</td>
<td style="text-align:right;">
0.0023329
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
ACER3/ELOVL4/HACD2/KDSR/SGMS2/SGPP1/SPTLC1/VAPA
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006376" class="uri">GO:0006376</a>
</td>
<td style="text-align:left;">
mRNA splice site selection
</td>
<td style="text-align:left;">
5/571
</td>
<td style="text-align:left;">
20/11462
</td>
<td style="text-align:right;">
0.0025036
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
CELF1/LOC113985802/LOC113985815/PTBP2/SRSF1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044257" class="uri">GO:0044257</a>
</td>
<td style="text-align:left;">
cellular protein catabolic process
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
13/11462
</td>
<td style="text-align:right;">
0.0030385
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
CLN8/IDE/RAB12/RIPK1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043161" class="uri">GO:0043161</a>
</td>
<td style="text-align:left;">
proteasome-mediated ubiquitin-dependent protein catabolic process
</td>
<td style="text-align:left;">
16/571
</td>
<td style="text-align:left;">
149/11462
</td>
<td style="text-align:right;">
0.0030477
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
ATXN3/CLOCK/GID8/KCTD5/NEDD4/NHLRC1/NHLRC3/PPP2R5C/PSMD10/RAD23B/RNF4/SPOPL/TBL1XR1/UBE2D3/UBE2W/USP44
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006851" class="uri">GO:0006851</a>
</td>
<td style="text-align:left;">
mitochondrial calcium ion transmembrane transport
</td>
<td style="text-align:left;">
5/571
</td>
<td style="text-align:left;">
23/11462
</td>
<td style="text-align:right;">
0.0048020
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
LETM1/MAIP1/MCUR1/MICU2/MICU3
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031398" class="uri">GO:0031398</a>
</td>
<td style="text-align:left;">
positive regulation of protein ubiquitination
</td>
<td style="text-align:left;">
9/571
</td>
<td style="text-align:left;">
66/11462
</td>
<td style="text-align:right;">
0.0051892
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
BCL10/CHFR/DERL1/MAPK9/NDFIP2/NHLRC1/PSMD10/TRAF6/UBE2D1
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002756" class="uri">GO:0002756</a>
</td>
<td style="text-align:left;">
MyD88-independent toll-like receptor signaling pathway
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
15/11462
</td>
<td style="text-align:right;">
0.0053578
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
RIPK1/TLR3/UBE2D1/UBE2D3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051560" class="uri">GO:0051560</a>
</td>
<td style="text-align:left;">
mitochondrial calcium ion homeostasis
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
16/11462
</td>
<td style="text-align:right;">
0.0068663
</td>
<td style="text-align:right;">
0.6093961
</td>
<td style="text-align:right;">
0.6025927
</td>
<td style="text-align:left;">
LETM1/MAIP1/MICU2/MICU3
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="honeydew1-module" class="section level3" number="3.0.3">
<h3><span class="header-section-number">3.0.3</span> honeydew1
module</h3>
<p>This module is potentially associated with social network strength.
Don’t need batch here because it wasn’t associated with this module.</p>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEhoneydew1
FALSE                            Sum Sq  Df F value  Pr(&gt;F)  
FALSE Year                      0.00588   1  0.7612 0.38498  
FALSE Tissue                    0.11252   9  1.6174 0.11989  
FALSE scale(strength.all_study) 0.02507   1  3.2428 0.07466 .
FALSE Residuals                 0.79620 103                  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEhoneydew1
FALSE                                   Sum Sq Df F value Pr(&gt;F)
FALSE (Intercept)                      0.00364  1  0.4316 0.5128
FALSE Year                             0.00608  1  0.7208 0.3980
FALSE Tissue                           0.10946  9  1.4417 0.1817
FALSE scale(strength.all_study)        0.00511  1  0.6055 0.4384
FALSE Tissue:scale(strength.all_study) 0.00319  9  0.0420 1.0000
FALSE Residuals                        0.79301 94</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEhoneydew1 ~ Year + Tissue * scale(strength.all_study), 
FALSE     data = mes)
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -0.21613 -0.02369  0.00467  0.04700  0.15303 
FALSE 
FALSE Coefficients:
FALSE                                        Estimate Std. Error t value Pr(&gt;|t|)  
FALSE (Intercept)                           0.0187459  0.0285347   0.657   0.5128  
FALSE Year2018                             -0.0212289  0.0250040  -0.849   0.3980  
FALSE TissueAH                             -0.0242310  0.0384095  -0.631   0.5297  
FALSE TissuePVN                            -0.0206393  0.0375009  -0.550   0.5834  
FALSE TissuePOM                            -0.0101864  0.0384304  -0.265   0.7915  
FALSE TissueICO                            -0.0371755  0.0384430  -0.967   0.3360  
FALSE TissueGCT                            -0.0724391  0.0375009  -1.932   0.0564 .
FALSE TissueAI                              0.0500201  0.0390097   1.282   0.2029  
FALSE TissueTNA                             0.0110174  0.0383814   0.287   0.7747  
FALSE TissueLS                              0.0154758  0.0375009   0.413   0.6808  
FALSE TissueBSTm                           -0.0077801  0.0375009  -0.207   0.8361  
FALSE scale(strength.all_study)             0.0215393  0.0276810   0.778   0.4384  
FALSE TissueAH:scale(strength.all_study)   -0.0041436  0.0374795  -0.111   0.9122  
FALSE TissuePVN:scale(strength.all_study)   0.0033188  0.0371918   0.089   0.9291  
FALSE TissuePOM:scale(strength.all_study)  -0.0125190  0.0384790  -0.325   0.7456  
FALSE TissueICO:scale(strength.all_study)   0.0052359  0.0379768   0.138   0.8906  
FALSE TissueGCT:scale(strength.all_study)  -0.0030444  0.0371918  -0.082   0.9349  
FALSE TissueAI:scale(strength.all_study)   -0.0039698  0.0448166  -0.089   0.9296  
FALSE TissueTNA:scale(strength.all_study)  -0.0008582  0.0372995  -0.023   0.9817  
FALSE TissueLS:scale(strength.all_study)    0.0023813  0.0371918   0.064   0.9491  
FALSE TissueBSTm:scale(strength.all_study)  0.0069860  0.0371918   0.188   0.8514  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.09185 on 94 degrees of freedom
FALSE Multiple R-squared:  0.207,   Adjusted R-squared:  0.03827 
FALSE F-statistic: 1.227 on 20 and 94 DF,  p-value: 0.2508</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEhoneydew1
FALSE                                   Sum Sq Df F value Pr(&gt;F)
FALSE (Intercept)                      0.00115  1  0.1369 0.7122
FALSE Tissue                           0.11078  9  1.4633 0.1729
FALSE scale(strength.all_study)        0.01017  1  1.2090 0.2743
FALSE Tissue:scale(strength.all_study) 0.00299  9  0.0396 1.0000
FALSE Residuals                        0.79909 95</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEhoneydew1 ~ Tissue + scale(strength.all_study) + 
FALSE     Tissue:scale(strength.all_study), data = mes)
FALSE 
FALSE Residuals:
FALSE       Min        1Q    Median        3Q       Max 
FALSE -0.209145 -0.030023  0.008395  0.040887  0.165932 
FALSE 
FALSE Coefficients:
FALSE                                        Estimate Std. Error t value Pr(&gt;|t|)  
FALSE (Intercept)                           0.0097981  0.0264781   0.370    0.712  
FALSE TissueAH                             -0.0254616  0.0383257  -0.664    0.508  
FALSE TissuePVN                            -0.0206393  0.0374457  -0.551    0.583  
FALSE TissuePOM                            -0.0102098  0.0383739  -0.266    0.791  
FALSE TissueICO                            -0.0366235  0.0383809  -0.954    0.342  
FALSE TissueGCT                            -0.0724391  0.0374457  -1.935    0.056 .
FALSE TissueAI                              0.0513413  0.0389213   1.319    0.190  
FALSE TissueTNA                             0.0099604  0.0383048   0.260    0.795  
FALSE TissueLS                              0.0154758  0.0374457   0.413    0.680  
FALSE TissueBSTm                           -0.0077801  0.0374457  -0.208    0.836  
FALSE scale(strength.all_study)             0.0288735  0.0262599   1.100    0.274  
FALSE TissueAH:scale(strength.all_study)   -0.0034568  0.0374157  -0.092    0.927  
FALSE TissuePVN:scale(strength.all_study)   0.0033188  0.0371371   0.089    0.929  
FALSE TissuePOM:scale(strength.all_study)  -0.0125464  0.0384224  -0.327    0.745  
FALSE TissueICO:scale(strength.all_study)   0.0047346  0.0379164   0.125    0.901  
FALSE TissueGCT:scale(strength.all_study)  -0.0030444  0.0371371  -0.082    0.935  
FALSE TissueAI:scale(strength.all_study)   -0.0008949  0.0446044  -0.020    0.984  
FALSE TissueTNA:scale(strength.all_study)  -0.0004918  0.0372422  -0.013    0.989  
FALSE TissueLS:scale(strength.all_study)    0.0023813  0.0371371   0.064    0.949  
FALSE TissueBSTm:scale(strength.all_study)  0.0069860  0.0371371   0.188    0.851  
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.09171 on 95 degrees of freedom
FALSE Multiple R-squared:  0.2009,  Adjusted R-squared:  0.04109 
FALSE F-statistic: 1.257 on 19 and 95 DF,  p-value: 0.231</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEhoneydew1
FALSE                            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue                    0.11321   9   1.631 0.1159238    
FALSE scale(strength.all_study) 0.09354   1  12.128 0.0007278 ***
FALSE Residuals                 0.80209 104                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/honeydew1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Not significant after controlling for batch, tissue and indivdual as
a random effect.</p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in honeydew1 module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0032922" class="uri">GO:0032922</a>
</td>
<td style="text-align:left;">
circadian regulation of gene expression
</td>
<td style="text-align:left;">
4/42
</td>
<td style="text-align:left;">
68/11462
</td>
<td style="text-align:right;">
0.0001070
</td>
<td style="text-align:right;">
0.0168006
</td>
<td style="text-align:right;">
0.0117148
</td>
<td style="text-align:left;">
CIART/NR1D1/PER2/PER3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005978" class="uri">GO:0005978</a>
</td>
<td style="text-align:left;">
glycogen biosynthetic process
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
14/11462
</td>
<td style="text-align:right;">
0.0011600
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
NR1D1/PER2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007623" class="uri">GO:0007623</a>
</td>
<td style="text-align:left;">
circadian rhythm
</td>
<td style="text-align:left;">
3/42
</td>
<td style="text-align:left;">
59/11462
</td>
<td style="text-align:right;">
0.0012896
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
LOC114003929/NR1D1/PER2
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042752" class="uri">GO:0042752</a>
</td>
<td style="text-align:left;">
regulation of circadian rhythm
</td>
<td style="text-align:left;">
3/42
</td>
<td style="text-align:left;">
61/11462
</td>
<td style="text-align:right;">
0.0014204
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
NR1D1/NR1D2/PER2
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048024" class="uri">GO:0048024</a>
</td>
<td style="text-align:left;">
regulation of mRNA splicing, via spliceosome
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
18/11462
</td>
<td style="text-align:right;">
0.0019323
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
KHDRBS1/SRSF3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006090" class="uri">GO:0006090</a>
</td>
<td style="text-align:left;">
pyruvate metabolic process
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
19/11462
</td>
<td style="text-align:right;">
0.0021547
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
LOC113983410/LOC113990030
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043153" class="uri">GO:0043153</a>
</td>
<td style="text-align:left;">
entrainment of circadian clock by photoperiod
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
20/11462
</td>
<td style="text-align:right;">
0.0023885
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
PER2/PER3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008652" class="uri">GO:0008652</a>
</td>
<td style="text-align:left;">
cellular amino acid biosynthetic process
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
21/11462
</td>
<td style="text-align:right;">
0.0026338
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
KYAT1/PYCR1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006783" class="uri">GO:0006783</a>
</td>
<td style="text-align:left;">
heme biosynthetic process
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
22/11462
</td>
<td style="text-align:right;">
0.0028905
</td>
<td style="text-align:right;">
0.0504229
</td>
<td style="text-align:right;">
0.0351591
</td>
<td style="text-align:left;">
LOC113982573/UROS
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009060" class="uri">GO:0009060</a>
</td>
<td style="text-align:left;">
aerobic respiration
</td>
<td style="text-align:left;">
2/42
</td>
<td style="text-align:left;">
25/11462
</td>
<td style="text-align:right;">
0.0037279
</td>
<td style="text-align:right;">
0.0532067
</td>
<td style="text-align:right;">
0.0371002
</td>
<td style="text-align:left;">
LOC113983410/LOC113990030
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
</div>
<div id="darkorange2" class="section level3" number="3.0.4">
<h3><span class="header-section-number">3.0.4</span> darkorange2</h3>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEdarkorange2
FALSE                            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Year                      0.00975   1  1.5311 0.2187535    
FALSE Tissue                    0.20752   9  3.6223 0.0005867 ***
FALSE scale(strength.all_study) 0.03385   1  5.3184 0.0231039 *  
FALSE Residuals                 0.65564 103                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE Anova Table (Type III tests)
FALSE 
FALSE Response: MEdarkorange2
FALSE                                   Sum Sq Df F value   Pr(&gt;F)   
FALSE (Intercept)                      0.00443  1  0.6449 0.423976   
FALSE Year                             0.00941  1  1.3688 0.244978   
FALSE Tissue                           0.20742  9  3.3531 0.001339 **
FALSE scale(strength.all_study)        0.01073  1  1.5606 0.214679   
FALSE Tissue:scale(strength.all_study) 0.00954  9  0.1542 0.997645   
FALSE Residuals                        0.64610 94                    
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEdarkorange2 ~ Year + Tissue * scale(strength.all_study), 
FALSE     data = mes)
FALSE 
FALSE Residuals:
FALSE       Min        1Q    Median        3Q       Max 
FALSE -0.144604 -0.053316 -0.000851  0.047907  0.241264 
FALSE 
FALSE Coefficients:
FALSE                                        Estimate Std. Error t value Pr(&gt;|t|)   
FALSE (Intercept)                           0.0206833  0.0257562   0.803   0.4240   
FALSE Year2018                              0.0264051  0.0225694   1.170   0.2450   
FALSE TissueAH                             -0.0516874  0.0346696  -1.491   0.1393   
FALSE TissuePVN                             0.0172869  0.0338494   0.511   0.6108   
FALSE TissuePOM                             0.0098605  0.0346884   0.284   0.7768   
FALSE TissueICO                             0.0042670  0.0346998   0.123   0.9024   
FALSE TissueGCT                             0.0094744  0.0338494   0.280   0.7802   
FALSE TissueAI                             -0.0876757  0.0352113  -2.490   0.0145 * 
FALSE TissueTNA                            -0.0465449  0.0346442  -1.344   0.1823   
FALSE TissueLS                             -0.0929944  0.0338494  -2.747   0.0072 **
FALSE TissueBSTm                           -0.0809117  0.0338494  -2.390   0.0188 * 
FALSE scale(strength.all_study)            -0.0312132  0.0249857  -1.249   0.2147   
FALSE TissueAH:scale(strength.all_study)    0.0067419  0.0338302   0.199   0.8425   
FALSE TissuePVN:scale(strength.all_study)  -0.0060643  0.0335705  -0.181   0.8570   
FALSE TissuePOM:scale(strength.all_study)   0.0236760  0.0347323   0.682   0.4971   
FALSE TissueICO:scale(strength.all_study)   0.0087445  0.0342790   0.255   0.7992   
FALSE TissueGCT:scale(strength.all_study)  -0.0002642  0.0335705  -0.008   0.9937   
FALSE TissueAI:scale(strength.all_study)    0.0045006  0.0404529   0.111   0.9117   
FALSE TissueTNA:scale(strength.all_study)  -0.0035565  0.0336677  -0.106   0.9161   
FALSE TissueLS:scale(strength.all_study)    0.0180131  0.0335705   0.537   0.5928   
FALSE TissueBSTm:scale(strength.all_study)  0.0127069  0.0335705   0.379   0.7059   
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.08291 on 94 degrees of freedom
FALSE Multiple R-squared:  0.3539,  Adjusted R-squared:  0.2164 
FALSE F-statistic: 2.574 on 20 and 94 DF,  p-value: 0.001183</code></pre>
<pre><code>FALSE 
FALSE Call:
FALSE lm(formula = MEdarkorange2 ~ Tissue + scale(strength.all_study) + 
FALSE     Tissue:scale(strength.all_study), data = mes)
FALSE 
FALSE Residuals:
FALSE      Min       1Q   Median       3Q      Max 
FALSE -0.14566 -0.04763  0.00457  0.04900  0.23445 
FALSE 
FALSE Coefficients:
FALSE                                        Estimate Std. Error t value Pr(&gt;|t|)   
FALSE (Intercept)                           0.0318128  0.0239816   1.327   0.1878   
FALSE TissueAH                             -0.0501567  0.0347121  -1.445   0.1518   
FALSE TissuePVN                             0.0172869  0.0339151   0.510   0.6114   
FALSE TissuePOM                             0.0098895  0.0347557   0.285   0.7766   
FALSE TissueICO                             0.0035804  0.0347621   0.103   0.9182   
FALSE TissueGCT                             0.0094744  0.0339151   0.279   0.7806   
FALSE TissueAI                             -0.0893190  0.0352515  -2.534   0.0129 * 
FALSE TissueTNA                            -0.0452301  0.0346931  -1.304   0.1955   
FALSE TissueLS                             -0.0929944  0.0339151  -2.742   0.0073 **
FALSE TissueBSTm                           -0.0809117  0.0339151  -2.386   0.0190 * 
FALSE scale(strength.all_study)            -0.0403356  0.0237839  -1.696   0.0932 . 
FALSE TissueAH:scale(strength.all_study)    0.0058876  0.0338879   0.174   0.8624   
FALSE TissuePVN:scale(strength.all_study)  -0.0060643  0.0336355  -0.180   0.8573   
FALSE TissuePOM:scale(strength.all_study)   0.0237101  0.0347996   0.681   0.4973   
FALSE TissueICO:scale(strength.all_study)   0.0093680  0.0343414   0.273   0.7856   
FALSE TissueGCT:scale(strength.all_study)  -0.0002642  0.0336355  -0.008   0.9937   
FALSE TissueAI:scale(strength.all_study)    0.0006761  0.0403987   0.017   0.9867   
FALSE TissueTNA:scale(strength.all_study)  -0.0040121  0.0337307  -0.119   0.9056   
FALSE TissueLS:scale(strength.all_study)    0.0180131  0.0336355   0.536   0.5935   
FALSE TissueBSTm:scale(strength.all_study)  0.0127069  0.0336355   0.378   0.7064   
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
FALSE 
FALSE Residual standard error: 0.08307 on 95 degrees of freedom
FALSE Multiple R-squared:  0.3445,  Adjusted R-squared:  0.2134 
FALSE F-statistic: 2.628 on 19 and 95 DF,  p-value: 0.001098</code></pre>
<pre><code>FALSE Anova Table (Type II tests)
FALSE 
FALSE Response: MEdarkorange2
FALSE                            Sum Sq  Df F value    Pr(&gt;F)    
FALSE Tissue                    0.20717   9  3.5979 0.0006209 ***
FALSE scale(strength.all_study) 0.13342   1 20.8531 1.366e-05 ***
FALSE Residuals                 0.66538 104                      
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre><code>FALSE # Effect Size for ANOVA (Type II)
FALSE 
FALSE Parameter                 | Eta2 (partial) |       95% CI
FALSE ---------------------------------------------------------
FALSE Year                      |           0.01 | [0.00, 1.00]
FALSE Tissue                    |           0.24 | [0.08, 1.00]
FALSE scale(strength.all_study) |           0.05 | [0.00, 1.00]
FALSE 
FALSE - One-sided CIs: upper bound fixed at [1.00].</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkorange2 module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006457" class="uri">GO:0006457</a>
</td>
<td style="text-align:left;">
protein folding
</td>
<td style="text-align:left;">
14/74
</td>
<td style="text-align:left;">
129/11462
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:left;">
AHSA1/CALR/CCT3/CCT5/CCT6A/DNAJB6/HSP90AA1/HSP90B1/HSPA8/LOC113994013/PDIA4/PDIA6/PTGES3/TCP1
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904851" class="uri">GO:1904851</a>
</td>
<td style="text-align:left;">
positive regulation of establishment of protein localization to telomere
</td>
<td style="text-align:left;">
6/74
</td>
<td style="text-align:left;">
11/11462
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:left;">
CCT3/CCT5/CCT6A/LOC113994013/TCP1/WRAP53
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904871" class="uri">GO:1904871</a>
</td>
<td style="text-align:left;">
positive regulation of protein localization to Cajal body
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
12/11462
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
9.00e-07
</td>
<td style="text-align:right;">
8.00e-07
</td>
<td style="text-align:left;">
CCT3/CCT5/CCT6A/LOC113994013/TCP1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042026" class="uri">GO:0042026</a>
</td>
<td style="text-align:left;">
protein refolding
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
16/11462
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
3.00e-06
</td>
<td style="text-align:right;">
2.70e-06
</td>
<td style="text-align:left;">
DNAJA4/HSP90AA1/HSPA2/HSPA5/HSPA8
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904874" class="uri">GO:1904874</a>
</td>
<td style="text-align:left;">
positive regulation of telomerase RNA localization to Cajal body
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
16/11462
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
3.00e-06
</td>
<td style="text-align:right;">
2.70e-06
</td>
<td style="text-align:left;">
CCT3/CCT5/CCT6A/LOC113994013/TCP1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006986" class="uri">GO:0006986</a>
</td>
<td style="text-align:left;">
response to unfolded protein
</td>
<td style="text-align:left;">
6/74
</td>
<td style="text-align:left;">
33/11462
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:right;">
3.50e-06
</td>
<td style="text-align:right;">
3.20e-06
</td>
<td style="text-align:left;">
HERPUD1/HSP90AA1/HSPA2/HSPA4/HSPA8/MANF
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051973" class="uri">GO:0051973</a>
</td>
<td style="text-align:left;">
positive regulation of telomerase activity
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
29/11462
</td>
<td style="text-align:right;">
1.0e-06
</td>
<td style="text-align:right;">
4.56e-05
</td>
<td style="text-align:right;">
4.13e-05
</td>
<td style="text-align:left;">
HSP90AA1/LOC113994013/PTGES3/TCP1/WRAP53
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007339" class="uri">GO:0007339</a>
</td>
<td style="text-align:left;">
binding of sperm to zona pellucida
</td>
<td style="text-align:left;">
4/74
</td>
<td style="text-align:left;">
13/11462
</td>
<td style="text-align:right;">
1.1e-06
</td>
<td style="text-align:right;">
4.56e-05
</td>
<td style="text-align:right;">
4.13e-05
</td>
<td style="text-align:left;">
CCT3/CCT5/LOC113994013/TCP1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032212" class="uri">GO:0032212</a>
</td>
<td style="text-align:left;">
positive regulation of telomere maintenance via telomerase
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
30/11462
</td>
<td style="text-align:right;">
1.2e-06
</td>
<td style="text-align:right;">
4.56e-05
</td>
<td style="text-align:right;">
4.13e-05
</td>
<td style="text-align:left;">
CCT3/CCT5/CCT6A/LOC113994013/TCP1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1901998" class="uri">GO:1901998</a>
</td>
<td style="text-align:left;">
toxin transport
</td>
<td style="text-align:left;">
5/74
</td>
<td style="text-align:left;">
30/11462
</td>
<td style="text-align:right;">
1.2e-06
</td>
<td style="text-align:right;">
4.56e-05
</td>
<td style="text-align:right;">
4.13e-05
</td>
<td style="text-align:left;">
CCT3/CCT5/HSPA5/LOC113994013/TCP1
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/darkorange2%20go-1.png" width="672" style="display: block; margin: auto;" /><img src="02_2_WGCNA_wholebrain_files/figure-html/darkorange2%20go-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="navajowhite2" class="section level3" number="3.0.5">
<h3><span class="header-section-number">3.0.5</span> navajowhite2</h3>
<p>This module is interesting because it’s associated with many
sex-steroid related genes in the hypothalamic regions.</p>
<pre><code>FALSE Analysis of Variance Table
FALSE 
FALSE Response: MEnavajowhite2
FALSE               Df  Sum Sq  Mean Sq  F value    Pr(&gt;F)    
FALSE Batch          1 0.00247 0.002474   8.8626  0.003701 ** 
FALSE Tissue         9 0.97041 0.107823 386.2757 &lt; 2.2e-16 ***
FALSE Status         1 0.00022 0.000217   0.7783  0.379924    
FALSE Tissue:Status  9 0.00066 0.000074   0.2637  0.982703    
FALSE Residuals     94 0.02624 0.000279                       
FALSE ---
FALSE Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This gene ontology analysis compared the genes in the navajowhite
module against all other genes in the dataset.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in navajowhite2 module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007218" class="uri">GO:0007218</a>
</td>
<td style="text-align:left;">
neuropeptide signaling pathway
</td>
<td style="text-align:left;">
27/676
</td>
<td style="text-align:left;">
83/11462
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
CALCB/GAL/GALR3/GPR139/GPR83/HCRT/LOC113987733/LOC113995733/LOC114000503/MCHR1/NMS/NMU/NPBWR2/NPVF/NPW/NTSR1/OPRK1/OPRL1/PDYN/PENK/PNOC/POMC/PROK2/QRFP/RXFP3/SSTR5/TENM1
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
68/676
</td>
<td style="text-align:left;">
437/11462
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
ADCY7/ADGRD1/ADGRG2/ADRA2A/AGTR1/ANXA1/CALCB/CASR/CMKLR1/CXCR4/FFAR4/FZD5/GAL/GALR3/GCGR/GHRH/GHSR/GLP1R/GNRH1/GPR135/GPR139/GPR176/GPR75/GPR83/HCRT/HEBP1/IAPP/LGR6/LOC113982931/LOC113987733/LOC113987774/LOC113989201/LOC113991146/LOC113993563/LOC113995733/LOC113996140/LOC114000503/LOC114001074/LOC114004322/MC3R/MCHR1/NMS/NMU/NPBWR2/NPR1/NPW/NTSR1/OBSCN/OPRK1/OPRL1/PDE3B/PDE7B/PDE8B/PDYN/PENK/PNOC/POMC/PRLH/PROK2/QRFP/RGS2/RORB/RXFP3/S1PR3/SSTR5/TRH/TRHR/TSHR
</td>
<td style="text-align:right;">
68
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007631" class="uri">GO:0007631</a>
</td>
<td style="text-align:left;">
feeding behavior
</td>
<td style="text-align:left;">
10/676
</td>
<td style="text-align:left;">
21/11462
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0000494
</td>
<td style="text-align:right;">
0.0000474
</td>
<td style="text-align:left;">
CALCB/GAL/GALR3/LOC113987774/LOC114000503/MCHR1/MRAP2/NEGR1/NPW/PRLH
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007200" class="uri">GO:0007200</a>
</td>
<td style="text-align:left;">
phospholipase C-activating G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
12/676
</td>
<td style="text-align:left;">
45/11462
</td>
<td style="text-align:right;">
0.0000076
</td>
<td style="text-align:right;">
0.0030374
</td>
<td style="text-align:right;">
0.0029185
</td>
<td style="text-align:left;">
AGTR1/CASR/CMKLR1/ESR1/FFAR4/GPR139/HCRT/LOC113991146/LOC114001074/MC3R/OPRK1/TRHR
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048704" class="uri">GO:0048704</a>
</td>
<td style="text-align:left;">
embryonic skeletal system morphogenesis
</td>
<td style="text-align:left;">
6/676
</td>
<td style="text-align:left;">
11/11462
</td>
<td style="text-align:right;">
0.0000147
</td>
<td style="text-align:right;">
0.0046962
</td>
<td style="text-align:right;">
0.0045123
</td>
<td style="text-align:left;">
LOC113998261/LOC114001942/PCGF2/PDGFRA/SATB2/SOX11
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007204" class="uri">GO:0007204</a>
</td>
<td style="text-align:left;">
positive regulation of cytosolic calcium ion concentration
</td>
<td style="text-align:left;">
16/676
</td>
<td style="text-align:left;">
86/11462
</td>
<td style="text-align:right;">
0.0000360
</td>
<td style="text-align:right;">
0.0095517
</td>
<td style="text-align:right;">
0.0091776
</td>
<td style="text-align:left;">
AGTR1/CALCB/CD36/CMKLR1/CXCR4/ESR1/FFAR4/GLP1R/HCRT/IAPP/LOC113991146/MCHR1/NMU/PDGFRA/PROK2/S1PR3
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006112" class="uri">GO:0006112</a>
</td>
<td style="text-align:left;">
energy reserve metabolic process
</td>
<td style="text-align:left;">
6/676
</td>
<td style="text-align:left;">
13/11462
</td>
<td style="text-align:right;">
0.0000494
</td>
<td style="text-align:right;">
0.0112482
</td>
<td style="text-align:right;">
0.0108077
</td>
<td style="text-align:left;">
LOC113985008/LOC113987774/LOC113999007/MRAP2/MYC/PRLH
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007189" class="uri">GO:0007189</a>
</td>
<td style="text-align:left;">
adenylate cyclase-activating G protein-coupled receptor signaling
pathway
</td>
<td style="text-align:left;">
14/676
</td>
<td style="text-align:left;">
76/11462
</td>
<td style="text-align:right;">
0.0001223
</td>
<td style="text-align:right;">
0.0243606
</td>
<td style="text-align:right;">
0.0234066
</td>
<td style="text-align:left;">
ADCY7/ADGRD1/ADGRG2/ADRA2A/CALCB/GCGR/GHRH/GLP1R/IAPP/LOC113991146/LOC113997521/MC3R/S1PR3/TSHR
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
5/676
</td>
<td style="text-align:left;">
10/11462
</td>
<td style="text-align:right;">
0.0001382
</td>
<td style="text-align:right;">
0.0244725
</td>
<td style="text-align:right;">
0.0235141
</td>
<td style="text-align:left;">
AR/ESR1/ESR2/NR3C1/PGR
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060412" class="uri">GO:0060412</a>
</td>
<td style="text-align:left;">
ventricular septum morphogenesis
</td>
<td style="text-align:left;">
9/676
</td>
<td style="text-align:left;">
36/11462
</td>
<td style="text-align:right;">
0.0001825
</td>
<td style="text-align:right;">
0.0290867
</td>
<td style="text-align:right;">
0.0279477
</td>
<td style="text-align:left;">
CITED2/LOC113994594/LOC113997556/LOC114001848/PROX1/SLIT2/SMAD7/SOX11/TGFB2
</td>
<td style="text-align:right;">
9
</td>
</tr>
</tbody>
</table>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/navajowhite-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="go-for-modules" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> GO for modules</h2>
</div>
</div>
<div id="module-connectivity" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Module
Connectivity</h1>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1   0.0126 33.80          0.961 6810.00   6810.00   6880
FALSE 2      2   0.1560 -9.21          0.878 3730.00   3730.00   4140
FALSE 3      3   0.1170 -2.41          0.898 2200.00   2180.00   2770
FALSE 4      4   0.2180 -1.99          0.934 1370.00   1350.00   2020
FALSE 5      5   0.3350 -1.86          0.951  896.00    872.00   1550
FALSE 6      6   0.4460 -1.77          0.968  609.00    582.00   1230
FALSE 7      7   0.5430 -1.77          0.974  427.00    399.00   1010
FALSE 8      8   0.6150 -1.77          0.976  309.00    280.00    841
FALSE 9      9   0.6690 -1.79          0.980  228.00    200.00    714
FALSE 10    10   0.7100 -1.83          0.976  173.00    146.00    614
FALSE 11    11   0.7480 -1.85          0.980  133.00    108.00    534
FALSE 12    12   0.7800 -1.86          0.984  104.00     80.70    469
FALSE 13    14   0.8190 -1.88          0.987   66.40     46.60    370
FALSE 14    16   0.8410 -1.91          0.987   44.40     28.00    299
FALSE 15    18   0.8530 -1.93          0.988   30.80     17.30    247
FALSE 16    20   0.8600 -1.95          0.987   22.10     11.00    207
FALSE 17    22   0.8720 -1.93          0.988   16.30      7.15    176
FALSE 18    24   0.8810 -1.92          0.991   12.20      4.73    151
FALSE 19    26   0.8830 -1.91          0.990    9.39      3.19    131
FALSE 20    28   0.8860 -1.91          0.989    7.32      2.19    115
FALSE 21    30   0.8810 -1.91          0.988    5.80      1.52    101</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1  0.04720 49.600          0.944 6810.00   6810.00   6880
FALSE 2      2  0.02060 -2.970          0.882 3760.00   3750.00   4170
FALSE 3      3  0.00267 -0.382          0.867 2240.00   2220.00   2830
FALSE 4      4  0.04100 -0.885          0.901 1410.00   1390.00   2080
FALSE 5      5  0.13700 -1.150          0.936  936.00    913.00   1620
FALSE 6      6  0.28700 -1.350          0.964  644.00    619.00   1300
FALSE 7      7  0.43900 -1.470          0.980  458.00    432.00   1070
FALSE 8      8  0.56100 -1.560          0.989  335.00    308.00    894
FALSE 9      9  0.64400 -1.640          0.987  251.00    224.00    761
FALSE 10    10  0.70900 -1.710          0.987  192.00    166.00    657
FALSE 11    11  0.75100 -1.750          0.985  149.00    125.00    572
FALSE 12    12  0.78300 -1.780          0.985  118.00     94.80    503
FALSE 13    14  0.83700 -1.820          0.994   76.40     56.60    400
FALSE 14    16  0.86200 -1.840          0.993   51.90     35.30    325
FALSE 15    18  0.87800 -1.870          0.992   36.50     22.60    269
FALSE 16    20  0.87300 -1.910          0.984   26.50     14.70    227
FALSE 17    22  0.88300 -1.910          0.986   19.70      9.83    193
FALSE 18    24  0.89400 -1.900          0.988   15.00      6.71    166
FALSE 19    26  0.90100 -1.890          0.990   11.60      4.65    145
FALSE 20    28  0.90800 -1.880          0.991    9.11      3.26    127
FALSE 21    30  0.90600 -1.870          0.987    7.27      2.32    112</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>How does module size influence connectivity?</p>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="module-preservation" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Module
Preservation</h1>
<p><img src="02_2_WGCNA_wholebrain_files/figure-html/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
