<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive Change Dec 2023" />


<title>WGCNA Hypothalamic Pituitary Gonadal Axis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Supp 1: Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_1_Checks_PCA.html">Supp 1 &amp; 2: Checks and System-wide PCA</a>
    </li>
    <li>
      <a href="./02_2_WGCNA_wholebrain.html">Supp 2: Whole-brain WGCNA</a>
    </li>
    <li>
      <a href="./02_3_WGCNA_HPG.html">Supp 2: HPG WGCNA</a>
    </li>
    <li>
      <a href="./03_DESeq2.html">Supp 3: Per Tissue Differential Expression</a>
    </li>
    <li>
      <a href="./04_candidate_gene_analysis.html">Supp 4: Candidate Gene Analysis</a>
    </li>
    <li>
      <a href="./05_WGCNA_single_tissue.html">Supp 5: Within Tissue WGCNA </a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Supp 6: Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WGCNA Hypothalamic Pituitary Gonadal
Axis</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male
cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive Change Dec 2023</h4>
<h4 class="date">Last Knit “2024-01-04”</h4>

</div>


<pre><code>FALSE Allowing parallel execution with up to 7 working processes.</code></pre>
<div id="subsetting-filtering" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Subsetting &amp;
Filtering</h1>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="connectivity-test-for-outliers" class="section level1"
number="2">
<h1><span class="header-section-number">2</span> Connectivity Test for
Outliers</h1>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>No outliers</p>
</div>
<div id="effect-of-batch-correction" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Effect of Batch
Correction</h1>
<p><img src="02_3_WGCNA_HPG_files/figure-html/batch%20correction-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/batch%20correction-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Again this seems to make the differences between batches even
larger…</p>
</div>
<div id="soft-threshold" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Soft Threshold</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co">#write.csv(datExpr0, file=&quot;../WGCNA_results/HPG/HPG_vsd.csv&quot;)</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>gsg <span class="ot">=</span> <span class="fu">goodSamplesGenes</span>(datExpr0, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>TRUE  Flagging genes and samples with too many missing values...
TRUE   ..step 1</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code></pre></div>
<pre><code>TRUE [1] TRUE</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>powers<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">11</span>, <span class="at">by =</span> <span class="dv">1</span>), <span class="fu">seq</span>(<span class="dv">12</span>, <span class="dv">30</span>, <span class="at">by =</span> <span class="dv">2</span>))</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>sft<span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr0, <span class="at">powerVector=</span>powers, <span class="at">verbose=</span><span class="dv">0</span>, <span class="at">networkType=</span><span class="st">&quot;signed&quot;</span>)</span></code></pre></div>
<pre><code>TRUE    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
TRUE 1      1 1.68e-05 -0.0505          0.697    7530      7510   7950
TRUE 2      2 9.41e-02  1.1500          0.115    4950      4890   6030
TRUE 3      3 3.81e-01  1.1000          0.597    3680      3650   5090
TRUE 4      4 4.29e-01  0.6830          0.681    2920      2880   4520
TRUE 5      5 2.85e-01  0.3660          0.574    2410      2350   4110
TRUE 6      6 8.00e-02  0.1400          0.402    2050      1950   3800
TRUE 7      7 4.58e-03 -0.0280          0.309    1770      1640   3560
TRUE 8      8 1.52e-01 -0.1670          0.337    1560      1400   3360
TRUE 9      9 3.56e-01 -0.2730          0.456    1390      1210   3190
TRUE 10    10 5.33e-01 -0.3630          0.592    1250      1050   3040
TRUE 11    11 6.52e-01 -0.4420          0.675    1130       922   2910
TRUE 12    12 7.28e-01 -0.5030          0.729    1030       812   2800
TRUE 13    14 8.32e-01 -0.6050          0.815     875       643   2600
TRUE 14    16 8.63e-01 -0.6910          0.835     754       519   2440
TRUE 15    18 8.94e-01 -0.7510          0.868     659       424   2300
TRUE 16    20 9.11e-01 -0.7920          0.886     583       349   2180
TRUE 17    22 9.22e-01 -0.8280          0.900     520       290   2080
TRUE 18    24 9.36e-01 -0.8590          0.918     468       243   1980
TRUE 19    26 9.39e-01 -0.8820          0.922     424       205   1900
TRUE 20    28 9.42e-01 -0.9010          0.927     386       174   1820
TRUE 21    30 9.42e-01 -0.9170          0.927     354       148   1760</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">0.9</span>;</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit,signed R^2&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>,</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>));</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>     <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>);</span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a><span class="co"># this line corresponds to using an R^2 cut-off of h</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.90</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>,</span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>], <span class="at">labels=</span>powers, <span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="02_3_WGCNA_HPG_files/figure-html/soft%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">22</span></span></code></pre></div>
<p>Seems like there is no strong effect on Scale Free Topology The
soft-threshold power I have selected here is 22, after which there is
little improvement in the Scale free topology fit.</p>
<div id="adjacency-matrix-and-tom-matrix" class="section level2"
number="4.1">
<h2><span class="header-section-number">4.1</span> Adjacency Matrix and
TOM Matrix</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">22</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a><span class="co">#datExpr0&lt;- read.csv(&quot;wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co">#rownames(datExpr0)&lt;- datExpr0$X</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a><span class="co">#datExpr0$X&lt;- NULL</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>adjacency<span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr0, <span class="at">power =</span> softPower, <span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>TOM<span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMType=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a><span class="co">#dissTOM&lt;- 1-TOM</span></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="fu">save</span>(adjacency, TOM, <span class="at">file=</span><span class="st">&quot;../WGCNA_results/HPG/HPG_network.RData&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="identify-co-expression-modules" class="section level1"
number="5">
<h1><span class="header-section-number">5</span> Identify co-expression
modules</h1>
<p>The first step is to identify modules of genes with similar gene
expression. Basically, the tool creates a hierarchical clustering of the
topological dissimilarity between genes.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;../WGCNA_results/HPG/HPG_network.RData&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>dissTOM<span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>TOM</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>geneTree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method=</span><span class="st">&quot;average&quot;</span>)</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a><span class="co">#plot(geneTree, xlab=&quot;&quot;, sub=&quot;&quot;, main= &quot;Gene Clustering on TOM-based dissimilarity&quot;, labels= FALSE, hang=0.04)</span></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>minModuleSize<span class="ot">&lt;-</span><span class="dv">30</span> </span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>dynamicMods<span class="ot">&lt;-</span><span class="fu">cutreeDynamic</span>(<span class="at">dendro=</span> geneTree, <span class="at">distM=</span> dissTOM, <span class="at">deepSplit=</span><span class="dv">2</span>, <span class="at">pamRespectsDendro=</span> <span class="cn">FALSE</span>, <span class="at">minClusterSize=</span> minModuleSize)</span></code></pre></div>
<pre><code>FALSE  ..cutHeight not given, setting it to 0.993  ===&gt;  99% of the (truncated) height range in dendro.
FALSE  ..done.</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co">#table(dynamicMods)</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>dynamicColors<span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span> <span class="fl">0.05</span>, <span class="at">main=</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span></code></pre></div>
<p><img src="02_3_WGCNA_HPG_files/figure-html/calculate%20modules-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now we try to merge some of these modules that are particularly
similar in expression based on a similarity threshold.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="co">#-----Merge modules whose expression profiles are very similar</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>MEList<span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, <span class="at">colors=</span> dynamicColors)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>MEs<span class="ot">=</span> MEList<span class="sc">$</span>eigengenes</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a><span class="co">#Calculate dissimilarity of module eigenegenes</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>MEDiss<span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(MEs)</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="co">#Cluster module eigengenes</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>METree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method=</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="co">#plot(METree, main= &quot;Clustering of module eigengenes&quot;, xlab= &quot;&quot;, sub= &quot;&quot;)</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>MEDissThres<span class="ot">=</span> <span class="fl">0.20</span> <span class="co"># i.e. merge modules with an r2 &gt; 0.90. This is stringent, could relax to reduce number of modules and increase module size.</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co">#abline(h=MEDissThres, col=&quot;red&quot;)</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>merge<span class="ot">=</span> <span class="fu">mergeCloseModules</span>(datExpr0, dynamicColors, <span class="at">cutHeight=</span> MEDissThres, <span class="at">verbose =</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>FALSE  mergeCloseModules: Merging modules whose distance is less than 0.2
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 19 module eigengenes in given set.
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 11 module eigengenes in given set.
FALSE    Calculating new MEs...
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 11 module eigengenes in given set.</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>mergedColors<span class="ot">=</span>merge<span class="sc">$</span>colors</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>mergedMEs<span class="ot">=</span> merge<span class="sc">$</span>newMEs</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, <span class="fu">cbind</span>(dynamicColors, mergedColors), <span class="fu">c</span>(<span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="st">&quot;Merged dynamic&quot;</span>), <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>moduleColors<span class="ot">=</span> mergedColors</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>colorOrder<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="fu">standardColors</span>(<span class="dv">50</span>))</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>moduleLabels<span class="ot">=</span> <span class="fu">match</span>(moduleColors, colorOrder)<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>MEs<span class="ot">=</span>mergedMEs</span></code></pre></div>
<p><br></p>
<div id="module-trait-correlation" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Module-trait
correlation</h2>
<p>Correlate the eigenvector of each of the 11 co-expression modules
with brain regions, individual and batch, as well as our interest
variables (mean testosterone, status, and social network strength)</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> key_brain</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(datTraits))</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>datTraits<span class="sc">$</span>Batch<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>Batch))</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>datTraits<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>Year))</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>datTraits<span class="sc">$</span>POM<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;POM&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>datTraits<span class="sc">$</span>VMH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;VMH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>datTraits<span class="sc">$</span>AH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;AH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>datTraits<span class="sc">$</span>PVN<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;PVN&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>datTraits<span class="sc">$</span>GON<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;GON&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>datTraits<span class="sc">$</span>PIT<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;PIT&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>datTraits<span class="sc">$</span>Bird<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>sampleID))</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>datTraits<span class="sc">$</span>Status2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(datTraits<span class="sc">$</span>Status<span class="sc">==</span><span class="st">&quot;territorial&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> <span class="fu">subset</span>(datTraits, <span class="at">select=</span><span class="fu">c</span>(<span class="st">&quot;Bird&quot;</span>,<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Year&quot;</span>,<span class="st">&quot;Status2&quot;</span>, <span class="st">&quot;mean_T&quot;</span>,<span class="st">&quot;strength.all_study&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;GON&quot;</span>,<span class="st">&quot;PIT&quot;</span>))</span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a><span class="fu">names</span>(datTraits)[<span class="fu">names</span>(datTraits)<span class="sc">==</span><span class="st">&quot;Status2&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Status&quot;</span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a></span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a><span class="co">#-----Define numbers of genes and samples</span></span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>nGenes <span class="ot">=</span> <span class="fu">ncol</span>(datExpr0);</span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>nSamples <span class="ot">=</span> <span class="fu">nrow</span>(datExpr0);</span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a><span class="co">#-----Recalculate MEs with color labels</span></span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>MEs0 <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, moduleColors)<span class="sc">$</span>eigengenes</span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a><span class="co">#-----Correlations of genes with eigengenes</span></span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a>moduleGeneCor<span class="ot">=</span><span class="fu">cor</span>(MEs,datExpr0)</span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a>moduleGenePvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleGeneCor, nSamples);</span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a>moduleTraitCor <span class="ot">=</span> <span class="fu">cor</span>(MEs, datTraits, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>);</span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a>moduleTraitPvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleTraitCor, nSamples);</span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a></span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a><span class="co">#---------------------Module-trait heatmap</span></span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a></span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a></span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a>textMatrix <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">signif</span>(moduleTraitCor, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>,</span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a>                   <span class="fu">signif</span>(moduleTraitPvalue, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>);</span>
<span id="cb25-38"><a href="#cb25-38" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">=</span> <span class="fu">dim</span>(moduleTraitCor)</span>
<span id="cb25-39"><a href="#cb25-39" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">10</span>, <span class="dv">3</span>, <span class="dv">3</span>));</span>
<span id="cb25-40"><a href="#cb25-40" tabindex="-1"></a><span class="co"># Display the correlation values within a heatmap plot</span></span>
<span id="cb25-41"><a href="#cb25-41" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> moduleTraitCor,</span>
<span id="cb25-42"><a href="#cb25-42" tabindex="-1"></a>               <span class="at">xLabels =</span> <span class="fu">names</span>(datTraits),</span>
<span id="cb25-43"><a href="#cb25-43" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb25-44"><a href="#cb25-44" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb25-45"><a href="#cb25-45" tabindex="-1"></a>               <span class="at">colorLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-46"><a href="#cb25-46" tabindex="-1"></a>               <span class="at">colors =</span> <span class="fu">blueWhiteRed</span>(<span class="dv">50</span>),</span>
<span id="cb25-47"><a href="#cb25-47" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix,</span>
<span id="cb25-48"><a href="#cb25-48" tabindex="-1"></a>               <span class="at">setStdMargins =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-49"><a href="#cb25-49" tabindex="-1"></a>               <span class="at">cex.text =</span> <span class="fl">0.5</span>,</span>
<span id="cb25-50"><a href="#cb25-50" tabindex="-1"></a>               <span class="at">zlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb25-51"><a href="#cb25-51" tabindex="-1"></a>               <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module-trait relationships in HPG&quot;</span>))</span></code></pre></div>
<p><img src="02_3_WGCNA_HPG_files/figure-html/trait%20correlation-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of genes per co-expression module
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
1792
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
2752
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
2296
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
94
</td>
</tr>
<tr>
<td style="text-align:left;">
grey60
</td>
<td style="text-align:right;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcyan
</td>
<td style="text-align:right;">
143
</td>
</tr>
<tr>
<td style="text-align:left;">
lightgreen
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta
</td>
<td style="text-align:right;">
1422
</td>
</tr>
<tr>
<td style="text-align:left;">
midnightblue
</td>
<td style="text-align:right;">
583
</td>
</tr>
<tr>
<td style="text-align:left;">
pink
</td>
<td style="text-align:right;">
1574
</td>
</tr>
<tr>
<td style="text-align:left;">
turquoise
</td>
<td style="text-align:right;">
4242
</td>
</tr>
</tbody>
</table>
<p>See the data portion of this repository to see the the module
membership and gene-significance results.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>folder<span class="ot">=</span><span class="st">&quot;../WGCNA_results/HPG/&quot;</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>test<span class="ot">=</span><span class="st">&quot;HPG&quot;</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>datME<span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr0,mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>datKME<span class="ot">&lt;-</span> <span class="fu">signedKME</span>(datExpr0, datME, <span class="at">outputColumnName=</span><span class="st">&quot;MM.&quot;</span>) <span class="co">#use the &quot;signed eigennode connectivity&quot; or module membership</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>MMPvalue <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(datKME), nSamples)) <span class="co"># Calculate module membership P-values</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>datKME<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(datKME)</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>MMPvalue<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(MMPvalue)</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>genes<span class="ot">=</span><span class="fu">names</span>(datExpr0)</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>genes,<span class="at">moduleColor=</span>moduleColors)</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>color<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, datKME, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>) <span class="co">#these are from your original WGCNA analysis </span></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a><span class="co">#head(color)</span></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(color), <span class="at">file =</span> <span class="fu">paste0</span>(folder,test,<span class="st">&quot;_results_ModuleMembership.csv&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>MMPvalue<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, MMPvalue, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a><span class="fu">write.csv</span>(MMPvalue, <span class="at">file=</span><span class="fu">paste0</span>(folder,test,<span class="st">&quot;_results_ModuleMembership_P-value.csv&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a><span class="do">#### gene-significance with traits of interest.</span></span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>Status) <span class="co">#change here for traits of interest</span></span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a><span class="fu">names</span>(trait) <span class="ot">=</span> <span class="st">&quot;status&quot;</span> <span class="co">#change here for traits of interest </span></span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a>modNames <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">names</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-27"><a href="#cb26-27" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb26-28"><a href="#cb26-28" tabindex="-1"></a></span>
<span id="cb26-29"><a href="#cb26-29" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>mean_T)</span>
<span id="cb26-30"><a href="#cb26-30" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;mean_T&quot;</span></span>
<span id="cb26-31"><a href="#cb26-31" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb26-32"><a href="#cb26-32" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb26-33"><a href="#cb26-33" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-34"><a href="#cb26-34" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-35"><a href="#cb26-35" tabindex="-1"></a>GS2<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb26-36"><a href="#cb26-36" tabindex="-1"></a></span>
<span id="cb26-37"><a href="#cb26-37" tabindex="-1"></a></span>
<span id="cb26-38"><a href="#cb26-38" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>strength.all_study)</span>
<span id="cb26-39"><a href="#cb26-39" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;strength&quot;</span></span>
<span id="cb26-40"><a href="#cb26-40" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb26-41"><a href="#cb26-41" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb26-42"><a href="#cb26-42" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-43"><a href="#cb26-43" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb26-44"><a href="#cb26-44" tabindex="-1"></a>GS3<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb26-45"><a href="#cb26-45" tabindex="-1"></a></span>
<span id="cb26-46"><a href="#cb26-46" tabindex="-1"></a>GS<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(GS)</span>
<span id="cb26-47"><a href="#cb26-47" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(GS,GS2, GS3)</span>
<span id="cb26-48"><a href="#cb26-48" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0,GS, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb26-49"><a href="#cb26-49" tabindex="-1"></a></span>
<span id="cb26-50"><a href="#cb26-50" tabindex="-1"></a>folder<span class="ot">=</span><span class="st">&quot;../WGCNA_results/HPG/&quot;</span></span>
<span id="cb26-51"><a href="#cb26-51" tabindex="-1"></a>test<span class="ot">=</span><span class="st">&quot;HPG&quot;</span></span>
<span id="cb26-52"><a href="#cb26-52" tabindex="-1"></a><span class="fu">write.csv</span>(GS, <span class="at">file=</span><span class="fu">paste0</span>(folder,test,<span class="st">&quot;_results_GeneSignificance.csv&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="intermodule-correlations" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Intermodule
correlations</h2>
<p>Given the network type used, all genes in a module can only go in the
same direction. This means that another module may represent genes that
are downregulated by the upregulation of genes in another module (or
vice-versa).</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Specify colors</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>colz<span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(MEs))</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a><span class="fu">names</span>(colz)<span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>ann_colors <span class="ot">=</span> <span class="fu">list</span>(<span class="at">module=</span>colz)</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>annotation_col<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">colnames</span>(MEs), <span class="at">module=</span><span class="fu">colnames</span>(MEs))</span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>correlation<span class="ot">&lt;-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a><span class="fu">pheatmap</span>(correlation, <span class="at">annotation_col =</span> annotation_col, <span class="at">annotation_colors =</span> ann_colors, <span class="at">annotation_legend=</span><span class="cn">FALSE</span>, <span class="at">legend_breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="fl">0.5</span>,<span class="dv">0</span>,<span class="fl">0.5</span>, <span class="dv">1</span>,<span class="dv">1</span>), </span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a><span class="at">main=</span><span class="st">&quot;module eigengene correlation in HPG&quot;</span>, <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;-1&quot;</span>, <span class="st">&quot;-0.5&quot;</span>, <span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.5&quot;</span>,<span class="st">&quot;1&quot;</span>,<span class="st">&quot;correlation</span><span class="sc">\n\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<p><img src="02_3_WGCNA_HPG_files/figure-html/intermodule%20correlations-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="go-for-modules" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> GO for modules</h2>
</div>
</div>
<div id="module-connectivity" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Module
Connectivity</h1>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1 0.000237 -0.1880         0.6950    7530      7510   7970
FALSE 2      2 0.096700  1.2400         0.0839    4980      4900   6050
FALSE 3      3 0.388000  1.2100         0.5830    3720      3690   5120
FALSE 4      4 0.452000  0.7790         0.6900    2970      2930   4560
FALSE 5      5 0.336000  0.4420         0.6140    2460      2400   4160
FALSE 6      6 0.125000  0.1980         0.4250    2100      2000   3850
FALSE 7      7 0.001520  0.0178         0.2780    1820      1690   3610
FALSE 8      8 0.083700 -0.1280         0.2790    1610      1450   3410
FALSE 9      9 0.258000 -0.2410         0.3760    1440      1250   3240
FALSE 10    10 0.429000 -0.3310         0.4900    1300      1100   3090
FALSE 11    11 0.578000 -0.4080         0.6060    1180       965   2970
FALSE 12    12 0.675000 -0.4750         0.6740    1080       855   2850
FALSE 13    14 0.781000 -0.5830         0.7570     916       683   2660
FALSE 14    16 0.836000 -0.6690         0.8020     793       554   2500
FALSE 15    18 0.866000 -0.7330         0.8310     696       456   2360
FALSE 16    20 0.893000 -0.7770         0.8630     618       380   2240
FALSE 17    22 0.906000 -0.8130         0.8790     553       318   2140
FALSE 18    24 0.914000 -0.8460         0.8900     500       268   2050
FALSE 19    26 0.928000 -0.8760         0.9090     454       229   1960
FALSE 20    28 0.935000 -0.8960         0.9200     415       195   1890
FALSE 21    30 0.938000 -0.9100         0.9250     381       168   1820</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/terr%20adj-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1 5.56e-06  0.0301          0.707    7530      7510   7950
FALSE 2      2 1.87e-01  1.7200          0.320    5000      4950   6060
FALSE 3      3 4.90e-01  1.3400          0.722    3750      3750   5140
FALSE 4      4 5.14e-01  0.8320          0.741    3000      3000   4570
FALSE 5      5 4.12e-01  0.4930          0.675    2500      2470   4170
FALSE 6      6 2.13e-01  0.2480          0.535    2130      2070   3870
FALSE 7      7 2.13e-02  0.0608          0.356    1860      1760   3620
FALSE 8      8 4.68e-02 -0.0830          0.325    1640      1510   3420
FALSE 9      9 2.51e-01 -0.2060          0.392    1470      1310   3250
FALSE 10    10 4.27e-01 -0.3030          0.496    1330      1150   3110
FALSE 11    11 5.65e-01 -0.3860          0.594    1210      1020   2980
FALSE 12    12 6.71e-01 -0.4500          0.682    1100       902   2870
FALSE 13    14 8.05e-01 -0.5570          0.795     942       724   2670
FALSE 14    16 8.56e-01 -0.6470          0.832     816       591   2510
FALSE 15    18 8.73e-01 -0.7050          0.845     717       487   2370
FALSE 16    20 8.92e-01 -0.7520          0.865     637       408   2250
FALSE 17    22 9.15e-01 -0.7920          0.892     571       344   2140
FALSE 18    24 9.31e-01 -0.8220          0.912     516       290   2050
FALSE 19    26 9.34e-01 -0.8480          0.915     469       249   1970
FALSE 20    28 9.36e-01 -0.8730          0.918     429       214   1890
FALSE 21    30 9.33e-01 -0.8930          0.914     394       185   1820</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/flo%20adj-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] TRUE</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_3_WGCNA_HPG_files/figure-html/k%20plots-1.png" width="672" style="display: block; margin: auto;" /><img src="02_3_WGCNA_HPG_files/figure-html/k%20plots-2.png" width="672" style="display: block; margin: auto;" /><img src="02_3_WGCNA_HPG_files/figure-html/k%20plots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>How does module size influence connectivity?</p>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="module-preservation" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Module
Preservation</h1>
<p><img src="02_3_WGCNA_HPG_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
