<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive change January 2022" />


<title>DESeq2 and PCA Analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_DESeq2.html">Differential Expression</a>
    </li>
    <li>
      <a href="./03_WGCNA_wholebrain.html">Whole-brain co-expression</a>
    </li>
    <li>
      <a href="./04_WGCNA_single_tissue.html">Co-expression in each tissue</a>
    </li>
    <li>
      <a href="./05_candidate_gene_analysis.html">Candidate Genes</a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DESeq2 and PCA Analysis</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive change January 2022</h4>
<h4 class="date">Last Knit “2022-05-26”</h4>

</div>


<p>Our primary aim is to explore whether among-individual variation in T modulated cooperative behaviour is associated with differences in gene expression. We have 5 main traits that were measured.</p>
<ul>
<li>Status: Territorial or Floater</li>
<li>Testosterone concentration
<ul>
<li><em>mean_T</em> is best estimate of among individual variation in T across all birds (’r summary(behav$mean_T)`). This measure is repeatable (Ryder et al 2020) within individuals and could be considered a hormonal phenotype.</li>
</ul></li>
<li>Strength</li>
<li>Mean number of social interactions per 10 hr day as detected by telemetry.</li>
</ul>
<p>Here, I take the filtered dataset from the QC process, and run a whole brain analysis, as well as individual tissue analysis. I also explore the relationships between our explanatory variables.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_key_Parsed_ReplicatesRemoved.csv&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>behav<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_unfiltered/PIPFIL_T_and_behav_data.csv&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(key)<span class="ot">&lt;-</span> key<span class="sc">$</span>X</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>key<span class="sc">$</span>Color_ID<span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;/&quot;</span>,<span class="st">&quot;&quot;</span>, key<span class="sc">$</span>Color_ID)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;Color_ID&quot;</span><span class="ot">=</span><span class="st">&quot;colorID&quot;</span>))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>behav<span class="sc">$</span>status<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(behav<span class="sc">$</span>status, <span class="fu">c</span>(<span class="st">&quot;floa&quot;</span><span class="ot">=</span><span class="st">&quot;floater&quot;</span>, <span class="st">&quot;terr&quot;</span><span class="ot">=</span><span class="st">&quot;territorial&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">merge</span>(key, behav, <span class="at">by=</span><span class="st">&quot;colorID&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">is.na</span>(key_behav<span class="sc">$</span>last_behav),]</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#create a data.frame with all of the observations. </span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> key[<span class="sc">!</span>key<span class="sc">$</span>X <span class="sc">%in%</span> key_behav<span class="sc">$</span>X,]</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>cols_not_key<span class="ot">&lt;-</span> <span class="fu">colnames</span>(key_behav)[<span class="sc">!</span><span class="fu">colnames</span>(key_behav) <span class="sc">%in%</span> <span class="fu">colnames</span>(key)]</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>cols_not_key_df<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(not_in_behav), <span class="at">ncol =</span> <span class="fu">length</span>(cols_not_key)))</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cols_not_key_df)<span class="ot">&lt;-</span> cols_not_key</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> <span class="fu">cbind</span>(not_in_behav, cols_not_key_df)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">rbind</span>(key_behav, not_in_behav)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>X</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Tissue, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;GON&quot;</span>,<span class="st">&quot;PIT&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;AI&quot;</span>,<span class="st">&quot;TNA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(<span class="fu">rownames</span>(key_behav)),]</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Class)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">revalue</span>(key_behav<span class="sc">$</span>Class, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;SCB floater&quot;</span><span class="ot">=</span><span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;DCB floater &quot;</span><span class="ot">=</span><span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;DCB territorial&quot;</span><span class="ot">=</span><span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Class, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>sampleID</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(key_behav<span class="sc">$</span>sampleID),]</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Year)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Batch)</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>key_behav<span class="sc">$</span>Status<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Status)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>key_behav_unique<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">duplicated</span>(key_behav<span class="sc">$</span>colorID),]</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="co">#what individuals are missing? </span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co">#key$in_behav&lt;- key$Harvest_ID %in% behav$Harvest_ID</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="co">#key[key$in_behav==FALSE, c(&quot;colorID&quot;, &quot;Harvest_ID&quot;)]</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co">#only unbanded individuals are missing from the larger behavioural dataset. </span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="co">#check that status matches between datasets</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co">#key_behav$status_match&lt;- ifelse(key_behav$status==key_behav$Status, &quot;same&quot;,&quot;not&quot;)</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co">#nrow(key_behav[key_behav$status_match==&quot;not&quot;,]) #0 GOOD!</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="co">#read the raw count data</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>data<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv&quot;</span>)</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(data)<span class="ot">&lt;-</span> data<span class="sc">$</span>X</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="do">### Gene Ontology</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>go_key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/Maggies_annotations_modifiedR.csv&quot;</span>)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>go_key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/pfil_GO_key_raw.csv&quot;</span>)</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_terms, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>go2gene_bp<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;P&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>go2gene_mf<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;F&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/ontology_obo_out.csv&quot;</span>)</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_obo, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;id&quot;</span><span class="ot">=</span><span class="st">&quot;GO_ID&quot;</span>))</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>go2name_bp<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;biological_process&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>go2name_mf<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;molecular_function&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span></code></pre></div>
<p><br><br></p>
<div id="exploration-of-the-behavioural-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Exploration of the behavioural data</h1>
<p>We were concerned about the signal from the last few days outweighing the gene-expression signal from their overall phenotype, so we will explore how different these measures are from eachother.</p>
<p><img src="02_DESeq2_files/figure-html/variable%20correlations-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="associations-between-variables" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Associations between variables</h2>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Given the significant relationship between degree and strength, I will only keep one of these variables - the strength. This was the most repeatable of the two sociality variables in Ryder et al. (2020).</p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/cor%20status-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>All significance tests cited in the figure above are based on just territorial and floater, but colour-coded according to plumage*status.</p>
<p><img src="02_DESeq2_files/figure-html/mean%20T%20and%20behav-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="nuisance-variables" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Nuisance variables</h2>
<p>In the previous QC exploration I found substantial DEGs according to collection Year and Sequencing Batch. Here we explore how the continuous variables are distributed across sampling years. The tests find there is a difference in the medians between years for both our interest traits.</p>
<p><img src="02_DESeq2_files/figure-html/variables%20according%20to%20Year-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
<div id="all-tissues-pca" class="section level1" number="2">
<h1><span class="header-section-number">2</span> All tissues PCA</h1>
<p>Plotting all the tissues</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> data</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(all_data)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes with avg read counts &lt;20</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>all_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(all_data, <span class="dv">1</span>, mean)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> all_data[all_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">20</span>,]</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>all_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>all_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(all_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(all_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>all_data<span class="ot">&lt;-</span> all_data[all_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>all_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#13652 genes.</span></span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Based on these results it seems like PC7 is associated with social Status. Let’s explore this PC further.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>test<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC7=</span>p<span class="sc">$</span>rotated<span class="sc">$</span>PC7)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>test<span class="ot">&lt;-</span> <span class="fu">cbind</span>(test, key_behav)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(lm(PC5 ~ Year + Tissue*Status,data=test ))</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>m1<span class="ot">&lt;-</span> <span class="fu">lme</span>(PC7 <span class="sc">~</span> Year <span class="sc">+</span> Tissue<span class="sc">*</span>Status, <span class="at">random=</span><span class="sc">~</span><span class="dv">1</span><span class="sc">|</span>Harvest_ID, <span class="at">data=</span>test)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>pc5test<span class="ot">&lt;-</span> <span class="fu">anova</span>(m1)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#pc5test&lt;- summary(aov(test$PC5 ~ test$Year + test$Status))</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>pc5test_anova <span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(<span class="fu">signif</span>(pc5test,<span class="dv">2</span>))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#,  theme = ttheme(&quot;mOrange&quot;))</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>e<span class="ot">&lt;-</span><span class="fu">biplot</span>(p, <span class="at">x=</span><span class="st">&quot;PC1&quot;</span>,<span class="at">y=</span><span class="st">&quot;PC7&quot;</span>, <span class="at">lab=</span><span class="cn">NULL</span>, <span class="at">colby=</span><span class="st">&quot;Tissue&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Status&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;&quot;</span>) <span class="sc">+</span> peri_theme</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>f<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x=</span>Status, <span class="at">y=</span>PC7)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>()<span class="sc">+</span> peri_theme  <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>) </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(e,<span class="fu">ggarrange</span>(pc5test_anova,f, <span class="at">nrow=</span><span class="dv">2</span>), <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">nrow=</span><span class="dv">1</span>,<span class="at">labels=</span><span class="fu">c</span>(LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]))</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/all%20tissues%20pc7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(filename=&quot;figure_alltissues_PC7.pdf&quot;, device=&quot;pdf&quot;,height =4 ,width=8, units=&quot;in&quot;)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(filename=&quot;figure_alltissues_PC7.png&quot;, device=&quot;png&quot;,height =4 ,width=8, units=&quot;in&quot;)</span></span></code></pre></div>
<p>Let’s see what genes are associated with each principal component. See GitHub repository for larger version.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> p<span class="sc">$</span>loadings</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>loadings<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(loadings)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>loadings<span class="ot">&lt;-</span> <span class="fu">merge</span>(loadings, go_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>plots<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>pcs<span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> pcs){</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#i=&quot;PC2&quot;</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> loadings[,<span class="fu">c</span>(i, <span class="st">&quot;best_anno&quot;</span>,<span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  sub<span class="ot">&lt;-</span> sub[<span class="fu">order</span>(sub[,i]),]</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  top30<span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">head</span>(sub,<span class="dv">15</span>), <span class="fu">tail</span>(sub,<span class="dv">15</span>))</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  top30<span class="sc">$</span>best_anno2<span class="ot">&lt;-</span> <span class="fu">factor</span>(top30<span class="sc">$</span>best_anno, <span class="at">levels=</span><span class="fu">unique</span>(top30<span class="sc">$</span>best_anno[<span class="fu">order</span>(top30[,i],top30<span class="sc">$</span>best_anno)]), <span class="at">ordered=</span><span class="cn">TRUE</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  top30<span class="sc">$</span>pos<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(top30[,i]<span class="sc">&lt;</span><span class="dv">0</span>, <span class="st">&quot;neg&quot;</span>, <span class="st">&quot;pos&quot;</span>)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  plots[[i]]<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top30, <span class="fu">aes</span>(<span class="at">x=</span>best_anno2, <span class="at">y=</span>top30[,i], <span class="at">fill=</span>pos)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="sc">+</span>  peri_theme <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;gene&quot;</span>, <span class="at">y=</span><span class="fu">paste</span>(i,<span class="st">&quot;loading&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;#0D8CFF&quot;</span>,<span class="st">&quot;#FF3300&quot;</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(<span class="at">plotlist=</span>plots, <span class="at">ncol=</span><span class="dv">5</span>, <span class="at">nrow=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/loadings%20all%20tissues-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(filename=&quot;figure_alltissue_loadings.pdf&quot;, device=&quot;pdf&quot;,height =8.5 ,width=16, units=&quot;in&quot;)</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(filename=&quot;figure_alltissue_loadings.png&quot;, device=&quot;png&quot;,height =8.5 ,width=16, units=&quot;in&quot;)</span></span></code></pre></div>
<p><br><br></p>
</div>
<div id="wholebrain-pca" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Wholebrain PCA</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>outliers<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;PFT2_POM_run1&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">!=</span><span class="st">&quot;GON&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(brain_behav, Tissue<span class="sc">!=</span><span class="st">&quot;PIT&quot;</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> brain_behav[<span class="sc">!</span><span class="fu">rownames</span>(brain_behav) <span class="sc">%in%</span> outliers,]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>brain_behav<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(brain_behav)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(brain_behav)]</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(brain_data)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes with avg read counts &lt;20</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>brain_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(brain_data, <span class="dv">1</span>, mean)</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> brain_data[brain_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">20</span>,]</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>brain_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>brain_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(brain_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(brain_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>brain_data<span class="ot">&lt;-</span> brain_data[brain_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>brain_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a><span class="co">#13652 genes.</span></span></code></pre></div>
<p>Looking for cohesive signatures across the whole brain. Overall, before this round of filtering for the brain dataset we have 16854 genes. After filtering we have 13653 genes. This final document is the result of lots of work in single tissues to determine outlier status that were identified in the QC process. We are excluding PFT2_POM_run1 because the analyses below indicated that it was an outlier within POM, other potentials identified in the QC process were not deemed outliers.</p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>PC5 and PC10 seem to be associated with our interest variables.</p>
<p><img src="02_DESeq2_files/figure-html/wholebrain%20pc5%20and%2010-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/loadings-1.png" width="1920" style="display: block; margin: auto;" /></p>
<div id="pc5-gene-ontology" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> PC5 Gene Ontology</h2>
<p>What are the gene ontology categories enriched in PC5?</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> loadings[<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;PC5&quot;</span>,<span class="st">&quot;best_anno&quot;</span>)]</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co">#res_go$logP&lt;- -log(abs(res_go$PC5), 10)</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#res_go$logP&lt;- ifelse(res_go$PC5&lt;0, -(res_go$logP), res_go$logP)</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> res_go[<span class="fu">order</span>(res_go<span class="sc">$</span>PC5, <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>geneList<span class="ot">&lt;-</span> res_go<span class="sc">$</span>PC5</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList)<span class="ot">&lt;-</span> res_go<span class="sc">$</span>gene</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>y<span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> go2gene_bp, <span class="at">TERM2NAME =</span> go2name_bp)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ridgeplot</span>(y) <span class="sc">+</span> <span class="fu">scale_fill_viridis</span>(<span class="at">direction=</span><span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span> peri_theme</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/PC5%20enrichment-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="batch-effects" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Batch Effects</h2>
<p>What happens if I use batch effect exclusion on the whole brain? This way I wouldn’t have to use single tissue</p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The PCA plot shows that this makes the differences between batches worse, even though the eigencorplot (correlation between PC axis and variable) doesn’t show it.</p>
<p>In previous whole brain analyses I have attempted to run full brain models in DESeq2 that accounts for individual. However, these analyses do not run because of unequal sample sizes among tissues (DESeq2 issue). Further, the PCA results suggests that accounting for batch effects is not helpful.</p>
<p><br><br></p>
</div>
</div>
<div id="gonads-gon" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Gonads (GON)</h1>
<p>I will use the first two tissues to demonstrate the processes applied across all tissues, but code echoing will be suppressed in future tissues.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">==</span><span class="st">&quot;GON&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(gon_key)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(gon_key)]</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(gon_data)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes with less than 5 reads</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, mean)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">5</span>,]</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(gon_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span></code></pre></div>
<p>Before filtering we had 16854 genes, after filtering for mean read count and excluding genes where &gt;50% of samples had a count of 0 we are left with 16169</p>
<div id="checking-the-sampling" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Checking the sampling</h2>
<p>Before I go into the analyses, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gonad samples for 16, with respect to T and the tracking data we have 12 samples.</p>
<p>Here, we will run all the data to check for DEGs against status, then subset the data to only include those individuals with T and behavioural data.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>fsb<span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Batch))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Batch), <span class="at">caption=</span><span class="fu">paste0</span>(<span class="st">&quot;Status across sequencing runs, fisher test p=&quot;</span>, fsb<span class="sc">$</span>p.value)) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>fsy<span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Year))</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Year), <span class="at">caption=</span><span class="fu">paste0</span>(<span class="st">&quot;Status across sampling years, fisher test p=&quot;</span>, fsy<span class="sc">$</span>p.value)) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sampling years, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
</div>
<div id="status-gon" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Status GON</h2>
<p>First, I will run through a data checking process. I will check for outliers, and then find out how our potential nuisance variables effect our gene expression data. If they have effects they will be incorporated into the final DESeq2 models.</p>
<p><br></p>
<div id="expression-data-exploration" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Expression data exploration</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_key, <span class="at">design=</span> <span class="sc">~</span> Status)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),] <span class="co">#remove any genes that didn&#39;t converge. </span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dd)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20status%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>vsd_data<span class="ot">&lt;-</span> <span class="fu">getVarianceStabilizedData</span>(dd)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(vsd_data, <span class="at">metadata =</span> gon_key)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_key<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;Status&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Batch&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;Gonads all samples PCA&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20status%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check to see how our interest and nuisance variables are correlated to the PC axes. This will help decide on model form for DESeq.</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">eigencorplot</span>(p, <span class="at">metavars =</span> <span class="fu">c</span>(<span class="st">&quot;time_kill&quot;</span>,<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Status&quot;</span>))</span></code></pre></div>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/GON%20status%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co">#gsg&lt;- goodSamplesGenes(datExpr0, verbose=3)</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gsg$allOK</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(datExpr0),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor)</span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20status%20set%20up-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Based on the eigencor plot, I will include Batch as a variable in my analyses, as it is associated with PC4.</p>
<p><br></p>
</div>
<div id="corrections-applied" class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Corrections applied</h3>
<p>Also let’s see how the data look when we account for Batch in our model by using limma::RemoveBatchEffect()</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#remove batch effect</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>mat<span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(vsd_data, gon_key<span class="sc">$</span>Batch)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(mat, <span class="at">metadata =</span> gon_key)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_key<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;Status&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Batch&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;PCA Gonads removeBatchEffect(Batch)&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20removeBatchEffect-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use this if Batch and Year are both pinged in the previous section. </span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#eigencorplot(p, metavars = c(&quot;Batch&quot;,&quot;Year&quot;, &quot;Status&quot;))</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>wgcnadata<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(mat))</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>traits<span class="ot">&lt;-</span> gon_key[, <span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>,  <span class="st">&quot;Year&quot;</span>)]</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>pilot<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;pilot&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>Run1<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;run1&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>Run2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;run2&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>yr2015<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2015&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>yr2017<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2017&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>traits<span class="sc">$</span>yr2018<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2018&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>traits<span class="ot">&lt;-</span> traits[,<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)] </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(wgcnadata),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>traitColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="fu">numbers2colors</span>(traits,<span class="at">signed=</span><span class="cn">FALSE</span>))</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(traitColors)[[<span class="dv">2</span>]]<span class="ot">=</span><span class="fu">paste</span>(<span class="fu">names</span>(traits))</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor,traitColors)</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20removeBatchEffect-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="analysis" class="section level3" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Analysis</h3>
<p>Now run DEseq with the updated model form and extract DEGs.</p>
<p>Formula ~ Batch + Status.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span> Batch <span class="sc">+</span> Status)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16169 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.012%
FALSE LFC &lt; 0 (down)     : 0, 0%
FALSE outliers [1]       : 18, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;Gonads Territorial vs Floater&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20Status2-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>Status_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>Status_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(res)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>Status_res<span class="ot">&lt;-</span> Status_res[<span class="fu">order</span>(Status_res<span class="sc">$</span>gene),]</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(Status_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_Status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>status_cols<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#86AD44&quot;</span>,<span class="st">&quot;#E54849&quot;</span>,<span class="st">&quot;#414042&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>gone<span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">1</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>gtwo<span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">2</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>gthree<span class="ot">&lt;-</span>  <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">3</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>gfour<span class="ot">&lt;-</span>  <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">4</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>gfive<span class="ot">&lt;-</span>  <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">5</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>gsix<span class="ot">&lt;-</span>  <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="fu">row.names</span>(res[<span class="dv">6</span>,]), <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;Class&quot;</span>,<span class="st">&quot;Status&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>vars<span class="ot">&lt;-</span> <span class="fu">list</span>(gone,gtwo,gthree,gfour,gfive,gsix)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>plots<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>){</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  dat<span class="ot">&lt;-</span> vars[[i]]</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  lfc<span class="ot">&lt;-</span> <span class="fu">signif</span>(res<span class="sc">$</span>log2FoldChange[i],<span class="dv">2</span>)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  padj<span class="ot">&lt;-</span> <span class="fu">formatC</span>(res<span class="sc">$</span>padj[i],<span class="dv">1</span>, <span class="at">format=</span><span class="st">&quot;e&quot;</span>)</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  plots[[i]]<span class="ot">&lt;-</span>  <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>Status, <span class="at">y=</span>count, <span class="at">color=</span>Class)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="fu">paste0</span>(<span class="fu">row.names</span>(res[i,])),<span class="at">subtitle=</span><span class="fu">paste0</span>(<span class="st">&quot;LFC=&quot;</span>,lfc,<span class="st">&quot; padj=&quot;</span>, padj),<span class="at">x=</span><span class="st">&quot;&quot;</span>, <span class="at">y=</span><span class="st">&quot;&quot;</span>) <span class="sc">+</span> peri_theme <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>status_cols, <span class="at">name=</span><span class="st">&quot;Age x Status&quot;</span>)</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(<span class="at">plotlist=</span>plots,<span class="at">ncol=</span><span class="dv">3</span>, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">common.legend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span> <span class="fu">annotate_figure</span>(g, <span class="at">top=</span><span class="fu">textGrob</span>(<span class="st">&quot;Top six DEGs by Status in GON&quot;</span>,<span class="at">gp=</span><span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">12</span>)), <span class="at">left=</span><span class="fu">textGrob</span>(<span class="st">&quot;Normalized counts&quot;</span>, <span class="at">rot =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">gp=</span><span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">12</span>)), <span class="at">bottom=</span><span class="fu">textGrob</span>(<span class="st">&quot;Status&quot;</span>,<span class="at">gp=</span><span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">12</span>)))</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a>g</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20status%20geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology" class="section level3" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Gene Ontology</h3>
<p>Conduct Gene Set enrichment analysis (GSEA) when there are few significant differentially expressed genes. This will leverage the whole list, but when we have a larger number I will use simple foreground/background enrichment.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> Status_res</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>res_go<span class="sc">$</span>logP<span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log</span>(res_go<span class="sc">$</span>pvalue, <span class="dv">10</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>res_go<span class="sc">$</span>logP<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(res_go<span class="sc">$</span>log2FoldChange<span class="sc">&lt;</span><span class="dv">0</span>, <span class="sc">-</span>(res_go<span class="sc">$</span>logP), res_go<span class="sc">$</span>logP)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>res_go<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_go)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res_go)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> <span class="fu">merge</span>(res_go, go_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> res_go[<span class="fu">order</span>(res_go<span class="sc">$</span>logP, <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>res_go<span class="ot">&lt;-</span> res_go[<span class="sc">!</span><span class="fu">is.na</span>(res_go<span class="sc">$</span>logP),]</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>res_go<span class="sc">$</span>pvalue<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(res_go<span class="sc">$</span>padj), <span class="cn">NA</span>, <span class="fu">as.numeric</span>(res_go<span class="sc">$</span>pvalue))</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>geneList<span class="ot">&lt;-</span> res_go<span class="sc">$</span>logP</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneList)<span class="ot">&lt;-</span> res_go<span class="sc">$</span>gene</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>y<span class="ot">&lt;-</span> <span class="fu">GSEA</span>(geneList, <span class="at">TERM2GENE =</span> go2gene_bp, <span class="at">TERM2NAME =</span> go2name_bp)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>gseaplot<span class="ot">&lt;-</span> <span class="fu">ridgeplot</span>(y) <span class="sc">+</span> peri_theme <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Top GO BP terms from GSEA&quot;</span>) <span class="sc">+</span> <span class="fu">scale_fill_viridis</span>()</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>gseaplot</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20Status%20BP-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Below is the code for making the figures in the supplement.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>g2<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(g,gseaplot,p, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">3</span>, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;C&quot;</span>))</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>g2</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;supp_figure_GON_status.png&quot;</span>, <span class="at">plot=</span>g2, <span class="at">device=</span><span class="st">&quot;png&quot;</span> ,<span class="at">height=</span><span class="dv">12</span>, <span class="at">width=</span><span class="fl">8.5</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="testosterone-gon" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Testosterone GON</h2>
<p>Given that the behaviors are best correlated with mean T, then this is our variable of interest</p>
<p><br></p>
<div id="expression-data-exploration-1" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Expression data exploration</h3>
<p>But first, we need to look at these new data in the gonads with the new subset with respect to our interest variables in the PCA.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">==</span><span class="st">&quot;GON&quot;</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(gon_behav, Batch<span class="sc">!=</span><span class="st">&quot;pilot&quot;</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(gon_behav)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[,<span class="fu">colnames</span>(gon_data) <span class="sc">%in%</span> <span class="fu">rownames</span>(gon_behav)]</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(gon_data)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes with less than 5 reads</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, mean)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">5</span>,]</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(gon_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span></code></pre></div>
<p>Before filtering we had 16169 genes, after filtering for mean read count and excluding genes where &gt;50% of samples had a count of 0 we are left with 16124…</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> mean_T)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),] <span class="co">#remove any genes that didn&#39;t converge. </span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co">#plotDispEsts(dd)</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>vsd_data<span class="ot">&lt;-</span> <span class="fu">getVarianceStabilizedData</span>(dd)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="co">#vsd_data&lt;- assay(vsd)</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">counts</span>(dd, <span class="at">normalized=</span><span class="cn">TRUE</span>), <span class="st">&quot;../DE_results/data_GON_norm.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(vsd_data, <span class="at">metadata =</span> gon_behav)</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a><span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_behav<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;mean_T&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Year&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;mean T&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">eigencorplot</span>(p, <span class="at">metavars =</span> <span class="fu">c</span>(<span class="st">&quot;time_kill&quot;</span>,<span class="st">&quot;Year&quot;</span>,<span class="st">&quot;Status&quot;</span>, <span class="st">&quot;mean_T&quot;</span>,<span class="st">&quot;strength.all_study&quot;</span>))</span></code></pre></div>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#gsg&lt;- goodSamplesGenes(datExpr0, verbose=3)</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co">#gsg$allOK</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(datExpr0),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor)</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-1" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-1" class="section level3" number="4.3.3">
<h3><span class="header-section-number">4.3.3</span> Analysis</h3>
<p>DESeq model specification ~ Year + mean_T</p>
<p>This chunk also includes a function I wrote to remove highly influential observations (as measured by Cook’s Distance) from the continuous data. I know from previous iterations of this analysis that there are observations that are highly influential and also <a href="https://support.bioconductor.org/p/126040/">DESeq2 does not exclude these from the results in continuous data</a>.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> Year <span class="sc">+</span> mean_T)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),]</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16124 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 33, 0.2%
FALSE LFC &lt; 0 (down)     : 111, 0.69%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>rm_continuous_outliers<span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">deseq=</span>x, <span class="at">results=</span>y, <span class="at">pct=</span><span class="fl">0.98</span>){</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  cooks<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">assays</span>(deseq)[[<span class="st">&quot;cooks&quot;</span>]])</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  p<span class="ot">=</span><span class="fu">ncol</span>(<span class="fu">model.matrix</span>(<span class="fu">design</span>(deseq), <span class="fu">colData</span>(deseq)))</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  n<span class="ot">=</span><span class="fu">nrow</span>(<span class="fu">colData</span>(deseq))</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  cd_cutoff<span class="ot">&lt;-</span> <span class="fu">qf</span>(pct,p,n<span class="sc">-</span>p)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  out<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">apply</span>(cooks, <span class="dv">1</span>, <span class="cf">function</span>(x){ x <span class="sc">&gt;</span> cd_cutoff })))</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  out<span class="sc">$</span>outlier<span class="ot">&lt;-</span> <span class="fu">apply</span>(out,<span class="dv">1</span>, any)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>  outliers<span class="ot">&lt;-</span> <span class="fu">rownames</span>(out)[out<span class="sc">$</span>outlier<span class="sc">==</span><span class="cn">TRUE</span>]</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>padj[<span class="fu">rownames</span>(results) <span class="sc">%in%</span> outliers]<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>pvalue[<span class="fu">rownames</span>(results) <span class="sc">%in%</span> outliers]<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results)</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16124 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 33, 0.2%
FALSE LFC &lt; 0 (down)     : 111, 0.69%
FALSE outliers [1]       : 37, 0.23%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>mean_T_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res[<span class="fu">order</span>(<span class="fu">rownames</span>(res)),])</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>mean_T_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(mean_T_res)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(mean_T_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_mean_T.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;Gonads Mean T&quot;</span>)</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20mean%20T%20DEseq-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/GON%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-1" class="section level3" number="4.3.4">
<h3><span class="header-section-number">4.3.4</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms by mean_T in GON
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0051412" class="uri">GO:0051412</a>
</td>
<td style="text-align:left;">
response to corticosterone
</td>
<td style="text-align:left;">
4/124
</td>
<td style="text-align:left;">
13/12501
</td>
<td style="text-align:right;">
0.0000062
</td>
<td style="text-align:right;">
0.0044303
</td>
<td style="text-align:right;">
0.0039963
</td>
<td style="text-align:left;">
AVPR1A/LOC113992757/STAR/TH
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001975" class="uri">GO:0001975</a>
</td>
<td style="text-align:left;">
response to amphetamine
</td>
<td style="text-align:left;">
4/124
</td>
<td style="text-align:left;">
24/12501
</td>
<td style="text-align:right;">
0.0000840
</td>
<td style="text-align:right;">
0.0302523
</td>
<td style="text-align:right;">
0.0272890
</td>
<td style="text-align:left;">
DBH/RGS4/SLC18A2/TH
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050729" class="uri">GO:0050729</a>
</td>
<td style="text-align:left;">
positive regulation of inflammatory response
</td>
<td style="text-align:left;">
5/124
</td>
<td style="text-align:left;">
57/12501
</td>
<td style="text-align:right;">
0.0002455
</td>
<td style="text-align:right;">
0.0500987
</td>
<td style="text-align:right;">
0.0451914
</td>
<td style="text-align:left;">
AGTR1/IFI35/LOC114003559/PDE2A/SNCA
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007626" class="uri">GO:0007626</a>
</td>
<td style="text-align:left;">
locomotory behavior
</td>
<td style="text-align:left;">
5/124
</td>
<td style="text-align:left;">
59/12501
</td>
<td style="text-align:right;">
0.0002889
</td>
<td style="text-align:right;">
0.0500987
</td>
<td style="text-align:right;">
0.0451914
</td>
<td style="text-align:left;">
C1QL1/CHRNA3/DBH/SLC18A2/TH
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010460" class="uri">GO:0010460</a>
</td>
<td style="text-align:left;">
positive regulation of heart rate
</td>
<td style="text-align:left;">
3/124
</td>
<td style="text-align:left;">
15/12501
</td>
<td style="text-align:right;">
0.0003973
</td>
<td style="text-align:right;">
0.0500987
</td>
<td style="text-align:right;">
0.0451914
</td>
<td style="text-align:left;">
AVPR1A/RGS4/SCN3B
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006469" class="uri">GO:0006469</a>
</td>
<td style="text-align:left;">
negative regulation of protein kinase activity
</td>
<td style="text-align:left;">
5/124
</td>
<td style="text-align:left;">
64/12501
</td>
<td style="text-align:right;">
0.0004230
</td>
<td style="text-align:right;">
0.0500987
</td>
<td style="text-align:right;">
0.0451914
</td>
<td style="text-align:left;">
LOC113982361/LOC114003559/SNCA/SOCS3/THY1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007268" class="uri">GO:0007268</a>
</td>
<td style="text-align:left;">
chemical synaptic transmission
</td>
<td style="text-align:left;">
8/124
</td>
<td style="text-align:left;">
184/12501
</td>
<td style="text-align:right;">
0.0004871
</td>
<td style="text-align:right;">
0.0500987
</td>
<td style="text-align:right;">
0.0451914
</td>
<td style="text-align:left;">
CHRNA3/DBH/GABRB1/HTR1E/SLC18A2/SNCA/SNCG/SYT1
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034605" class="uri">GO:0034605</a>
</td>
<td style="text-align:left;">
cellular response to heat
</td>
<td style="text-align:left;">
4/124
</td>
<td style="text-align:left;">
40/12501
</td>
<td style="text-align:right;">
0.0006394
</td>
<td style="text-align:right;">
0.0557705
</td>
<td style="text-align:right;">
0.0503076
</td>
<td style="text-align:left;">
BAG3/EIF2S1/PDCD6/STAC
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031103" class="uri">GO:0031103</a>
</td>
<td style="text-align:left;">
axon regeneration
</td>
<td style="text-align:left;">
3/124
</td>
<td style="text-align:left;">
18/12501
</td>
<td style="text-align:right;">
0.0006971
</td>
<td style="text-align:right;">
0.0557705
</td>
<td style="text-align:right;">
0.0503076
</td>
<td style="text-align:left;">
GAP43/ISL1/LOC113990764
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051281" class="uri">GO:0051281</a>
</td>
<td style="text-align:left;">
positive regulation of release of sequestered calcium ion into cytosol
</td>
<td style="text-align:left;">
3/124
</td>
<td style="text-align:left;">
20/12501
</td>
<td style="text-align:right;">
0.0009599
</td>
<td style="text-align:right;">
0.0691153
</td>
<td style="text-align:right;">
0.0623452
</td>
<td style="text-align:left;">
CEMIP/SNCA/THY1
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status" class="section level3" number="4.3.5">
<h3><span class="header-section-number">4.3.5</span> mean T x Status</h3>
<p>For pulling out candidate genes with interactions</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> Year <span class="sc">+</span> Status <span class="sc">+</span> mean_T <span class="sc">+</span> Status<span class="sc">:</span>mean_T)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),]</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>, <span class="at">name =</span> <span class="st">&quot;Statusterritorial.mean_T&quot;</span>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16124 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 6, 0.037%
FALSE LFC &lt; 0 (down)     : 9, 0.056%
FALSE outliers [1]       : 29, 0.18%
FALSE low counts [2]     : 311, 1.9%
FALSE (mean count &lt; 8)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>interaction_res<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>interaction_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>interaction_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="st">&quot;GON&quot;</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(interaction_res, <span class="at">file=</span><span class="st">&quot;../DE_results/interactions_GON_mean_T_Status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;GON mean Testosterone * Status&quot;</span>)</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-15-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-gon" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Strength GON</h2>
<div id="analysis-2" class="section level3" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Analysis</h3>
<p>formula ~ Year + scale(Strength)</p>
<p>Without scaling strength, DESeq2 would throw a warning, recommending the variable be scaled.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>dd<span class="sc">$</span>strength.all_study<span class="ot">&lt;-</span> <span class="fu">scale</span>(dd<span class="sc">$</span>strength.all_study)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span>Year <span class="sc">+</span> strength.all_study)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16124 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 10, 0.062%
FALSE LFC &lt; 0 (down)     : 11, 0.068%
FALSE outliers [1]       : 45, 0.28%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>strength_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res[<span class="fu">order</span>(<span class="fu">rownames</span>(res)),])</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>strength_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(strength_res)</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(strength_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_Strength.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;GON Social Network Strength&quot;</span>)</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/GON%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/GON%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-2" class="section level3" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/GON%20strength%20BP-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="pituitary-pit" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Pituitary (PIT)</h1>
<div id="checking-the-sampling-1" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Checking the sampling</h2>
<p>Before filtering we had 16854 genes, after filtering for mean read count and excluding genes where &gt;50% of samples had a count of 0 we are left with 14812</p>
<p>Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pit samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.22
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/PIT%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="expression-data-exploration-2" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/PIT%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-2" class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks controlling for Batch also removes the association with Year. Let’s just include Batch.</p>
<p><br></p>
</div>
</div>
<div id="status-pit" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Status PIT</h2>
<div id="analysis-3" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<p>In this section, I find results with different p-value distributions. I was following <a href="https://www.huber.embl.de/users/klaus/Teaching/DESeq2-Analysis.pdf">THIS</a> tutorial by Bernd Klaus and it recommended using <code>fdrtool</code> to correct these p-values.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span> Batch <span class="sc">+</span> Status)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res[<span class="sc">!</span><span class="fu">is.na</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res1[<span class="sc">!</span><span class="fu">is.na</span>(res1<span class="sc">$</span>pvalue),]</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>res1<span class="sc">$</span>padj<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>res1.fdr<span class="ot">&lt;-</span> <span class="fu">fdrtool</span>(res1<span class="sc">$</span>stat, <span class="at">statistic=</span><span class="st">&quot;normal&quot;</span>, <span class="at">plot=</span><span class="cn">FALSE</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>res1<span class="sc">$</span>padj<span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res1.fdr<span class="sc">$</span>pval, <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>outs<span class="ot">&lt;-</span> res[<span class="fu">is.na</span>(res<span class="sc">$</span>padj),] <span class="co"># add those outliers back in... </span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>res1<span class="ot">&lt;-</span> <span class="fu">rbind</span>(res1, outs)</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res1[<span class="fu">order</span>(res1<span class="sc">$</span>padj),]</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res1)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 8, 0.054%
FALSE LFC &lt; 0 (down)     : 6, 0.041%
FALSE outliers [1]       : 91, 0.61%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>status_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res1)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>status_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(status_res)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>status_res<span class="ot">&lt;-</span> status_res[<span class="fu">order</span>(status_res<span class="sc">$</span>gene),]</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(status_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_PIT_status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;Pituitary Territorial vs Floater&quot;</span>)</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res1, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;FDR tool corrected PIT Status&quot;</span>)</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res1.fdr<span class="sc">$</span>pval, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;P-value correction &#39;fdrtool&#39;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&#39;Corrected&#39; P-value&quot;</span>)</span></code></pre></div>
<p><img src="02_DESeq2_files/figure-html/PIT%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-ontology-3" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/PIT%20Status%20GO-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-pit" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Testosterone PIT</h2>
<div id="analysis-4" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Analysis</h3>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 14, 0.095%
FALSE LFC &lt; 0 (down)     : 5, 0.034%
FALSE outliers [1]       : 62, 0.42%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-4" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/PIT%20mean%20T%20BP-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/PIT%20mean%20T%20BP-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/PIT%20mean%20T%20BP-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-1" class="section level3" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 24, 0.16%
FALSE LFC &lt; 0 (down)     : 67, 0.45%
FALSE outliers [1]       : 37, 0.25%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20mean%20T%20x%20Status-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-23-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-pit" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Strength PIT</h2>
<div id="analysis-5" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> Analysis</h3>
<p>Formula: ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 387, 2.6%
FALSE LFC &lt; 0 (down)     : 113, 0.76%
FALSE outliers [1]       : 207, 1.4%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/PIT%20strength%20geneplots-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>As you can see above, PFT3 has consistently high leverage on the social network analysis. Even if I play around with the outlier theshold this I am always getting non-linear plots like above. So, let’s see what happens when I remove this sample.</p>
<p><br></p>
</div>
<div id="re-analysis---outlier-removed" class="section level3" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Re-analysis - outlier removed</h3>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 109, 0.74%
FALSE LFC &lt; 0 (down)     : 13, 0.088%
FALSE outliers [1]       : 58, 0.39%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20Strength2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PIT%20strength%20geneplots2-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-5" class="section level3" number="5.4.3">
<h3><span class="header-section-number">5.4.3</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/PIT%20strength%20BP-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in PIT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0046849" class="uri">GO:0046849</a>
</td>
<td style="text-align:left;">
bone remodeling
</td>
<td style="text-align:left;">
3/100
</td>
<td style="text-align:left;">
14/12040
</td>
<td style="text-align:right;">
0.0001894
</td>
<td style="text-align:right;">
0.0765610
</td>
<td style="text-align:right;">
0.0748556
</td>
<td style="text-align:left;">
LTBP3/TPH1/WNT16
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007015" class="uri">GO:0007015</a>
</td>
<td style="text-align:left;">
actin filament organization
</td>
<td style="text-align:left;">
6/100
</td>
<td style="text-align:left;">
109/12040
</td>
<td style="text-align:right;">
0.0002867
</td>
<td style="text-align:right;">
0.0765610
</td>
<td style="text-align:right;">
0.0748556
</td>
<td style="text-align:left;">
CORO1B/INPPL1/LOC113988218/LOC113992312/SAMD14/SORBS3
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071260" class="uri">GO:0071260</a>
</td>
<td style="text-align:left;">
cellular response to mechanical stimulus
</td>
<td style="text-align:left;">
4/100
</td>
<td style="text-align:left;">
59/12040
</td>
<td style="text-align:right;">
0.0014369
</td>
<td style="text-align:right;">
0.2557690
</td>
<td style="text-align:right;">
0.2500718
</td>
<td style="text-align:left;">
DAG1/PDE2A/TNFRSF10B/TNFRSF8
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006833" class="uri">GO:0006833</a>
</td>
<td style="text-align:left;">
water transport
</td>
<td style="text-align:left;">
2/100
</td>
<td style="text-align:left;">
10/12040
</td>
<td style="text-align:right;">
0.0029428
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
LOC113983498/LOC113999239
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006790" class="uri">GO:0006790</a>
</td>
<td style="text-align:left;">
sulfur compound metabolic process
</td>
<td style="text-align:left;">
2/100
</td>
<td style="text-align:left;">
11/12040
</td>
<td style="text-align:right;">
0.0035774
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
LOC113994298/SUOX
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903779" class="uri">GO:1903779</a>
</td>
<td style="text-align:left;">
regulation of cardiac conduction
</td>
<td style="text-align:left;">
3/100
</td>
<td style="text-align:left;">
39/12040
</td>
<td style="text-align:right;">
0.0040906
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
LOC114003417/NPR1/STIM1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1901796" class="uri">GO:1901796</a>
</td>
<td style="text-align:left;">
regulation of signal transduction by p53 class mediator
</td>
<td style="text-align:left;">
5/100
</td>
<td style="text-align:left;">
129/12040
</td>
<td style="text-align:right;">
0.0043682
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
CDK5R1/LOC113989520/LOC113998423/PRKAG1/STK11
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010212" class="uri">GO:0010212</a>
</td>
<td style="text-align:left;">
response to ionizing radiation
</td>
<td style="text-align:left;">
3/100
</td>
<td style="text-align:left;">
42/12040
</td>
<td style="text-align:right;">
0.0050470
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
BABAM1/LOC113998423/STK11
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045860" class="uri">GO:0045860</a>
</td>
<td style="text-align:left;">
positive regulation of protein kinase activity
</td>
<td style="text-align:left;">
3/100
</td>
<td style="text-align:left;">
42/12040
</td>
<td style="text-align:right;">
0.0050470
</td>
<td style="text-align:right;">
0.2994552
</td>
<td style="text-align:right;">
0.2927849
</td>
<td style="text-align:left;">
DAG1/MAPRE3/PRKAG1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002009" class="uri">GO:0002009</a>
</td>
<td style="text-align:left;">
morphogenesis of an epithelium
</td>
<td style="text-align:left;">
2/100
</td>
<td style="text-align:left;">
14/12040
</td>
<td style="text-align:right;">
0.0058237
</td>
<td style="text-align:right;">
0.3109833
</td>
<td style="text-align:right;">
0.3040562
</td>
<td style="text-align:left;">
DAG1/SLC9A3R1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/PIT%20strength%20BP-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/PIT%20strength%20BP-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="ventromedial-hypothalamus-vmh" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Ventromedial Hypothalamus (VMH)</h1>
<div id="checking-the-sampling-2" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Checking the sampling</h2>
<p>Before filtering we had 16854 genes, after filtering for mean read count and excluding genes where &gt;50% of samples had a count of 0 we are left with 15220</p>
<p>Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have vmh samples for 16, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.76
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00078
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/VMH%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The majority of our data in VMH is in run 1.</p>
<div id="expression-data-exploration-3" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/VMH%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/VMH%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-3" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Corrections applied</h3>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks like controlling for Batch is sufficient here, as Batch is highly correlated with Year.</p>
</div>
</div>
<div id="status-vmh" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Status VMH</h2>
<div id="analysis-6" class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15220 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 4, 0.026%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-6" class="section level3" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/VMH%20status%20GO-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-vmh" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Testosterone VMH</h2>
<p><br></p>
<div id="expression-data-exploration-4" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/VMH%20DESeq%20continuous%20variable%20subset-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20DESeq%20continuous%20variable%20subset-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-4" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-29-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Batch does not account for year here… so have to include both…</p>
<p><br></p>
</div>
<div id="analysis-7" class="section level3" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Analysis</h3>
<p>Formula: ~ Batch + Year + mean_T</p>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 8, 0.053%
FALSE outliers [1]       : 52, 0.34%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/vmh%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-7" class="section level3" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/VMH%20mean%20T%20BP-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-2" class="section level3" number="6.3.5">
<h3><span class="header-section-number">6.3.5</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 45, 0.3%
FALSE LFC &lt; 0 (down)     : 54, 0.36%
FALSE outliers [1]       : 141, 0.93%
FALSE low counts [2]     : 275, 1.8%
FALSE (mean count &lt; 7)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20mean%20T%20x%20Status-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-31-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-vmh" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Strength VMH</h2>
<div id="analysis-8" class="section level3" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Analysis</h3>
<p>Formula ~ Batch + Year + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 28, 0.18%
FALSE LFC &lt; 0 (down)     : 19, 0.13%
FALSE outliers [1]       : 82, 0.54%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/vmh%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/VMH%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-8" class="section level3" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/VMH%20strength%20BP-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/VMH%20strength%20BP-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="anterior-hypothalamus-ah" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Anterior Hypothalamus (AH)</h1>
<p><br></p>
<div id="checking-the-sampling-3" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Checking the sampling</h2>
<p>For the AH, we started with 16854 genes but after filering we have 14926 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ah samples for 11, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-5" class="section level3" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/AH%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-5" class="section level3" number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>PFT1 and PFT11 cluster with the BSTm as previously identified by the brain tissue PCA plot and the all tissue WGCNA dendrogram. We know that there were issues with the sectioning plane in individual PFT1 that could affect this tissue, but not PFT11.</p>
<p>The effects of removing these samples will be explored for each analysis vs removing a randomly selected pair of samples. These results (not shown) showed similar effects, so these samples were not excluded.</p>
<p><br></p>
</div>
</div>
<div id="status-ah" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Status AH</h2>
<div id="analysis-9" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Analysis</h3>
<p>Formula = ~ Year + Status</p>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 1, 0.0067%
FALSE outliers [1]       : 28, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 1, 0.0067%
FALSE outliers [1]       : 28, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-9" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-34-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-ah" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Testosterone AH</h2>
<div id="analysis-10" class="section level3" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> Analysis</h3>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 23, 0.15%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-10" class="section level3" number="7.3.2">
<h3><span class="header-section-number">7.3.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-3" class="section level3" number="7.3.3">
<h3><span class="header-section-number">7.3.3</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 14, 0.094%
FALSE LFC &lt; 0 (down)     : 27, 0.18%
FALSE outliers [1]       : 29, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-39-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-ah" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Strength AH</h2>
<div id="analysis-11" class="section level3" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14893 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 76, 0.51%
FALSE LFC &lt; 0 (down)     : 109, 0.73%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AH%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-11" class="section level3" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in AH
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0043928" class="uri">GO:0043928</a>
</td>
<td style="text-align:left;">
exonucleolytic catabolism of deadenylated mRNA
</td>
<td style="text-align:left;">
4/122
</td>
<td style="text-align:left;">
37/12117
</td>
<td style="text-align:right;">
0.0004997
</td>
<td style="text-align:right;">
0.1824968
</td>
<td style="text-align:right;">
0.1824968
</td>
<td style="text-align:left;">
DDX6/DIS3/LOC113984931/LOC113990767
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0017148" class="uri">GO:0017148</a>
</td>
<td style="text-align:left;">
negative regulation of translation
</td>
<td style="text-align:left;">
5/122
</td>
<td style="text-align:left;">
69/12117
</td>
<td style="text-align:right;">
0.0006403
</td>
<td style="text-align:right;">
0.1824968
</td>
<td style="text-align:right;">
0.1824968
</td>
<td style="text-align:left;">
DDX6/EIF4E3/IGF2BP3/LOC113999202/PAIP2B
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061061" class="uri">GO:0061061</a>
</td>
<td style="text-align:left;">
muscle structure development
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
11/12117
</td>
<td style="text-align:right;">
0.0052112
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
CAPN3/PRICKLE4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042472" class="uri">GO:0042472</a>
</td>
<td style="text-align:left;">
inner ear morphogenesis
</td>
<td style="text-align:left;">
3/122
</td>
<td style="text-align:left;">
39/12117
</td>
<td style="text-align:right;">
0.0069892
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
HMX2/HMX3/USH1C
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007605" class="uri">GO:0007605</a>
</td>
<td style="text-align:left;">
sensory perception of sound
</td>
<td style="text-align:left;">
5/122
</td>
<td style="text-align:left;">
123/12117
</td>
<td style="text-align:right;">
0.0080150
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
EPS8L2/LOC113987015/OTOG/TSPEAR/USH1C
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016075" class="uri">GO:0016075</a>
</td>
<td style="text-align:left;">
rRNA catabolic process
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
14/12117
</td>
<td style="text-align:right;">
0.0084538
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
DIS3/LOC113990767
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035810" class="uri">GO:0035810</a>
</td>
<td style="text-align:left;">
positive regulation of urine volume
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
14/12117
</td>
<td style="text-align:right;">
0.0084538
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
HAS2/LOC113993563
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045943" class="uri">GO:0045943</a>
</td>
<td style="text-align:left;">
positive regulation of transcription by RNA polymerase I
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
14/12117
</td>
<td style="text-align:right;">
0.0084538
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
DHX33/LOC113997168
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007274" class="uri">GO:0007274</a>
</td>
<td style="text-align:left;">
neuromuscular synaptic transmission
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
16/12117
</td>
<td style="text-align:right;">
0.0110025
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
CHAT/STAC3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006956" class="uri">GO:0006956</a>
</td>
<td style="text-align:left;">
complement activation
</td>
<td style="text-align:left;">
2/122
</td>
<td style="text-align:left;">
18/12117
</td>
<td style="text-align:right;">
0.0138457
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:right;">
0.4535007
</td>
<td style="text-align:left;">
C1S/C3
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="paraventricular-nucleus-pvn" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Paraventricular Nucleus (PVN)</h1>
<p><br></p>
<div id="checking-the-sampling-4" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Checking the sampling</h2>
<p>For the PVN, we started with 16854 genes but after filering we have 15151 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pvn samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<div id="expression-data-exploration-6" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/PVN%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks like we don’t have any outliers, nor do we need to include Year (or Batch) as a covariate.</p>
<p><br></p>
</div>
</div>
<div id="status-pvn" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Status PVN</h2>
<div id="analysis-12" class="section level3" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Analysis</h3>
<p>Formula = ~ Status</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 27, 0.18%
FALSE LFC &lt; 0 (down)     : 15, 0.099%
FALSE outliers [1]       : 19, 0.13%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-12" class="section level3" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms by status in PVN
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
4/33
</td>
<td style="text-align:left;">
38/12236
</td>
<td style="text-align:right;">
0.0000030
</td>
<td style="text-align:right;">
0.0006673
</td>
<td style="text-align:right;">
0.0004758
</td>
<td style="text-align:left;">
CFAP73/DNAH1/LOC113981955/TTC25
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036159" class="uri">GO:0036159</a>
</td>
<td style="text-align:left;">
inner dynein arm assembly
</td>
<td style="text-align:left;">
3/33
</td>
<td style="text-align:left;">
15/12236
</td>
<td style="text-align:right;">
0.0000080
</td>
<td style="text-align:right;">
0.0008750
</td>
<td style="text-align:right;">
0.0006238
</td>
<td style="text-align:left;">
CFAP73/DNAH1/LOC113981955
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061640" class="uri">GO:0061640</a>
</td>
<td style="text-align:left;">
cytoskeleton-dependent cytokinesis
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
19/12236
</td>
<td style="text-align:right;">
0.0011721
</td>
<td style="text-align:right;">
0.0859509
</td>
<td style="text-align:right;">
0.0612760
</td>
<td style="text-align:left;">
LOC113992196/SEPT4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2001244" class="uri">GO:2001244</a>
</td>
<td style="text-align:left;">
positive regulation of intrinsic apoptotic signaling pathway
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
23/12236
</td>
<td style="text-align:right;">
0.0017224
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
LOC113992196/SEPT4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001501" class="uri">GO:0001501</a>
</td>
<td style="text-align:left;">
skeletal system development
</td>
<td style="text-align:left;">
3/33
</td>
<td style="text-align:left;">
99/12236
</td>
<td style="text-align:right;">
0.0023505
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
CNMD/LOC113991284/LOC113997556
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051491" class="uri">GO:0051491</a>
</td>
<td style="text-align:left;">
positive regulation of filopodium assembly
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
31/12236
</td>
<td style="text-align:right;">
0.0031234
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
LOC113996134/RIPOR2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006695" class="uri">GO:0006695</a>
</td>
<td style="text-align:left;">
cholesterol biosynthetic process
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
34/12236
</td>
<td style="text-align:right;">
0.0037492
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
DHCR7/FDFT1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0017157" class="uri">GO:0017157</a>
</td>
<td style="text-align:left;">
regulation of exocytosis
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
34/12236
</td>
<td style="text-align:right;">
0.0037492
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
LOC113992196/SEPT4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045540" class="uri">GO:0045540</a>
</td>
<td style="text-align:left;">
regulation of cholesterol biosynthetic process
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
34/12236
</td>
<td style="text-align:right;">
0.0037492
</td>
<td style="text-align:right;">
0.0916478
</td>
<td style="text-align:right;">
0.0653375
</td>
<td style="text-align:left;">
DHCR7/FDFT1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034613" class="uri">GO:0034613</a>
</td>
<td style="text-align:left;">
cellular protein localization
</td>
<td style="text-align:left;">
2/33
</td>
<td style="text-align:left;">
54/12236
</td>
<td style="text-align:right;">
0.0092476
</td>
<td style="text-align:right;">
0.1394724
</td>
<td style="text-align:right;">
0.0994325
</td>
<td style="text-align:left;">
LOC113992196/SEPT4
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-44-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-44-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-pvn" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Testosterone PVN</h2>
<div id="analysis-13" class="section level3" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 7, 0.046%
FALSE LFC &lt; 0 (down)     : 6, 0.04%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-13" class="section level3" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-46-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-4" class="section level3" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 13, 0.086%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-48-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-pvn" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Strength PVN</h2>
<div id="analysis-14" class="section level3" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 29, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/PVN%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-14" class="section level3" number="8.4.2">
<h3><span class="header-section-number">8.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-49-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="medial-preoptic-area-pom" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Medial Preoptic Area (POM)</h1>
<div id="checking-the-sampling-5" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Checking the sampling</h2>
<p>For the POM, we started with 16854 genes but after filering we have 15318 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have pom samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00055
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-7" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/POM%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/POM%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-6" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-51-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-51-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Don’t need to control for year because accounting for Batch seems to fix that.</p>
<p><br></p>
</div>
<div id="outlier-removed" class="section level3" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> Outlier removed</h3>
<p>PFT2 was previously identified as a potential outlier, and it’s coming out quite separately here, even after all the corrections for batch effects. So, this one I am going to REMOVE.</p>
<p><br></p>
<p>##Status POM</p>
<p>###Analysis</p>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15314 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 16, 0.1%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-15" class="section level3" number="9.1.4">
<h3><span class="header-section-number">9.1.4</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-52-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-pom" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Testosterone POM</h2>
<div id="expression-data-exploration-8" class="section level3" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-15" class="section level3" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 34, 0.22%
FALSE LFC &lt; 0 (down)     : 11, 0.072%
FALSE outliers [1]       : 29, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-16" class="section level3" number="9.2.3">
<h3><span class="header-section-number">9.2.3</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/POM%20mean%20T%20GO-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms by mean_T in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0042311" class="uri">GO:0042311</a>
</td>
<td style="text-align:left;">
vasodilation
</td>
<td style="text-align:left;">
3/40
</td>
<td style="text-align:left;">
21/12294
</td>
<td style="text-align:right;">
0.0000408
</td>
<td style="text-align:right;">
0.0105951
</td>
<td style="text-align:right;">
0.007764
</td>
<td style="text-align:left;">
ITGA11/KLF2/NOS1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035264" class="uri">GO:0035264</a>
</td>
<td style="text-align:left;">
multicellular organism growth
</td>
<td style="text-align:left;">
3/40
</td>
<td style="text-align:left;">
66/12294
</td>
<td style="text-align:right;">
0.0012669
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
AR/KLF2/LOC113993027
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009968" class="uri">GO:0009968</a>
</td>
<td style="text-align:left;">
negative regulation of signal transduction
</td>
<td style="text-align:left;">
2/40
</td>
<td style="text-align:left;">
34/12294
</td>
<td style="text-align:right;">
0.0054219
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
RGS8/RGS9BP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006644" class="uri">GO:0006644</a>
</td>
<td style="text-align:left;">
phospholipid metabolic process
</td>
<td style="text-align:left;">
2/40
</td>
<td style="text-align:left;">
37/12294
</td>
<td style="text-align:right;">
0.0063973
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
PLA2G4A/PLPPR1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009749" class="uri">GO:0009749</a>
</td>
<td style="text-align:left;">
response to glucose
</td>
<td style="text-align:left;">
2/40
</td>
<td style="text-align:left;">
54/12294
</td>
<td style="text-align:right;">
0.0132758
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
LOC113982296/PRLH
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001701" class="uri">GO:0001701</a>
</td>
<td style="text-align:left;">
in utero embryonic development
</td>
<td style="text-align:left;">
3/40
</td>
<td style="text-align:left;">
159/12294
</td>
<td style="text-align:right;">
0.0147806
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
AR/KLF2/LOC114001232
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008584" class="uri">GO:0008584</a>
</td>
<td style="text-align:left;">
male gonad development
</td>
<td style="text-align:left;">
2/40
</td>
<td style="text-align:left;">
70/12294
</td>
<td style="text-align:right;">
0.0216857
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
AR/TESC
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006936" class="uri">GO:0006936</a>
</td>
<td style="text-align:left;">
muscle contraction
</td>
<td style="text-align:left;">
2/40
</td>
<td style="text-align:left;">
80/12294
</td>
<td style="text-align:right;">
0.0278046
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
ITGA11/NOS1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
1/40
</td>
<td style="text-align:left;">
10/12294
</td>
<td style="text-align:right;">
0.0320755
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032026" class="uri">GO:0032026</a>
</td>
<td style="text-align:left;">
response to magnesium ion
</td>
<td style="text-align:left;">
1/40
</td>
<td style="text-align:left;">
10/12294
</td>
<td style="text-align:right;">
0.0320755
</td>
<td style="text-align:right;">
0.1544064
</td>
<td style="text-align:right;">
0.113148
</td>
<td style="text-align:left;">
TNFRSF11B
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/POM%20mean%20T%20GO-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-5" class="section level3" number="9.2.4">
<h3><span class="header-section-number">9.2.4</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 20, 0.13%
FALSE LFC &lt; 0 (down)     : 10, 0.065%
FALSE outliers [1]       : 13, 0.085%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-56-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-pom" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Strength POM</h2>
<div id="analysis-16" class="section level3" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 5, 0.033%
FALSE LFC &lt; 0 (down)     : 10, 0.065%
FALSE outliers [1]       : 48, 0.31%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/POM%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-17" class="section level3" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-57-1.png" width="1728" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-57-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="dorsomedial-intercollicular-nucleus-ico" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Dorsomedial Intercollicular nucleus (ICo)</h1>
<p><br></p>
<div id="checking-the-sampling-6" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Checking the sampling</h2>
<p>For the ICO, we started with 16854 genes but after filering we have 15069 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ico samples for 11, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.41
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.015
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/ICO%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="expression-data-exploration-9" class="section level3" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/ICO%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-7" class="section level3" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-59-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-59-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="status-ico" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Status ICO</h2>
<div id="analysis-17" class="section level3" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15069 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 1, 0.0066%
FALSE outliers [1]       : 6, 0.04%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-18" class="section level3" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-60-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-61-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-ico" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Testosterone ICO</h2>
<div id="expression-data-exploration-10" class="section level3" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-8" class="section level3" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-63-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-63-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Still need to control for Year here…</p>
<p><br></p>
</div>
<div id="analysis-18" class="section level3" number="10.3.3">
<h3><span class="header-section-number">10.3.3</span> Analysis</h3>
<p>Formula = ~ Batch + Year + mean_T</p>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 121, 0.81%
FALSE LFC &lt; 0 (down)     : 40, 0.27%
FALSE outliers [1]       : 16, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-19" class="section level3" number="10.3.4">
<h3><span class="header-section-number">10.3.4</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-66-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms by mean_T in ICO
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
8/114
</td>
<td style="text-align:left;">
38/12146
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000010
</td>
<td style="text-align:right;">
0.0000009
</td>
<td style="text-align:left;">
CFAP70/DNAH1/DNAH9/DNAI1/DNAI2/LOC113981955/LOC113998190/LOC114000425
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007018" class="uri">GO:0007018</a>
</td>
<td style="text-align:left;">
microtubule-based movement
</td>
<td style="text-align:left;">
8/114
</td>
<td style="text-align:left;">
68/12146
</td>
<td style="text-align:right;">
0.0000002
</td>
<td style="text-align:right;">
0.0000590
</td>
<td style="text-align:right;">
0.0000556
</td>
<td style="text-align:left;">
DNAH1/DNAH3/DNAH9/DNAI1/DNAI2/KIF24/LOC113998190/LOC114000425
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060285" class="uri">GO:0060285</a>
</td>
<td style="text-align:left;">
cilium-dependent cell motility
</td>
<td style="text-align:left;">
4/114
</td>
<td style="text-align:left;">
14/12146
</td>
<td style="text-align:right;">
0.0000069
</td>
<td style="text-align:right;">
0.0012381
</td>
<td style="text-align:right;">
0.0011686
</td>
<td style="text-align:left;">
CFAP44/DNAH1/DNAH3/LOC113984044
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060271" class="uri">GO:0060271</a>
</td>
<td style="text-align:left;">
cilium assembly
</td>
<td style="text-align:left;">
9/114
</td>
<td style="text-align:left;">
180/12146
</td>
<td style="text-align:right;">
0.0000481
</td>
<td style="text-align:right;">
0.0065205
</td>
<td style="text-align:right;">
0.0061545
</td>
<td style="text-align:left;">
CEP126/CFAP44/CFAP54/CFAP70/DNAI2/GSN/KIF24/LOC113982241/LOC113998190
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007368" class="uri">GO:0007368</a>
</td>
<td style="text-align:left;">
determination of left/right symmetry
</td>
<td style="text-align:left;">
5/114
</td>
<td style="text-align:left;">
50/12146
</td>
<td style="text-align:right;">
0.0001009
</td>
<td style="text-align:right;">
0.0109382
</td>
<td style="text-align:right;">
0.0103242
</td>
<td style="text-align:left;">
DNAI1/DNAI2/LOC113984044/LOC113998190/ZIC3
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904707" class="uri">GO:1904707</a>
</td>
<td style="text-align:left;">
positive regulation of vascular associated smooth muscle cell proliferation
</td>
<td style="text-align:left;">
4/114
</td>
<td style="text-align:left;">
34/12146
</td>
<td style="text-align:right;">
0.0002747
</td>
<td style="text-align:right;">
0.0238414
</td>
<td style="text-align:right;">
0.0225033
</td>
<td style="text-align:left;">
ADAMTS1/HPGD/LOC113991532/MMP2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030317" class="uri">GO:0030317</a>
</td>
<td style="text-align:left;">
flagellated sperm motility
</td>
<td style="text-align:left;">
4/114
</td>
<td style="text-align:left;">
35/12146
</td>
<td style="text-align:right;">
0.0003079
</td>
<td style="text-align:right;">
0.0238414
</td>
<td style="text-align:right;">
0.0225033
</td>
<td style="text-align:left;">
CFAP65/DNAH1/DNAI1/LOC113998190
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036158" class="uri">GO:0036158</a>
</td>
<td style="text-align:left;">
outer dynein arm assembly
</td>
<td style="text-align:left;">
3/114
</td>
<td style="text-align:left;">
18/12146
</td>
<td style="text-align:right;">
0.0005929
</td>
<td style="text-align:right;">
0.0401715
</td>
<td style="text-align:right;">
0.0379168
</td>
<td style="text-align:left;">
DNAI1/DNAI2/LOC113998190
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007288" class="uri">GO:0007288</a>
</td>
<td style="text-align:left;">
sperm axoneme assembly
</td>
<td style="text-align:left;">
3/114
</td>
<td style="text-align:left;">
20/12146
</td>
<td style="text-align:right;">
0.0008171
</td>
<td style="text-align:right;">
0.0492087
</td>
<td style="text-align:right;">
0.0464468
</td>
<td style="text-align:left;">
CFAP44/CFAP65/DNAH1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030030" class="uri">GO:0030030</a>
</td>
<td style="text-align:left;">
cell projection organization
</td>
<td style="text-align:left;">
3/114
</td>
<td style="text-align:left;">
34/12146
</td>
<td style="text-align:right;">
0.0038989
</td>
<td style="text-align:right;">
0.2113210
</td>
<td style="text-align:right;">
0.1994601
</td>
<td style="text-align:left;">
CCDC78/DNAH9/LOC114000425
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-66-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-6" class="section level3" number="10.3.5">
<h3><span class="header-section-number">10.3.5</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 28, 0.19%
FALSE LFC &lt; 0 (down)     : 47, 0.31%
FALSE outliers [1]       : 15, 0.1%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-68-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-ico" class="section level2" number="10.4">
<h2><span class="header-section-number">10.4</span> Strength ICO</h2>
<div id="analysis-19" class="section level3" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> Analysis</h3>
<p>Formula ~ Batch + Year + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 39, 0.26%
FALSE LFC &lt; 0 (down)     : 29, 0.19%
FALSE outliers [1]       : 11, 0.073%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/ICO%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-20" class="section level3" number="10.4.2">
<h3><span class="header-section-number">10.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-70-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-71-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="midbrain-central-grey-gct" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Midbrain Central Grey (GCt)</h1>
<p><br></p>
<div id="checking-the-sampling-7" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Checking the sampling</h2>
<p>For the GCT, we started with 16854 genes but after filering we have 15099 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have gct samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.76
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.18
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/GCT%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>given there is only 1 GCT sample from the pilot, I will just exclude the pilot data.</p>
<p><br></p>
<div id="expression-data-exploration-11" class="section level3" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/GCT%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-9" class="section level3" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-73-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-73-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-73-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="status-gct" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Status GCT</h2>
<div id="analysis-20" class="section level3" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15099 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 17, 0.11%
FALSE LFC &lt; 0 (down)     : 12, 0.079%
FALSE outliers [1]       : 14, 0.093%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-21" class="section level3" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-74-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-75-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-gct" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Testosterone GCT</h2>
</div>
<div id="expression-data-exploration-12" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> Expression data exploration</h2>
<p><img src="02_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-3.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><br></p>
<div id="analysis-21" class="section level3" number="11.4.1">
<h3><span class="header-section-number">11.4.1</span> Analysis</h3>
<p>Formula = ~ Batch + mean_T</p>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 4, 0.026%
FALSE LFC &lt; 0 (down)     : 13, 0.086%
FALSE outliers [1]       : 33, 0.22%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-22" class="section level3" number="11.4.2">
<h3><span class="header-section-number">11.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-77-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-77-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-7" class="section level3" number="11.4.3">
<h3><span class="header-section-number">11.4.3</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 216, 1.4%
FALSE LFC &lt; 0 (down)     : 203, 1.3%
FALSE outliers [1]       : 23, 0.15%
FALSE low counts [2]     : 874, 5.8%
FALSE (mean count &lt; 11)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-79-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-gct" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> Strength GCT</h2>
<div id="analysis-22" class="section level3" number="11.5.1">
<h3><span class="header-section-number">11.5.1</span> Analysis</h3>
<p>Formula ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 34, 0.23%
FALSE LFC &lt; 0 (down)     : 29, 0.19%
FALSE outliers [1]       : 41, 0.27%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/GCT%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-23" class="section level3" number="11.5.2">
<h3><span class="header-section-number">11.5.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-80-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-80-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="nucleus-taenia-tna" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Nucleus Taenia (TnA)</h1>
<p><br></p>
<div id="checking-the-sampling-8" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Checking the sampling</h2>
<p>For the TNA, we started with 16854 genes but after filering we have 14693 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have tna samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00075
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/TNA%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="expression-data-exploration-13" class="section level3" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/TNA%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/TNA%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-10" class="section level3" number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-82-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-82-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-82-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>PFT1 was previously identified as an outlier based on WGCNA dendrograms only. But here it’s not consistently dentified as an outlier.Also previous analysis shows that it doesn’t have a strong influence on the results, so I am going to keep it in this data. Looks like we only need to control for Batch, as it removes the effects of year on our data.</p>
<p><br></p>
</div>
</div>
<div id="status-tna" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Status TNA</h2>
<div id="analysis-23" class="section level3" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14693 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 5, 0.034%
FALSE LFC &lt; 0 (down)     : 8, 0.054%
FALSE outliers [1]       : 24, 0.16%
FALSE low counts [2]     : 1415, 9.6%
FALSE (mean count &lt; 17)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-24" class="section level3" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-83-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-83-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-tna" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Testosterone TNA</h2>
<div id="expression-data-exploration-14" class="section level3" number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/TNA%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-11" class="section level3" number="12.3.2">
<h3><span class="header-section-number">12.3.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-85-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-85-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-85-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Taking the batch effect into account seems to remove the association with Year too. Only need to include Batch in the model.</p>
</div>
<div id="analysis-24" class="section level3" number="12.3.3">
<h3><span class="header-section-number">12.3.3</span> Analysis</h3>
<p>Formula = ~ Batch + mean_T</p>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 99, 0.68%
FALSE LFC &lt; 0 (down)     : 143, 0.98%
FALSE outliers [1]       : 37, 0.25%
FALSE low counts [2]     : 280, 1.9%
FALSE (mean count &lt; 6)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-25" class="section level3" number="12.3.4">
<h3><span class="header-section-number">12.3.4</span> Gene Ontology</h3>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-87-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in TnA
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007613" class="uri">GO:0007613</a>
</td>
<td style="text-align:left;">
memory
</td>
<td style="text-align:left;">
7/210
</td>
<td style="text-align:left;">
73/11903
</td>
<td style="text-align:right;">
0.0002925
</td>
<td style="text-align:right;">
0.1888650
</td>
<td style="text-align:right;">
0.1803886
</td>
<td style="text-align:left;">
CEBPB/FGF13/LGMN/LOC113991532/PPP1R1B/PTN/SLC24A2
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061098" class="uri">GO:0061098</a>
</td>
<td style="text-align:left;">
positive regulation of protein tyrosine kinase activity
</td>
<td style="text-align:left;">
5/210
</td>
<td style="text-align:left;">
37/11903
</td>
<td style="text-align:right;">
0.0004486
</td>
<td style="text-align:right;">
0.1888650
</td>
<td style="text-align:right;">
0.1803886
</td>
<td style="text-align:left;">
ABI1/AFAP1L2/DLG3/EPHA4/LOC113988363
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0017158" class="uri">GO:0017158</a>
</td>
<td style="text-align:left;">
regulation of calcium ion-dependent exocytosis
</td>
<td style="text-align:left;">
3/210
</td>
<td style="text-align:left;">
13/11903
</td>
<td style="text-align:right;">
0.0013587
</td>
<td style="text-align:right;">
0.2874335
</td>
<td style="text-align:right;">
0.2745333
</td>
<td style="text-align:left;">
SYT12/SYT7/SYT9
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034614" class="uri">GO:0034614</a>
</td>
<td style="text-align:left;">
cellular response to reactive oxygen species
</td>
<td style="text-align:left;">
4/210
</td>
<td style="text-align:left;">
29/11903
</td>
<td style="text-align:right;">
0.0015821
</td>
<td style="text-align:right;">
0.2874335
</td>
<td style="text-align:right;">
0.2745333
</td>
<td style="text-align:left;">
AGAP3/JUN/LOC114000703/MMP9
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030048" class="uri">GO:0030048</a>
</td>
<td style="text-align:left;">
actin filament-based movement
</td>
<td style="text-align:left;">
3/210
</td>
<td style="text-align:left;">
14/11903
</td>
<td style="text-align:right;">
0.0017068
</td>
<td style="text-align:right;">
0.2874335
</td>
<td style="text-align:right;">
0.2745333
</td>
<td style="text-align:left;">
MYH9/WASL/WIPF3
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001836" class="uri">GO:0001836</a>
</td>
<td style="text-align:left;">
release of cytochrome c from mitochondria
</td>
<td style="text-align:left;">
3/210
</td>
<td style="text-align:left;">
15/11903
</td>
<td style="text-align:right;">
0.0021060
</td>
<td style="text-align:right;">
0.2955393
</td>
<td style="text-align:right;">
0.2822753
</td>
<td style="text-align:left;">
DNM1L/JUN/LOC114000703
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904707" class="uri">GO:1904707</a>
</td>
<td style="text-align:left;">
positive regulation of vascular associated smooth muscle cell proliferation
</td>
<td style="text-align:left;">
4/210
</td>
<td style="text-align:left;">
34/11903
</td>
<td style="text-align:right;">
0.0028840
</td>
<td style="text-align:right;">
0.3052112
</td>
<td style="text-align:right;">
0.2915131
</td>
<td style="text-align:left;">
JUN/LOC113991532/LOC114000703/MMP9
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060291" class="uri">GO:0060291</a>
</td>
<td style="text-align:left;">
long-term synaptic potentiation
</td>
<td style="text-align:left;">
4/210
</td>
<td style="text-align:left;">
35/11903
</td>
<td style="text-align:right;">
0.0032118
</td>
<td style="text-align:right;">
0.3052112
</td>
<td style="text-align:right;">
0.2915131
</td>
<td style="text-align:left;">
CRHR2/PTN/SLC24A2/SYT12
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007612" class="uri">GO:0007612</a>
</td>
<td style="text-align:left;">
learning
</td>
<td style="text-align:left;">
5/210
</td>
<td style="text-align:left;">
61/11903
</td>
<td style="text-align:right;">
0.0043519
</td>
<td style="text-align:right;">
0.3052112
</td>
<td style="text-align:right;">
0.2915131
</td>
<td style="text-align:left;">
FGF13/JUN/LOC114000703/PTN/SLC24A2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007052" class="uri">GO:0007052</a>
</td>
<td style="text-align:left;">
mitotic spindle organization
</td>
<td style="text-align:left;">
4/210
</td>
<td style="text-align:left;">
39/11903
</td>
<td style="text-align:right;">
0.0047763
</td>
<td style="text-align:right;">
0.3052112
</td>
<td style="text-align:right;">
0.2915131
</td>
<td style="text-align:left;">
MAP4/RMDN1/SPC25/STMN1
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-87-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-87-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-8" class="section level3" number="12.3.5">
<h3><span class="header-section-number">12.3.5</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 33, 0.23%
FALSE LFC &lt; 0 (down)     : 14, 0.096%
FALSE outliers [1]       : 22, 0.15%
FALSE low counts [2]     : 3099, 21%
FALSE (mean count &lt; 58)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-89-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-tna" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Strength TNA</h2>
<div id="analysis-25" class="section level3" number="12.4.1">
<h3><span class="header-section-number">12.4.1</span> Analysis</h3>
<p>Formula ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 22, 0.15%
FALSE LFC &lt; 0 (down)     : 34, 0.23%
FALSE outliers [1]       : 38, 0.26%
FALSE low counts [2]     : 839, 5.7%
FALSE (mean count &lt; 11)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/TNA%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-26" class="section level3" number="12.4.2">
<h3><span class="header-section-number">12.4.2</span> Gene Ontology</h3>
<p>There are no enriched terms here.</p>
<p><br><br></p>
</div>
</div>
</div>
<div id="arcopallium-intermedium-ai" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Arcopallium intermedium (Ai)</h1>
<p><br></p>
<div id="checking-the-sampling-9" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Checking the sampling</h2>
<p>For the AI, we started with 16854 genes but after filering we have 14598 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ai samples for 11, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.54
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.0027
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-15" class="section level3" number="13.1.1">
<h3><span class="header-section-number">13.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/AI%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/AI%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><br></p>
</div>
<div id="corrections-applied-12" class="section level3" number="13.1.2">
<h3><span class="header-section-number">13.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-92-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-92-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-92-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Removing the batch effect effectively controls for Year.</p>
<p><br></p>
</div>
</div>
<div id="status-ai" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Status AI</h2>
<div id="analysis-26" class="section level3" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14598 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 11, 0.075%
FALSE LFC &lt; 0 (down)     : 7, 0.048%
FALSE outliers [1]       : 10, 0.069%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 1)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-27" class="section level3" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-93-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-94-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-ai" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Testosterone AI</h2>
<div id="expression-data-exploration-16" class="section level3" number="13.3.1">
<h3><span class="header-section-number">13.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>No association with year.</p>
<p><br></p>
</div>
<div id="analysis-27" class="section level3" number="13.3.2">
<h3><span class="header-section-number">13.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 13, 0.089%
FALSE LFC &lt; 0 (down)     : 30, 0.21%
FALSE outliers [1]       : 25, 0.17%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-28" class="section level3" number="13.3.3">
<h3><span class="header-section-number">13.3.3</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-96-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-97-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-9" class="section level3" number="13.3.4">
<h3><span class="header-section-number">13.3.4</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 4, 0.027%
FALSE LFC &lt; 0 (down)     : 4, 0.027%
FALSE outliers [1]       : 493, 3.4%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-99-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-ai" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Strength AI</h2>
<div id="analysis-28" class="section level3" number="13.4.1">
<h3><span class="header-section-number">13.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 18, 0.12%
FALSE LFC &lt; 0 (down)     : 16, 0.11%
FALSE outliers [1]       : 21, 0.14%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/AI%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-29" class="section level3" number="13.4.2">
<h3><span class="header-section-number">13.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-100-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="lateral-septum-ls" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Lateral Septum (Ls)</h1>
<p><br></p>
<div id="checking-the-sampling-10" class="section level2" number="14.1">
<h2><span class="header-section-number">14.1</span> Checking the sampling</h2>
<p>For the LS, we started with 16854 genes but after filering we have 14899 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have ls samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.0022
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<div id="expression-data-exploration-17" class="section level3" number="14.1.1">
<h3><span class="header-section-number">14.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/LS%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-13" class="section level3" number="14.1.2">
<h3><span class="header-section-number">14.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-102-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-102-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-102-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>PFT2 was previously identified as an outlier, but using the quantitative threshold and controlling for batch it doesn’t come out as different. Controlling for Batch adequately controls for Year.</p>
</div>
</div>
<div id="status-ls" class="section level2" number="14.2">
<h2><span class="header-section-number">14.2</span> Status LS</h2>
<div id="analysis-29" class="section level3" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14899 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 16, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 1)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-30" class="section level3" number="14.2.2">
<h3><span class="header-section-number">14.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-103-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-103-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-ls" class="section level2" number="14.3">
<h2><span class="header-section-number">14.3</span> Testosterone LS</h2>
<div id="expression-data-exploration-18" class="section level3" number="14.3.1">
<h3><span class="header-section-number">14.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-30" class="section level3" number="14.3.2">
<h3><span class="header-section-number">14.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 62, 0.42%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-31" class="section level3" number="14.3.3">
<h3><span class="header-section-number">14.3.3</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-105-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-105-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-10" class="section level3" number="14.3.4">
<h3><span class="header-section-number">14.3.4</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 14, 0.094%
FALSE LFC &lt; 0 (down)     : 25, 0.17%
FALSE outliers [1]       : 29, 0.2%
FALSE low counts [2]     : 572, 3.9%
FALSE (mean count &lt; 8)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-107-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-ls" class="section level2" number="14.4">
<h2><span class="header-section-number">14.4</span> Strength LS</h2>
<div id="analysis-31" class="section level3" number="14.4.1">
<h3><span class="header-section-number">14.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 17, 0.11%
FALSE LFC &lt; 0 (down)     : 81, 0.55%
FALSE outliers [1]       : 53, 0.36%
FALSE low counts [2]     : 6849, 46%
FALSE (mean count &lt; 303)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/LS%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-32" class="section level3" number="14.4.2">
<h3><span class="header-section-number">14.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-109-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="02_DESeq2_files/figure-html/unnamed-chunk-110-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in LS
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0048538" class="uri">GO:0048538</a>
</td>
<td style="text-align:left;">
thymus development
</td>
<td style="text-align:left;">
3/83
</td>
<td style="text-align:left;">
32/12010
</td>
<td style="text-align:right;">
0.0013661
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
LMO4/PSEN1/ZMPSTE24
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000244" class="uri">GO:0000244</a>
</td>
<td style="text-align:left;">
spliceosomal tri-snRNP complex assembly
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
10/12010
</td>
<td style="text-align:right;">
0.0020484
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
PRPF8/SART3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050927" class="uri">GO:0050927</a>
</td>
<td style="text-align:left;">
positive regulation of positive chemotaxis
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
11/12010
</td>
<td style="text-align:right;">
0.0024924
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
S1PR1/VEGFA
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048593" class="uri">GO:0048593</a>
</td>
<td style="text-align:left;">
camera-type eye morphogenesis
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
13/12010
</td>
<td style="text-align:right;">
0.0035032
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
SHROOM2/VEGFA
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030500" class="uri">GO:0030500</a>
</td>
<td style="text-align:left;">
regulation of bone mineralization
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
14/12010
</td>
<td style="text-align:right;">
0.0040687
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
S1PR1/ZMPSTE24
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000009" class="uri">GO:2000009</a>
</td>
<td style="text-align:left;">
negative regulation of protein localization to cell surface
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
14/12010
</td>
<td style="text-align:right;">
0.0040687
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
GPM6B/LEPROT
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007193" class="uri">GO:0007193</a>
</td>
<td style="text-align:left;">
adenylate cyclase-inhibiting G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
3/83
</td>
<td style="text-align:left;">
49/12010
</td>
<td style="text-align:right;">
0.0046643
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
LOC113992576/NPY1R/S1PR1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090557" class="uri">GO:0090557</a>
</td>
<td style="text-align:left;">
establishment of endothelial intestinal barrier
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
15/12010
</td>
<td style="text-align:right;">
0.0046737
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
RAB1A/RAP2C
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031532" class="uri">GO:0031532</a>
</td>
<td style="text-align:left;">
actin cytoskeleton reorganization
</td>
<td style="text-align:left;">
3/83
</td>
<td style="text-align:left;">
55/12010
</td>
<td style="text-align:right;">
0.0064477
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
LOC113988212/RALA/S1PR1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032402" class="uri">GO:0032402</a>
</td>
<td style="text-align:left;">
melanosome transport
</td>
<td style="text-align:left;">
2/83
</td>
<td style="text-align:left;">
18/12010
</td>
<td style="text-align:right;">
0.0067194
</td>
<td style="text-align:right;">
0.2756505
</td>
<td style="text-align:right;">
0.2636709
</td>
<td style="text-align:left;">
BLOC1S5/RAB1A
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-110-2.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-110-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
<div id="bed-nucleus-of-the-stria-terminalis-bstm" class="section level1" number="15">
<h1><span class="header-section-number">15</span> Bed Nucleus of the Stria Terminalis (BSTm)</h1>
<p><br></p>
<div id="checking-the-sampling-11" class="section level2" number="15.1">
<h2><span class="header-section-number">15.1</span> Checking the sampling</h2>
<p>For the BSTm, we started with 16854 genes but after filering we have 15031 genes. Before I go into more detail, I am going to check that the sampling is equal across batches with respect to our variables of interest. With respect to Status we have bstm samples for 12, with respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00055
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-19" class="section level3" number="15.1.1">
<h3><span class="header-section-number">15.1.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/BSTm%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-14" class="section level3" number="15.1.2">
<h3><span class="header-section-number">15.1.2</span> Corrections applied</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-112-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-112-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-112-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Batch controls effectively for Year.</p>
<p><br></p>
</div>
</div>
<div id="status-bstm" class="section level2" number="15.2">
<h2><span class="header-section-number">15.2</span> Status BSTm</h2>
<div id="analysis-32" class="section level3" number="15.2.1">
<h3><span class="header-section-number">15.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15031 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 6, 0.04%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 5, 0.033%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20Status%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-33" class="section level3" number="15.2.2">
<h3><span class="header-section-number">15.2.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-114-1.png" width="1728" style="display: block; margin: auto;" /> <img src="02_DESeq2_files/figure-html/unnamed-chunk-115-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="testosterone-bstm" class="section level2" number="15.3">
<h2><span class="header-section-number">15.3</span> Testosterone BSTm</h2>
<div id="expression-data-exploration-20" class="section level3" number="15.3.1">
<h3><span class="header-section-number">15.3.1</span> Expression data exploration</h3>
<p><img src="02_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-33" class="section level3" number="15.3.2">
<h3><span class="header-section-number">15.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 32, 0.21%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20mean_T%20Geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-34" class="section level3" number="15.3.3">
<h3><span class="header-section-number">15.3.3</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-117-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-117-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="mean-t-x-status-11" class="section level3" number="15.3.4">
<h3><span class="header-section-number">15.3.4</span> mean T x Status</h3>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 11, 0.073%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-119-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="strength-bstm" class="section level2" number="15.4">
<h2><span class="header-section-number">15.4</span> Strength BSTm</h2>
<div id="analysis-34" class="section level3" number="15.4.1">
<h3><span class="header-section-number">15.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 8, 0.053%
FALSE LFC &lt; 0 (down)     : 7, 0.047%
FALSE outliers [1]       : 26, 0.17%
FALSE low counts [2]     : 870, 5.8%
FALSE (mean count &lt; 11)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="02_DESeq2_files/figure-html/BSTm%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-35" class="section level3" number="15.4.2">
<h3><span class="header-section-number">15.4.2</span> Gene Ontology</h3>
<p><img src="02_DESeq2_files/figure-html/unnamed-chunk-121-1.png" width="672" style="display: block; margin: auto;" /><img src="02_DESeq2_files/figure-html/unnamed-chunk-121-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Volcano plots and similarity of gene expression between tissues is in another Rmarkdown on this site.</p>
<pre><code>FALSE R version 4.0.2 (2020-06-22)
FALSE Platform: x86_64-w64-mingw32/x64 (64-bit)
FALSE Running under: Windows 10 x64 (build 19044)
FALSE 
FALSE Matrix products: default
FALSE 
FALSE locale:
FALSE [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
FALSE [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
FALSE [5] LC_TIME=English_Australia.1252    
FALSE 
FALSE attached base packages:
FALSE  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
FALSE  [8] datasets  methods   base     
FALSE 
FALSE other attached packages:
FALSE  [1] nlme_3.1-148                GGally_2.1.0               
FALSE  [3] viridis_0.6.1               viridisLite_0.4.0          
FALSE  [5] rrvgo_1.0.2                 UpSetR_1.4.0               
FALSE  [7] limma_3.44.3                flashClust_1.01-2          
FALSE  [9] org.Hs.eg.db_3.11.4         AnnotationDbi_1.50.0       
FALSE [11] clusterProfiler_3.16.1      WGCNA_1.69                 
FALSE [13] fastcluster_1.1.25          dynamicTreeCut_1.63-1      
FALSE [15] wesanderson_0.3.6           VennDiagram_1.6.20         
FALSE [17] futile.logger_1.4.3         kableExtra_1.3.1           
FALSE [19] DESeq2_1.28.1               SummarizedExperiment_1.18.1
FALSE [21] DelayedArray_0.14.0         matrixStats_0.56.0         
FALSE [23] Biobase_2.48.0              GenomicRanges_1.40.0       
FALSE [25] GenomeInfoDb_1.24.0         IRanges_2.22.2             
FALSE [27] S4Vectors_0.26.1            BiocGenerics_0.34.0        
FALSE [29] lubridate_1.8.0             stringr_1.4.0              
FALSE [31] fdrtool_1.2.15              PCAtools_2.0.0             
FALSE [33] cowplot_1.0.0               lattice_0.20-41            
FALSE [35] ggrepel_0.8.2               ggpubr_0.4.0               
FALSE [37] gridExtra_2.3               ggplot2_3.3.5              
FALSE [39] reshape2_1.4.4              plyr_1.8.6                 
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] ragg_1.2.1                tidyr_1.1.4              
FALSE   [3] acepack_1.4.1             bit64_0.9-7              
FALSE   [5] knitr_1.29                irlba_2.3.3              
FALSE   [7] data.table_1.13.0         rpart_4.1-15             
FALSE   [9] RCurl_1.98-1.2            doParallel_1.0.15        
FALSE  [11] generics_0.0.2            snow_0.4-3               
FALSE  [13] preprocessCore_1.50.0     lambda.r_1.2.4           
FALSE  [15] RSQLite_2.2.0             europepmc_0.4            
FALSE  [17] bit_4.0.4                 enrichplot_1.8.1         
FALSE  [19] webshot_0.5.2             xml2_1.3.2               
FALSE  [21] httpuv_1.5.4              assertthat_0.2.1         
FALSE  [23] xfun_0.29                 hms_1.1.1                
FALSE  [25] jquerylib_0.1.4           evaluate_0.14            
FALSE  [27] promises_1.1.1            fansi_0.4.1              
FALSE  [29] progress_1.2.2            readxl_1.3.1             
FALSE  [31] igraph_1.2.6              DBI_1.1.0                
FALSE  [33] geneplotter_1.66.0        htmlwidgets_1.5.3        
FALSE  [35] reshape_0.8.8             purrr_0.3.4              
FALSE  [37] ellipsis_0.3.2            dplyr_1.0.7              
FALSE  [39] backports_1.1.7           annotate_1.66.0          
FALSE  [41] gridBase_0.4-7            vctrs_0.3.8              
FALSE  [43] abind_1.4-5               withr_2.4.3              
FALSE  [45] ggforce_0.3.2             triebeard_0.3.0          
FALSE  [47] treemap_2.4-2             checkmate_2.0.0          
FALSE  [49] prettyunits_1.1.1         cluster_2.1.0            
FALSE  [51] DOSE_3.14.0               crayon_1.4.2             
FALSE  [53] genefilter_1.70.0         pkgconfig_2.0.3          
FALSE  [55] slam_0.1-47               labeling_0.3             
FALSE  [57] tweenr_1.0.1              wordcloud_2.6            
FALSE  [59] nnet_7.3-14               rlang_0.4.12             
FALSE  [61] lifecycle_1.0.1           downloader_0.4           
FALSE  [63] rsvd_1.0.3                cellranger_1.1.0         
FALSE  [65] polyclip_1.10-0           Matrix_1.2-18            
FALSE  [67] urltools_1.7.3            carData_3.0-4            
FALSE  [69] base64enc_0.1-3           ggridges_0.5.3           
FALSE  [71] pheatmap_1.0.12           png_0.1-7                
FALSE  [73] bitops_1.0-6              blob_1.2.1               
FALSE  [75] DelayedMatrixStats_1.10.0 qvalue_2.20.0            
FALSE  [77] jpeg_0.1-8.1              rstatix_0.6.0            
FALSE  [79] gridGraphics_0.5-1        ggsignif_0.6.0           
FALSE  [81] scales_1.1.1              memoise_1.1.0            
FALSE  [83] magrittr_2.0.1            zlibbioc_1.34.0          
FALSE  [85] compiler_4.0.2            scatterpie_0.1.5         
FALSE  [87] dqrng_0.2.1               RColorBrewer_1.1-2       
FALSE  [89] XVector_0.28.0            htmlTable_2.0.1          
FALSE  [91] formatR_1.8               Formula_1.2-3            
FALSE  [93] MASS_7.3-51.6             tidyselect_1.1.0         
FALSE  [95] stringi_1.4.6             forcats_0.5.1            
FALSE  [97] textshaping_0.3.6         highr_0.8                
FALSE  [99] yaml_2.2.1                GOSemSim_2.14.2          
FALSE [101] BiocSingular_1.4.0        locfit_1.5-9.4           
FALSE [103] latticeExtra_0.6-29       fastmatch_1.1-0          
FALSE [105] tools_4.0.2               rio_0.5.16               
FALSE [107] rstudioapi_0.13           foreach_1.5.0            
FALSE [109] foreign_0.8-80            farver_2.0.3             
FALSE [111] ggraph_2.0.5              digest_0.6.25            
FALSE [113] rvcheck_0.1.8             BiocManager_1.30.10      
FALSE [115] shiny_1.5.0               Rcpp_1.0.7               
FALSE [117] car_3.0-8                 broom_0.7.10             
FALSE [119] later_1.1.0.1             httr_1.4.2               
FALSE [121] colorspace_1.4-1          rvest_1.0.2              
FALSE [123] XML_3.99-0.5              splines_4.0.2            
FALSE [125] graphlayouts_0.7.1        ggplotify_0.0.5          
FALSE [127] systemfonts_1.0.3         xtable_1.8-4             
FALSE [129] jsonlite_1.7.2            futile.options_1.0.1     
FALSE [131] tidygraph_1.2.0           NLP_0.2-1                
FALSE [133] R6_2.4.1                  tm_0.7-8                 
FALSE [135] Hmisc_4.4-0               pillar_1.6.4             
FALSE [137] htmltools_0.5.2           mime_0.9                 
FALSE [139] glue_1.5.1                fastmap_1.1.0            
FALSE [141] BiocParallel_1.22.0       codetools_0.2-16         
FALSE [143] fgsea_1.14.0              utf8_1.1.4               
FALSE [145] tibble_3.1.6              curl_4.3                 
FALSE [147] zip_2.0.4                 GO.db_3.11.4             
FALSE [149] openxlsx_4.1.5            survival_3.2-3           
FALSE [151] rmarkdown_2.11            munsell_0.5.0            
FALSE [153] DO.db_2.9                 GenomeInfoDbData_1.2.3   
FALSE [155] iterators_1.0.12          impute_1.62.0            
FALSE [157] haven_2.3.1               gtable_0.3.0</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
