<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive change November 2023" />


<title>DESeq2 and PCA Analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Supp 1: Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_1_Checks_PCA.html">Supp 1 &amp; 2: Checks and System-wide PCA</a>
    </li>
    <li>
      <a href="./02_2_WGCNA_wholebrain.html">Supp 2: Whole-brain WGCNA</a>
    </li>
    <li>
      <a href="./02_3_WGCNA_HPG.html">Supp 2: HPG WGCNA</a>
    </li>
    <li>
      <a href="./03_DESeq2.html">Supp 3: Per Tissue Differential Expression</a>
    </li>
    <li>
      <a href="./04_candidate_gene_analysis.html">Supp 4: Candidate Gene Analysis</a>
    </li>
    <li>
      <a href="./05_WGCNA_single_tissue.html">Supp 5: Within Tissue WGCNA </a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Supp 6: Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DESeq2 and PCA Analysis</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male
cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive change November 2023</h4>
<h4 class="date">Last Knit “2024-01-04”</h4>

</div>


<p>Our primary aim is to explore whether among-individual variation in T
modulated cooperative behaviour is associated with differences in gene
expression. We have 5 main traits that were measured.</p>
<ul>
<li>Status: Territorial or Floater</li>
<li>Testosterone concentration
<ul>
<li><em>mean_T</em> is best estimate of among individual variation in T
across all birds (’r summary(behav$mean_T)`). This measure is repeatable
(Ryder et al 2020) within individuals and could be considered a hormonal
phenotype.</li>
</ul></li>
<li>Strength</li>
<li>Mean number of social interactions per 10 hr day as detected by
telemetry.</li>
</ul>
<p>Here, I take the filtered dataset from the QC process, and run a
whole brain analysis, as well as individual tissue analysis. I also
explore the relationships between our explanatory variables.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_key_Parsed_ReplicatesRemoved.csv&quot;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>behav<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_unfiltered/PIPFIL_T_and_behav_data.csv&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">rownames</span>(key)<span class="ot">&lt;-</span> key<span class="sc">$</span>X</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>key<span class="sc">$</span>Color_ID<span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;/&quot;</span>,<span class="st">&quot;&quot;</span>, key<span class="sc">$</span>Color_ID)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;Color_ID&quot;</span><span class="ot">=</span><span class="st">&quot;colorID&quot;</span>))</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>behav<span class="sc">$</span>status<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(behav<span class="sc">$</span>status, <span class="fu">c</span>(<span class="st">&quot;floa&quot;</span><span class="ot">=</span><span class="st">&quot;floater&quot;</span>, <span class="st">&quot;terr&quot;</span><span class="ot">=</span><span class="st">&quot;territorial&quot;</span>))</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">merge</span>(key, behav, <span class="at">by=</span><span class="st">&quot;colorID&quot;</span>)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">is.na</span>(key_behav<span class="sc">$</span>last_behav),]</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="co">#create a data.frame with all of the observations. </span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> key[<span class="sc">!</span>key<span class="sc">$</span>X <span class="sc">%in%</span> key_behav<span class="sc">$</span>X,]</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>cols_not_key<span class="ot">&lt;-</span> <span class="fu">colnames</span>(key_behav)[<span class="sc">!</span><span class="fu">colnames</span>(key_behav) <span class="sc">%in%</span> <span class="fu">colnames</span>(key)]</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>cols_not_key_df<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(not_in_behav), <span class="at">ncol =</span> <span class="fu">length</span>(cols_not_key)))</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="fu">colnames</span>(cols_not_key_df)<span class="ot">&lt;-</span> cols_not_key</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> <span class="fu">cbind</span>(not_in_behav, cols_not_key_df)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">rbind</span>(key_behav, not_in_behav)</span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>X</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>key_behav<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Tissue, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;GON&quot;</span>,<span class="st">&quot;PIT&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;AI&quot;</span>,<span class="st">&quot;TNA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(<span class="fu">rownames</span>(key_behav)),]</span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Class)</span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">revalue</span>(key_behav<span class="sc">$</span>Class, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;SCB floater&quot;</span><span class="ot">=</span><span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;DCB floater &quot;</span><span class="ot">=</span><span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;DCB territorial&quot;</span><span class="ot">=</span><span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Class, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>sampleID</span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(key_behav<span class="sc">$</span>sampleID),]</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a>key_behav<span class="sc">$</span>Year<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Year)</span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a>key_behav<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Batch)</span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a>key_behav<span class="sc">$</span>Status<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(key_behav<span class="sc">$</span>Status)</span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>key_behav_unique<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">duplicated</span>(key_behav<span class="sc">$</span>colorID),]</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="co">#read the raw count data</span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a>data<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv&quot;</span>)</span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a>data<span class="sc">$</span>X[data<span class="sc">$</span>X<span class="sc">==</span><span class="st">&quot;LOC113993669&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CYP19A1&quot;</span></span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a>data<span class="sc">$</span>X[data<span class="sc">$</span>X<span class="sc">==</span><span class="st">&quot;LOC113983511&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;OXT&quot;</span></span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a>data<span class="sc">$</span>X[data<span class="sc">$</span>X<span class="sc">==</span><span class="st">&quot;LOC113983498&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVP&quot;</span></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a>data<span class="sc">$</span>X[data<span class="sc">$</span>X<span class="sc">==</span><span class="st">&quot;LOC113982601&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVPR2&quot;</span></span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a><span class="fu">rownames</span>(data)<span class="ot">&lt;-</span> data<span class="sc">$</span>X</span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a>data<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a></span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a><span class="do">### Gene Ontology</span></span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/pfil_GO_key_raw.csv&quot;</span>)</span>
<span id="cb1-50"><a href="#cb1-50" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_terms, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-51"><a href="#cb1-51" tabindex="-1"></a>go2gene_bp<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;P&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb1-52"><a href="#cb1-52" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/ontology_obo_out.csv&quot;</span>)</span>
<span id="cb1-53"><a href="#cb1-53" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_obo, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;id&quot;</span><span class="ot">=</span><span class="st">&quot;GO_ID&quot;</span>))</span>
<span id="cb1-54"><a href="#cb1-54" tabindex="-1"></a>go2name_bp<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;biological_process&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span>
<span id="cb1-55"><a href="#cb1-55" tabindex="-1"></a></span>
<span id="cb1-56"><a href="#cb1-56" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" tabindex="-1"></a><span class="do">## annotating the LOC no</span></span>
<span id="cb1-58"><a href="#cb1-58" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/Maggies_annotations_modifiedR.csv&quot;</span>)</span>
<span id="cb1-59"><a href="#cb1-59" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(genes_key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb1-60"><a href="#cb1-60" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113993669&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;CYP19A1&quot;</span></span>
<span id="cb1-61"><a href="#cb1-61" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113983511&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;OXT&quot;</span></span>
<span id="cb1-62"><a href="#cb1-62" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113983498&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVP&quot;</span></span>
<span id="cb1-63"><a href="#cb1-63" tabindex="-1"></a>genes_key<span class="sc">$</span>gene[genes_key<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;LOC113982601&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;AVPR2&quot;</span></span>
<span id="cb1-64"><a href="#cb1-64" tabindex="-1"></a>data_genes<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span><span class="fu">rownames</span>(data))</span>
<span id="cb1-65"><a href="#cb1-65" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">merge</span>(data_genes, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-66"><a href="#cb1-66" tabindex="-1"></a><span class="co">#genes_key&lt;- genes_key[which(grepl(&quot;LOC[0-9]+&quot;,genes_key$gene)),]</span></span>
<span id="cb1-67"><a href="#cb1-67" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" tabindex="-1"></a>genes_key<span class="sc">$</span>display_gene_ID<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;LOC[0-9]+&quot;</span>, genes_key<span class="sc">$</span>gene) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;LOC[0-9]+&quot;</span>,genes_key<span class="sc">$</span>best_anno) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(genes_key<span class="sc">$</span>best_anno), <span class="fu">paste0</span>(genes_key<span class="sc">$</span>gene,<span class="st">&quot; (&quot;</span>,genes_key<span class="sc">$</span>best_anno,<span class="st">&quot;)&quot;</span>), <span class="fu">as.character</span>(genes_key<span class="sc">$</span>gene))</span>
<span id="cb1-69"><a href="#cb1-69" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> genes_key[,<span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;best_anno&quot;</span>,<span class="st">&quot;display_gene_ID&quot;</span>)]</span>
<span id="cb1-70"><a href="#cb1-70" tabindex="-1"></a>  </span>
<span id="cb1-71"><a href="#cb1-71" tabindex="-1"></a>candidates2<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AR&quot;</span>, <span class="st">&quot;SRD5A2&quot;</span>, <span class="st">&quot;CYP19A1&quot;</span>, <span class="st">&quot;ESR1&quot;</span>, <span class="st">&quot;ESR2&quot;</span>, <span class="st">&quot;PGR&quot;</span>,<span class="st">&quot;GNRH1&quot;</span>, <span class="st">&quot;PRL&quot;</span>,<span class="st">&quot;PRLR&quot;</span>,<span class="st">&quot;VIP&quot;</span>, <span class="st">&quot;VIPR1&quot;</span>, <span class="st">&quot;VIPR2&quot;</span>,<span class="st">&quot;OXT&quot;</span>, <span class="st">&quot;OXTR&quot;</span>, <span class="st">&quot;AVP&quot;</span>, <span class="st">&quot;AVPR1A&quot;</span>, <span class="st">&quot;AVPR1B&quot;</span>, <span class="st">&quot;AVPR2&quot;</span>)</span></code></pre></div>
<p>I have to run each tissue separately in part because of linear
combinations of batch variables and tissues (some tissues were sequenced
only on a single flow cell) that means DESeq2 throws a “model not full
rank” error. But also we are interested in the within-tissue gene
expression.</p>
<p><br><br></p>
<div id="gonads-gon" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Gonads (GON)</h1>
<p>I will use the first two tissues to demonstrate the processes applied
across all tissues, but code echoing will be suppressed in future
tissues so as to save space.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>tissue<span class="ot">=</span><span class="st">&quot;GON&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">==</span>tissue)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(gon_key)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(gon_key)]</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(gon_data)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#remove genes with less than 5 reads</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, mean)</span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">5</span>,]</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(gon_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span></code></pre></div>
<p>Before filtering we had 16854 genes, after filtering for mean read
count and excluding genes where &gt;50% of samples had a count of 0 we
are left with 16169</p>
<div id="checking-the-sampling" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Checking the
sampling</h2>
<p>Before I go into the analyses, I am going to check that the sampling
is equal across batches with respect to our variables of interest. With
respect to Status we have gonad samples for 16, with respect to T and
the tracking data we have 12 samples.</p>
<p>Here, we will run all the data to check for DEGs against status, then
subset the data to only include those individuals with T and behavioural
data.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>fsb<span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Batch))</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Batch), <span class="at">caption=</span><span class="fu">paste0</span>(<span class="st">&quot;Status across sequencing runs, fisher test p=&quot;</span>, fsb<span class="sc">$</span>p.value)) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>fsy<span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Year))</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">table</span>(gon_key<span class="sc">$</span>Status, gon_key<span class="sc">$</span>Year), <span class="at">caption=</span><span class="fu">paste0</span>(<span class="st">&quot;Status across sampling years, fisher test p=&quot;</span>, fsy<span class="sc">$</span>p.value)) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sampling years, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
</div>
<div id="status-gon" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Status GON</h2>
<p>First, I will run through a data checking process. I will check for
outliers, and then find out how our potential nuisance variables effect
our gene expression data. If they have effects they will be incorporated
into the final DESeq2 models.</p>
<p><br></p>
<div id="expression-data-exploration" class="section level3"
number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Expression data
exploration</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_key, <span class="at">design=</span> <span class="sc">~</span> Status)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),] <span class="co">#remove any genes that didn&#39;t converge. </span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dd)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co">#plot a PCA of our samples to look for suspicious samples and any obvious patterning with respect to our interest variables</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>vsd_data<span class="ot">&lt;-</span> <span class="fu">getVarianceStabilizedData</span>(dd)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(vsd_data, <span class="at">metadata =</span> gon_key)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>a<span class="ot">&lt;-</span> <span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_key<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;Status&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Batch&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;Gonads all samples PCA&quot;</span>) <span class="sc">+</span> peri_figure</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>a</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="do">## check to see how our interest and nuisance variables are correlated to the PC axes. This will help decide on model form for DESeq.</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>b<span class="ot">&lt;-</span> <span class="fu">eigencorplot</span>(p, <span class="at">metavars =</span> <span class="fu">c</span>(<span class="st">&quot;time_kill&quot;</span>,<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Status&quot;</span>))</span></code></pre></div>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>b</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#use WGCNA connectivity measure to identify outliers.</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#gsg&lt;- goodSamplesGenes(datExpr0, verbose=3)</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#gsg$allOK</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(datExpr0),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor)</span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20set%20up-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Based on the eigencor plot, I will include Batch as a variable in my
analyses, as it is associated with PC4.</p>
<p><br></p>
<div id="corrections-applied" class="section level4" number="1.2.1.1">
<h4><span class="header-section-number">1.2.1.1</span> Corrections
applied</h4>
<p>Also let’s see how the data look when we account for Batch in our
model by using limma::RemoveBatchEffect()</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co">#remove batch effect</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>mat<span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(vsd_data, gon_key<span class="sc">$</span>Batch)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(mat, <span class="at">metadata =</span> gon_key)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_key<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;Status&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Batch&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;PCA Gonads removeBatchEffect(Batch)&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20removeBatchEffect-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co">#use this if Batch and Year are both pinged in the previous section. </span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">eigencorplot</span>(p, <span class="at">metavars =</span> <span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Status&quot;</span>))</span></code></pre></div>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/GON%20removeBatchEffect-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>wgcnadata<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(mat))</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>traits<span class="ot">&lt;-</span> gon_key[, <span class="fu">c</span>(<span class="st">&quot;Batch&quot;</span>,  <span class="st">&quot;Year&quot;</span>)]</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>traits<span class="sc">$</span>pilot<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;pilot&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>traits<span class="sc">$</span>Run1<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;run1&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>traits<span class="sc">$</span>Run2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Batch<span class="sc">==</span><span class="st">&quot;run2&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-8"><a href="#cb14-8" tabindex="-1"></a>traits<span class="sc">$</span>yr2015<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2015&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-9"><a href="#cb14-9" tabindex="-1"></a>traits<span class="sc">$</span>yr2017<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2017&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-10"><a href="#cb14-10" tabindex="-1"></a>traits<span class="sc">$</span>yr2018<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(traits<span class="sc">$</span>Year<span class="sc">==</span><span class="st">&quot;2018&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb14-11"><a href="#cb14-11" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" tabindex="-1"></a>traits<span class="ot">&lt;-</span> traits[,<span class="sc">-</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)] </span>
<span id="cb14-13"><a href="#cb14-13" tabindex="-1"></a></span>
<span id="cb14-14"><a href="#cb14-14" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(wgcnadata),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb14-15"><a href="#cb14-15" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb14-16"><a href="#cb14-16" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb14-17"><a href="#cb14-17" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb14-18"><a href="#cb14-18" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb14-19"><a href="#cb14-19" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb14-20"><a href="#cb14-20" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb14-21"><a href="#cb14-21" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb14-22"><a href="#cb14-22" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb14-23"><a href="#cb14-23" tabindex="-1"></a>traitColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="fu">numbers2colors</span>(traits,<span class="at">signed=</span><span class="cn">FALSE</span>))</span>
<span id="cb14-24"><a href="#cb14-24" tabindex="-1"></a><span class="fu">dimnames</span>(traitColors)[[<span class="dv">2</span>]]<span class="ot">=</span><span class="fu">paste</span>(<span class="fu">names</span>(traits))</span>
<span id="cb14-25"><a href="#cb14-25" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor,traitColors)</span>
<span id="cb14-26"><a href="#cb14-26" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb14-28"><a href="#cb14-28" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb14-29"><a href="#cb14-29" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20removeBatchEffect-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="analysis" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Analysis</h3>
<p>Now run DEseq with the updated model form and extract DEGs.</p>
<p>Formula ~ Batch + Status.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>varname<span class="ot">=</span><span class="st">&quot;Status&quot;</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span> Batch <span class="sc">+</span> Status)</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16169 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.012%
FALSE LFC &lt; 0 (down)     : 0, 0%
FALSE outliers [1]       : 18, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;MA-plot&quot;</span>,tissue,<span class="st">&quot;exp&quot;</span>,<span class="fu">design</span>(dd)))</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20Status2-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">design</span>(dd))</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">paste0</span>(des, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res)</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>out_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(out_res)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">merge</span>(out_res, genes_key,<span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> out_res[<span class="fu">order</span>(out_res<span class="sc">$</span>pvalue),]</span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>out_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> tissue</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>out_res<span class="sc">$</span>design<span class="ot">&lt;-</span> des</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>out_res<span class="sc">$</span>fdrtool<span class="ot">&lt;-</span> <span class="st">&quot;no&quot;</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a><span class="fu">write.csv</span>(out_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_Status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">gene_plots</span>(out_res,<span class="fu">c</span>(<span class="st">&quot;Status&quot;</span>,<span class="st">&quot;Class&quot;</span>),<span class="dv">6</span>, status_cols)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>g</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20status%20geneplotting-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology" class="section level3" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Gene Ontology</h3>
<p>Conduct Gene Set enrichment analysis (GSEA) when there are few
significant differentially expressed genes. This will leverage the whole
list. Then, I will also conduct foreground background testing to get an
explicit view of either the top genes after FDR correction (q&lt;0.1),
or the top genes without FDR correction (p&lt;0.05)</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">res_gsea_bp</span>(out_res)</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>gseaplot</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20Status%20GSEA-1.png" width="1728" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">write.csv</span>(gsearesult, <span class="at">file=</span><span class="fu">paste0</span>(<span class="st">&quot;../DE_results/GSEA_BP_&quot;</span>,tissue,<span class="st">&quot;_&quot;</span>,varname,<span class="st">&quot;.csv&quot;</span>), <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">res_go_bp</span>(out_res)</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>go_plot_result</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20Status%20enrichment-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(enrich_results[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">1</span>], <span class="at">caption=</span><span class="fu">paste0</span>(<span class="st">&quot;Top 10 enriched GO BP in &quot;</span>,tissue, <span class="st">&quot; according to &quot;</span>,varname)) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GON according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
21/817
</td>
<td style="text-align:left;">
79/12553
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000333
</td>
<td style="text-align:right;">
0.0000326
</td>
<td style="text-align:left;">
CPAMD8/LOC113992168/KNG1/SERPINB1/SERPINA10/SPP2/ITIH4/FETUB/LOC113991485/ITIH2/VTN/LOC113989405/C3/SERPINC1/SERPINF2/AGT/BIRC6/HRG/C5/USP14/SERPINF1
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030449" class="uri">GO:0030449</a>
</td>
<td style="text-align:left;">
regulation of complement activation
</td>
<td style="text-align:left;">
13/817
</td>
<td style="text-align:left;">
37/12553
</td>
<td style="text-align:right;">
0.0000003
</td>
<td style="text-align:right;">
0.0002476
</td>
<td style="text-align:right;">
0.0002422
</td>
<td style="text-align:left;">
C6/LOC113984930/LOC113992168/F2/C8B/VTN/C3/CFI/LOC113986226/C9/C5/C8A/CFH
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042730" class="uri">GO:0042730</a>
</td>
<td style="text-align:left;">
fibrinolysis
</td>
<td style="text-align:left;">
8/817
</td>
<td style="text-align:left;">
14/12553
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.0003848
</td>
<td style="text-align:right;">
0.0003765
</td>
<td style="text-align:left;">
FGB/F2/FGG/PLG/LOC113992335/SERPINF2/FGA/HRG
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051918" class="uri">GO:0051918</a>
</td>
<td style="text-align:left;">
negative regulation of fibrinolysis
</td>
<td style="text-align:left;">
7/817
</td>
<td style="text-align:left;">
11/12553
</td>
<td style="text-align:right;">
0.0000013
</td>
<td style="text-align:right;">
0.0004807
</td>
<td style="text-align:right;">
0.0004702
</td>
<td style="text-align:left;">
F2/PLG/LOC113992335/SERPINF2/APOH/THBD/HRG
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002576" class="uri">GO:0002576</a>
</td>
<td style="text-align:left;">
platelet degranulation
</td>
<td style="text-align:left;">
22/817
</td>
<td style="text-align:left;">
108/12553
</td>
<td style="text-align:right;">
0.0000014
</td>
<td style="text-align:right;">
0.0004807
</td>
<td style="text-align:right;">
0.0004702
</td>
<td style="text-align:left;">
CLEC3B/SPARC/FGB/FGG/KNG1/SPP2/ITIH4/ENDOD1/PLG/F5/ALB/LOC113992335/SERPINF2/FGA/APOH/LOC114001803/LOC113992533/HRG/TMX3/LOC113989827/TMSB4X/APOOL
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007596" class="uri">GO:0007596</a>
</td>
<td style="text-align:left;">
blood coagulation
</td>
<td style="text-align:left;">
24/817
</td>
<td style="text-align:left;">
128/12553
</td>
<td style="text-align:right;">
0.0000022
</td>
<td style="text-align:right;">
0.0006423
</td>
<td style="text-align:right;">
0.0006284
</td>
<td style="text-align:left;">
GATA5/ANO6/FGB/F2/FGG/SERPINA10/PROC/LOC113995989/F9/F7/PLG/F5/SERPINC1/LOC113992335/CAPZB/FGA/LOC113995424/THBD/IRF2/COL1A2/RCOR1/PRSS56/COL1A1/GATA2
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006957" class="uri">GO:0006957</a>
</td>
<td style="text-align:left;">
complement activation, alternative pathway
</td>
<td style="text-align:left;">
8/817
</td>
<td style="text-align:left;">
18/12553
</td>
<td style="text-align:right;">
0.0000076
</td>
<td style="text-align:right;">
0.0017094
</td>
<td style="text-align:right;">
0.0016724
</td>
<td style="text-align:left;">
LOC113984930/C8B/C3/LOC113986226/C9/C5/C8A/CFH
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044267" class="uri">GO:0044267</a>
</td>
<td style="text-align:left;">
cellular protein metabolic process
</td>
<td style="text-align:left;">
27/817
</td>
<td style="text-align:left;">
165/12553
</td>
<td style="text-align:right;">
0.0000078
</td>
<td style="text-align:right;">
0.0017094
</td>
<td style="text-align:right;">
0.0016724
</td>
<td style="text-align:left;">
STC2/LAMB2/CP/LOC113992168/F2/FGG/KNG1/SERPINA10/SPP2/PROC/GSN/LOC114003546/CCN1/PLG/ITIH2/F5/C3/SERPINC1/ALB/APOA5/LOC113992335/LOC113991601/FGA/TTR/MMP2/P4HB/SNCA
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:left;">
9/817
</td>
<td style="text-align:left;">
27/12553
</td>
<td style="text-align:right;">
0.0000323
</td>
<td style="text-align:right;">
0.0063096
</td>
<td style="text-align:right;">
0.0061727
</td>
<td style="text-align:left;">
C6/LOC113992168/C8B/C3/CFI/LOC113986226/C9/C5/C8A
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
817
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019835" class="uri">GO:0019835</a>
</td>
<td style="text-align:left;">
cytolysis
</td>
<td style="text-align:left;">
5/817
</td>
<td style="text-align:left;">
10/12553
</td>
<td style="text-align:right;">
0.0002207
</td>
<td style="text-align:right;">
0.0352323
</td>
<td style="text-align:right;">
0.0344678
</td>
<td style="text-align:left;">
C6/C8B/C9/C5/C8A
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
817
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>go_table<span class="ot">&lt;-</span> enrich_results[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>,<span class="st">&quot;Description&quot;</span>,<span class="st">&quot;GeneRatio&quot;</span>,<span class="st">&quot;pvalue&quot;</span>,<span class="st">&quot;p.adjust&quot;</span>)]</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>go_table<span class="ot">&lt;-</span> go_table[<span class="sc">!</span><span class="fu">is.na</span>(go_table<span class="sc">$</span>ID),]</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>go_table<span class="sc">$</span>pvalue<span class="ot">&lt;-</span> <span class="fu">signif</span>(go_table<span class="sc">$</span>pvalue,<span class="dv">2</span>)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>go_table<span class="sc">$</span>p.adjust<span class="ot">&lt;-</span> <span class="fu">signif</span>(go_table<span class="sc">$</span>p.adjust,<span class="dv">2</span>)</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>go_table<span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(<span class="fu">as.data.frame</span>(go_table),<span class="at">theme =</span> <span class="fu">ttheme</span>(<span class="at">base_size =</span> <span class="dv">6</span>,<span class="at">padding=</span><span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">10</span>),<span class="st">&quot;pt&quot;</span>)),<span class="at">rows=</span><span class="cn">NULL</span>) <span class="sc">%&gt;%</span> <span class="fu">tab_add_title</span>(<span class="at">text =</span> main.title, <span class="at">size=</span><span class="dv">6</span>, <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;line&quot;</span>))<span class="sc">%&gt;%</span> <span class="fu">tab_add_title</span>(<span class="at">text =</span> main.title, <span class="at">size=</span><span class="dv">6</span>, <span class="at">padding =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">&quot;line&quot;</span>))</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a><span class="fu">write.csv</span>(enrich_results, <span class="at">file=</span><span class="fu">paste0</span>(<span class="st">&quot;../DE_results/GO_BP_&quot;</span>,tissue,<span class="st">&quot;_&quot;</span>,varname,<span class="st">&quot;.csv&quot;</span>), <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>Below is the code for making the figures in the supplement.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>g2<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(<span class="fu">ggarrange</span>(vp,g,<span class="at">widths=</span><span class="fu">c</span>(<span class="fl">0.7</span>,<span class="dv">1</span>), <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>),<span class="at">ncol=</span><span class="dv">2</span>,<span class="at">nrow=</span><span class="dv">1</span>),<span class="fu">ggarrange</span>(gseaplot,go_table, <span class="at">ncol=</span><span class="dv">2</span>,<span class="at">nrow=</span><span class="dv">1</span>, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;C&quot;</span>,<span class="st">&quot;D&quot;</span>)), <span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="co">#g2</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;../DE_results/supp_figure_GON_status.png&quot;</span>, <span class="at">plot=</span>g2, <span class="at">device=</span><span class="st">&quot;png&quot;</span> ,<span class="at">height=</span><span class="dv">180</span>, <span class="at">width=</span><span class="dv">200</span>, <span class="at">units=</span><span class="st">&quot;mm&quot;</span>, <span class="at">bg=</span><span class="st">&quot;white&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="testosterone-gon" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Testosterone GON</h2>
<p>Given that the behaviors are best correlated with mean T, then this
is our variable of interest</p>
<p><br></p>
<div id="expression-data-exploration-1" class="section level3"
number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Expression data
exploration</h3>
<p>But first, we need to look at these new data in the gonads with the
new subset with respect to our interest variables in the PCA.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span> <span class="fu">ls</span>()[<span class="sc">!</span>(<span class="fu">ls</span>() <span class="sc">%in%</span> <span class="fu">c</span>(keep,<span class="st">&quot;gon_data&quot;</span>,<span class="st">&quot;gon_behav&quot;</span>))])</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>tissue<span class="ot">=</span><span class="st">&quot;GON&quot;</span></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">==</span>tissue)</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(gon_behav, Batch<span class="sc">!=</span><span class="st">&quot;pilot&quot;</span>)</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>gon_behav<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(gon_behav)</span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(gon_behav)]</span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(gon_data)</span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a><span class="co">#remove genes with less than 5 reads</span></span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, mean)</span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">5</span>,]</span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(gon_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb28-19"><a href="#cb28-19" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span></code></pre></div>
<p>Before filtering we had 16854 genes, after filtering for mean read
count and excluding genes where &gt;50% of samples had a count of 0 we
are left with 16137…</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> mean_T)</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),] <span class="co">#remove any genes that didn&#39;t converge. </span></span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a><span class="co">#plotDispEsts(dd)</span></span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>vsd_data<span class="ot">&lt;-</span> <span class="fu">getVarianceStabilizedData</span>(dd)</span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a><span class="co">#vsd_data&lt;- assay(vsd)</span></span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">counts</span>(dd, <span class="at">normalized=</span><span class="cn">TRUE</span>), <span class="st">&quot;../DE_results/data_GON_norm.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">pca</span>(vsd_data, <span class="at">metadata =</span> gon_behav)</span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a><span class="fu">biplot</span>(p, <span class="at">lab=</span>gon_behav<span class="sc">$</span>Harvest_ID, <span class="at">colby=</span><span class="st">&quot;mean_T&quot;</span>, <span class="at">shape=</span><span class="st">&quot;Year&quot;</span>, <span class="at">legendPosition=</span><span class="st">&quot;right&quot;</span>, <span class="at">title=</span><span class="st">&quot;mean T in GON&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">eigencorplot</span>(p, <span class="at">metavars =</span> <span class="fu">c</span>(<span class="st">&quot;time_kill&quot;</span>,<span class="st">&quot;Year&quot;</span>,<span class="st">&quot;Status&quot;</span>, <span class="st">&quot;mean_T&quot;</span>,<span class="st">&quot;strength.all_study&quot;</span>))</span></code></pre></div>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a><span class="co">#gsg&lt;- goodSamplesGenes(datExpr0, verbose=3)</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a><span class="co">#gsg$allOK</span></span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>A<span class="ot">=</span><span class="fu">adjacency</span>(<span class="fu">t</span>(datExpr0),<span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a><span class="co">#-----Calculate whole network connectivity</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>k<span class="ot">=</span><span class="fu">as.numeric</span>(<span class="fu">apply</span>(A,<span class="dv">2</span>,sum))<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a><span class="co">#-----Standardized connectivity</span></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a>Z.k<span class="ot">=</span><span class="fu">scale</span>(k)</span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>thresholdZ.k<span class="ot">=</span><span class="sc">-</span><span class="fl">2.5</span> </span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>outlierColor<span class="ot">=</span><span class="fu">ifelse</span>(Z.k<span class="sc">&lt;</span>thresholdZ.k,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>A), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a><span class="co">#-----Convert traits to colors</span></span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a></span>
<span id="cb32-15"><a href="#cb32-15" tabindex="-1"></a>datColors<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">outlier=</span>outlierColor)</span>
<span id="cb32-16"><a href="#cb32-16" tabindex="-1"></a><span class="co">#-----Plot the sample dendrogram</span></span>
<span id="cb32-17"><a href="#cb32-17" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(sampleTree,<span class="at">groupLabels=</span><span class="fu">names</span>(datColors),</span>
<span id="cb32-18"><a href="#cb32-18" tabindex="-1"></a>                    <span class="at">colors=</span>datColors,<span class="at">main=</span><span class="st">&quot;Sample dendrogram and trait heatmap&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20continuous%20variable%20setup-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-1" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-1" class="section level3" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Analysis</h3>
<p>DESeq model specification ~ Year + mean_T</p>
<p>This chunk also includes a function I wrote to remove highly
influential observations (as measured by Cook’s Distance) from the
continuous data. I know from previous iterations of this analysis that
there are observations that are highly influential and also <a
href="https://support.bioconductor.org/p/126040/">DESeq2 does not
exclude these from the results in continuous data</a>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>varname<span class="ot">=</span><span class="st">&quot;mean_T&quot;</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> Year <span class="sc">+</span> mean_T)</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),]</span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16137 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 33, 0.2%
FALSE LFC &lt; 0 (down)     : 111, 0.69%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="co">#DEseq2 doesn&#39;t do outlier removal for continuous variables. I&#39;m doing that here with a bespoke function</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16137 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 33, 0.2%
FALSE LFC &lt; 0 (down)     : 111, 0.69%
FALSE outliers [1]       : 37, 0.23%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">design</span>(dd))</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">paste0</span>(des, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res)</span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>out_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(out_res)</span>
<span id="cb38-5"><a href="#cb38-5" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">merge</span>(out_res, genes_key,<span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb38-6"><a href="#cb38-6" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> out_res[<span class="fu">order</span>(out_res<span class="sc">$</span>pvalue),]</span>
<span id="cb38-7"><a href="#cb38-7" tabindex="-1"></a>out_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> tissue</span>
<span id="cb38-8"><a href="#cb38-8" tabindex="-1"></a>out_res<span class="sc">$</span>design<span class="ot">&lt;-</span> des</span>
<span id="cb38-9"><a href="#cb38-9" tabindex="-1"></a>out_res<span class="sc">$</span>fdrtool<span class="ot">&lt;-</span> <span class="st">&quot;no&quot;</span></span>
<span id="cb38-10"><a href="#cb38-10" tabindex="-1"></a></span>
<span id="cb38-11"><a href="#cb38-11" tabindex="-1"></a><span class="fu">write.csv</span>(out_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_mean_T.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-12"><a href="#cb38-12" tabindex="-1"></a></span>
<span id="cb38-13"><a href="#cb38-13" tabindex="-1"></a></span>
<span id="cb38-14"><a href="#cb38-14" tabindex="-1"></a></span>
<span id="cb38-15"><a href="#cb38-15" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb38-16"><a href="#cb38-16" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;MA-plot&quot;</span>,tissue,<span class="st">&quot;exp&quot;</span>,des))</span>
<span id="cb38-17"><a href="#cb38-17" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20mean%20T%20DEseq-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/GON%20mean_T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/GON%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-1" class="section level3" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/GON%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/GON%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GON according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0050729" class="uri">GO:0050729</a>
</td>
<td style="text-align:left;">
positive regulation of inflammatory response
</td>
<td style="text-align:left;">
16/1105
</td>
<td style="text-align:left;">
57/12541
</td>
<td style="text-align:right;">
0.0000213
</td>
<td style="text-align:right;">
0.0305125
</td>
<td style="text-align:right;">
0.0298555
</td>
<td style="text-align:left;">
PDE2A/AGTR1/LOC114003559/IFI35/SNCA/LPL/LOC114000059/TNIP1/TLR2/TGM2/IL1RL1/CEBPB/PLA2G7/CLOCK/ETS1/APP
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051412" class="uri">GO:0051412</a>
</td>
<td style="text-align:left;">
response to corticosterone
</td>
<td style="text-align:left;">
7/1105
</td>
<td style="text-align:left;">
13/12541
</td>
<td style="text-align:right;">
0.0000432
</td>
<td style="text-align:right;">
0.0305125
</td>
<td style="text-align:right;">
0.0298555
</td>
<td style="text-align:left;">
STAR/TH/AVPR1A/LOC113992757/STARD3/LOC113998857/NEFL
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050679" class="uri">GO:0050679</a>
</td>
<td style="text-align:left;">
positive regulation of epithelial cell proliferation
</td>
<td style="text-align:left;">
15/1105
</td>
<td style="text-align:left;">
55/12541
</td>
<td style="text-align:right;">
0.0000555
</td>
<td style="text-align:right;">
0.0305125
</td>
<td style="text-align:right;">
0.0298555
</td>
<td style="text-align:left;">
NR4A3/LOC113991446/HTRA1/OSR2/LOC113991466/MYC/LOC113986457/BMP4/BMP5/FGFR2/IGF1/TWIST1/HRAS/FGF1/ATP7A
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
19/1105
</td>
<td style="text-align:left;">
81/12541
</td>
<td style="text-align:right;">
0.0000593
</td>
<td style="text-align:right;">
0.0305125
</td>
<td style="text-align:right;">
0.0298555
</td>
<td style="text-align:left;">
LOC113990217/RPL31/RPL37/RPS20/RPS25/RPL39/RPS12/RPL23/RPL38/RPL22/RPL35/RPL29/RPL32/RPL23A/RPS6/RPL9/RPL5/RPL27/RPL10A
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019083" class="uri">GO:0019083</a>
</td>
<td style="text-align:left;">
viral transcription
</td>
<td style="text-align:left;">
20/1105
</td>
<td style="text-align:left;">
100/12541
</td>
<td style="text-align:right;">
0.0003898
</td>
<td style="text-align:right;">
0.1578854
</td>
<td style="text-align:right;">
0.1544856
</td>
<td style="text-align:left;">
LOC113990217/RPL31/RPL37/RPS20/RPS25/RPL39/RPS12/RPL23/RPL38/RPL22/RPL35/RPL29/RPL32/RPL23A/RPS6/RPL9/RPL5/TPR/RPL27/RPL10A
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006825" class="uri">GO:0006825</a>
</td>
<td style="text-align:left;">
copper ion transport
</td>
<td style="text-align:left;">
6/1105
</td>
<td style="text-align:left;">
13/12541
</td>
<td style="text-align:right;">
0.0004603
</td>
<td style="text-align:right;">
0.1578854
</td>
<td style="text-align:right;">
0.1544856
</td>
<td style="text-align:left;">
ATOX1/HEPHL1/SLC31A1/FKBP4/ATP7A/CUTC
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0055010" class="uri">GO:0055010</a>
</td>
<td style="text-align:left;">
ventricular cardiac muscle tissue morphogenesis
</td>
<td style="text-align:left;">
7/1105
</td>
<td style="text-align:left;">
18/12541
</td>
<td style="text-align:right;">
0.0005396
</td>
<td style="text-align:right;">
0.1586287
</td>
<td style="text-align:right;">
0.1552129
</td>
<td style="text-align:left;">
ISL1/FOXC1/FGFR2/LOC113998261/FKBP1A/PTCD2/TPM1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046928" class="uri">GO:0046928</a>
</td>
<td style="text-align:left;">
regulation of neurotransmitter secretion
</td>
<td style="text-align:left;">
6/1105
</td>
<td style="text-align:left;">
14/12541
</td>
<td style="text-align:right;">
0.0007457
</td>
<td style="text-align:right;">
0.1918374
</td>
<td style="text-align:right;">
0.1877065
</td>
<td style="text-align:left;">
SNCG/SNCAIP/CPLX2/CPLX1/RHOT1/MCTP1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021510" class="uri">GO:0021510</a>
</td>
<td style="text-align:left;">
spinal cord development
</td>
<td style="text-align:left;">
8/1105
</td>
<td style="text-align:left;">
25/12541
</td>
<td style="text-align:right;">
0.0009806
</td>
<td style="text-align:right;">
0.1955375
</td>
<td style="text-align:right;">
0.1913269
</td>
<td style="text-align:left;">
BAG3/PTN/PLXDC1/SUZ12/PHGDH/SRD5A1/NEFL/PTPRS
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1105
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050772" class="uri">GO:0050772</a>
</td>
<td style="text-align:left;">
positive regulation of axonogenesis
</td>
<td style="text-align:left;">
9/1105
</td>
<td style="text-align:left;">
31/12541
</td>
<td style="text-align:right;">
0.0010495
</td>
<td style="text-align:right;">
0.1955375
</td>
<td style="text-align:right;">
0.1913269
</td>
<td style="text-align:left;">
SHOX2/METRN/STK25/CHODL/NGF/PLXND1/MAP6/NEFL/SLITRK1
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1105
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="status-x-mean-t-gon" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Status x mean T
GON</h2>
<p>For pulling out candidate genes with interactions</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span> <span class="fu">ls</span>()[<span class="sc">!</span>(<span class="fu">ls</span>() <span class="sc">%in%</span> <span class="fu">c</span>(keep,<span class="st">&quot;gon_data&quot;</span>,<span class="st">&quot;gon_behav&quot;</span>,<span class="st">&quot;tissue&quot;</span>))])</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>varname<span class="ot">=</span><span class="st">&quot;Status x mean_T&quot;</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_behav, <span class="at">design=</span> <span class="sc">~</span> Year <span class="sc">+</span> Status <span class="sc">+</span> mean_T <span class="sc">+</span> Status<span class="sc">:</span>mean_T)</span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd,<span class="at">test=</span><span class="st">&quot;LRT&quot;</span>,<span class="at">reduced=</span><span class="sc">~</span>Year <span class="sc">+</span> Status <span class="sc">+</span> mean_T)</span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a><span class="co">#dd&lt;- DESeq(dd)</span></span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a><span class="co">#dd&lt;- dd[which(mcols(dd)$betaConv),]</span></span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a><span class="co">#summary(res)</span></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16137 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 5, 0.031%
FALSE LFC &lt; 0 (down)     : 7, 0.043%
FALSE outliers [1]       : 29, 0.18%
FALSE low counts [2]     : 933, 5.8%
FALSE (mean count &lt; 14)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">design</span>(dd))</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">paste0</span>(des, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res)</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>out_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(out_res)</span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">merge</span>(out_res, genes_key,<span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> out_res[<span class="fu">order</span>(out_res<span class="sc">$</span>padj),]</span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a>out_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> tissue</span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a>out_res<span class="sc">$</span>design<span class="ot">&lt;-</span> des</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>out_res<span class="sc">$</span>n_p005<span class="ot">&lt;-</span> <span class="fu">nrow</span>(out_res[<span class="fu">which</span>(out_res<span class="sc">$</span>pvalue<span class="sc">&lt;</span><span class="fl">0.05</span>),])</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>out_res<span class="sc">$</span>n_q01<span class="ot">&lt;-</span> <span class="fu">nrow</span>(out_res[<span class="fu">which</span>(out_res<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.1</span>),])</span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>out_res<span class="sc">$</span>Test<span class="ot">&lt;-</span> <span class="st">&quot;LRT_full_interaction&quot;</span></span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>out_res<span class="sc">$</span>fdrtool<span class="ot">&lt;-</span> <span class="st">&quot;no&quot;</span></span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a></span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a><span class="fu">write.csv</span>(out_res, <span class="at">file=</span><span class="st">&quot;../DE_results/interactions_GON_mean_T_Status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a></span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a></span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a></span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;MA-plot&quot;</span>,tissue,<span class="st">&quot;exp&quot;</span>,des))</span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p>How sensitive is the candidate gene results to model form?</p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-5-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-gon" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Strength GON</h2>
<div id="analysis-2" class="section level3" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> Analysis</h3>
<p>formula ~ Year + scale(Strength)</p>
<p>Without scaling strength, DESeq2 would throw a warning, recommending
the variable be scaled.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span> <span class="fu">ls</span>()[<span class="sc">!</span>(<span class="fu">ls</span>() <span class="sc">%in%</span> <span class="fu">c</span>(keep,<span class="st">&quot;gon_data&quot;</span>,<span class="st">&quot;gon_behav&quot;</span>,<span class="st">&quot;tissue&quot;</span>,<span class="st">&quot;dd&quot;</span>))])</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>varname<span class="ot">=</span><span class="st">&quot;Strength&quot;</span></span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>dd<span class="sc">$</span>strength.all_study<span class="ot">&lt;-</span> <span class="fu">scale</span>(dd<span class="sc">$</span>strength.all_study)</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span>Year <span class="sc">+</span> strength.all_study)</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a></span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">rm_continuous_outliers</span>(dd,res)</span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 16137 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 10, 0.062%
FALSE LFC &lt; 0 (down)     : 11, 0.068%
FALSE outliers [1]       : 45, 0.28%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>res<span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">design</span>(dd))</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">paste0</span>(des, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb48-5"><a href="#cb48-5" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res)</span>
<span id="cb48-6"><a href="#cb48-6" tabindex="-1"></a>out_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(out_res)</span>
<span id="cb48-7"><a href="#cb48-7" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">merge</span>(out_res, genes_key,<span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb48-8"><a href="#cb48-8" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> out_res[<span class="fu">order</span>(out_res<span class="sc">$</span>padj),]</span>
<span id="cb48-9"><a href="#cb48-9" tabindex="-1"></a>out_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> tissue</span>
<span id="cb48-10"><a href="#cb48-10" tabindex="-1"></a>out_res<span class="sc">$</span>design<span class="ot">&lt;-</span> des</span>
<span id="cb48-11"><a href="#cb48-11" tabindex="-1"></a>out_res<span class="sc">$</span>fdrtool<span class="ot">&lt;-</span> <span class="st">&quot;no&quot;</span></span>
<span id="cb48-12"><a href="#cb48-12" tabindex="-1"></a></span>
<span id="cb48-13"><a href="#cb48-13" tabindex="-1"></a></span>
<span id="cb48-14"><a href="#cb48-14" tabindex="-1"></a><span class="fu">write.csv</span>(out_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_GON_Strength.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb48-15"><a href="#cb48-15" tabindex="-1"></a></span>
<span id="cb48-16"><a href="#cb48-16" tabindex="-1"></a></span>
<span id="cb48-17"><a href="#cb48-17" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb48-18"><a href="#cb48-18" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;MA-plot&quot;</span>,tissue,<span class="st">&quot;exp&quot;</span>,des))</span>
<span id="cb48-19"><a href="#cb48-19" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/GON%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/GON%20Strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/GON%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-2" class="section level3" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/GON%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/GON%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GON according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0045540" class="uri">GO:0045540</a>
</td>
<td style="text-align:left;">
regulation of cholesterol biosynthetic process
</td>
<td style="text-align:left;">
13/919
</td>
<td style="text-align:left;">
35/12541
</td>
<td style="text-align:right;">
0.0000005
</td>
<td style="text-align:right;">
0.0009732
</td>
<td style="text-align:right;">
0.0009732
</td>
<td style="text-align:left;">
SC5D/MVK/FDFT1/FASN/SEC14L2/ELOVL6/LOC113987727/HMGCS1/LOC114002832/MVD/ACACA/SREBF1/LSS
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006695" class="uri">GO:0006695</a>
</td>
<td style="text-align:left;">
cholesterol biosynthetic process
</td>
<td style="text-align:left;">
12/919
</td>
<td style="text-align:left;">
35/12541
</td>
<td style="text-align:right;">
0.0000038
</td>
<td style="text-align:right;">
0.0035789
</td>
<td style="text-align:right;">
0.0035789
</td>
<td style="text-align:left;">
MVK/FDFT1/MSMO1/DHCR24/LOC113987727/HMGCS1/LOC114002832/MVD/ACLY/APOA5/LSS/APOA1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044267" class="uri">GO:0044267</a>
</td>
<td style="text-align:left;">
cellular protein metabolic process
</td>
<td style="text-align:left;">
28/919
</td>
<td style="text-align:left;">
164/12541
</td>
<td style="text-align:right;">
0.0000211
</td>
<td style="text-align:right;">
0.0112266
</td>
<td style="text-align:right;">
0.0112266
</td>
<td style="text-align:left;">
UBA52/SERPINA10/ADORA2A/LOC113992168/PLG/ITIH2/SERPINC1/F2/TTR/FN1/ALB/TGOLN2/SPP2/LOC113992658/FGG/APOA5/F5/FGA/IGFBP4/CALB1/SERPIND1/SHISA5/LOC113996623/GGA1/CCN1/FSTL3/APOA1/LOC113990797
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042730" class="uri">GO:0042730</a>
</td>
<td style="text-align:left;">
fibrinolysis
</td>
<td style="text-align:left;">
7/919
</td>
<td style="text-align:left;">
14/12541
</td>
<td style="text-align:right;">
0.0000241
</td>
<td style="text-align:right;">
0.0112266
</td>
<td style="text-align:right;">
0.0112266
</td>
<td style="text-align:left;">
PLG/HRG/F2/FGB/FGG/SERPINF2/FGA
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
16/919
</td>
<td style="text-align:left;">
79/12541
</td>
<td style="text-align:right;">
0.0001629
</td>
<td style="text-align:right;">
0.0534749
</td>
<td style="text-align:right;">
0.0534749
</td>
<td style="text-align:left;">
SERPINA10/VTN/LOC113992168/LOC113989405/ITIH2/HRG/SERPINC1/PTTG1/FETUB/SPOCK3/SPP2/SERPINF2/SERPIND1/GAPDH/ITIH4/AGT
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904158" class="uri">GO:1904158</a>
</td>
<td style="text-align:left;">
axonemal central apparatus assembly
</td>
<td style="text-align:left;">
7/919
</td>
<td style="text-align:left;">
18/12541
</td>
<td style="text-align:right;">
0.0001721
</td>
<td style="text-align:right;">
0.0534749
</td>
<td style="text-align:right;">
0.0534749
</td>
<td style="text-align:left;">
LOC113997903/LOC113997531/LOC113995484/LOC114001265/LOC113997354/LOC113983738/DNAJB13
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070328" class="uri">GO:0070328</a>
</td>
<td style="text-align:left;">
triglyceride homeostasis
</td>
<td style="text-align:left;">
7/919
</td>
<td style="text-align:left;">
19/12541
</td>
<td style="text-align:right;">
0.0002554
</td>
<td style="text-align:right;">
0.0680072
</td>
<td style="text-align:right;">
0.0680072
</td>
<td style="text-align:left;">
ABCG5/ANGPTL3/APOA5/ANGPTL4/GIP/LIPC/APOA1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006694" class="uri">GO:0006694</a>
</td>
<td style="text-align:left;">
steroid biosynthetic process
</td>
<td style="text-align:left;">
7/919
</td>
<td style="text-align:left;">
20/12541
</td>
<td style="text-align:right;">
0.0003682
</td>
<td style="text-align:right;">
0.0857953
</td>
<td style="text-align:right;">
0.0857953
</td>
<td style="text-align:left;">
FDFT1/LOC114002832/LOC113992757/STARD3/STAR/LOC113983393/LSS
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008652" class="uri">GO:0008652</a>
</td>
<td style="text-align:left;">
cellular amino acid biosynthetic process
</td>
<td style="text-align:left;">
7/919
</td>
<td style="text-align:left;">
21/12541
</td>
<td style="text-align:right;">
0.0005177
</td>
<td style="text-align:right;">
0.0957789
</td>
<td style="text-align:right;">
0.0957789
</td>
<td style="text-align:left;">
OAT/PAH/ASPG/GLUL/GLS/GLUD1/ALDH18A1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060438" class="uri">GO:0060438</a>
</td>
<td style="text-align:left;">
trachea development
</td>
<td style="text-align:left;">
6/919
</td>
<td style="text-align:left;">
16/12541
</td>
<td style="text-align:right;">
0.0006437
</td>
<td style="text-align:right;">
0.0957789
</td>
<td style="text-align:right;">
0.0957789
</td>
<td style="text-align:left;">
LOC113997903/LOC113997531/LOC113995484/LOC114001265/LOC113997354/LOC113983738
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
919
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="pituitary-pit" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Pituitary (PIT)</h1>
<div id="checking-the-sampling-1" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Checking the
sampling</h2>
<p>Before filtering we had 16854 genes, after filtering for mean read
count and excluding genes where &gt;50% of samples had a count of 0 we
are left with 14812</p>
<p>Before I go into more detail, I am going to check that the sampling
is equal across batches with respect to our variables of interest. With
respect to Status we have pit samples for 12, with respect to T and the
tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.22
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<p><img src="03_DESeq2_files/figure-html/PIT%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="expression-data-exploration-2" class="section level3"
number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/PIT%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-2" class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks controlling for Batch also removes the association with Year.
Let’s just include Batch.</p>
<p><br></p>
</div>
</div>
<div id="status-pit" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Status PIT</h2>
<div id="analysis-3" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<p>In this section, I find results with different p-value distributions.
I was following <a
href="https://www.huber.embl.de/users/klaus/Teaching/DESeq2-Analysis.pdf">THIS</a>
tutorial by Bernd Klaus and it recommended using <code>fdrtool</code> to
correct these p-values.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a>varname<span class="ot">=</span><span class="st">&quot;Status&quot;</span></span>
<span id="cb53-2"><a href="#cb53-2" tabindex="-1"></a><span class="fu">design</span>(dd)<span class="ot">&lt;-</span> <span class="fu">formula</span>(<span class="sc">~</span> Batch <span class="sc">+</span> Status)</span>
<span id="cb53-3"><a href="#cb53-3" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span>
<span id="cb53-4"><a href="#cb53-4" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">results</span>(dd, <span class="at">alpha=</span><span class="fl">0.1</span>)</span>
<span id="cb53-5"><a href="#cb53-5" tabindex="-1"></a></span>
<span id="cb53-6"><a href="#cb53-6" tabindex="-1"></a></span>
<span id="cb53-7"><a href="#cb53-7" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res[<span class="sc">!</span><span class="fu">is.na</span>(res<span class="sc">$</span>padj),]</span>
<span id="cb53-8"><a href="#cb53-8" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res1[<span class="sc">!</span><span class="fu">is.na</span>(res1<span class="sc">$</span>pvalue),]</span>
<span id="cb53-9"><a href="#cb53-9" tabindex="-1"></a>res1<span class="sc">$</span>padj<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb53-10"><a href="#cb53-10" tabindex="-1"></a>res1.fdr<span class="ot">&lt;-</span> <span class="fu">fdrtool</span>(res1<span class="sc">$</span>stat, <span class="at">statistic=</span><span class="st">&quot;normal&quot;</span>, <span class="at">plot=</span><span class="cn">FALSE</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb53-11"><a href="#cb53-11" tabindex="-1"></a>res1<span class="sc">$</span>padj<span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res1.fdr<span class="sc">$</span>pval, <span class="at">method=</span><span class="st">&quot;BH&quot;</span>)</span>
<span id="cb53-12"><a href="#cb53-12" tabindex="-1"></a></span>
<span id="cb53-13"><a href="#cb53-13" tabindex="-1"></a>outs<span class="ot">&lt;-</span> res[<span class="fu">is.na</span>(res<span class="sc">$</span>padj),] <span class="co"># add those outliers back in... </span></span>
<span id="cb53-14"><a href="#cb53-14" tabindex="-1"></a>res1<span class="ot">&lt;-</span> <span class="fu">rbind</span>(res1, outs)</span>
<span id="cb53-15"><a href="#cb53-15" tabindex="-1"></a>res1<span class="ot">&lt;-</span> res1[<span class="fu">order</span>(res1<span class="sc">$</span>padj),]</span>
<span id="cb53-16"><a href="#cb53-16" tabindex="-1"></a><span class="fu">summary</span>(res1)</span></code></pre></div>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 8, 0.054%
FALSE LFC &lt; 0 (down)     : 6, 0.041%
FALSE outliers [1]       : 91, 0.61%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">design</span>(dd))</span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>des<span class="ot">&lt;-</span> <span class="fu">paste0</span>(des, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb55-3"><a href="#cb55-3" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res1)</span>
<span id="cb55-4"><a href="#cb55-4" tabindex="-1"></a>out_res<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">row.names</span>(out_res)</span>
<span id="cb55-5"><a href="#cb55-5" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> <span class="fu">merge</span>(out_res, genes_key,<span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb55-6"><a href="#cb55-6" tabindex="-1"></a>out_res<span class="ot">&lt;-</span> out_res[<span class="fu">order</span>(out_res<span class="sc">$</span>padj),]</span>
<span id="cb55-7"><a href="#cb55-7" tabindex="-1"></a>out_res<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> tissue</span>
<span id="cb55-8"><a href="#cb55-8" tabindex="-1"></a>out_res<span class="sc">$</span>design<span class="ot">&lt;-</span> des</span>
<span id="cb55-9"><a href="#cb55-9" tabindex="-1"></a>out_res<span class="sc">$</span>fdrtool<span class="ot">&lt;-</span> <span class="st">&quot;yes&quot;</span></span>
<span id="cb55-10"><a href="#cb55-10" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" tabindex="-1"></a><span class="fu">write.csv</span>(out_res, <span class="at">file=</span><span class="st">&quot;../DE_results/results_PIT_status.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb55-12"><a href="#cb55-12" tabindex="-1"></a></span>
<span id="cb55-13"><a href="#cb55-13" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb55-14"><a href="#cb55-14" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;MA-plot&quot;</span>,tissue,<span class="st">&quot;exp&quot;</span>,des))</span>
<span id="cb55-15"><a href="#cb55-15" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;uncorrected P-value&quot;</span>)</span>
<span id="cb55-16"><a href="#cb55-16" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res1, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>), <span class="at">main=</span><span class="st">&quot;FDR tool corrected PIT Status&quot;</span>)</span>
<span id="cb55-17"><a href="#cb55-17" tabindex="-1"></a><span class="fu">hist</span>(res1.fdr<span class="sc">$</span>pval, <span class="at">breaks=</span><span class="dv">20</span>, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main=</span><span class="st">&quot;P-value correction &#39;fdrtool&#39;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&#39;Corrected&#39; P-value&quot;</span>)</span></code></pre></div>
<p><img src="03_DESeq2_files/figure-html/PIT%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PIT%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-ontology-3" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Gene Ontology</h3>
<p><br></p>
<p><img src="03_DESeq2_files/figure-html/PIT%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PIT%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PIT according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007422" class="uri">GO:0007422</a>
</td>
<td style="text-align:left;">
peripheral nervous system development
</td>
<td style="text-align:left;">
7/426
</td>
<td style="text-align:left;">
23/12076
</td>
<td style="text-align:right;">
0.0000097
</td>
<td style="text-align:right;">
0.0125810
</td>
<td style="text-align:right;">
0.0123228
</td>
<td style="text-align:left;">
LOC113999850/SOX10/SERPINI1/LOC113986199/PMP22/NTF3/GFRA3
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051384" class="uri">GO:0051384</a>
</td>
<td style="text-align:left;">
response to glucocorticoid
</td>
<td style="text-align:left;">
8/426
</td>
<td style="text-align:left;">
41/12076
</td>
<td style="text-align:right;">
0.0000770
</td>
<td style="text-align:right;">
0.0498862
</td>
<td style="text-align:right;">
0.0488623
</td>
<td style="text-align:left;">
PAM/ALPL/SDC1/LOC113993087/ADIPOQ/CASP7/TYMS/SPARC
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032496" class="uri">GO:0032496</a>
</td>
<td style="text-align:left;">
response to lipopolysaccharide
</td>
<td style="text-align:left;">
12/426
</td>
<td style="text-align:left;">
100/12076
</td>
<td style="text-align:right;">
0.0002011
</td>
<td style="text-align:right;">
0.0752137
</td>
<td style="text-align:right;">
0.0736699
</td>
<td style="text-align:left;">
GCH1/ALPL/FMO1/CASP7/PCK1/CNR1/DCN/VCAM1/OPRM1/SPARC/CNP/LIAS
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045776" class="uri">GO:0045776</a>
</td>
<td style="text-align:left;">
negative regulation of blood pressure
</td>
<td style="text-align:left;">
5/426
</td>
<td style="text-align:left;">
17/12076
</td>
<td style="text-align:right;">
0.0002323
</td>
<td style="text-align:right;">
0.0752137
</td>
<td style="text-align:right;">
0.0736699
</td>
<td style="text-align:left;">
GCH1/ADIPOQ/RNLS/CNR1/GJA5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
9/426
</td>
<td style="text-align:left;">
73/12076
</td>
<td style="text-align:right;">
0.0010203
</td>
<td style="text-align:right;">
0.2106709
</td>
<td style="text-align:right;">
0.2063470
</td>
<td style="text-align:left;">
SERPIND1/ITIH2/LOC113991485/SERPINF1/COL6A3/SERPINF2/SERPINI1/ITIH4/LOC113992706
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006957" class="uri">GO:0006957</a>
</td>
<td style="text-align:left;">
complement activation, alternative pathway
</td>
<td style="text-align:left;">
4/426
</td>
<td style="text-align:left;">
14/12076
</td>
<td style="text-align:right;">
0.0011533
</td>
<td style="text-align:right;">
0.2106709
</td>
<td style="text-align:right;">
0.2063470
</td>
<td style="text-align:left;">
CFH/C8B/C7/LOC113986241
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042552" class="uri">GO:0042552</a>
</td>
<td style="text-align:left;">
myelination
</td>
<td style="text-align:left;">
7/426
</td>
<td style="text-align:left;">
47/12076
</td>
<td style="text-align:right;">
0.0012002
</td>
<td style="text-align:right;">
0.2106709
</td>
<td style="text-align:right;">
0.2063470
</td>
<td style="text-align:left;">
ADGRG6/MPZ/LOC114000755/PLLP/DHH/MAL2/BCAS1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006836" class="uri">GO:0006836</a>
</td>
<td style="text-align:left;">
neurotransmitter transport
</td>
<td style="text-align:left;">
5/426
</td>
<td style="text-align:left;">
24/12076
</td>
<td style="text-align:right;">
0.0013014
</td>
<td style="text-align:right;">
0.2106709
</td>
<td style="text-align:right;">
0.2063470
</td>
<td style="text-align:left;">
SLC22A3/LOC113990781/SLC6A2/SLC29A1/SLC6A4
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071300" class="uri">GO:0071300</a>
</td>
<td style="text-align:left;">
cellular response to retinoic acid
</td>
<td style="text-align:left;">
7/426
</td>
<td style="text-align:left;">
51/12076
</td>
<td style="text-align:right;">
0.0019578
</td>
<td style="text-align:right;">
0.2445362
</td>
<td style="text-align:right;">
0.2395173
</td>
<td style="text-align:left;">
SERPINF1/BRINP2/EPHA3/OSR1/PCK1/ADNP2/SLC6A4
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007200" class="uri">GO:0007200</a>
</td>
<td style="text-align:left;">
phospholipase C-activating G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
6/426
</td>
<td style="text-align:left;">
38/12076
</td>
<td style="text-align:right;">
0.0019694
</td>
<td style="text-align:right;">
0.2445362
</td>
<td style="text-align:right;">
0.2395173
</td>
<td style="text-align:left;">
PTGER3/ESR1/P2RY2/CMKLR1/HTR2B/OPRM1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
426
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="testosterone-pit" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Testosterone PIT</h2>
<div id="analysis-4" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Analysis</h3>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 14, 0.095%
FALSE LFC &lt; 0 (down)     : 5, 0.034%
FALSE outliers [1]       : 62, 0.42%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PIT%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-4" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/PIT%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PIT%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PIT according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0010759" class="uri">GO:0010759</a>
</td>
<td style="text-align:left;">
positive regulation of macrophage chemotaxis
</td>
<td style="text-align:left;">
4/470
</td>
<td style="text-align:left;">
13/12076
</td>
<td style="text-align:right;">
0.0012238
</td>
<td style="text-align:right;">
0.5854897
</td>
<td style="text-align:right;">
0.5782776
</td>
<td style="text-align:left;">
LOC113985892/IL34/CSF1/LOC113983017
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045087" class="uri">GO:0045087</a>
</td>
<td style="text-align:left;">
innate immune response
</td>
<td style="text-align:left;">
20/470
</td>
<td style="text-align:left;">
245/12076
</td>
<td style="text-align:right;">
0.0014440
</td>
<td style="text-align:right;">
0.5854897
</td>
<td style="text-align:right;">
0.5782776
</td>
<td style="text-align:left;">
LOC114003556/ALPK1/IL34/PGLYRP2/SUSD4/INAVA/LOC113996654/CSF1/LOC113992821/SARM1/LOC113992189/LOC113983017/HEXIM1/SRC/NOD1/SLC15A4/S100B/LOC113986435/STYK1/SYK
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002009" class="uri">GO:0002009</a>
</td>
<td style="text-align:left;">
morphogenesis of an epithelium
</td>
<td style="text-align:left;">
4/470
</td>
<td style="text-align:left;">
14/12076
</td>
<td style="text-align:right;">
0.0016610
</td>
<td style="text-align:right;">
0.5854897
</td>
<td style="text-align:right;">
0.5782776
</td>
<td style="text-align:left;">
LOC113990794/SOX10/DAG1/TIMELESS
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046849" class="uri">GO:0046849</a>
</td>
<td style="text-align:left;">
bone remodeling
</td>
<td style="text-align:left;">
4/470
</td>
<td style="text-align:left;">
14/12076
</td>
<td style="text-align:right;">
0.0016610
</td>
<td style="text-align:right;">
0.5854897
</td>
<td style="text-align:right;">
0.5782776
</td>
<td style="text-align:left;">
P3H4/LTBP3/WNT16/TPH1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006813" class="uri">GO:0006813</a>
</td>
<td style="text-align:left;">
potassium ion transport
</td>
<td style="text-align:left;">
7/470
</td>
<td style="text-align:left;">
48/12076
</td>
<td style="text-align:right;">
0.0023882
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
KCNG1/KCNA4/KCNN1/LOC113993556/KCNJ15/KCNAB1/KCNS2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007015" class="uri">GO:0007015</a>
</td>
<td style="text-align:left;">
actin filament organization
</td>
<td style="text-align:left;">
11/470
</td>
<td style="text-align:left;">
109/12076
</td>
<td style="text-align:right;">
0.0034092
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
SHROOM4/LOC114003556/SORBS3/CORO6/ERMN/MARCKSL1/LOC113992312/ELMO1/LOC113988218/SAMD14/TMSB15B
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000134" class="uri">GO:2000134</a>
</td>
<td style="text-align:left;">
negative regulation of G1/S transition of mitotic cell cycle
</td>
<td style="text-align:left;">
5/470
</td>
<td style="text-align:left;">
27/12076
</td>
<td style="text-align:right;">
0.0034699
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
GFI1B/DACT1/CDKN1A/CDKN2C/EZH2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032755" class="uri">GO:0032755</a>
</td>
<td style="text-align:left;">
positive regulation of interleukin-6 production
</td>
<td style="text-align:left;">
7/470
</td>
<td style="text-align:left;">
52/12076
</td>
<td style="text-align:right;">
0.0037979
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
LOC114003556/LOC114000755/INAVA/IL6R/NOD1/LOC113986435/SYK
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030335" class="uri">GO:0030335</a>
</td>
<td style="text-align:left;">
positive regulation of cell migration
</td>
<td style="text-align:left;">
16/470
</td>
<td style="text-align:left;">
197/12076
</td>
<td style="text-align:right;">
0.0043787
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
LOC113991284/MYLK/DAPK3/HBEGF/LOC113996864/PLP1/CSF1/CXCL12/CTSH/LOC113983017/BMP2/TMSB15B/PAK1/SMO/NTRK3/FLT1
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006337" class="uri">GO:0006337</a>
</td>
<td style="text-align:left;">
nucleosome disassembly
</td>
<td style="text-align:left;">
4/470
</td>
<td style="text-align:left;">
18/12076
</td>
<td style="text-align:right;">
0.0044873
</td>
<td style="text-align:right;">
0.6235461
</td>
<td style="text-align:right;">
0.6158653
</td>
<td style="text-align:left;">
LOC113985909/LOC113985906/SMARCC2/LOC113997381
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
470
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 1205, 8.1%
FALSE LFC &lt; 0 (down)     : 1401, 9.5%
FALSE outliers [1]       : 37, 0.25%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20mean%20T%20x%20Status-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-12-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-pit" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Strength PIT</h2>
<div id="analysis-5" class="section level3" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Analysis</h3>
<p>Formula: ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 387, 2.6%
FALSE LFC &lt; 0 (down)     : 113, 0.76%
FALSE outliers [1]       : 207, 1.4%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20strength%20geneplotting-1.png" width="1344" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/PIT%20strength%20geneplotting-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>As you can see above, PFT3 has consistently high leverage on the
social network analysis. Even if I play around with the outlier theshold
this I am always getting non-linear plots like above. So, let’s see what
happens when I remove this sample.</p>
<p><br></p>
</div>
<div id="re-analysis---outlier-removed" class="section level3"
number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Re-analysis -
outlier removed</h3>
<pre><code>FALSE 
FALSE out of 14812 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 109, 0.74%
FALSE LFC &lt; 0 (down)     : 13, 0.088%
FALSE outliers [1]       : 58, 0.39%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20Strength2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PIT%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PIT%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-5" class="section level3" number="2.5.3">
<h3><span class="header-section-number">2.5.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/PIT%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PIT%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PIT according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0019934" class="uri">GO:0019934</a>
</td>
<td style="text-align:left;">
cGMP-mediated signaling
</td>
<td style="text-align:left;">
7/1174
</td>
<td style="text-align:left;">
15/12076
</td>
<td style="text-align:right;">
0.0002572
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
PDE2A/NPR1/NPR2/EDNRB/LOC113993563/HTR2C/AQP1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043116" class="uri">GO:0043116</a>
</td>
<td style="text-align:left;">
negative regulation of vascular permeability
</td>
<td style="text-align:left;">
7/1174
</td>
<td style="text-align:left;">
16/12076
</td>
<td style="text-align:right;">
0.0004190
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
PDE2A/CLDN5/ANGPT4/DDAH1/ADORA2A/LOC113996612/ARHGAP35
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009566" class="uri">GO:0009566</a>
</td>
<td style="text-align:left;">
fertilization
</td>
<td style="text-align:left;">
7/1174
</td>
<td style="text-align:left;">
17/12076
</td>
<td style="text-align:right;">
0.0006528
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
TDRD9/BAX/AAAS/LOC113990833/TDRKH/SYCP2/YBX3
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034097" class="uri">GO:0034097</a>
</td>
<td style="text-align:left;">
response to cytokine
</td>
<td style="text-align:left;">
11/1174
</td>
<td style="text-align:left;">
38/12076
</td>
<td style="text-align:right;">
0.0007119
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
LOC114001824/MAPKAPK2/MCL1/TIMP2/LOC113992821/BCL2L1/IL6R/LOC113984425/ALDH1A2/TNFRSF11A/TYMS
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046777" class="uri">GO:0046777</a>
</td>
<td style="text-align:left;">
protein autophosphorylation
</td>
<td style="text-align:left;">
28/1174
</td>
<td style="text-align:left;">
155/12076
</td>
<td style="text-align:right;">
0.0009200
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
STK11/MAP3K11/CAD/MAPKAPK2/DAPK3/CHEK2/LOC114000515/PIM1/MARK2/MKNK2/CAMK2B/VRK2/FGFR1/ACVR1B/LOC113983017/STK39/CDK5/MAP3K12/LOC113994408/DYRK1A/CSNK1G2/LOC113999057/ERBB2/DDR2/FER/EIF2AK2/NTRK1/ATP13A2
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016525" class="uri">GO:0016525</a>
</td>
<td style="text-align:left;">
negative regulation of angiogenesis
</td>
<td style="text-align:left;">
17/1174
</td>
<td style="text-align:left;">
77/12076
</td>
<td style="text-align:right;">
0.0009651
</td>
<td style="text-align:right;">
0.3249333
</td>
<td style="text-align:right;">
0.3247639
</td>
<td style="text-align:left;">
PRL/NPR1/PTPRM/AGBL5/CLDN5/ADGRB2/HSPG2/ISM1/LOC113993563/EPHA2/SEMA3E/SPRY2/KLF4/GADD45A/THBS4/FASLG/HGS
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1902430" class="uri">GO:1902430</a>
</td>
<td style="text-align:left;">
negative regulation of amyloid-beta formation
</td>
<td style="text-align:left;">
6/1174
</td>
<td style="text-align:left;">
14/12076
</td>
<td style="text-align:right;">
0.0012596
</td>
<td style="text-align:right;">
0.3562365
</td>
<td style="text-align:right;">
0.3560508
</td>
<td style="text-align:left;">
LOC113988015/LOC113987965/SORL1/CHRNA7/PIN1/PRNP
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030259" class="uri">GO:0030259</a>
</td>
<td style="text-align:left;">
lipid glycosylation
</td>
<td style="text-align:left;">
5/1174
</td>
<td style="text-align:left;">
10/12076
</td>
<td style="text-align:right;">
0.0014289
</td>
<td style="text-align:right;">
0.3562365
</td>
<td style="text-align:right;">
0.3560508
</td>
<td style="text-align:left;">
ST3GAL2/ST3GAL4/B3GALT1/LOC114003933/GBGT1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007015" class="uri">GO:0007015</a>
</td>
<td style="text-align:left;">
actin filament organization
</td>
<td style="text-align:left;">
21/1174
</td>
<td style="text-align:left;">
109/12076
</td>
<td style="text-align:right;">
0.0016920
</td>
<td style="text-align:right;">
0.3562365
</td>
<td style="text-align:right;">
0.3560508
</td>
<td style="text-align:left;">
SAMD14/LOC113988218/LOC113992312/CORO1B/SORBS3/INPPL1/KPTN/LOC114003556/TPM4/TMOD1/CORO6/MYO1D/SHROOM4/FAT1/RHOH/HIP1/SHROOM2/LMOD1/HIP1R/MYO5C/RHOB
</td>
<td style="text-align:right;">
21
</td>
<td style="text-align:right;">
1174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070527" class="uri">GO:0070527</a>
</td>
<td style="text-align:left;">
platelet aggregation
</td>
<td style="text-align:left;">
11/1174
</td>
<td style="text-align:left;">
42/12076
</td>
<td style="text-align:right;">
0.0017635
</td>
<td style="text-align:right;">
0.3562365
</td>
<td style="text-align:right;">
0.3560508
</td>
<td style="text-align:left;">
HSPB1/ITGA2B/MYL9/STXBP1/COMP/FIBP/PEAR1/LOC114002136/RAP2B/ITGB3/METAP1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1174
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="ventromedial-hypothalamus-vmh" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Ventromedial
Hypothalamus (VMH)</h1>
<div id="checking-the-sampling-2" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Checking the
sampling</h2>
<p>Before filtering we had 16854 genes, after filtering for mean read
count and excluding genes where &gt;50% of samples had a count of 0 we
are left with 15220</p>
<p>Before I go into more detail, I am going to check that the sampling
is equal across batches with respect to our variables of interest. With
respect to Status we have vmh samples for 16, with respect to T and the
tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.76
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00078
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<p><img src="03_DESeq2_files/figure-html/VMH%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The majority of our data in VMH is in run 1.</p>
<div id="expression-data-exploration-3" class="section level3"
number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/VMH%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/VMH%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-3" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Corrections
applied</h3>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks like controlling for Batch is sufficient here, as Batch is
highly correlated with Year.</p>
</div>
</div>
<div id="status-vmh" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Status VMH</h2>
<div id="analysis-6" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15220 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 4, 0.026%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/VMH%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-6" class="section level3" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/VMH%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/VMH%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in VMH according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0032981" class="uri">GO:0032981</a>
</td>
<td style="text-align:left;">
mitochondrial respiratory chain complex I assembly
</td>
<td style="text-align:left;">
10/615
</td>
<td style="text-align:left;">
49/12283
</td>
<td style="text-align:right;">
0.0001285
</td>
<td style="text-align:right;">
0.1376210
</td>
<td style="text-align:right;">
0.1376210
</td>
<td style="text-align:left;">
ACAD9/NDUFA5/NDUFS7/TIMM21/NDUFA10/NDUFS3/NDUFV2/NDUFB9/NDUFA8/NDUFAF3
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
13/615
</td>
<td style="text-align:left;">
81/12283
</td>
<td style="text-align:right;">
0.0001773
</td>
<td style="text-align:right;">
0.1376210
</td>
<td style="text-align:right;">
0.1376210
</td>
<td style="text-align:left;">
RPS15A/RPS23/RPS29/RPL27A/RPL35A/RPL19/RPS21/RPS24/UBA52/LOC113989206/RPS6/RPS8/RPL6
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000184" class="uri">GO:0000184</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
</td>
<td style="text-align:left;">
14/615
</td>
<td style="text-align:left;">
112/12283
</td>
<td style="text-align:right;">
0.0013789
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:left;">
RPS15A/RPS23/RPS29/RPL27A/RPL35A/RPL19/RPS21/RPS24/UBA52/PNLDC1/LOC113989206/RPS6/RPS8/RPL6
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019083" class="uri">GO:0019083</a>
</td>
<td style="text-align:left;">
viral transcription
</td>
<td style="text-align:left;">
13/615
</td>
<td style="text-align:left;">
100/12283
</td>
<td style="text-align:right;">
0.0014081
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:left;">
RPS15A/RPS23/RPS29/RPL27A/RPL35A/RPL19/RPS21/RPS24/UBA52/LOC113989206/RPS6/RPS8/RPL6
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051604" class="uri">GO:0051604</a>
</td>
<td style="text-align:left;">
protein maturation
</td>
<td style="text-align:left;">
5/615
</td>
<td style="text-align:left;">
18/12283
</td>
<td style="text-align:right;">
0.0015383
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:right;">
0.4774939
</td>
<td style="text-align:left;">
TSPAN5/LOC114002160/STUB1/TSPAN15/TSPAN33
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006413" class="uri">GO:0006413</a>
</td>
<td style="text-align:left;">
translational initiation
</td>
<td style="text-align:left;">
14/615
</td>
<td style="text-align:left;">
119/12283
</td>
<td style="text-align:right;">
0.0024636
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:left;">
RPS15A/RPS23/RPS29/RPL27A/RPL35A/RPL19/RPS21/RPS24/UBA52/EIF3J/LOC113989206/RPS6/RPS8/RPL6
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006120" class="uri">GO:0006120</a>
</td>
<td style="text-align:left;">
mitochondrial electron transport, NADH to ubiquinone
</td>
<td style="text-align:left;">
7/615
</td>
<td style="text-align:left;">
38/12283
</td>
<td style="text-align:right;">
0.0024770
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:left;">
NDUFA5/NDUFS7/NDUFA10/NDUFS3/NDUFV2/NDUFB9/NDUFA8
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:left;">
16/615
</td>
<td style="text-align:left;">
148/12283
</td>
<td style="text-align:right;">
0.0030068
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:left;">
IGF2BP3/RPS15A/RPS23/RPS29/RPL27A/RPL35A/RPL19/RPS21/MRPL13/RPS24/GGT7/LOC113989206/RMND1/RPS6/RPS8/RPL6
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070207" class="uri">GO:0070207</a>
</td>
<td style="text-align:left;">
protein homotrimerization
</td>
<td style="text-align:left;">
4/615
</td>
<td style="text-align:left;">
13/12283
</td>
<td style="text-align:right;">
0.0030966
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:right;">
0.5339842
</td>
<td style="text-align:left;">
CLYBL/LOC113994845/LOC113996480/LOC113989423
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
615
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031145" class="uri">GO:0031145</a>
</td>
<td style="text-align:left;">
anaphase-promoting complex-dependent catabolic process
</td>
<td style="text-align:left;">
9/615
</td>
<td style="text-align:left;">
65/12283
</td>
<td style="text-align:right;">
0.0048481
</td>
<td style="text-align:right;">
0.7524203
</td>
<td style="text-align:right;">
0.7524203
</td>
<td style="text-align:left;">
UBB/ANAPC7/PSMF1/ANAPC2/PSMB7/PSMB4/UBA52/LOC113998129/PSMC5
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
615
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-vmh" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Testosterone VMH</h2>
<p><br></p>
<div id="expression-data-exploration-4" class="section level3"
number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/VMH%20DESeq%20continuous%20variable%20subset-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20DESeq%20continuous%20variable%20subset-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-4" class="section level3" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Batch does not account for year here… so have to include both…</p>
<p><br></p>
</div>
<div id="analysis-7" class="section level3" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Analysis</h3>
<p>Formula: ~ Batch + Year + mean_T</p>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 8, 0.053%
FALSE outliers [1]       : 52, 0.34%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/VMH%20mean%20T%20%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-7" class="section level3" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/VMH%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/VMH%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in VMH according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
22/514
</td>
<td style="text-align:left;">
209/12262
</td>
<td style="text-align:right;">
0.0000660
</td>
<td style="text-align:right;">
0.0918544
</td>
<td style="text-align:right;">
0.0918544
</td>
<td style="text-align:left;">
VIT/COL9A1/COL1A1/COL7A1/PDGFRA/COL5A2/ADAMTS14/CCDC80/ITGB3/FBLN5/COLQ/NR2E1/LOC113997171/COL4A2/ADAMTS7/FN1/TGFBI/ADAMTSL5/LOC113998180/LAMB2/LOC113982147/LOC113993281
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060285" class="uri">GO:0060285</a>
</td>
<td style="text-align:left;">
cilium-dependent cell motility
</td>
<td style="text-align:left;">
5/514
</td>
<td style="text-align:left;">
13/12262
</td>
<td style="text-align:right;">
0.0001235
</td>
<td style="text-align:right;">
0.0918544
</td>
<td style="text-align:right;">
0.0918544
</td>
<td style="text-align:left;">
DNAH1/DNAH12/RFX3/CCDC65/LOC113984044
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007611" class="uri">GO:0007611</a>
</td>
<td style="text-align:left;">
learning or memory
</td>
<td style="text-align:left;">
8/514
</td>
<td style="text-align:left;">
50/12262
</td>
<td style="text-align:right;">
0.0010246
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
GRIN2A/FZD9/S100B/LHX8/MME/CASP3/PSEN1/PTPRZ1
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090023" class="uri">GO:0090023</a>
</td>
<td style="text-align:left;">
positive regulation of neutrophil chemotaxis
</td>
<td style="text-align:left;">
5/514
</td>
<td style="text-align:left;">
21/12262
</td>
<td style="text-align:right;">
0.0014776
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
LOC113988506/LOC113985892/DNM1L/LOC113988345/RIPOR2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007601" class="uri">GO:0007601</a>
</td>
<td style="text-align:left;">
visual perception
</td>
<td style="text-align:left;">
15/514
</td>
<td style="text-align:left;">
153/12262
</td>
<td style="text-align:right;">
0.0019106
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
MFRP/COL1A1/LOC113999869/LOC113993535/NDP/PDE6G/LOC114004117/LOC113987867/NR2E1/ZIC2/SLC24A1/TGFBI/GLRB/LAMB2/LOC113982931
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042542" class="uri">GO:0042542</a>
</td>
<td style="text-align:left;">
response to hydrogen peroxide
</td>
<td style="text-align:left;">
7/514
</td>
<td style="text-align:left;">
44/12262
</td>
<td style="text-align:right;">
0.0021723
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
COL1A1/DUSP1/STARD3/CASP3/FKBP1B/STK26/PRDX3
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002244" class="uri">GO:0002244</a>
</td>
<td style="text-align:left;">
hematopoietic progenitor cell differentiation
</td>
<td style="text-align:left;">
7/514
</td>
<td style="text-align:left;">
45/12262
</td>
<td style="text-align:right;">
0.0024815
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
LOC114000366/PDGFRA/PTPRC/TENT2/LOC113997171/PSEN1/SIN3A
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060041" class="uri">GO:0060041</a>
</td>
<td style="text-align:left;">
retina development in camera-type eye
</td>
<td style="text-align:left;">
7/514
</td>
<td style="text-align:left;">
45/12262
</td>
<td style="text-align:right;">
0.0024815
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:right;">
0.4615527
</td>
<td style="text-align:left;">
MFRP/TENT2/LOC114004117/LOC113987867/NR2E1/LAMB2/LOC113982931
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036159" class="uri">GO:0036159</a>
</td>
<td style="text-align:left;">
inner dynein arm assembly
</td>
<td style="text-align:left;">
4/514
</td>
<td style="text-align:left;">
15/12262
</td>
<td style="text-align:right;">
0.0028794
</td>
<td style="text-align:right;">
0.4760682
</td>
<td style="text-align:right;">
0.4760682
</td>
<td style="text-align:left;">
LOC113981955/DNAH1/DNAH12/TEKT2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
514
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001947" class="uri">GO:0001947</a>
</td>
<td style="text-align:left;">
heart looping
</td>
<td style="text-align:left;">
7/514
</td>
<td style="text-align:left;">
48/12262
</td>
<td style="text-align:right;">
0.0036142
</td>
<td style="text-align:right;">
0.5377997
</td>
<td style="text-align:right;">
0.5377997
</td>
<td style="text-align:left;">
NDRG4/ZIC3/WNT5A/VANGL2/TMED2/PSEN1/SHH
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
514
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="status-x-meant" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Status x meanT</h2>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 10, 0.066%
FALSE LFC &lt; 0 (down)     : 16, 0.11%
FALSE outliers [1]       : 141, 0.93%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20mean%20T%20x%20Status-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-20-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-vmh" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Strength VMH</h2>
<div id="analysis-8" class="section level3" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Analysis</h3>
<p>Formula ~ Batch + Year + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15162 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 28, 0.18%
FALSE LFC &lt; 0 (down)     : 19, 0.13%
FALSE outliers [1]       : 82, 0.54%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/vmh%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/VMH%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/VMH%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-8" class="section level3" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/VMH%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/VMH%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in VMH according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0042472" class="uri">GO:0042472</a>
</td>
<td style="text-align:left;">
inner ear morphogenesis
</td>
<td style="text-align:left;">
5/174
</td>
<td style="text-align:left;">
40/12262
</td>
<td style="text-align:right;">
0.0002391
</td>
<td style="text-align:right;">
0.1730775
</td>
<td style="text-align:right;">
0.1643205
</td>
<td style="text-align:left;">
FGFR2/COL2A1/USH1C/FOXG1/KCNQ4
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006488" class="uri">GO:0006488</a>
</td>
<td style="text-align:left;">
dolichol-linked oligosaccharide biosynthetic process
</td>
<td style="text-align:left;">
3/174
</td>
<td style="text-align:left;">
15/12262
</td>
<td style="text-align:right;">
0.0011271
</td>
<td style="text-align:right;">
0.2719956
</td>
<td style="text-align:right;">
0.2582337
</td>
<td style="text-align:left;">
ALG1/SRD5A3/ALG13
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036159" class="uri">GO:0036159</a>
</td>
<td style="text-align:left;">
inner dynein arm assembly
</td>
<td style="text-align:left;">
3/174
</td>
<td style="text-align:left;">
15/12262
</td>
<td style="text-align:right;">
0.0011271
</td>
<td style="text-align:right;">
0.2719956
</td>
<td style="text-align:right;">
0.2582337
</td>
<td style="text-align:left;">
DNAH12/LOC113981955/CCDC39
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007605" class="uri">GO:0007605</a>
</td>
<td style="text-align:left;">
sensory perception of sound
</td>
<td style="text-align:left;">
7/174
</td>
<td style="text-align:left;">
126/12262
</td>
<td style="text-align:right;">
0.0021050
</td>
<td style="text-align:right;">
0.3234432
</td>
<td style="text-align:right;">
0.3070782
</td>
<td style="text-align:left;">
LOC113990629/COL1A1/OTOG/COL2A1/USH1C/KCNQ4/ATP6V0A4
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
4/174
</td>
<td style="text-align:left;">
40/12262
</td>
<td style="text-align:right;">
0.0024034
</td>
<td style="text-align:right;">
0.3234432
</td>
<td style="text-align:right;">
0.3070782
</td>
<td style="text-align:left;">
CFAP61/DNAH12/LOC113981955/CCDC39
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010765" class="uri">GO:0010765</a>
</td>
<td style="text-align:left;">
positive regulation of sodium ion transport
</td>
<td style="text-align:left;">
3/174
</td>
<td style="text-align:left;">
20/12262
</td>
<td style="text-align:right;">
0.0026805
</td>
<td style="text-align:right;">
0.3234432
</td>
<td style="text-align:right;">
0.3070782
</td>
<td style="text-align:left;">
LOC113986459/LOC113988585/LOC113999252
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021537" class="uri">GO:0021537</a>
</td>
<td style="text-align:left;">
telencephalon development
</td>
<td style="text-align:left;">
3/174
</td>
<td style="text-align:left;">
23/12262
</td>
<td style="text-align:right;">
0.0040363
</td>
<td style="text-align:right;">
0.3331375
</td>
<td style="text-align:right;">
0.3162820
</td>
<td style="text-align:left;">
LOC113986459/LOC113988585/RFX4
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050679" class="uri">GO:0050679</a>
</td>
<td style="text-align:left;">
positive regulation of epithelial cell proliferation
</td>
<td style="text-align:left;">
4/174
</td>
<td style="text-align:left;">
52/12262
</td>
<td style="text-align:right;">
0.0062423
</td>
<td style="text-align:right;">
0.3331375
</td>
<td style="text-align:right;">
0.3162820
</td>
<td style="text-align:left;">
LOC113986459/LOC113988585/FGFR2/WDR48
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000077" class="uri">GO:0000077</a>
</td>
<td style="text-align:left;">
DNA damage checkpoint
</td>
<td style="text-align:left;">
3/174
</td>
<td style="text-align:left;">
29/12262
</td>
<td style="text-align:right;">
0.0078260
</td>
<td style="text-align:right;">
0.3331375
</td>
<td style="text-align:right;">
0.3162820
</td>
<td style="text-align:left;">
DOT1L/CLSPN/ERCC6
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000076" class="uri">GO:0000076</a>
</td>
<td style="text-align:left;">
DNA replication checkpoint
</td>
<td style="text-align:left;">
2/174
</td>
<td style="text-align:left;">
10/12262
</td>
<td style="text-align:right;">
0.0083599
</td>
<td style="text-align:right;">
0.3331375
</td>
<td style="text-align:right;">
0.3162820
</td>
<td style="text-align:left;">
CLSPN/TIPIN
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
174
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="anterior-hypothalamus-ah" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Anterior Hypothalamus
(AH)</h1>
<p><br></p>
<div id="checking-the-sampling-3" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Checking the
sampling</h2>
<p>For the AH, we started with 16854 genes but after filering we have
14926 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have ah samples for 11, with respect
to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-5" class="section level3"
number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/AH%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
<p>PFT1 and PFT11 cluster with the BSTm as previously identified by the
brain tissue PCA plot and the all tissue WGCNA dendrogram. We know that
there were issues with the sectioning plane in individual PFT1 that
could affect this tissue, but not PFT11.</p>
<p>The effects of removing these samples will be explored for each
analysis vs removing a randomly selected pair of samples. These results
(not shown) showed similar effects, so these samples were not
excluded.</p>
<p><br></p>
</div>
</div>
<div id="status-ah" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Status AH</h2>
<div id="analysis-9" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Analysis</h3>
<p>Formula = ~ Status</p>
<p><img src="03_DESeq2_files/figure-html/AH%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AH%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-9" class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AH%20Status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AH%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AH according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0071850" class="uri">GO:0071850</a>
</td>
<td style="text-align:left;">
mitotic cell cycle arrest
</td>
<td style="text-align:left;">
7/1048
</td>
<td style="text-align:left;">
14/12134
</td>
<td style="text-align:right;">
0.0000701
</td>
<td style="text-align:right;">
0.1385595
</td>
<td style="text-align:right;">
0.1385595
</td>
<td style="text-align:left;">
DUSP1/CDC14A/RGCC/CDKN1A/DDIAS/CDC14B/FAP
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051017" class="uri">GO:0051017</a>
</td>
<td style="text-align:left;">
actin filament bundle assembly
</td>
<td style="text-align:left;">
9/1048
</td>
<td style="text-align:left;">
29/12134
</td>
<td style="text-align:right;">
0.0005283
</td>
<td style="text-align:right;">
0.5222015
</td>
<td style="text-align:right;">
0.5222015
</td>
<td style="text-align:left;">
DPYSL3/USH1C/ADD2/AIF1L/CALD1/ESPN/EZR/FAM107A/SHROOM2
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000226" class="uri">GO:0000226</a>
</td>
<td style="text-align:left;">
microtubule cytoskeleton organization
</td>
<td style="text-align:left;">
20/1048
</td>
<td style="text-align:left;">
109/12134
</td>
<td style="text-align:right;">
0.0009542
</td>
<td style="text-align:right;">
0.6288193
</td>
<td style="text-align:right;">
0.6288193
</td>
<td style="text-align:left;">
LOC113996267/LOC113991233/TUBB3/WEE1/MAPT/TACC1/CDK5/CDK5RAP2/SON/APC2/CDC14A/LOC113995052/TBCD/TUBA8/TUBG1/GAS2/SIK3/TBCEL/HAUS6/CDC14B
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010389" class="uri">GO:0010389</a>
</td>
<td style="text-align:left;">
regulation of G2/M transition of mitotic cell cycle
</td>
<td style="text-align:left;">
15/1048
</td>
<td style="text-align:left;">
75/12134
</td>
<td style="text-align:right;">
0.0016430
</td>
<td style="text-align:right;">
0.8120696
</td>
<td style="text-align:right;">
0.8120696
</td>
<td style="text-align:left;">
LOC113996267/LOC113984333/CCP110/ALMS1/LOC113999915/DCTN2/CDK5RAP2/PRKAR2B/LOC113995052/NEDD1/PPP2R1A/TUBG1/LOC113986004/TPX2/HAUS6
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051604" class="uri">GO:0051604</a>
</td>
<td style="text-align:left;">
protein maturation
</td>
<td style="text-align:left;">
6/1048
</td>
<td style="text-align:left;">
18/12134
</td>
<td style="text-align:right;">
0.0030639
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
TSPAN33/LMF2/TSPAN14/GALNT2/TSPAN32/LMF1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000278" class="uri">GO:0000278</a>
</td>
<td style="text-align:left;">
mitotic cell cycle
</td>
<td style="text-align:left;">
19/1048
</td>
<td style="text-align:left;">
113/12134
</td>
<td style="text-align:right;">
0.0035755
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
LOC113996267/RRM1/CLTC/LOC113991233/TUBB3/LOC114001298/CDC14A/LOC113995052/NUDT15/TBCD/TUBA8/TADA3/TUBG1/TPX2/ABRAXAS2/LOC113997232/YEATS4/CDC14B/WAPL
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070509" class="uri">GO:0070509</a>
</td>
<td style="text-align:left;">
calcium ion import
</td>
<td style="text-align:left;">
7/1048
</td>
<td style="text-align:left;">
25/12134
</td>
<td style="text-align:right;">
0.0042182
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
LOC113989062/MCUR1/LOC113994017/CDK5/LOC114000517/LOC113993166/SLC30A1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000086" class="uri">GO:0000086</a>
</td>
<td style="text-align:left;">
G2/M transition of mitotic cell cycle
</td>
<td style="text-align:left;">
20/1048
</td>
<td style="text-align:left;">
127/12134
</td>
<td style="text-align:right;">
0.0060890
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
LOC113996267/LOC113984333/CCP110/USH1C/WEE1/ALMS1/FBXL17/LOC113999915/DCTN2/CDK7/CDK5RAP2/PRKAR2B/LOC113995052/NEDD1/PPP2R1A/TUBG1/CDKN1A/LOC113986004/TPD52L1/HAUS6
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051895" class="uri">GO:0051895</a>
</td>
<td style="text-align:left;">
negative regulation of focal adhesion assembly
</td>
<td style="text-align:left;">
6/1048
</td>
<td style="text-align:left;">
21/12134
</td>
<td style="text-align:right;">
0.0071579
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
LOC113988419/LOC113991284/SRC/BCAS3/FAM107A/LRP1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1048
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016485" class="uri">GO:0016485</a>
</td>
<td style="text-align:left;">
protein processing
</td>
<td style="text-align:left;">
11/1048
</td>
<td style="text-align:left;">
56/12134
</td>
<td style="text-align:right;">
0.0075717
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:right;">
0.9303018
</td>
<td style="text-align:left;">
NCSTN/LOC113991284/ECEL1/PCSK7/AEBP1/MMEL1/ECE2/FURIN/ADAM10/PCSK6/PSEN1
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1048
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-ah" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Testosterone AH</h2>
<div id="analysis-10" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Analysis</h3>
<p>Formula ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 1, 0.0067%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 33, 0.22%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AH%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" />
<br></p>
</div>
<div id="gene-ontology-10" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AH%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AH%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AH according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0042572" class="uri">GO:0042572</a>
</td>
<td style="text-align:left;">
retinol metabolic process
</td>
<td style="text-align:left;">
8/632
</td>
<td style="text-align:left;">
32/12134
</td>
<td style="text-align:right;">
0.0001789
</td>
<td style="text-align:right;">
0.2810090
</td>
<td style="text-align:right;">
0.2810090
</td>
<td style="text-align:left;">
ALDH1A3/LOC113995092/LOC113996821/LOC113999436/LOC114001898/LOC114001962/PNPLA2/RETSAT
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072205" class="uri">GO:0072205</a>
</td>
<td style="text-align:left;">
metanephric collecting duct development
</td>
<td style="text-align:left;">
4/632
</td>
<td style="text-align:left;">
10/12134
</td>
<td style="text-align:right;">
0.0011897
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
CALB1/DLG5/LOC114001962/SHH
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006414" class="uri">GO:0006414</a>
</td>
<td style="text-align:left;">
translational elongation
</td>
<td style="text-align:left;">
5/632
</td>
<td style="text-align:left;">
17/12134
</td>
<td style="text-align:right;">
0.0013827
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
ABTB1/EEF2/GTPBP1/LOC113991380/TSFM
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042552" class="uri">GO:0042552</a>
</td>
<td style="text-align:left;">
myelination
</td>
<td style="text-align:left;">
8/632
</td>
<td style="text-align:left;">
45/12134
</td>
<td style="text-align:right;">
0.0020169
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
ACSBG1/ATRN/CMTM8/EGR2/LOC114001565/LOC114003174/NFASC/SERINC5
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009617" class="uri">GO:0009617</a>
</td>
<td style="text-align:left;">
response to bacterium
</td>
<td style="text-align:left;">
8/632
</td>
<td style="text-align:left;">
46/12134
</td>
<td style="text-align:right;">
0.0023324
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
C3/GPM6A/LOC113995953/LPL/LSM5/NAALADL2/RAB29/UPK1B
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061640" class="uri">GO:0061640</a>
</td>
<td style="text-align:left;">
cytoskeleton-dependent cytokinesis
</td>
<td style="text-align:left;">
5/632
</td>
<td style="text-align:left;">
19/12134
</td>
<td style="text-align:right;">
0.0023826
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
NOX5/LOC113992196/SEPT10/SEPT2/SEPT4
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030301" class="uri">GO:0030301</a>
</td>
<td style="text-align:left;">
cholesterol transport
</td>
<td style="text-align:left;">
4/632
</td>
<td style="text-align:left;">
12/12134
</td>
<td style="text-align:right;">
0.0025793
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:right;">
0.5788714
</td>
<td style="text-align:left;">
APOA1/CD36/LCAT/OSBPL5
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043691" class="uri">GO:0043691</a>
</td>
<td style="text-align:left;">
reverse cholesterol transport
</td>
<td style="text-align:left;">
4/632
</td>
<td style="text-align:left;">
13/12134
</td>
<td style="text-align:right;">
0.0035736
</td>
<td style="text-align:right;">
0.7017611
</td>
<td style="text-align:right;">
0.7017611
</td>
<td style="text-align:left;">
APOA1/APOC3/LCAT/LOC113997963
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072089" class="uri">GO:0072089</a>
</td>
<td style="text-align:left;">
stem cell proliferation
</td>
<td style="text-align:left;">
4/632
</td>
<td style="text-align:left;">
14/12134
</td>
<td style="text-align:right;">
0.0047992
</td>
<td style="text-align:right;">
0.7821617
</td>
<td style="text-align:right;">
0.7821617
</td>
<td style="text-align:left;">
ABCB5/CD34/FGF2/RNF43
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
632
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042110" class="uri">GO:0042110</a>
</td>
<td style="text-align:left;">
T cell activation
</td>
<td style="text-align:left;">
5/632
</td>
<td style="text-align:left;">
23/12134
</td>
<td style="text-align:right;">
0.0058025
</td>
<td style="text-align:right;">
0.7821617
</td>
<td style="text-align:right;">
0.7821617
</td>
<td style="text-align:left;">
CD44/KIF13B/PPP3CA/PPP3CB/RAB29
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
632
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="status-x-mean-t" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Status x mean T</h2>
<pre><code>FALSE 
FALSE out of 14926 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 1220, 8.2%
FALSE LFC &lt; 0 (down)     : 1340, 9%
FALSE outliers [1]       : 32, 0.21%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-25-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-ah" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Strength AH</h2>
<div id="analysis-11" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14900 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 1, 0.0067%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AH%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AH%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-11" class="section level3" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AH%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AH%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AH according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0008217" class="uri">GO:0008217</a>
</td>
<td style="text-align:left;">
regulation of blood pressure
</td>
<td style="text-align:left;">
9/478
</td>
<td style="text-align:left;">
58/12115
</td>
<td style="text-align:right;">
0.0004087
</td>
<td style="text-align:right;">
0.5554226
</td>
<td style="text-align:right;">
0.5554226
</td>
<td style="text-align:left;">
LOC113990682/NPR2/ATP1A1/GCGR/GCH1/MC3R/ACTA2/LOC113993563/EXTL1
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090305" class="uri">GO:0090305</a>
</td>
<td style="text-align:left;">
nucleic acid phosphodiester bond hydrolysis
</td>
<td style="text-align:left;">
8/478
</td>
<td style="text-align:left;">
52/12115
</td>
<td style="text-align:right;">
0.0009033
</td>
<td style="text-align:right;">
0.6137750
</td>
<td style="text-align:right;">
0.6137750
</td>
<td style="text-align:left;">
RAD9B/PLSCR1/GEN1/RBBP8/ASTE1/ENPP1/ENPP2/LOC114001323
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042493" class="uri">GO:0042493</a>
</td>
<td style="text-align:left;">
response to drug
</td>
<td style="text-align:left;">
19/478
</td>
<td style="text-align:left;">
226/12115
</td>
<td style="text-align:right;">
0.0015653
</td>
<td style="text-align:right;">
0.7091026
</td>
<td style="text-align:right;">
0.7091026
</td>
<td style="text-align:left;">
DAB1/SFRP2/LOC114000060/LOC114000287/MCM7/ATP1A1/WFDC1/ABCB5/GNAO1/GGH/APOD/LOC113989520/TP73/SORD/ADIPOR2/ATR/AGPAT2/PAM/SREBF1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006911" class="uri">GO:0006911</a>
</td>
<td style="text-align:left;">
phagocytosis, engulfment
</td>
<td style="text-align:left;">
4/478
</td>
<td style="text-align:left;">
17/12115
</td>
<td style="text-align:right;">
0.0037808
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
ARHGAP12/DOCK1/MFGE8/GSN
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051000" class="uri">GO:0051000</a>
</td>
<td style="text-align:left;">
positive regulation of nitric-oxide synthase activity
</td>
<td style="text-align:left;">
4/478
</td>
<td style="text-align:left;">
17/12115
</td>
<td style="text-align:right;">
0.0037808
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
NOS1AP/PIK3CB/GCH1/S100A1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006814" class="uri">GO:0006814</a>
</td>
<td style="text-align:left;">
sodium ion transport
</td>
<td style="text-align:left;">
7/478
</td>
<td style="text-align:left;">
52/12115
</td>
<td style="text-align:right;">
0.0040939
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
SLC5A3/SLC17A8/SLC12A3/SLC5A11/SLC13A4/SLC38A3/NDUFA9
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006027" class="uri">GO:0006027</a>
</td>
<td style="text-align:left;">
glycosaminoglycan catabolic process
</td>
<td style="text-align:left;">
5/478
</td>
<td style="text-align:left;">
28/12115
</td>
<td style="text-align:right;">
0.0043368
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
IDS/GLB1L2/SDC1/SGSH/FUCA1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007009" class="uri">GO:0007009</a>
</td>
<td style="text-align:left;">
plasma membrane organization
</td>
<td style="text-align:left;">
5/478
</td>
<td style="text-align:left;">
28/12115
</td>
<td style="text-align:right;">
0.0043368
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
LOC113987015/LOC113993146/TLCD2/LOC113991140/LOC113993758
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061025" class="uri">GO:0061025</a>
</td>
<td style="text-align:left;">
membrane fusion
</td>
<td style="text-align:left;">
6/478
</td>
<td style="text-align:left;">
42/12115
</td>
<td style="text-align:right;">
0.0057369
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
LOC113987015/LOC113993146/LOC113991140/LOC113993758/VAMP3/NSFL1C
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
478
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034341" class="uri">GO:0034341</a>
</td>
<td style="text-align:left;">
response to interferon-gamma
</td>
<td style="text-align:left;">
3/478
</td>
<td style="text-align:left;">
10/12115
</td>
<td style="text-align:right;">
0.0059520
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:right;">
0.7170805
</td>
<td style="text-align:left;">
SLC11A1/CALCOCO2/GCH1
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
478
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="paraventricular-nucleus-pvn" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Paraventricular Nucleus
(PVN)</h1>
<p><br></p>
<div id="checking-the-sampling-4" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Checking the
sampling</h2>
<p>For the PVN, we started with 16854 genes but after filering we have
15151 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have pvn samples for 12, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<div id="expression-data-exploration-6" class="section level3"
number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/PVN%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Looks like we don’t have any outliers, nor do we need to include Year
(or Batch) as a covariate.</p>
<p><br></p>
</div>
</div>
<div id="status-pvn" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Status PVN</h2>
<div id="analysis-12" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Analysis</h3>
<p>Formula = ~ Status</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 27, 0.18%
FALSE LFC &lt; 0 (down)     : 15, 0.099%
FALSE outliers [1]       : 19, 0.13%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PVN%20status%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-12" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/PVN%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PVN%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PVN according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
11/447
</td>
<td style="text-align:left;">
38/12245
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0000898
</td>
<td style="text-align:right;">
0.0000853
</td>
<td style="text-align:left;">
DNAH1/TTC25/LOC113981955/CFAP73/LOC113986306/DNAAF1/DNAI1/LOC114000425/LOC114000433/TTC29/ARMC4
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
12/447
</td>
<td style="text-align:left;">
70/12245
</td>
<td style="text-align:right;">
0.0000075
</td>
<td style="text-align:right;">
0.0050385
</td>
<td style="text-align:right;">
0.0047860
</td>
<td style="text-align:left;">
ITIH5/PROS1/A2ML1/ANXA2/LOC113992168/CRIM1/CD109/LOC113991485/LOC113989407/LOC113992817/SERPINB5/WFDC1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006695" class="uri">GO:0006695</a>
</td>
<td style="text-align:left;">
cholesterol biosynthetic process
</td>
<td style="text-align:left;">
8/447
</td>
<td style="text-align:left;">
34/12245
</td>
<td style="text-align:right;">
0.0000233
</td>
<td style="text-align:right;">
0.0104184
</td>
<td style="text-align:right;">
0.0098962
</td>
<td style="text-align:left;">
FDFT1/DHCR7/DHCR24/HMGCS1/LOC113987727/APOA1/INSIG1/ACLY
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045540" class="uri">GO:0045540</a>
</td>
<td style="text-align:left;">
regulation of cholesterol biosynthetic process
</td>
<td style="text-align:left;">
7/447
</td>
<td style="text-align:left;">
34/12245
</td>
<td style="text-align:right;">
0.0001883
</td>
<td style="text-align:right;">
0.0631879
</td>
<td style="text-align:right;">
0.0600208
</td>
<td style="text-align:left;">
FDFT1/DHCR7/HMGCS1/SREBF2/LOC113987727/SC5D/SREBF1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904659" class="uri">GO:1904659</a>
</td>
<td style="text-align:left;">
glucose transmembrane transport
</td>
<td style="text-align:left;">
5/447
</td>
<td style="text-align:left;">
17/12245
</td>
<td style="text-align:right;">
0.0002725
</td>
<td style="text-align:right;">
0.0731458
</td>
<td style="text-align:right;">
0.0694797
</td>
<td style="text-align:left;">
SLC5A3/MFSD4A/SLC2A12/LOC113984297/SLC5A1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007288" class="uri">GO:0007288</a>
</td>
<td style="text-align:left;">
sperm axoneme assembly
</td>
<td style="text-align:left;">
5/447
</td>
<td style="text-align:left;">
20/12245
</td>
<td style="text-align:right;">
0.0006236
</td>
<td style="text-align:right;">
0.1394892
</td>
<td style="text-align:right;">
0.1324978
</td>
<td style="text-align:left;">
DNAH1/CFAP65/CFAP44/NEURL1B/MEIG1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032570" class="uri">GO:0032570</a>
</td>
<td style="text-align:left;">
response to progesterone
</td>
<td style="text-align:left;">
6/447
</td>
<td style="text-align:left;">
31/12245
</td>
<td style="text-align:right;">
0.0007745
</td>
<td style="text-align:right;">
0.1484808
</td>
<td style="text-align:right;">
0.1410387
</td>
<td style="text-align:left;">
LOC113997556/TGFB2/CAV1/LOC113992301/SREBF1/GABRB1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071711" class="uri">GO:0071711</a>
</td>
<td style="text-align:left;">
basement membrane organization
</td>
<td style="text-align:left;">
4/447
</td>
<td style="text-align:left;">
13/12245
</td>
<td style="text-align:right;">
0.0009634
</td>
<td style="text-align:right;">
0.1616051
</td>
<td style="text-align:right;">
0.1535052
</td>
<td style="text-align:left;">
CAV1/EXTL1/NID1/RIC8A
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032094" class="uri">GO:0032094</a>
</td>
<td style="text-align:left;">
response to food
</td>
<td style="text-align:left;">
4/447
</td>
<td style="text-align:left;">
14/12245
</td>
<td style="text-align:right;">
0.0013101
</td>
<td style="text-align:right;">
0.1758144
</td>
<td style="text-align:right;">
0.1670023
</td>
<td style="text-align:left;">
SREBF1/LOC113992180/OPRM1/CPS1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
447
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048566" class="uri">GO:0048566</a>
</td>
<td style="text-align:left;">
embryonic digestive tract development
</td>
<td style="text-align:left;">
4/447
</td>
<td style="text-align:left;">
14/12245
</td>
<td style="text-align:right;">
0.0013101
</td>
<td style="text-align:right;">
0.1758144
</td>
<td style="text-align:right;">
0.1670023
</td>
<td style="text-align:left;">
LOC113997556/TGFB2/RARRES2/ALDH1A2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
447
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-pvn" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Testosterone PVN</h2>
<div id="analysis-13" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 7, 0.046%
FALSE LFC &lt; 0 (down)     : 6, 0.04%
FALSE outliers [1]       : 0, 0%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PVN%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-13" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/PVN%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PVN%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PVN according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0034587" class="uri">GO:0034587</a>
</td>
<td style="text-align:left;">
piRNA metabolic process
</td>
<td style="text-align:left;">
4/399
</td>
<td style="text-align:left;">
11/12245
</td>
<td style="text-align:right;">
0.0003055
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
MAEL/PIWIL1/FKBP6/HENMT1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
30/399
</td>
<td style="text-align:left;">
484/12245
</td>
<td style="text-align:right;">
0.0005584
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
POMC/PRLH/C3/MLNR/LOC113987091/PDGFRL/CYSLTR1/HTR2A/TACR3/ADGRV1/LOC113996140/LOC113997278/HRH3/FZD8/LOC114000970/GPR171/C3AR1/ADGRF4/PTHLH/HEBP1/FGD2/RASD1/LOC113993299/GRP/MCF2L/GPR6/LOC113999236/LOC114003556/ANXA1/LPAR1
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021542" class="uri">GO:0021542</a>
</td>
<td style="text-align:left;">
dentate gyrus development
</td>
<td style="text-align:left;">
4/399
</td>
<td style="text-align:left;">
14/12245
</td>
<td style="text-align:right;">
0.0008573
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
BTG2/EMX2/PROX1/ATAT1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050900" class="uri">GO:0050900</a>
</td>
<td style="text-align:left;">
leukocyte migration
</td>
<td style="text-align:left;">
10/399
</td>
<td style="text-align:left;">
97/12245
</td>
<td style="text-align:right;">
0.0012129
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
LOC113984271/CD34/LOC113984305/GPC1/COL1A1/GRB2/ITGB3/SHC1/CD44/LOC113989681
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048705" class="uri">GO:0048705</a>
</td>
<td style="text-align:left;">
skeletal system morphogenesis
</td>
<td style="text-align:left;">
5/399
</td>
<td style="text-align:left;">
27/12245
</td>
<td style="text-align:right;">
0.0015989
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
SPEF2/LOC113985828/HHIP/FMN1/LOC113996506
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006911" class="uri">GO:0006911</a>
</td>
<td style="text-align:left;">
phagocytosis, engulfment
</td>
<td style="text-align:left;">
4/399
</td>
<td style="text-align:left;">
17/12245
</td>
<td style="text-align:right;">
0.0018862
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
LOC113984271/LOC113984305/LOC113992914/GULP1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051591" class="uri">GO:0051591</a>
</td>
<td style="text-align:left;">
response to cAMP
</td>
<td style="text-align:left;">
5/399
</td>
<td style="text-align:left;">
28/12245
</td>
<td style="text-align:right;">
0.0018951
</td>
<td style="text-align:right;">
0.3676509
</td>
<td style="text-align:right;">
0.3553683
</td>
<td style="text-align:left;">
JUN/DUSP1/FOS/COL1A1/LOC114003020
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035924" class="uri">GO:0035924</a>
</td>
<td style="text-align:left;">
cellular response to vascular endothelial growth factor stimulus
</td>
<td style="text-align:left;">
5/399
</td>
<td style="text-align:left;">
29/12245
</td>
<td style="text-align:right;">
0.0022296
</td>
<td style="text-align:right;">
0.3784718
</td>
<td style="text-align:right;">
0.3658277
</td>
<td style="text-align:left;">
NR4A1/MAP2K3/FLT4/SPRY2/ANXA1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042127" class="uri">GO:0042127</a>
</td>
<td style="text-align:left;">
regulation of cell population proliferation
</td>
<td style="text-align:left;">
10/399
</td>
<td style="text-align:left;">
107/12245
</td>
<td style="text-align:right;">
0.0025434
</td>
<td style="text-align:right;">
0.3819735
</td>
<td style="text-align:right;">
0.3692124
</td>
<td style="text-align:left;">
JUN/PLA2G4A/NFKBIA/LOC113983017/LAMA5/JUND/SIX3/TGFB2/EPS15L1/LOC114003556
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
399
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035865" class="uri">GO:0035865</a>
</td>
<td style="text-align:left;">
cellular response to potassium ion
</td>
<td style="text-align:left;">
3/399
</td>
<td style="text-align:left;">
10/12245
</td>
<td style="text-align:right;">
0.0034734
</td>
<td style="text-align:right;">
0.3819735
</td>
<td style="text-align:right;">
0.3692124
</td>
<td style="text-align:left;">
CRHBP/PPARGC1A/LOC113983406
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
399
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="status-x-mean-t-1" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Status x mean T</h2>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 1, 0.0066%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 13, 0.086%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-32-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-pvn" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Strength PVN</h2>
<div id="analysis-14" class="section level3" number="5.5.1">
<h3><span class="header-section-number">5.5.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15151 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 29, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/PVN%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/PVN%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-14" class="section level3" number="5.5.2">
<h3><span class="header-section-number">5.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/PVN%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/PVN%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in PVN according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0016082" class="uri">GO:0016082</a>
</td>
<td style="text-align:left;">
synaptic vesicle priming
</td>
<td style="text-align:left;">
8/1044
</td>
<td style="text-align:left;">
20/12245
</td>
<td style="text-align:right;">
0.0001349
</td>
<td style="text-align:right;">
0.2591926
</td>
<td style="text-align:right;">
0.2591926
</td>
<td style="text-align:left;">
LOC113993146/SNAP25/LOC113987015/LOC113991140/LOC113993758/LOC114001826/LOC113986517/STXBP1
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016070" class="uri">GO:0016070</a>
</td>
<td style="text-align:left;">
RNA metabolic process
</td>
<td style="text-align:left;">
11/1044
</td>
<td style="text-align:left;">
47/12245
</td>
<td style="text-align:right;">
0.0016303
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
HNRNPA0/DDX54/DUSP11/HNRNPA2B1/HNRNPA3/HNRNPU/PCBP2/POLR2C/PTBP1/NCBP1/HNRNPK
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0120009" class="uri">GO:0120009</a>
</td>
<td style="text-align:left;">
intermembrane lipid transfer
</td>
<td style="text-align:left;">
9/1044
</td>
<td style="text-align:left;">
36/12245
</td>
<td style="text-align:right;">
0.0026239
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
ABCA1/ABCA3/PITPNM2/PITPNC1/PLEKHA3/TTPA/OSBP/PITPNM1/PRELID3A
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006342" class="uri">GO:0006342</a>
</td>
<td style="text-align:left;">
chromatin silencing
</td>
<td style="text-align:left;">
6/1044
</td>
<td style="text-align:left;">
18/12245
</td>
<td style="text-align:right;">
0.0028693
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
MECP2/EED/HMGB1/PCGF2/MORF4L1/LOC113997379
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006959" class="uri">GO:0006959</a>
</td>
<td style="text-align:left;">
humoral immune response
</td>
<td style="text-align:left;">
7/1044
</td>
<td style="text-align:left;">
24/12245
</td>
<td style="text-align:right;">
0.0030460
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
RBPJ/LOC113992699/LOC114001683/NOTCH1/NOTCH2/TNFRSF21/LOC114001688
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030316" class="uri">GO:0030316</a>
</td>
<td style="text-align:left;">
osteoclast differentiation
</td>
<td style="text-align:left;">
7/1044
</td>
<td style="text-align:left;">
24/12245
</td>
<td style="text-align:right;">
0.0030460
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
TCIRG1/CSF1R/LOC113996964/SBNO2/LOC114001688/GAB2/OSTM1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090026" class="uri">GO:0090026</a>
</td>
<td style="text-align:left;">
positive regulation of monocyte chemotaxis
</td>
<td style="text-align:left;">
5/1044
</td>
<td style="text-align:left;">
13/12245
</td>
<td style="text-align:right;">
0.0032106
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:right;">
0.8810927
</td>
<td style="text-align:left;">
CREB3L1/CREB3L2/MOSPD2/HMGB1/LOC113991763
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051402" class="uri">GO:0051402</a>
</td>
<td style="text-align:left;">
neuron apoptotic process
</td>
<td style="text-align:left;">
9/1044
</td>
<td style="text-align:left;">
38/12245
</td>
<td style="text-align:right;">
0.0038941
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:left;">
XIAP/ATN1/BNIP3/BOK/NCSTN/SCN1A/TNFRSF21/LOC113994464/BID
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010812" class="uri">GO:0010812</a>
</td>
<td style="text-align:left;">
negative regulation of cell-substrate adhesion
</td>
<td style="text-align:left;">
5/1044
</td>
<td style="text-align:left;">
14/12245
</td>
<td style="text-align:right;">
0.0046473
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:left;">
COL1A1/ANGPT2/CORO2B/NOTCH1/DMTN
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1044
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007568" class="uri">GO:0007568</a>
</td>
<td style="text-align:left;">
aging
</td>
<td style="text-align:left;">
20/1044
</td>
<td style="text-align:left;">
126/12245
</td>
<td style="text-align:right;">
0.0048296
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:right;">
0.9026353
</td>
<td style="text-align:left;">
FBXO4/CLDN1/ATG7/ENDOG/PAX2/EEF2/RPN2/CTNNA1/GCLM/GNAO1/HTR2A/LOC114000703/NUDT1/PENK/PITX3/PPARGC1A/VCAM1/WRN/ITGB2/NAPEPLD
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
1044
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="medial-preoptic-area-pom" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Medial Preoptic Area
(POM)</h1>
<div id="checking-the-sampling-5" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Checking the
sampling</h2>
<p>For the POM, we started with 16854 genes but after filering we have
15318 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have pom samples for 12, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00055
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-7" class="section level3"
number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/POM%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/POM%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-5" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-35-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-35-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Don’t need to control for year because accounting for Batch seems to
fix that.</p>
<p><br></p>
</div>
<div id="outlier-removed" class="section level3" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Outlier
removed</h3>
<p>PFT2 was previously identified as a potential outlier, and it’s
coming out quite separately here, even after all the corrections for
batch effects. So, this one I am going to REMOVE.</p>
<p><br></p>
<p>##Status POM</p>
<p>###Analysis</p>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15314 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 16, 0.1%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/POM%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-15" class="section level3" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/POM%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/POM%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in POM according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0042391" class="uri">GO:0042391</a>
</td>
<td style="text-align:left;">
regulation of membrane potential
</td>
<td style="text-align:left;">
14/628
</td>
<td style="text-align:left;">
82/12335
</td>
<td style="text-align:right;">
0.0000602
</td>
<td style="text-align:right;">
0.0967256
</td>
<td style="text-align:right;">
0.0954846
</td>
<td style="text-align:left;">
GABRA4/LOC113991128/LOC113990721/GRIN1/CHRNB4/GABRA5/CHRNA7/KCNH7/KCNH8/LOC113984913/RIMS4/CHRNA5/KCNA1/KCNH6
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071679" class="uri">GO:0071679</a>
</td>
<td style="text-align:left;">
commissural neuron axon guidance
</td>
<td style="text-align:left;">
5/628
</td>
<td style="text-align:left;">
12/12335
</td>
<td style="text-align:right;">
0.0001977
</td>
<td style="text-align:right;">
0.1589313
</td>
<td style="text-align:right;">
0.1568921
</td>
<td style="text-align:left;">
PTCH1/NFIB/ROBO3/VEGFA/NRP1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006816" class="uri">GO:0006816</a>
</td>
<td style="text-align:left;">
calcium ion transport
</td>
<td style="text-align:left;">
11/628
</td>
<td style="text-align:left;">
70/12335
</td>
<td style="text-align:right;">
0.0007607
</td>
<td style="text-align:right;">
0.4077390
</td>
<td style="text-align:right;">
0.4025075
</td>
<td style="text-align:left;">
ATP2B2/LOC113993165/GRIN3B/LOC113994017/CACNA1D/CHRNA7/LOC113982108/PSEN2/ITPR1/CACNA1E/CACNB3
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071805" class="uri">GO:0071805</a>
</td>
<td style="text-align:left;">
potassium ion transmembrane transport
</td>
<td style="text-align:left;">
13/628
</td>
<td style="text-align:left;">
99/12335
</td>
<td style="text-align:right;">
0.0014905
</td>
<td style="text-align:right;">
0.5329363
</td>
<td style="text-align:right;">
0.5260984
</td>
<td style="text-align:left;">
SLC9A3/CCDC51/KCNA4/KCND3/KCNH7/KCNH8/KCNK12/KCNK18/LOC113984913/KCNA1/KCNH6/KCNK15/SLC24A2
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008589" class="uri">GO:0008589</a>
</td>
<td style="text-align:left;">
regulation of smoothened signaling pathway
</td>
<td style="text-align:left;">
5/628
</td>
<td style="text-align:left;">
18/12335
</td>
<td style="text-align:right;">
0.0016571
</td>
<td style="text-align:right;">
0.5329363
</td>
<td style="text-align:right;">
0.5260984
</td>
<td style="text-align:left;">
PTCH1/INTU/IFT81/C2CD3/OTX2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050877" class="uri">GO:0050877</a>
</td>
<td style="text-align:left;">
nervous system process
</td>
<td style="text-align:left;">
7/628
</td>
<td style="text-align:left;">
39/12335
</td>
<td style="text-align:right;">
0.0031741
</td>
<td style="text-align:right;">
0.6453285
</td>
<td style="text-align:right;">
0.6370485
</td>
<td style="text-align:left;">
GABRA4/LOC113991128/LOC113990721/CHRNB4/GABRA5/CHRNA7/CHRNA5
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021756" class="uri">GO:0021756</a>
</td>
<td style="text-align:left;">
striatum development
</td>
<td style="text-align:left;">
4/628
</td>
<td style="text-align:left;">
13/12335
</td>
<td style="text-align:right;">
0.0032905
</td>
<td style="text-align:right;">
0.6453285
</td>
<td style="text-align:right;">
0.6370485
</td>
<td style="text-align:left;">
CNTNAP2/DRD2/BBS4/SECISBP2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007268" class="uri">GO:0007268</a>
</td>
<td style="text-align:left;">
chemical synaptic transmission
</td>
<td style="text-align:left;">
20/628
</td>
<td style="text-align:left;">
203/12335
</td>
<td style="text-align:right;">
0.0035735
</td>
<td style="text-align:right;">
0.6453285
</td>
<td style="text-align:right;">
0.6370485
</td>
<td style="text-align:left;">
GABRA4/SV2B/LOC113993165/LOC113991128/SLC18A2/HTR5A/SDCBP2/LOC113990721/LOC113994017/GRIN1/CHRNB4/GABRA5/CHRNA7/LOC113982386/SYN2/CHRM4/CHRNA5/KCNA1/CACNA1E/CACNB3
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060384" class="uri">GO:0060384</a>
</td>
<td style="text-align:left;">
innervation
</td>
<td style="text-align:left;">
4/628
</td>
<td style="text-align:left;">
14/12335
</td>
<td style="text-align:right;">
0.0044231
</td>
<td style="text-align:right;">
0.6453285
</td>
<td style="text-align:right;">
0.6370485
</td>
<td style="text-align:left;">
VCAM1/SULF2/GABRA5/SERPINE2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060749" class="uri">GO:0060749</a>
</td>
<td style="text-align:left;">
mammary gland alveolus development
</td>
<td style="text-align:left;">
4/628
</td>
<td style="text-align:left;">
14/12335
</td>
<td style="text-align:right;">
0.0044231
</td>
<td style="text-align:right;">
0.6453285
</td>
<td style="text-align:right;">
0.6370485
</td>
<td style="text-align:left;">
AR/SOCS2/VEGFA/LOC113998275
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
628
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-pom" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Testosterone POM</h2>
<div id="expression-data-exploration-8" class="section level3"
number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-15" class="section level3" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 34, 0.22%
FALSE LFC &lt; 0 (down)     : 11, 0.072%
FALSE outliers [1]       : 29, 0.19%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/POM%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-16" class="section level3" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/POM%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/POM%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in POM according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0070593" class="uri">GO:0070593</a>
</td>
<td style="text-align:left;">
dendrite self-avoidance
</td>
<td style="text-align:left;">
6/493
</td>
<td style="text-align:left;">
20/12310
</td>
<td style="text-align:right;">
0.0000960
</td>
<td style="text-align:right;">
0.1208173
</td>
<td style="text-align:right;">
0.1201815
</td>
<td style="text-align:left;">
VSTM2L/MYPN/CNTN2/ROBO4/NEXN/ROBO3
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010575" class="uri">GO:0010575</a>
</td>
<td style="text-align:left;">
positive regulation of vascular endothelial growth factor production
</td>
<td style="text-align:left;">
6/493
</td>
<td style="text-align:left;">
22/12310
</td>
<td style="text-align:right;">
0.0001726
</td>
<td style="text-align:right;">
0.1208173
</td>
<td style="text-align:right;">
0.1201815
</td>
<td style="text-align:left;">
C3/FLT4/BRCA1/IL6ST/RORA/CCBE1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
35/493
</td>
<td style="text-align:left;">
488/12310
</td>
<td style="text-align:right;">
0.0006065
</td>
<td style="text-align:right;">
0.2830267
</td>
<td style="text-align:right;">
0.2815371
</td>
<td style="text-align:left;">
PRLH/RGS8/MC5R/LHCGR/OPRL1/C3/GNRH1/GPR50/GPR171/RAC2/FGD3/MCF2L/RXFP3/GAL/SST/LOC113993299/CCL28/IAPP/RAPGEFL1/TSHR/LOC113989272/HRH3/SORT1/LOC114002880/GPR78/LOC113982386/SSTR1/P2RY6/HTR5A/LOC114004322/NTSR1/RIC8A/NPY/TIAM2/LOC113996140
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007565" class="uri">GO:0007565</a>
</td>
<td style="text-align:left;">
female pregnancy
</td>
<td style="text-align:left;">
8/493
</td>
<td style="text-align:left;">
52/12310
</td>
<td style="text-align:right;">
0.0009956
</td>
<td style="text-align:right;">
0.3484516
</td>
<td style="text-align:right;">
0.3466176
</td>
<td style="text-align:left;">
PAPPA/GNRH1/CRHBP/CORIN/LOC113989272/FOS/THBD/LOC113982386
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046849" class="uri">GO:0046849</a>
</td>
<td style="text-align:left;">
bone remodeling
</td>
<td style="text-align:left;">
4/493
</td>
<td style="text-align:left;">
14/12310
</td>
<td style="text-align:right;">
0.0018461
</td>
<td style="text-align:right;">
0.3793144
</td>
<td style="text-align:right;">
0.3773180
</td>
<td style="text-align:left;">
NOTCH2/TPH1/P3H4/RASSF2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007399" class="uri">GO:0007399</a>
</td>
<td style="text-align:left;">
nervous system development
</td>
<td style="text-align:left;">
22/493
</td>
<td style="text-align:left;">
279/12310
</td>
<td style="text-align:right;">
0.0019248
</td>
<td style="text-align:right;">
0.3793144
</td>
<td style="text-align:right;">
0.3773180
</td>
<td style="text-align:left;">
PLPPR1/CPLX2/GDPD4/NOTCH2/NRN1L/IGF2BP3/ZC3H12A/GPSM1/HES4/CAMK1D/LOC113991778/GDPD5/TTLL7/MTR/VEGFA/SMARCD1/PCDH7/RAPGEFL1/FOS/BPNT1/PRPS1/NR2E1
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007187" class="uri">GO:0007187</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway, coupled to cyclic
nucleotide second messenger
</td>
<td style="text-align:left;">
7/493
</td>
<td style="text-align:left;">
46/12310
</td>
<td style="text-align:right;">
0.0021843
</td>
<td style="text-align:right;">
0.3793144
</td>
<td style="text-align:right;">
0.3773180
</td>
<td style="text-align:left;">
MC5R/LHCGR/TSHR/HRH3/SSTR1/HTR5A/NPY
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:left;">
5/493
</td>
<td style="text-align:left;">
24/12310
</td>
<td style="text-align:right;">
0.0022802
</td>
<td style="text-align:right;">
0.3793144
</td>
<td style="text-align:right;">
0.3773180
</td>
<td style="text-align:left;">
LOC113992168/C3/LOC113984305/LOC113984271/CFI
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002064" class="uri">GO:0002064</a>
</td>
<td style="text-align:left;">
epithelial cell development
</td>
<td style="text-align:left;">
4/493
</td>
<td style="text-align:left;">
15/12310
</td>
<td style="text-align:right;">
0.0024384
</td>
<td style="text-align:right;">
0.3793144
</td>
<td style="text-align:right;">
0.3773180
</td>
<td style="text-align:left;">
LOC113992388/LOC114000433/LOC113987257/TP63
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
493
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033628" class="uri">GO:0033628</a>
</td>
<td style="text-align:left;">
regulation of cell adhesion mediated by integrin
</td>
<td style="text-align:left;">
4/493
</td>
<td style="text-align:left;">
17/12310
</td>
<td style="text-align:right;">
0.0039897
</td>
<td style="text-align:right;">
0.5430211
</td>
<td style="text-align:right;">
0.5401631
</td>
<td style="text-align:left;">
TESC/PTK2/LOC114002160/LYN
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
493
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-1" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 16, 0.1%
FALSE LFC &lt; 0 (down)     : 8, 0.052%
FALSE outliers [1]       : 13, 0.085%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-40-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-pom" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Strength POM</h2>
<div id="analysis-16" class="section level3" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15268 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 5, 0.033%
FALSE LFC &lt; 0 (down)     : 10, 0.065%
FALSE outliers [1]       : 48, 0.31%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/POM%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/POM%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" />
<br></p>
</div>
<div id="gene-ontology-17" class="section level3" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/POM%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/POM%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in POM according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0043312" class="uri">GO:0043312</a>
</td>
<td style="text-align:left;">
neutrophil degranulation
</td>
<td style="text-align:left;">
41/699
</td>
<td style="text-align:left;">
375/12310
</td>
<td style="text-align:right;">
0.0000408
</td>
<td style="text-align:right;">
0.0694043
</td>
<td style="text-align:right;">
0.0686333
</td>
<td style="text-align:left;">
HPSE/GNS/CD59/LOC114003217/TCIRG1/FUCA1/PSMD14/RAB18/TOLLIP/RAB7A/YPEL5/LOC113990017/SVIP/AP1M1/FGL2/GRN/FUCA2/SLC27A5/CRISPLD2/CTSB/SNAP25/APRT/COMMD9/LOC113992610/LTA4H/PRDX4/RAB24/HSP90AB1/PGAM1/PSMC3/ARHGAP45/RAP2C/LOC114002464/GUSB/PSMD12/PTPRC/PAFAH1B2/ACTR10/CREG1/DEGS1/LOC113996661
</td>
<td style="text-align:right;">
41
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030574" class="uri">GO:0030574</a>
</td>
<td style="text-align:left;">
collagen catabolic process
</td>
<td style="text-align:left;">
7/699
</td>
<td style="text-align:left;">
27/12310
</td>
<td style="text-align:right;">
0.0006035
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
MRC2/FURIN/ADAMTS14/MMP23B/CTSB/MMP15/LOC113997326
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016082" class="uri">GO:0016082</a>
</td>
<td style="text-align:left;">
synaptic vesicle priming
</td>
<td style="text-align:left;">
6/699
</td>
<td style="text-align:left;">
20/12310
</td>
<td style="text-align:right;">
0.0006399
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758/LOC113993146/LOC114002475/SNAP25
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030010" class="uri">GO:0030010</a>
</td>
<td style="text-align:left;">
establishment of cell polarity
</td>
<td style="text-align:left;">
8/699
</td>
<td style="text-align:left;">
37/12310
</td>
<td style="text-align:right;">
0.0009222
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
TCIRG1/SDCCAG8/RAB11FIP2/ARHGEF11/HSP90AB1/KIF26B/LOC113998953/CRK
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016079" class="uri">GO:0016079</a>
</td>
<td style="text-align:left;">
synaptic vesicle exocytosis
</td>
<td style="text-align:left;">
6/699
</td>
<td style="text-align:left;">
23/12310
</td>
<td style="text-align:right;">
0.0014401
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758/LOC113993146/SNAP25/CPLX1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1900006" class="uri">GO:1900006</a>
</td>
<td style="text-align:left;">
positive regulation of dendrite development
</td>
<td style="text-align:left;">
5/699
</td>
<td style="text-align:left;">
16/12310
</td>
<td style="text-align:right;">
0.0015037
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
OPA1/LRP8/LOC113989207/CRTC1/PTN
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2001224" class="uri">GO:2001224</a>
</td>
<td style="text-align:left;">
positive regulation of neuron migration
</td>
<td style="text-align:left;">
5/699
</td>
<td style="text-align:left;">
16/12310
</td>
<td style="text-align:right;">
0.0015037
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
PLAA/LOC113989207/SEMA3A/LOC113989827/LOC114003556
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019079" class="uri">GO:0019079</a>
</td>
<td style="text-align:left;">
viral genome replication
</td>
<td style="text-align:left;">
4/699
</td>
<td style="text-align:left;">
10/12310
</td>
<td style="text-align:right;">
0.0016432
</td>
<td style="text-align:right;">
0.3493887
</td>
<td style="text-align:right;">
0.3455077
</td>
<td style="text-align:left;">
KPNA6/HTATSF1/CTBP2/LAMTOR5
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033628" class="uri">GO:0033628</a>
</td>
<td style="text-align:left;">
regulation of cell adhesion mediated by integrin
</td>
<td style="text-align:left;">
5/699
</td>
<td style="text-align:left;">
17/12310
</td>
<td style="text-align:right;">
0.0020317
</td>
<td style="text-align:right;">
0.3839828
</td>
<td style="text-align:right;">
0.3797175
</td>
<td style="text-align:left;">
LYN/LOC114002160/TESC/LOC113998953/CRK
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
699
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051014" class="uri">GO:0051014</a>
</td>
<td style="text-align:left;">
actin filament severing
</td>
<td style="text-align:left;">
4/699
</td>
<td style="text-align:left;">
12/12310
</td>
<td style="text-align:right;">
0.0035355
</td>
<td style="text-align:right;">
0.5622677
</td>
<td style="text-align:right;">
0.5560220
</td>
<td style="text-align:left;">
SCIN/VILL/CFL2/SVIL
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
699
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="dorsomedial-intercollicular-nucleus-ico" class="section level1"
number="7">
<h1><span class="header-section-number">7</span> Dorsomedial
Intercollicular nucleus (ICo)</h1>
<p><br></p>
<div id="checking-the-sampling-6" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Checking the
sampling</h2>
<p>For the ICO, we started with 16854 genes but after filering we have
15069 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have ico samples for 11, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.41
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.015
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="03_DESeq2_files/figure-html/ICO%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="expression-data-exploration-9" class="section level3"
number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/ICO%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-6" class="section level3" number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-42-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-42-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="status-ico" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Status ICO</h2>
<div id="analysis-17" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15069 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 1, 0.0066%
FALSE outliers [1]       : 6, 0.04%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/ICO%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-18" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/ICO%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/ICO%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in ICO according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
15/532
</td>
<td style="text-align:left;">
81/12198
</td>
<td style="text-align:right;">
0.0000018
</td>
<td style="text-align:right;">
0.0025620
</td>
<td style="text-align:right;">
0.0025620
</td>
<td style="text-align:left;">
LOC114002081/RPS3/RPL22/RPL23A/RPL27/RPL37/RPL38/RPL4/RPL6/RPL7A/RPS14/RPS29/RPS4X/RPS6/UBA52
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019083" class="uri">GO:0019083</a>
</td>
<td style="text-align:left;">
viral transcription
</td>
<td style="text-align:left;">
16/532
</td>
<td style="text-align:left;">
100/12198
</td>
<td style="text-align:right;">
0.0000062
</td>
<td style="text-align:right;">
0.0043095
</td>
<td style="text-align:right;">
0.0043095
</td>
<td style="text-align:left;">
LOC114002081/RPS3/NUP160/RPL22/RPL23A/RPL27/RPL37/RPL38/RPL4/RPL6/RPL7A/RPS14/RPS29/RPS4X/RPS6/UBA52
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000184" class="uri">GO:0000184</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
</td>
<td style="text-align:left;">
15/532
</td>
<td style="text-align:left;">
111/12198
</td>
<td style="text-align:right;">
0.0000915
</td>
<td style="text-align:right;">
0.0426901
</td>
<td style="text-align:right;">
0.0426901
</td>
<td style="text-align:left;">
LOC114002081/RPS3/RPL22/RPL23A/RPL27/RPL37/RPL38/RPL4/RPL6/RPL7A/RPS14/RPS29/RPS4X/RPS6/UBA52
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006413" class="uri">GO:0006413</a>
</td>
<td style="text-align:left;">
translational initiation
</td>
<td style="text-align:left;">
15/532
</td>
<td style="text-align:left;">
118/12198
</td>
<td style="text-align:right;">
0.0001840
</td>
<td style="text-align:right;">
0.0643631
</td>
<td style="text-align:right;">
0.0643631
</td>
<td style="text-align:left;">
LOC114002081/RPS3/RPL22/RPL23A/RPL27/RPL37/RPL38/RPL4/RPL6/RPL7A/RPS14/RPS29/RPS4X/RPS6/UBA52
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045540" class="uri">GO:0045540</a>
</td>
<td style="text-align:left;">
regulation of cholesterol biosynthetic process
</td>
<td style="text-align:left;">
7/532
</td>
<td style="text-align:left;">
34/12198
</td>
<td style="text-align:right;">
0.0005556
</td>
<td style="text-align:right;">
0.1554559
</td>
<td style="text-align:right;">
0.1554559
</td>
<td style="text-align:left;">
DHCR7/FDFT1/GGPS1/LPCAT3/LSS/SC5D/SREBF1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015949" class="uri">GO:0015949</a>
</td>
<td style="text-align:left;">
nucleobase-containing small molecule interconversion
</td>
<td style="text-align:left;">
6/532
</td>
<td style="text-align:left;">
26/12198
</td>
<td style="text-align:right;">
0.0007299
</td>
<td style="text-align:right;">
0.1701948
</td>
<td style="text-align:right;">
0.1701948
</td>
<td style="text-align:left;">
AK2/AK9/CMPK1/GSR/LOC113996758/RRM1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:left;">
15/532
</td>
<td style="text-align:left;">
148/12198
</td>
<td style="text-align:right;">
0.0020197
</td>
<td style="text-align:right;">
0.3846192
</td>
<td style="text-align:right;">
0.3846192
</td>
<td style="text-align:left;">
LOC114002081/RPS3/RPL22/RPL23A/RPL27/RPL37/RPL38/RPL4/RPL6/RPL7A/RPS14/RPS29/RPS4X/RPS6/TNIP1
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006936" class="uri">GO:0006936</a>
</td>
<td style="text-align:left;">
muscle contraction
</td>
<td style="text-align:left;">
10/532
</td>
<td style="text-align:left;">
79/12198
</td>
<td style="text-align:right;">
0.0021994
</td>
<td style="text-align:right;">
0.3846192
</td>
<td style="text-align:right;">
0.3846192
</td>
<td style="text-align:left;">
GAMT/TNNI1/CACNA1S/CLCN1/ITGA1/LOC114001935/MYL6/MYL9/TPM2/TTN
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007076" class="uri">GO:0007076</a>
</td>
<td style="text-align:left;">
mitotic chromosome condensation
</td>
<td style="text-align:left;">
4/532
</td>
<td style="text-align:left;">
15/12198
</td>
<td style="text-align:right;">
0.0033248
</td>
<td style="text-align:right;">
0.4651398
</td>
<td style="text-align:right;">
0.4651398
</td>
<td style="text-align:left;">
AKAP8L/LOC113992565/SMC4/TTN
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
532
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070542" class="uri">GO:0070542</a>
</td>
<td style="text-align:left;">
response to fatty acid
</td>
<td style="text-align:left;">
4/532
</td>
<td style="text-align:left;">
15/12198
</td>
<td style="text-align:right;">
0.0033248
</td>
<td style="text-align:right;">
0.4651398
</td>
<td style="text-align:right;">
0.4651398
</td>
<td style="text-align:left;">
CAT/FOXO1/FOXO3/LOC113996085
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
532
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-ico" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Testosterone ICO</h2>
<div id="expression-data-exploration-10" class="section level3"
number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-7" class="section level3" number="7.3.2">
<h3><span class="header-section-number">7.3.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-44-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Still need to control for Year here…</p>
<p><br></p>
</div>
<div id="analysis-18" class="section level3" number="7.3.3">
<h3><span class="header-section-number">7.3.3</span> Analysis</h3>
<p>Formula = ~ Batch + Year + mean_T</p>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 121, 0.81%
FALSE LFC &lt; 0 (down)     : 40, 0.27%
FALSE outliers [1]       : 16, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/ICO%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-19" class="section level3" number="7.3.4">
<h3><span class="header-section-number">7.3.4</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/ICO%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/ICO%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in ICO according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
17/473
</td>
<td style="text-align:left;">
38/12155
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
LOC113981955/DNAI1/DNAI2/DNAH9/LOC113998190/CFAP70/DNAH1/LOC114000425/ARMC4/LOC114000433/LRRC6/DNAH12/CFAP221/LOC114001265/CCDC40/TTC25/DNAH5
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060285" class="uri">GO:0060285</a>
</td>
<td style="text-align:left;">
cilium-dependent cell motility
</td>
<td style="text-align:left;">
8/473
</td>
<td style="text-align:left;">
14/12155
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
0.0000079
</td>
<td style="text-align:right;">
0.0000077
</td>
<td style="text-align:left;">
LOC113984044/DNAH1/DNAH3/CFAP44/DNAH12/LOC113982904/EFHC1/EFHC2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030317" class="uri">GO:0030317</a>
</td>
<td style="text-align:left;">
flagellated sperm motility
</td>
<td style="text-align:left;">
10/473
</td>
<td style="text-align:left;">
35/12155
</td>
<td style="text-align:right;">
6.00e-07
</td>
<td style="text-align:right;">
0.0002391
</td>
<td style="text-align:right;">
0.0002348
</td>
<td style="text-align:left;">
CFAP65/DNAI1/LOC113998190/DNAH1/LRRC6/ROPN1L/TCTE1/CCDC40/DNAH5/ENKUR
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007368" class="uri">GO:0007368</a>
</td>
<td style="text-align:left;">
determination of left/right symmetry
</td>
<td style="text-align:left;">
11/473
</td>
<td style="text-align:left;">
50/12155
</td>
<td style="text-align:right;">
2.60e-06
</td>
<td style="text-align:right;">
0.0007342
</td>
<td style="text-align:right;">
0.0007210
</td>
<td style="text-align:left;">
ZIC3/DNAI1/DNAI2/LOC113998190/LOC113984044/LOC113992699/FOXJ1/ARMC4/BBS7/LOC113982904/DNAH5
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036158" class="uri">GO:0036158</a>
</td>
<td style="text-align:left;">
outer dynein arm assembly
</td>
<td style="text-align:left;">
7/473
</td>
<td style="text-align:left;">
18/12155
</td>
<td style="text-align:right;">
2.80e-06
</td>
<td style="text-align:right;">
0.0007342
</td>
<td style="text-align:right;">
0.0007210
</td>
<td style="text-align:left;">
DNAI1/DNAI2/LOC113998190/ARMC4/LRRC6/TTC25/DNAH5
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035082" class="uri">GO:0035082</a>
</td>
<td style="text-align:left;">
axoneme assembly
</td>
<td style="text-align:left;">
7/473
</td>
<td style="text-align:left;">
21/12155
</td>
<td style="text-align:right;">
9.30e-06
</td>
<td style="text-align:right;">
0.0019436
</td>
<td style="text-align:right;">
0.0019086
</td>
<td style="text-align:left;">
CFAP74/JHY/LOC113992699/FOXJ1/CFAP46/CCDC40/RSPH1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007018" class="uri">GO:0007018</a>
</td>
<td style="text-align:left;">
microtubule-based movement
</td>
<td style="text-align:left;">
12/473
</td>
<td style="text-align:left;">
68/12155
</td>
<td style="text-align:right;">
1.05e-05
</td>
<td style="text-align:right;">
0.0019436
</td>
<td style="text-align:right;">
0.0019086
</td>
<td style="text-align:left;">
DNAI1/DNAI2/DNAH9/LOC113998190/DNAH1/DNAH3/KIF24/LOC114000425/LOC114000985/DNAH12/DNAH10/DNAH5
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007601" class="uri">GO:0007601</a>
</td>
<td style="text-align:left;">
visual perception
</td>
<td style="text-align:left;">
18/473
</td>
<td style="text-align:left;">
148/12155
</td>
<td style="text-align:right;">
1.74e-05
</td>
<td style="text-align:right;">
0.0028142
</td>
<td style="text-align:right;">
0.0027635
</td>
<td style="text-align:left;">
GUCA1B/GUCA1C/NDP/ZIC2/COL1A1/LAMB2/LOC113984118/PDE6G/SOX14/USH2A/MYO3A/BBS7/MFRP/RORB/HSF4/NR2E1/GJC1/LOC113999235
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0018149" class="uri">GO:0018149</a>
</td>
<td style="text-align:left;">
peptide cross-linking
</td>
<td style="text-align:left;">
5/473
</td>
<td style="text-align:left;">
12/12155
</td>
<td style="text-align:right;">
5.51e-05
</td>
<td style="text-align:right;">
0.0079406
</td>
<td style="text-align:right;">
0.0077974
</td>
<td style="text-align:left;">
F13A1/TGM1/LOC113988333/TGM2/LOC113992706
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003351" class="uri">GO:0003351</a>
</td>
<td style="text-align:left;">
epithelial cilium movement involved in extracellular fluid movement
</td>
<td style="text-align:left;">
5/473
</td>
<td style="text-align:left;">
13/12155
</td>
<td style="text-align:right;">
8.67e-05
</td>
<td style="text-align:right;">
0.0112426
</td>
<td style="text-align:right;">
0.0110399
</td>
<td style="text-align:left;">
LOC113998190/SPAG17/ROPN1L/CCDC40/DNAH5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
473
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-2" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 1194, 8%
FALSE LFC &lt; 0 (down)     : 1086, 7.3%
FALSE outliers [1]       : 15, 0.1%
FALSE low counts [2]     : 1446, 9.7%
FALSE (mean count &lt; 18)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-47-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-47-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-48-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-ico" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> Strength ICO</h2>
<div id="analysis-19" class="section level3" number="7.5.1">
<h3><span class="header-section-number">7.5.1</span> Analysis</h3>
<p>Formula ~ Batch + Year + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14968 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 39, 0.26%
FALSE LFC &lt; 0 (down)     : 29, 0.19%
FALSE outliers [1]       : 11, 0.073%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/ICO%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/ICO%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-20" class="section level3" number="7.5.2">
<h3><span class="header-section-number">7.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/ICO%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/ICO%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in ICO according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007605" class="uri">GO:0007605</a>
</td>
<td style="text-align:left;">
sensory perception of sound
</td>
<td style="text-align:left;">
15/306
</td>
<td style="text-align:left;">
127/12155
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.0007127
</td>
<td style="text-align:right;">
0.0006741
</td>
<td style="text-align:left;">
LOC113987015/LOC113993758/LOC113991140/BARHL1/ALDH7A1/COL4A3/GABRA5/ATP2B2/EPS8L2/USP53/LOC114000755/COL1A1/LRP2/LOC113993146/COL2A1
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
7/306
</td>
<td style="text-align:left;">
38/12155
</td>
<td style="text-align:right;">
0.0000386
</td>
<td style="text-align:right;">
0.0201160
</td>
<td style="text-align:right;">
0.0190272
</td>
<td style="text-align:left;">
DNAI2/LOC114000425/DNAI1/LOC113981955/CFAP221/DNAH9/CCDC40
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070268" class="uri">GO:0070268</a>
</td>
<td style="text-align:left;">
cornification
</td>
<td style="text-align:left;">
6/306
</td>
<td style="text-align:left;">
28/12155
</td>
<td style="text-align:right;">
0.0000572
</td>
<td style="text-align:right;">
0.0201160
</td>
<td style="text-align:right;">
0.0190272
</td>
<td style="text-align:left;">
LOC114001273/DSG2/DSP/LOC114000789/LOC114002150/LOC113991144
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007601" class="uri">GO:0007601</a>
</td>
<td style="text-align:left;">
visual perception
</td>
<td style="text-align:left;">
13/306
</td>
<td style="text-align:left;">
148/12155
</td>
<td style="text-align:right;">
0.0000919
</td>
<td style="text-align:right;">
0.0234574
</td>
<td style="text-align:right;">
0.0221877
</td>
<td style="text-align:left;">
GUCA1B/GRM8/RPE65/GJC1/COL18A1/GLRA1/CNGB1/COL1A1/COL2A1/BBS4/LAMB2/LOC113999235/PAX6
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016082" class="uri">GO:0016082</a>
</td>
<td style="text-align:left;">
synaptic vesicle priming
</td>
<td style="text-align:left;">
5/306
</td>
<td style="text-align:left;">
20/12155
</td>
<td style="text-align:right;">
0.0001112
</td>
<td style="text-align:right;">
0.0234574
</td>
<td style="text-align:right;">
0.0221877
</td>
<td style="text-align:left;">
LOC113987015/LOC113993758/LOC113991140/LOC113993146/UNC13C
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031424" class="uri">GO:0031424</a>
</td>
<td style="text-align:left;">
keratinization
</td>
<td style="text-align:left;">
5/306
</td>
<td style="text-align:left;">
22/12155
</td>
<td style="text-align:right;">
0.0001812
</td>
<td style="text-align:right;">
0.0318573
</td>
<td style="text-align:right;">
0.0301329
</td>
<td style="text-align:left;">
LOC114001273/DSG2/DSP/LOC114000789/LOC114002150
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
15/306
</td>
<td style="text-align:left;">
208/12155
</td>
<td style="text-align:right;">
0.0002497
</td>
<td style="text-align:right;">
0.0376358
</td>
<td style="text-align:right;">
0.0355986
</td>
<td style="text-align:left;">
COL4A4/COL4A3/LOC113993281/ADAMTS7/COL8A2/COL7A1/COL18A1/VCAN/COL1A1/COL8A1/COL2A1/COL4A6/COL6A2/LAMB2/LAMA3
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043651" class="uri">GO:0043651</a>
</td>
<td style="text-align:left;">
linoleic acid metabolic process
</td>
<td style="text-align:left;">
4/306
</td>
<td style="text-align:left;">
16/12155
</td>
<td style="text-align:right;">
0.0005642
</td>
<td style="text-align:right;">
0.0700335
</td>
<td style="text-align:right;">
0.0662427
</td>
<td style="text-align:left;">
LOC113991921/LOC114001775/LOC114002601/LOC114002603
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035987" class="uri">GO:0035987</a>
</td>
<td style="text-align:left;">
endodermal cell differentiation
</td>
<td style="text-align:left;">
5/306
</td>
<td style="text-align:left;">
28/12155
</td>
<td style="text-align:right;">
0.0005974
</td>
<td style="text-align:right;">
0.0700335
</td>
<td style="text-align:right;">
0.0662427
</td>
<td style="text-align:left;">
LOC113983959/COL7A1/COL8A1/COL4A6/LAMA3
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
306
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007155" class="uri">GO:0007155</a>
</td>
<td style="text-align:left;">
cell adhesion
</td>
<td style="text-align:left;">
19/306
</td>
<td style="text-align:left;">
331/12155
</td>
<td style="text-align:right;">
0.0007230
</td>
<td style="text-align:right;">
0.0762806
</td>
<td style="text-align:right;">
0.0721516
</td>
<td style="text-align:left;">
LOC113983959/FERMT2/SORBS3/CNTNAP2/COL4A3/CD9/CDHR4/LOC113988059/LOC113995955/LOC113993281/DSG2/PGM5/COL7A1/COL18A1/VCAN/COL8A1/COL6A2/STAB1/CX3CL1
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
306
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="midbrain-central-grey-gct" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Midbrain Central Grey
(GCt)</h1>
<p><br></p>
<div id="checking-the-sampling-7" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Checking the
sampling</h2>
<p>For the GCT, we started with 16854 genes but after filering we have
15099 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have gct samples for 12, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.76
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.18
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p><img src="03_DESeq2_files/figure-html/GCT%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>given there is only 1 GCT sample from the pilot, I will just exclude
the pilot data.</p>
<p><br></p>
<div id="expression-data-exploration-11" class="section level3"
number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/GCT%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-8" class="section level3" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-51-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-51-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="status-gct" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Status GCT</h2>
<div id="analysis-20" class="section level3" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15099 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 17, 0.11%
FALSE LFC &lt; 0 (down)     : 12, 0.079%
FALSE outliers [1]       : 14, 0.093%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/GCT%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" />
<br></p>
</div>
<div id="gene-ontology-21" class="section level3" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/GCT%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/GCT%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GCT according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0042738" class="uri">GO:0042738</a>
</td>
<td style="text-align:left;">
exogenous drug catabolic process
</td>
<td style="text-align:left;">
4/294
</td>
<td style="text-align:left;">
10/12236
</td>
<td style="text-align:right;">
0.0000612
</td>
<td style="text-align:right;">
0.0428880
</td>
<td style="text-align:right;">
0.0407714
</td>
<td style="text-align:left;">
LOC113983773/NOS1/LOC114002604/LOC113988755
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042572" class="uri">GO:0042572</a>
</td>
<td style="text-align:left;">
retinol metabolic process
</td>
<td style="text-align:left;">
6/294
</td>
<td style="text-align:left;">
31/12236
</td>
<td style="text-align:right;">
0.0000811
</td>
<td style="text-align:right;">
0.0428880
</td>
<td style="text-align:right;">
0.0407714
</td>
<td style="text-align:left;">
LOC114001898/LOC113983773/LOC113988755/LOC113991249/BCO1/DGAT2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007160" class="uri">GO:0007160</a>
</td>
<td style="text-align:left;">
cell-matrix adhesion
</td>
<td style="text-align:left;">
9/294
</td>
<td style="text-align:left;">
79/12236
</td>
<td style="text-align:right;">
0.0001113
</td>
<td style="text-align:right;">
0.0428880
</td>
<td style="text-align:right;">
0.0407714
</td>
<td style="text-align:left;">
NID1/COL3A1/FN1/ADAMTS12/ITGB5/NPNT/FGG/ITGA2/NID2
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071356" class="uri">GO:0071356</a>
</td>
<td style="text-align:left;">
cellular response to tumor necrosis factor
</td>
<td style="text-align:left;">
8/294
</td>
<td style="text-align:left;">
82/12236
</td>
<td style="text-align:right;">
0.0007711
</td>
<td style="text-align:right;">
0.2115720
</td>
<td style="text-align:right;">
0.2011302
</td>
<td style="text-align:left;">
CLDN1/LOC114000824/CRHBP/KLF2/ADAMTS12/NPNT/EDN1/LOC114003556
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030522" class="uri">GO:0030522</a>
</td>
<td style="text-align:left;">
intracellular receptor signaling pathway
</td>
<td style="text-align:left;">
5/294
</td>
<td style="text-align:left;">
32/12236
</td>
<td style="text-align:right;">
0.0009151
</td>
<td style="text-align:right;">
0.2115720
</td>
<td style="text-align:right;">
0.2011302
</td>
<td style="text-align:left;">
LOC113996216/LOC113995289/SREBF1/LOC113982931/NR2E1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
13/294
</td>
<td style="text-align:left;">
213/12236
</td>
<td style="text-align:right;">
0.0019262
</td>
<td style="text-align:right;">
0.3205493
</td>
<td style="text-align:right;">
0.3047290
</td>
<td style="text-align:left;">
NID1/VIT/COL3A1/LAMB2/LOC113999927/FN1/ADAMTS12/ITGB5/NPNT/FGG/ITGA2/NID2/NR2E1
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006805" class="uri">GO:0006805</a>
</td>
<td style="text-align:left;">
xenobiotic metabolic process
</td>
<td style="text-align:left;">
6/294
</td>
<td style="text-align:left;">
57/12236
</td>
<td style="text-align:right;">
0.0023716
</td>
<td style="text-align:right;">
0.3205493
</td>
<td style="text-align:right;">
0.3047290
</td>
<td style="text-align:left;">
LOC113983773/LOC114002604/LOC113996216/LOC113995289/LOC113988755/LOC113991029
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006629" class="uri">GO:0006629</a>
</td>
<td style="text-align:left;">
lipid metabolic process
</td>
<td style="text-align:left;">
8/294
</td>
<td style="text-align:left;">
98/12236
</td>
<td style="text-align:right;">
0.0024487
</td>
<td style="text-align:right;">
0.3205493
</td>
<td style="text-align:right;">
0.3047290
</td>
<td style="text-align:left;">
LCAT/SC5D/LOC113983773/LOC113998978/SREBF1/LOC113991249/CPNE7/DGAT2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0018149" class="uri">GO:0018149</a>
</td>
<td style="text-align:left;">
peptide cross-linking
</td>
<td style="text-align:left;">
3/294
</td>
<td style="text-align:left;">
12/12236
</td>
<td style="text-align:right;">
0.0025714
</td>
<td style="text-align:right;">
0.3205493
</td>
<td style="text-align:right;">
0.3047290
</td>
<td style="text-align:left;">
COL3A1/FN1/F13A1
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043085" class="uri">GO:0043085</a>
</td>
<td style="text-align:left;">
positive regulation of catalytic activity
</td>
<td style="text-align:left;">
4/294
</td>
<td style="text-align:left;">
25/12236
</td>
<td style="text-align:right;">
0.0027729
</td>
<td style="text-align:right;">
0.3205493
</td>
<td style="text-align:right;">
0.3047290
</td>
<td style="text-align:left;">
LOC114000824/FGFR4/SLC5A3/DCP1B
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
294
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-gct" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Testosterone GCT</h2>
<div id="expression-data-exploration-12" class="section level3"
number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-3.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/GCT%20Continuous%20data%20setup-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><br></p>
</div>
<div id="analysis-21" class="section level3" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Analysis</h3>
<p>Formula = ~ Batch + mean_T</p>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 4, 0.026%
FALSE LFC &lt; 0 (down)     : 13, 0.086%
FALSE outliers [1]       : 33, 0.22%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/GCT%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-22" class="section level3" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/GCT%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/GCT%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GCT according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007596" class="uri">GO:0007596</a>
</td>
<td style="text-align:left;">
blood coagulation
</td>
<td style="text-align:left;">
14/380
</td>
<td style="text-align:left;">
115/12237
</td>
<td style="text-align:right;">
0.0000126
</td>
<td style="text-align:right;">
0.0170537
</td>
<td style="text-align:right;">
0.0160700
</td>
<td style="text-align:left;">
FGG/GATA2/F13A1/ENTPD1/LOC114003293/COL1A1/VWF/P2RX3/LOC113995989/SERPIND1/MICAL1/CDC42/ANO6/P2RX4
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032703" class="uri">GO:0032703</a>
</td>
<td style="text-align:left;">
negative regulation of interleukin-2 production
</td>
<td style="text-align:left;">
6/380
</td>
<td style="text-align:left;">
21/12237
</td>
<td style="text-align:right;">
0.0000315
</td>
<td style="text-align:right;">
0.0213293
</td>
<td style="text-align:right;">
0.0200990
</td>
<td style="text-align:left;">
LOC113986241/CD34/LOC114003293/LOC113993091/LOC113993083/PTPRC
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:left;">
6/380
</td>
<td style="text-align:left;">
23/12237
</td>
<td style="text-align:right;">
0.0000556
</td>
<td style="text-align:right;">
0.0250948
</td>
<td style="text-align:right;">
0.0236473
</td>
<td style="text-align:left;">
LOC113986241/C3/LOC113984271/LOC113984305/LOC113992168/C1R
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045766" class="uri">GO:0045766</a>
</td>
<td style="text-align:left;">
positive regulation of angiogenesis
</td>
<td style="text-align:left;">
12/380
</td>
<td style="text-align:left;">
105/12237
</td>
<td style="text-align:right;">
0.0000977
</td>
<td style="text-align:right;">
0.0330929
</td>
<td style="text-align:right;">
0.0311841
</td>
<td style="text-align:left;">
AQP1/APELA/GATA2/PRKD1/CD34/C3/ITGB2/EPHA1/WNT5A/VEGFC/NR2E1/ISL1
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000178" class="uri">GO:2000178</a>
</td>
<td style="text-align:left;">
negative regulation of neural precursor cell proliferation
</td>
<td style="text-align:left;">
4/380
</td>
<td style="text-align:left;">
10/12237
</td>
<td style="text-align:right;">
0.0001657
</td>
<td style="text-align:right;">
0.0448999
</td>
<td style="text-align:right;">
0.0423101
</td>
<td style="text-align:left;">
BTG2/GATA2/LIMS2/NR2E1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035735" class="uri">GO:0035735</a>
</td>
<td style="text-align:left;">
intraciliary transport involved in cilium assembly
</td>
<td style="text-align:left;">
7/380
</td>
<td style="text-align:left;">
40/12237
</td>
<td style="text-align:right;">
0.0002026
</td>
<td style="text-align:right;">
0.0457568
</td>
<td style="text-align:right;">
0.0431175
</td>
<td style="text-align:left;">
HSPB11/DYNLL1/DYNC2LI1/KIF17/TTC26/IFT74/CLUAP1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006631" class="uri">GO:0006631</a>
</td>
<td style="text-align:left;">
fatty acid metabolic process
</td>
<td style="text-align:left;">
8/380
</td>
<td style="text-align:left;">
55/12237
</td>
<td style="text-align:right;">
0.0002726
</td>
<td style="text-align:right;">
0.0527687
</td>
<td style="text-align:right;">
0.0497250
</td>
<td style="text-align:left;">
GPAM/HACL1/ACSBG2/C3/LOC113995989/TH/AASDH/LYPLA2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007411" class="uri">GO:0007411</a>
</td>
<td style="text-align:left;">
axon guidance
</td>
<td style="text-align:left;">
17/380
</td>
<td style="text-align:left;">
212/12237
</td>
<td style="text-align:right;">
0.0003323
</td>
<td style="text-align:right;">
0.0562798
</td>
<td style="text-align:right;">
0.0530335
</td>
<td style="text-align:left;">
EMB/CXCR4/LOC113999057/PRKCQ/CSF1R/LOC114003293/EPHA1/ARX/MAPK7/LOC113988477/GBX2/WNT5A/SEMA3C/LYPLA2/BSG/NPTN/TNR
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006957" class="uri">GO:0006957</a>
</td>
<td style="text-align:left;">
complement activation, alternative pathway
</td>
<td style="text-align:left;">
4/380
</td>
<td style="text-align:left;">
14/12237
</td>
<td style="text-align:right;">
0.0007154
</td>
<td style="text-align:right;">
0.0866905
</td>
<td style="text-align:right;">
0.0816901
</td>
<td style="text-align:left;">
LOC113986241/C3/LOC113993091/LOC113993083
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050776" class="uri">GO:0050776</a>
</td>
<td style="text-align:left;">
regulation of immune response
</td>
<td style="text-align:left;">
6/380
</td>
<td style="text-align:left;">
36/12237
</td>
<td style="text-align:right;">
0.0007623
</td>
<td style="text-align:right;">
0.0866905
</td>
<td style="text-align:right;">
0.0816901
</td>
<td style="text-align:left;">
CD34/C3/ITGB2/COL1A1/CD99/COL3A1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
380
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-3" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 206, 1.4%
FALSE LFC &lt; 0 (down)     : 189, 1.3%
FALSE outliers [1]       : 23, 0.15%
FALSE low counts [2]     : 583, 3.9%
FALSE (mean count &lt; 9)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-54-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-54-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-55-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-gct" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Strength GCT</h2>
<div id="analysis-22" class="section level3" number="8.5.1">
<h3><span class="header-section-number">8.5.1</span> Analysis</h3>
<p>Formula ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15105 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 34, 0.23%
FALSE LFC &lt; 0 (down)     : 29, 0.19%
FALSE outliers [1]       : 41, 0.27%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/GCT%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/GCT%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-23" class="section level3" number="8.5.2">
<h3><span class="header-section-number">8.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/GCT%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/GCT%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in GCT according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
23/1035
</td>
<td style="text-align:left;">
81/12237
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0002586
</td>
<td style="text-align:right;">
0.0002547
</td>
<td style="text-align:left;">
RPL39/RPL14/RPS24/RPL10A/RPSA/RPL34/RPS7/RPS8/RPS20/RPL31/RPL5/RPS3/LOC113987193/RPS25/RPL12/RPS15A/SSR3/RPL35A/RPS3A/RPL35/RPS5/RPS4X/RPL17
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006413" class="uri">GO:0006413</a>
</td>
<td style="text-align:left;">
translational initiation
</td>
<td style="text-align:left;">
27/1035
</td>
<td style="text-align:left;">
119/12237
</td>
<td style="text-align:right;">
0.0000016
</td>
<td style="text-align:right;">
0.0015486
</td>
<td style="text-align:right;">
0.0015254
</td>
<td style="text-align:left;">
RPL39/RPL14/RPS24/RPL10A/EIF4A2/RPSA/RPL34/PABPC1/RPS7/RPS8/RPS20/RPL31/EIF4E3/RPL5/LOC113985040/RPS3/LOC113987193/RPS25/RPL12/RPS15A/RPL35A/RPS3A/RPL35/LOC114003727/RPS5/RPS4X/RPL17
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:left;">
30/1035
</td>
<td style="text-align:left;">
148/12237
</td>
<td style="text-align:right;">
0.0000052
</td>
<td style="text-align:right;">
0.0033240
</td>
<td style="text-align:right;">
0.0032743
</td>
<td style="text-align:left;">
RPL39/IGF2BP3/RPL14/RPS24/RPL10A/RPSA/PSTK/RPL34/CPEB3/RPS7/RPS8/RPS20/MRPL18/RPL31/RSL24D1/RPL5/RPS3/LOC113987193/RPS25/RPL12/RPS15A/RPL35A/RPS3A/RPL35/LOC114003727/RPS5/MRPL55/RPS4X/GGT7/RPL17
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019083" class="uri">GO:0019083</a>
</td>
<td style="text-align:left;">
viral transcription
</td>
<td style="text-align:left;">
23/1035
</td>
<td style="text-align:left;">
100/12237
</td>
<td style="text-align:right;">
0.0000074
</td>
<td style="text-align:right;">
0.0035429
</td>
<td style="text-align:right;">
0.0034898
</td>
<td style="text-align:left;">
RPL39/RPL14/RPS24/RPL10A/RPSA/RPL34/RPS7/RPS8/RPS20/RPL31/RPL5/RPS3/LOC113987193/RPS25/RPL12/RPS15A/RPL35A/RPS3A/RPL35/RAE1/RPS5/RPS4X/RPL17
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000184" class="uri">GO:0000184</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
</td>
<td style="text-align:left;">
24/1035
</td>
<td style="text-align:left;">
112/12237
</td>
<td style="text-align:right;">
0.0000170
</td>
<td style="text-align:right;">
0.0065104
</td>
<td style="text-align:right;">
0.0064129
</td>
<td style="text-align:left;">
RPL39/RPL14/RPS24/RPL10A/RPSA/RPL34/PABPC1/RPS7/RPS8/RPS20/RPL31/RPL5/RPS3/LOC113987193/RPS25/RPL12/RPS15A/RPL35A/RPS3A/RPL35/LOC114003727/RPS5/RPS4X/RPL17
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070050" class="uri">GO:0070050</a>
</td>
<td style="text-align:left;">
neuron cellular homeostasis
</td>
<td style="text-align:left;">
7/1035
</td>
<td style="text-align:left;">
14/12237
</td>
<td style="text-align:right;">
0.0000612
</td>
<td style="text-align:right;">
0.0195424
</td>
<td style="text-align:right;">
0.0192497
</td>
<td style="text-align:left;">
TAOK1/NELL2/SLC6A2/MAP1A/CX3CL1/PRKAA1/LOC113995995
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
35/1035
</td>
<td style="text-align:left;">
213/12237
</td>
<td style="text-align:right;">
0.0001016
</td>
<td style="text-align:right;">
0.0277843
</td>
<td style="text-align:right;">
0.0273682
</td>
<td style="text-align:left;">
COL5A2/MMP2/ITGA2B/SH3PXD2A/LOC113982345/TNC/DAG1/LOC113991284/THBS1/HAPLN1/VTN/MMP15/ADAMTS3/ITGB2/LOC113998785/ADAMTS14/LUM/CCDC80/COL5A1/ACAN/SMOC2/TGFBI/MMP23B/COL6A3/PRDX4/ADAMTSL5/ELN/LAMA5/LOC113999927/ATXN1L/COL18A1/NF1/PDGFRA/SPARC/COL6A2
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035082" class="uri">GO:0035082</a>
</td>
<td style="text-align:left;">
axoneme assembly
</td>
<td style="text-align:left;">
8/1035
</td>
<td style="text-align:left;">
21/12237
</td>
<td style="text-align:right;">
0.0001908
</td>
<td style="text-align:right;">
0.0456705
</td>
<td style="text-align:right;">
0.0449864
</td>
<td style="text-align:left;">
LOC113992699/LOC113990666/RSPH1/JHY/CFAP206/RSPH9/LOC113986306/LOC113984124
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0018095" class="uri">GO:0018095</a>
</td>
<td style="text-align:left;">
protein polyglutamylation
</td>
<td style="text-align:left;">
5/1035
</td>
<td style="text-align:left;">
10/12237
</td>
<td style="text-align:right;">
0.0007526
</td>
<td style="text-align:right;">
0.1601461
</td>
<td style="text-align:right;">
0.1577473
</td>
<td style="text-align:left;">
TTLL7/TPGS2/TTLL4/CEP41/TTLL11
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1035
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006811" class="uri">GO:0006811</a>
</td>
<td style="text-align:left;">
ion transport
</td>
<td style="text-align:left;">
16/1035
</td>
<td style="text-align:left;">
81/12237
</td>
<td style="text-align:right;">
0.0010739
</td>
<td style="text-align:right;">
0.2056489
</td>
<td style="text-align:right;">
0.2025685
</td>
<td style="text-align:left;">
SLC22A15/STEAP1/SLC24A2/SLC26A1/LOC113988621/TRPA1/SLC12A2/LOC113991506/AQP11/WNK1/SLC26A6/GLRA1/LASP1/SLC1A3/LOC113989700/SLC24A3
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
1035
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="nucleus-taenia-tna" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Nucleus Taenia
(TnA)</h1>
<p><br></p>
<div id="checking-the-sampling-8" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Checking the
sampling</h2>
<p>For the TNA, we started with 16854 genes but after filering we have
14693 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have tna samples for 12, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00075
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="03_DESeq2_files/figure-html/TNA%20sampling-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="expression-data-exploration-13" class="section level3"
number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/TNA%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/TNA%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-9" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-57-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-57-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-57-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>PFT1 was previously identified as an outlier based on WGCNA
dendrograms only. But here it’s not consistently dentified as an
outlier.Also previous analysis shows that it doesn’t have a strong
influence on the results, so I am going to keep it in this data. Looks
like we only need to control for Batch, as it removes the effects of
year on our data.</p>
<p><br></p>
</div>
</div>
<div id="status-tna" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Status TNA</h2>
<div id="analysis-23" class="section level3" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14693 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 5, 0.034%
FALSE LFC &lt; 0 (down)     : 8, 0.054%
FALSE outliers [1]       : 24, 0.16%
FALSE low counts [2]     : 1415, 9.6%
FALSE (mean count &lt; 17)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/TNA%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-24" class="section level3" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/TNA%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/TNA%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in TNA according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
16/963
</td>
<td style="text-align:left;">
70/11975
</td>
<td style="text-align:right;">
0.0001062
</td>
<td style="text-align:right;">
0.1979001
</td>
<td style="text-align:right;">
0.1978107
</td>
<td style="text-align:left;">
LOC113991485/ITIH5/ANXA2/LOC113992817/LOC113991484/SPINT1/APP/LOC113992168/CRIM1/SERPINB1/USP14/TIMP4/A2M/SERPINE1/LOC113991482/LOC113991487
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060291" class="uri">GO:0060291</a>
</td>
<td style="text-align:left;">
long-term synaptic potentiation
</td>
<td style="text-align:left;">
10/963
</td>
<td style="text-align:left;">
36/11975
</td>
<td style="text-align:right;">
0.0003965
</td>
<td style="text-align:right;">
0.2559777
</td>
<td style="text-align:right;">
0.2558621
</td>
<td style="text-align:left;">
SYT12/CRHR2/NLGN1/ARC/SLC24A2/PLK2/PTN/CRH/ITPR3/LRRTM2
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051965" class="uri">GO:0051965</a>
</td>
<td style="text-align:left;">
positive regulation of synapse assembly
</td>
<td style="text-align:left;">
12/963
</td>
<td style="text-align:left;">
50/11975
</td>
<td style="text-align:right;">
0.0004723
</td>
<td style="text-align:right;">
0.2559777
</td>
<td style="text-align:right;">
0.2558621
</td>
<td style="text-align:left;">
EPHB2/CUX2/AMIGO3/NLGN1/PTPRD/LRRN3/SLITRK6/EEF2K/SLITRK2/IL1RAP/LRRTM3/LRRTM2
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
16/963
</td>
<td style="text-align:left;">
81/11975
</td>
<td style="text-align:right;">
0.0006227
</td>
<td style="text-align:right;">
0.2559777
</td>
<td style="text-align:right;">
0.2558621
</td>
<td style="text-align:left;">
LOC114002081/LOC113998898/RPL4/RPL31/RPL19/RPS27A/RPS23/RPL37/RPS21/RPL7/RPS29/RPL30/RPS5/RPL23A/SRP19/RPS3A
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006936" class="uri">GO:0006936</a>
</td>
<td style="text-align:left;">
muscle contraction
</td>
<td style="text-align:left;">
15/963
</td>
<td style="text-align:left;">
74/11975
</td>
<td style="text-align:right;">
0.0006866
</td>
<td style="text-align:right;">
0.2559777
</td>
<td style="text-align:right;">
0.2558621
</td>
<td style="text-align:left;">
ITGA1/CALD1/GAMT/VCL/TPM2/CACNA1H/SORBS3/ITGA11/LOC113993584/LOC113999652/PXN/CALM1/LOC113999542/LOC113999078/LOC114001935
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007093" class="uri">GO:0007093</a>
</td>
<td style="text-align:left;">
mitotic cell cycle checkpoint
</td>
<td style="text-align:left;">
5/963
</td>
<td style="text-align:left;">
12/11975
</td>
<td style="text-align:right;">
0.0016323
</td>
<td style="text-align:right;">
0.3612802
</td>
<td style="text-align:right;">
0.3611170
</td>
<td style="text-align:left;">
NABP2/CHFR/PLK2/PLK3/NABP1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090050" class="uri">GO:0090050</a>
</td>
<td style="text-align:left;">
positive regulation of cell migration involved in sprouting angiogenesis
</td>
<td style="text-align:left;">
7/963
</td>
<td style="text-align:left;">
23/11975
</td>
<td style="text-align:right;">
0.0016572
</td>
<td style="text-align:right;">
0.3612802
</td>
<td style="text-align:right;">
0.3611170
</td>
<td style="text-align:left;">
RHOJ/PLK2/VEGFA/PTGS2/KDR/LOC113997171/LOC114000824
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035633" class="uri">GO:0035633</a>
</td>
<td style="text-align:left;">
maintenance of blood-brain barrier
</td>
<td style="text-align:left;">
8/963
</td>
<td style="text-align:left;">
30/11975
</td>
<td style="text-align:right;">
0.0020087
</td>
<td style="text-align:right;">
0.3612802
</td>
<td style="text-align:right;">
0.3611170
</td>
<td style="text-align:left;">
TJP3/ITGB1/LOC114001565/CDH5/VCL/CLDN1/TJP1/SLC12A2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042127" class="uri">GO:0042127</a>
</td>
<td style="text-align:left;">
regulation of cell population proliferation
</td>
<td style="text-align:left;">
18/963
</td>
<td style="text-align:left;">
107/11975
</td>
<td style="text-align:right;">
0.0020783
</td>
<td style="text-align:right;">
0.3612802
</td>
<td style="text-align:right;">
0.3611170
</td>
<td style="text-align:left;">
LOC114000703/SIX3/TOPORS/LTK/LOC114000971/MEAK7/XIAP/DUSP15/MAFG/JUN/IL4R/TGFB3/LOC114001747/CCDC88A/SHH/SLC7A11/AGTR1/LOC114000824
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
963
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090141" class="uri">GO:0090141</a>
</td>
<td style="text-align:left;">
positive regulation of mitochondrial fission
</td>
<td style="text-align:left;">
6/963
</td>
<td style="text-align:left;">
18/11975
</td>
<td style="text-align:right;">
0.0021270
</td>
<td style="text-align:right;">
0.3612802
</td>
<td style="text-align:right;">
0.3611170
</td>
<td style="text-align:left;">
MIEF2/BNIP3/MIEF1/KDR/LOC113997171/SPIRE1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
963
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-tna" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Testosterone TNA</h2>
<div id="expression-data-exploration-14" class="section level3"
number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/TNA%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-10" class="section level3" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-59-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-59-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Taking the batch effect into account seems to remove the association
with Year too. Only need to include Batch in the model.</p>
</div>
<div id="analysis-24" class="section level3" number="9.3.3">
<h3><span class="header-section-number">9.3.3</span> Analysis</h3>
<p>Formula = ~ Batch + mean_T</p>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 99, 0.68%
FALSE LFC &lt; 0 (down)     : 143, 0.98%
FALSE outliers [1]       : 37, 0.25%
FALSE low counts [2]     : 280, 1.9%
FALSE (mean count &lt; 6)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/TNA%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-25" class="section level3" number="9.3.4">
<h3><span class="header-section-number">9.3.4</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/TNA%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/TNA%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in TNA according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0060291" class="uri">GO:0060291</a>
</td>
<td style="text-align:left;">
long-term synaptic potentiation
</td>
<td style="text-align:left;">
13/1251
</td>
<td style="text-align:left;">
36/11931
</td>
<td style="text-align:right;">
0.0000398
</td>
<td style="text-align:right;">
0.0798071
</td>
<td style="text-align:right;">
0.0794405
</td>
<td style="text-align:left;">
SLC24A2/CRHR2/PTN/SYT12/PLK2/NLGN1/NR2E1/RASGRF2/LOC114003174/RGS14/SLC1A1/LOC113985801/MECP2
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
18/1251
</td>
<td style="text-align:left;">
69/11931
</td>
<td style="text-align:right;">
0.0001951
</td>
<td style="text-align:right;">
0.1574843
</td>
<td style="text-align:right;">
0.1567608
</td>
<td style="text-align:left;">
LOC113992168/ANXA2/SPOCK1/ITIH2/SPOCK3/TFPI2/PROS1/TIMP4/SPINT1/SERPIND1/LOC113999601/A2M/LOC113991484/LOC113992708/SERPINB5/APLP2/SERPINF1/APP
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045446" class="uri">GO:0045446</a>
</td>
<td style="text-align:left;">
endothelial cell differentiation
</td>
<td style="text-align:left;">
6/1251
</td>
<td style="text-align:left;">
11/11931
</td>
<td style="text-align:right;">
0.0003808
</td>
<td style="text-align:right;">
0.1574843
</td>
<td style="text-align:right;">
0.1567608
</td>
<td style="text-align:left;">
PTN/JAG2/CXCR4/S1PR1/LOC113997171/SCUBE1
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001755" class="uri">GO:0001755</a>
</td>
<td style="text-align:left;">
neural crest cell migration
</td>
<td style="text-align:left;">
13/1251
</td>
<td style="text-align:left;">
44/11931
</td>
<td style="text-align:right;">
0.0004024
</td>
<td style="text-align:right;">
0.1574843
</td>
<td style="text-align:right;">
0.1567608
</td>
<td style="text-align:left;">
RADIL/LOC113982696/LOC113984504/SEMA4G/RET/KITLG/SEMA4C/SEMA3C/LOC114000376/SEMA3A/SEMA3B/SEMA4B/SEMA4A
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048843" class="uri">GO:0048843</a>
</td>
<td style="text-align:left;">
negative regulation of axon extension involved in axon guidance
</td>
<td style="text-align:left;">
9/1251
</td>
<td style="text-align:left;">
24/11931
</td>
<td style="text-align:right;">
0.0004508
</td>
<td style="text-align:right;">
0.1574843
</td>
<td style="text-align:right;">
0.1567608
</td>
<td style="text-align:left;">
LOC113982696/LOC113984504/SEMA4G/SEMA4C/SEMA3C/SEMA3A/SEMA3B/SEMA4B/SEMA4A
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071526" class="uri">GO:0071526</a>
</td>
<td style="text-align:left;">
semaphorin-plexin signaling pathway
</td>
<td style="text-align:left;">
11/1251
</td>
<td style="text-align:left;">
34/11931
</td>
<td style="text-align:right;">
0.0004713
</td>
<td style="text-align:right;">
0.1574843
</td>
<td style="text-align:right;">
0.1567608
</td>
<td style="text-align:left;">
LOC113982696/LOC113984504/SEMA4G/SEMA4C/MET/SEMA3C/SEMA3A/SEMA3B/SEMA4B/LOC113986256/SEMA4A
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050919" class="uri">GO:0050919</a>
</td>
<td style="text-align:left;">
negative chemotaxis
</td>
<td style="text-align:left;">
11/1251
</td>
<td style="text-align:left;">
38/11931
</td>
<td style="text-align:right;">
0.0013385
</td>
<td style="text-align:right;">
0.2642404
</td>
<td style="text-align:right;">
0.2630265
</td>
<td style="text-align:left;">
LOC113982696/LOC113984504/SEMA4G/SEMA4C/RHOA/SEMA3C/SEMA3A/SEMA3B/SEMA4B/FLRT3/SEMA4A
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007613" class="uri">GO:0007613</a>
</td>
<td style="text-align:left;">
memory
</td>
<td style="text-align:left;">
17/1251
</td>
<td style="text-align:left;">
74/11931
</td>
<td style="text-align:right;">
0.0013965
</td>
<td style="text-align:right;">
0.2642404
</td>
<td style="text-align:right;">
0.2630265
</td>
<td style="text-align:left;">
SLC24A2/CEBPB/PPP1R1B/PTN/LGMN/FGF13/LOC113991532/PLK2/KALRN/CCK/ASIC3/KCNK2/LOC113989303/LOC114003174/ADCY8/LOC113985801/PAK5
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1990573" class="uri">GO:1990573</a>
</td>
<td style="text-align:left;">
potassium ion import across plasma membrane
</td>
<td style="text-align:left;">
10/1251
</td>
<td style="text-align:left;">
33/11931
</td>
<td style="text-align:right;">
0.0014975
</td>
<td style="text-align:right;">
0.2642404
</td>
<td style="text-align:right;">
0.2630265
</td>
<td style="text-align:left;">
KCNJ6/SLC12A2/LOC113999542/FXYD2/LOC113996443/ATP1B1/LOC113986621/KCNJ3/ATP1B3/ATP1A1
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1251
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070374" class="uri">GO:0070374</a>
</td>
<td style="text-align:left;">
positive regulation of ERK1 and ERK2 cascade
</td>
<td style="text-align:left;">
26/1251
</td>
<td style="text-align:left;">
136/11931
</td>
<td style="text-align:right;">
0.0017453
</td>
<td style="text-align:right;">
0.2642404
</td>
<td style="text-align:right;">
0.2630265
</td>
<td style="text-align:left;">
CALCR/FGFR4/CFLAR/LOC113991532/FGFR3/RAP1B/GPER1/CX3CL1/PDGFC/LOC113997171/MAP2K1/APELA/PRKCA/SRC/HTR2C/FGFR2/BMP4/ANGPT4/ALKAL2/P2RY1/RIPK2/BMP2/DDT/PTPRC/SLC30A10/APP
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
1251
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-4" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 29, 0.2%
FALSE LFC &lt; 0 (down)     : 13, 0.089%
FALSE outliers [1]       : 22, 0.15%
FALSE low counts [2]     : 3947, 27%
FALSE (mean count &lt; 94)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-62-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/unnamed-chunk-62-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-63-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-tna" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> Strength TNA</h2>
<div id="analysis-25" class="section level3" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Analysis</h3>
<p>Formula ~ Batch + scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14614 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 22, 0.15%
FALSE LFC &lt; 0 (down)     : 34, 0.23%
FALSE outliers [1]       : 38, 0.26%
FALSE low counts [2]     : 839, 5.7%
FALSE (mean count &lt; 11)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/TNA%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/TNA%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-26" class="section level3" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/TNA%20strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/TNA%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in TNA according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0050771" class="uri">GO:0050771</a>
</td>
<td style="text-align:left;">
negative regulation of axonogenesis
</td>
<td style="text-align:left;">
7/911
</td>
<td style="text-align:left;">
20/11931
</td>
<td style="text-align:right;">
0.0004733
</td>
<td style="text-align:right;">
0.4902006
</td>
<td style="text-align:right;">
0.4902006
</td>
<td style="text-align:left;">
LOC113988556/DAB1/LOC113990764/LRP4/RUFY3/LINGO1/RTN4
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043951" class="uri">GO:0043951</a>
</td>
<td style="text-align:left;">
negative regulation of cAMP-mediated signaling
</td>
<td style="text-align:left;">
6/911
</td>
<td style="text-align:left;">
15/11931
</td>
<td style="text-align:right;">
0.0005352
</td>
<td style="text-align:right;">
0.4902006
</td>
<td style="text-align:right;">
0.4902006
</td>
<td style="text-align:left;">
LOC113994469/LOC113993433/LOC113993299/LPAR1/APLNR/PDE3A
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000310" class="uri">GO:2000310</a>
</td>
<td style="text-align:left;">
regulation of NMDA receptor activity
</td>
<td style="text-align:left;">
8/911
</td>
<td style="text-align:left;">
31/11931
</td>
<td style="text-align:right;">
0.0018130
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:left;">
MEF2C/DAPK1/ACTN2/DLG3/OPRM1/GRIA4/GRIA1/RASGRF2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034975" class="uri">GO:0034975</a>
</td>
<td style="text-align:left;">
protein folding in endoplasmic reticulum
</td>
<td style="text-align:left;">
5/911
</td>
<td style="text-align:left;">
13/11931
</td>
<td style="text-align:right;">
0.0019666
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:left;">
HSPA5/HSP90B1/PDIA3/CALR/DNAJC3
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035023" class="uri">GO:0035023</a>
</td>
<td style="text-align:left;">
regulation of Rho protein signal transduction
</td>
<td style="text-align:left;">
7/911
</td>
<td style="text-align:left;">
25/11931
</td>
<td style="text-align:right;">
0.0020954
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:right;">
0.7677461
</td>
<td style="text-align:left;">
ARHGEF28/ARHGDIB/ARHGEF10/STARD8/LOC114000754/AKAP13/RASGRF2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
50/911
</td>
<td style="text-align:left;">
446/11931
</td>
<td style="text-align:right;">
0.0037799
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:left;">
LOC113994469/HTR1D/LANCL1/PLEKHG5/LOC113993090/GNAI3/PREX1/CCKBR/LOC113994699/LOC113993433/LOC113983719/CX3CR1/P2RY13/LOC113989258/ARHGEF26/ADORA1/LOC113993299/OPRM1/PLXNB1/ABR/GPRC5C/LPAR1/APLNR/TACR1/LPAR2/AKAP13/GRP/RIC8B/GPR85/GRM2/GRM7/RASGRF2/GPR162/GPR146/CARTPT/FZD5/QRFPR/PDE3A/AVPR1A/CALCR/LOC113983132/DGKQ/ARHGEF11/RAMP3/HTR1E/ADGRG2/CCK/CRHR2/LOC114003560/POMC
</td>
<td style="text-align:right;">
50
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006986" class="uri">GO:0006986</a>
</td>
<td style="text-align:left;">
response to unfolded protein
</td>
<td style="text-align:left;">
8/911
</td>
<td style="text-align:left;">
35/11931
</td>
<td style="text-align:right;">
0.0041211
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:left;">
FICD/HSP90AB1/HSPA8/HSP90AA1/MANF/UBE2J2/HSPA4/ERP27
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070166" class="uri">GO:0070166</a>
</td>
<td style="text-align:left;">
enamel mineralization
</td>
<td style="text-align:left;">
4/911
</td>
<td style="text-align:left;">
10/11931
</td>
<td style="text-align:right;">
0.0048811
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:left;">
TCIRG1/FAM20A/LOC113984874/STIM1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0086004" class="uri">GO:0086004</a>
</td>
<td style="text-align:left;">
regulation of cardiac muscle cell contraction
</td>
<td style="text-align:left;">
4/911
</td>
<td style="text-align:left;">
10/11931
</td>
<td style="text-align:right;">
0.0048811
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:left;">
ATP1A1/KCNJ2/ADORA1/LOC113993299
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
911
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030217" class="uri">GO:0030217</a>
</td>
<td style="text-align:left;">
T cell differentiation
</td>
<td style="text-align:left;">
6/911
</td>
<td style="text-align:left;">
22/11931
</td>
<td style="text-align:right;">
0.0050256
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:right;">
0.8309344
</td>
<td style="text-align:left;">
TCIRG1/DLL4/PREX1/PTPRC/PATZ1/IL7R
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
911
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="arcopallium-intermedium-ai" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Arcopallium
intermedium (Ai)</h1>
<p><br></p>
<div id="checking-the-sampling-9" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Checking the
sampling</h2>
<p>For the AI, we started with 16854 genes but after filering we have
14598 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have ai samples for 11, with respect
to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=0.54
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.0027
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-15" class="section level3"
number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/AI%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /><img src="03_DESeq2_files/figure-html/AI%20DEseq1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><br></p>
</div>
<div id="corrections-applied-11" class="section level3" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-65-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-65-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Removing the batch effect effectively controls for Year.</p>
<p><br></p>
</div>
</div>
<div id="status-ai" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Status AI</h2>
<div id="analysis-26" class="section level3" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14598 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 11, 0.075%
FALSE LFC &lt; 0 (down)     : 7, 0.048%
FALSE outliers [1]       : 10, 0.069%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 1)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AI%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-27" class="section level3" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AI%20status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AI%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AI according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0071470" class="uri">GO:0071470</a>
</td>
<td style="text-align:left;">
cellular response to osmotic stress
</td>
<td style="text-align:left;">
5/463
</td>
<td style="text-align:left;">
13/11914
</td>
<td style="text-align:right;">
0.0000862
</td>
<td style="text-align:right;">
0.1158070
</td>
<td style="text-align:right;">
0.1158070
</td>
<td style="text-align:left;">
LOC113991485/LOC113991484/LOC113991487/LOC113991482/LOC114001824
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002088" class="uri">GO:0002088</a>
</td>
<td style="text-align:left;">
lens development in camera-type eye
</td>
<td style="text-align:left;">
6/463
</td>
<td style="text-align:left;">
27/11914
</td>
<td style="text-align:right;">
0.0004919
</td>
<td style="text-align:right;">
0.2676867
</td>
<td style="text-align:right;">
0.2676867
</td>
<td style="text-align:left;">
GJE1/DLG1/PITX3/PROX1/UNC45A/CRYBB2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045670" class="uri">GO:0045670</a>
</td>
<td style="text-align:left;">
regulation of osteoclast differentiation
</td>
<td style="text-align:left;">
4/463
</td>
<td style="text-align:left;">
11/11914
</td>
<td style="text-align:right;">
0.0005975
</td>
<td style="text-align:right;">
0.2676867
</td>
<td style="text-align:right;">
0.2676867
</td>
<td style="text-align:left;">
BGLAP/CEBPB/CAMK4/MITF
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006271" class="uri">GO:0006271</a>
</td>
<td style="text-align:left;">
DNA strand elongation involved in DNA replication
</td>
<td style="text-align:left;">
4/463
</td>
<td style="text-align:left;">
12/11914
</td>
<td style="text-align:right;">
0.0008689
</td>
<td style="text-align:right;">
0.2919463
</td>
<td style="text-align:right;">
0.2919463
</td>
<td style="text-align:left;">
GINS1/MCM4/POLD2/POLD3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043278" class="uri">GO:0043278</a>
</td>
<td style="text-align:left;">
response to morphine
</td>
<td style="text-align:left;">
5/463
</td>
<td style="text-align:left;">
23/11914
</td>
<td style="text-align:right;">
0.0016327
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:left;">
SRR/PENK/CXCR4/PITX3/LOC113984425
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032201" class="uri">GO:0032201</a>
</td>
<td style="text-align:left;">
telomere maintenance via semi-conservative replication
</td>
<td style="text-align:left;">
5/463
</td>
<td style="text-align:left;">
25/11914
</td>
<td style="text-align:right;">
0.0024181
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:left;">
POLE2/POLE3/LOC114000574/POLD2/POLD3
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035023" class="uri">GO:0035023</a>
</td>
<td style="text-align:left;">
regulation of Rho protein signal transduction
</td>
<td style="text-align:left;">
5/463
</td>
<td style="text-align:left;">
25/11914
</td>
<td style="text-align:right;">
0.0024181
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:left;">
LOC114003556/ARHGEF10/ARHGDIB/STARD8/PDCD10
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006260" class="uri">GO:0006260</a>
</td>
<td style="text-align:left;">
DNA replication
</td>
<td style="text-align:left;">
12/463
</td>
<td style="text-align:left;">
121/11914
</td>
<td style="text-align:right;">
0.0025939
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:left;">
POLE2/POLE3/LOC114000574/MCM4/SDE2/MCM5/BRCA1/CDK2/NFIA/ORC6/LOC114001320/POLD2
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001676" class="uri">GO:0001676</a>
</td>
<td style="text-align:left;">
long-chain fatty acid metabolic process
</td>
<td style="text-align:left;">
4/463
</td>
<td style="text-align:left;">
16/11914
</td>
<td style="text-align:right;">
0.0028230
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:right;">
0.4215656
</td>
<td style="text-align:left;">
ACAD9/ACSL4/ACSL1/SLC27A6
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
463
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
8/463
</td>
<td style="text-align:left;">
67/11914
</td>
<td style="text-align:right;">
0.0042717
</td>
<td style="text-align:right;">
0.5626977
</td>
<td style="text-align:right;">
0.5626977
</td>
<td style="text-align:left;">
LOC113991485/LOC113992168/LOC113991484/LOC113991487/LOC113992709/LOC113991482/C3/CRB2
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
463
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-ai" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Testosterone AI</h2>
<div id="expression-data-exploration-16" class="section level3"
number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>No association with year.</p>
<p><br></p>
</div>
<div id="analysis-27" class="section level3" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 13, 0.089%
FALSE LFC &lt; 0 (down)     : 30, 0.21%
FALSE outliers [1]       : 25, 0.17%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AI%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-28" class="section level3" number="10.3.3">
<h3><span class="header-section-number">10.3.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AI%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AI%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AI according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0060438" class="uri">GO:0060438</a>
</td>
<td style="text-align:left;">
trachea development
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
10/11926
</td>
<td style="text-align:right;">
0.0000244
</td>
<td style="text-align:right;">
0.0273859
</td>
<td style="text-align:right;">
0.0273859
</td>
<td style="text-align:left;">
LOC113987257/LOC113994177/LOC114000433/LOC113999122/LOC113992388
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002064" class="uri">GO:0002064</a>
</td>
<td style="text-align:left;">
epithelial cell development
</td>
<td style="text-align:left;">
6/490
</td>
<td style="text-align:left;">
17/11926
</td>
<td style="text-align:right;">
0.0000392
</td>
<td style="text-align:right;">
0.0273859
</td>
<td style="text-align:right;">
0.0273859
</td>
<td style="text-align:left;">
LOC113987257/LOC113994177/LOC114000433/LOC113999122/LOC113992388/CXCR4
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1904158" class="uri">GO:1904158</a>
</td>
<td style="text-align:left;">
axonemal central apparatus assembly
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
12/11926
</td>
<td style="text-align:right;">
0.0000715
</td>
<td style="text-align:right;">
0.0332832
</td>
<td style="text-align:right;">
0.0332832
</td>
<td style="text-align:left;">
LOC113987257/LOC113994177/LOC114000433/LOC113999122/LOC113992388
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010976" class="uri">GO:0010976</a>
</td>
<td style="text-align:left;">
positive regulation of neuron projection development
</td>
<td style="text-align:left;">
13/490
</td>
<td style="text-align:left;">
107/11926
</td>
<td style="text-align:right;">
0.0004303
</td>
<td style="text-align:right;">
0.1502731
</td>
<td style="text-align:right;">
0.1502731
</td>
<td style="text-align:left;">
PTN/CNTN1/NDRG4/LOC113990140/CAMK1/LOC113996443/CFLAR/TWF1/AMIGO1/LOC114002148/FYN/PLPPR5/RGS2
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072657" class="uri">GO:0072657</a>
</td>
<td style="text-align:left;">
protein localization to membrane
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
18/11926
</td>
<td style="text-align:right;">
0.0006299
</td>
<td style="text-align:right;">
0.1759910
</td>
<td style="text-align:right;">
0.1759910
</td>
<td style="text-align:left;">
TM9SF3/TM9SF2/HPS6/HGS/PACSIN1
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016082" class="uri">GO:0016082</a>
</td>
<td style="text-align:left;">
synaptic vesicle priming
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
20/11926
</td>
<td style="text-align:right;">
0.0010648
</td>
<td style="text-align:right;">
0.2479230
</td>
<td style="text-align:right;">
0.2479230
</td>
<td style="text-align:left;">
LOC113991140/SNAP29/STXBP1/LOC113993758/LOC113987015
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061025" class="uri">GO:0061025</a>
</td>
<td style="text-align:left;">
membrane fusion
</td>
<td style="text-align:left;">
7/490
</td>
<td style="text-align:left;">
42/11926
</td>
<td style="text-align:right;">
0.0014639
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:left;">
LOC113991140/SNAP29/LOC113993758/DNM1L/LOC113987015/LOC113993578/DNM3
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021591" class="uri">GO:0021591</a>
</td>
<td style="text-align:left;">
ventricular system development
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
22/11926
</td>
<td style="text-align:right;">
0.0016899
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:left;">
LOC113987257/LOC113994177/LOC114000433/LOC113999122/LOC113992388
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045742" class="uri">GO:0045742</a>
</td>
<td style="text-align:left;">
positive regulation of epidermal growth factor receptor signaling
pathway
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
22/11926
</td>
<td style="text-align:right;">
0.0016899
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:right;">
0.2623042
</td>
<td style="text-align:left;">
AFAP1L2/PDE6G/LOC113992190/HAP1/CNOT9
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006749" class="uri">GO:0006749</a>
</td>
<td style="text-align:left;">
glutathione metabolic process
</td>
<td style="text-align:left;">
6/490
</td>
<td style="text-align:left;">
36/11926
</td>
<td style="text-align:right;">
0.0031798
</td>
<td style="text-align:right;">
0.4386326
</td>
<td style="text-align:right;">
0.4386326
</td>
<td style="text-align:left;">
LOC113990807/LOC113990865/LOC113989543/SLC1A1/LOC113984987/LOC113990605
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
490
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-5" class="section level2" number="10.4">
<h2><span class="header-section-number">10.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 4, 0.027%
FALSE LFC &lt; 0 (down)     : 3, 0.021%
FALSE outliers [1]       : 493, 3.4%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-69-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-ai" class="section level2" number="10.5">
<h2><span class="header-section-number">10.5</span> Strength AI</h2>
<div id="analysis-28" class="section level3" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14628 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 18, 0.12%
FALSE LFC &lt; 0 (down)     : 16, 0.11%
FALSE outliers [1]       : 21, 0.14%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/AI%20Strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/AI%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-29" class="section level3" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/AI%20Strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/AI%20Strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in AI according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0055072" class="uri">GO:0055072</a>
</td>
<td style="text-align:left;">
iron ion homeostasis
</td>
<td style="text-align:left;">
5/274
</td>
<td style="text-align:left;">
24/11926
</td>
<td style="text-align:right;">
0.0001833
</td>
<td style="text-align:right;">
0.1303095
</td>
<td style="text-align:right;">
0.1290652
</td>
<td style="text-align:left;">
B2M/HEPHL1/TMPRSS6/PICALM/STEAP3
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
14/274
</td>
<td style="text-align:left;">
205/11926
</td>
<td style="text-align:right;">
0.0002749
</td>
<td style="text-align:right;">
0.1303095
</td>
<td style="text-align:right;">
0.1290652
</td>
<td style="text-align:left;">
HSPG2/LOC113999927/ADAMTS14/SERPINB5/VWA1/COL4A4/COL1A1/ITGA2B/COL7A1/CSGALNACT1/ITGA4/ITGB6/MMP23B/LOC113989166
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043122" class="uri">GO:0043122</a>
</td>
<td style="text-align:left;">
regulation of I-kappaB kinase/NF-kappaB signaling
</td>
<td style="text-align:left;">
4/274
</td>
<td style="text-align:left;">
24/11926
</td>
<td style="text-align:right;">
0.0020153
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
UFL1/ZFAND6/SQSTM1/ANGPT1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045765" class="uri">GO:0045765</a>
</td>
<td style="text-align:left;">
regulation of angiogenesis
</td>
<td style="text-align:left;">
4/274
</td>
<td style="text-align:left;">
27/11926
</td>
<td style="text-align:right;">
0.0031535
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
EGLN1/HTATIP2/ADGRA2/BMPER
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007611" class="uri">GO:0007611</a>
</td>
<td style="text-align:left;">
learning or memory
</td>
<td style="text-align:left;">
5/274
</td>
<td style="text-align:left;">
46/11926
</td>
<td style="text-align:right;">
0.0039183
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
B2M/VIP/SLC1A1/CRHBP/PICALM
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019222" class="uri">GO:0019222</a>
</td>
<td style="text-align:left;">
regulation of metabolic process
</td>
<td style="text-align:left;">
3/274
</td>
<td style="text-align:left;">
15/11926
</td>
<td style="text-align:right;">
0.0044469
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
MC3R/LOC113983719/MC5R
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051289" class="uri">GO:0051289</a>
</td>
<td style="text-align:left;">
protein homotetramerization
</td>
<td style="text-align:left;">
5/274
</td>
<td style="text-align:left;">
48/11926
</td>
<td style="text-align:right;">
0.0047163
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
CBY1/B2M/LOC113984533/CBR4/KIF25
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043588" class="uri">GO:0043588</a>
</td>
<td style="text-align:left;">
skin development
</td>
<td style="text-align:left;">
4/274
</td>
<td style="text-align:left;">
31/11926
</td>
<td style="text-align:right;">
0.0052622
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
DACT2/COL1A1/LOC113984533/ITGB6
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2001224" class="uri">GO:2001224</a>
</td>
<td style="text-align:left;">
positive regulation of neuron migration
</td>
<td style="text-align:left;">
3/274
</td>
<td style="text-align:left;">
16/11926
</td>
<td style="text-align:right;">
0.0053811
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
SEMA6A/LOC114003556/SEMA3A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044267" class="uri">GO:0044267</a>
</td>
<td style="text-align:left;">
cellular protein metabolic process
</td>
<td style="text-align:left;">
9/274
</td>
<td style="text-align:left;">
144/11926
</td>
<td style="text-align:right;">
0.0059933
</td>
<td style="text-align:right;">
0.5075303
</td>
<td style="text-align:right;">
0.5026838
</td>
<td style="text-align:left;">
HSPG2/VWA1/ALB/B2M/CDH2/TSPAN5/MFGE8/APH1A/IGF2
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
274
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="lateral-septum-ls" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Lateral Septum
(Ls)</h1>
<p><br></p>
<div id="checking-the-sampling-10" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Checking the
sampling</h2>
<p>For the LS, we started with 16854 genes but after filering we have
14899 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have ls samples for 12, with respect
to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.0022
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<div id="expression-data-exploration-17" class="section level3"
number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/LS%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="corrections-applied-12" class="section level3" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-71-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-71-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-71-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>PFT2 was previously identified as an outlier, but using the
quantitative threshold and controlling for batch it doesn’t come out as
different. Controlling for Batch adequately controls for Year.</p>
</div>
</div>
<div id="status-ls" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Status LS</h2>
<div id="analysis-29" class="section level3" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 14899 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 2, 0.013%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 16, 0.11%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 1)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/LS%20status%20geneplots-1.png" width="1344" style="display: block; margin: auto;" />
<br></p>
</div>
<div id="gene-ontology-30" class="section level3" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/LS%20Status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/LS%20Status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in LS according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0036159" class="uri">GO:0036159</a>
</td>
<td style="text-align:left;">
inner dynein arm assembly
</td>
<td style="text-align:left;">
7/609
</td>
<td style="text-align:left;">
15/12088
</td>
<td style="text-align:right;">
0.0000036
</td>
<td style="text-align:right;">
0.0057326
</td>
<td style="text-align:right;">
0.0056371
</td>
<td style="text-align:left;">
LRRC6/LOC113981955/DNAAF5/CFAP73/DNAH1/DNAAF1/DNAH12
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003341" class="uri">GO:0003341</a>
</td>
<td style="text-align:left;">
cilium movement
</td>
<td style="text-align:left;">
10/609
</td>
<td style="text-align:left;">
38/12088
</td>
<td style="text-align:right;">
0.0000128
</td>
<td style="text-align:right;">
0.0102328
</td>
<td style="text-align:right;">
0.0100622
</td>
<td style="text-align:left;">
ARMC4/LRRC6/LOC113981955/DNAAF5/CFAP73/DNAH1/TTC29/DNAAF1/DNAH12/DNAH9
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071377" class="uri">GO:0071377</a>
</td>
<td style="text-align:left;">
cellular response to glucagon stimulus
</td>
<td style="text-align:left;">
7/609
</td>
<td style="text-align:left;">
22/12088
</td>
<td style="text-align:right;">
0.0000699
</td>
<td style="text-align:right;">
0.0371562
</td>
<td style="text-align:right;">
0.0365370
</td>
<td style="text-align:left;">
LOC113984696/CPS1/LOC113995499/ADCY5/PRKAR2A/LOC113999236/LOC113983744
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060271" class="uri">GO:0060271</a>
</td>
<td style="text-align:left;">
cilium assembly
</td>
<td style="text-align:left;">
22/609
</td>
<td style="text-align:left;">
181/12088
</td>
<td style="text-align:right;">
0.0001126
</td>
<td style="text-align:right;">
0.0416889
</td>
<td style="text-align:right;">
0.0409942
</td>
<td style="text-align:left;">
TMEM237/CC2D2A/IFT88/WDR35/CEP126/CFAP54/IFT74/TCTN2/LOC113992699/ARL13B/INTU/CEP89/NUDCD3/CFAP44/POC1B/ATXN10/KIF27/DNAAF1/IQCB1/SSX2IP/TMEM67/PCNT
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007018" class="uri">GO:0007018</a>
</td>
<td style="text-align:left;">
microtubule-based movement
</td>
<td style="text-align:left;">
12/609
</td>
<td style="text-align:left;">
68/12088
</td>
<td style="text-align:right;">
0.0001307
</td>
<td style="text-align:right;">
0.0416889
</td>
<td style="text-align:right;">
0.0409942
</td>
<td style="text-align:left;">
LOC114000985/DNAH10/LOC113982299/DNAH3/KIFC3/DNAH1/KIF6/KIF27/KIF21B/DNAH12/DNAH9/LOC113996658
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007190" class="uri">GO:0007190</a>
</td>
<td style="text-align:left;">
activation of adenylate cyclase activity
</td>
<td style="text-align:left;">
8/609
</td>
<td style="text-align:left;">
33/12088
</td>
<td style="text-align:right;">
0.0001794
</td>
<td style="text-align:right;">
0.0476815
</td>
<td style="text-align:right;">
0.0468869
</td>
<td style="text-align:left;">
LOC113984696/CALCR/DRD5/LOC114001840/ADCY5/ADORA3/LOC113999236/LOC113983744
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030317" class="uri">GO:0030317</a>
</td>
<td style="text-align:left;">
flagellated sperm motility
</td>
<td style="text-align:left;">
8/609
</td>
<td style="text-align:left;">
35/12088
</td>
<td style="text-align:right;">
0.0002781
</td>
<td style="text-align:right;">
0.0633726
</td>
<td style="text-align:right;">
0.0623166
</td>
<td style="text-align:left;">
CFAP65/LRRC6/TCTE1/DNAH1/PLTP/DNAJA1/LOC113999236/LOC113983744
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015718" class="uri">GO:0015718</a>
</td>
<td style="text-align:left;">
monocarboxylic acid transport
</td>
<td style="text-align:left;">
5/609
</td>
<td style="text-align:left;">
14/12088
</td>
<td style="text-align:right;">
0.0004370
</td>
<td style="text-align:right;">
0.0712075
</td>
<td style="text-align:right;">
0.0700209
</td>
<td style="text-align:left;">
SLC16A9/SLC16A5/LOC113990781/LOC113996953/SLC6A12
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060285" class="uri">GO:0060285</a>
</td>
<td style="text-align:left;">
cilium-dependent cell motility
</td>
<td style="text-align:left;">
5/609
</td>
<td style="text-align:left;">
14/12088
</td>
<td style="text-align:right;">
0.0004370
</td>
<td style="text-align:right;">
0.0712075
</td>
<td style="text-align:right;">
0.0700209
</td>
<td style="text-align:left;">
DNAH3/CFAP44/DNAH1/DNAH12/EFHC2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
609
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007283" class="uri">GO:0007283</a>
</td>
<td style="text-align:left;">
spermatogenesis
</td>
<td style="text-align:left;">
24/609
</td>
<td style="text-align:left;">
226/12088
</td>
<td style="text-align:right;">
0.0004464
</td>
<td style="text-align:right;">
0.0712075
</td>
<td style="text-align:right;">
0.0700209
</td>
<td style="text-align:left;">
LOC113982299/SPEF2/GJA1/GAMT/MSH4/CFAP54/LOC113992699/NUP62CL/SEPT4/ACVR2A/DHX36/LOC113987746/KATNAL1/LOC113992196/MNS1/DNMT3A/TDRD9/DNAJA1/UBR2/GGT7/PUM1/STK11/LOC113985256/LOC113996658
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
609
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-ls" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Testosterone LS</h2>
<div id="expression-data-exploration-18" class="section level3"
number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-30" class="section level3" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 62, 0.42%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/LS%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-31" class="section level3" number="11.3.3">
<h3><span class="header-section-number">11.3.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/LS%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/LS%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in LS according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0043525" class="uri">GO:0043525</a>
</td>
<td style="text-align:left;">
positive regulation of neuron apoptotic process
</td>
<td style="text-align:left;">
7/385
</td>
<td style="text-align:left;">
45/12041
</td>
<td style="text-align:right;">
0.0005140
</td>
<td style="text-align:right;">
0.3788809
</td>
<td style="text-align:right;">
0.3646207
</td>
<td style="text-align:left;">
FASLG/BAX/LOC113991532/ITGA11/ATF4/PRNP/CDC42
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0022010" class="uri">GO:0022010</a>
</td>
<td style="text-align:left;">
central nervous system myelination
</td>
<td style="text-align:left;">
4/385
</td>
<td style="text-align:left;">
13/12041
</td>
<td style="text-align:right;">
0.0005849
</td>
<td style="text-align:right;">
0.3788809
</td>
<td style="text-align:right;">
0.3646207
</td>
<td style="text-align:left;">
LOC113990764/ASPA/LOC113991947/PLP1
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016525" class="uri">GO:0016525</a>
</td>
<td style="text-align:left;">
negative regulation of angiogenesis
</td>
<td style="text-align:left;">
9/385
</td>
<td style="text-align:left;">
78/12041
</td>
<td style="text-align:right;">
0.0008260
</td>
<td style="text-align:right;">
0.3788809
</td>
<td style="text-align:right;">
0.3646207
</td>
<td style="text-align:left;">
CNMD/SPRY2/FASLG/THBS4/CLDN5/PLK2/APOH/SPARC/PDE3B
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008217" class="uri">GO:0008217</a>
</td>
<td style="text-align:left;">
regulation of blood pressure
</td>
<td style="text-align:left;">
7/385
</td>
<td style="text-align:left;">
54/12041
</td>
<td style="text-align:right;">
0.0015680
</td>
<td style="text-align:right;">
0.4929332
</td>
<td style="text-align:right;">
0.4743803
</td>
<td style="text-align:left;">
SGK1/STK39/POMC/SOD1/ACVRL1/CD34/LOC113992470
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003151" class="uri">GO:0003151</a>
</td>
<td style="text-align:left;">
outflow tract morphogenesis
</td>
<td style="text-align:left;">
6/385
</td>
<td style="text-align:left;">
41/12041
</td>
<td style="text-align:right;">
0.0017912
</td>
<td style="text-align:right;">
0.4929332
</td>
<td style="text-align:right;">
0.4743803
</td>
<td style="text-align:left;">
TBX2/NIPBL/SOX11/CLDN5/TGFBR3/HIF1A
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010466" class="uri">GO:0010466</a>
</td>
<td style="text-align:left;">
negative regulation of peptidase activity
</td>
<td style="text-align:left;">
3/385
</td>
<td style="text-align:left;">
10/12041
</td>
<td style="text-align:right;">
0.0032917
</td>
<td style="text-align:right;">
0.5361629
</td>
<td style="text-align:right;">
0.5159830
</td>
<td style="text-align:left;">
LOC113992709/LOC113992708/LOC113999601
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1900264" class="uri">GO:1900264</a>
</td>
<td style="text-align:left;">
positive regulation of DNA-directed DNA polymerase activity
</td>
<td style="text-align:left;">
3/385
</td>
<td style="text-align:left;">
10/12041
</td>
<td style="text-align:right;">
0.0032917
</td>
<td style="text-align:right;">
0.5361629
</td>
<td style="text-align:right;">
0.5159830
</td>
<td style="text-align:left;">
PCNA/RFC4/RFC3
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010738" class="uri">GO:0010738</a>
</td>
<td style="text-align:left;">
regulation of protein kinase A signaling
</td>
<td style="text-align:left;">
3/385
</td>
<td style="text-align:left;">
11/12041
</td>
<td style="text-align:right;">
0.0044195
</td>
<td style="text-align:right;">
0.5361629
</td>
<td style="text-align:right;">
0.5159830
</td>
<td style="text-align:left;">
AIP/PRKAR1A/AKAP12
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070987" class="uri">GO:0070987</a>
</td>
<td style="text-align:left;">
error-free translesion synthesis
</td>
<td style="text-align:left;">
4/385
</td>
<td style="text-align:left;">
22/12041
</td>
<td style="text-align:right;">
0.0047659
</td>
<td style="text-align:right;">
0.5361629
</td>
<td style="text-align:right;">
0.5159830
</td>
<td style="text-align:left;">
PCNA/RFC4/RCHY1/RFC3
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
385
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045778" class="uri">GO:0045778</a>
</td>
<td style="text-align:left;">
positive regulation of ossification
</td>
<td style="text-align:left;">
3/385
</td>
<td style="text-align:left;">
12/12041
</td>
<td style="text-align:right;">
0.0057542
</td>
<td style="text-align:right;">
0.5361629
</td>
<td style="text-align:right;">
0.5159830
</td>
<td style="text-align:left;">
NIPBL/SOX11/WNT5A
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
385
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-6" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 16, 0.11%
FALSE LFC &lt; 0 (down)     : 23, 0.16%
FALSE outliers [1]       : 29, 0.2%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 4)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-74-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-75-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-ls" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> Strength LS</h2>
<div id="analysis-31" class="section level3" number="11.5.1">
<h3><span class="header-section-number">11.5.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 14818 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 17, 0.11%
FALSE LFC &lt; 0 (down)     : 81, 0.55%
FALSE outliers [1]       : 53, 0.36%
FALSE low counts [2]     : 6849, 46%
FALSE (mean count &lt; 303)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/LS%20Strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/LS%20Strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-32" class="section level3" number="11.5.2">
<h3><span class="header-section-number">11.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/LS%20Strength%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/LS%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in LS according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0045176" class="uri">GO:0045176</a>
</td>
<td style="text-align:left;">
apical protein localization
</td>
<td style="text-align:left;">
8/1426
</td>
<td style="text-align:left;">
14/12041
</td>
<td style="text-align:right;">
0.0000589
</td>
<td style="text-align:right;">
0.1214821
</td>
<td style="text-align:right;">
0.1214821
</td>
<td style="text-align:left;">
SHROOM2/ARF4/GOPC/RDX/ATP8B1/INSC/JAM3/SHROOM3
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045116" class="uri">GO:0045116</a>
</td>
<td style="text-align:left;">
protein neddylation
</td>
<td style="text-align:left;">
6/1426
</td>
<td style="text-align:left;">
11/12041
</td>
<td style="text-align:right;">
0.0007430
</td>
<td style="text-align:right;">
0.6108154
</td>
<td style="text-align:right;">
0.6108154
</td>
<td style="text-align:left;">
DCUN1D1/UBA3/DCUN1D4/RNF7/UBE2M/DCUN1D2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006888" class="uri">GO:0006888</a>
</td>
<td style="text-align:left;">
endoplasmic reticulum to Golgi vesicle-mediated transport
</td>
<td style="text-align:left;">
32/1426
</td>
<td style="text-align:left;">
153/12041
</td>
<td style="text-align:right;">
0.0008891
</td>
<td style="text-align:right;">
0.6108154
</td>
<td style="text-align:right;">
0.6108154
</td>
<td style="text-align:left;">
COG1/RAB1A/USE1/GORASP1/COL7A1/MCFD2/TMED2/TMED7/ARF4/RAB2A/DYNC1I1/TMED5/TRAPPC2/DYNC1LI1/CNIH1/GOPC/KDELR2/TMED4/LMAN1/CAPZA2/SPTBN1/COG6/ANK2/LMAN2L/SEC24C/SPAST/COG4/SCFD1/SEC24B/VCP/BET1/TEX261
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006904" class="uri">GO:0006904</a>
</td>
<td style="text-align:left;">
vesicle docking involved in exocytosis
</td>
<td style="text-align:left;">
11/1426
</td>
<td style="text-align:left;">
34/12041
</td>
<td style="text-align:right;">
0.0013235
</td>
<td style="text-align:right;">
0.6213255
</td>
<td style="text-align:right;">
0.6213255
</td>
<td style="text-align:left;">
RAB12/STXBP1/VPS45/RAB3A/LOC113982219/VPS33B/VPS11/SCFD1/RAB8B/LOC113997864/LOC114001550
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015031" class="uri">GO:0015031</a>
</td>
<td style="text-align:left;">
protein transport
</td>
<td style="text-align:left;">
53/1426
</td>
<td style="text-align:left;">
298/12041
</td>
<td style="text-align:right;">
0.0015073
</td>
<td style="text-align:right;">
0.6213255
</td>
<td style="text-align:right;">
0.6213255
</td>
<td style="text-align:left;">
COG1/GPM6B/USE1/PSEN1/SCAMP1/GORASP1/GOLT1B/MCFD2/SNAP29/MLC1/LIN7C/ATG4D/SNX3/SNX24/RAB2A/RAB9A/RAB39A/GABARAPL2/GOPC/SFT2D1/APBA2/RAB33B/LOC113982219/RP2/KDELR2/PTPN23/VPS33B/ATG4C/YIPF5/WASHC3/PIK3R2/LOC113995129/COG6/ANK2/ARRDC4/LMAN2L/VTA1/COG4/EXOC2/VPS52/PHAX/RUFY1/BET1/COL1A1/LOC113992565/LOC113996864/LOC113997864/LOC114000510/LOC114001550/MYO1D/RAMP1/S100A13/TESC
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050821" class="uri">GO:0050821</a>
</td>
<td style="text-align:left;">
protein stabilization
</td>
<td style="text-align:left;">
32/1426
</td>
<td style="text-align:left;">
160/12041
</td>
<td style="text-align:right;">
0.0019270
</td>
<td style="text-align:right;">
0.6619383
</td>
<td style="text-align:right;">
0.6619383
</td>
<td style="text-align:left;">
LAMP1/STXBP1/SUGT1/BAG1/LAMP2/CAMLG/RAB21/CHP1/NOP53/NAA16/WFS1/NLK/EP300/CCNH/GNAQ/ANK2/ATP1B1/HSP90AB1/STX12/USP36/SUMO1/BAG4/HYPK/USP7/PFN2/LOC113990352/PTEN/CRTAP/LOC113992637/TAF9B/TESC/UBE2B
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043405" class="uri">GO:0043405</a>
</td>
<td style="text-align:left;">
regulation of MAP kinase activity
</td>
<td style="text-align:left;">
5/1426
</td>
<td style="text-align:left;">
10/12041
</td>
<td style="text-align:right;">
0.0034880
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:left;">
KSR1/DUSP19/HGS/TRIB2/PPEF2
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0097502" class="uri">GO:0097502</a>
</td>
<td style="text-align:left;">
mannosylation
</td>
<td style="text-align:left;">
5/1426
</td>
<td style="text-align:left;">
10/12041
</td>
<td style="text-align:right;">
0.0034880
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:left;">
ALG12/ALG1/ALG9/ALG2/ALG3
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008654" class="uri">GO:0008654</a>
</td>
<td style="text-align:left;">
phospholipid biosynthetic process
</td>
<td style="text-align:left;">
8/1426
</td>
<td style="text-align:left;">
23/12041
</td>
<td style="text-align:right;">
0.0036128
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:left;">
LOC113988212/MBOAT2/AGPAT5/GPAM/FITM2/LPGAT1/PCYT2/LOC113992314
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1426
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070979" class="uri">GO:0070979</a>
</td>
<td style="text-align:left;">
protein K11-linked ubiquitination
</td>
<td style="text-align:left;">
9/1426
</td>
<td style="text-align:left;">
28/12041
</td>
<td style="text-align:right;">
0.0037776
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:right;">
0.7546298
</td>
<td style="text-align:left;">
UBE2D3/UBE2W/UBE2E3/UBE2A/FZR1/ANAPC1/RNF4/UBE2L3/UBE2B
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1426
</td>
</tr>
</tbody>
</table>
<p><br><br></p>
</div>
</div>
</div>
<div id="bed-nucleus-of-the-stria-terminalis-bstm"
class="section level1" number="12">
<h1><span class="header-section-number">12</span> Bed Nucleus of the
Stria Terminalis (BSTm)</h1>
<p><br></p>
<div id="checking-the-sampling-11" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Checking the
sampling</h2>
<p>For the BSTm, we started with 16854 genes but after filering we have
15031 genes. Before I go into more detail, I am going to check that the
sampling is equal across batches with respect to our variables of
interest. With respect to Status we have bstm samples for 12, with
respect to T and the tracking data.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Status across sequencing runs, fisher test p=1
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
pilot
</th>
<th style="text-align:right;">
run2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
floater
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
territorial
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Sampling years distributed across batches, fisher test p=0.00055
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
2015
</th>
<th style="text-align:right;">
2017
</th>
<th style="text-align:right;">
2018
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pilot
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
run2
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><br></p>
<div id="expression-data-exploration-19" class="section level3"
number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/BSTm%20DEseq1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20DEseq1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="corrections-applied-13" class="section level3" number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> Corrections
applied</h3>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-79-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-79-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-79-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Batch controls effectively for Year.</p>
<p><br></p>
</div>
</div>
<div id="status-bstm" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Status BSTm</h2>
<div id="analysis-32" class="section level3" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Analysis</h3>
<p>Formula = ~ Batch + Status</p>
<pre><code>FALSE 
FALSE out of 15031 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 6, 0.04%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 5, 0.033%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20status%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/BSTm%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-33" class="section level3" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/BSTm%20Status%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/BSTm%20status%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in BSTm according to Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0048706" class="uri">GO:0048706</a>
</td>
<td style="text-align:left;">
embryonic skeletal system development
</td>
<td style="text-align:left;">
8/1014
</td>
<td style="text-align:left;">
21/12177
</td>
<td style="text-align:right;">
0.0001711
</td>
<td style="text-align:right;">
0.2413900
</td>
<td style="text-align:right;">
0.2411523
</td>
<td style="text-align:left;">
COL1A1/SP3/WNT11/NOG/PBX1/SHH/LOC113989878/MBTD1
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061564" class="uri">GO:0061564</a>
</td>
<td style="text-align:left;">
axon development
</td>
<td style="text-align:left;">
7/1014
</td>
<td style="text-align:left;">
17/12177
</td>
<td style="text-align:right;">
0.0002509
</td>
<td style="text-align:right;">
0.2413900
</td>
<td style="text-align:right;">
0.2411523
</td>
<td style="text-align:left;">
DYNLT1/NEFH/LOC114000515/PLP1/LOC113996738/LOC113991704/TSPAN2
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001666" class="uri">GO:0001666</a>
</td>
<td style="text-align:left;">
response to hypoxia
</td>
<td style="text-align:left;">
23/1014
</td>
<td style="text-align:left;">
136/12177
</td>
<td style="text-align:right;">
0.0008088
</td>
<td style="text-align:right;">
0.5010804
</td>
<td style="text-align:right;">
0.5005869
</td>
<td style="text-align:left;">
CHRNA4/VEGFC/BECN1/BNIP3/ITPR2/LOC113992301/SMAD4/CD24/DPP4/LOC113989812/SOD2/THBS1/HSD11B2/UCP3/WDR83/PRKAA1/PKM/PLOD1/P2RX3/CAT/SRF/HIF1A/ADORA1
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021762" class="uri">GO:0021762</a>
</td>
<td style="text-align:left;">
substantia nigra development
</td>
<td style="text-align:left;">
10/1014
</td>
<td style="text-align:left;">
39/12177
</td>
<td style="text-align:right;">
0.0010426
</td>
<td style="text-align:right;">
0.5010804
</td>
<td style="text-align:right;">
0.5005869
</td>
<td style="text-align:left;">
CKB/LOC114001565/RAD1/ZNF148/GNB4/PLP1/ATP5PF/CNP/GLUD1/LOC114000755
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006091" class="uri">GO:0006091</a>
</td>
<td style="text-align:left;">
generation of precursor metabolites and energy
</td>
<td style="text-align:left;">
9/1014
</td>
<td style="text-align:left;">
34/12177
</td>
<td style="text-align:right;">
0.0014454
</td>
<td style="text-align:right;">
0.5010804
</td>
<td style="text-align:right;">
0.5005869
</td>
<td style="text-align:left;">
SLC25A4/XYLB/ENPP1/LOC113996683/MCHR1/LOC113983432/PHKA2/ACO2/LOC113986170
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051028" class="uri">GO:0051028</a>
</td>
<td style="text-align:left;">
mRNA transport
</td>
<td style="text-align:left;">
9/1014
</td>
<td style="text-align:left;">
35/12177
</td>
<td style="text-align:right;">
0.0018042
</td>
<td style="text-align:right;">
0.5010804
</td>
<td style="text-align:right;">
0.5005869
</td>
<td style="text-align:left;">
XPO1/IWS1/LOC113992455/ZC3H3/LOC113992247/IGF2BP2/IGF2BP3/CETN3/NUTF2
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021522" class="uri">GO:0021522</a>
</td>
<td style="text-align:left;">
spinal cord motor neuron differentiation
</td>
<td style="text-align:left;">
6/1014
</td>
<td style="text-align:left;">
17/12177
</td>
<td style="text-align:right;">
0.0018231
</td>
<td style="text-align:right;">
0.5010804
</td>
<td style="text-align:right;">
0.5005869
</td>
<td style="text-align:left;">
ISL1/SOX4/PTCH1/SHH/LOC113997538/OLIG2
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001501" class="uri">GO:0001501</a>
</td>
<td style="text-align:left;">
skeletal system development
</td>
<td style="text-align:left;">
17/1014
</td>
<td style="text-align:left;">
96/12177
</td>
<td style="text-align:right;">
0.0022457
</td>
<td style="text-align:right;">
0.5400944
</td>
<td style="text-align:right;">
0.5395625
</td>
<td style="text-align:left;">
HAPLN2/LOC113993027/COL1A1/MMP16/TLL1/CNMD/SOX4/UFD1/NOG/ACAN/CHRD/PAPSS2/USP1/HAPLN4/IGF1/HDAC4/TNFRSF11B
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009653" class="uri">GO:0009653</a>
</td>
<td style="text-align:left;">
anatomical structure morphogenesis
</td>
<td style="text-align:left;">
14/1014
</td>
<td style="text-align:left;">
74/12177
</td>
<td style="text-align:right;">
0.0028417
</td>
<td style="text-align:right;">
0.6050786
</td>
<td style="text-align:right;">
0.6044827
</td>
<td style="text-align:left;">
DKK3/SOX4/SMAD4/SIAH1/FOXD1/BICD1/SMAD7/CPM/SOX21/TRIM13/IGF2BP2/SMAD5/F2R/IGF2BP3
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
1014
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006120" class="uri">GO:0006120</a>
</td>
<td style="text-align:left;">
mitochondrial electron transport, NADH to ubiquinone
</td>
<td style="text-align:left;">
9/1014
</td>
<td style="text-align:left;">
38/12177
</td>
<td style="text-align:right;">
0.0033228
</td>
<td style="text-align:right;">
0.6050786
</td>
<td style="text-align:right;">
0.6044827
</td>
<td style="text-align:left;">
NDUFB3/NDUFB5/NDUFV3/NDUFA4/NDUFS1/COQ9/NDUFB8/NDUFA12/NDUFA6
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1014
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="testosterone-bstm" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Testosterone
BSTm</h2>
<div id="expression-data-exploration-20" class="section level3"
number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> Expression data
exploration</h3>
<p><img src="03_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Status&quot;
FALSE [1] &quot;Year&quot;</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20continuous%20data%20set%20up-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="analysis-33" class="section level3" number="12.3.2">
<h3><span class="header-section-number">12.3.2</span> Analysis</h3>
<p>Formula = ~ mean_T</p>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 0, 0%
FALSE LFC &lt; 0 (down)     : 3, 0.02%
FALSE outliers [1]       : 32, 0.21%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20mean%20T-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20mean%20T%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/BSTm%20mean%20T%20geneplots-1.png" width="1344" style="display: block; margin: auto;" />
<br></p>
</div>
<div id="gene-ontology-34" class="section level3" number="12.3.3">
<h3><span class="header-section-number">12.3.3</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/BSTm%20mean%20T%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/BSTm%20mean%20T%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in BSTm according to mean_T
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0055119" class="uri">GO:0055119</a>
</td>
<td style="text-align:left;">
relaxation of cardiac muscle
</td>
<td style="text-align:left;">
4/585
</td>
<td style="text-align:left;">
11/12171
</td>
<td style="text-align:right;">
0.0013297
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
ATP1B1/ATP2A2/GSN/KCNJ2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021591" class="uri">GO:0021591</a>
</td>
<td style="text-align:left;">
ventricular system development
</td>
<td style="text-align:left;">
5/585
</td>
<td style="text-align:left;">
19/12171
</td>
<td style="text-align:right;">
0.0016710
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
LOC114000433/LOC113997903/LOC113994177/NME5/SEMA6D
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034614" class="uri">GO:0034614</a>
</td>
<td style="text-align:left;">
cellular response to reactive oxygen species
</td>
<td style="text-align:left;">
6/585
</td>
<td style="text-align:left;">
29/12171
</td>
<td style="text-align:right;">
0.0022134
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
PRDX3/MAPK1/MPV17/MAPK9/MMP2/PDGFRA
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000729" class="uri">GO:0000729</a>
</td>
<td style="text-align:left;">
DNA double-strand break processing
</td>
<td style="text-align:left;">
4/585
</td>
<td style="text-align:left;">
15/12171
</td>
<td style="text-align:right;">
0.0047171
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
LOC113998423/NBN/SMARCAD1/DNA2
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006695" class="uri">GO:0006695</a>
</td>
<td style="text-align:left;">
cholesterol biosynthetic process
</td>
<td style="text-align:left;">
6/585
</td>
<td style="text-align:left;">
34/12171
</td>
<td style="text-align:right;">
0.0051131
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
GGPS1/MSMO1/LOC113997963/DHCR24/LOC114002832/PLPP6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019233" class="uri">GO:0019233</a>
</td>
<td style="text-align:left;">
sensory perception of pain
</td>
<td style="text-align:left;">
6/585
</td>
<td style="text-align:left;">
34/12171
</td>
<td style="text-align:right;">
0.0051131
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
NPY1R/MMEL1/MAPK1/KCND2/NIPSNAP1/PRDM12
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051932" class="uri">GO:0051932</a>
</td>
<td style="text-align:left;">
synaptic transmission, GABAergic
</td>
<td style="text-align:left;">
4/585
</td>
<td style="text-align:left;">
16/12171
</td>
<td style="text-align:right;">
0.0060536
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
GABRG1/RAC3/GABRA2/LOC113990089
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031623" class="uri">GO:0031623</a>
</td>
<td style="text-align:left;">
receptor internalization
</td>
<td style="text-align:left;">
6/585
</td>
<td style="text-align:left;">
38/12171
</td>
<td style="text-align:right;">
0.0089263
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
ACKR3/CALCRL/CAV1/CD36/CXCR2/LOC113993578
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006891" class="uri">GO:0006891</a>
</td>
<td style="text-align:left;">
intra-Golgi vesicle-mediated transport
</td>
<td style="text-align:left;">
5/585
</td>
<td style="text-align:left;">
28/12171
</td>
<td style="text-align:right;">
0.0098851
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
RAB33B/COG5/COG6/RAB6A/VTI1A
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006693" class="uri">GO:0006693</a>
</td>
<td style="text-align:left;">
prostaglandin metabolic process
</td>
<td style="text-align:left;">
3/585
</td>
<td style="text-align:left;">
10/12171
</td>
<td style="text-align:right;">
0.0102877
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:right;">
0.866266
</td>
<td style="text-align:left;">
LOC114004119/PTGR2/ZADH2
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
585
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
</div>
<div id="mean-t-x-status-7" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> mean T x Status</h2>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 3, 0.02%
FALSE LFC &lt; 0 (down)     : 2, 0.013%
FALSE outliers [1]       : 11, 0.073%
FALSE low counts [2]     : 0, 0%
FALSE (mean count &lt; 5)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20mean%20T%20x%20Status-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-83-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/unnamed-chunk-84-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="strength-bstm" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Strength BSTm</h2>
<div id="analysis-34" class="section level3" number="12.5.1">
<h3><span class="header-section-number">12.5.1</span> Analysis</h3>
<p>Formula ~ scale(Strength)</p>
<pre><code>FALSE 
FALSE out of 15010 with nonzero total read count
FALSE adjusted p-value &lt; 0.1
FALSE LFC &gt; 0 (up)       : 8, 0.053%
FALSE LFC &lt; 0 (down)     : 7, 0.047%
FALSE outliers [1]       : 26, 0.17%
FALSE low counts [2]     : 870, 5.8%
FALSE (mean count &lt; 11)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20Strength-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>FALSE null device 
FALSE           1</code></pre>
<p><img src="03_DESeq2_files/figure-html/BSTm%20strength%20volcano%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_DESeq2_files/figure-html/BSTm%20strength%20geneplots-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="gene-ontology-35" class="section level3" number="12.5.2">
<h3><span class="header-section-number">12.5.2</span> Gene Ontology</h3>
<p><img src="03_DESeq2_files/figure-html/BSTm%20BP%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
<img src="03_DESeq2_files/figure-html/BSTm%20strength%20Enrich-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Top 10 enriched GO BP in BSTm according to Strength
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:right;">
n_annotated
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:1903077" class="uri">GO:1903077</a>
</td>
<td style="text-align:left;">
negative regulation of protein localization to plasma membrane
</td>
<td style="text-align:left;">
7/979
</td>
<td style="text-align:left;">
14/12171
</td>
<td style="text-align:right;">
0.0000442
</td>
<td style="text-align:right;">
0.0853381
</td>
<td style="text-align:right;">
0.0846167
</td>
<td style="text-align:left;">
PICALM/PID1/MRAP2/BCL2L1/TMEM59/PPFIA1/TMBIM1
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072659" class="uri">GO:0072659</a>
</td>
<td style="text-align:left;">
protein localization to plasma membrane
</td>
<td style="text-align:left;">
24/979
</td>
<td style="text-align:left;">
141/12171
</td>
<td style="text-align:right;">
0.0003458
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
LOC113988212/GPR158/LOC114002160/RAB3C/SLC4A1/KCNIP4/CDH2/ATP1B1/DPP6/MRAP2/LOC113988625/RAB15/PKP2/IFT20/ATP1B3/ZDHHC3/DLG1/EMP2/RAB12/ACTN2/TPBG/EPB41L3/TTC7B/GAK
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030225" class="uri">GO:0030225</a>
</td>
<td style="text-align:left;">
macrophage differentiation
</td>
<td style="text-align:left;">
6/979
</td>
<td style="text-align:left;">
14/12171
</td>
<td style="text-align:right;">
0.0004558
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
MMP9/VEGFA/CSF1R/CDC42/SIRT1/BMP4
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043647" class="uri">GO:0043647</a>
</td>
<td style="text-align:left;">
inositol phosphate metabolic process
</td>
<td style="text-align:left;">
11/979
</td>
<td style="text-align:left;">
44/12171
</td>
<td style="text-align:right;">
0.0005573
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
PLCD3/PLCH2/INPP5J/IPMK/PLCB4/PLCG1/PTEN/PLCG2/INPPL1/MINPP1/PPIP5K2
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043588" class="uri">GO:0043588</a>
</td>
<td style="text-align:left;">
skin development
</td>
<td style="text-align:left;">
9/979
</td>
<td style="text-align:left;">
32/12171
</td>
<td style="text-align:right;">
0.0006996
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
PKD1/DACT2/PDGFA/LOC113984010/SLITRK5/COL5A1/LOC113998649/PTCH2/FRAS1
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030534" class="uri">GO:0030534</a>
</td>
<td style="text-align:left;">
adult behavior
</td>
<td style="text-align:left;">
8/979
</td>
<td style="text-align:left;">
26/12171
</td>
<td style="text-align:right;">
0.0007167
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
PCDH17/LOC113984666/LOC113995009/GABRG2/CNTNAP2/PTEN/SLITRK5/LOC113982203
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042391" class="uri">GO:0042391</a>
</td>
<td style="text-align:left;">
regulation of membrane potential
</td>
<td style="text-align:left;">
16/979
</td>
<td style="text-align:left;">
82/12171
</td>
<td style="text-align:right;">
0.0007195
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
GABRG1/GABRA5/PXK/CHRNB3/GLRA2/GABRG2/KCNMA1/LOC113989700/LOC113999820/CHRNA7/GLRB/HCN1/DLG1/GABRA2/ACTN2/LOC113996568
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007268" class="uri">GO:0007268</a>
</td>
<td style="text-align:left;">
chemical synaptic transmission
</td>
<td style="text-align:left;">
30/979
</td>
<td style="text-align:left;">
202/12171
</td>
<td style="text-align:right;">
0.0007629
</td>
<td style="text-align:right;">
0.1840597
</td>
<td style="text-align:right;">
0.1825037
</td>
<td style="text-align:left;">
GABRG1/GABRA5/NPTX2/GALR3/SV2C/CHRNB3/LOC113984666/GLRA2/LOC113995009/GABRG2/LOC113992038/LOC113989700/LOC113999820/GAD2/CHRNA7/GLRB/APBA1/SLITRK5/KCND2/ATXN3/HTR7/DLG1/GABRA2/LOC114001565/PAFAH1B1/MPZL1/SLC6A1/SLC1A1/DTNA/LOC113996568
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050877" class="uri">GO:0050877</a>
</td>
<td style="text-align:left;">
nervous system process
</td>
<td style="text-align:left;">
10/979
</td>
<td style="text-align:left;">
40/12171
</td>
<td style="text-align:right;">
0.0009868
</td>
<td style="text-align:right;">
0.2116229
</td>
<td style="text-align:right;">
0.2098339
</td>
<td style="text-align:left;">
GABRG1/GABRA5/CHRNB3/GLRA2/GABRG2/LOC113999820/CHRNA7/GLRB/GABRA2/LOC113996568
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
979
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032228" class="uri">GO:0032228</a>
</td>
<td style="text-align:left;">
regulation of synaptic transmission, GABAergic
</td>
<td style="text-align:left;">
5/979
</td>
<td style="text-align:left;">
13/12171
</td>
<td style="text-align:right;">
0.0024811
</td>
<td style="text-align:right;">
0.4353275
</td>
<td style="text-align:right;">
0.4316474
</td>
<td style="text-align:left;">
KRAS/PLCL2/SYN3/PTEN/LOC113995955
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
979
</td>
</tr>
</tbody>
</table>
<p>Volcano plots and similarity of gene expression between tissues is in
another Rmarkdown on this site.</p>
<pre><code>FALSE R version 4.0.2 (2020-06-22)
FALSE Platform: x86_64-w64-mingw32/x64 (64-bit)
FALSE Running under: Windows 10 x64 (build 19045)
FALSE 
FALSE Matrix products: default
FALSE 
FALSE locale:
FALSE [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
FALSE [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
FALSE [5] LC_TIME=English_Australia.1252    
FALSE 
FALSE attached base packages:
FALSE  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
FALSE  [8] datasets  methods   base     
FALSE 
FALSE other attached packages:
FALSE  [1] viridis_0.6.1               viridisLite_0.4.0          
FALSE  [3] UpSetR_1.4.0                limma_3.44.3               
FALSE  [5] flashClust_1.01-2           clusterProfiler_3.16.1     
FALSE  [7] WGCNA_1.69                  fastcluster_1.1.25         
FALSE  [9] dynamicTreeCut_1.63-1       wesanderson_0.3.6          
FALSE [11] VennDiagram_1.6.20          futile.logger_1.4.3        
FALSE [13] kableExtra_1.3.1            DESeq2_1.28.1              
FALSE [15] SummarizedExperiment_1.18.1 DelayedArray_0.14.0        
FALSE [17] matrixStats_0.56.0          Biobase_2.48.0             
FALSE [19] GenomicRanges_1.40.0        GenomeInfoDb_1.24.0        
FALSE [21] IRanges_2.22.2              S4Vectors_0.26.1           
FALSE [23] BiocGenerics_0.34.0         lubridate_1.8.0            
FALSE [25] stringr_1.4.0               fdrtool_1.2.15             
FALSE [27] PCAtools_2.0.0              cowplot_1.0.0              
FALSE [29] lattice_0.20-41             ggrepel_0.8.2              
FALSE [31] ggpubr_0.4.0                gridExtra_2.3              
FALSE [33] ggplot2_3.4.4               reshape2_1.4.4             
FALSE [35] plyr_1.8.6                 
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] utf8_1.1.4                tidyselect_1.1.0         
FALSE   [3] RSQLite_2.2.0             AnnotationDbi_1.50.0     
FALSE   [5] htmlwidgets_1.5.3         BiocParallel_1.22.0      
FALSE   [7] scatterpie_0.1.5          munsell_0.5.0            
FALSE   [9] ragg_1.2.1                codetools_0.2-16         
FALSE  [11] preprocessCore_1.50.0     withr_2.5.2              
FALSE  [13] colorspace_1.4-1          GOSemSim_2.14.2          
FALSE  [15] highr_0.8                 knitr_1.29               
FALSE  [17] rstudioapi_0.13           ggsignif_0.6.0           
FALSE  [19] DOSE_3.14.0               labeling_0.3             
FALSE  [21] urltools_1.7.3            GenomeInfoDbData_1.2.3   
FALSE  [23] polyclip_1.10-0           bit64_0.9-7              
FALSE  [25] farver_2.0.3              downloader_0.4           
FALSE  [27] vctrs_0.6.5               generics_0.0.2           
FALSE  [29] lambda.r_1.2.4            xfun_0.29                
FALSE  [31] R6_2.4.1                  doParallel_1.0.15        
FALSE  [33] graphlayouts_0.7.1        rsvd_1.0.3               
FALSE  [35] locfit_1.5-9.4            gridGraphics_0.5-1       
FALSE  [37] bitops_1.0-6              fgsea_1.14.0             
FALSE  [39] assertthat_0.2.1          scales_1.3.0             
FALSE  [41] ggraph_2.0.5              nnet_7.3-14              
FALSE  [43] enrichplot_1.8.1          gtable_0.3.0             
FALSE  [45] tidygraph_1.2.0           rlang_1.1.2              
FALSE  [47] genefilter_1.70.0         systemfonts_1.0.3        
FALSE  [49] splines_4.0.2             rstatix_0.7.0            
FALSE  [51] acepack_1.4.1             impute_1.62.0            
FALSE  [53] broom_0.7.10              europepmc_0.4            
FALSE  [55] checkmate_2.0.0           BiocManager_1.30.10      
FALSE  [57] yaml_2.2.1                abind_1.4-5              
FALSE  [59] backports_1.1.7           qvalue_2.20.0            
FALSE  [61] Hmisc_4.4-0               tools_4.0.2              
FALSE  [63] ggplotify_0.0.5           ellipsis_0.3.2           
FALSE  [65] jquerylib_0.1.4           RColorBrewer_1.1-2       
FALSE  [67] ggridges_0.5.3            Rcpp_1.0.7               
FALSE  [69] base64enc_0.1-3           progress_1.2.2           
FALSE  [71] zlibbioc_1.34.0           purrr_0.3.4              
FALSE  [73] RCurl_1.98-1.2            prettyunits_1.1.1        
FALSE  [75] rpart_4.1-15              haven_2.3.1              
FALSE  [77] cluster_2.1.0             magrittr_2.0.1           
FALSE  [79] data.table_1.13.0         futile.options_1.0.1     
FALSE  [81] DO.db_2.9                 openxlsx_4.1.5           
FALSE  [83] triebeard_0.3.0           hms_1.1.1                
FALSE  [85] evaluate_0.14             xtable_1.8-4             
FALSE  [87] XML_3.99-0.5              rio_0.5.16               
FALSE  [89] jpeg_0.1-8.1              readxl_1.3.1             
FALSE  [91] compiler_4.0.2            tibble_3.1.6             
FALSE  [93] crayon_1.4.2              htmltools_0.5.2          
FALSE  [95] snow_0.4-3                Formula_1.2-3            
FALSE  [97] tidyr_1.1.4               geneplotter_1.66.0       
FALSE  [99] DBI_1.1.0                 tweenr_1.0.1             
FALSE [101] formatR_1.8               MASS_7.3-51.6            
FALSE [103] Matrix_1.6-4              car_3.0-8                
FALSE [105] cli_3.6.1                 igraph_1.2.6             
FALSE [107] forcats_0.5.1             pkgconfig_2.0.3          
FALSE [109] rvcheck_0.1.8             foreign_0.8-80           
FALSE [111] xml2_1.3.2                foreach_1.5.0            
FALSE [113] annotate_1.66.0           dqrng_0.2.1              
FALSE [115] webshot_0.5.2             XVector_0.28.0           
FALSE [117] rvest_1.0.2               digest_0.6.25            
FALSE [119] rmarkdown_2.11            cellranger_1.1.0         
FALSE [121] fastmatch_1.1-0           htmlTable_2.0.1          
FALSE [123] DelayedMatrixStats_1.10.0 curl_4.3                 
FALSE [125] lifecycle_1.0.4           jsonlite_1.7.2           
FALSE [127] carData_3.0-4             fansi_0.4.1              
FALSE [129] pillar_1.6.4              fastmap_1.1.0            
FALSE [131] httr_1.4.2                survival_3.2-3           
FALSE [133] GO.db_3.11.4              glue_1.5.1               
FALSE [135] zip_2.0.4                 png_0.1-7                
FALSE [137] iterators_1.0.12          bit_4.0.4                
FALSE [139] ggforce_0.3.2             stringi_1.4.6            
FALSE [141] blob_1.2.1                textshaping_0.3.6        
FALSE [143] BiocSingular_1.4.0        latticeExtra_0.6-29      
FALSE [145] memoise_1.1.0             dplyr_1.0.7              
FALSE [147] irlba_2.3.3</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
