<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive Change May 2021" />


<title>Whole brain Co-expression</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_DESeq2.html">Differential Expression</a>
    </li>
    <li>
      <a href="./03_WGCNA_wholebrain.html">Whole-brain co-expression</a>
    </li>
    <li>
      <a href="./04_WGCNA_single_tissue.html">Co-expression in each tissue</a>
    </li>
    <li>
      <a href="./05_candidate_gene_analysis.html">Candidate Genes</a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Whole brain Co-expression</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive Change May 2021</h4>
<h4 class="date">Last Knit “2022-05-30”</h4>

</div>


<pre><code>FALSE Allowing parallel execution with up to 7 working processes.</code></pre>
<p><br><br></p>
<p>In <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/gbb.12560">Horton et al (2020)</a>, they identified bunches of genes with similar expression patterns (co-expression modules) that aligned with differences among brain nuclei. The co-expression analysis here (using WGCNA), has two aims:</p>
<ul>
<li><p>Replicate the results of Horton et al (2020)</p></li>
<li><p>To identify if there are co-expression modules that are shared amongst brain regions that are associated with our traits of interest.</p></li>
</ul>
<p>This second aim is in part a complement to the whole brain PCA analysis done in the <a href="https://periperipatus.github.io/PIFI_brain_transcriptome/02_DESeq2_FINAL.html">DESeq2 document</a>.</p>
<div id="data-quality-control" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Data Quality Control</h1>
<p>This is where I originally ran the test on the effect of batch correction on the whole brain dataset (now also presented in the <a href="https://periperipatus.github.io/PIFI_brain_transcriptome/02_DESeq2_FINAL.html">DESeq2 document</a>). As identified in that document and the QC process, <code>PFT2_POM</code> sample was removed from this analysis.</p>
<p><img src="03_WGCNA_wholebrain_files/figure-html/batch%20correction-1.png" width="3360" angle=90 style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/batch%20correction-2.png" width="3360" angle=90 style="display: block; margin: auto;" /></p>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<pre><code>FALSE [1] TRUE</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/batch%20correction-3.png" width="3360" angle=90 style="display: block; margin: auto;" /></p>
<p>The analysis above also suggests PFT1_TNA_run2 is an outlier, based on the connectivity to the remainder of the analysis. In the analysis below that excludes the pilot variables I ran it in combinations that did and did not include this sample (not all shown here). Those analyses that did include the sample showed the same outlier status (not shown), therefore I have removed it for this version.</p>
<p>What do the data look like without the pilot samples (Horton et al 2020) and batch correction? Here I rerun the same analyses as above but excluding the batch correction and without the pilot samples.</p>
<p><br></p>
<div id="removing-the-pilot-data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Removing the pilot data</h2>
<p><img src="03_WGCNA_wholebrain_files/figure-html/removing%20pilot%20data-1.png" width="672" style="display: block; margin: auto;" /><img src="03_WGCNA_wholebrain_files/figure-html/removing%20pilot%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>FALSE [1] &quot;Batch&quot;
FALSE [1] &quot;Tissue&quot;
FALSE [1] &quot;Status&quot;</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/removing%20pilot%20data-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>The PCA plot without batch correction still shows that tissues (and different batches) are clustered more closely. Perhaps in part this is still because batches are not evenly distributed across tissues.</p>
<p>Going forward with the WGCNA analysis on all brain samples, I will exclude pilot data and I will not conduct batch correction. However, I will include a batch variable in the trait-module correlation matrix.</p>
<p><br><br></p>
</div>
</div>
<div id="wgcna-analysis" class="section level1" number="2">
<h1><span class="header-section-number">2</span> WGCNA Analysis</h1>
<p>Using the dataset without the pilot data and the two outliers <code>PFT2_POM_run1</code> and <code>PFT1_TNA_run2</code>, I will now commence the main co-expression analysis.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(vsd_data))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(datExpr0, file=&quot;wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>gsg <span class="ot">=</span> <span class="fu">goodSamplesGenes</span>(wgcnadata, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>FALSE  Flagging genes and samples with too many missing values...
FALSE   ..step 1</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<p><br></p>
<div id="soft-threshold-selection" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Soft Threshold Selection</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>powers<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">11</span>, <span class="at">by =</span> <span class="dv">1</span>), <span class="fu">seq</span>(<span class="dv">12</span>, <span class="dv">26</span>, <span class="at">by =</span> <span class="dv">2</span>))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>sft<span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr0, <span class="at">powerVector=</span>powers, <span class="at">verbose=</span><span class="dv">0</span>, <span class="at">networkType=</span><span class="st">&quot;signed&quot;</span>)</span></code></pre></div>
<pre><code>FALSE    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
FALSE 1      1    0.128 108.00          0.968 6810.00   6810.00   6870
FALSE 2      2    0.174  -9.19          0.893 3720.00   3710.00   4120
FALSE 3      3    0.111  -2.24          0.923 2170.00   2160.00   2750
FALSE 4      4    0.208  -1.89          0.945 1340.00   1320.00   2010
FALSE 5      5    0.343  -1.81          0.966  868.00    846.00   1540
FALSE 6      6    0.456  -1.74          0.978  584.00    559.00   1220
FALSE 7      7    0.558  -1.72          0.984  407.00    380.00    997
FALSE 8      8    0.641  -1.77          0.984  291.00    264.00    831
FALSE 9      9    0.695  -1.81          0.979  214.00    188.00    705
FALSE 10    10    0.734  -1.82          0.980  161.00    135.00    606
FALSE 11    11    0.757  -1.85          0.975  123.00     99.40    526
FALSE 12    12    0.784  -1.87          0.976   95.80     73.80    462
FALSE 13    14    0.821  -1.90          0.981   60.70     41.90    364
FALSE 14    16    0.845  -1.90          0.985   40.30     24.70    294
FALSE 15    18    0.859  -1.93          0.985   27.80     15.00    243
FALSE 16    20    0.873  -1.92          0.987   19.90      9.39    203
FALSE 17    22    0.870  -1.94          0.981   14.60      6.00    172
FALSE 18    24    0.876  -1.93          0.983   10.90      3.91    148
FALSE 19    26    0.883  -1.91          0.983    8.36      2.60    128</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">0.9</span>;</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit,signed R^2&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>));</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>);</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># this line corresponds to using an R^2 cut-off of h</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.90</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>,</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>], <span class="at">labels=</span>powers, <span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/soft%20threshold-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">20</span></span></code></pre></div>
<p>The soft-threshold power I have selected here is 20, after which there is no improvement in scale free topology.</p>
<p><br></p>
</div>
<div id="adjacency-and-topological-overlap-matrices" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Adjacency and Topological Overlap matrices</h2>
<p>The adjacency matrix and Topological Overlap was made on an external server to save my PC some RAMs.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">20</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;wholebrain_vsd_nobatchrm.csv&quot;</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(datExpr0)<span class="ot">&lt;-</span> datExpr0<span class="sc">$</span>X</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>datExpr0<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>adjacency<span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr0, <span class="at">power =</span> softPower, <span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>TOM<span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMtype=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co">#dissTOM&lt;- 1-TOM</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(adjacency, TOM, <span class="at">file=</span><span class="st">&quot;wholebrain_network.RData&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="identify-co-expression-modules" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Identify co-expression modules</h2>
<p>The first step is to identify modules of genes with similar gene expression. Basically, the tool creates a hierarchical clustering of the topological dissimilarity between genes.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#for the rmarkdown knit this data is already loaded above.</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#datExpr0&lt;- read.csv(&quot;../WGCNA_results/all_tissues/wholebrain_vsd_nobatchrm.csv&quot;)</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(datExpr0)&lt;- datExpr0$X</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#datExpr0$X&lt;- NULL</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;../WGCNA_results/all_tissues/wholebrain_network.RData&quot;</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>dissTOM<span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>TOM</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>geneTree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method=</span><span class="st">&quot;average&quot;</span>)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(geneTree, xlab=&quot;&quot;, sub=&quot;&quot;, main= &quot;Gene Clustering on TOM-based dissimilarity&quot;, labels= FALSE, hang=0.04)</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>minModuleSize<span class="ot">&lt;-</span><span class="dv">30</span> </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>dynamicMods<span class="ot">&lt;-</span><span class="fu">cutreeDynamic</span>(<span class="at">dendro=</span> geneTree, <span class="at">distM=</span> dissTOM, <span class="at">deepSplit=</span><span class="dv">2</span>, <span class="at">pamRespectsDendro=</span> <span class="cn">FALSE</span>, <span class="at">minClusterSize=</span> minModuleSize)</span></code></pre></div>
<pre><code>FALSE  ..cutHeight not given, setting it to 0.998  ===&gt;  99% of the (truncated) height range in dendro.
FALSE  ..done.</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dynamicMods)</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>dynamicColors<span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span> <span class="fl">0.05</span>, <span class="at">main=</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/calculate%20modules-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now we try to merge some of these modules that are particularly similar in expression based on a similarity threshold.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Merge modules whose expression profiles are very similar</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>MEList<span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, <span class="at">colors=</span> dynamicColors)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>MEs<span class="ot">=</span> MEList<span class="sc">$</span>eigengenes</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate dissimilarity of module eigenegenes</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>MEDiss<span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(MEs)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Cluster module eigengenes</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>METree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method=</span> <span class="st">&quot;average&quot;</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(METree, main= &quot;Clustering of module eigengenes&quot;, xlab= &quot;&quot;, sub= &quot;&quot;)</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>MEDissThres<span class="ot">=</span> <span class="fl">0.30</span> <span class="co"># i.e. merge modules with an r2 &gt; 0.90. This is stringent, could relax to reduce number of modules and increase module size.</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co">#abline(h=MEDissThres, col=&quot;red&quot;)</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>merge<span class="ot">=</span> <span class="fu">mergeCloseModules</span>(datExpr0, dynamicColors, <span class="at">cutHeight=</span> MEDissThres, <span class="at">verbose =</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>FALSE  mergeCloseModules: Merging modules whose distance is less than 0.3
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 61 module eigengenes in given set.
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 27 module eigengenes in given set.
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 22 module eigengenes in given set.
FALSE    Calculating new MEs...
FALSE    multiSetMEs: Calculating module MEs.
FALSE      Working on set 1 ...
FALSE      moduleEigengenes: Calculating 22 module eigengenes in given set.</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>mergedColors<span class="ot">=</span>merge<span class="sc">$</span>colors</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>mergedMEs<span class="ot">=</span> merge<span class="sc">$</span>newMEs</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, <span class="fu">cbind</span>(dynamicColors, mergedColors), <span class="fu">c</span>(<span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="st">&quot;Merged dynamic&quot;</span>), <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>moduleColors<span class="ot">=</span> mergedColors</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>colorOrder<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="fu">standardColors</span>(<span class="dv">50</span>))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>moduleLabels<span class="ot">=</span> <span class="fu">match</span>(moduleColors, colorOrder)<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>MEs<span class="ot">=</span>mergedMEs</span></code></pre></div>
<p><br></p>
</div>
<div id="module-trait-correlation" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Module-trait correlation</h2>
<p>Correlate the eigenvector of each of the 22 co-expression modules with brain regions, individual and batch, as well as our interest variables (mean testosterone, status, and social network strength)</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> key_brain</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(datTraits))</span></code></pre></div>
<pre><code>FALSE [1] TRUE</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>Batch<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;run1&quot;</span>,datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>Ai<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;AI&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>TnA<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TNA&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>BSTm<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;BST&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>ICO<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;ICO&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>GCT<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;GCT&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>LS<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;LS&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>POM<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;POM&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>VMH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;VMH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>AH<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;AH&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>PVN<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;PVN&quot;</span>, datTraits<span class="sc">$</span>sampleID), <span class="dv">1</span>,<span class="dv">0</span>)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>Bird<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>sampleID))</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>datTraits<span class="sc">$</span>Status2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(datTraits<span class="sc">$</span>Status<span class="sc">==</span><span class="st">&quot;territorial&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> <span class="fu">subset</span>(datTraits, <span class="at">select=</span><span class="fu">c</span>(<span class="st">&quot;Bird&quot;</span>,<span class="st">&quot;Batch&quot;</span>,<span class="st">&quot;Status2&quot;</span>, <span class="st">&quot;mean_T&quot;</span>,<span class="st">&quot;strength.all_study&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;Ai&quot;</span>,<span class="st">&quot;TnA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(datTraits)[<span class="fu">names</span>(datTraits)<span class="sc">==</span><span class="st">&quot;Status2&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Status&quot;</span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Define numbers of genes and samples</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>nGenes <span class="ot">=</span> <span class="fu">ncol</span>(datExpr0);</span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>nSamples <span class="ot">=</span> <span class="fu">nrow</span>(datExpr0);</span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Recalculate MEs with color labels</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>MEs0 <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, moduleColors)<span class="sc">$</span>eigengenes</span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a><span class="co">#-----Correlations of genes with eigengenes</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>moduleGeneCor<span class="ot">=</span><span class="fu">cor</span>(MEs,datExpr0)</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>moduleGenePvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleGeneCor, nSamples);</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>moduleTraitCor <span class="ot">=</span> <span class="fu">cor</span>(MEs, datTraits, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>);</span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>moduleTraitPvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleTraitCor, nSamples);</span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------------Module-trait heatmap</span></span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>textMatrix <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">signif</span>(moduleTraitCor, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>,</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">signif</span>(moduleTraitPvalue, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>);</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">=</span> <span class="fu">dim</span>(moduleTraitCor)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">10</span>, <span class="dv">3</span>, <span class="dv">3</span>));</span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the correlation values within a heatmap plot</span></span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> moduleTraitCor,</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>               <span class="at">xLabels =</span> <span class="fu">names</span>(datTraits),</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>               <span class="at">colorLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>               <span class="at">colors =</span> <span class="fu">blueWhiteRed</span>(<span class="dv">50</span>),</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix,</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>               <span class="at">setStdMargins =</span> <span class="cn">FALSE</span>,</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">cex.text =</span> <span class="fl">0.5</span>,</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>               <span class="at">zlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>               <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module-trait relationships in whole brain&quot;</span>))</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/trait%20correlation-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><br></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Number of genes per co-expression module
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
1143
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
891
</td>
</tr>
<tr>
<td style="text-align:left;">
cyan
</td>
<td style="text-align:right;">
680
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
1487
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
509
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen
</td>
<td style="text-align:right;">
136
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
91
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen4
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
216
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
275
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
50
</td>
</tr>
<tr>
<td style="text-align:left;">
ivory
</td>
<td style="text-align:right;">
164
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple3
</td>
<td style="text-align:right;">
333
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite2
</td>
<td style="text-align:right;">
808
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
662
</td>
</tr>
<tr>
<td style="text-align:left;">
plum1
</td>
<td style="text-align:right;">
488
</td>
</tr>
<tr>
<td style="text-align:left;">
purple
</td>
<td style="text-align:right;">
1592
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna3
</td>
<td style="text-align:right;">
1316
</td>
</tr>
<tr>
<td style="text-align:left;">
tan
</td>
<td style="text-align:right;">
1212
</td>
</tr>
<tr>
<td style="text-align:left;">
violet
</td>
<td style="text-align:right;">
350
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow
</td>
<td style="text-align:right;">
628
</td>
</tr>
<tr>
<td style="text-align:left;">
yellowgreen
</td>
<td style="text-align:right;">
536
</td>
</tr>
</tbody>
</table>
<p>See the data portion of this repository to see the the module membership and gene-significance results.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>datME<span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr0,mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>datKME<span class="ot">&lt;-</span> <span class="fu">signedKME</span>(datExpr0, datME, <span class="at">outputColumnName=</span><span class="st">&quot;MM.&quot;</span>) <span class="co">#use the &quot;signed eigennode connectivity&quot; or module membership</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>MMPvalue <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(datKME), nSamples)) <span class="co"># Calculate module membership P-values</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>datKME<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(datKME)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>MMPvalue<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(MMPvalue)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>genes<span class="ot">=</span><span class="fu">names</span>(datExpr0)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>genes,<span class="at">moduleColor=</span>moduleColors)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>color<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, datKME, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>) <span class="co">#these are from your original WGCNA analysis </span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co">#head(color)</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(color), <span class="at">file =</span> <span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/all_tissues/wholebrain_results_ModuleMembership.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;_&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>MMPvalue<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, MMPvalue, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(MMPvalue, <span class="at">file=</span><span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/all_tissues/wholebrain_results_ModuleMembership_P-value.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;_&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="do">#### gene-significance with traits of interest.</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>Status) <span class="co">#change here for traits of interest</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(trait) <span class="ot">=</span> <span class="st">&quot;status&quot;</span> <span class="co">#change here for traits of interest </span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>modNames <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">names</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>mean_T)</span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;mean_T&quot;</span></span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>GS2<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>strength.all_study)</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;strength&quot;</span></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>GS3<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>GS<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(GS)</span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(GS,GS2, GS3)</span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0,GS, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(GS, <span class="at">file=</span><span class="st">&quot;../WGCNA_results/all_tissues/wholebrain_results_GeneSignificance.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="intermodule-correlations" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> intermodule correlations</h2>
<p>Given the network type used, all genes in a module can only go in the same direction. This means that another module may represent genes that are downregulated by the upregulation of genes in another module (or vice-versa).</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify colors</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>colz<span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(MEs))</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colz)<span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>ann_colors <span class="ot">=</span> <span class="fu">list</span>(<span class="at">module=</span>colz)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>annotation_col<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">colnames</span>(MEs), <span class="at">module=</span><span class="fu">colnames</span>(MEs))</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>correlation<span class="ot">&lt;-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(correlation, <span class="at">annotation_col =</span> annotation_col, <span class="at">annotation_colors =</span> ann_colors, <span class="at">annotation_legend=</span><span class="cn">FALSE</span>, <span class="at">legend_breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="fl">0.5</span>,<span class="dv">0</span>,<span class="fl">0.5</span>, <span class="dv">1</span>,<span class="dv">1</span>), </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="at">main=</span><span class="st">&quot;module eigengene correlation in whole brain&quot;</span>, <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;-1&quot;</span>, <span class="st">&quot;-0.5&quot;</span>, <span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.5&quot;</span>,<span class="st">&quot;1&quot;</span>,<span class="st">&quot;correlation</span><span class="sc">\n\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/intermodule%20correlations-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="interesting-modules" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Interesting Modules</h2>
<p>There are three modules that appear to be associated with our traits of interest. Let’s explore these further. Because the module-trait correlation procedure cannot account for tissues taken from the same individual, I will test to see if these relationships still hold when accounting for this in a linear mixed model.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmerTest)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co">#merge module eigengenes with trait matrix.</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>mes<span class="ot">&lt;-</span> MEs</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>mes<span class="sc">$</span>sampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mes)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>mes<span class="ot">&lt;-</span> <span class="fu">merge</span>(key_brain, mes, <span class="at">by=</span><span class="st">&quot;sampleID&quot;</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>mes<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">as.factor</span>(mes<span class="sc">$</span>Tissue)</span></code></pre></div>
<div id="darkturquoise-module" class="section level3" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> darkturquoise module</h3>
<p>This module is potentially associated with social status.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lme</span>(MEdarkturquoise<span class="sc">~</span>Batch <span class="sc">+</span> Tissue<span class="sc">*</span>Status,<span class="at">random=</span><span class="sc">~</span> <span class="dv">1</span><span class="sc">|</span>Harvest_ID,<span class="at">data=</span>mes)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(m1)</span></code></pre></div>
<pre><code>FALSE               numDF denDF   F-value p-value
FALSE (Intercept)       1    84  0.000295  0.9863
FALSE Batch             1    84 14.471512  0.0003
FALSE Tissue            9    84  4.324336  0.0001
FALSE Status            1    10  4.864112  0.0520
FALSE Tissue:Status     9    84  1.250563  0.2762</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Diagnostic plot </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(m1)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/darkturquoise-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot the module eigengene against our traits of interest</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data=</span>mes, <span class="fu">aes</span>(<span class="at">x=</span>Tissue, <span class="at">y=</span>MEdarkturquoise, <span class="at">color=</span>status, <span class="at">shape=</span>Batch), <span class="at">position=</span>dodge) <span class="sc">+</span> <span class="fu">geom_boxplot</span>(<span class="at">data=</span>mes, <span class="fu">aes</span>(<span class="at">x=</span>Tissue, <span class="at">y=</span>MEdarkturquoise, <span class="at">color=</span>status), <span class="at">fill=</span><span class="cn">NA</span>, <span class="at">outlier.colour =</span> <span class="cn">NA</span>) <span class="sc">+</span> peri_theme <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;darkturquoise module eigengene vs status&quot;</span>)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/darkturquoise-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Not very convinced with the relationship with Status when accounting for batch and individual ID, though some tissues appear to be stronger than others - e.g. VMH and BSTm.</p>
<p>Let’s now plot the top hub gene. This gene is the one with the highest module membership score. UBA52 has a module membership score ~0.9</p>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="paleturquoise" class="section level3" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> paleturquoise</h3>
<p>This module is potentially associated with status.</p>
<pre><code>FALSE               numDF denDF  F-value p-value
FALSE (Intercept)       1    84  0.00000  1.0000
FALSE Batch             1    84 69.76363  &lt;.0001
FALSE Tissue            9    84  2.04425  0.0441
FALSE Status            1    10  6.13781  0.0327
FALSE Tissue:Status     9    84  0.91679  0.5149</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/pale%20turquoise-1.png" width="672" style="display: block; margin: auto;" /><img src="03_WGCNA_wholebrain_files/figure-html/pale%20turquoise-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The top hub gene for this module is KCTD5 with a MM score of ~0.95 <img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>whichModule<span class="ot">=</span><span class="st">&quot;paleturquoise&quot;</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>nTop<span class="ot">&lt;-</span><span class="dv">30</span> <span class="co"># I use all the genes in the module, but could limit to top n genes based on MM</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="co">#extract the expression of the top 30 genes and sort them based on module membership. </span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>modcol<span class="ot">=</span><span class="fu">paste</span>(<span class="st">&quot;MM.&quot;</span>,whichModule,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>module_genes<span class="ot">&lt;-</span> color[color<span class="sc">$</span>moduleColor<span class="sc">==</span>whichModule,]</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="fu">verboseScatterplot</span>(<span class="fu">abs</span>(module_genes[, modcol]),</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abs</span>(GS[GS<span class="sc">$</span>moduleColor<span class="sc">==</span>whichModule, <span class="st">&quot;GS.status&quot;</span>]),</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="at">xlab =</span> <span class="fu">paste</span>(<span class="st">&quot;Module Membership in&quot;</span>, whichModule, <span class="st">&quot;module&quot;</span>),</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="at">ylab =</span> <span class="st">&quot;Gene significance for status&quot;</span>,</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module membership vs. gene significance</span><span class="sc">\n</span><span class="st">&quot;</span>),</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="at">cex.main =</span> <span class="fl">1.2</span>, <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.axis =</span> <span class="fl">1.2</span>, <span class="at">col =</span> whichModule)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>sorted<span class="ot">=</span>color<span class="sc">$</span>gene[<span class="fu">order</span>(color[,modcol],<span class="at">decreasing=</span>T)]</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>hubs<span class="ot">=</span>sorted[<span class="dv">1</span><span class="sc">:</span>nTop]</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>hubs<span class="ot">=</span>datExpr0[,<span class="fu">match</span>(hubs, <span class="fu">colnames</span>(datExpr0))]</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="do">### Sort individual data based on the trait value</span></span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>expDesign<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">rownames</span>(key_brain), <span class="at">mean_T=</span>key_brain[,<span class="st">&quot;mean_T&quot;</span>], <span class="at">Status=</span>key_brain[,<span class="st">&quot;Status&quot;</span>])</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>expDesign<span class="ot">&lt;-</span> expDesign[<span class="fu">order</span>(expDesign<span class="sc">$</span>Status),]</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>ann_cols<span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Status=</span><span class="fu">c</span>(<span class="at">territorial=</span><span class="st">&quot;#414042&quot;</span>,<span class="at">floater=</span><span class="st">&quot;#E54849&quot;</span>))</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>hubs<span class="ot">&lt;-</span> hubs[<span class="fu">match</span>(<span class="fu">rownames</span>(expDesign), <span class="fu">rownames</span>(hubs)), ]</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">#original heatmap  </span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(hubs,<span class="at">scale=</span><span class="st">&quot;column&quot;</span>, <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,<span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,<span class="at">show_rownames =</span> <span class="cn">TRUE</span>, <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,<span class="at">border_color=</span><span class="cn">NA</span>,  <span class="at">main=</span><span class="fu">paste</span>(whichModule,<span class="st">&quot;module in brain&quot;</span>), <span class="at">annotation_row=</span>expDesign, <span class="at">annotation_colors=</span>ann_cols)</span></code></pre></div>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in paleturquoise module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0036444" class="uri">GO:0036444</a>
</td>
<td style="text-align:left;">
calcium import into the mitochondrion
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
10/11465
</td>
<td style="text-align:right;">
0.0010049
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
MAIP1/MCUR1/MICU2/MICU3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043484" class="uri">GO:0043484</a>
</td>
<td style="text-align:left;">
regulation of RNA splicing
</td>
<td style="text-align:left;">
7/571
</td>
<td style="text-align:left;">
36/11465
</td>
<td style="text-align:right;">
0.0017305
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
CELF1/LOC114001870/MBNL2/PTBP2/RBM12/RBM12B/TRA2B
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030148" class="uri">GO:0030148</a>
</td>
<td style="text-align:left;">
sphingolipid biosynthetic process
</td>
<td style="text-align:left;">
8/571
</td>
<td style="text-align:left;">
48/11465
</td>
<td style="text-align:right;">
0.0023291
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
ACER3/ELOVL4/HACD2/KDSR/SGMS2/SGPP1/SPTLC1/VAPA
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006376" class="uri">GO:0006376</a>
</td>
<td style="text-align:left;">
mRNA splice site selection
</td>
<td style="text-align:left;">
5/571
</td>
<td style="text-align:left;">
20/11465
</td>
<td style="text-align:right;">
0.0025007
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
CELF1/LOC113985802/LOC113985815/PTBP2/SRSF1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044257" class="uri">GO:0044257</a>
</td>
<td style="text-align:left;">
cellular protein catabolic process
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
13/11465
</td>
<td style="text-align:right;">
0.0030357
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
CLN8/IDE/RAB12/RIPK1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043161" class="uri">GO:0043161</a>
</td>
<td style="text-align:left;">
proteasome-mediated ubiquitin-dependent protein catabolic process
</td>
<td style="text-align:left;">
16/571
</td>
<td style="text-align:left;">
149/11465
</td>
<td style="text-align:right;">
0.0030399
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
ATXN3/CLOCK/GID8/KCTD5/NEDD4/NHLRC1/NHLRC3/PPP2R5C/PSMD10/RAD23B/RNF4/SPOPL/TBL1XR1/UBE2D3/UBE2W/USP44
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006851" class="uri">GO:0006851</a>
</td>
<td style="text-align:left;">
mitochondrial calcium ion transmembrane transport
</td>
<td style="text-align:left;">
5/571
</td>
<td style="text-align:left;">
23/11465
</td>
<td style="text-align:right;">
0.0047967
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
LETM1/MAIP1/MCUR1/MICU2/MICU3
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031398" class="uri">GO:0031398</a>
</td>
<td style="text-align:left;">
positive regulation of protein ubiquitination
</td>
<td style="text-align:left;">
9/571
</td>
<td style="text-align:left;">
66/11465
</td>
<td style="text-align:right;">
0.0051804
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
BCL10/CHFR/DERL1/MAPK9/NDFIP2/NHLRC1/PSMD10/TRAF6/UBE2D1
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002756" class="uri">GO:0002756</a>
</td>
<td style="text-align:left;">
MyD88-independent toll-like receptor signaling pathway
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
15/11465
</td>
<td style="text-align:right;">
0.0053529
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
RIPK1/TLR3/UBE2D1/UBE2D3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051560" class="uri">GO:0051560</a>
</td>
<td style="text-align:left;">
mitochondrial calcium ion homeostasis
</td>
<td style="text-align:left;">
4/571
</td>
<td style="text-align:left;">
16/11465
</td>
<td style="text-align:right;">
0.0068600
</td>
<td style="text-align:right;">
0.6074856
</td>
<td style="text-align:right;">
0.6007034
</td>
<td style="text-align:left;">
LETM1/MAIP1/MICU2/MICU3
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><img src="03_WGCNA_wholebrain_files/figure-html/paleturqoise-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="honeydew1-module" class="section level3" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> honeydew1 module</h3>
<p>This module is potentially associated with social network strength.</p>
<pre><code>FALSE                           numDF denDF  F-value p-value
FALSE (Intercept)                   1    84  0.00693  0.9339
FALSE Batch                         1    84 24.58860  &lt;.0001
FALSE Tissue                        9    84 45.74681  &lt;.0001
FALSE strength.all_study            1    10  1.28478  0.2835
FALSE Tissue:strength.all_study     9    84  0.75868  0.6544</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/honeydew1-1.png" width="672" style="display: block; margin: auto;" /><img src="03_WGCNA_wholebrain_files/figure-html/honeydew1-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Not significant after controlling for batch, tissue and indivdual as a random effect.</p>
<p><br></p>
</div>
<div id="darkorange2" class="section level3" number="2.6.4">
<h3><span class="header-section-number">2.6.4</span> darkorange2</h3>
<pre><code>FALSE                           numDF denDF  F-value p-value
FALSE (Intercept)                   1    84  0.00718  0.9327
FALSE Batch                         1    84 38.02610  &lt;.0001
FALSE Tissue                        9    84 60.78957  &lt;.0001
FALSE strength.all_study            1    10  2.17888  0.1707
FALSE Tissue:strength.all_study     9    84  2.24083  0.0269</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="navajowhite2" class="section level3" number="2.6.5">
<h3><span class="header-section-number">2.6.5</span> navajowhite2</h3>
<p>This module is interesting because it’s associated with many sex-steroid related genes in the hypothalamic regions.</p>
<pre><code>FALSE               numDF denDF  F-value p-value
FALSE (Intercept)       1    84   0.0008  0.9770
FALSE Batch             1    84   8.6451  0.0042
FALSE Tissue            9    84 407.7426  &lt;.0001
FALSE Status            1    10   0.5609  0.4712
FALSE Tissue:Status     9    84   0.2616  0.9830</code></pre>
<p><img src="03_WGCNA_wholebrain_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This gene ontology analysis compared the genes in the navajowhite module against all other genes in the dataset.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in navajowhite2 module
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
70/679
</td>
<td style="text-align:left;">
439/11465
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
ADCY7/ADGRD1/ADGRG2/ADRA2A/AGTR1/ANXA1/CALCB/CASR/CMKLR1/CXCR4/FFAR4/FZD5/GAL/GALR3/GCGR/GHRH/GHSR/GLP1R/GNRH1/GPR135/GPR139/GPR176/GPR75/GPR83/HCRT/HEBP1/IAPP/LGR6/LOC113982931/LOC113983498/LOC113983511/LOC113987733/LOC113987774/LOC113989201/LOC113991146/LOC113993563/LOC113995733/LOC113996140/LOC114000503/LOC114001074/LOC114004322/MC3R/MCHR1/NMS/NMU/NPBWR2/NPR1/NPW/NTSR1/OBSCN/OPRK1/OPRL1/PDE3B/PDE7B/PDE8B/PDYN/PENK/PNOC/POMC/PRLH/PROK2/QRFP/RGS2/RORB/RXFP3/S1PR3/SSTR5/TRH/TRHR/TSHR
</td>
<td style="text-align:right;">
70
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007218" class="uri">GO:0007218</a>
</td>
<td style="text-align:left;">
neuropeptide signaling pathway
</td>
<td style="text-align:left;">
27/679
</td>
<td style="text-align:left;">
83/11465
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:left;">
CALCB/GAL/GALR3/GPR139/GPR83/HCRT/LOC113987733/LOC113995733/LOC114000503/MCHR1/NMS/NMU/NPBWR2/NPVF/NPW/NTSR1/OPRK1/OPRL1/PDYN/PENK/PNOC/POMC/PROK2/QRFP/RXFP3/SSTR5/TENM1
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007631" class="uri">GO:0007631</a>
</td>
<td style="text-align:left;">
feeding behavior
</td>
<td style="text-align:left;">
10/679
</td>
<td style="text-align:left;">
21/11465
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
0.0000516
</td>
<td style="text-align:right;">
0.0000494
</td>
<td style="text-align:left;">
CALCB/GAL/GALR3/LOC113987774/LOC114000503/MCHR1/MRAP2/NEGR1/NPW/PRLH
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007204" class="uri">GO:0007204</a>
</td>
<td style="text-align:left;">
positive regulation of cytosolic calcium ion concentration
</td>
<td style="text-align:left;">
18/679
</td>
<td style="text-align:left;">
88/11465
</td>
<td style="text-align:right;">
0.0000031
</td>
<td style="text-align:right;">
0.0012417
</td>
<td style="text-align:right;">
0.0011896
</td>
<td style="text-align:left;">
AGTR1/CALCB/CD36/CMKLR1/CXCR4/ESR1/FFAR4/GLP1R/HCRT/IAPP/LOC113983498/LOC113983511/LOC113991146/MCHR1/NMU/PDGFRA/PROK2/S1PR3
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007200" class="uri">GO:0007200</a>
</td>
<td style="text-align:left;">
phospholipase C-activating G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
12/679
</td>
<td style="text-align:left;">
45/11465
</td>
<td style="text-align:right;">
0.0000080
</td>
<td style="text-align:right;">
0.0025483
</td>
<td style="text-align:right;">
0.0024413
</td>
<td style="text-align:left;">
AGTR1/CASR/CMKLR1/ESR1/FFAR4/GPR139/HCRT/LOC113991146/LOC114001074/MC3R/OPRK1/TRHR
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048704" class="uri">GO:0048704</a>
</td>
<td style="text-align:left;">
embryonic skeletal system morphogenesis
</td>
<td style="text-align:left;">
6/679
</td>
<td style="text-align:left;">
11/11465
</td>
<td style="text-align:right;">
0.0000151
</td>
<td style="text-align:right;">
0.0040286
</td>
<td style="text-align:right;">
0.0038594
</td>
<td style="text-align:left;">
LOC113998261/LOC114001942/PCGF2/PDGFRA/SATB2/SOX11
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006112" class="uri">GO:0006112</a>
</td>
<td style="text-align:left;">
energy reserve metabolic process
</td>
<td style="text-align:left;">
6/679
</td>
<td style="text-align:left;">
13/11465
</td>
<td style="text-align:right;">
0.0000506
</td>
<td style="text-align:right;">
0.0115740
</td>
<td style="text-align:right;">
0.0110881
</td>
<td style="text-align:left;">
LOC113985008/LOC113987774/LOC113999007/MRAP2/MYC/PRLH
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007189" class="uri">GO:0007189</a>
</td>
<td style="text-align:left;">
adenylate cyclase-activating G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
14/679
</td>
<td style="text-align:left;">
76/11465
</td>
<td style="text-align:right;">
0.0001278
</td>
<td style="text-align:right;">
0.0250877
</td>
<td style="text-align:right;">
0.0240344
</td>
<td style="text-align:left;">
ADCY7/ADGRD1/ADGRG2/ADRA2A/CALCB/GCGR/GHRH/GLP1R/IAPP/LOC113991146/LOC113997521/MC3R/S1PR3/TSHR
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
5/679
</td>
<td style="text-align:left;">
10/11465
</td>
<td style="text-align:right;">
0.0001409
</td>
<td style="text-align:right;">
0.0250877
</td>
<td style="text-align:right;">
0.0240344
</td>
<td style="text-align:left;">
AR/ESR1/ESR2/NR3C1/PGR
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060412" class="uri">GO:0060412</a>
</td>
<td style="text-align:left;">
ventricular septum morphogenesis
</td>
<td style="text-align:left;">
9/679
</td>
<td style="text-align:left;">
36/11465
</td>
<td style="text-align:right;">
0.0001883
</td>
<td style="text-align:right;">
0.0281013
</td>
<td style="text-align:right;">
0.0269214
</td>
<td style="text-align:left;">
CITED2/LOC113994594/LOC113997556/LOC114001848/PROX1/SLIT2/SMAD7/SOX11/TGFB2
</td>
<td style="text-align:right;">
9
</td>
</tr>
</tbody>
</table>
<p><img src="03_WGCNA_wholebrain_files/figure-html/navajowhite-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
