<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Last Substantive Change April 2021" />


<title>WGCNA Analyses Within-Tissues</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./01_reading_QC.html">Supp 1: Data QC and filtering</a>
    </li>
    <li>
      <a href="./02_1_Checks_PCA.html">Supp 1 &amp; 2: Checks and System-wide PCA</a>
    </li>
    <li>
      <a href="./02_2_WGCNA_wholebrain.html">Supp 2: Whole-brain WGCNA</a>
    </li>
    <li>
      <a href="./02_3_WGCNA_HPG.html">Supp 2: HPG WGCNA</a>
    </li>
    <li>
      <a href="./03_DESeq2.html">Supp 3: Per Tissue Differential Expression</a>
    </li>
    <li>
      <a href="./04_candidate_gene_analysis.html">Supp 4: Candidate Gene Analysis</a>
    </li>
    <li>
      <a href="./05_WGCNA_single_tissue.html">Supp 5: Within Tissue WGCNA </a>
    </li>
    <li>
      <a href="./06_tissue_summary_similarity.html">Supp 6: Summaries and overlap analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WGCNA Analyses Within-Tissues</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male
cooperative behavior in a wild bird</h3>
<h4 class="author">Last Substantive Change April 2021</h4>
<h4 class="date">Last Knit “2024-01-06”</h4>

</div>


<pre><code>## Warning: package &#39;network&#39; was built under R version 4.0.3</code></pre>
<pre><code>## network: Classes for Relational Data
## Version 1.16.1 created on 2020-10-06.
## copyright (c) 2005, Carter T. Butts, University of California-Irvine
##                     Mark S. Handcock, University of California -- Los Angeles
##                     David R. Hunter, Penn State University
##                     Martina Morris, University of Washington
##                     Skye Bender-deMoll, University of Washington
##  For citation information, type citation(&quot;network&quot;).
##  Type help(&quot;network-package&quot;) to get started.</code></pre>
<pre><code>## 
## Attaching package: &#39;network&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:igraph&#39;:
## 
##     %c%, %s%, add.edges, add.vertices, delete.edges, delete.vertices,
##     get.edge.attribute, get.edges, get.vertex.attribute, is.bipartite,
##     is.directed, list.edge.attributes, list.vertex.attributes,
##     set.edge.attribute, set.vertex.attribute</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     is.discrete</code></pre>
<pre><code>## Loading required package: BiocFileCache</code></pre>
<pre><code>## Loading required package: dbplyr</code></pre>
<pre><code>## Warning: package &#39;dbplyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>## 
## Attaching package: &#39;dbplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     ident, sql</code></pre>
<pre><code>## 
## Attaching package: &#39;AnnotationHub&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     cache</code></pre>
<pre><code>## 
## Attaching package: &#39;enrichplot&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:GGally&#39;:
## 
##     ggtable</code></pre>
<pre><code>## The following object is masked from &#39;package:ggpubr&#39;:
## 
##     color_palette</code></pre>
<pre><code>## The following object is masked from &#39;package:lattice&#39;:
## 
##     dotplot</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_key_Parsed_ReplicatesRemoved.csv&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>behav<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_unfiltered/PIPFIL_T_and_behav_data.csv&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">rownames</span>(key)<span class="ot">&lt;-</span> key<span class="sc">$</span>X</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>key<span class="sc">$</span>Color_ID<span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;/&quot;</span>,<span class="st">&quot;&quot;</span>, key<span class="sc">$</span>Color_ID)</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;Color_ID&quot;</span><span class="ot">=</span><span class="st">&quot;colorID&quot;</span>))</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>behav<span class="sc">$</span>status<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(behav<span class="sc">$</span>status, <span class="fu">c</span>(<span class="st">&quot;floa&quot;</span><span class="ot">=</span><span class="st">&quot;floater&quot;</span>, <span class="st">&quot;terr&quot;</span><span class="ot">=</span><span class="st">&quot;territorial&quot;</span>))</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">merge</span>(key, behav, <span class="at">by=</span><span class="st">&quot;colorID&quot;</span>)</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="sc">!</span><span class="fu">is.na</span>(key_behav<span class="sc">$</span>last_behav),]</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co">#create a data.frame with all of the observations. </span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> key[<span class="sc">!</span>key<span class="sc">$</span>X <span class="sc">%in%</span> key_behav<span class="sc">$</span>X,]</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>cols_not_key<span class="ot">&lt;-</span> <span class="fu">colnames</span>(key_behav)[<span class="sc">!</span><span class="fu">colnames</span>(key_behav) <span class="sc">%in%</span> <span class="fu">colnames</span>(key)]</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a>cols_not_key_df<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(not_in_behav), <span class="at">ncol =</span> <span class="fu">length</span>(cols_not_key)))</span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a><span class="fu">colnames</span>(cols_not_key_df)<span class="ot">&lt;-</span> cols_not_key</span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a>not_in_behav<span class="ot">&lt;-</span> <span class="fu">cbind</span>(not_in_behav, cols_not_key_df)</span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">rbind</span>(key_behav, not_in_behav)</span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>X</span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, <span class="at">select=</span><span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Tissue&quot;</span>, <span class="st">&quot;Batch&quot;</span>, <span class="st">&quot;Status&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;mean_T&quot;</span>, <span class="st">&quot;strength.all_study&quot;</span>))</span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a><span class="co">#key_behav$Class&lt;- word(key_behav$Class,1)</span></span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a>key_behav<span class="sc">$</span>Tissue<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Tissue, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;GON&quot;</span>,<span class="st">&quot;PIT&quot;</span>,<span class="st">&quot;VMH&quot;</span>,<span class="st">&quot;AH&quot;</span>,<span class="st">&quot;PVN&quot;</span>,<span class="st">&quot;POM&quot;</span>,<span class="st">&quot;ICO&quot;</span>,<span class="st">&quot;GCT&quot;</span>,<span class="st">&quot;AI&quot;</span>,<span class="st">&quot;TNA&quot;</span>,<span class="st">&quot;LS&quot;</span>, <span class="st">&quot;BSTm&quot;</span>))</span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(<span class="fu">rownames</span>(key_behav)),]</span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(key_behav<span class="sc">$</span>Class, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;SCB floater&quot;</span><span class="ot">=</span><span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;DCB floater &quot;</span><span class="ot">=</span><span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;DCB territorial&quot;</span><span class="ot">=</span><span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb17-28"><a href="#cb17-28" tabindex="-1"></a>key_behav<span class="sc">$</span>Class<span class="ot">&lt;-</span> <span class="fu">factor</span>(key_behav<span class="sc">$</span>Class, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Predefinitive floater&quot;</span>, <span class="st">&quot;Definitive floater&quot;</span>, <span class="st">&quot;Territorial&quot;</span>))</span>
<span id="cb17-29"><a href="#cb17-29" tabindex="-1"></a><span class="fu">rownames</span>(key_behav)<span class="ot">&lt;-</span> key_behav<span class="sc">$</span>sampleID</span>
<span id="cb17-30"><a href="#cb17-30" tabindex="-1"></a>key_behav<span class="ot">&lt;-</span> key_behav[<span class="fu">order</span>(key_behav<span class="sc">$</span>sampleID),]</span>
<span id="cb17-31"><a href="#cb17-31" tabindex="-1"></a></span>
<span id="cb17-32"><a href="#cb17-32" tabindex="-1"></a></span>
<span id="cb17-33"><a href="#cb17-33" tabindex="-1"></a><span class="co">#read the raw count data</span></span>
<span id="cb17-34"><a href="#cb17-34" tabindex="-1"></a>data<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../data_filtered/data_RawCounts_all_ReplicatesRemoved_antisense_V2.csv&quot;</span>)</span>
<span id="cb17-35"><a href="#cb17-35" tabindex="-1"></a></span>
<span id="cb17-36"><a href="#cb17-36" tabindex="-1"></a><span class="fu">rownames</span>(data)<span class="ot">&lt;-</span> data<span class="sc">$</span>X</span>
<span id="cb17-37"><a href="#cb17-37" tabindex="-1"></a>data<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-38"><a href="#cb17-38" tabindex="-1"></a></span>
<span id="cb17-39"><a href="#cb17-39" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/Maggies_annotations_modifiedR.csv&quot;</span>)</span>
<span id="cb17-40"><a href="#cb17-40" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(genes_key, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb17-41"><a href="#cb17-41" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" tabindex="-1"></a>data_genes<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span><span class="fu">rownames</span>(data))</span>
<span id="cb17-43"><a href="#cb17-43" tabindex="-1"></a>genes_key<span class="ot">&lt;-</span> <span class="fu">merge</span>(data_genes, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-44"><a href="#cb17-44" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" tabindex="-1"></a>genes_key<span class="sc">$</span>display_gene_ID<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(genes_key<span class="sc">$</span>best_anno), <span class="fu">as.character</span>(genes_key<span class="sc">$</span>gene), <span class="fu">ifelse</span>(genes_key<span class="sc">$</span>best_anno<span class="sc">!=</span>genes_key<span class="sc">$</span>gene, <span class="fu">paste0</span>(genes_key<span class="sc">$</span>gene,<span class="st">&quot;/&quot;</span>, genes_key<span class="sc">$</span>best_anno), <span class="fu">as.character</span>(genes_key<span class="sc">$</span>best_anno))) </span>
<span id="cb17-46"><a href="#cb17-46" tabindex="-1"></a>  </span>
<span id="cb17-47"><a href="#cb17-47" tabindex="-1"></a>  </span>
<span id="cb17-48"><a href="#cb17-48" tabindex="-1"></a></span>
<span id="cb17-49"><a href="#cb17-49" tabindex="-1"></a>                                </span>
<span id="cb17-50"><a href="#cb17-50" tabindex="-1"></a></span>
<span id="cb17-51"><a href="#cb17-51" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/pfil_GO_key_raw.csv&quot;</span>)</span>
<span id="cb17-52"><a href="#cb17-52" tabindex="-1"></a>go_terms<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_terms, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span><span class="ot">=</span><span class="st">&quot;gene&quot;</span>))</span>
<span id="cb17-53"><a href="#cb17-53" tabindex="-1"></a>go2gene_bp<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;P&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb17-54"><a href="#cb17-54" tabindex="-1"></a>go2gene_mf<span class="ot">&lt;-</span> go_terms[<span class="fu">which</span>(go_terms<span class="sc">$</span>Aspect<span class="sc">==</span><span class="st">&quot;F&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;gene&quot;</span>)]</span>
<span id="cb17-55"><a href="#cb17-55" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../GO_annotations/ontology_obo_out.csv&quot;</span>)</span>
<span id="cb17-56"><a href="#cb17-56" tabindex="-1"></a>go_obo<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">rename</span>(go_obo, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;id&quot;</span><span class="ot">=</span><span class="st">&quot;GO_ID&quot;</span>))</span>
<span id="cb17-57"><a href="#cb17-57" tabindex="-1"></a>go2name_bp<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;biological_process&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span>
<span id="cb17-58"><a href="#cb17-58" tabindex="-1"></a>go2name_mf<span class="ot">&lt;-</span> go_obo[<span class="fu">which</span>(go_obo<span class="sc">$</span>namespace<span class="sc">==</span><span class="st">&quot;molecular_function&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;GO_ID&quot;</span>, <span class="st">&quot;name&quot;</span>)]</span></code></pre></div>
<p><br><br></p>
<div id="gonads-gon" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Gonads (GON)</h1>
<p>Like the DESeq2 document, the Gonads section will contain the code
used throughout the the analysis to achieve the full set of results.
First, each dataset is filtered conservatively as per the DESeq2
workflow, and then any batch effects identified in the DEseq workflow
are removed here. Then we create a midweight bicorrelation weighted
signed network using the recommended options. The Topological Overlap
Matrix is also signed.</p>
<p><br></p>
<div id="data-quality-control" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Data quality
control</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>tissue<span class="ot">&lt;-</span> <span class="st">&quot;GON&quot;</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">subset</span>(key_behav, Tissue<span class="sc">==</span>tissue)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>gon_key<span class="ot">&lt;-</span> <span class="fu">droplevels</span>(gon_key)</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> data[,<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(gon_key)]</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>start<span class="ot">&lt;-</span> <span class="fu">nrow</span>(gon_data)</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#remove genes with less than 5 reads</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, mean)</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>avg_count<span class="sc">&gt;</span><span class="dv">5</span>,]</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>gon_data<span class="sc">$</span>avg_count<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb18-13"><a href="#cb18-13" tabindex="-1"></a><span class="co">#gon_data$mad&lt;- apply(gon_data,1,mad)</span></span>
<span id="cb18-14"><a href="#cb18-14" tabindex="-1"></a><span class="co">#gon_data$var&lt;- apply(gon_data, 1, var) # could filter by variance rather than MAD maybe &lt;0.2 but these filtering methods all seem to achieve this. </span></span>
<span id="cb18-15"><a href="#cb18-15" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" tabindex="-1"></a><span class="co">#remove genes where &gt;50% of samples have 0 gene expression</span></span>
<span id="cb18-17"><a href="#cb18-17" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span> <span class="fu">apply</span>(gon_data, <span class="dv">1</span>, <span class="cf">function</span>(x)<span class="fu">length</span>(x[x<span class="sc">==</span><span class="dv">0</span>]))</span>
<span id="cb18-18"><a href="#cb18-18" tabindex="-1"></a>thresh<span class="ot">&lt;-</span> <span class="fu">ncol</span>(gon_data)<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb18-19"><a href="#cb18-19" tabindex="-1"></a>gon_data<span class="ot">&lt;-</span> gon_data[gon_data<span class="sc">$</span>percent_0<span class="sc">&lt;=</span>thresh,]</span>
<span id="cb18-20"><a href="#cb18-20" tabindex="-1"></a>gon_data<span class="sc">$</span>percent_0<span class="ot">&lt;-</span><span class="cn">NULL</span></span>
<span id="cb18-21"><a href="#cb18-21" tabindex="-1"></a><span class="fu">nrow</span>(gon_data)</span></code></pre></div>
<pre><code>## [1] 16169</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>gon_data, <span class="at">colData=</span>gon_key, <span class="at">design=</span> <span class="sc">~</span> Batch)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>dd<span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dd)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 175 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>dd<span class="ot">&lt;-</span> dd[<span class="fu">which</span>(<span class="fu">mcols</span>(dd)<span class="sc">$</span>betaConv),]</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>vsd_data<span class="ot">&lt;-</span> <span class="fu">getVarianceStabilizedData</span>(dd)</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>mat<span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">removeBatchEffect</span>(vsd_data, gon_key<span class="sc">$</span>Batch)</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(mat))</span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>gsg <span class="ot">=</span> <span class="fu">goodSamplesGenes</span>(datExpr0, <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##  Flagging genes and samples with too many missing values...
##   ..step 1</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="co">#write.csv(datExpr0, file=&quot;GON_vsd_batch_rm.csv&quot;)</span></span></code></pre></div>
<p><br></p>
</div>
<div id="soft-threshold-selection" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Soft-threshold
selection</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>powers<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">11</span>, <span class="at">by =</span> <span class="dv">1</span>), <span class="fu">seq</span>(<span class="dv">12</span>, <span class="dv">26</span>, <span class="at">by =</span> <span class="dv">2</span>))</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>sft<span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(datExpr0, <span class="at">powerVector=</span>powers, <span class="at">verbose=</span><span class="dv">0</span>, <span class="at">networkType=</span><span class="st">&quot;signed&quot;</span>)</span></code></pre></div>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0406 16.20          0.920  8090.0   8090.00   8330
## 2      2   0.0284 -3.75          0.684  4440.0   4410.00   4990
## 3      3   0.3020 -5.40          0.739  2610.0   2570.00   3360
## 4      4   0.6060 -4.84          0.877  1630.0   1580.00   2450
## 5      5   0.7690 -4.13          0.946  1070.0   1020.00   1870
## 6      6   0.8240 -3.56          0.972   726.0    675.00   1490
## 7      7   0.8620 -3.20          0.987   511.0    462.00   1230
## 8      8   0.8840 -2.95          0.994   370.0    324.00   1030
## 9      9   0.8980 -2.75          0.996   275.0    232.00    880
## 10    10   0.9070 -2.60          0.995   209.0    169.00    762
## 11    11   0.9090 -2.49          0.992   162.0    125.00    668
## 12    12   0.9090 -2.40          0.987   128.0     93.40    591
## 13    14   0.9020 -2.26          0.977    82.9     54.60    474
## 14    16   0.8920 -2.16          0.967    56.6     33.10    389
## 15    18   0.8740 -2.07          0.949    40.3     20.80    325
## 16    20   0.8780 -1.97          0.946    29.7     13.50    276
## 17    22   0.8660 -1.90          0.933    22.5      8.97    237
## 18    24   0.8510 -1.84          0.913    17.4      6.12    206
## 19    26   0.8540 -1.76          0.909    13.8      4.28    181</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft power selected will be the first power that that reaches
scale free topology.</p>
<p><br></p>
</div>
<div id="adjacency-matrix" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Adjacency Matrix</h2>
<p>The adjacency matrix was run on an HPC to speed up run time and
prevent overloading my personal computer.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>softPower<span class="ot">=</span><span class="dv">10</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>datExpr0<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;GON_vsd_batch_rm.csv&quot;</span>)</span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a><span class="fu">rownames</span>(datExpr0)<span class="ot">&lt;-</span> datExpr0<span class="sc">$</span>X</span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a>datExpr0<span class="sc">$</span>X<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a>adjacency<span class="ot">&lt;-</span> <span class="fu">adjacency</span>(datExpr0, <span class="at">power =</span> softPower, <span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a>TOM<span class="ot">&lt;-</span> <span class="fu">TOMsimilarity</span>(adjacency, <span class="at">TOMtype=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a><span class="co">#dissTOM&lt;- 1-TOM</span></span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a><span class="fu">save</span>(adjacency, dissTOM, <span class="at">file=</span><span class="st">&quot;GON_network.RData&quot;</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="identify-co-expression-modules" class="section level2"
number="1.4">
<h2><span class="header-section-number">1.4</span> Identify
co-expression modules</h2>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;../WGCNA_results/within_tissues/GON_network.RData&quot;</span>)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>dissTOM<span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>TOM</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>geneTree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(dissTOM), <span class="at">method=</span><span class="st">&quot;average&quot;</span>)</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a></span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a><span class="co">#plot(geneTree, xlab=&quot;&quot;, sub=&quot;&quot;, main= &quot;Gene Clustering on TOM-based dissimilarity&quot;, labels= FALSE, hang=0.04)</span></span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>minModuleSize<span class="ot">&lt;-</span><span class="dv">30</span> </span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>dynamicMods<span class="ot">&lt;-</span><span class="fu">cutreeDynamic</span>(<span class="at">dendro=</span> geneTree, <span class="at">distM=</span> dissTOM, <span class="at">deepSplit=</span><span class="dv">2</span>, <span class="at">pamRespectsDendro=</span> <span class="cn">FALSE</span>, <span class="at">minClusterSize=</span> minModuleSize)</span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a><span class="co">#table(dynamicMods)</span></span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>dynamicColors<span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span> <span class="fl">0.05</span>, <span class="at">main=</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="co">#-----Merge modules whose expression profiles are very similar</span></span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>MEList<span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, <span class="at">colors=</span> dynamicColors)</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>MEs<span class="ot">=</span> MEList<span class="sc">$</span>eigengenes</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a><span class="co">#Calculate dissimilarity of module eigenegenes</span></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>MEDiss<span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(MEs)</span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a><span class="co">#Cluster module eigengenes</span></span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>METree<span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(MEDiss), <span class="at">method=</span> <span class="st">&quot;average&quot;</span>)</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>##  mergeCloseModules: Merging modules whose distance is less than 0.3
##    multiSetMEs: Calculating module MEs.
##      Working on set 1 ...
##      moduleEigengenes: Calculating 17 module eigengenes in given set.
##    multiSetMEs: Calculating module MEs.
##      Working on set 1 ...
##      moduleEigengenes: Calculating 11 module eigengenes in given set.
##    multiSetMEs: Calculating module MEs.
##      Working on set 1 ...
##      moduleEigengenes: Calculating 10 module eigengenes in given set.
##    Calculating new MEs...
##    multiSetMEs: Calculating module MEs.
##      Working on set 1 ...
##      moduleEigengenes: Calculating 10 module eigengenes in given set.</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(geneTree, <span class="fu">cbind</span>(dynamicColors, mergedColors), <span class="fu">c</span>(<span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="st">&quot;Merged dynamic&quot;</span>), <span class="at">dendroLabels=</span> <span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span> <span class="cn">TRUE</span>, <span class="at">guideHang=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a>moduleColors<span class="ot">=</span> mergedColors</span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a>colorOrder<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>, <span class="fu">standardColors</span>(<span class="dv">50</span>))</span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a>moduleLabels<span class="ot">=</span> <span class="fu">match</span>(moduleColors, colorOrder)<span class="sc">-</span><span class="dv">1</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a>MEs<span class="ot">=</span>mergedMEs</span></code></pre></div>
<p><br></p>
</div>
<div id="module-trait-correlation" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Module-trait
correlation</h2>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> gon_key</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a><span class="fu">all.equal</span>(<span class="fu">rownames</span>(datExpr0), <span class="fu">rownames</span>(datTraits))</span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a></span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>datTraits<span class="sc">$</span>sampleID<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.factor</span>(datTraits<span class="sc">$</span>sampleID))</span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>datTraits<span class="sc">$</span>Status2<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(datTraits<span class="sc">$</span>Status<span class="sc">==</span><span class="st">&quot;territorial&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a></span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a>datTraits<span class="ot">&lt;-</span> <span class="fu">subset</span>(datTraits, <span class="at">select=</span><span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>,  <span class="st">&quot;Status2&quot;</span>, <span class="st">&quot;mean_T&quot;</span>, <span class="st">&quot;strength.all_study&quot;</span>))</span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a><span class="fu">names</span>(datTraits)[<span class="fu">names</span>(datTraits)<span class="sc">==</span><span class="st">&quot;Status2&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Status&quot;</span></span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a><span class="fu">names</span>(datTraits)[<span class="fu">names</span>(datTraits)<span class="sc">==</span><span class="st">&quot;sampleID&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Individual&quot;</span></span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a></span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a></span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a><span class="co">#-----Define numbers of genes and samples</span></span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a>nGenes <span class="ot">=</span> <span class="fu">ncol</span>(datExpr0);</span>
<span id="cb46-16"><a href="#cb46-16" tabindex="-1"></a>nSamples <span class="ot">=</span> <span class="fu">nrow</span>(datExpr0);</span>
<span id="cb46-17"><a href="#cb46-17" tabindex="-1"></a><span class="co">#-----Recalculate MEs with color labels</span></span>
<span id="cb46-18"><a href="#cb46-18" tabindex="-1"></a>MEs0 <span class="ot">=</span> <span class="fu">moduleEigengenes</span>(datExpr0, moduleColors)<span class="sc">$</span>eigengenes</span>
<span id="cb46-19"><a href="#cb46-19" tabindex="-1"></a>MEs <span class="ot">=</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb46-20"><a href="#cb46-20" tabindex="-1"></a></span>
<span id="cb46-21"><a href="#cb46-21" tabindex="-1"></a><span class="co">#-----Correlations of genes with eigengenes</span></span>
<span id="cb46-22"><a href="#cb46-22" tabindex="-1"></a>moduleGeneCor<span class="ot">=</span><span class="fu">cor</span>(MEs,datExpr0)</span>
<span id="cb46-23"><a href="#cb46-23" tabindex="-1"></a>moduleGenePvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleGeneCor, nSamples);</span>
<span id="cb46-24"><a href="#cb46-24" tabindex="-1"></a></span>
<span id="cb46-25"><a href="#cb46-25" tabindex="-1"></a>moduleTraitCor <span class="ot">=</span> <span class="fu">cor</span>(MEs, datTraits, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>);</span>
<span id="cb46-26"><a href="#cb46-26" tabindex="-1"></a>moduleTraitPvalue <span class="ot">=</span> <span class="fu">corPvalueStudent</span>(moduleTraitCor, nSamples);</span>
<span id="cb46-27"><a href="#cb46-27" tabindex="-1"></a></span>
<span id="cb46-28"><a href="#cb46-28" tabindex="-1"></a></span>
<span id="cb46-29"><a href="#cb46-29" tabindex="-1"></a>wgcnasum<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(moduleTraitPvalue)</span>
<span id="cb46-30"><a href="#cb46-30" tabindex="-1"></a></span>
<span id="cb46-31"><a href="#cb46-31" tabindex="-1"></a>wgcnasum<span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="at">Tissue=</span>tissue, </span>
<span id="cb46-32"><a href="#cb46-32" tabindex="-1"></a>                                    <span class="at">Individual=</span><span class="fu">nrow</span>(wgcnasum[wgcnasum<span class="sc">$</span>Individual<span class="sc">&lt;</span><span class="fl">0.05</span>,]),</span>
<span id="cb46-33"><a href="#cb46-33" tabindex="-1"></a>                                <span class="at">Status=</span><span class="fu">nrow</span>(wgcnasum[wgcnasum<span class="sc">$</span>Status<span class="sc">&lt;</span><span class="fl">0.05</span>,]),</span>
<span id="cb46-34"><a href="#cb46-34" tabindex="-1"></a>                                    <span class="at">mean_T=</span><span class="fu">nrow</span>(wgcnasum[wgcnasum<span class="sc">$</span>mean_T<span class="sc">&lt;</span><span class="fl">0.05</span>,]), <span class="at">strength=</span><span class="fu">nrow</span>(wgcnasum[wgcnasum<span class="sc">$</span>strength.all_study<span class="sc">&lt;</span><span class="fl">0.05</span>,]), </span>
<span id="cb46-35"><a href="#cb46-35" tabindex="-1"></a>                       <span class="at">status_meanT=</span><span class="fu">nrow</span>(wgcnasum[<span class="fu">which</span>(wgcnasum<span class="sc">$</span>Status<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> wgcnasum<span class="sc">$</span>mean_T<span class="sc">&lt;</span><span class="fl">0.05</span>),]),</span>
<span id="cb46-36"><a href="#cb46-36" tabindex="-1"></a>                       <span class="at">meanT_strength=</span><span class="fu">nrow</span>(wgcnasum[<span class="fu">which</span>(wgcnasum<span class="sc">$</span>mean_T<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> wgcnasum<span class="sc">$</span>strength.all_study<span class="sc">&lt;</span><span class="fl">0.05</span>),]),</span>
<span id="cb46-37"><a href="#cb46-37" tabindex="-1"></a>                       <span class="at">status_strength=</span><span class="fu">nrow</span>(wgcnasum[<span class="fu">which</span>(wgcnasum<span class="sc">$</span>Status<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> wgcnasum<span class="sc">$</span>strength.all_study<span class="sc">&lt;</span><span class="fl">0.05</span>),]),</span>
<span id="cb46-38"><a href="#cb46-38" tabindex="-1"></a>                       <span class="at">total_modules=</span><span class="fu">ncol</span>(MEs))</span>
<span id="cb46-39"><a href="#cb46-39" tabindex="-1"></a></span>
<span id="cb46-40"><a href="#cb46-40" tabindex="-1"></a>wgcna_summary<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb46-41"><a href="#cb46-41" tabindex="-1"></a>wgcna_summary[[tissue]]<span class="ot">&lt;-</span> wgcnasum</span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a><span class="co">#---------------------Module-trait heatmap</span></span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a></span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a>textMatrix <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">signif</span>(moduleTraitCor, <span class="dv">2</span>), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">(&quot;</span>,</span>
<span id="cb47-5"><a href="#cb47-5" tabindex="-1"></a>                   <span class="fu">signif</span>(moduleTraitPvalue, <span class="dv">1</span>), <span class="st">&quot;)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>);</span>
<span id="cb47-6"><a href="#cb47-6" tabindex="-1"></a><span class="fu">dim</span>(textMatrix) <span class="ot">=</span> <span class="fu">dim</span>(moduleTraitCor)</span>
<span id="cb47-7"><a href="#cb47-7" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="fl">8.5</span>, <span class="dv">3</span>, <span class="dv">3</span>));</span>
<span id="cb47-8"><a href="#cb47-8" tabindex="-1"></a><span class="co"># Display the correlation values within a heatmap plot</span></span>
<span id="cb47-9"><a href="#cb47-9" tabindex="-1"></a><span class="fu">labeledHeatmap</span>(<span class="at">Matrix =</span> moduleTraitCor,</span>
<span id="cb47-10"><a href="#cb47-10" tabindex="-1"></a>               <span class="at">xLabels =</span> <span class="fu">names</span>(datTraits),</span>
<span id="cb47-11"><a href="#cb47-11" tabindex="-1"></a>               <span class="at">yLabels =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb47-12"><a href="#cb47-12" tabindex="-1"></a>               <span class="at">ySymbols =</span> <span class="fu">names</span>(MEs),</span>
<span id="cb47-13"><a href="#cb47-13" tabindex="-1"></a>               <span class="at">colorLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-14"><a href="#cb47-14" tabindex="-1"></a>               <span class="at">colors =</span> <span class="fu">blueWhiteRed</span>(<span class="dv">50</span>),</span>
<span id="cb47-15"><a href="#cb47-15" tabindex="-1"></a>               <span class="at">textMatrix =</span> textMatrix,</span>
<span id="cb47-16"><a href="#cb47-16" tabindex="-1"></a>               <span class="at">setStdMargins =</span> <span class="cn">FALSE</span>,</span>
<span id="cb47-17"><a href="#cb47-17" tabindex="-1"></a>               <span class="at">cex.text =</span> <span class="fl">0.5</span>,</span>
<span id="cb47-18"><a href="#cb47-18" tabindex="-1"></a>               <span class="at">zlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb47-19"><a href="#cb47-19" tabindex="-1"></a>               <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module-trait relationships in&quot;</span>, tissue))</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GON%20heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in GON
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
1361
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
2584
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
2686
</td>
</tr>
<tr>
<td style="text-align:left;">
cyan
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
green
</td>
<td style="text-align:right;">
1431
</td>
</tr>
<tr>
<td style="text-align:left;">
grey60
</td>
<td style="text-align:right;">
215
</td>
</tr>
<tr>
<td style="text-align:left;">
midnightblue
</td>
<td style="text-align:right;">
1393
</td>
</tr>
<tr>
<td style="text-align:left;">
pink
</td>
<td style="text-align:right;">
590
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon
</td>
<td style="text-align:right;">
243
</td>
</tr>
<tr>
<td style="text-align:left;">
turquoise
</td>
<td style="text-align:right;">
5526
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="module-membership-and-gene-significance" class="section level2"
number="1.6">
<h2><span class="header-section-number">1.6</span> Module Membership and
Gene Significance</h2>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="do">#### module membership</span></span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a></span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>datME<span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(datExpr0,mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a>datKME<span class="ot">&lt;-</span> <span class="fu">signedKME</span>(datExpr0, datME, <span class="at">outputColumnName=</span><span class="st">&quot;MM.&quot;</span>) <span class="co">#use the &quot;signed eigennode connectivity&quot; or module membership</span></span>
<span id="cb48-5"><a href="#cb48-5" tabindex="-1"></a></span>
<span id="cb48-6"><a href="#cb48-6" tabindex="-1"></a>MMPvalue <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(datKME), nSamples)) <span class="co"># Calculate module membership P-values</span></span>
<span id="cb48-7"><a href="#cb48-7" tabindex="-1"></a>datKME<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(datKME)</span>
<span id="cb48-8"><a href="#cb48-8" tabindex="-1"></a>MMPvalue<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(MMPvalue)</span>
<span id="cb48-9"><a href="#cb48-9" tabindex="-1"></a>genes<span class="ot">=</span><span class="fu">names</span>(datExpr0)</span>
<span id="cb48-10"><a href="#cb48-10" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>genes,<span class="at">moduleColor=</span>moduleColors)</span>
<span id="cb48-11"><a href="#cb48-11" tabindex="-1"></a>geneInfo0 <span class="ot">&lt;-</span>  <span class="fu">merge</span>(geneInfo0, genes_key, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-12"><a href="#cb48-12" tabindex="-1"></a>color<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, datKME, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>) <span class="co">#these are from your original WGCNA analysis </span></span>
<span id="cb48-13"><a href="#cb48-13" tabindex="-1"></a><span class="co">#head(color)</span></span>
<span id="cb48-14"><a href="#cb48-14" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">as.data.frame</span>(color), <span class="at">file =</span> <span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/within_tissues/&quot;</span>,tissue,<span class="st">&quot;results_ModuleMembership.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb48-15"><a href="#cb48-15" tabindex="-1"></a>MMPvalue<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0, MMPvalue, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb48-16"><a href="#cb48-16" tabindex="-1"></a><span class="fu">write.csv</span>(MMPvalue, <span class="at">file=</span><span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/within_tissues/&quot;</span>,tissue,<span class="st">&quot;_results_ModuleMembership_P-value.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb48-17"><a href="#cb48-17" tabindex="-1"></a></span>
<span id="cb48-18"><a href="#cb48-18" tabindex="-1"></a><span class="do">#### gene-significance</span></span>
<span id="cb48-19"><a href="#cb48-19" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>Status) <span class="co">#change here for traits of interest</span></span>
<span id="cb48-20"><a href="#cb48-20" tabindex="-1"></a><span class="fu">names</span>(trait) <span class="ot">=</span> <span class="st">&quot;status&quot;</span> <span class="co">#change here for traits of interest </span></span>
<span id="cb48-21"><a href="#cb48-21" tabindex="-1"></a>modNames <span class="ot">=</span> <span class="fu">substring</span>(<span class="fu">names</span>(MEs), <span class="dv">3</span>)</span>
<span id="cb48-22"><a href="#cb48-22" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb48-23"><a href="#cb48-23" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb48-24"><a href="#cb48-24" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-25"><a href="#cb48-25" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-26"><a href="#cb48-26" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb48-27"><a href="#cb48-27" tabindex="-1"></a></span>
<span id="cb48-28"><a href="#cb48-28" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>mean_T)</span>
<span id="cb48-29"><a href="#cb48-29" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;mean_T&quot;</span></span>
<span id="cb48-30"><a href="#cb48-30" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb48-31"><a href="#cb48-31" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb48-32"><a href="#cb48-32" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-33"><a href="#cb48-33" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-34"><a href="#cb48-34" tabindex="-1"></a>GS2<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb48-35"><a href="#cb48-35" tabindex="-1"></a></span>
<span id="cb48-36"><a href="#cb48-36" tabindex="-1"></a></span>
<span id="cb48-37"><a href="#cb48-37" tabindex="-1"></a>trait <span class="ot">=</span> <span class="fu">as.data.frame</span>(datTraits<span class="sc">$</span>strength.all_study)</span>
<span id="cb48-38"><a href="#cb48-38" tabindex="-1"></a><span class="fu">names</span>(trait)<span class="ot">=</span> <span class="st">&quot;strength&quot;</span></span>
<span id="cb48-39"><a href="#cb48-39" tabindex="-1"></a>geneTraitSignificance <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cor</span>(datExpr0, trait, <span class="at">use =</span> <span class="st">&quot;p&quot;</span>))</span>
<span id="cb48-40"><a href="#cb48-40" tabindex="-1"></a>GSPvalue <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">corPvalueStudent</span>(<span class="fu">as.matrix</span>(geneTraitSignificance), nSamples))</span>
<span id="cb48-41"><a href="#cb48-41" tabindex="-1"></a><span class="fu">names</span>(geneTraitSignificance) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-42"><a href="#cb48-42" tabindex="-1"></a><span class="fu">names</span>(GSPvalue) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;p.GS.&quot;</span>, <span class="fu">names</span>(trait), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb48-43"><a href="#cb48-43" tabindex="-1"></a>GS3<span class="ot">&lt;-</span> <span class="fu">cbind</span>(geneTraitSignificance,GSPvalue)</span>
<span id="cb48-44"><a href="#cb48-44" tabindex="-1"></a></span>
<span id="cb48-45"><a href="#cb48-45" tabindex="-1"></a>GS<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">rownames</span>(GS)</span>
<span id="cb48-46"><a href="#cb48-46" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">cbind</span>(GS,GS2, GS3)</span>
<span id="cb48-47"><a href="#cb48-47" tabindex="-1"></a>GS<span class="ot">&lt;-</span> <span class="fu">merge</span>(geneInfo0,GS, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb48-48"><a href="#cb48-48" tabindex="-1"></a></span>
<span id="cb48-49"><a href="#cb48-49" tabindex="-1"></a></span>
<span id="cb48-50"><a href="#cb48-50" tabindex="-1"></a><span class="fu">write.csv</span>(GS, <span class="at">file=</span><span class="fu">paste</span>(<span class="st">&quot;../WGCNA_results/within_tissues/&quot;</span>,tissue, <span class="st">&quot;_results_GeneSignificance.csv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><br></p>
</div>
<div id="intermodule-correlations" class="section level2" number="1.7">
<h2><span class="header-section-number">1.7</span> Intermodule
correlations</h2>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a><span class="co"># Specify colors</span></span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a>colz<span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(MEs))</span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a><span class="fu">names</span>(colz)<span class="ot">&lt;-</span> <span class="fu">colnames</span>(MEs)</span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a>ann_colors <span class="ot">=</span> <span class="fu">list</span>(<span class="at">module=</span>colz)</span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a></span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>annotation_col<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">colnames</span>(MEs), <span class="at">module=</span><span class="fu">colnames</span>(MEs))</span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a></span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a></span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>correlation<span class="ot">&lt;-</span> <span class="fu">cor</span>(MEs)</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a><span class="fu">pheatmap</span>(correlation, <span class="at">annotation_col =</span> annotation_col, <span class="at">annotation_colors =</span> ann_colors, <span class="at">annotation_legend=</span><span class="cn">FALSE</span>, <span class="at">legend_breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="fl">0.5</span>,<span class="dv">0</span>,<span class="fl">0.5</span>, <span class="dv">1</span>,<span class="dv">1</span>), </span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a><span class="at">main=</span><span class="fu">paste</span>(<span class="st">&quot;module eigengene correlations in&quot;</span>, tissue, <span class="at">sep=</span><span class="st">&quot; &quot;</span>), <span class="at">legend_labels =</span> <span class="fu">c</span>(<span class="st">&quot;-1&quot;</span>, <span class="st">&quot;-0.5&quot;</span>, <span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.5&quot;</span>,<span class="st">&quot;1&quot;</span>,<span class="st">&quot;correlation</span><span class="sc">\n\n</span><span class="st">&quot;</span>))</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="interesting-modules" class="section level2" number="1.8">
<h2><span class="header-section-number">1.8</span> Interesting
modules</h2>
<p><br></p>
<div id="pink-module---mean-testosterone" class="section level3"
number="1.8.1">
<h3><span class="header-section-number">1.8.1</span> Pink module - mean
testosterone</h3>
<p>Here I extract the genes associated with a given module and explore
them further. This module is potentially associated with mean
testosterone. The WGCNA manual says that a good association will show a
relationship between the module membership scores of each gene and the
gene-significance for that gene.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a>whichModule<span class="ot">=</span><span class="st">&quot;pink&quot;</span></span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a>modcol<span class="ot">=</span><span class="fu">paste</span>(<span class="st">&quot;MM.&quot;</span>,whichModule,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a>module_genes<span class="ot">&lt;-</span> color[color<span class="sc">$</span>moduleColor<span class="sc">==</span>whichModule,]</span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a></span>
<span id="cb50-6"><a href="#cb50-6" tabindex="-1"></a></span>
<span id="cb50-7"><a href="#cb50-7" tabindex="-1"></a><span class="fu">verboseScatterplot</span>(<span class="fu">abs</span>(module_genes[, modcol]),</span>
<span id="cb50-8"><a href="#cb50-8" tabindex="-1"></a><span class="fu">abs</span>(GS[GS<span class="sc">$</span>moduleColor<span class="sc">==</span>whichModule, <span class="st">&quot;GS.mean_T&quot;</span>]),</span>
<span id="cb50-9"><a href="#cb50-9" tabindex="-1"></a><span class="at">xlab =</span> <span class="fu">paste</span>(<span class="st">&quot;Module Membership in&quot;</span>, whichModule, <span class="st">&quot;module&quot;</span>),</span>
<span id="cb50-10"><a href="#cb50-10" tabindex="-1"></a><span class="at">ylab =</span> <span class="st">&quot;Gene significance for mean Testosterone&quot;</span>,</span>
<span id="cb50-11"><a href="#cb50-11" tabindex="-1"></a><span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Module membership vs. gene significance</span><span class="sc">\n</span><span class="st">&quot;</span>),</span>
<span id="cb50-12"><a href="#cb50-12" tabindex="-1"></a><span class="at">cex.main =</span> <span class="fl">1.2</span>, <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.axis =</span> <span class="fl">1.2</span>, <span class="at">col =</span> whichModule)</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GON%20pink%20module-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="top-30-hub-genes" class="section level4" number="1.8.1.1">
<h4><span class="header-section-number">1.8.1.1</span> Top 30 hub
genes</h4>
<p>Next, I am going to explore the top 30 hub genes based on highest
module membership</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>nTop<span class="ot">&lt;-</span><span class="dv">30</span> </span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a><span class="co">#extract the expression of the top 30 genes and sort them based on module membership. </span></span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>sorted<span class="ot">=</span>color<span class="sc">$</span>gene[<span class="fu">order</span>(color[,modcol],<span class="at">decreasing=</span>T)]</span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a>hubs<span class="ot">=</span>sorted[<span class="dv">1</span><span class="sc">:</span>nTop]</span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a>hubs<span class="ot">=</span>datExpr0[,<span class="fu">match</span>(hubs, <span class="fu">colnames</span>(datExpr0))]</span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a></span>
<span id="cb51-7"><a href="#cb51-7" tabindex="-1"></a></span>
<span id="cb51-8"><a href="#cb51-8" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" tabindex="-1"></a><span class="do">### Sort individual data based on the trait value</span></span>
<span id="cb51-10"><a href="#cb51-10" tabindex="-1"></a>expDesign<span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names=</span><span class="fu">rownames</span>(gon_key), <span class="at">mean_T=</span>gon_key[,<span class="st">&quot;mean_T&quot;</span>], <span class="at">Status=</span>gon_key[,<span class="st">&quot;Status&quot;</span>])</span>
<span id="cb51-11"><a href="#cb51-11" tabindex="-1"></a>expDesign<span class="ot">&lt;-</span> expDesign[<span class="fu">order</span>(expDesign<span class="sc">$</span>mean_T),]</span>
<span id="cb51-12"><a href="#cb51-12" tabindex="-1"></a>ann_cols<span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Status=</span><span class="fu">c</span>(<span class="at">territorial=</span><span class="st">&quot;#414042&quot;</span>,<span class="at">floater=</span><span class="st">&quot;#E54849&quot;</span>))</span>
<span id="cb51-13"><a href="#cb51-13" tabindex="-1"></a>hubs<span class="ot">&lt;-</span> hubs[<span class="fu">match</span>(<span class="fu">rownames</span>(expDesign), <span class="fu">rownames</span>(hubs)), ]</span>
<span id="cb51-14"><a href="#cb51-14" tabindex="-1"></a></span>
<span id="cb51-15"><a href="#cb51-15" tabindex="-1"></a><span class="co">#plot heatmap</span></span>
<span id="cb51-16"><a href="#cb51-16" tabindex="-1"></a><span class="fu">pheatmap</span>(hubs,<span class="at">scale=</span><span class="st">&quot;column&quot;</span>, <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,<span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,<span class="at">show_rownames =</span> <span class="cn">TRUE</span>, <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,<span class="at">border_color=</span><span class="cn">NA</span>,  <span class="at">main=</span><span class="fu">paste</span>(whichModule,<span class="st">&quot;module in&quot;</span>,tissue), <span class="at">annotation_row=</span>expDesign, <span class="at">annotation_colors=</span>ann_cols)</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<p>Now, let’s plot the top gene in this module to see its relationship
with mean testosterone</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>gone<span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dd, <span class="at">gene=</span><span class="st">&quot;CHRNA3&quot;</span>, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;mean_T&quot;</span>, <span class="st">&quot;Class&quot;</span>), <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a><span class="fu">ggplot</span>(gone, <span class="fu">aes</span>(<span class="at">x=</span>mean_T, <span class="at">y=</span>count)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;CHRNA3 in GON&quot;</span>,<span class="at">x=</span><span class="st">&quot;mean T&quot;</span>, <span class="at">y=</span><span class="st">&quot;Normalised Counts&quot;</span>) <span class="sc">+</span> peri_theme</span></code></pre></div>
<pre><code>## Warning: Removed 4 rows containing missing values
## (`geom_point()`).</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a><span class="do">##### TOM BASED NETWORK ####</span></span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a><span class="do">## get names of genes from the module</span></span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" tabindex="-1"></a></span>
<span id="cb54-5"><a href="#cb54-5" tabindex="-1"></a>probes <span class="ot">&lt;-</span> <span class="fu">names</span>(datExpr0)</span>
<span id="cb54-6"><a href="#cb54-6" tabindex="-1"></a>inModule <span class="ot">&lt;-</span> moduleColors<span class="sc">==</span>whichModule <span class="co">#creates logical vector of whether that gene is in the module</span></span>
<span id="cb54-7"><a href="#cb54-7" tabindex="-1"></a>modProbes <span class="ot">&lt;-</span> probes[inModule]</span>
<span id="cb54-8"><a href="#cb54-8" tabindex="-1"></a></span>
<span id="cb54-9"><a href="#cb54-9" tabindex="-1"></a></span>
<span id="cb54-10"><a href="#cb54-10" tabindex="-1"></a><span class="co"># Select the corresponding Topological Overlap</span></span>
<span id="cb54-11"><a href="#cb54-11" tabindex="-1"></a>modTOM <span class="ot">&lt;-</span> TOM[inModule, inModule] <span class="co">#TOM object does not have any gene names so need to use the positional information from the logical vector.</span></span>
<span id="cb54-12"><a href="#cb54-12" tabindex="-1"></a></span>
<span id="cb54-13"><a href="#cb54-13" tabindex="-1"></a><span class="fu">dimnames</span>(modTOM) <span class="ot">&lt;-</span> <span class="fu">list</span>(modProbes, modProbes)</span>
<span id="cb54-14"><a href="#cb54-14" tabindex="-1"></a></span>
<span id="cb54-15"><a href="#cb54-15" tabindex="-1"></a>IMConn<span class="ot">&lt;-</span> <span class="fu">softConnectivity</span>(datExpr0[, modProbes], <span class="at">power =</span> softPower, <span class="at">type=</span><span class="st">&quot;signed&quot;</span>)</span>
<span id="cb54-16"><a href="#cb54-16" tabindex="-1"></a><span class="fu">names</span>(IMConn)<span class="ot">&lt;-</span> modProbes</span>
<span id="cb54-17"><a href="#cb54-17" tabindex="-1"></a>IMConn<span class="ot">&lt;-</span> <span class="fu">sort</span>(IMConn, <span class="at">decreasing=</span>T) <span class="co">#rank by highest connectivity</span></span>
<span id="cb54-18"><a href="#cb54-18" tabindex="-1"></a>nTop<span class="ot">&lt;-</span><span class="dv">30</span></span>
<span id="cb54-19"><a href="#cb54-19" tabindex="-1"></a>top<span class="ot">&lt;-</span> IMConn[<span class="dv">1</span><span class="sc">:</span>nTop] <span class="co">#Take the top genes </span></span>
<span id="cb54-20"><a href="#cb54-20" tabindex="-1"></a>top30TOM<span class="ot">&lt;-</span> modTOM[<span class="fu">names</span>(top),<span class="fu">names</span>(top)] </span>
<span id="cb54-21"><a href="#cb54-21" tabindex="-1"></a><span class="co">#rename the TOM columns and rows to reflect the locno + gene ID. </span></span>
<span id="cb54-22"><a href="#cb54-22" tabindex="-1"></a>top_genes<span class="ot">&lt;-</span> module_genes[module_genes<span class="sc">$</span>gene <span class="sc">%in%</span> <span class="fu">names</span>(top),]</span>
<span id="cb54-23"><a href="#cb54-23" tabindex="-1"></a>top_genes<span class="ot">&lt;-</span> top_genes[<span class="fu">match</span>(<span class="fu">names</span>(top), top_genes<span class="sc">$</span>gene), ]</span>
<span id="cb54-24"><a href="#cb54-24" tabindex="-1"></a><span class="fu">colnames</span>(top30TOM)<span class="ot">&lt;-</span> top_genes<span class="sc">$</span>display_gene_ID</span>
<span id="cb54-25"><a href="#cb54-25" tabindex="-1"></a><span class="fu">rownames</span>(top30TOM)<span class="ot">&lt;-</span> top_genes<span class="sc">$</span>display_gene_ID</span>
<span id="cb54-26"><a href="#cb54-26" tabindex="-1"></a>vis <span class="ot">=</span> <span class="fu">exportNetworkToVisANT</span>(top30TOM,</span>
<span id="cb54-27"><a href="#cb54-27" tabindex="-1"></a><span class="at">weighted =</span> <span class="cn">TRUE</span>,</span>
<span id="cb54-28"><a href="#cb54-28" tabindex="-1"></a><span class="at">threshold =</span> <span class="fl">0.00</span>)</span>
<span id="cb54-29"><a href="#cb54-29" tabindex="-1"></a><span class="co">#take the top 25% of TOM connections</span></span>
<span id="cb54-30"><a href="#cb54-30" tabindex="-1"></a>vis<span class="ot">&lt;-</span> vis[vis<span class="sc">$</span>weight<span class="sc">&gt;</span><span class="fu">summary</span>(vis<span class="sc">$</span>weight)[<span class="dv">5</span>],]</span>
<span id="cb54-31"><a href="#cb54-31" tabindex="-1"></a>CoExpGraph <span class="ot">&lt;-</span> <span class="fu">graph.data.frame</span>(vis, <span class="at">directed=</span><span class="cn">FALSE</span>)</span>
<span id="cb54-32"><a href="#cb54-32" tabindex="-1"></a>nDegree <span class="ot">&lt;-</span> <span class="fu">degree</span>(CoExpGraph, <span class="at">v=</span><span class="fu">V</span>(CoExpGraph), <span class="at">mode=</span><span class="fu">c</span>(<span class="st">&quot;total&quot;</span>))</span>
<span id="cb54-33"><a href="#cb54-33" tabindex="-1"></a><span class="fu">V</span>(CoExpGraph)<span class="sc">$</span>degree <span class="ot">&lt;-</span> nDegree</span></code></pre></div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">ggnetwork</span>(CoExpGraph, <span class="at">layout=</span><span class="fu">with_fr</span>()), <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">xend =</span> xend, <span class="at">yend =</span> yend))<span class="sc">+</span><span class="fu">geom_edges</span>(<span class="fu">aes</span>(<span class="at">color=</span>weight))<span class="sc">+</span><span class="fu">geom_nodes</span>(<span class="fu">aes</span>(x,y, <span class="at">size=</span>degree),<span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">alpha=</span><span class="fl">0.7</span>) <span class="sc">+</span><span class="fu">scale_color_gradient</span>(<span class="at">low=</span><span class="st">&quot;#ffd71f&quot;</span>, <span class="at">high=</span><span class="st">&quot;#e50000&quot;</span>)<span class="sc">+</span><span class="fu">theme_blank</span>()<span class="sc">+</span> <span class="fu">geom_nodetext_repel</span>(<span class="fu">aes</span>(<span class="at">label=</span>name))</span></code></pre></div>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-ontology" class="section level4" number="1.8.1.2">
<h4><span class="header-section-number">1.8.1.2</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in pink module in GON
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0001525" class="uri">GO:0001525</a>
</td>
<td style="text-align:left;">
angiogenesis
</td>
<td style="text-align:left;">
25/490
</td>
<td style="text-align:left;">
205/12555
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
0.0006380
</td>
<td style="text-align:right;">
0.0005905
</td>
<td style="text-align:left;">
ACVR1/ANXA2/APOLD1/ARHGAP22/CCN2/EGF/ENG/FGF1/FOXC1/HSPG2/ID1/JUN/LOC113990968/LOC114000824/LOC114001403/LOC114001849/MAPK14/MMP2/NRCAM/PLXND1/PROK2/ROBO4/SCG2/TGFBI/THY1
</td>
<td style="text-align:right;">
25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001975" class="uri">GO:0001975</a>
</td>
<td style="text-align:left;">
response to amphetamine
</td>
<td style="text-align:left;">
8/490
</td>
<td style="text-align:left;">
24/12555
</td>
<td style="text-align:right;">
0.0000022
</td>
<td style="text-align:right;">
0.0016401
</td>
<td style="text-align:right;">
0.0015179
</td>
<td style="text-align:left;">
ASIC3/DBH/PPP1R9B/RGS10/RGS2/RGS4/SLC18A2/TH
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007268" class="uri">GO:0007268</a>
</td>
<td style="text-align:left;">
chemical synaptic transmission
</td>
<td style="text-align:left;">
22/490
</td>
<td style="text-align:left;">
188/12555
</td>
<td style="text-align:right;">
0.0000041
</td>
<td style="text-align:right;">
0.0020970
</td>
<td style="text-align:right;">
0.0019408
</td>
<td style="text-align:left;">
AKAP5/CACNA1E/CHRNA3/CPLX1/CRH/DBH/FGF12/GAD2/HTR1E/HTR4/LOC113983406/LOC113990721/NPY/PLP1/PMP22/SLC18A2/SLC6A2/SNCA/SNCG/SV2B/SYN2/SYT1
</td>
<td style="text-align:right;">
22
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044267" class="uri">GO:0044267</a>
</td>
<td style="text-align:left;">
cellular protein metabolic process
</td>
<td style="text-align:left;">
20/490
</td>
<td style="text-align:left;">
165/12555
</td>
<td style="text-align:right;">
0.0000066
</td>
<td style="text-align:right;">
0.0025184
</td>
<td style="text-align:right;">
0.0023308
</td>
<td style="text-align:left;">
ADORA2B/ANO8/B2M/CHGB/CTSH/FURIN/HSPG2/LAMB2/LGALS1/LOC113986369/LOC114003910/LYZ/MMP2/SCG2/SCG3/SFTPB/SFTPC/SNCA/SNCAIP/TGFBI
</td>
<td style="text-align:right;">
20
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009749" class="uri">GO:0009749</a>
</td>
<td style="text-align:left;">
response to glucose
</td>
<td style="text-align:left;">
10/490
</td>
<td style="text-align:left;">
51/12555
</td>
<td style="text-align:right;">
0.0000226
</td>
<td style="text-align:right;">
0.0068942
</td>
<td style="text-align:right;">
0.0063809
</td>
<td style="text-align:left;">
ACVR1C/ADIPOQ/CCN2/FKBP1B/GJA1/LOC113992227/LPL/SARM1/SREBF1/TXNIP
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010628" class="uri">GO:0010628</a>
</td>
<td style="text-align:left;">
positive regulation of gene expression
</td>
<td style="text-align:left;">
27/490
</td>
<td style="text-align:left;">
310/12555
</td>
<td style="text-align:right;">
0.0000826
</td>
<td style="text-align:right;">
0.0189665
</td>
<td style="text-align:right;">
0.0175543
</td>
<td style="text-align:left;">
CCN2/CITED2/CRH/CTSH/EGF/FGF1/GATA5/GJA1/ID1/IL34/LOC113992691/LOC114001703/LRRC32/MAPK14/MYC/NOS1AP/NR5A1/OSR1/OSR2/PDE2A/PLP1/SLC26A9/SLC6A4/STAR/STAT3/TWIST1/ZNF804A
</td>
<td style="text-align:right;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051412" class="uri">GO:0051412</a>
</td>
<td style="text-align:left;">
response to corticosterone
</td>
<td style="text-align:left;">
5/490
</td>
<td style="text-align:left;">
13/12555
</td>
<td style="text-align:right;">
0.0000880
</td>
<td style="text-align:right;">
0.0189665
</td>
<td style="text-align:right;">
0.0175543
</td>
<td style="text-align:left;">
CRH/LOC113992757/NEFL/STAR/TH
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009617" class="uri">GO:0009617</a>
</td>
<td style="text-align:left;">
response to bacterium
</td>
<td style="text-align:left;">
9/490
</td>
<td style="text-align:left;">
49/12555
</td>
<td style="text-align:right;">
0.0000996
</td>
<td style="text-align:right;">
0.0189665
</td>
<td style="text-align:right;">
0.0175543
</td>
<td style="text-align:left;">
ADIPOQ/CCDC80/CLPS/DGKB/FKBP5/LOC114003559/LPL/RGS1/VIL1
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050679" class="uri">GO:0050679</a>
</td>
<td style="text-align:left;">
positive regulation of epithelial cell proliferation
</td>
<td style="text-align:left;">
9/490
</td>
<td style="text-align:left;">
55/12555
</td>
<td style="text-align:right;">
0.0002504
</td>
<td style="text-align:right;">
0.0399249
</td>
<td style="text-align:right;">
0.0369522
</td>
<td style="text-align:left;">
ATP7A/DLX5/FGF1/ID1/MYC/NR4A3/OSR1/OSR2/TWIST1
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010977" class="uri">GO:0010977</a>
</td>
<td style="text-align:left;">
negative regulation of neuron projection development
</td>
<td style="text-align:left;">
9/490
</td>
<td style="text-align:left;">
56/12555
</td>
<td style="text-align:right;">
0.0002882
</td>
<td style="text-align:right;">
0.0399249
</td>
<td style="text-align:right;">
0.0369522
</td>
<td style="text-align:left;">
B2M/INPP5J/LGALS1/LOC114000824/NR2F1/PRAG1/RTN4RL2/STMN2/VIM
</td>
<td style="text-align:right;">
9
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GON%20pink%20GO-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GON%20pink%20GO-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="pituitary-pit" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Pituitary (PIT)</h1>
<p>The code for this tissue is mostly above, but each section the R
environment is cleared to only keep the base data.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a>keep<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;data&quot;</span>, <span class="st">&quot;peri_theme&quot;</span>, <span class="st">&quot;key&quot;</span>,<span class="st">&quot;key_behav&quot;</span>,<span class="st">&quot;dodge&quot;</span>,<span class="st">&quot;genes_key&quot;</span>, <span class="st">&quot;keep&quot;</span>, <span class="st">&quot;go2gene_mf&quot;</span>,<span class="st">&quot;go2name_mf&quot;</span>,<span class="st">&quot;go2gene_bp&quot;</span>,<span class="st">&quot;go2name_bp&quot;</span>, <span class="st">&quot;wgcna_summary&quot;</span>)</span>
<span id="cb59-2"><a href="#cb59-2" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span> <span class="fu">ls</span>()[<span class="sc">!</span>(<span class="fu">ls</span>() <span class="sc">%in%</span> keep)])</span></code></pre></div>
<pre><code>## [1] 14812</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 101 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##  Flagging genes and samples with too many missing values...
##   ..step 1</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1 0.003630 -4.320          0.856  7410.0    7420.0   7640
## 2      2 0.000786  0.449          0.935  4240.0    4230.0   4720
## 3      3 0.002610 -0.375          0.918  2650.0    2640.0   3310
## 4      4 0.027300 -0.758          0.906  1780.0    1750.0   2490
## 5      5 0.091300 -1.030          0.896  1250.0    1220.0   1960
## 6      6 0.181000 -1.180          0.892   911.0     881.0   1600
## 7      7 0.279000 -1.310          0.890   687.0     654.0   1340
## 8      8 0.375000 -1.420          0.892   531.0     498.0   1140
## 9      9 0.455000 -1.490          0.896   419.0     386.0    985
## 10    10 0.517000 -1.580          0.894   337.0     304.0    863
## 11    11 0.578000 -1.620          0.908   275.0     243.0    764
## 12    12 0.619000 -1.670          0.911   228.0     196.0    681
## 13    14 0.675000 -1.750          0.914   161.0     133.0    554
## 14    16 0.707000 -1.810          0.913   118.0      92.7    460
## 15    18 0.712000 -1.890          0.903    89.3      66.3    389
## 16    20 0.732000 -1.920          0.912    69.1      48.5    334
## 17    22 0.751000 -1.930          0.922    54.6      36.4    290
## 18    24 0.757000 -1.970          0.925    43.8      27.8    254
## 19    26 0.775000 -1.970          0.934    35.7      21.5    224</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for PIT is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-13-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/PIT%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in PIT
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
1328
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
1594
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
156
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
208
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
2779
</td>
</tr>
<tr>
<td style="text-align:left;">
coral2
</td>
<td style="text-align:right;">
382
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgreen
</td>
<td style="text-align:right;">
397
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
883
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen2
</td>
<td style="text-align:right;">
589
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
253
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
126
</td>
</tr>
<tr>
<td style="text-align:left;">
darkslateblue
</td>
<td style="text-align:right;">
936
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
72
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick3
</td>
<td style="text-align:right;">
1601
</td>
</tr>
<tr>
<td style="text-align:left;">
floralwhite
</td>
<td style="text-align:right;">
295
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcyan1
</td>
<td style="text-align:right;">
133
</td>
</tr>
<tr>
<td style="text-align:left;">
lightgreen
</td>
<td style="text-align:right;">
1798
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered3
</td>
<td style="text-align:right;">
84
</td>
</tr>
<tr>
<td style="text-align:left;">
palevioletred2
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
pink4
</td>
<td style="text-align:right;">
265
</td>
</tr>
<tr>
<td style="text-align:left;">
plum1
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
plum2
</td>
<td style="text-align:right;">
176
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna4
</td>
<td style="text-align:right;">
56
</td>
</tr>
<tr>
<td style="text-align:left;">
violet
</td>
<td style="text-align:right;">
375
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="interesting-modules-1" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Interesting
Modules</h2>
<div id="floralwhite-module---strength" class="section level3"
number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> floralwhite module
- strength</h3>
<p>This module is associated with social network strength.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20floralwhite%20%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20floralwhite%20%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>##  softConnectivity: FYI: connecitivty of genes with less than 4 valid samples will be returned as NA.
##  ..calculating connectivities..</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20floralwhite%20network%20plotting%20TOM-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>This module is too influenced by PFT3. This is a sample I removed in
the DESeq2 analysis. Disregard.</p>
<p><br></p>
</div>
<div id="pink4-module---strength" class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> pink4 module -
strength</h3>
<p>This module is also associated with social network strength</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20pink4%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20pink4%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-1" class="section level4" number="2.1.2.1">
<h4><span class="header-section-number">2.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in pink4 module in PIT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007288" class="uri">GO:0007288</a>
</td>
<td style="text-align:left;">
sperm axoneme assembly
</td>
<td style="text-align:left;">
4/214
</td>
<td style="text-align:left;">
20/12079
</td>
<td style="text-align:right;">
0.0003714
</td>
<td style="text-align:right;">
0.1318112
</td>
<td style="text-align:right;">
0.1312432
</td>
<td style="text-align:left;">
ARMC2/CFAP206/SPEF2/TTLL1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031647" class="uri">GO:0031647</a>
</td>
<td style="text-align:left;">
regulation of protein stability
</td>
<td style="text-align:left;">
7/214
</td>
<td style="text-align:left;">
76/12079
</td>
<td style="text-align:right;">
0.0003854
</td>
<td style="text-align:right;">
0.1318112
</td>
<td style="text-align:right;">
0.1312432
</td>
<td style="text-align:left;">
APTX/ATF7IP/DPM2/LOC113993664/MOSMO/PINX1/QRSL1
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006362" class="uri">GO:0006362</a>
</td>
<td style="text-align:left;">
transcription elongation from RNA polymerase I promoter
</td>
<td style="text-align:left;">
3/214
</td>
<td style="text-align:left;">
23/12079
</td>
<td style="text-align:right;">
0.0074766
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
CCNH/ERCC3/TAF1A
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050821" class="uri">GO:0050821</a>
</td>
<td style="text-align:left;">
protein stabilization
</td>
<td style="text-align:left;">
8/214
</td>
<td style="text-align:left;">
162/12079
</td>
<td style="text-align:right;">
0.0082095
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
ATF7IP/CCNH/CCT6A/GAPDH/GTPBP4/MFSD1/NCLN/TBL1XR1
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006363" class="uri">GO:0006363</a>
</td>
<td style="text-align:left;">
termination of RNA polymerase I transcription
</td>
<td style="text-align:left;">
3/214
</td>
<td style="text-align:left;">
24/12079
</td>
<td style="text-align:right;">
0.0084346
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
CCNH/ERCC3/TAF1A
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006890" class="uri">GO:0006890</a>
</td>
<td style="text-align:left;">
retrograde vesicle-mediated transport, Golgi to endoplasmic reticulum
</td>
<td style="text-align:left;">
5/214
</td>
<td style="text-align:left;">
73/12079
</td>
<td style="text-align:right;">
0.0094384
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
ARFGAP2/COPB1/KIFAP3/PITPNB/RINT1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006361" class="uri">GO:0006361</a>
</td>
<td style="text-align:left;">
transcription initiation from RNA polymerase I promoter
</td>
<td style="text-align:left;">
3/214
</td>
<td style="text-align:left;">
25/12079
</td>
<td style="text-align:right;">
0.0094614
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
CCNH/ERCC3/TAF1A
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006310" class="uri">GO:0006310</a>
</td>
<td style="text-align:left;">
DNA recombination
</td>
<td style="text-align:left;">
4/214
</td>
<td style="text-align:left;">
51/12079
</td>
<td style="text-align:right;">
0.0125004
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
ACTR8/KIN/LOC113982971/SUPV3L1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000226" class="uri">GO:0000226</a>
</td>
<td style="text-align:left;">
microtubule cytoskeleton organization
</td>
<td style="text-align:left;">
6/214
</td>
<td style="text-align:left;">
109/12079
</td>
<td style="text-align:right;">
0.0128965
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
GAPDH/KNSTRN/MARK3/PRKCZ/TTLL1/TUBA8
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070085" class="uri">GO:0070085</a>
</td>
<td style="text-align:left;">
glycosylation
</td>
<td style="text-align:left;">
2/214
</td>
<td style="text-align:left;">
11/12079
</td>
<td style="text-align:right;">
0.0154662
</td>
<td style="text-align:right;">
0.6320762
</td>
<td style="text-align:right;">
0.6293525
</td>
<td style="text-align:left;">
B4GALT4/LOC113992792
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20pink4%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PIT%20pink4%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
</div>
<div id="ventromedial-hypothalamus-vmh" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Ventromedial
Hypothalamus (VMH)</h1>
<pre><code>## [1] 15223</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 102 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1 0.008020 12.600          0.986  7610.0   7610.00   7770
## 2      2 0.000212 -0.471          0.940  4210.0   4200.00   4600
## 3      3 0.080100 -3.750          0.890  2500.0   2480.00   3050
## 4      4 0.196000 -3.140          0.903  1580.0   1550.00   2160
## 5      5 0.329000 -2.720          0.928  1040.0   1010.00   1600
## 6      6 0.473000 -2.710          0.949   715.0    686.00   1240
## 7      7 0.567000 -2.620          0.955   506.0    478.00    985
## 8      8 0.632000 -2.530          0.961   368.0    341.00    802
## 9      9 0.685000 -2.440          0.966   274.0    248.00    664
## 10    10 0.716000 -2.400          0.964   208.0    184.00    558
## 11    11 0.747000 -2.330          0.968   160.0    139.00    474
## 12    12 0.772000 -2.290          0.972   126.0    106.00    408
## 13    14 0.800000 -2.300          0.970    80.4     64.00    314
## 14    16 0.821000 -2.300          0.970    53.8     40.30    248
## 15    18 0.831000 -2.290          0.967    37.3     26.20    201
## 16    20 0.840000 -2.270          0.969    26.7     17.50    165
## 17    22 0.851000 -2.250          0.972    19.6     12.00    138
## 18    24 0.855000 -2.240          0.972    14.7      8.42    117
## 19    26 0.859000 -2.230          0.972    11.2      6.03    101</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for VMH is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/VMH%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in VMH
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
223
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
103
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
590
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
2088
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
2842
</td>
</tr>
<tr>
<td style="text-align:left;">
brown2
</td>
<td style="text-align:right;">
50
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
981
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
1033
</td>
</tr>
<tr>
<td style="text-align:left;">
coral2
</td>
<td style="text-align:right;">
125
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgreen
</td>
<td style="text-align:right;">
901
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen4
</td>
<td style="text-align:right;">
72
</td>
</tr>
<tr>
<td style="text-align:left;">
darkslateblue
</td>
<td style="text-align:right;">
99
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
906
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
floralwhite
</td>
<td style="text-align:right;">
530
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
620
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred4
</td>
<td style="text-align:right;">
474
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcoral
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcyan1
</td>
<td style="text-align:right;">
253
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta
</td>
<td style="text-align:right;">
512
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumorchid
</td>
<td style="text-align:right;">
65
</td>
</tr>
<tr>
<td style="text-align:left;">
midnightblue
</td>
<td style="text-align:right;">
1746
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
292
</td>
</tr>
<tr>
<td style="text-align:left;">
royalblue
</td>
<td style="text-align:right;">
208
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue2
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
steelblue
</td>
<td style="text-align:right;">
155
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle
</td>
<td style="text-align:right;">
42
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle3
</td>
<td style="text-align:right;">
46
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="interesting-modules-2" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Interesting
Modules</h2>
<div id="lightcoral-module---statusmean-testosterone"
class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> lightcoral module -
Status/mean testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/VMH%20lightcoral%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/VMH%20lightcoral%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>The top hub gene was not significantly associated with either mean_T
or Status in the DESeq2 analysis.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="gene-ontology-2" class="section level4" number="3.1.1.1">
<h4><span class="header-section-number">3.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in lightcoral module in VMH
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0032094" class="uri">GO:0032094</a>
</td>
<td style="text-align:left;">
response to food
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
15/12287
</td>
<td style="text-align:right;">
0.0013361
</td>
<td style="text-align:right;">
0.1789233
</td>
<td style="text-align:right;">
0.1543029
</td>
<td style="text-align:left;">
BCL10/OPRM1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008154" class="uri">GO:0008154</a>
</td>
<td style="text-align:left;">
actin polymerization or depolymerization
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
16/12287
</td>
<td style="text-align:right;">
0.0015234
</td>
<td style="text-align:right;">
0.1789233
</td>
<td style="text-align:right;">
0.1543029
</td>
<td style="text-align:left;">
CAPG/WASL
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001676" class="uri">GO:0001676</a>
</td>
<td style="text-align:left;">
long-chain fatty acid metabolic process
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
19/12287
</td>
<td style="text-align:right;">
0.0021557
</td>
<td style="text-align:right;">
0.1789233
</td>
<td style="text-align:right;">
0.1543029
</td>
<td style="text-align:left;">
ACAD9/ACSL4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000310" class="uri">GO:2000310</a>
</td>
<td style="text-align:left;">
regulation of NMDA receptor activity
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
31/12287
</td>
<td style="text-align:right;">
0.0057008
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
LOC113983406/OPRM1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002224" class="uri">GO:0002224</a>
</td>
<td style="text-align:left;">
toll-like receptor signaling pathway
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
33/12287
</td>
<td style="text-align:right;">
0.0064432
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
BCL10/IRAK4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043280" class="uri">GO:0043280</a>
</td>
<td style="text-align:left;">
positive regulation of cysteine-type endopeptidase activity involved in
apoptotic process
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
36/12287
</td>
<td style="text-align:right;">
0.0076346
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
BCL10/GRAMD4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007254" class="uri">GO:0007254</a>
</td>
<td style="text-align:left;">
JNK cascade
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
39/12287
</td>
<td style="text-align:right;">
0.0089175
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
IRAK4/MAP4K3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016192" class="uri">GO:0016192</a>
</td>
<td style="text-align:left;">
vesicle-mediated transport
</td>
<td style="text-align:left;">
3/45
</td>
<td style="text-align:left;">
130/12287
</td>
<td style="text-align:right;">
0.0118818
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
ARL5B/LOC113995129/LOC113998023
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051056" class="uri">GO:0051056</a>
</td>
<td style="text-align:left;">
regulation of small GTPase mediated signal transduction
</td>
<td style="text-align:left;">
3/45
</td>
<td style="text-align:left;">
140/12287
</td>
<td style="text-align:right;">
0.0144937
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
ARHGEF7/LOC113995129/LOC113998023
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001843" class="uri">GO:0001843</a>
</td>
<td style="text-align:left;">
neural tube closure
</td>
<td style="text-align:left;">
2/45
</td>
<td style="text-align:left;">
66/12287
</td>
<td style="text-align:right;">
0.0242541
</td>
<td style="text-align:right;">
0.2080429
</td>
<td style="text-align:right;">
0.1794156
</td>
<td style="text-align:left;">
BCL10/SEC24B
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/VMH%20lightcoral%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="anterior-hypothalamus-ah" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Anterior Hypothalamus
(AH)</h1>
<pre><code>## [1] 14928</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1 0.000105 -2.410          0.986  7460.0    7460.0   7550
## 2      2 0.065400  4.700          0.810  4330.0    4330.0   4730
## 3      3 0.052000  1.330          0.796  2760.0    2760.0   3340
## 4      4 0.003500 -0.203          0.819  1880.0    1880.0   2550
## 5      5 0.077200 -0.705          0.850  1350.0    1340.0   2050
## 6      6 0.196000 -0.959          0.881  1000.0     989.0   1700
## 7      7 0.324000 -1.130          0.906   768.0     749.0   1440
## 8      8 0.429000 -1.250          0.925   603.0     579.0   1240
## 9      9 0.510000 -1.340          0.940   483.0     455.0   1090
## 10    10 0.577000 -1.420          0.945   393.0     364.0    967
## 11    11 0.632000 -1.490          0.953   325.0     295.0    866
## 12    12 0.666000 -1.550          0.954   272.0     241.0    781
## 13    14 0.724000 -1.620          0.965   196.0     166.0    648
## 14    16 0.761000 -1.680          0.970   147.0     118.0    549
## 15    18 0.792000 -1.730          0.976   113.0      85.8    472
## 16    20 0.808000 -1.770          0.977    88.7      63.9    411
## 17    22 0.821000 -1.800          0.978    71.0      48.6    362
## 18    24 0.833000 -1.820          0.982    57.8      37.5    321
## 19    26 0.833000 -1.850          0.979    47.6      29.4    288</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for AH is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/AH%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in AH
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite2
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
2152
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
383
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
1283
</td>
</tr>
<tr>
<td style="text-align:left;">
blue4
</td>
<td style="text-align:right;">
141
</td>
</tr>
<tr>
<td style="text-align:left;">
blueviolet
</td>
<td style="text-align:right;">
239
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
131
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen2
</td>
<td style="text-align:right;">
193
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
147
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen3
</td>
<td style="text-align:right;">
1754
</td>
</tr>
<tr>
<td style="text-align:left;">
darkslateblue
</td>
<td style="text-align:right;">
107
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
578
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick3
</td>
<td style="text-align:right;">
417
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick4
</td>
<td style="text-align:right;">
72
</td>
</tr>
<tr>
<td style="text-align:left;">
greenyellow
</td>
<td style="text-align:right;">
1308
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
178
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred3
</td>
<td style="text-align:right;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush2
</td>
<td style="text-align:right;">
64
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcyan
</td>
<td style="text-align:right;">
310
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink2
</td>
<td style="text-align:right;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
lightslateblue
</td>
<td style="text-align:right;">
152
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta4
</td>
<td style="text-align:right;">
123
</td>
</tr>
<tr>
<td style="text-align:left;">
maroon
</td>
<td style="text-align:right;">
96
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple1
</td>
<td style="text-align:right;">
56
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple2
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite
</td>
<td style="text-align:right;">
104
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered4
</td>
<td style="text-align:right;">
1148
</td>
</tr>
<tr>
<td style="text-align:left;">
plum
</td>
<td style="text-align:right;">
317
</td>
</tr>
<tr>
<td style="text-align:left;">
plum4
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
royalblue
</td>
<td style="text-align:right;">
2177
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon1
</td>
<td style="text-align:right;">
45
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue2
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
tan
</td>
<td style="text-align:right;">
469
</td>
</tr>
<tr>
<td style="text-align:left;">
violet
</td>
<td style="text-align:right;">
137
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow3
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow4
</td>
<td style="text-align:right;">
86
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="interesting-modules-3" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Interesting
Modules</h2>
<div id="mediumpurple1-module---status" class="section level3"
number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> mediumpurple1
module - Status</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AH%20mediumpurple1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AH%20mediumpurple1%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-23-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-3" class="section level4" number="4.1.1.1">
<h4><span class="header-section-number">4.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in mediumpurple1 module in AH
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0016082" class="uri">GO:0016082</a>
</td>
<td style="text-align:left;">
synaptic vesicle priming
</td>
<td style="text-align:left;">
3/48
</td>
<td style="text-align:left;">
20/12138
</td>
<td style="text-align:right;">
0.0000631
</td>
<td style="text-align:right;">
0.0101610
</td>
<td style="text-align:right;">
0.0086999
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016079" class="uri">GO:0016079</a>
</td>
<td style="text-align:left;">
synaptic vesicle exocytosis
</td>
<td style="text-align:left;">
3/48
</td>
<td style="text-align:left;">
23/12138
</td>
<td style="text-align:right;">
0.0000972
</td>
<td style="text-align:right;">
0.0101610
</td>
<td style="text-align:right;">
0.0086999
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007009" class="uri">GO:0007009</a>
</td>
<td style="text-align:left;">
plasma membrane organization
</td>
<td style="text-align:left;">
3/48
</td>
<td style="text-align:left;">
28/12138
</td>
<td style="text-align:right;">
0.0001774
</td>
<td style="text-align:right;">
0.0123577
</td>
<td style="text-align:right;">
0.0105807
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006406" class="uri">GO:0006406</a>
</td>
<td style="text-align:left;">
mRNA export from nucleus
</td>
<td style="text-align:left;">
4/48
</td>
<td style="text-align:left;">
96/12138
</td>
<td style="text-align:right;">
0.0005477
</td>
<td style="text-align:right;">
0.0249930
</td>
<td style="text-align:right;">
0.0213992
</td>
<td style="text-align:left;">
DDX25/LOC113993656/LOC114004318/NUP160
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0061025" class="uri">GO:0061025</a>
</td>
<td style="text-align:left;">
membrane fusion
</td>
<td style="text-align:left;">
3/48
</td>
<td style="text-align:left;">
42/12138
</td>
<td style="text-align:right;">
0.0005979
</td>
<td style="text-align:right;">
0.0249930
</td>
<td style="text-align:right;">
0.0213992
</td>
<td style="text-align:left;">
LOC113987015/LOC113991140/LOC113993758
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0038063" class="uri">GO:0038063</a>
</td>
<td style="text-align:left;">
collagen-activated tyrosine kinase receptor signaling pathway
</td>
<td style="text-align:left;">
2/48
</td>
<td style="text-align:left;">
13/12138
</td>
<td style="text-align:right;">
0.0011617
</td>
<td style="text-align:right;">
0.0346858
</td>
<td style="text-align:right;">
0.0296983
</td>
<td style="text-align:left;">
LOC113998275/OTOL1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098789" class="uri">GO:0098789</a>
</td>
<td style="text-align:left;">
pre-mRNA cleavage required for polyadenylation
</td>
<td style="text-align:left;">
2/48
</td>
<td style="text-align:left;">
13/12138
</td>
<td style="text-align:right;">
0.0011617
</td>
<td style="text-align:right;">
0.0346858
</td>
<td style="text-align:right;">
0.0296983
</td>
<td style="text-align:left;">
LOC113993656/LOC114004318
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006388" class="uri">GO:0006388</a>
</td>
<td style="text-align:left;">
tRNA splicing, via endonucleolytic cleavage and ligation
</td>
<td style="text-align:left;">
2/48
</td>
<td style="text-align:left;">
15/12138
</td>
<td style="text-align:right;">
0.0015560
</td>
<td style="text-align:right;">
0.0406503
</td>
<td style="text-align:right;">
0.0348051
</td>
<td style="text-align:left;">
LOC113993656/LOC114004318
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016254" class="uri">GO:0016254</a>
</td>
<td style="text-align:left;">
preassembly of GPI anchor in ER membrane
</td>
<td style="text-align:left;">
2/48
</td>
<td style="text-align:left;">
16/12138
</td>
<td style="text-align:right;">
0.0017738
</td>
<td style="text-align:right;">
0.0411915
</td>
<td style="text-align:right;">
0.0352685
</td>
<td style="text-align:left;">
PIGV/PIGX
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006637" class="uri">GO:0006637</a>
</td>
<td style="text-align:left;">
acyl-CoA metabolic process
</td>
<td style="text-align:left;">
2/48
</td>
<td style="text-align:left;">
22/12138
</td>
<td style="text-align:right;">
0.0033633
</td>
<td style="text-align:right;">
0.0702937
</td>
<td style="text-align:right;">
0.0601860
</td>
<td style="text-align:left;">
GPAT4/HMGCL
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AH%20mediumpurple1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="tan-module---status" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> tan module -
Status</h3>
<p>Also the inverse module of the mediumpurple1 module based on the
intermodule correlation plot</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AH%20tan%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AH%20tan%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-4" class="section level4" number="4.1.2.1">
<h4><span class="header-section-number">4.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in tan module in AH
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0070125" class="uri">GO:0070125</a>
</td>
<td style="text-align:left;">
mitochondrial translational elongation
</td>
<td style="text-align:left;">
10/351
</td>
<td style="text-align:left;">
87/12138
</td>
<td style="text-align:right;">
0.0001990
</td>
<td style="text-align:right;">
0.2147199
</td>
<td style="text-align:right;">
0.2128242
</td>
<td style="text-align:left;">
CHCHD1/MRPL14/MRPL24/MRPL35/MRPL36/MRPL51/MRPL54/MRPS18C/MRPS22/TSFM
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016075" class="uri">GO:0016075</a>
</td>
<td style="text-align:left;">
rRNA catabolic process
</td>
<td style="text-align:left;">
4/351
</td>
<td style="text-align:left;">
14/12138
</td>
<td style="text-align:right;">
0.0005467
</td>
<td style="text-align:right;">
0.2949453
</td>
<td style="text-align:right;">
0.2923413
</td>
<td style="text-align:left;">
EXOSC4/EXOSC6/TRIR/XRN1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070126" class="uri">GO:0070126</a>
</td>
<td style="text-align:left;">
mitochondrial translational termination
</td>
<td style="text-align:left;">
9/351
</td>
<td style="text-align:left;">
87/12138
</td>
<td style="text-align:right;">
0.0008986
</td>
<td style="text-align:right;">
0.3231911
</td>
<td style="text-align:right;">
0.3203377
</td>
<td style="text-align:left;">
CHCHD1/MRPL14/MRPL24/MRPL35/MRPL36/MRPL51/MRPL54/MRPS18C/MRPS22
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0055088" class="uri">GO:0055088</a>
</td>
<td style="text-align:left;">
lipid homeostasis
</td>
<td style="text-align:left;">
5/351
</td>
<td style="text-align:left;">
31/12138
</td>
<td style="text-align:right;">
0.0017973
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
ABHD5/CEBPA/DDX3X/LOC114000584/LOC114001595
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090503" class="uri">GO:0090503</a>
</td>
<td style="text-align:left;">
RNA phosphodiester bond hydrolysis, exonucleolytic
</td>
<td style="text-align:left;">
5/351
</td>
<td style="text-align:left;">
33/12138
</td>
<td style="text-align:right;">
0.0023947
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
DCP2/EXOSC4/EXOSC6/TRIR/XRN1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009650" class="uri">GO:0009650</a>
</td>
<td style="text-align:left;">
UV protection
</td>
<td style="text-align:left;">
3/351
</td>
<td style="text-align:left;">
10/12138
</td>
<td style="text-align:right;">
0.0024730
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
ERCC4/LOC113992976/XPA
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071028" class="uri">GO:0071028</a>
</td>
<td style="text-align:left;">
nuclear mRNA surveillance
</td>
<td style="text-align:left;">
3/351
</td>
<td style="text-align:left;">
10/12138
</td>
<td style="text-align:right;">
0.0024730
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
EXOSC4/EXOSC6/XRN1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030308" class="uri">GO:0030308</a>
</td>
<td style="text-align:left;">
negative regulation of cell growth
</td>
<td style="text-align:left;">
9/351
</td>
<td style="text-align:left;">
104/12138
</td>
<td style="text-align:right;">
0.0031382
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
ADIPOR1/APBB1/BTG1/DDX3X/LOC113996102/LOC113999325/PPP2CA/PSRC1/VGLL4
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045930" class="uri">GO:0045930</a>
</td>
<td style="text-align:left;">
negative regulation of mitotic cell cycle
</td>
<td style="text-align:left;">
4/351
</td>
<td style="text-align:left;">
22/12138
</td>
<td style="text-align:right;">
0.0033278
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
BTG1/CDKN1C/FOXC1/LOC113996102
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009411" class="uri">GO:0009411</a>
</td>
<td style="text-align:left;">
response to UV
</td>
<td style="text-align:left;">
5/351
</td>
<td style="text-align:left;">
36/12138
</td>
<td style="text-align:right;">
0.0035441
</td>
<td style="text-align:right;">
0.3598964
</td>
<td style="text-align:right;">
0.3567189
</td>
<td style="text-align:left;">
ERCC4/ERCC6/LOC113992976/LOC114000584/LOC114001595
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AH%20tan%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="paraventricular-nucleus-pvn" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Paraventricular Nucleus
(PVN)</h1>
<pre><code>## [1] 15153</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 15 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0195 19.20          0.743  7580.0    7580.0   7660
## 2      2   0.4010 -9.84          0.711  4300.0    4240.0   4880
## 3      3   0.4960 -3.59          0.645  2660.0    2570.0   3470
## 4      4   0.6420 -2.55          0.757  1760.0    1660.0   2670
## 5      5   0.7320 -2.27          0.844  1220.0    1110.0   2180
## 6      6   0.8040 -2.09          0.901   883.0     776.0   1840
## 7      7   0.8580 -1.99          0.937   658.0     556.0   1580
## 8      8   0.8910 -1.93          0.957   504.0     408.0   1390
## 9      9   0.9080 -1.90          0.964   395.0     306.0   1240
## 10    10   0.9180 -1.87          0.967   315.0     233.0   1110
## 11    11   0.9320 -1.83          0.975   256.0     180.0   1010
## 12    12   0.9330 -1.82          0.974   211.0     141.0    924
## 13    14   0.9330 -1.79          0.969   148.0      89.5    785
## 14    16   0.9400 -1.76          0.974   108.0      59.1    679
## 15    18   0.9440 -1.73          0.976    81.4      40.2    596
## 16    20   0.9390 -1.71          0.972    62.9      28.1    528
## 17    22   0.9390 -1.69          0.974    49.7      20.2    472
## 18    24   0.9340 -1.68          0.971    39.9      14.8    424
## 19    26   0.9240 -1.68          0.964    32.6      11.0    384</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for PVN is 9.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/PVN%20correlate%20with%20traits-1.png" width="960" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in PVN
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
438
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
5377
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgreen
</td>
<td style="text-align:right;">
383
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen
</td>
<td style="text-align:right;">
348
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange
</td>
<td style="text-align:right;">
394
</td>
</tr>
<tr>
<td style="text-align:left;">
darkred
</td>
<td style="text-align:right;">
664
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
181
</td>
</tr>
<tr>
<td style="text-align:left;">
green
</td>
<td style="text-align:right;">
555
</td>
</tr>
<tr>
<td style="text-align:left;">
greenyellow
</td>
<td style="text-align:right;">
886
</td>
</tr>
<tr>
<td style="text-align:left;">
midnightblue
</td>
<td style="text-align:right;">
646
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
159
</td>
</tr>
<tr>
<td style="text-align:left;">
saddlebrown
</td>
<td style="text-align:right;">
106
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna3
</td>
<td style="text-align:right;">
73
</td>
</tr>
<tr>
<td style="text-align:left;">
steelblue
</td>
<td style="text-align:right;">
106
</td>
</tr>
<tr>
<td style="text-align:left;">
turquoise
</td>
<td style="text-align:right;">
4631
</td>
</tr>
<tr>
<td style="text-align:left;">
white
</td>
<td style="text-align:right;">
129
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>It doesn’t appear that our interesting module <em>white</em> is
associated with any other module of interest.</p>
<p><br></p>
<div id="interesting-modules-4" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Interesting
Modules</h2>
<div id="white-module---status" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> white module -
status</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PVN%20white%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/PVN%20white%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-5" class="section level4" number="5.1.1.1">
<h4><span class="header-section-number">5.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in white module in PVN
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006388" class="uri">GO:0006388</a>
</td>
<td style="text-align:left;">
tRNA splicing, via endonucleolytic cleavage and ligation
</td>
<td style="text-align:left;">
3/104
</td>
<td style="text-align:left;">
15/12249
</td>
<td style="text-align:right;">
0.0002512
</td>
<td style="text-align:right;">
0.1233371
</td>
<td style="text-align:right;">
0.1197807
</td>
<td style="text-align:left;">
CLP1/CPSF4/RTCB
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006270" class="uri">GO:0006270</a>
</td>
<td style="text-align:left;">
DNA replication initiation
</td>
<td style="text-align:left;">
3/104
</td>
<td style="text-align:left;">
28/12249
</td>
<td style="text-align:right;">
0.0016693
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
MCM3/MCM5/ORC1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032922" class="uri">GO:0032922</a>
</td>
<td style="text-align:left;">
circadian regulation of gene expression
</td>
<td style="text-align:left;">
4/104
</td>
<td style="text-align:left;">
70/12249
</td>
<td style="text-align:right;">
0.0029266
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
DRD3/LOC113996216/LOC113999640/USP2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006094" class="uri">GO:0006094</a>
</td>
<td style="text-align:left;">
gluconeogenesis
</td>
<td style="text-align:left;">
3/104
</td>
<td style="text-align:left;">
35/12249
</td>
<td style="text-align:right;">
0.0031947
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
ENO1/MDH1/PGK1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000727" class="uri">GO:0000727</a>
</td>
<td style="text-align:left;">
double-strand break repair via break-induced replication
</td>
<td style="text-align:left;">
2/104
</td>
<td style="text-align:left;">
11/12249
</td>
<td style="text-align:right;">
0.0037356
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
MCM3/MCM5
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045598" class="uri">GO:0045598</a>
</td>
<td style="text-align:left;">
regulation of fat cell differentiation
</td>
<td style="text-align:left;">
2/104
</td>
<td style="text-align:left;">
11/12249
</td>
<td style="text-align:right;">
0.0037356
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
LOC113996216/LOC113999640
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098609" class="uri">GO:0098609</a>
</td>
<td style="text-align:left;">
cell-cell adhesion
</td>
<td style="text-align:left;">
5/104
</td>
<td style="text-align:left;">
124/12249
</td>
<td style="text-align:right;">
0.0040602
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
ANXA2/FAT2/FBLIM1/LOC113988363/PDLIM1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006953" class="uri">GO:0006953</a>
</td>
<td style="text-align:left;">
acute-phase response
</td>
<td style="text-align:left;">
2/104
</td>
<td style="text-align:left;">
12/12249
</td>
<td style="text-align:right;">
0.0044580
</td>
<td style="text-align:right;">
0.2736082
</td>
<td style="text-align:right;">
0.2657187
</td>
<td style="text-align:left;">
AHSG/IL6R
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000956" class="uri">GO:0000956</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process
</td>
<td style="text-align:left;">
2/104
</td>
<td style="text-align:left;">
13/12249
</td>
<td style="text-align:right;">
0.0052395
</td>
<td style="text-align:right;">
0.2858411
</td>
<td style="text-align:right;">
0.2775989
</td>
<td style="text-align:left;">
LOC113987020/XRN2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903146" class="uri">GO:1903146</a>
</td>
<td style="text-align:left;">
regulation of autophagy of mitochondrion
</td>
<td style="text-align:left;">
2/104
</td>
<td style="text-align:left;">
15/12249
</td>
<td style="text-align:right;">
0.0069756
</td>
<td style="text-align:right;">
0.3425008
</td>
<td style="text-align:right;">
0.3326248
</td>
<td style="text-align:left;">
ATP13A2/PINK1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/PVN%20white%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="medial-preoptic-area-pom" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Medial Preoptic Area
(POM)</h1>
<p>Note that the PFT2 POM sample was removed from this analysis as it
was previously found to be an outlier.</p>
<pre><code>## [1] 15315</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 28 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0199  20.40          0.939 7660.00   7660.00 7760.0
## 2      2   0.4150 -33.70          0.906 4170.00   4160.00 4490.0
## 3      3   0.4780 -16.50          0.936 2420.00   2410.00 2830.0
## 4      4   0.4710  -9.84          0.939 1490.00   1470.00 1900.0
## 5      5   0.4360  -6.51          0.928  951.00    934.00 1340.0
## 6      6   0.4420  -4.83          0.924  632.00    615.00  975.0
## 7      7   0.4810  -4.06          0.930  433.00    418.00  733.0
## 8      8   0.5500  -3.84          0.946  304.00    291.00  573.0
## 9      9   0.6080  -3.71          0.956  219.00    207.00  456.0
## 10    10   0.6650  -3.68          0.966  161.00    151.00  370.0
## 11    11   0.7080  -3.62          0.972  120.00    111.00  304.0
## 12    12   0.7390  -3.53          0.975   91.50     83.60  253.0
## 13    14   0.8040  -3.36          0.986   55.10     48.90  181.0
## 14    16   0.8520  -3.19          0.992   34.80     29.90  134.0
## 15    18   0.8910  -3.01          0.996   22.80     19.00  102.0
## 16    20   0.9160  -2.91          0.997   15.50     12.50   79.4
## 17    22   0.9350  -2.81          0.998   10.80      8.42   63.3
## 18    24   0.9490  -2.73          0.997    7.74      5.83   51.7
## 19    26   0.9530  -2.72          0.997    5.67      4.10   44.3</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for POM is 20.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/POM%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite1
</td>
<td style="text-align:right;">
157
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
199
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
316
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
769
</td>
</tr>
<tr>
<td style="text-align:left;">
blue3
</td>
<td style="text-align:right;">
48
</td>
</tr>
<tr>
<td style="text-align:left;">
blue4
</td>
<td style="text-align:right;">
164
</td>
</tr>
<tr>
<td style="text-align:left;">
blueviolet
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
881
</td>
</tr>
<tr>
<td style="text-align:left;">
brown1
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
brown2
</td>
<td style="text-align:right;">
661
</td>
</tr>
<tr>
<td style="text-align:left;">
chocolate4
</td>
<td style="text-align:right;">
144
</td>
</tr>
<tr>
<td style="text-align:left;">
coral
</td>
<td style="text-align:right;">
2067
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
155
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
185
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
233
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen1
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen2
</td>
<td style="text-align:right;">
225
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange
</td>
<td style="text-align:right;">
674
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
97
</td>
</tr>
<tr>
<td style="text-align:left;">
darkred
</td>
<td style="text-align:right;">
164
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen1
</td>
<td style="text-align:right;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen2
</td>
<td style="text-align:right;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen3
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen4
</td>
<td style="text-align:right;">
145
</td>
</tr>
<tr>
<td style="text-align:left;">
darkslateblue
</td>
<td style="text-align:right;">
380
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
158
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
71
</td>
</tr>
<tr>
<td style="text-align:left;">
deeppink1
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick2
</td>
<td style="text-align:right;">
106
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick3
</td>
<td style="text-align:right;">
153
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick4
</td>
<td style="text-align:right;">
74
</td>
</tr>
<tr>
<td style="text-align:left;">
floralwhite
</td>
<td style="text-align:right;">
391
</td>
</tr>
<tr>
<td style="text-align:left;">
green4
</td>
<td style="text-align:right;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred4
</td>
<td style="text-align:right;">
283
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush
</td>
<td style="text-align:right;">
43
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush2
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
lightblue3
</td>
<td style="text-align:right;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcoral
</td>
<td style="text-align:right;">
433
</td>
</tr>
<tr>
<td style="text-align:left;">
lightgreen
</td>
<td style="text-align:right;">
177
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink1
</td>
<td style="text-align:right;">
111
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink2
</td>
<td style="text-align:right;">
108
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink4
</td>
<td style="text-align:right;">
88
</td>
</tr>
<tr>
<td style="text-align:left;">
lightslateblue
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
lightsteelblue
</td>
<td style="text-align:right;">
1161
</td>
</tr>
<tr>
<td style="text-align:left;">
lightsteelblue1
</td>
<td style="text-align:right;">
163
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta3
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
maroon
</td>
<td style="text-align:right;">
152
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple1
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple4
</td>
<td style="text-align:right;">
156
</td>
</tr>
<tr>
<td style="text-align:left;">
moccasin
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite
</td>
<td style="text-align:right;">
334
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite1
</td>
<td style="text-align:right;">
203
</td>
</tr>
<tr>
<td style="text-align:left;">
orange
</td>
<td style="text-align:right;">
251
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered1
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
pink3
</td>
<td style="text-align:right;">
52
</td>
</tr>
<tr>
<td style="text-align:left;">
pink4
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
plum1
</td>
<td style="text-align:right;">
107
</td>
</tr>
<tr>
<td style="text-align:left;">
powderblue
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
purple
</td>
<td style="text-align:right;">
1021
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon
</td>
<td style="text-align:right;">
216
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon2
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna3
</td>
<td style="text-align:right;">
112
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue
</td>
<td style="text-align:right;">
134
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue1
</td>
<td style="text-align:right;">
82
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue4
</td>
<td style="text-align:right;">
109
</td>
</tr>
<tr>
<td style="text-align:left;">
tan3
</td>
<td style="text-align:right;">
45
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-31-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="interesting-modules-5" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Interesting
Modules</h2>
<div id="floralwhite-module---statusmean-t" class="section level3"
number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> floralwhite module
- Status/mean T</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20floralwhite%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/POM%20floralwhite%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-6" class="section level4" number="6.1.1.1">
<h4><span class="header-section-number">6.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in floralwhite module in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007186" class="uri">GO:0007186</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
30/309
</td>
<td style="text-align:left;">
495/12339
</td>
<td style="text-align:right;">
0.0000069
</td>
<td style="text-align:right;">
0.0074407
</td>
<td style="text-align:right;">
0.0071573
</td>
<td style="text-align:left;">
CXCR2/FZD9/GAL/GALR1/GHRH/GNAI2/GNRH1/GPR21/GPR63/GRPR/HRH2/HTR5A/IAPP/JAK2/LHCGR/LOC113982386/LOC113993087/LOC113993299/LOC114002880/MCHR1/OPRL1/PDE3B/PRLH/QRFPR/RGS20/RORB/RXFP3/SST/SSTR2/UTS2B
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007218" class="uri">GO:0007218</a>
</td>
<td style="text-align:left;">
neuropeptide signaling pathway
</td>
<td style="text-align:left;">
10/309
</td>
<td style="text-align:left;">
88/12339
</td>
<td style="text-align:right;">
0.0000672
</td>
<td style="text-align:right;">
0.0362788
</td>
<td style="text-align:right;">
0.0348968
</td>
<td style="text-align:left;">
GAL/GALR1/GRPR/LOC113993087/MCHR1/OPRL1/QRFPR/RXFP3/SCG5/SSTR2
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051560" class="uri">GO:0051560</a>
</td>
<td style="text-align:left;">
mitochondrial calcium ion homeostasis
</td>
<td style="text-align:left;">
4/309
</td>
<td style="text-align:left;">
16/12339
</td>
<td style="text-align:right;">
0.0005532
</td>
<td style="text-align:right;">
0.1989591
</td>
<td style="text-align:right;">
0.1913796
</td>
<td style="text-align:left;">
IMMT/MAIP1/MCUB/MICU3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019915" class="uri">GO:0019915</a>
</td>
<td style="text-align:left;">
lipid storage
</td>
<td style="text-align:left;">
4/309
</td>
<td style="text-align:left;">
18/12339
</td>
<td style="text-align:right;">
0.0008940
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
CD36/LDAH/NRIP1/PNPLA2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007611" class="uri">GO:0007611</a>
</td>
<td style="text-align:left;">
learning or memory
</td>
<td style="text-align:left;">
6/309
</td>
<td style="text-align:left;">
50/12339
</td>
<td style="text-align:right;">
0.0014771
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
CHRNA7/CRHBP/FZD9/GRPR/LOC113982386/PAFAH1B1
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
3/309
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0016376
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
AR/NR3C1/PGR
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036444" class="uri">GO:0036444</a>
</td>
<td style="text-align:left;">
calcium import into the mitochondrion
</td>
<td style="text-align:left;">
3/309
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0016376
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
MAIP1/MCUB/MICU3
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007399" class="uri">GO:0007399</a>
</td>
<td style="text-align:left;">
nervous system development
</td>
<td style="text-align:left;">
16/309
</td>
<td style="text-align:left;">
280/12339
</td>
<td style="text-align:right;">
0.0018609
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
ACTL6B/BPNT1/FGF13/FZD9/GDPD5/GFRA3/GPSM1/LOC113991778/LSAMP/NEURL1B/NRN1/NRN1L/OLFM1/PCSK2/PHF10/PLPPR1
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007565" class="uri">GO:0007565</a>
</td>
<td style="text-align:left;">
female pregnancy
</td>
<td style="text-align:left;">
6/309
</td>
<td style="text-align:left;">
53/12339
</td>
<td style="text-align:right;">
0.0020045
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
CORIN/CRHBP/GNRH1/IGFBP2/LOC113982386/PAPPA
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090280" class="uri">GO:0090280</a>
</td>
<td style="text-align:left;">
positive regulation of calcium ion import
</td>
<td style="text-align:left;">
3/309
</td>
<td style="text-align:left;">
11/12339
</td>
<td style="text-align:right;">
0.0022102
</td>
<td style="text-align:right;">
0.2384799
</td>
<td style="text-align:right;">
0.2293948
</td>
<td style="text-align:left;">
LOC113982386/LOC114000001/STC1
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20floralwhite%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20floralwhite%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="pink3-module---strength" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> pink3 module -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20pink3%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/POM%20pink3%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-7" class="section level4" number="6.1.2.1">
<h4><span class="header-section-number">6.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in pink3 module in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0036444" class="uri">GO:0036444</a>
</td>
<td style="text-align:left;">
calcium import into the mitochondrion
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0004767
</td>
<td style="text-align:right;">
0.0533864
</td>
<td style="text-align:right;">
0.0446343
</td>
<td style="text-align:left;">
OPA1/SMDT1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015693" class="uri">GO:0015693</a>
</td>
<td style="text-align:left;">
magnesium ion transport
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
12/12339
</td>
<td style="text-align:right;">
0.0006962
</td>
<td style="text-align:right;">
0.0533864
</td>
<td style="text-align:right;">
0.0446343
</td>
<td style="text-align:left;">
NIPAL3/TUSC3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903830" class="uri">GO:1903830</a>
</td>
<td style="text-align:left;">
magnesium ion transmembrane transport
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
13/12339
</td>
<td style="text-align:right;">
0.0008210
</td>
<td style="text-align:right;">
0.0533864
</td>
<td style="text-align:right;">
0.0446343
</td>
<td style="text-align:left;">
NIPAL3/TUSC3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048208" class="uri">GO:0048208</a>
</td>
<td style="text-align:left;">
COPII vesicle coating
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
60/12339
</td>
<td style="text-align:right;">
0.0010217
</td>
<td style="text-align:right;">
0.0533864
</td>
<td style="text-align:right;">
0.0446343
</td>
<td style="text-align:left;">
CD59/CTSC/PDCD6
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006094" class="uri">GO:0006094</a>
</td>
<td style="text-align:left;">
gluconeogenesis
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
36/12339
</td>
<td style="text-align:right;">
0.0063185
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
GOT1/PGAM1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006110" class="uri">GO:0006110</a>
</td>
<td style="text-align:left;">
regulation of glycolytic process
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
40/12339
</td>
<td style="text-align:right;">
0.0077576
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
NUP35/PGAM1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0022900" class="uri">GO:0022900</a>
</td>
<td style="text-align:left;">
electron transport chain
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
46/12339
</td>
<td style="text-align:right;">
0.0101654
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
LOC114003129/NDUFAF2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006487" class="uri">GO:0006487</a>
</td>
<td style="text-align:left;">
protein N-linked glycosylation
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
48/12339
</td>
<td style="text-align:right;">
0.0110326
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
DOLPP1/TUSC3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032981" class="uri">GO:0032981</a>
</td>
<td style="text-align:left;">
mitochondrial respiratory chain complex I assembly
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
49/12339
</td>
<td style="text-align:right;">
0.0114780
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
NDUFAF2/NDUFB1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036297" class="uri">GO:0036297</a>
</td>
<td style="text-align:left;">
interstrand cross-link repair
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
50/12339
</td>
<td style="text-align:right;">
0.0119313
</td>
<td style="text-align:right;">
0.1905773
</td>
<td style="text-align:right;">
0.1593344
</td>
<td style="text-align:left;">
FANCE/NEIL3
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20pink3%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="mediumpurple1---mean-testosterone" class="section level3"
number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> mediumpurple1 -
mean testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20mediumpurple1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/POM%20mediumpurple1%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-8" class="section level4" number="6.1.3.1">
<h4><span class="header-section-number">6.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in mediumpurple1 module in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007084" class="uri">GO:0007084</a>
</td>
<td style="text-align:left;">
mitotic nuclear envelope reassembly
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0004767
</td>
<td style="text-align:right;">
0.0517006
</td>
<td style="text-align:right;">
0.0420245
</td>
<td style="text-align:left;">
LMNA/PPP2R2A
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033622" class="uri">GO:0033622</a>
</td>
<td style="text-align:left;">
integrin activation
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0004767
</td>
<td style="text-align:right;">
0.0517006
</td>
<td style="text-align:right;">
0.0420245
</td>
<td style="text-align:left;">
LOC113991759/TLN1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036498" class="uri">GO:0036498</a>
</td>
<td style="text-align:left;">
IRE1-mediated unfolded protein response
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
50/12339
</td>
<td style="text-align:right;">
0.0005988
</td>
<td style="text-align:right;">
0.0517006
</td>
<td style="text-align:right;">
0.0420245
</td>
<td style="text-align:left;">
LMNA/LOC113991759/TLN1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031398" class="uri">GO:0031398</a>
</td>
<td style="text-align:left;">
positive regulation of protein ubiquitination
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
66/12339
</td>
<td style="text-align:right;">
0.0013476
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
AXIN1/LOC113992269/TRAF6
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032147" class="uri">GO:0032147</a>
</td>
<td style="text-align:left;">
activation of protein kinase activity
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
67/12339
</td>
<td style="text-align:right;">
0.0014075
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
AXIN1/LOC113985569/TRAF6
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000060" class="uri">GO:2000060</a>
</td>
<td style="text-align:left;">
positive regulation of ubiquitin-dependent protein catabolic process
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
19/12339
</td>
<td style="text-align:right;">
0.0017773
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
AXIN1/LOC113992269
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006936" class="uri">GO:0006936</a>
</td>
<td style="text-align:left;">
muscle contraction
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
81/12339
</td>
<td style="text-align:right;">
0.0024275
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
LOC113991759/PXN/TLN1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007160" class="uri">GO:0007160</a>
</td>
<td style="text-align:left;">
cell-matrix adhesion
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
82/12339
</td>
<td style="text-align:right;">
0.0025139
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
ADAMTS13/PXN/SNED1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007229" class="uri">GO:0007229</a>
</td>
<td style="text-align:left;">
integrin-mediated signaling pathway
</td>
<td style="text-align:left;">
3/41
</td>
<td style="text-align:left;">
82/12339
</td>
<td style="text-align:right;">
0.0025139
</td>
<td style="text-align:right;">
0.0723435
</td>
<td style="text-align:right;">
0.0588039
</td>
<td style="text-align:left;">
ADAMTS13/LOC113991759/TLN1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030866" class="uri">GO:0030866</a>
</td>
<td style="text-align:left;">
cortical actin cytoskeleton organization
</td>
<td style="text-align:left;">
2/41
</td>
<td style="text-align:left;">
31/12339
</td>
<td style="text-align:right;">
0.0047128
</td>
<td style="text-align:right;">
0.1220611
</td>
<td style="text-align:right;">
0.0992165
</td>
<td style="text-align:left;">
LOC113991759/TLN1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20mediumpurple1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="sienna3---strength" class="section level3" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> sienna3 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20sienna3%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/POM%20sienna3%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-35-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-9" class="section level4" number="6.1.4.1">
<h4><span class="header-section-number">6.1.4.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in sienna3 module in POM
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0030433" class="uri">GO:0030433</a>
</td>
<td style="text-align:left;">
ubiquitin-dependent ERAD pathway
</td>
<td style="text-align:left;">
4/92
</td>
<td style="text-align:left;">
58/12339
</td>
<td style="text-align:right;">
0.0009028
</td>
<td style="text-align:right;">
0.2416790
</td>
<td style="text-align:right;">
0.2368751
</td>
<td style="text-align:left;">
DNAJB14/LOC113995895/RNF103/UBE2G2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006891" class="uri">GO:0006891</a>
</td>
<td style="text-align:left;">
intra-Golgi vesicle-mediated transport
</td>
<td style="text-align:left;">
3/92
</td>
<td style="text-align:left;">
28/12339
</td>
<td style="text-align:right;">
0.0011481
</td>
<td style="text-align:right;">
0.2416790
</td>
<td style="text-align:right;">
0.2368751
</td>
<td style="text-align:left;">
GABARAPL2/GOSR1/NAPG
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006885" class="uri">GO:0006885</a>
</td>
<td style="text-align:left;">
regulation of pH
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
10/12339
</td>
<td style="text-align:right;">
0.0023802
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
EDNRB/SLC26A4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045600" class="uri">GO:0045600</a>
</td>
<td style="text-align:left;">
positive regulation of fat cell differentiation
</td>
<td style="text-align:left;">
3/92
</td>
<td style="text-align:left;">
41/12339
</td>
<td style="text-align:right;">
0.0034838
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
CREBL2/LOC113993849/SYAP1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031573" class="uri">GO:0031573</a>
</td>
<td style="text-align:left;">
intra-S DNA damage checkpoint
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
13/12339
</td>
<td style="text-align:right;">
0.0040661
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
ATF2/EME1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006112" class="uri">GO:0006112</a>
</td>
<td style="text-align:left;">
energy reserve metabolic process
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
15/12339
</td>
<td style="text-align:right;">
0.0054208
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
GNAS/KL
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031396" class="uri">GO:0031396</a>
</td>
<td style="text-align:left;">
regulation of protein ubiquitination
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
15/12339
</td>
<td style="text-align:right;">
0.0054208
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
DCUN1D1/PINK1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033146" class="uri">GO:0033146</a>
</td>
<td style="text-align:left;">
regulation of intracellular estrogen receptor signaling pathway
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
15/12339
</td>
<td style="text-align:right;">
0.0054208
</td>
<td style="text-align:right;">
0.2852676
</td>
<td style="text-align:right;">
0.2795972
</td>
<td style="text-align:left;">
LOC113993849/UFSP2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045071" class="uri">GO:0045071</a>
</td>
<td style="text-align:left;">
negative regulation of viral genome replication
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
17/12339
</td>
<td style="text-align:right;">
0.0069536
</td>
<td style="text-align:right;">
0.3252720
</td>
<td style="text-align:right;">
0.3188064
</td>
<td style="text-align:left;">
BTBD17/EIF2AK2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051443" class="uri">GO:0051443</a>
</td>
<td style="text-align:left;">
positive regulation of ubiquitin-protein transferase activity
</td>
<td style="text-align:left;">
2/92
</td>
<td style="text-align:left;">
18/12339
</td>
<td style="text-align:right;">
0.0077850
</td>
<td style="text-align:right;">
0.3277496
</td>
<td style="text-align:right;">
0.3212347
</td>
<td style="text-align:left;">
DCUN1D1/PINK1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/POM%20sienna3%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="dorsomedial-intercollicular-nucleus-ico" class="section level1"
number="7">
<h1><span class="header-section-number">7</span> Dorsomedial
Intercollicular Nucleus (ICo)</h1>
<pre><code>## [1] 15071</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0312 25.70          0.970  7540.0    7540.0   7670
## 2      2   0.0490 -4.86          0.903  4250.0    4230.0   4730
## 3      3   0.1430 -3.42          0.860  2600.0    2570.0   3280
## 4      4   0.2250 -2.48          0.856  1700.0    1660.0   2420
## 5      5   0.3260 -2.05          0.880  1160.0    1120.0   1860
## 6      6   0.4310 -1.98          0.902   830.0     788.0   1490
## 7      7   0.5200 -2.00          0.915   610.0     568.0   1230
## 8      8   0.5820 -2.02          0.918   460.0     420.0   1030
## 9      9   0.6320 -2.04          0.923   355.0     317.0    883
## 10    10   0.6730 -2.04          0.931   279.0     243.0    762
## 11    11   0.6950 -2.05          0.932   223.0     189.0    665
## 12    12   0.7230 -2.04          0.939   180.0     149.0    584
## 13    14   0.7460 -2.06          0.940   123.0      95.7    462
## 14    16   0.7750 -2.04          0.949    86.8      63.6    373
## 15    18   0.7800 -2.05          0.947    63.4      43.6    306
## 16    20   0.7980 -2.02          0.952    47.6      30.7    255
## 17    22   0.8110 -1.99          0.956    36.5      22.1    215
## 18    24   0.8190 -1.97          0.956    28.5      16.3    183
## 19    26   0.8340 -1.92          0.960    22.7      12.2    157</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for ICO is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## [1] TRUE</code></pre>
<img src="05_WGCNA_single_tissue_files/figure-html/ICO%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in ICO
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
1297
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
brown2
</td>
<td style="text-align:right;">
618
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
1201
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
1339
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
68
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
513
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen3
</td>
<td style="text-align:right;">
2480
</td>
</tr>
<tr>
<td style="text-align:left;">
darkturquoise
</td>
<td style="text-align:right;">
363
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
631
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick4
</td>
<td style="text-align:right;">
265
</td>
</tr>
<tr>
<td style="text-align:left;">
greenyellow
</td>
<td style="text-align:right;">
521
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew
</td>
<td style="text-align:right;">
45
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
85
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred4
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush2
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush3
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink3
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
maroon
</td>
<td style="text-align:right;">
237
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple3
</td>
<td style="text-align:right;">
116
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite1
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered4
</td>
<td style="text-align:right;">
2718
</td>
</tr>
<tr>
<td style="text-align:left;">
palevioletred3
</td>
<td style="text-align:right;">
90
</td>
</tr>
<tr>
<td style="text-align:left;">
plum3
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon4
</td>
<td style="text-align:right;">
92
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue
</td>
<td style="text-align:right;">
169
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle1
</td>
<td style="text-align:right;">
93
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle3
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
turquoise
</td>
<td style="text-align:right;">
1097
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There appears to be no correlations between modules of interest.</p>
<div id="interesting-modules-6" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Interesting
Modules</h2>
<div id="lavenderblush2---strength" class="section level3"
number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> lavenderblush2 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20lavenderblush2%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20lavenderblush2%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-10" class="section level4" number="7.1.1.1">
<h4><span class="header-section-number">7.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in lavenderblush2 module in ICO
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0089718" class="uri">GO:0089718</a>
</td>
<td style="text-align:left;">
amino acid import across plasma membrane
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
10/12201
</td>
<td style="text-align:right;">
0.0003966
</td>
<td style="text-align:right;">
0.0662263
</td>
<td style="text-align:right;">
0.0563539
</td>
<td style="text-align:left;">
LOC114001992/SLC22A2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007265" class="uri">GO:0007265</a>
</td>
<td style="text-align:left;">
Ras protein signal transduction
</td>
<td style="text-align:left;">
3/37
</td>
<td style="text-align:left;">
65/12201
</td>
<td style="text-align:right;">
0.0009853
</td>
<td style="text-align:right;">
0.0822706
</td>
<td style="text-align:right;">
0.0700065
</td>
<td style="text-align:left;">
CDKN1A/JUN/NRAS
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042908" class="uri">GO:0042908</a>
</td>
<td style="text-align:left;">
xenobiotic transport
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
22/12201
</td>
<td style="text-align:right;">
0.0019896
</td>
<td style="text-align:right;">
0.1075022
</td>
<td style="text-align:right;">
0.0914768
</td>
<td style="text-align:left;">
LOC114001992/SLC22A2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051591" class="uri">GO:0051591</a>
</td>
<td style="text-align:left;">
response to cAMP
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
28/12201
</td>
<td style="text-align:right;">
0.0032186
</td>
<td style="text-align:right;">
0.1075022
</td>
<td style="text-align:right;">
0.0914768
</td>
<td style="text-align:left;">
JUN/LOC113993849
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071479" class="uri">GO:0071479</a>
</td>
<td style="text-align:left;">
cellular response to ionizing radiation
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
28/12201
</td>
<td style="text-align:right;">
0.0032186
</td>
<td style="text-align:right;">
0.1075022
</td>
<td style="text-align:right;">
0.0914768
</td>
<td style="text-align:left;">
CDKN1A/SWI5
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000079" class="uri">GO:0000079</a>
</td>
<td style="text-align:left;">
regulation of cyclin-dependent protein serine/threonine kinase activity
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
43/12201
</td>
<td style="text-align:right;">
0.0074726
</td>
<td style="text-align:right;">
0.1853735
</td>
<td style="text-align:right;">
0.1577398
</td>
<td style="text-align:left;">
CDKN1A/CNPPD1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006977" class="uri">GO:0006977</a>
</td>
<td style="text-align:left;">
DNA damage response, signal transduction by p53 class mediator resulting
in cell cycle arrest
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
47/12201
</td>
<td style="text-align:right;">
0.0088778
</td>
<td style="text-align:right;">
0.1853735
</td>
<td style="text-align:right;">
0.1577398
</td>
<td style="text-align:left;">
CDKN1A/LOC113993849
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090305" class="uri">GO:0090305</a>
</td>
<td style="text-align:left;">
nucleic acid phosphodiester bond hydrolysis
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
52/12201
</td>
<td style="text-align:right;">
0.0107870
</td>
<td style="text-align:right;">
0.1853735
</td>
<td style="text-align:right;">
0.1577398
</td>
<td style="text-align:left;">
LOC113989190/ZRANB3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006606" class="uri">GO:0006606</a>
</td>
<td style="text-align:left;">
protein import into nucleus
</td>
<td style="text-align:left;">
2/37
</td>
<td style="text-align:left;">
65/12201
</td>
<td style="text-align:right;">
0.0165091
</td>
<td style="text-align:right;">
0.1853735
</td>
<td style="text-align:right;">
0.1577398
</td>
<td style="text-align:left;">
CDKN1A/SNUPN
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006098" class="uri">GO:0006098</a>
</td>
<td style="text-align:left;">
pentose-phosphate shunt
</td>
<td style="text-align:left;">
1/37
</td>
<td style="text-align:left;">
10/12201
</td>
<td style="text-align:right;">
0.0299258
</td>
<td style="text-align:right;">
0.1853735
</td>
<td style="text-align:right;">
0.1577398
</td>
<td style="text-align:left;">
RPIA
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20lavenderblush2%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20lavenderblush2%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="darkturquoise---mean-testosterone" class="section level3"
number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> darkturquoise -
mean testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20darkturquoise%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20darkturquoise%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-11" class="section level4" number="7.1.2.1">
<h4><span class="header-section-number">7.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkturquoise module in ICO
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0043392" class="uri">GO:0043392</a>
</td>
<td style="text-align:left;">
negative regulation of DNA binding
</td>
<td style="text-align:left;">
4/289
</td>
<td style="text-align:left;">
21/12201
</td>
<td style="text-align:right;">
0.0013420
</td>
<td style="text-align:right;">
0.6394570
</td>
<td style="text-align:right;">
0.6393909
</td>
<td style="text-align:left;">
CTNNBIP1/HDAC2/LOC113990539/NFKBIA
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034765" class="uri">GO:0034765</a>
</td>
<td style="text-align:left;">
regulation of ion transmembrane transport
</td>
<td style="text-align:left;">
8/289
</td>
<td style="text-align:left;">
93/12201
</td>
<td style="text-align:right;">
0.0016049
</td>
<td style="text-align:right;">
0.6394570
</td>
<td style="text-align:right;">
0.6393909
</td>
<td style="text-align:left;">
GPR89B/KCNA6/KCNG3/KCNH8/KCNJ6/KCNJ8/LOC113993165/TMEM37
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036499" class="uri">GO:0036499</a>
</td>
<td style="text-align:left;">
PERK-mediated unfolded protein response
</td>
<td style="text-align:left;">
3/289
</td>
<td style="text-align:left;">
11/12201
</td>
<td style="text-align:right;">
0.0018845
</td>
<td style="text-align:right;">
0.6394570
</td>
<td style="text-align:right;">
0.6393909
</td>
<td style="text-align:left;">
ATF3/LOC113998959/LOC114003207
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007263" class="uri">GO:0007263</a>
</td>
<td style="text-align:left;">
nitric oxide mediated signal transduction
</td>
<td style="text-align:left;">
3/289
</td>
<td style="text-align:left;">
14/12201
</td>
<td style="text-align:right;">
0.0039439
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
CD36/NPY2R/RASD1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030517" class="uri">GO:0030517</a>
</td>
<td style="text-align:left;">
negative regulation of axon extension
</td>
<td style="text-align:left;">
3/289
</td>
<td style="text-align:left;">
14/12201
</td>
<td style="text-align:right;">
0.0039439
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
KIAA0319/RNF6/RTN4R
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006370" class="uri">GO:0006370</a>
</td>
<td style="text-align:left;">
7-methylguanosine mRNA capping
</td>
<td style="text-align:left;">
4/289
</td>
<td style="text-align:left;">
31/12201
</td>
<td style="text-align:right;">
0.0058603
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
GTF2F2/LOC113983394/POLR2J/RNGTT
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042246" class="uri">GO:0042246</a>
</td>
<td style="text-align:left;">
tissue regeneration
</td>
<td style="text-align:left;">
3/289
</td>
<td style="text-align:left;">
17/12201
</td>
<td style="text-align:right;">
0.0069911
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
GAP43/LOC113998959/LOC114003207
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043254" class="uri">GO:0043254</a>
</td>
<td style="text-align:left;">
regulation of protein-containing complex assembly
</td>
<td style="text-align:left;">
3/289
</td>
<td style="text-align:left;">
17/12201
</td>
<td style="text-align:right;">
0.0069911
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
CD36/HJURP/INSM1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032355" class="uri">GO:0032355</a>
</td>
<td style="text-align:left;">
response to estradiol
</td>
<td style="text-align:left;">
6/289
</td>
<td style="text-align:left;">
75/12201
</td>
<td style="text-align:right;">
0.0086076
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
CASP8/DHH/EZH2/HPGD/MBD3/TXNIP
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032496" class="uri">GO:0032496</a>
</td>
<td style="text-align:left;">
response to lipopolysaccharide
</td>
<td style="text-align:left;">
7/289
</td>
<td style="text-align:left;">
100/12201
</td>
<td style="text-align:right;">
0.0095616
</td>
<td style="text-align:right;">
0.7171947
</td>
<td style="text-align:right;">
0.7171206
</td>
<td style="text-align:left;">
CASP8/CSF2RB/DUSP10/HDAC2/HPGD/KCNJ8/LOC113998857
</td>
<td style="text-align:right;">
7
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20darkturquoise%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="maroon---mean-testosterone" class="section level3"
number="7.1.3">
<h3><span class="header-section-number">7.1.3</span> maroon - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20maroon%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20maroon%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-41-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-12" class="section level4" number="7.1.3.1">
<h4><span class="header-section-number">7.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in maroon module in ICO
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006508" class="uri">GO:0006508</a>
</td>
<td style="text-align:left;">
proteolysis
</td>
<td style="text-align:left;">
17/205
</td>
<td style="text-align:left;">
250/12201
</td>
<td style="text-align:right;">
0.0000011
</td>
<td style="text-align:right;">
0.0008534
</td>
<td style="text-align:right;">
0.0007831
</td>
<td style="text-align:left;">
ACE/ADAM8/CAPN3/CAPN5/LGMN/LOC113983631/LOC114004136/MASP1/METAP1D/MMP2/MMP28/PAPLN/PM20D1/PRSS23/RHBDF1/TINAG/USP25
</td>
<td style="text-align:right;">
17
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060078" class="uri">GO:0060078</a>
</td>
<td style="text-align:left;">
regulation of postsynaptic membrane potential
</td>
<td style="text-align:left;">
6/205
</td>
<td style="text-align:left;">
38/12201
</td>
<td style="text-align:right;">
0.0000369
</td>
<td style="text-align:right;">
0.0149480
</td>
<td style="text-align:right;">
0.0137169
</td>
<td style="text-align:left;">
GABRA5/GABRA6/GABRB2/GRIK1/GRIN3A/LOC113990089
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060384" class="uri">GO:0060384</a>
</td>
<td style="text-align:left;">
innervation
</td>
<td style="text-align:left;">
4/205
</td>
<td style="text-align:left;">
15/12201
</td>
<td style="text-align:right;">
0.0000914
</td>
<td style="text-align:right;">
0.0194990
</td>
<td style="text-align:right;">
0.0178931
</td>
<td style="text-align:left;">
GABRA5/GABRB2/LRIG1/SULF2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030514" class="uri">GO:0030514</a>
</td>
<td style="text-align:left;">
negative regulation of BMP signaling pathway
</td>
<td style="text-align:left;">
6/205
</td>
<td style="text-align:left;">
45/12201
</td>
<td style="text-align:right;">
0.0000986
</td>
<td style="text-align:right;">
0.0194990
</td>
<td style="text-align:right;">
0.0178931
</td>
<td style="text-align:left;">
BMPER/CTDSPL2/SFRP2/SKOR2/SOSTDC1/VWC2L
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051932" class="uri">GO:0051932</a>
</td>
<td style="text-align:left;">
synaptic transmission, GABAergic
</td>
<td style="text-align:left;">
4/205
</td>
<td style="text-align:left;">
16/12201
</td>
<td style="text-align:right;">
0.0001202
</td>
<td style="text-align:right;">
0.0194990
</td>
<td style="text-align:right;">
0.0178931
</td>
<td style="text-align:left;">
GABRA5/GABRA6/GABRB2/LOC113990089
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007215" class="uri">GO:0007215</a>
</td>
<td style="text-align:left;">
glutamate receptor signaling pathway
</td>
<td style="text-align:left;">
4/205
</td>
<td style="text-align:left;">
18/12201
</td>
<td style="text-align:right;">
0.0001969
</td>
<td style="text-align:right;">
0.0236565
</td>
<td style="text-align:right;">
0.0217083
</td>
<td style="text-align:left;">
GRID2/GRIK1/GRIN2A/PLCB1
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030198" class="uri">GO:0030198</a>
</td>
<td style="text-align:left;">
extracellular matrix organization
</td>
<td style="text-align:left;">
12/205
</td>
<td style="text-align:left;">
208/12201
</td>
<td style="text-align:right;">
0.0002042
</td>
<td style="text-align:right;">
0.0236565
</td>
<td style="text-align:right;">
0.0217083
</td>
<td style="text-align:left;">
ADAMTS12/BCAN/COL6A1/EGFLAM/LAMA3/LOC113998335/MMP2/MMP28/NDNF/PAPLN/PTX3/VWA1
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042472" class="uri">GO:0042472</a>
</td>
<td style="text-align:left;">
inner ear morphogenesis
</td>
<td style="text-align:left;">
5/205
</td>
<td style="text-align:left;">
36/12201
</td>
<td style="text-align:right;">
0.0003147
</td>
<td style="text-align:right;">
0.0318980
</td>
<td style="text-align:right;">
0.0292710
</td>
<td style="text-align:left;">
GATA2/NTN1/OTX1/PAX2/SPRY2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090102" class="uri">GO:0090102</a>
</td>
<td style="text-align:left;">
cochlea development
</td>
<td style="text-align:left;">
4/205
</td>
<td style="text-align:left;">
23/12201
</td>
<td style="text-align:right;">
0.0005334
</td>
<td style="text-align:right;">
0.0475213
</td>
<td style="text-align:right;">
0.0436078
</td>
<td style="text-align:left;">
GABRA5/GABRB2/GATA2/PAX2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007268" class="uri">GO:0007268</a>
</td>
<td style="text-align:left;">
chemical synaptic transmission
</td>
<td style="text-align:left;">
11/205
</td>
<td style="text-align:left;">
201/12201
</td>
<td style="text-align:right;">
0.0005860
</td>
<td style="text-align:right;">
0.0475213
</td>
<td style="text-align:right;">
0.0436078
</td>
<td style="text-align:left;">
CACNB4/DTNA/GABRA5/GABRA6/GABRB2/GRIK1/GRIN2A/GRM4/LOC113990089/NPTX1/NTSR1
</td>
<td style="text-align:right;">
11
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20maroon%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/ICO%20maroon%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="midbrain-central-grey-gct" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Midbrain Central Grey
(GCt)</h1>
<pre><code>## [1] 15099</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 11717 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1  0.00530 -12.10          0.974  7550.0    7550.0   7660
## 2      2  0.00467  -1.64          0.886  4270.0    4270.0   4610
## 3      3  0.11400  -3.30          0.881  2630.0    2630.0   3140
## 4      4  0.15500  -2.43          0.877  1730.0    1720.0   2300
## 5      5  0.21600  -2.10          0.884  1190.0    1180.0   1760
## 6      6  0.29000  -1.97          0.894   852.0     837.0   1400
## 7      7  0.36600  -1.90          0.905   630.0     612.0   1140
## 8      8  0.43600  -1.87          0.916   478.0     458.0    947
## 9      9  0.50200  -1.89          0.922   371.0     350.0    802
## 10    10  0.55800  -1.91          0.929   293.0     272.0    688
## 11    11  0.60100  -1.92          0.934   235.0     214.0    597
## 12    12  0.63700  -1.95          0.938   191.0     171.0    524
## 13    14  0.69900  -1.97          0.951   131.0     112.0    412
## 14    16  0.72900  -2.03          0.954    93.7      76.7    336
## 15    18  0.74400  -2.10          0.952    69.0      54.0    281
## 16    20  0.76500  -2.14          0.960    52.1      39.0    238
## 17    22  0.77000  -2.18          0.960    40.3      28.6    204
## 18    24  0.78400  -2.20          0.966    31.7      21.5    177
## 19    26  0.79200  -2.21          0.970    25.3      16.4    154</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-42-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for GCT is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-43-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/GCT%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in GCT
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite1
</td>
<td style="text-align:right;">
34
</td>
</tr>
<tr>
<td style="text-align:left;">
antiquewhite2
</td>
<td style="text-align:right;">
132
</td>
</tr>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
95
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
2333
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
914
</td>
</tr>
<tr>
<td style="text-align:left;">
brown2
</td>
<td style="text-align:right;">
156
</td>
</tr>
<tr>
<td style="text-align:left;">
chocolate4
</td>
<td style="text-align:right;">
79
</td>
</tr>
<tr>
<td style="text-align:left;">
coral
</td>
<td style="text-align:right;">
113
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
179
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
360
</td>
</tr>
<tr>
<td style="text-align:left;">
cyan
</td>
<td style="text-align:right;">
1252
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgreen
</td>
<td style="text-align:right;">
194
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
408
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen2
</td>
<td style="text-align:right;">
1719
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
1267
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen2
</td>
<td style="text-align:right;">
40
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen3
</td>
<td style="text-align:right;">
174
</td>
</tr>
<tr>
<td style="text-align:left;">
deeppink
</td>
<td style="text-align:right;">
1490
</td>
</tr>
<tr>
<td style="text-align:left;">
firebrick3
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
floralwhite
</td>
<td style="text-align:right;">
582
</td>
</tr>
<tr>
<td style="text-align:left;">
green4
</td>
<td style="text-align:right;">
40
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred3
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcoral
</td>
<td style="text-align:right;">
83
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink2
</td>
<td style="text-align:right;">
47
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink3
</td>
<td style="text-align:right;">
70
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple1
</td>
<td style="text-align:right;">
134
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple2
</td>
<td style="text-align:right;">
155
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple4
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite
</td>
<td style="text-align:right;">
1335
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered3
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
pink4
</td>
<td style="text-align:right;">
302
</td>
</tr>
<tr>
<td style="text-align:left;">
plum
</td>
<td style="text-align:right;">
89
</td>
</tr>
<tr>
<td style="text-align:left;">
plum4
</td>
<td style="text-align:right;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
saddlebrown
</td>
<td style="text-align:right;">
161
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon2
</td>
<td style="text-align:right;">
72
</td>
</tr>
<tr>
<td style="text-align:left;">
sienna4
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue2
</td>
<td style="text-align:right;">
152
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue4
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
tan
</td>
<td style="text-align:right;">
230
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow4
</td>
<td style="text-align:right;">
92
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There appears to be a strong correlation between <em>skyblue2</em>
and <em>green4</em>, which are associated with strength.</p>
<p><br></p>
<div id="interesting-modules-7" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Interesting
Modules</h2>
<div id="skyblue2---strength" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> skyblue2 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20skyblue2%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20skyblue2%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>I’m not convinced. Let’s move on to correlations with stronger
p-values</p>
<p><br></p>
</div>
<div id="lightpink3---mean-testosterone" class="section level3"
number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> lightpink3 - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20lightpink3%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20lightpink3%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-46-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-13" class="section level4" number="8.1.2.1">
<h4><span class="header-section-number">8.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in lightpink3 module in GCT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0070528" class="uri">GO:0070528</a>
</td>
<td style="text-align:left;">
protein kinase C signaling
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
10/12240
</td>
<td style="text-align:right;">
0.0009691
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
LOC113994699/PLEK
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033540" class="uri">GO:0033540</a>
</td>
<td style="text-align:left;">
fatty acid beta-oxidation using acyl-CoA oxidase
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
13/12240
</td>
<td style="text-align:right;">
0.0016646
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
EHHADH/HSD17B4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032703" class="uri">GO:0032703</a>
</td>
<td style="text-align:left;">
negative regulation of interleukin-2 production
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
21/12240
</td>
<td style="text-align:right;">
0.0043738
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
CD34/LOC113986241
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043392" class="uri">GO:0043392</a>
</td>
<td style="text-align:left;">
negative regulation of DNA binding
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
21/12240
</td>
<td style="text-align:right;">
0.0043738
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
JUN/NFKBIA
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
24/12240
</td>
<td style="text-align:right;">
0.0056963
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
LOC113984271/LOC113986241
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034080" class="uri">GO:0034080</a>
</td>
<td style="text-align:left;">
CENP-A containing nucleosome assembly
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
28/12240
</td>
<td style="text-align:right;">
0.0077074
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
CENPT/HJURP
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000082" class="uri">GO:0000082</a>
</td>
<td style="text-align:left;">
G1/S transition of mitotic cell cycle
</td>
<td style="text-align:left;">
3/58
</td>
<td style="text-align:left;">
87/12240
</td>
<td style="text-align:right;">
0.0080763
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
EIF4EBP1/MCM2/PLK2
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051591" class="uri">GO:0051591</a>
</td>
<td style="text-align:left;">
response to cAMP
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
29/12240
</td>
<td style="text-align:right;">
0.0082533
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
DUSP1/JUN
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046627" class="uri">GO:0046627</a>
</td>
<td style="text-align:left;">
negative regulation of insulin receptor signaling pathway
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
33/12240
</td>
<td style="text-align:right;">
0.0106042
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
PRKCQ/SOCS1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045746" class="uri">GO:0045746</a>
</td>
<td style="text-align:left;">
negative regulation of Notch signaling pathway
</td>
<td style="text-align:left;">
2/58
</td>
<td style="text-align:left;">
37/12240
</td>
<td style="text-align:right;">
0.0132150
</td>
<td style="text-align:right;">
0.2332786
</td>
<td style="text-align:right;">
0.2190439
</td>
<td style="text-align:left;">
NFKBIA/TCIM
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20lightpink3%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="antiquewhite1---strength" class="section level3"
number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> antiquewhite1 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite1%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-47-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-14" class="section level4" number="8.1.3.1">
<h4><span class="header-section-number">8.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in antiquewhite1 module in GCT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0043044" class="uri">GO:0043044</a>
</td>
<td style="text-align:left;">
ATP-dependent chromatin remodeling
</td>
<td style="text-align:left;">
2/27
</td>
<td style="text-align:left;">
36/12240
</td>
<td style="text-align:right;">
0.0028188
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
MBD3/SMARCA5
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1990830" class="uri">GO:1990830</a>
</td>
<td style="text-align:left;">
cellular response to leukemia inhibitory factor
</td>
<td style="text-align:left;">
2/27
</td>
<td style="text-align:left;">
77/12240
</td>
<td style="text-align:right;">
0.0123845
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
SMARCA5/SPRY2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006613" class="uri">GO:0006613</a>
</td>
<td style="text-align:left;">
cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
10/12240
</td>
<td style="text-align:right;">
0.0218491
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
SRP14
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
10/12240
</td>
<td style="text-align:right;">
0.0218491
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0106074" class="uri">GO:0106074</a>
</td>
<td style="text-align:left;">
aminoacyl-tRNA metabolism involved in translational fidelity
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
10/12240
</td>
<td style="text-align:right;">
0.0218491
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
DTD1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002031" class="uri">GO:0002031</a>
</td>
<td style="text-align:left;">
G protein-coupled receptor internalization
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0240085
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
SAG
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042474" class="uri">GO:0042474</a>
</td>
<td style="text-align:left;">
middle ear morphogenesis
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0240085
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
TSHZ1
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050678" class="uri">GO:0050678</a>
</td>
<td style="text-align:left;">
regulation of epithelial cell proliferation
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0240085
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070914" class="uri">GO:0070914</a>
</td>
<td style="text-align:left;">
UV-damage excision repair
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0240085
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
DDB2
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098974" class="uri">GO:0098974</a>
</td>
<td style="text-align:left;">
postsynaptic actin cytoskeleton organization
</td>
<td style="text-align:left;">
1/27
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0240085
</td>
<td style="text-align:right;">
0.1458146
</td>
<td style="text-align:right;">
0.1121232
</td>
<td style="text-align:left;">
LOC113998505
</td>
<td style="text-align:right;">
1
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="antiquewhite2---strength" class="section level3"
number="8.1.4">
<h3><span class="header-section-number">8.1.4</span> antiquewhite2 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite2%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite2%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-48-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-15" class="section level4" number="8.1.4.1">
<h4><span class="header-section-number">8.1.4.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in antiquewhite2 module in GCT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0009566" class="uri">GO:0009566</a>
</td>
<td style="text-align:left;">
fertilization
</td>
<td style="text-align:left;">
3/95
</td>
<td style="text-align:left;">
21/12240
</td>
<td style="text-align:right;">
0.0005443
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:left;">
AAAS/LOC114000708/MEIOB
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046854" class="uri">GO:0046854</a>
</td>
<td style="text-align:left;">
phosphatidylinositol phosphorylation
</td>
<td style="text-align:left;">
4/95
</td>
<td style="text-align:left;">
50/12240
</td>
<td style="text-align:right;">
0.0005968
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:left;">
FAM126A/PIK3C3/PIK3CB/PIK3R6
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090305" class="uri">GO:0090305</a>
</td>
<td style="text-align:left;">
nucleic acid phosphodiester bond hydrolysis
</td>
<td style="text-align:left;">
4/95
</td>
<td style="text-align:left;">
54/12240
</td>
<td style="text-align:right;">
0.0008003
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:right;">
0.1091090
</td>
<td style="text-align:left;">
DCLRE1C/EXD2/LOC114001323/MEIOB
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010975" class="uri">GO:0010975</a>
</td>
<td style="text-align:left;">
regulation of neuron projection development
</td>
<td style="text-align:left;">
3/95
</td>
<td style="text-align:left;">
29/12240
</td>
<td style="text-align:right;">
0.0014298
</td>
<td style="text-align:right;">
0.1461951
</td>
<td style="text-align:right;">
0.1461951
</td>
<td style="text-align:left;">
IL1RAPL1/LOC113992691/MARK1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019229" class="uri">GO:0019229</a>
</td>
<td style="text-align:left;">
regulation of vasoconstriction
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
12/12240
</td>
<td style="text-align:right;">
0.0037400
</td>
<td style="text-align:right;">
0.2549409
</td>
<td style="text-align:right;">
0.2549409
</td>
<td style="text-align:left;">
ECE1/EDN3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0036092" class="uri">GO:0036092</a>
</td>
<td style="text-align:left;">
phosphatidylinositol-3-phosphate biosynthetic process
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
12/12240
</td>
<td style="text-align:right;">
0.0037400
</td>
<td style="text-align:right;">
0.2549409
</td>
<td style="text-align:right;">
0.2549409
</td>
<td style="text-align:left;">
PIK3C3/PIK3CB
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002009" class="uri">GO:0002009</a>
</td>
<td style="text-align:left;">
morphogenesis of an epithelium
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
14/12240
</td>
<td style="text-align:right;">
0.0051048
</td>
<td style="text-align:right;">
0.2982672
</td>
<td style="text-align:right;">
0.2982672
</td>
<td style="text-align:left;">
FRAS1/RIPK4
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000729" class="uri">GO:0000729</a>
</td>
<td style="text-align:left;">
DNA double-strand break processing
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
15/12240
</td>
<td style="text-align:right;">
0.0058605
</td>
<td style="text-align:right;">
0.2996201
</td>
<td style="text-align:right;">
0.2996201
</td>
<td style="text-align:left;">
EXD2/LOC114001323
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048873" class="uri">GO:0048873</a>
</td>
<td style="text-align:left;">
homeostasis of number of cells within a tissue
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
21/12240
</td>
<td style="text-align:right;">
0.0113724
</td>
<td style="text-align:right;">
0.3877285
</td>
<td style="text-align:right;">
0.3877285
</td>
<td style="text-align:left;">
ADD1/LOC114000708
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090023" class="uri">GO:0090023</a>
</td>
<td style="text-align:left;">
positive regulation of neutrophil chemotaxis
</td>
<td style="text-align:left;">
2/95
</td>
<td style="text-align:left;">
21/12240
</td>
<td style="text-align:right;">
0.0113724
</td>
<td style="text-align:right;">
0.3877285
</td>
<td style="text-align:right;">
0.3877285
</td>
<td style="text-align:left;">
MCU/MOSPD2
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20antiquewhite2%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="coral1---strength" class="section level3" number="8.1.5">
<h3><span class="header-section-number">8.1.5</span> coral1 -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20coral1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20coral1%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-49-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-16" class="section level4" number="8.1.5.1">
<h4><span class="header-section-number">8.1.5.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in coral1 module in GCT
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007275" class="uri">GO:0007275</a>
</td>
<td style="text-align:left;">
multicellular organism development
</td>
<td style="text-align:left;">
11/149
</td>
<td style="text-align:left;">
303/12240
</td>
<td style="text-align:right;">
0.0011903
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
CDH8/DGKD/EPHA5/ERBB4/ERMARD/NELFA/PAQR5/STRBP/TDRD9/VIRMA/WIF1
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071392" class="uri">GO:0071392</a>
</td>
<td style="text-align:left;">
cellular response to estradiol stimulus
</td>
<td style="text-align:left;">
3/149
</td>
<td style="text-align:left;">
31/12240
</td>
<td style="text-align:right;">
0.0061911
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
ESR2/NCOA3/NRIP1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030518" class="uri">GO:0030518</a>
</td>
<td style="text-align:left;">
intracellular steroid hormone receptor signaling pathway
</td>
<td style="text-align:left;">
2/149
</td>
<td style="text-align:left;">
10/12240
</td>
<td style="text-align:right;">
0.0062129
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
ESR2/NR3C1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000010" class="uri">GO:2000010</a>
</td>
<td style="text-align:left;">
positive regulation of protein localization to cell surface
</td>
<td style="text-align:left;">
2/149
</td>
<td style="text-align:left;">
11/12240
</td>
<td style="text-align:right;">
0.0075332
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
ERBB4/MAP1A
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042472" class="uri">GO:0042472</a>
</td>
<td style="text-align:left;">
inner ear morphogenesis
</td>
<td style="text-align:left;">
3/149
</td>
<td style="text-align:left;">
36/12240
</td>
<td style="text-align:right;">
0.0094090
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
ALDH1A3/GBX2/PRRX1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0018105" class="uri">GO:0018105</a>
</td>
<td style="text-align:left;">
peptidyl-serine phosphorylation
</td>
<td style="text-align:left;">
6/149
</td>
<td style="text-align:left;">
148/12240
</td>
<td style="text-align:right;">
0.0094306
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
CAB39/CAMK1D/MAPK14/MAPKAPK5/SGK2/WNK1
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032793" class="uri">GO:0032793</a>
</td>
<td style="text-align:left;">
positive regulation of CREB transcription factor activity
</td>
<td style="text-align:left;">
2/149
</td>
<td style="text-align:left;">
14/12240
</td>
<td style="text-align:right;">
0.0121698
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
CAMK1D/EPHA5
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002028" class="uri">GO:0002028</a>
</td>
<td style="text-align:left;">
regulation of sodium ion transport
</td>
<td style="text-align:left;">
2/149
</td>
<td style="text-align:left;">
15/12240
</td>
<td style="text-align:right;">
0.0139309
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
LOC113983777/WNK1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050773" class="uri">GO:0050773</a>
</td>
<td style="text-align:left;">
regulation of dendrite development
</td>
<td style="text-align:left;">
2/149
</td>
<td style="text-align:left;">
15/12240
</td>
<td style="text-align:right;">
0.0139309
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
CAMK1D/CSMD3
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001755" class="uri">GO:0001755</a>
</td>
<td style="text-align:left;">
neural crest cell migration
</td>
<td style="text-align:left;">
3/149
</td>
<td style="text-align:left;">
45/12240
</td>
<td style="text-align:right;">
0.0172755
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:right;">
0.5366347
</td>
<td style="text-align:left;">
ERBB4/GBX2/SEMA5B
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/GCT%20coral1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="nucleus-taenia-tna" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Nucleus Taenia
(TnA)</h1>
<pre><code>## [1] 14696</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 33 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq    slope truncated.R.sq mean.k. median.k. max.k.
## 1      1 9.27e-03 15.90000          0.870 7350.00   7350.00 7430.0
## 2      2 2.09e-03 -2.39000          0.926 4010.00   4000.00 4300.0
## 3      3 4.48e-03  1.51000          0.968 2330.00   2330.00 2730.0
## 4      4 1.51e-02  1.62000          0.973 1440.00   1430.00 1840.0
## 5      5 1.31e-02  1.02000          0.971  922.00    912.00 1310.0
## 6      6 1.39e-02  0.77800          0.973  614.00    604.00  958.0
## 7      7 3.84e-03  0.31700          0.964  422.00    412.00  724.0
## 8      8 4.59e-06  0.00917          0.955  298.00    288.00  564.0
## 9      9 2.28e-02 -0.57300          0.944  215.00    206.00  452.0
## 10    10 9.46e-02 -1.07000          0.938  159.00    150.00  369.0
## 11    11 2.05e-01 -1.44000          0.940  119.00    111.00  306.0
## 12    12 3.35e-01 -1.76000          0.952   90.60     83.60  257.0
## 13    14 5.72e-01 -2.18000          0.978   54.90     49.10  186.0
## 14    16 7.14e-01 -2.41000          0.991   34.80     30.10  140.0
## 15    18 7.92e-01 -2.54000          0.996   22.90     19.10  108.0
## 16    20 8.29e-01 -2.64000          0.993   15.60     12.50   84.7
## 17    22 8.55e-01 -2.68000          0.994   10.90      8.45   67.8
## 18    24 8.75e-01 -2.68000          0.997    7.83      5.83   55.1
## 19    26 8.92e-01 -2.67000          0.998    5.73      4.11   45.4</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-50-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for TNA is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-51-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/TNA%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in TNA
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite2
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
140
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
1096
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
1787
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
470
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
106
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
1579
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
2067
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
254
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange
</td>
<td style="text-align:right;">
1383
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
1848
</td>
</tr>
<tr>
<td style="text-align:left;">
darkseagreen4
</td>
<td style="text-align:right;">
372
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew
</td>
<td style="text-align:right;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
lavenderblush3
</td>
<td style="text-align:right;">
217
</td>
</tr>
<tr>
<td style="text-align:left;">
lightsteelblue
</td>
<td style="text-align:right;">
612
</td>
</tr>
<tr>
<td style="text-align:left;">
lightyellow
</td>
<td style="text-align:right;">
585
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta4
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple4
</td>
<td style="text-align:right;">
49
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite1
</td>
<td style="text-align:right;">
60
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered3
</td>
<td style="text-align:right;">
81
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered4
</td>
<td style="text-align:right;">
954
</td>
</tr>
<tr>
<td style="text-align:left;">
palevioletred3
</td>
<td style="text-align:right;">
95
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue1
</td>
<td style="text-align:right;">
82
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue2
</td>
<td style="text-align:right;">
84
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue4
</td>
<td style="text-align:right;">
44
</td>
</tr>
<tr>
<td style="text-align:left;">
tan
</td>
<td style="text-align:right;">
457
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle2
</td>
<td style="text-align:right;">
99
</td>
</tr>
</tbody>
</table>
<p>Based on inspection of the module memberships it seems that
<em>coral1</em> is the inverse of <em>blue</em> and/or
<em>lightyellow</em></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-52-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
<div id="interesting-modules-8" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Interesting
Modules</h2>
<div id="coral1---mean-testosterone" class="section level3"
number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> coral1 - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20coral1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20coral1%20module-2.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20coral1%20module-3.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-53-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-17" class="section level4" number="9.1.1.1">
<h4><span class="header-section-number">9.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in coral1 module in TNA
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0001755" class="uri">GO:0001755</a>
</td>
<td style="text-align:left;">
neural crest cell migration
</td>
<td style="text-align:left;">
16/1250
</td>
<td style="text-align:left;">
45/11978
</td>
<td style="text-align:right;">
0.0000062
</td>
<td style="text-align:right;">
0.0122582
</td>
<td style="text-align:right;">
0.0122582
</td>
<td style="text-align:left;">
ACVR1/ALX1/ARTN/EDN3/EFNB1/LOC113982696/LOC113984504/LOC113988059/RADIL/SEMA3B/SEMA4A/SEMA4B/SEMA4D/SEMA4G/SHH/SOX8
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007411" class="uri">GO:0007411</a>
</td>
<td style="text-align:left;">
axon guidance
</td>
<td style="text-align:left;">
41/1250
</td>
<td style="text-align:left;">
208/11978
</td>
<td style="text-align:right;">
0.0000432
</td>
<td style="text-align:right;">
0.0429695
</td>
<td style="text-align:right;">
0.0429695
</td>
<td style="text-align:left;">
ARTN/B3GNT2/BMP7/DOK5/DRAXIN/EFNA1/EFNB1/EPHA1/GAB2/GFRA3/GPC1/LGI1/LOC113982696/LOC113983904/LOC113984118/LOC113984504/LOC113988059/LOC113990338/LRTM2/MAPK7/NEO1/NEXN/NPTN/NRP2/NTN4/PALLD/RAP1GAP/SEMA3B/SEMA4A/SEMA4B/SEMA4D/SEMA4G/SHC3/SHH/SIAH1/SLIT2/SRC/TUBB3/UNC5A/VLDLR/VSTM2L
</td>
<td style="text-align:right;">
41
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098962" class="uri">GO:0098962</a>
</td>
<td style="text-align:left;">
regulation of postsynaptic neurotransmitter receptor activity
</td>
<td style="text-align:left;">
8/1250
</td>
<td style="text-align:left;">
18/11978
</td>
<td style="text-align:right;">
0.0002306
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:left;">
BEGAIN/CACNG4/DLGAP3/LOC113982730/LOC114000047/NPTX1/NPTX2/SRC
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050919" class="uri">GO:0050919</a>
</td>
<td style="text-align:left;">
negative chemotaxis
</td>
<td style="text-align:left;">
12/1250
</td>
<td style="text-align:left;">
38/11978
</td>
<td style="text-align:right;">
0.0003237
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:left;">
LOC113982696/LOC113984504/LOC113988059/LRTM2/NRP2/PDGFA/SEMA3B/SEMA4A/SEMA4B/SEMA4D/SEMA4G/SLIT2
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035195" class="uri">GO:0035195</a>
</td>
<td style="text-align:left;">
gene silencing by miRNA
</td>
<td style="text-align:left;">
6/1250
</td>
<td style="text-align:left;">
11/11978
</td>
<td style="text-align:right;">
0.0003710
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:right;">
0.1476696
</td>
<td style="text-align:left;">
CNOT6/CNOT8/EIF6/LIMD1/MOV10L1/WTIP
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007267" class="uri">GO:0007267</a>
</td>
<td style="text-align:left;">
cell-cell signaling
</td>
<td style="text-align:left;">
26/1250
</td>
<td style="text-align:left;">
130/11978
</td>
<td style="text-align:right;">
0.0008199
</td>
<td style="text-align:right;">
0.2719451
</td>
<td style="text-align:right;">
0.2719451
</td>
<td style="text-align:left;">
ADORA2A/ADRA1A/BMP2/CCN5/CX3CL1/DHH/ECE2/EDN3/EFNA1/EFNB1/ESR2/FGF18/FGFBP2/GJA5/GPNMB/GUCA1B/INHBA/LOC113988059/LOC113991921/LOC113992470/LOC113993120/PDGFA/POMC/SEMA3B/SHH/SSTR3
</td>
<td style="text-align:right;">
26
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071470" class="uri">GO:0071470</a>
</td>
<td style="text-align:left;">
cellular response to osmotic stress
</td>
<td style="text-align:left;">
6/1250
</td>
<td style="text-align:left;">
13/11978
</td>
<td style="text-align:right;">
0.0011461
</td>
<td style="text-align:right;">
0.3258145
</td>
<td style="text-align:right;">
0.3258145
</td>
<td style="text-align:left;">
LOC113991483/LOC113991484/LOC113991485/LOC113991486/LOC114001824/PKD2
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009880" class="uri">GO:0009880</a>
</td>
<td style="text-align:left;">
embryonic pattern specification
</td>
<td style="text-align:left;">
7/1250
</td>
<td style="text-align:left;">
18/11978
</td>
<td style="text-align:right;">
0.0014951
</td>
<td style="text-align:right;">
0.3305788
</td>
<td style="text-align:right;">
0.3305788
</td>
<td style="text-align:left;">
BMP7/EFNB1/RIPPLY2/SATB2/SHH/SMAD3/ZBTB16
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0055074" class="uri">GO:0055074</a>
</td>
<td style="text-align:left;">
calcium ion homeostasis
</td>
<td style="text-align:left;">
7/1250
</td>
<td style="text-align:left;">
18/11978
</td>
<td style="text-align:right;">
0.0014951
</td>
<td style="text-align:right;">
0.3305788
</td>
<td style="text-align:right;">
0.3305788
</td>
<td style="text-align:left;">
CAMK2N1/CIB2/FAM20A/KL/LOC114004119/SNCA/TNFSF11
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051764" class="uri">GO:0051764</a>
</td>
<td style="text-align:left;">
actin crosslink formation
</td>
<td style="text-align:left;">
5/1250
</td>
<td style="text-align:left;">
10/11978
</td>
<td style="text-align:right;">
0.0019733
</td>
<td style="text-align:right;">
0.3926931
</td>
<td style="text-align:right;">
0.3926931
</td>
<td style="text-align:left;">
BAIAP2/DPYSL3/GAS2/GAS2L1/MARCKS
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20coral1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20coral1%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="blue---mean-testosterone" class="section level3"
number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> blue - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20blue%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20blue%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-54-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-18" class="section level4" number="9.1.2.1">
<h4><span class="header-section-number">9.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in blue module in TNA
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006120" class="uri">GO:0006120</a>
</td>
<td style="text-align:left;">
mitochondrial electron transport, NADH to ubiquinone
</td>
<td style="text-align:left;">
17/1486
</td>
<td style="text-align:left;">
38/11978
</td>
<td style="text-align:right;">
0.0000008
</td>
<td style="text-align:right;">
0.0016443
</td>
<td style="text-align:right;">
0.0016443
</td>
<td style="text-align:left;">
NDUFA1/NDUFA11/NDUFA4/NDUFA4L2/NDUFA5/NDUFAF1/NDUFB11/NDUFB3/NDUFB4/NDUFB5/NDUFB7/NDUFS1/NDUFS4/NDUFS5/NDUFS6/NDUFS8/NDUFV3
</td>
<td style="text-align:right;">
17
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032981" class="uri">GO:0032981</a>
</td>
<td style="text-align:left;">
mitochondrial respiratory chain complex I assembly
</td>
<td style="text-align:left;">
18/1486
</td>
<td style="text-align:left;">
49/11978
</td>
<td style="text-align:right;">
0.0000113
</td>
<td style="text-align:right;">
0.0118183
</td>
<td style="text-align:right;">
0.0118183
</td>
<td style="text-align:left;">
ACAD9/ECSIT/LOC113987270/NDUFA1/NDUFA11/NDUFA5/NDUFAF1/NDUFB11/NDUFB3/NDUFB4/NDUFB5/NDUFB7/NDUFS1/NDUFS4/NDUFS5/NDUFS6/NDUFS8/NDUFV3
</td>
<td style="text-align:right;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048013" class="uri">GO:0048013</a>
</td>
<td style="text-align:left;">
ephrin receptor signaling pathway
</td>
<td style="text-align:left;">
25/1486
</td>
<td style="text-align:left;">
92/11978
</td>
<td style="text-align:right;">
0.0000975
</td>
<td style="text-align:right;">
0.0679597
</td>
<td style="text-align:right;">
0.0679597
</td>
<td style="text-align:left;">
ACTB/ACTR2/ACTR3/ARPC1A/ARPC1B/ARPC2/ARPC5/CDC42/EPHA4/EPHA8/EPHB2/EPHB3/GIT1/HCK/KALRN/LOC113993801/LOC113998953/NCK2/NCSTN/NGEF/RAC1/RHOA/SDC2/SS18/WASL
</td>
<td style="text-align:right;">
25
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034314" class="uri">GO:0034314</a>
</td>
<td style="text-align:left;">
Arp2/3 complex-mediated actin nucleation
</td>
<td style="text-align:left;">
8/1486
</td>
<td style="text-align:left;">
15/11978
</td>
<td style="text-align:right;">
0.0001580
</td>
<td style="text-align:right;">
0.0825658
</td>
<td style="text-align:right;">
0.0825658
</td>
<td style="text-align:left;">
ACTR2/ACTR3/ARPC1A/ARPC1B/ARPC2/ARPC5/ARPC5L/WHAMM
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1902600" class="uri">GO:1902600</a>
</td>
<td style="text-align:left;">
proton transmembrane transport
</td>
<td style="text-align:left;">
21/1486
</td>
<td style="text-align:left;">
75/11978
</td>
<td style="text-align:right;">
0.0002186
</td>
<td style="text-align:right;">
0.0913576
</td>
<td style="text-align:right;">
0.0913576
</td>
<td style="text-align:left;">
ATP12A/ATP2C1/ATP5MF/ATP6AP1L/ATP6V0B/ATP6V0E1/ATP6V1E1/HVCN1/LOC113983953/LOC113984288/LOC113985114/LOC113985195/LOC113985992/LOC114000835/LOC114003365/NDUFA4/NDUFA4L2/SLC17A5/SLC25A27/SLC2A10/SLC9A1
</td>
<td style="text-align:right;">
21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035970" class="uri">GO:0035970</a>
</td>
<td style="text-align:left;">
peptidyl-threonine dephosphorylation
</td>
<td style="text-align:left;">
9/1486
</td>
<td style="text-align:left;">
20/11978
</td>
<td style="text-align:right;">
0.0003162
</td>
<td style="text-align:right;">
0.1101596
</td>
<td style="text-align:right;">
0.1101596
</td>
<td style="text-align:left;">
DUSP4/DUSP5/DUSP6/DUSP7/LOC113997232/PPM1A/PPM1B/PPM1E/PPP2CA
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043687" class="uri">GO:0043687</a>
</td>
<td style="text-align:left;">
post-translational protein modification
</td>
<td style="text-align:left;">
55/1486
</td>
<td style="text-align:left;">
285/11978
</td>
<td style="text-align:right;">
0.0005068
</td>
<td style="text-align:right;">
0.1513247
</td>
<td style="text-align:right;">
0.1513247
</td>
<td style="text-align:left;">
ANKRD9/APP/ASB1/ASB16/ASB7/BMP4/BTBD6/C3/CCN1/COP1/COPS5/COPS7A/CP/DCAF13/DCAF17/DCUN1D1/ELOC/EVA1A/FBXL5/FBXL7/FBXO22/FBXO32/FBXO9/FUCA2/IGFBP7/KLHL13/KLHL20/KLHL42/LAMC1/LOC113984874/LOC113987294/LOC113992168/LOC113999119/LOC113999601/MIA3/NUB1/PROC/PSMA1/PSMA3/PSMA4/PSMA7/PSMD10/PSMD11/PSME4/RAB2A/RAB5A/RBBP7/SCG2/SDC2/SKP1/SKP2/SOCS3/SPSB4/UBE2F/UBE2M
</td>
<td style="text-align:right;">
55
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043401" class="uri">GO:0043401</a>
</td>
<td style="text-align:left;">
steroid hormone mediated signaling pathway
</td>
<td style="text-align:left;">
8/1486
</td>
<td style="text-align:left;">
18/11978
</td>
<td style="text-align:right;">
0.0007617
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:left;">
BMP4/GPER1/LEF1/NR2E1/NR4A3/PAQR7/PGRMC2/PPARD
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008277" class="uri">GO:0008277</a>
</td>
<td style="text-align:left;">
regulation of G protein-coupled receptor signaling pathway
</td>
<td style="text-align:left;">
14/1486
</td>
<td style="text-align:left;">
45/11978
</td>
<td style="text-align:right;">
0.0007642
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:left;">
ADCYAP1/GIT1/HOMER2/LOC114003843/PDCL/RGS11/RGS12/RGS14/RGS16/RGS2/RGS4/RGS5/ZDHHC3/ZDHHC7
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002281" class="uri">GO:0002281</a>
</td>
<td style="text-align:left;">
macrophage activation involved in immune response
</td>
<td style="text-align:left;">
6/1486
</td>
<td style="text-align:left;">
11/11978
</td>
<td style="text-align:right;">
0.0009564
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:right;">
0.1665755
</td>
<td style="text-align:left;">
IFI35/LOC113985745/LOC113986453/LOC113994309/LOC113998967/NMI
</td>
<td style="text-align:right;">
6
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20blue%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="lightyellow---mean-testosterone" class="section level3"
number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> lightyellow - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20lightyellow%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20lightyellow%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-55-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-19" class="section level4" number="9.1.3.1">
<h4><span class="header-section-number">9.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in lightyellow module in TNA
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006695" class="uri">GO:0006695</a>
</td>
<td style="text-align:left;">
cholesterol biosynthetic process
</td>
<td style="text-align:left;">
10/504
</td>
<td style="text-align:left;">
34/11978
</td>
<td style="text-align:right;">
0.0000008
</td>
<td style="text-align:right;">
0.0011542
</td>
<td style="text-align:right;">
0.0011542
</td>
<td style="text-align:left;">
FDFT1/HMGCR/HMGCS1/HSD17B7/INSIG1/INSIG2/LOC113987727/LOC114002832/MSMO1/SQLE
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016126" class="uri">GO:0016126</a>
</td>
<td style="text-align:left;">
sterol biosynthetic process
</td>
<td style="text-align:left;">
6/504
</td>
<td style="text-align:left;">
13/11978
</td>
<td style="text-align:right;">
0.0000072
</td>
<td style="text-align:right;">
0.0049447
</td>
<td style="text-align:right;">
0.0049447
</td>
<td style="text-align:left;">
CH25H/HMGCR/INSIG1/INSIG2/MSMO1/SQLE
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045540" class="uri">GO:0045540</a>
</td>
<td style="text-align:left;">
regulation of cholesterol biosynthetic process
</td>
<td style="text-align:left;">
8/504
</td>
<td style="text-align:left;">
33/11978
</td>
<td style="text-align:right;">
0.0000509
</td>
<td style="text-align:right;">
0.0233422
</td>
<td style="text-align:right;">
0.0233422
</td>
<td style="text-align:left;">
FDFT1/GPAM/HMGCR/HMGCS1/LOC113987727/LOC114002832/SQLE/SREBF2
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071404" class="uri">GO:0071404</a>
</td>
<td style="text-align:left;">
cellular response to low-density lipoprotein particle stimulus
</td>
<td style="text-align:left;">
5/504
</td>
<td style="text-align:left;">
16/11978
</td>
<td style="text-align:right;">
0.0003839
</td>
<td style="text-align:right;">
0.1321546
</td>
<td style="text-align:right;">
0.1321546
</td>
<td style="text-align:left;">
CD81/CD9/HMGCS1/PPARG/SREBF2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0033189" class="uri">GO:0033189</a>
</td>
<td style="text-align:left;">
response to vitamin A
</td>
<td style="text-align:left;">
4/504
</td>
<td style="text-align:left;">
10/11978
</td>
<td style="text-align:right;">
0.0005311
</td>
<td style="text-align:right;">
0.1462786
</td>
<td style="text-align:right;">
0.1462786
</td>
<td style="text-align:left;">
CAT/LOC113999589/LRAT/PPARG
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006625" class="uri">GO:0006625</a>
</td>
<td style="text-align:left;">
protein targeting to peroxisome
</td>
<td style="text-align:left;">
9/504
</td>
<td style="text-align:left;">
66/11978
</td>
<td style="text-align:right;">
0.0016812
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:left;">
ACOX1/AGPS/CAT/DAO/GSTK1/LOC113996048/NUDT19/PEX26/UBE2D3
</td>
<td style="text-align:right;">
9
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019432" class="uri">GO:0019432</a>
</td>
<td style="text-align:left;">
triglyceride biosynthetic process
</td>
<td style="text-align:left;">
4/504
</td>
<td style="text-align:left;">
14/11978
</td>
<td style="text-align:right;">
0.0022133
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:left;">
ACSL4/GK/GPAM/LPIN2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000009" class="uri">GO:2000009</a>
</td>
<td style="text-align:left;">
negative regulation of protein localization to cell surface
</td>
<td style="text-align:left;">
4/504
</td>
<td style="text-align:left;">
14/11978
</td>
<td style="text-align:right;">
0.0022133
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:left;">
LEPROT/LEPROTL1/PICALM/TAX1BP3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006629" class="uri">GO:0006629</a>
</td>
<td style="text-align:left;">
lipid metabolic process
</td>
<td style="text-align:left;">
11/504
</td>
<td style="text-align:left;">
96/11978
</td>
<td style="text-align:right;">
0.0022807
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:left;">
ACOX1/ACSL4/CH25H/CRK/CRKL/HMGCS1/LCAT/LPIN2/PPARG/SREBF2/TTPA
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035666" class="uri">GO:0035666</a>
</td>
<td style="text-align:left;">
TRIF-dependent toll-like receptor signaling pathway
</td>
<td style="text-align:left;">
5/504
</td>
<td style="text-align:left;">
23/11978
</td>
<td style="text-align:right;">
0.0023173
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:right;">
0.3190921
</td>
<td style="text-align:left;">
BIRC2/LOC113996048/TLR6/TRAF3/UBE2D3
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20lightyellow%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/TNA%20lightyellow%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="arcopallium-intermedium-ai" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Arcopallium
Intermedium (Ai)</h1>
<pre><code>## [1] 14598</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 71 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0107 25.90          0.978  7300.0   7300.00   7370
## 2      2   0.1290 -8.30          0.891  4070.0   4060.00   4460
## 3      3   0.2070 -4.32          0.909  2460.0   2440.00   3040
## 4      4   0.2960 -3.16          0.936  1580.0   1550.00   2230
## 5      5   0.3840 -2.66          0.952  1070.0   1040.00   1710
## 6      6   0.4630 -2.39          0.961   745.0    716.00   1360
## 7      7   0.5500 -2.26          0.973   538.0    509.00   1110
## 8      8   0.6310 -2.19          0.983   399.0    370.00    929
## 9      9   0.6880 -2.14          0.986   303.0    275.00    788
## 10    10   0.7350 -2.13          0.987   234.0    208.00    677
## 11    11   0.7710 -2.12          0.987   184.0    160.00    588
## 12    12   0.7890 -2.14          0.980   147.0    124.00    516
## 13    14   0.8370 -2.11          0.984    97.2     77.90    406
## 14    16   0.8630 -2.10          0.983    67.1     50.60    327
## 15    18   0.8790 -2.09          0.983    47.9     34.00    270
## 16    20   0.8890 -2.09          0.982    35.2     23.40    227
## 17    22   0.8720 -2.11          0.964    26.5     16.60    193
## 18    24   0.8630 -2.13          0.957    20.4     11.90    166
## 19    26   0.8750 -2.11          0.962    15.9      8.74    145</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-56-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for AI is 18.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-57-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/AI%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in AI
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite2
</td>
<td style="text-align:right;">
1026
</td>
</tr>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
720
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
182
</td>
</tr>
<tr>
<td style="text-align:left;">
blue2
</td>
<td style="text-align:right;">
68
</td>
</tr>
<tr>
<td style="text-align:left;">
brown2
</td>
<td style="text-align:right;">
68
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
104
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
94
</td>
</tr>
<tr>
<td style="text-align:left;">
coral2
</td>
<td style="text-align:right;">
90
</td>
</tr>
<tr>
<td style="text-align:left;">
coral3
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgreen
</td>
<td style="text-align:right;">
1538
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
341
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
1289
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen
</td>
<td style="text-align:right;">
1686
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen4
</td>
<td style="text-align:right;">
166
</td>
</tr>
<tr>
<td style="text-align:left;">
darkslateblue
</td>
<td style="text-align:right;">
482
</td>
</tr>
<tr>
<td style="text-align:left;">
darkviolet
</td>
<td style="text-align:right;">
65
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
honeydew1
</td>
<td style="text-align:right;">
309
</td>
</tr>
<tr>
<td style="text-align:left;">
indianred4
</td>
<td style="text-align:right;">
1184
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcoral
</td>
<td style="text-align:right;">
274
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcyan1
</td>
<td style="text-align:right;">
496
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink3
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta
</td>
<td style="text-align:right;">
2702
</td>
</tr>
<tr>
<td style="text-align:left;">
magenta4
</td>
<td style="text-align:right;">
59
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple2
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple4
</td>
<td style="text-align:right;">
53
</td>
</tr>
<tr>
<td style="text-align:left;">
orange
</td>
<td style="text-align:right;">
629
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered3
</td>
<td style="text-align:right;">
76
</td>
</tr>
<tr>
<td style="text-align:left;">
palevioletred3
</td>
<td style="text-align:right;">
98
</td>
</tr>
<tr>
<td style="text-align:left;">
plum
</td>
<td style="text-align:right;">
193
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon2
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue2
</td>
<td style="text-align:right;">
86
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle
</td>
<td style="text-align:right;">
64
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle1
</td>
<td style="text-align:right;">
99
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow3
</td>
<td style="text-align:right;">
50
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-58-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Of our interesting modules, it appears that <em>thistle1</em> is
negatively correlated with <em>skyblue2</em></p>
<p><br></p>
<div id="interesting-modules-9" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Interesting
Modules</h2>
<div id="thistle1---mean-testosterone" class="section level3"
number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> thistle1 - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AI%20thistle1%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AI%20thistle1%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-59-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-20" class="section level4" number="10.1.1.1">
<h4><span class="header-section-number">10.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in thistle1 module in AI
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0006783" class="uri">GO:0006783</a>
</td>
<td style="text-align:left;">
heme biosynthetic process
</td>
<td style="text-align:left;">
3/84
</td>
<td style="text-align:left;">
23/11917
</td>
<td style="text-align:right;">
0.0005404
</td>
<td style="text-align:right;">
0.1945428
</td>
<td style="text-align:right;">
0.1939740
</td>
<td style="text-align:left;">
ALAS1/CPOX/SLC25A39
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007005" class="uri">GO:0007005</a>
</td>
<td style="text-align:left;">
mitochondrion organization
</td>
<td style="text-align:left;">
4/84
</td>
<td style="text-align:left;">
84/11917
</td>
<td style="text-align:right;">
0.0028888
</td>
<td style="text-align:right;">
0.3805802
</td>
<td style="text-align:right;">
0.3794674
</td>
<td style="text-align:left;">
ALAS1/DNAJA3/PYROXD2/SIRT3
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045453" class="uri">GO:0045453</a>
</td>
<td style="text-align:left;">
bone resorption
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
14/11917
</td>
<td style="text-align:right;">
0.0042287
</td>
<td style="text-align:right;">
0.3805802
</td>
<td style="text-align:right;">
0.3794674
</td>
<td style="text-align:left;">
ACP5/NCDN
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1900181" class="uri">GO:1900181</a>
</td>
<td style="text-align:left;">
negative regulation of protein localization to nucleus
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
14/11917
</td>
<td style="text-align:right;">
0.0042287
</td>
<td style="text-align:right;">
0.3805802
</td>
<td style="text-align:right;">
0.3794674
</td>
<td style="text-align:left;">
LZTS2/TRIM29
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001921" class="uri">GO:0001921</a>
</td>
<td style="text-align:left;">
positive regulation of receptor recycling
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
16/11917
</td>
<td style="text-align:right;">
0.0055255
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
ANXA2/LOC113996134
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072593" class="uri">GO:0072593</a>
</td>
<td style="text-align:left;">
reactive oxygen species metabolic process
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
23/11917
</td>
<td style="text-align:right;">
0.0112839
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
LOC113988359/NDUFA13
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007131" class="uri">GO:0007131</a>
</td>
<td style="text-align:left;">
reciprocal meiotic recombination
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
26/11917
</td>
<td style="text-align:right;">
0.0142987
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
KLHDC3/RAD51D
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016525" class="uri">GO:0016525</a>
</td>
<td style="text-align:left;">
negative regulation of angiogenesis
</td>
<td style="text-align:left;">
3/84
</td>
<td style="text-align:left;">
74/11917
</td>
<td style="text-align:right;">
0.0152921
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
COL4A3/ISM1/LOC114002556
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000134" class="uri">GO:2000134</a>
</td>
<td style="text-align:left;">
negative regulation of G1/S transition of mitotic cell cycle
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
28/11917
</td>
<td style="text-align:right;">
0.0164802
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
CDKN2C/RBL1
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009408" class="uri">GO:0009408</a>
</td>
<td style="text-align:left;">
response to heat
</td>
<td style="text-align:left;">
2/84
</td>
<td style="text-align:left;">
30/11917
</td>
<td style="text-align:right;">
0.0187941
</td>
<td style="text-align:right;">
0.3924384
</td>
<td style="text-align:right;">
0.3912909
</td>
<td style="text-align:left;">
DNAJA3/RPS6KB1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/thistle1%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="skyblue2---mean-testosterone" class="section level3"
number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> skyblue2 - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AI%20skyblue2%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AI%20skyblue2%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-60-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-21" class="section level4" number="10.1.2.1">
<h4><span class="header-section-number">10.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in skyblue2 module in AI
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0098696" class="uri">GO:0098696</a>
</td>
<td style="text-align:left;">
regulation of neurotransmitter receptor localization to postsynaptic
specialization membrane
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
10/11917
</td>
<td style="text-align:right;">
0.0019340
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
ADAM10/PRKCZ
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006509" class="uri">GO:0006509</a>
</td>
<td style="text-align:left;">
membrane protein ectodomain proteolysis
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
19/11917
</td>
<td style="text-align:right;">
0.0070672
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
ADAM10/SPPL2A
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001954" class="uri">GO:0001954</a>
</td>
<td style="text-align:left;">
positive regulation of cell-matrix adhesion
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
20/11917
</td>
<td style="text-align:right;">
0.0078185
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
EPHA1/PRKCZ
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0097194" class="uri">GO:0097194</a>
</td>
<td style="text-align:left;">
execution phase of apoptosis
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
20/11917
</td>
<td style="text-align:right;">
0.0078185
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
CFLAR/STK24
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001817" class="uri">GO:0001817</a>
</td>
<td style="text-align:left;">
regulation of cytokine production
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
21/11917
</td>
<td style="text-align:right;">
0.0086041
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
IGF2BP3/LRRC19
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005977" class="uri">GO:0005977</a>
</td>
<td style="text-align:left;">
glycogen metabolic process
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
23/11917
</td>
<td style="text-align:right;">
0.0102765
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
IL6ST/LOC113990070
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010803" class="uri">GO:0010803</a>
</td>
<td style="text-align:left;">
regulation of tumor necrosis factor-mediated signaling pathway
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
23/11917
</td>
<td style="text-align:right;">
0.0102765
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
LOC113989730/SPPL2A
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032715" class="uri">GO:0032715</a>
</td>
<td style="text-align:left;">
negative regulation of interleukin-6 production
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
26/11917
</td>
<td style="text-align:right;">
0.0130308
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
FOXJ1/SOCS5
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0021510" class="uri">GO:0021510</a>
</td>
<td style="text-align:left;">
spinal cord development
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
28/11917
</td>
<td style="text-align:right;">
0.0150254
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
PTN/SUZ12
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051781" class="uri">GO:0051781</a>
</td>
<td style="text-align:left;">
positive regulation of cell division
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
31/11917
</td>
<td style="text-align:right;">
0.0182459
</td>
<td style="text-align:right;">
0.3458354
</td>
<td style="text-align:right;">
0.3407388
</td>
<td style="text-align:left;">
GAREM1/PTN
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/skyblue2%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="palevioletred3---mean-testosterone" class="section level3"
number="10.1.3">
<h3><span class="header-section-number">10.1.3</span> palevioletred3 -
mean testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AI%20palevioletred3%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AI%20palevioletred3%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Not very convincing relationship</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-61-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-22" class="section level4" number="10.1.3.1">
<h4><span class="header-section-number">10.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in palevioletred3 module in AI
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:2000772" class="uri">GO:2000772</a>
</td>
<td style="text-align:left;">
regulation of cellular senescence
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
11/11917
</td>
<td style="text-align:right;">
0.0023535
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
RSL1D1/ZMPSTE24
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2001214" class="uri">GO:2001214</a>
</td>
<td style="text-align:left;">
positive regulation of vasculogenesis
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
12/11917
</td>
<td style="text-align:right;">
0.0028120
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
ADM/CD34
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042632" class="uri">GO:0042632</a>
</td>
<td style="text-align:left;">
cholesterol homeostasis
</td>
<td style="text-align:left;">
3/80
</td>
<td style="text-align:left;">
52/11917
</td>
<td style="text-align:right;">
0.0050824
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
LOC113994745/NPC2/NUS1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031398" class="uri">GO:0031398</a>
</td>
<td style="text-align:left;">
positive regulation of protein ubiquitination
</td>
<td style="text-align:left;">
3/80
</td>
<td style="text-align:left;">
66/11917
</td>
<td style="text-align:right;">
0.0098415
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
BCL10/TBC1D7/UBE2L3
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010951" class="uri">GO:0010951</a>
</td>
<td style="text-align:left;">
negative regulation of endopeptidase activity
</td>
<td style="text-align:left;">
3/80
</td>
<td style="text-align:left;">
67/11917
</td>
<td style="text-align:right;">
0.0102538
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
LOC113992168/PROS1/SPOCK3
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001889" class="uri">GO:0001889</a>
</td>
<td style="text-align:left;">
liver development
</td>
<td style="text-align:left;">
3/80
</td>
<td style="text-align:left;">
68/11917
</td>
<td style="text-align:right;">
0.0106759
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
FPGS/SRD5A1/ZMPSTE24
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008360" class="uri">GO:0008360</a>
</td>
<td style="text-align:left;">
regulation of cell shape
</td>
<td style="text-align:left;">
4/80
</td>
<td style="text-align:left;">
135/11917
</td>
<td style="text-align:right;">
0.0128347
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
CDC42SE2/RHOA/TBCCD1/ZMPSTE24
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016239" class="uri">GO:0016239</a>
</td>
<td style="text-align:left;">
positive regulation of macroautophagy
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
27/11917
</td>
<td style="text-align:right;">
0.0140126
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
SCOC/SESN2
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000209" class="uri">GO:0000209</a>
</td>
<td style="text-align:left;">
protein polyubiquitination
</td>
<td style="text-align:left;">
5/80
</td>
<td style="text-align:left;">
220/11917
</td>
<td style="text-align:right;">
0.0160870
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
FBXO11/LOC113996737/SKP1/UBE2L3/UBE2V1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030449" class="uri">GO:0030449</a>
</td>
<td style="text-align:left;">
regulation of complement activation
</td>
<td style="text-align:left;">
2/80
</td>
<td style="text-align:left;">
30/11917
</td>
<td style="text-align:right;">
0.0171426
</td>
<td style="text-align:right;">
0.3208474
</td>
<td style="text-align:right;">
0.3203173
</td>
<td style="text-align:left;">
LOC113992168/PROS1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/palevioletred3%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="lightcoral---mean-testosterone" class="section level3"
number="10.1.4">
<h3><span class="header-section-number">10.1.4</span> lightcoral - mean
testosterone</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/AI%20lightcoral%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/AI%20lightcoral%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>Not a convincing relationship.</p>
<p><br><br></p>
</div>
</div>
</div>
<div id="lateral-septum-ls" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Lateral Septum
(Ls)</h1>
<pre><code>## [1] 14901</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 56 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.0556 39.00          0.907  7450.0   7450.00   7570
## 2      2   0.0483 -5.20          0.860  4130.0   4110.00   4480
## 3      3   0.1620 -4.09          0.900  2460.0   2450.00   3000
## 4      4   0.2850 -3.28          0.937  1560.0   1540.00   2160
## 5      5   0.4100 -2.85          0.962  1040.0   1010.00   1640
## 6      6   0.5170 -2.61          0.976   716.0    689.00   1290
## 7      7   0.6110 -2.48          0.985   510.0    483.00   1050
## 8      8   0.6900 -2.40          0.992   373.0    347.00    869
## 9      9   0.7470 -2.33          0.994   280.0    254.00    734
## 10    10   0.7940 -2.31          0.996   214.0    190.00    631
## 11    11   0.8250 -2.29          0.998   167.0    144.00    549
## 12    12   0.8450 -2.28          0.996   132.0    111.00    484
## 13    14   0.8770 -2.25          0.995    85.7     67.70    384
## 14    16   0.9000 -2.21          0.997    58.3     43.10    313
## 15    18   0.9140 -2.17          0.997    41.1     28.30    260
## 16    20   0.9240 -2.13          0.998    29.9     19.10    219
## 17    22   0.9300 -2.10          0.997    22.3     13.20    187
## 18    24   0.9340 -2.07          0.998    17.0      9.34    161
## 19    26   0.9330 -2.04          0.996    13.2      6.71    141</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-62-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for LS is 16.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-63-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/LS%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in LS
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
78
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
98
</td>
</tr>
<tr>
<td style="text-align:left;">
black
</td>
<td style="text-align:right;">
2039
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
1386
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
1569
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
1884
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
80
</td>
</tr>
<tr>
<td style="text-align:left;">
coral2
</td>
<td style="text-align:right;">
398
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
256
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
352
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange
</td>
<td style="text-align:right;">
818
</td>
</tr>
<tr>
<td style="text-align:left;">
darkred
</td>
<td style="text-align:right;">
284
</td>
</tr>
<tr>
<td style="text-align:left;">
floralwhite
</td>
<td style="text-align:right;">
466
</td>
</tr>
<tr>
<td style="text-align:left;">
green
</td>
<td style="text-align:right;">
1271
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
lightcoral
</td>
<td style="text-align:right;">
50
</td>
</tr>
<tr>
<td style="text-align:left;">
lightgreen
</td>
<td style="text-align:right;">
223
</td>
</tr>
<tr>
<td style="text-align:left;">
lightsteelblue1
</td>
<td style="text-align:right;">
961
</td>
</tr>
<tr>
<td style="text-align:left;">
maroon
</td>
<td style="text-align:right;">
161
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite2
</td>
<td style="text-align:right;">
357
</td>
</tr>
<tr>
<td style="text-align:left;">
orangered4
</td>
<td style="text-align:right;">
294
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
309
</td>
</tr>
<tr>
<td style="text-align:left;">
palevioletred3
</td>
<td style="text-align:right;">
90
</td>
</tr>
<tr>
<td style="text-align:left;">
plum
</td>
<td style="text-align:right;">
62
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue3
</td>
<td style="text-align:right;">
200
</td>
</tr>
<tr>
<td style="text-align:left;">
yellow
</td>
<td style="text-align:right;">
1083
</td>
</tr>
<tr>
<td style="text-align:left;">
yellowgreen
</td>
<td style="text-align:right;">
128
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-64-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There doesn’t appear to be any obviously negative correlations with
our modules of interest.</p>
<div id="interesting-modules-10" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Interesting
Modules</h2>
<div id="darkmagenta---strength" class="section level3" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> darkmagenta -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20darkmagenta%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/LS%20darkmagenta%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## Warning: Removed 3 rows containing missing values
## (`geom_point()`).</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-65-1.png" width="480" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-66-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-23" class="section level4" number="11.1.1.1">
<h4><span class="header-section-number">11.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkmagenta module in LS
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0016032" class="uri">GO:0016032</a>
</td>
<td style="text-align:left;">
viral process
</td>
<td style="text-align:left;">
21/295
</td>
<td style="text-align:left;">
406/12091
</td>
<td style="text-align:right;">
0.0009823
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
AAAS/ANKRD17/ATF7/BRD2/DCAF1/DNAJB12/E4F1/EP300/IL2RB/KAT2A/LOC113985512/LOC113989518/LOC113999689/LOC114003727/MPDZ/NUP214/SETD2/SPEN/SSRP1/TPR/UBR4
</td>
<td style="text-align:right;">
21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043967" class="uri">GO:0043967</a>
</td>
<td style="text-align:left;">
histone H4 acetylation
</td>
<td style="text-align:left;">
5/295
</td>
<td style="text-align:left;">
33/12091
</td>
<td style="text-align:right;">
0.0011321
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
EP300/EP400/LOC113997056/LOC113998036/LOC113998501
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0046854" class="uri">GO:0046854</a>
</td>
<td style="text-align:left;">
phosphatidylinositol phosphorylation
</td>
<td style="text-align:left;">
6/295
</td>
<td style="text-align:left;">
50/12091
</td>
<td style="text-align:right;">
0.0012919
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
LOC113983741/LOC113988212/LOC113992314/PI4K2A/PIK3R2/SOCS7
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045652" class="uri">GO:0045652</a>
</td>
<td style="text-align:left;">
regulation of megakaryocyte differentiation
</td>
<td style="text-align:left;">
5/295
</td>
<td style="text-align:left;">
35/12091
</td>
<td style="text-align:right;">
0.0014882
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
EP300/KMT2A/L3MBTL1/TNRC6B/TNRC6C
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035194" class="uri">GO:0035194</a>
</td>
<td style="text-align:left;">
post-transcriptional gene silencing by RNA
</td>
<td style="text-align:left;">
3/295
</td>
<td style="text-align:left;">
10/12091
</td>
<td style="text-align:right;">
0.0015190
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
HELZ/TNRC6B/TNRC6C
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051726" class="uri">GO:0051726</a>
</td>
<td style="text-align:left;">
regulation of cell cycle
</td>
<td style="text-align:left;">
8/295
</td>
<td style="text-align:left;">
96/12091
</td>
<td style="text-align:right;">
0.0023630
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
BOP1/CSNK2A2/EP300/HIPK2/L3MBTL1/NUP214/PRDM11/SON
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010761" class="uri">GO:0010761</a>
</td>
<td style="text-align:left;">
fibroblast migration
</td>
<td style="text-align:left;">
3/295
</td>
<td style="text-align:left;">
12/12091
</td>
<td style="text-align:right;">
0.0026857
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
ARID5B/LOC113988212/LOC113992314
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006110" class="uri">GO:0006110</a>
</td>
<td style="text-align:left;">
regulation of glycolytic process
</td>
<td style="text-align:left;">
5/295
</td>
<td style="text-align:left;">
40/12091
</td>
<td style="text-align:right;">
0.0027317
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
AAAS/EP300/LOC113999689/NUP214/TPR
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007595" class="uri">GO:0007595</a>
</td>
<td style="text-align:left;">
lactation
</td>
<td style="text-align:left;">
5/295
</td>
<td style="text-align:left;">
42/12091
</td>
<td style="text-align:right;">
0.0033944
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
CAD/LOC113997056/LOC113998036/MED1/NCOR2
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019827" class="uri">GO:0019827</a>
</td>
<td style="text-align:left;">
stem cell population maintenance
</td>
<td style="text-align:left;">
5/295
</td>
<td style="text-align:left;">
42/12091
</td>
<td style="text-align:right;">
0.0033944
</td>
<td style="text-align:right;">
0.2529708
</td>
<td style="text-align:right;">
0.2436046
</td>
<td style="text-align:left;">
EIF4ENIF1/LOC113993147/LOC113999502/MED30/RIF1
</td>
<td style="text-align:right;">
5
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20darkmagenta%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20darkmagenta%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="black---strength" class="section level3" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> black -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20black%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/LS%20black%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## Warning: Removed 3 rows containing missing values
## (`geom_point()`).</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-67-1.png" width="480" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-68-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-24" class="section level4" number="11.1.2.1">
<h4><span class="header-section-number">11.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in black module in LS
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0007281" class="uri">GO:0007281</a>
</td>
<td style="text-align:left;">
germ cell development
</td>
<td style="text-align:left;">
10/1654
</td>
<td style="text-align:left;">
19/12091
</td>
<td style="text-align:right;">
0.0000635
</td>
<td style="text-align:right;">
0.1183396
</td>
<td style="text-align:right;">
0.1180611
</td>
<td style="text-align:left;">
ACVR1/AKT1/ANGPT2/BMP4/CELF4/FBXW11/PAFAH1B1/RPS6KB1/SMAD5/TDRD1
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006376" class="uri">GO:0006376</a>
</td>
<td style="text-align:left;">
mRNA splice site selection
</td>
<td style="text-align:left;">
10/1654
</td>
<td style="text-align:left;">
20/12091
</td>
<td style="text-align:right;">
0.0001114
</td>
<td style="text-align:right;">
0.1183396
</td>
<td style="text-align:right;">
0.1180611
</td>
<td style="text-align:left;">
CELF2/CELF4/LOC113985802/LOC113985815/LOC114002198/LUC7L/LUC7L3/RBMX/SRSF1/SRSF6
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005979" class="uri">GO:0005979</a>
</td>
<td style="text-align:left;">
regulation of glycogen biosynthetic process
</td>
<td style="text-align:left;">
6/1654
</td>
<td style="text-align:left;">
10/12091
</td>
<td style="text-align:right;">
0.0008323
</td>
<td style="text-align:right;">
0.5895228
</td>
<td style="text-align:right;">
0.5881357
</td>
<td style="text-align:left;">
AKT1/INPP5K/PPP1CB/PPP1R3C/PPP1R3D/PPP1R3G
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035338" class="uri">GO:0035338</a>
</td>
<td style="text-align:left;">
long-chain fatty-acyl-CoA biosynthetic process
</td>
<td style="text-align:left;">
8/1654
</td>
<td style="text-align:left;">
18/12091
</td>
<td style="text-align:right;">
0.0014705
</td>
<td style="text-align:right;">
0.7811873
</td>
<td style="text-align:right;">
0.7793493
</td>
<td style="text-align:left;">
ACSL3/ACSL4/ELOVL2/ELOVL4/ELOVL5/ELOVL6/HSD17B12/TECR
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032869" class="uri">GO:0032869</a>
</td>
<td style="text-align:left;">
cellular response to insulin stimulus
</td>
<td style="text-align:left;">
21/1654
</td>
<td style="text-align:left;">
84/12091
</td>
<td style="text-align:right;">
0.0038217
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
AKT1/C2CD5/CAPN10/CPEB2/FER/HDAC9/INPP5K/INSIG2/LOC113985802/LOC113985815/LPIN2/PTEN/RAB12/RAB8B/RHOQ/RPS6KB1/SORBS1/STAR/SYAP1/TRARG1/YWHAG
</td>
<td style="text-align:right;">
21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006661" class="uri">GO:0006661</a>
</td>
<td style="text-align:left;">
phosphatidylinositol biosynthetic process
</td>
<td style="text-align:left;">
19/1654
</td>
<td style="text-align:left;">
74/12091
</td>
<td style="text-align:right;">
0.0042222
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
CDIPT/CDS1/DGKE/FIG4/INPP4A/INPP5K/MTMR12/MTMR6/MTMR8/PIK3CA/PIK3R4/PIP4K2A/PIP4K2B/PITPNM3/PLEKHA3/PTEN/RAB4A/RAB5A/SACM1L
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0016024" class="uri">GO:0016024</a>
</td>
<td style="text-align:left;">
CDP-diacylglycerol biosynthetic process
</td>
<td style="text-align:left;">
6/1654
</td>
<td style="text-align:left;">
13/12091
</td>
<td style="text-align:right;">
0.0047103
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
AGPAT3/AGPAT5/CDS1/LCLAT1/LOC113995899/TAMM41
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032482" class="uri">GO:0032482</a>
</td>
<td style="text-align:left;">
Rab protein signal transduction
</td>
<td style="text-align:left;">
6/1654
</td>
<td style="text-align:left;">
13/12091
</td>
<td style="text-align:right;">
0.0047103
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
RAB12/RAB21/RAB33B/RAB39A/RAB4A/RAB9A
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045842" class="uri">GO:0045842</a>
</td>
<td style="text-align:left;">
positive regulation of mitotic metaphase/anaphase transition
</td>
<td style="text-align:left;">
6/1654
</td>
<td style="text-align:left;">
13/12091
</td>
<td style="text-align:right;">
0.0047103
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
ANAPC11/ANAPC4/ESPL1/LOC113991003/NSMCE2/RB1
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071470" class="uri">GO:0071470</a>
</td>
<td style="text-align:left;">
cellular response to osmotic stress
</td>
<td style="text-align:left;">
6/1654
</td>
<td style="text-align:left;">
13/12091
</td>
<td style="text-align:right;">
0.0047103
</td>
<td style="text-align:right;">
0.8005247
</td>
<td style="text-align:right;">
0.7986411
</td>
<td style="text-align:left;">
DDX3X/LOC113991482/LOC113991485/LOC113991486/LRRC8D/ZFAND1
</td>
<td style="text-align:right;">
6
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20black%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/LS%20black%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br><br></p>
</div>
</div>
</div>
</div>
<div id="bed-nucleus-of-the-stria-terminalis-bstm"
class="section level1" number="12">
<h1><span class="header-section-number">12</span> Bed Nucleus of the
Stria Terminalis (BSTm)</h1>
<pre><code>## [1] 15033</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 35 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>##    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
## 1      1 8.90e-06   0.819          0.932  7520.0   7520.00   7570
## 2      2 4.60e-01 -19.200          0.919  4140.0   4120.00   4520
## 3      3 4.54e-01  -6.940          0.905  2450.0   2420.00   3000
## 4      4 4.95e-01  -4.130          0.915  1540.0   1500.00   2140
## 5      5 5.62e-01  -3.070          0.935  1010.0    973.00   1590
## 6      6 6.37e-01  -2.750          0.952   689.0    652.00   1250
## 7      7 6.93e-01  -2.670          0.960   486.0    450.00   1010
## 8      8 7.32e-01  -2.640          0.961   351.0    318.00    841
## 9      9 7.65e-01  -2.600          0.965   260.0    230.00    713
## 10    10 7.99e-01  -2.540          0.977   197.0    169.00    613
## 11    11 8.14e-01  -2.520          0.977   152.0    126.00    535
## 12    12 8.28e-01  -2.510          0.977   119.0     95.20    471
## 13    14 8.51e-01  -2.460          0.980    75.6     56.50    375
## 14    16 8.78e-01  -2.390          0.987    50.4     34.80    306
## 15    18 8.91e-01  -2.350          0.989    34.9     22.30    255
## 16    20 8.89e-01  -2.330          0.984    24.9     14.60    216
## 17    22 9.04e-01  -2.280          0.991    18.3      9.91    185
## 18    24 9.15e-01  -2.220          0.994    13.7      6.84    160
## 19    26 9.15e-01  -2.190          0.992    10.5      4.82    140</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-69-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>The soft-power selected for BSTm is 22.</p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-70-1.png" width="960" style="display: block; margin: auto;" /></p>
<img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20correlate%20with%20traits-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Genes per module in BSTm
</caption>
<thead>
<tr>
<th style="text-align:left;">
moduleColors
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
antiquewhite4
</td>
<td style="text-align:right;">
712
</td>
</tr>
<tr>
<td style="text-align:left;">
bisque4
</td>
<td style="text-align:right;">
111
</td>
</tr>
<tr>
<td style="text-align:left;">
blue
</td>
<td style="text-align:right;">
1123
</td>
</tr>
<tr>
<td style="text-align:left;">
brown
</td>
<td style="text-align:right;">
1067
</td>
</tr>
<tr>
<td style="text-align:left;">
brown4
</td>
<td style="text-align:right;">
1377
</td>
</tr>
<tr>
<td style="text-align:left;">
coral1
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
coral2
</td>
<td style="text-align:right;">
490
</td>
</tr>
<tr>
<td style="text-align:left;">
darkgrey
</td>
<td style="text-align:right;">
720
</td>
</tr>
<tr>
<td style="text-align:left;">
darkmagenta
</td>
<td style="text-align:right;">
317
</td>
</tr>
<tr>
<td style="text-align:left;">
darkolivegreen
</td>
<td style="text-align:right;">
929
</td>
</tr>
<tr>
<td style="text-align:left;">
darkorange2
</td>
<td style="text-align:right;">
1057
</td>
</tr>
<tr>
<td style="text-align:left;">
darkred
</td>
<td style="text-align:right;">
492
</td>
</tr>
<tr>
<td style="text-align:left;">
green
</td>
<td style="text-align:right;">
471
</td>
</tr>
<tr>
<td style="text-align:left;">
grey
</td>
<td style="text-align:right;">
141
</td>
</tr>
<tr>
<td style="text-align:left;">
grey60
</td>
<td style="text-align:right;">
996
</td>
</tr>
<tr>
<td style="text-align:left;">
ivory
</td>
<td style="text-align:right;">
220
</td>
</tr>
<tr>
<td style="text-align:left;">
lightpink4
</td>
<td style="text-align:right;">
82
</td>
</tr>
<tr>
<td style="text-align:left;">
maroon
</td>
<td style="text-align:right;">
317
</td>
</tr>
<tr>
<td style="text-align:left;">
mediumpurple3
</td>
<td style="text-align:right;">
128
</td>
</tr>
<tr>
<td style="text-align:left;">
navajowhite2
</td>
<td style="text-align:right;">
87
</td>
</tr>
<tr>
<td style="text-align:left;">
paleturquoise
</td>
<td style="text-align:right;">
337
</td>
</tr>
<tr>
<td style="text-align:left;">
royalblue
</td>
<td style="text-align:right;">
222
</td>
</tr>
<tr>
<td style="text-align:left;">
saddlebrown
</td>
<td style="text-align:right;">
169
</td>
</tr>
<tr>
<td style="text-align:left;">
salmon4
</td>
<td style="text-align:right;">
2361
</td>
</tr>
<tr>
<td style="text-align:left;">
skyblue
</td>
<td style="text-align:right;">
940
</td>
</tr>
<tr>
<td style="text-align:left;">
thistle1
</td>
<td style="text-align:right;">
96
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-71-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There appears to be a strong correlation between <em>darkred</em> and
<em>salmon4</em>, and inverse between <em>darkmagenta</em> and
<em>paleturquoise</em></p>
<p><br></p>
<div id="interesting-modules-11" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Interesting
Modules</h2>
<div id="darkred---strength" class="section level3" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> darkred -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkred%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkred%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-72-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-25" class="section level4" number="12.1.1.1">
<h4><span class="header-section-number">12.1.1.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkred module in BSTm
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0038007" class="uri">GO:0038007</a>
</td>
<td style="text-align:left;">
netrin-activated signaling pathway
</td>
<td style="text-align:left;">
5/420
</td>
<td style="text-align:left;">
14/12181
</td>
<td style="text-align:right;">
0.0000736
</td>
<td style="text-align:right;">
0.0904506
</td>
<td style="text-align:right;">
0.0869563
</td>
<td style="text-align:left;">
DSCAM/LOC113983017/LOC113988456/LOC113993074/UNC5A
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1900034" class="uri">GO:1900034</a>
</td>
<td style="text-align:left;">
regulation of cellular response to heat
</td>
<td style="text-align:left;">
10/420
</td>
<td style="text-align:left;">
70/12181
</td>
<td style="text-align:right;">
0.0001329
</td>
<td style="text-align:right;">
0.0904506
</td>
<td style="text-align:right;">
0.0869563
</td>
<td style="text-align:left;">
AAAS/ATR/CAMK2B/CREBBP/HSP90AB1/LOC113998422/MLST8/NUP54/NUP62CL/NUP85
</td>
<td style="text-align:right;">
10
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090303" class="uri">GO:0090303</a>
</td>
<td style="text-align:left;">
positive regulation of wound healing
</td>
<td style="text-align:left;">
6/420
</td>
<td style="text-align:left;">
26/12181
</td>
<td style="text-align:right;">
0.0002075
</td>
<td style="text-align:right;">
0.0941361
</td>
<td style="text-align:right;">
0.0904994
</td>
<td style="text-align:left;">
ACTG1/DMTN/LOC113983017/LOC113983228/LOC113993074/OCLN
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043984" class="uri">GO:0043984</a>
</td>
<td style="text-align:left;">
histone H4-K16 acetylation
</td>
<td style="text-align:left;">
5/420
</td>
<td style="text-align:left;">
21/12181
</td>
<td style="text-align:right;">
0.0006128
</td>
<td style="text-align:right;">
0.2085137
</td>
<td style="text-align:right;">
0.2004582
</td>
<td style="text-align:left;">
KANSL1/KANSL3/KAT7/KMT2A/MSL1
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006909" class="uri">GO:0006909</a>
</td>
<td style="text-align:left;">
phagocytosis
</td>
<td style="text-align:left;">
7/420
</td>
<td style="text-align:left;">
46/12181
</td>
<td style="text-align:right;">
0.0009214
</td>
<td style="text-align:right;">
0.2508137
</td>
<td style="text-align:right;">
0.2411241
</td>
<td style="text-align:left;">
LOC113988212/LOC113992314/LOC113993578/LRP1/RUBCN/SYT7/TUSC2
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051899" class="uri">GO:0051899</a>
</td>
<td style="text-align:left;">
membrane depolarization
</td>
<td style="text-align:left;">
5/420
</td>
<td style="text-align:left;">
26/12181
</td>
<td style="text-align:right;">
0.0017189
</td>
<td style="text-align:right;">
0.3342980
</td>
<td style="text-align:right;">
0.3213832
</td>
<td style="text-align:left;">
CACNA1E/CACNB3/CHRNB2/LOC113993166/LOC114000517
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060964" class="uri">GO:0060964</a>
</td>
<td style="text-align:left;">
regulation of gene silencing by miRNA
</td>
<td style="text-align:left;">
7/420
</td>
<td style="text-align:left;">
51/12181
</td>
<td style="text-align:right;">
0.0017194
</td>
<td style="text-align:right;">
0.3342980
</td>
<td style="text-align:right;">
0.3213832
</td>
<td style="text-align:left;">
AAAS/LOC113983589/LOC113992597/LOC114003727/NUP54/NUP62CL/NUP85
</td>
<td style="text-align:right;">
7
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0032786" class="uri">GO:0032786</a>
</td>
<td style="text-align:left;">
positive regulation of DNA-templated transcription, elongation
</td>
<td style="text-align:left;">
4/420
</td>
<td style="text-align:left;">
17/12181
</td>
<td style="text-align:right;">
0.0023199
</td>
<td style="text-align:right;">
0.3351089
</td>
<td style="text-align:right;">
0.3221628
</td>
<td style="text-align:left;">
CCNT1/ELL/KAT7/LOC113988658
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0038096" class="uri">GO:0038096</a>
</td>
<td style="text-align:left;">
Fc-gamma receptor signaling pathway involved in phagocytosis
</td>
<td style="text-align:left;">
8/420
</td>
<td style="text-align:left;">
69/12181
</td>
<td style="text-align:right;">
0.0024789
</td>
<td style="text-align:right;">
0.3351089
</td>
<td style="text-align:right;">
0.3221628
</td>
<td style="text-align:left;">
ACTG1/BAIAP2/ELMO2/HSP90AB1/LOC113983017/LOC113983228/LOC113993074/NCKIPSD
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034622" class="uri">GO:0034622</a>
</td>
<td style="text-align:left;">
cellular protein-containing complex assembly
</td>
<td style="text-align:left;">
4/420
</td>
<td style="text-align:left;">
18/12181
</td>
<td style="text-align:right;">
0.0029024
</td>
<td style="text-align:right;">
0.3351089
</td>
<td style="text-align:right;">
0.3221628
</td>
<td style="text-align:left;">
DNAJB12/LOC113995101/LOC113998422/TRAF2
</td>
<td style="text-align:right;">
4
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkred%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkred%20go-2.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkred%20go-3.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="darkmagenta---strength-1" class="section level3"
number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> darkmagenta -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkmagenta%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkmagenta%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-73-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-26" class="section level4" number="12.1.2.1">
<h4><span class="header-section-number">12.1.2.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in darkmagenta module in BSTm
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0055119" class="uri">GO:0055119</a>
</td>
<td style="text-align:left;">
relaxation of cardiac muscle
</td>
<td style="text-align:left;">
3/256
</td>
<td style="text-align:left;">
11/12181
</td>
<td style="text-align:right;">
0.0013359
</td>
<td style="text-align:right;">
0.5366384
</td>
<td style="text-align:right;">
0.5205780
</td>
<td style="text-align:left;">
ATP1A1/GSN/RGS2
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010887" class="uri">GO:0010887</a>
</td>
<td style="text-align:left;">
negative regulation of cholesterol storage
</td>
<td style="text-align:left;">
3/256
</td>
<td style="text-align:left;">
12/12181
</td>
<td style="text-align:right;">
0.0017537
</td>
<td style="text-align:right;">
0.5366384
</td>
<td style="text-align:right;">
0.5205780
</td>
<td style="text-align:left;">
ABCA7/LOC113986559/PPARD
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051014" class="uri">GO:0051014</a>
</td>
<td style="text-align:left;">
actin filament severing
</td>
<td style="text-align:left;">
3/256
</td>
<td style="text-align:left;">
12/12181
</td>
<td style="text-align:right;">
0.0017537
</td>
<td style="text-align:right;">
0.5366384
</td>
<td style="text-align:right;">
0.5205780
</td>
<td style="text-align:left;">
CAPG/FMNL1/GSN
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000377" class="uri">GO:2000377</a>
</td>
<td style="text-align:left;">
regulation of reactive oxygen species metabolic process
</td>
<td style="text-align:left;">
3/256
</td>
<td style="text-align:left;">
14/12181
</td>
<td style="text-align:right;">
0.0028127
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
BCO2/MPV17/PINK1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002028" class="uri">GO:0002028</a>
</td>
<td style="text-align:left;">
regulation of sodium ion transport
</td>
<td style="text-align:left;">
3/256
</td>
<td style="text-align:left;">
15/12181
</td>
<td style="text-align:right;">
0.0034618
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
ADRB2/ATP1A1/LOC113988625
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006904" class="uri">GO:0006904</a>
</td>
<td style="text-align:left;">
vesicle docking involved in exocytosis
</td>
<td style="text-align:left;">
4/256
</td>
<td style="text-align:left;">
34/12181
</td>
<td style="text-align:right;">
0.0053839
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
CPLX2/RAB15/RABEPK/VPS33B
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043407" class="uri">GO:0043407</a>
</td>
<td style="text-align:left;">
negative regulation of MAP kinase activity
</td>
<td style="text-align:left;">
4/256
</td>
<td style="text-align:left;">
34/12181
</td>
<td style="text-align:right;">
0.0053839
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
CDK5RAP3/LOC113991889/PIK3CB/RGS2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008203" class="uri">GO:0008203</a>
</td>
<td style="text-align:left;">
cholesterol metabolic process
</td>
<td style="text-align:left;">
5/256
</td>
<td style="text-align:left;">
58/12181
</td>
<td style="text-align:right;">
0.0072969
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
ABCA7/LOC113986559/LOC113991750/LRP5/PPARD
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030168" class="uri">GO:0030168</a>
</td>
<td style="text-align:left;">
platelet activation
</td>
<td style="text-align:left;">
6/256
</td>
<td style="text-align:left;">
83/12181
</td>
<td style="text-align:right;">
0.0079613
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
ADAMTS13/DGKZ/LOC113984938/LOC113991889/LOC113997278/PIK3CB
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051056" class="uri">GO:0051056</a>
</td>
<td style="text-align:left;">
regulation of small GTPase mediated signal transduction
</td>
<td style="text-align:left;">
8/256
</td>
<td style="text-align:left;">
141/12181
</td>
<td style="text-align:right;">
0.0098161
</td>
<td style="text-align:right;">
0.5634260
</td>
<td style="text-align:right;">
0.5465639
</td>
<td style="text-align:left;">
ARHGAP17/ARHGAP19/ARHGEF16/INPP5B/LOC113997278/OPHN1/RHOU/SYDE2
</td>
<td style="text-align:right;">
8
</td>
</tr>
</tbody>
</table>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkmagenta%20go-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<pre><code>## Warning in if (class(try(col2rgb(palette), silent = TRUE)) == &quot;try-error&quot;)
## stop(&quot;color palette is not correct&quot;): the condition has length &gt; 1 and only the
## first element will be used</code></pre>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20darkmagenta%20go-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="saddlebrown---strength" class="section level3" number="12.1.3">
<h3><span class="header-section-number">12.1.3</span> saddlebrown -
strength</h3>
<p><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20saddlebrown%20module-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/BSTm%20saddlebrown%20module-2.png" width="960" style="display: block; margin: auto;" /></p>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-74-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="gene-ontology-27" class="section level4" number="12.1.3.1">
<h4><span class="header-section-number">12.1.3.1</span> Gene
Ontology</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
BP enriched GO terms in saddlebrown module in BSTm
</caption>
<thead>
<tr>
<th style="text-align:left;">
ID
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:left;">
GeneRatio
</th>
<th style="text-align:left;">
BgRatio
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
p.adjust
</th>
<th style="text-align:right;">
qvalue
</th>
<th style="text-align:left;">
geneID
</th>
<th style="text-align:right;">
Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0015701" class="uri">GO:0015701</a>
</td>
<td style="text-align:left;">
bicarbonate transport
</td>
<td style="text-align:left;">
5/128
</td>
<td style="text-align:left;">
33/12181
</td>
<td style="text-align:right;">
0.0000222
</td>
<td style="text-align:right;">
0.0113948
</td>
<td style="text-align:right;">
0.0110398
</td>
<td style="text-align:left;">
LOC113995938/LOC114003675/RHAG/SLC4A1/SLC4A4
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0042307" class="uri">GO:0042307</a>
</td>
<td style="text-align:left;">
positive regulation of protein import into nucleus
</td>
<td style="text-align:left;">
5/128
</td>
<td style="text-align:left;">
37/12181
</td>
<td style="text-align:right;">
0.0000394
</td>
<td style="text-align:right;">
0.0113948
</td>
<td style="text-align:right;">
0.0110398
</td>
<td style="text-align:left;">
DMAP1/LOC113989827/LOC114001403/SMO/ZC3H12A
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000379" class="uri">GO:2000379</a>
</td>
<td style="text-align:left;">
positive regulation of reactive oxygen species metabolic process
</td>
<td style="text-align:left;">
4/128
</td>
<td style="text-align:left;">
20/12181
</td>
<td style="text-align:right;">
0.0000495
</td>
<td style="text-align:right;">
0.0113948
</td>
<td style="text-align:right;">
0.0110398
</td>
<td style="text-align:left;">
LOC114001403/NFE2L2/PDGFRB/ZC3H12A
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000036" class="uri">GO:2000036</a>
</td>
<td style="text-align:left;">
regulation of stem cell population maintenance
</td>
<td style="text-align:left;">
3/128
</td>
<td style="text-align:left;">
13/12181
</td>
<td style="text-align:right;">
0.0003001
</td>
<td style="text-align:right;">
0.0518481
</td>
<td style="text-align:right;">
0.0502329
</td>
<td style="text-align:left;">
SAV1/SMO/TAL1
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050801" class="uri">GO:0050801</a>
</td>
<td style="text-align:left;">
ion homeostasis
</td>
<td style="text-align:left;">
3/128
</td>
<td style="text-align:left;">
18/12181
</td>
<td style="text-align:right;">
0.0008240
</td>
<td style="text-align:right;">
0.1100942
</td>
<td style="text-align:right;">
0.1066645
</td>
<td style="text-align:left;">
SLC4A1/SLC4A4/WNK4
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006635" class="uri">GO:0006635</a>
</td>
<td style="text-align:left;">
fatty acid beta-oxidation
</td>
<td style="text-align:left;">
4/128
</td>
<td style="text-align:left;">
42/12181
</td>
<td style="text-align:right;">
0.0009560
</td>
<td style="text-align:right;">
0.1100942
</td>
<td style="text-align:right;">
0.1066645
</td>
<td style="text-align:left;">
DECR1/ECH1/LOC114001403/SESN2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0001525" class="uri">GO:0001525</a>
</td>
<td style="text-align:left;">
angiogenesis
</td>
<td style="text-align:left;">
8/128
</td>
<td style="text-align:left;">
203/12181
</td>
<td style="text-align:right;">
0.0013439
</td>
<td style="text-align:right;">
0.1326657
</td>
<td style="text-align:right;">
0.1285329
</td>
<td style="text-align:left;">
IL18/JUN/LOC113989827/LOC114001403/PLXND1/TAL1/WASF2/ZC3H12A
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098869" class="uri">GO:0098869</a>
</td>
<td style="text-align:left;">
cellular oxidant detoxification
</td>
<td style="text-align:left;">
4/128
</td>
<td style="text-align:left;">
55/12181
</td>
<td style="text-align:right;">
0.0026226
</td>
<td style="text-align:right;">
0.2265230
</td>
<td style="text-align:right;">
0.2194663
</td>
<td style="text-align:left;">
LOC113986834/LOC114003675/SELENOW/SESN2
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043123" class="uri">GO:0043123</a>
</td>
<td style="text-align:left;">
positive regulation of I-kappaB kinase/NF-kappaB signaling
</td>
<td style="text-align:left;">
6/128
</td>
<td style="text-align:left;">
146/12181
</td>
<td style="text-align:right;">
0.0043831
</td>
<td style="text-align:right;">
0.3224010
</td>
<td style="text-align:right;">
0.3123574
</td>
<td style="text-align:left;">
CARD11/LOC113989827/NEK6/TMEM101/TRIM32/WLS
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043249" class="uri">GO:0043249</a>
</td>
<td style="text-align:left;">
erythrocyte maturation
</td>
<td style="text-align:left;">
2/128
</td>
<td style="text-align:left;">
10/12181
</td>
<td style="text-align:right;">
0.0046657
</td>
<td style="text-align:right;">
0.3224010
</td>
<td style="text-align:right;">
0.3123574
</td>
<td style="text-align:left;">
LOC114001063/TAL1
</td>
<td style="text-align:right;">
2
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="wgcna-summary" class="section level1" number="13">
<h1><span class="header-section-number">13</span> WGCNA summary</h1>
<p><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-75-1.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-75-2.png" width="960" style="display: block; margin: auto;" /><img src="05_WGCNA_single_tissue_files/figure-html/unnamed-chunk-75-3.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Session Info</h1>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
## [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
## [5] LC_TIME=English_Australia.1252    
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] enrichplot_1.8.1            AnnotationHub_2.20.2       
##  [3] BiocFileCache_1.12.0        dbplyr_2.1.1               
##  [5] network_1.16.1              RColorBrewer_1.1-2         
##  [7] MASS_7.3-51.6               dplyr_1.0.7                
##  [9] ggnetwork_0.5.8             igraph_1.2.6               
## [11] ggfortify_0.4.11            GGally_2.1.0               
## [13] effectsize_0.8.6            car_3.0-8                  
## [15] carData_3.0-4               viridis_0.6.1              
## [17] viridisLite_0.4.0           rrvgo_1.0.2                
## [19] UpSetR_1.4.0                limma_3.44.3               
## [21] flashClust_1.01-2           org.Hs.eg.db_3.11.4        
## [23] AnnotationDbi_1.50.0        clusterProfiler_3.16.1     
## [25] fdrtool_1.2.15              ggpubr_0.4.0               
## [27] gridExtra_2.3               wesanderson_0.3.6          
## [29] VennDiagram_1.6.20          futile.logger_1.4.3        
## [31] kableExtra_1.3.1            WGCNA_1.69                 
## [33] fastcluster_1.1.25          dynamicTreeCut_1.63-1      
## [35] pheatmap_1.0.12             lubridate_1.8.0            
## [37] DESeq2_1.28.1               SummarizedExperiment_1.18.1
## [39] DelayedArray_0.14.0         matrixStats_0.56.0         
## [41] Biobase_2.48.0              GenomicRanges_1.40.0       
## [43] GenomeInfoDb_1.24.0         IRanges_2.22.2             
## [45] S4Vectors_0.26.1            BiocGenerics_0.34.0        
## [47] stringr_1.4.0               PCAtools_2.0.0             
## [49] cowplot_1.0.0               lattice_0.20-41            
## [51] ggrepel_0.8.2               ggplot2_3.4.4              
## [53] reshape2_1.4.4              plyr_1.8.6                 
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.1                ragg_1.2.1                   
##   [3] tidyr_1.1.4                   acepack_1.4.1                
##   [5] bit64_0.9-7                   knitr_1.29                   
##   [7] irlba_2.3.3                   data.table_1.13.0            
##   [9] rpart_4.1-15                  RCurl_1.98-1.2               
##  [11] doParallel_1.0.15             generics_0.0.2               
##  [13] snow_0.4-3                    preprocessCore_1.50.0        
##  [15] lambda.r_1.2.4                RSQLite_2.2.0                
##  [17] europepmc_0.4                 bit_4.0.4                    
##  [19] webshot_0.5.2                 xml2_1.3.2                   
##  [21] httpuv_1.5.4                  assertthat_0.2.1             
##  [23] xfun_0.29                     hms_1.1.1                    
##  [25] jquerylib_0.1.4               evaluate_0.14                
##  [27] promises_1.1.1                fansi_0.4.1                  
##  [29] progress_1.2.2                readxl_1.3.1                 
##  [31] DBI_1.1.0                     geneplotter_1.66.0           
##  [33] htmlwidgets_1.5.3             reshape_0.8.8                
##  [35] purrr_0.3.4                   ellipsis_0.3.2               
##  [37] backports_1.1.7               insight_0.19.6               
##  [39] annotate_1.66.0               gridBase_0.4-7               
##  [41] vctrs_0.6.5                   abind_1.4-5                  
##  [43] withr_2.5.2                   ggforce_0.3.2                
##  [45] triebeard_0.3.0               treemap_2.4-2                
##  [47] checkmate_2.0.0               prettyunits_1.1.1            
##  [49] cluster_2.1.0                 DOSE_3.14.0                  
##  [51] crayon_1.4.2                  genefilter_1.70.0            
##  [53] pkgconfig_2.0.3               slam_0.1-47                  
##  [55] labeling_0.3                  tweenr_1.0.1                 
##  [57] nlme_3.1-148                  wordcloud_2.6                
##  [59] nnet_7.3-14                   rlang_1.1.2                  
##  [61] lifecycle_1.0.4               downloader_0.4               
##  [63] rsvd_1.0.3                    cellranger_1.1.0             
##  [65] polyclip_1.10-0               datawizard_0.9.0             
##  [67] Matrix_1.6-4                  urltools_1.7.3               
##  [69] base64enc_0.1-3               ggridges_0.5.3               
##  [71] png_0.1-7                     parameters_0.21.3            
##  [73] bitops_1.0-6                  blob_1.2.1                   
##  [75] DelayedMatrixStats_1.10.0     qvalue_2.20.0                
##  [77] jpeg_0.1-8.1                  rstatix_0.7.0                
##  [79] gridGraphics_0.5-1            ggsignif_0.6.0               
##  [81] scales_1.3.0                  memoise_1.1.0                
##  [83] magrittr_2.0.1                zlibbioc_1.34.0              
##  [85] compiler_4.0.2                scatterpie_0.1.5             
##  [87] dqrng_0.2.1                   cli_3.6.1                    
##  [89] XVector_0.28.0                htmlTable_2.0.1              
##  [91] formatR_1.8                   Formula_1.2-3                
##  [93] mgcv_1.8-31                   tidyselect_1.1.0             
##  [95] stringi_1.4.6                 forcats_0.5.1                
##  [97] textshaping_0.3.6             highr_0.8                    
##  [99] yaml_2.2.1                    GOSemSim_2.14.2              
## [101] BiocSingular_1.4.0            locfit_1.5-9.4               
## [103] latticeExtra_0.6-29           fastmatch_1.1-0              
## [105] tools_4.0.2                   rio_0.5.16                   
## [107] rstudioapi_0.13               foreach_1.5.0                
## [109] foreign_0.8-80                farver_2.0.3                 
## [111] ggraph_2.0.5                  digest_0.6.25                
## [113] rvcheck_0.1.8                 BiocManager_1.30.10          
## [115] shiny_1.5.0                   Rcpp_1.0.7                   
## [117] broom_0.7.10                  BiocVersion_3.11.1           
## [119] later_1.1.0.1                 httr_1.4.2                   
## [121] colorspace_1.4-1              rvest_1.0.2                  
## [123] XML_3.99-0.5                  splines_4.0.2                
## [125] graphlayouts_0.7.1            ggplotify_0.0.5              
## [127] systemfonts_1.0.3             xtable_1.8-4                 
## [129] jsonlite_1.7.2                futile.options_1.0.1         
## [131] tidygraph_1.2.0               NLP_0.2-1                    
## [133] R6_2.4.1                      tm_0.7-8                     
## [135] Hmisc_4.4-0                   pillar_1.6.4                 
## [137] htmltools_0.5.2               mime_0.9                     
## [139] glue_1.5.1                    fastmap_1.1.0                
## [141] BiocParallel_1.22.0           interactiveDisplayBase_1.26.3
## [143] codetools_0.2-16              fgsea_1.14.0                 
## [145] utf8_1.1.4                    tibble_3.1.6                 
## [147] curl_4.3                      zip_2.0.4                    
## [149] GO.db_3.11.4                  openxlsx_4.1.5               
## [151] survival_3.2-3                rmarkdown_2.11               
## [153] munsell_0.5.0                 DO.db_2.9                    
## [155] GenomeInfoDbData_1.2.3        iterators_1.0.12             
## [157] impute_1.62.0                 haven_2.3.1                  
## [159] gtable_0.3.0                  bayestestR_0.13.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
