<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peri Bolton" />

<meta name="date" content="2022-02-06" />

<title>Candidate Gene Analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PIFI neurogenomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./candidate_gene_analysis.html">Candidate Genes</a>
    </li>
    <li>
      <a href="./test.html">Test</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/periperipatus/PIFI_brain_transcriptome">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Candidate Gene Analysis</h1>
<h3 class="subtitle">For manuscript: Neurogenomic landscape of male cooperative behavior in a wild bird</h3>
<h4 class="author">Peri Bolton</h4>
<h4 class="date">2022-02-06</h4>

</div>


<p>This document contains the differential expression results from DESeq2 for our candidate genes. These candidates are: <code>AR, SRD5A2, LOC113993669, ESR1, ESR2, PGR, GNRH1, PRL, PRLR, VIP, VIPR1, VIPR2, LOC113983511, OXTR, LOC113983498, AVPR1A, AVPR1B, LOC113982601</code>.</p>
<ul>
<li><p>LOC113983511 is OXT (Mesotocin)</p></li>
<li><p>LOC113993669 is CYP19A1 (Aromatase)</p></li>
<li><p>LOC113983498 is vasotocin-neurophysin (AVP precursor)</p></li>
<li><p>LOC113982601 is AVPR2.</p></li>
</ul>
<p>Originally I was correcting for multiple comparisons using the raw p-values for the number of genes expressed in a tissue (using Benjamini-Hochberg). However, these were more conservative than the corrected p-values from DESeq2. Therefore, i only present the corrected p-values from DESeq2 based on the number of genes expressed in that tissue.</p>
<div id="logfoldchange-heatmaps" class="section level1" number="1">
<h1><span class="header-section-number">1</span> LogfoldChange heatmaps</h1>
<p>A simple way to present the candidate gene data is through this grid. These are the raw results that went into the publication - but some results will have been subsequently manually excluded on the basis of R^2 and strongly influential individuals.</p>
<div id="social-status" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Social Status</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>results_data_status<span class="ot">&lt;-</span> results_data[<span class="fu">grepl</span>(<span class="st">&quot;status&quot;</span>, results_data<span class="sc">$</span>Test,<span class="at">ignore.case =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>results_data_status<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="fu">factor</span>(results_data_status<span class="sc">$</span>gene, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;AR&quot;</span>, <span class="st">&quot;SRD5A2&quot;</span>, <span class="st">&quot;LOC113993669&quot;</span>, <span class="st">&quot;ESR1&quot;</span>, <span class="st">&quot;ESR2&quot;</span>, <span class="st">&quot;PGR&quot;</span>,<span class="st">&quot;GNRH1&quot;</span>, <span class="st">&quot;PRL&quot;</span>,<span class="st">&quot;PRLR&quot;</span>,<span class="st">&quot;VIP&quot;</span>, <span class="st">&quot;VIPR1&quot;</span>, <span class="st">&quot;VIPR2&quot;</span>,<span class="st">&quot;LOC113983511&quot;</span>, <span class="st">&quot;OXTR&quot;</span>, <span class="st">&quot;LOC113983498&quot;</span>, <span class="st">&quot;AVPR1A&quot;</span>, <span class="st">&quot;AVPR1B&quot;</span>, <span class="st">&quot;LOC113982601&quot;</span>))</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>results_data_status<span class="sc">$</span>gene<span class="ot">&lt;-</span> plyr<span class="sc">::</span><span class="fu">revalue</span>(results_data_status<span class="sc">$</span>gene, <span class="at">replace=</span><span class="fu">c</span>(<span class="st">&quot;LOC113993669&quot;</span><span class="ot">=</span><span class="st">&quot;CYP19A1&quot;</span>,<span class="st">&quot;LOC113983511&quot;</span><span class="ot">=</span><span class="st">&quot;OXT&quot;</span>, <span class="st">&quot;LOC113983498&quot;</span><span class="ot">=</span><span class="st">&quot;AVP&quot;</span>, <span class="st">&quot;LOC113982601&quot;</span><span class="ot">=</span><span class="st">&quot;AVPR2&quot;</span>))</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>res_data_status<span class="ot">&lt;-</span> results_data_status[,<span class="fu">c</span>(<span class="st">&quot;log2FoldChange&quot;</span>, <span class="st">&quot;gene&quot;</span>,<span class="st">&quot;Tissue&quot;</span>)]</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>res_data_status<span class="ot">&lt;-</span> <span class="fu">dcast</span>(res_data_status, gene <span class="sc">~</span>Tissue, <span class="at">value.var=</span><span class="st">&quot;log2FoldChange&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_data_status)<span class="ot">&lt;-</span> res_data_status<span class="sc">$</span>gene</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>res_data_status<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>res_data_status<span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(res_data_status)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>res_int<span class="ot">&lt;-</span> results_int[,<span class="fu">c</span>(<span class="st">&quot;log2FoldChange&quot;</span>, <span class="st">&quot;gene&quot;</span>,<span class="st">&quot;Tissue&quot;</span>)]</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>res_int<span class="ot">&lt;-</span> <span class="fu">dcast</span>(res_int, gene <span class="sc">~</span>Tissue, <span class="at">value.var=</span><span class="st">&quot;log2FoldChange&quot;</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(res_int)<span class="ot">&lt;-</span> res_int<span class="sc">$</span>gene</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>res_int<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>res_int<span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(res_int)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>p_data_status<span class="ot">&lt;-</span> results_data_status[,<span class="fu">c</span>(<span class="st">&quot;pvalue&quot;</span>, <span class="st">&quot;gene&quot;</span>,<span class="st">&quot;Tissue&quot;</span>)]</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>p_data_status<span class="ot">&lt;-</span> <span class="fu">dcast</span>(p_data_status, gene <span class="sc">~</span>Tissue, <span class="at">value.var=</span><span class="st">&quot;pvalue&quot;</span>)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(p_data_status)<span class="ot">&lt;-</span> p_data_status<span class="sc">$</span>gene</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>p_data_status<span class="sc">$</span>gene<span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>p_data_status<span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(p_data_status)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>sig<span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;&quot;</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(p_data_status),<span class="at">ncol=</span><span class="fu">ncol</span>(p_data_status))</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;&quot;</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(p_data_status),<span class="at">ncol=</span><span class="fu">ncol</span>(p_data_status))</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>logsig<span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="st">&quot;&quot;</span>, <span class="at">nrow=</span><span class="fu">nrow</span>(p_data_status),<span class="at">ncol=</span><span class="fu">ncol</span>(p_data_status))</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(p_data_status)){</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(p_data_status)){</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    sig[i,j]<span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">ifelse</span>(p_data_status[i,j]<span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="fu">signif</span>(p_data_status[i,j],<span class="dv">1</span>),<span class="st">&quot;&quot;</span>))</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(res_data_status<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> sig<span class="sc">!=</span><span class="st">&quot;&quot;</span>,<span class="sc">-</span><span class="dv">1</span>,<span class="fu">ifelse</span>(res_data_status<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> sig<span class="sc">!=</span><span class="st">&quot;&quot;</span>,<span class="dv">1</span>,<span class="cn">NA</span>))</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>logsig<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(res<span class="sc">==-</span><span class="dv">1</span>,<span class="sc">-</span>(<span class="sc">-</span><span class="fu">log10</span>(p_data_status)), <span class="fu">ifelse</span>(res<span class="sc">==</span><span class="dv">1</span>, <span class="sc">-</span><span class="fu">log10</span>(p_data_status),<span class="cn">NA</span>))</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>sig<span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(sig),<span class="st">&quot;&quot;</span>, <span class="fu">as.character</span>(sig))</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(res, <span class="at">na_col =</span> <span class="st">&quot;white&quot;</span>, <span class="at">cluster_rows=</span><span class="cn">FALSE</span>, <span class="at">cluster_cols=</span><span class="cn">FALSE</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">border=</span><span class="cn">NA</span>, <span class="at">display_numbers =</span> sig, <span class="at">color=</span><span class="fu">blueWhiteRed</span>(<span class="dv">20</span>), <span class="at">border_color =</span> <span class="st">&quot;grey90&quot;</span>, <span class="at">legend=</span><span class="cn">FALSE</span>, <span class="at">main=</span><span class="st">&quot;Raw candidate gene results from social status&quot;</span>)</span></code></pre></div>
<p><img src="06_candidate_gene_analysis_files/figure-html/Status-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mean-testosterone-phenotype" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> mean testosterone phenotype</h2>
<p><img src="06_candidate_gene_analysis_files/figure-html/mean%20T-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="status-by-mean-t-interaction" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Status by mean T interaction</h2>
<p><img src="06_candidate_gene_analysis_files/figure-html/status%20by%20mean%20T-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cooperative-tendency-social-network-strength" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Cooperative tendency (Social network strength)</h2>
<p><img src="06_candidate_gene_analysis_files/figure-html/strength-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gonads" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Gonads</h1>
<p>The next sections are scatter plots for each of the tissues showing what each of the genes are doing for manual validation.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Gonads (GON). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
AVPR1A
</td>
<td style="text-align:right;">
188.293
</td>
<td style="text-align:right;">
0.793
</td>
<td style="text-align:right;">
0.195
</td>
<td style="text-align:right;">
4.059
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.014
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
PRLR
</td>
<td style="text-align:right;">
1011.603
</td>
<td style="text-align:right;">
0.324
</td>
<td style="text-align:right;">
0.139
</td>
<td style="text-align:right;">
2.332
</td>
<td style="text-align:right;">
0.020
</td>
<td style="text-align:right;">
0.442
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
VIPR1
</td>
<td style="text-align:right;">
120.523
</td>
<td style="text-align:right;">
0.459
</td>
<td style="text-align:right;">
0.200
</td>
<td style="text-align:right;">
2.292
</td>
<td style="text-align:right;">
0.022
</td>
<td style="text-align:right;">
0.590
</td>
</tr>
</tbody>
</table>
<p>Here only the data with continuous variables were at all significant so these will be plotted with the extracted Normalized data.</p>
<p><img src="06_candidate_gene_analysis_files/figure-html/GON%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="status-specific-gene-expression-in-the-gonads" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Status specific gene expression in the Gonads</h2>
<p>If not already identified as a possible interaction from multiple incidences in the single trait comparison, I also used DESeq2 to extract possible interactions.</p>
<p>Then, I am going to use AIC on simple linear models to decide amongst models to decide whether interaction effects or single term models better explain the data.</p>
<p>Note that the DESeq2 model for Status only is actually GeneExp ~ Batch + Status, and includes 4 additional samples. However, the linear models here don’t differ in biological interpretation.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>VIPR_GON_lm_Status<span class="ot">&lt;-</span> <span class="fu">lm</span>(VIPR2 <span class="sc">~</span> Year <span class="sc">+</span> Status, <span class="at">data=</span>VIPR2)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>VIPR_GON_lm_meanT<span class="ot">&lt;-</span> <span class="fu">lm</span>(VIPR2 <span class="sc">~</span> Year <span class="sc">+</span> mean_T, <span class="at">data=</span>VIPR2)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>VIPR_GON_lm_int<span class="ot">&lt;-</span> <span class="fu">lm</span>(VIPR2 <span class="sc">~</span> Year <span class="sc">+</span> Status<span class="sc">*</span>mean_T, <span class="at">data=</span>VIPR2)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#against the NULL model</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>VIPR_GON_lm_null<span class="ot">&lt;-</span> <span class="fu">lm</span>(VIPR2 <span class="sc">~</span> Year, <span class="at">data=</span>VIPR2)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>model<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;VIPR2 ~ Year&quot;</span>, <span class="st">&quot;VIPR2 ~ Year + Status&quot;</span>, <span class="st">&quot;VIPR2 ~ Year + mean T&quot;</span>,<span class="st">&quot;VIPR2 ~ Year + Status * mean T&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>AIC<span class="ot">=</span><span class="fu">c</span>(<span class="fu">AIC</span>(VIPR_GON_lm_null), <span class="fu">AIC</span>(VIPR_GON_lm_Status),<span class="fu">AIC</span>(VIPR_GON_lm_meanT),<span class="fu">AIC</span>(VIPR_GON_lm_int))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>deltaAIC<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="cn">NA</span>, AIC[<span class="dv">1</span>]<span class="sc">-</span>AIC[<span class="dv">2</span>], AIC[<span class="dv">1</span>]<span class="sc">-</span>AIC[<span class="dv">3</span>], AIC[<span class="dv">1</span>]<span class="sc">-</span>AIC[<span class="dv">4</span>])</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>lm_p<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">anova</span>(VIPR_GON_lm_null)<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">1</span>],<span class="fu">anova</span>(VIPR_GON_lm_Status)<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>],<span class="fu">anova</span>(VIPR_GON_lm_meanT)<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">2</span>], <span class="fu">anova</span>(VIPR_GON_lm_int)<span class="sc">$</span><span class="st">&#39;Pr(&gt;F)&#39;</span>[<span class="dv">4</span>])</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>DESeq2_raw_p<span class="ot">=</span><span class="fu">c</span>(<span class="cn">NA</span>,results_gon<span class="sc">$</span>pvalue[results_gon<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;VIPR2&quot;</span> <span class="sc">&amp;</span> results_gon<span class="sc">$</span>Test<span class="sc">==</span><span class="st">&quot;Status&quot;</span>], results_gon<span class="sc">$</span>pvalue[results_gon<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;VIPR2&quot;</span> <span class="sc">&amp;</span> results_gon<span class="sc">$</span>Test<span class="sc">==</span><span class="st">&quot;mean T&quot;</span>], results_int<span class="sc">$</span>pvalue[results_int<span class="sc">$</span>gene<span class="sc">==</span><span class="st">&quot;VIPR2&quot;</span> <span class="sc">&amp;</span> results_int<span class="sc">$</span>Tissue<span class="sc">==</span><span class="st">&quot;GON&quot;</span>])</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">data.frame</span>(model, AIC, deltaAIC, lm_p, DESeq2_raw_p),<span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">padding=</span><span class="dv">20</span>, <span class="at">caption=</span><span class="st">&quot;Model comparisons in the Gonads for VIPR2&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>()</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model comparisons in the Gonads for VIPR2
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIPR2 ~ Year
</td>
<td style="text-align:right;">
150.6986
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.3814937
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR2 ~ Year + Status
</td>
<td style="text-align:right;">
152.5878
</td>
<td style="text-align:right;">
-1.8892097
</td>
<td style="text-align:right;">
0.7791835
</td>
<td style="text-align:right;">
0.9993856
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR2 ~ Year + mean T
</td>
<td style="text-align:right;">
151.2701
</td>
<td style="text-align:right;">
-0.5715250
</td>
<td style="text-align:right;">
0.3139082
</td>
<td style="text-align:right;">
0.0614576
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR2 ~ Year + Status * mean T
</td>
<td style="text-align:right;">
150.0352
</td>
<td style="text-align:right;">
0.6633609
</td>
<td style="text-align:right;">
0.0933626
</td>
<td style="text-align:right;">
0.0061059
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>b<span class="ot">&lt;-</span> <span class="fu">ggtexttable</span>(<span class="fu">data.frame</span>(<span class="at">model=</span>model, <span class="at">AIC=</span><span class="fu">signif</span>(AIC,<span class="dv">4</span>), <span class="at">deltaAIC=</span><span class="fu">signif</span>(deltaAIC,<span class="dv">1</span>), <span class="at">lm_p=</span><span class="fu">signif</span>(lm_p,<span class="dv">1</span>), <span class="at">DESeq_raw_p=</span><span class="fu">signif</span>(DESeq2_raw_p,<span class="dv">1</span>)), <span class="at">rows=</span><span class="cn">NULL</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>figure<span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(a,b, <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>), <span class="at">nrow=</span><span class="dv">2</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(filename=&quot;../Candidate_gene_results/GON_interaction.png&quot;, figure,device=&quot;png&quot;, width=15, height=20, units=&quot;cm&quot;, dpi=300, bg=&quot;white&quot;)</span></span></code></pre></div>
</div>
</div>
<div id="pituitary" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Pituitary</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Pituitary (PIT). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
AVPR1B
</td>
<td style="text-align:right;">
1814.603
</td>
<td style="text-align:right;">
0.298
</td>
<td style="text-align:right;">
0.137
</td>
<td style="text-align:right;">
2.172
</td>
<td style="text-align:right;">
0.030
</td>
<td style="text-align:right;">
0.446
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
ESR1
</td>
<td style="text-align:right;">
1725.149
</td>
<td style="text-align:right;">
-0.661
</td>
<td style="text-align:right;">
0.253
</td>
<td style="text-align:right;">
-2.613
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
0.433
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
ESR2
</td>
<td style="text-align:right;">
15.496
</td>
<td style="text-align:right;">
-0.735
</td>
<td style="text-align:right;">
0.364
</td>
<td style="text-align:right;">
-2.020
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.509
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
123.156
</td>
<td style="text-align:right;">
6.417
</td>
<td style="text-align:right;">
1.578
</td>
<td style="text-align:right;">
4.066
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.023
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
118.037
</td>
<td style="text-align:right;">
4.366
</td>
<td style="text-align:right;">
1.243
</td>
<td style="text-align:right;">
3.513
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.074
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113983511
</td>
<td style="text-align:right;">
107.249
</td>
<td style="text-align:right;">
4.522
</td>
<td style="text-align:right;">
1.424
</td>
<td style="text-align:right;">
3.176
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.190
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
LOC113983511
</td>
<td style="text-align:right;">
62.816
</td>
<td style="text-align:right;">
3.493
</td>
<td style="text-align:right;">
1.211
</td>
<td style="text-align:right;">
2.885
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.211
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
PRL
</td>
<td style="text-align:right;">
379818.057
</td>
<td style="text-align:right;">
-0.553
</td>
<td style="text-align:right;">
0.136
</td>
<td style="text-align:right;">
-4.059
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.031
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
VIPR2
</td>
<td style="text-align:right;">
322.934
</td>
<td style="text-align:right;">
-0.519
</td>
<td style="text-align:right;">
0.218
</td>
<td style="text-align:right;">
-2.382
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
0.632
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
VIPR2
</td>
<td style="text-align:right;">
335.526
</td>
<td style="text-align:right;">
-0.426
</td>
<td style="text-align:right;">
0.160
</td>
<td style="text-align:right;">
-2.662
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.279
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/pit%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="status-specific-gene-expression-in-the-pituitary" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Status specific gene expression in the Pituitary</h2>
<p>I am excluding AVP and OXT because the expression patterns are odd.</p>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the pituitary (pit) for how AVPR1A expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch
</td>
<td style="text-align:right;">
115.118
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.909
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + Status
</td>
<td style="text-align:right;">
117.005
</td>
<td style="text-align:right;">
-1.886
</td>
<td style="text-align:right;">
0.776
</td>
<td style="text-align:right;">
0.786
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + mean T
</td>
<td style="text-align:right;">
115.991
</td>
<td style="text-align:right;">
-0.873
</td>
<td style="text-align:right;">
0.371
</td>
<td style="text-align:right;">
0.512
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
113.144
</td>
<td style="text-align:right;">
1.975
</td>
<td style="text-align:right;">
0.073
</td>
<td style="text-align:right;">
0.030
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="ventromedial-hypothalamus" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Ventromedial Hypothalamus</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Ventromedial Hypothalamus (VMH). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
951.594
</td>
<td style="text-align:right;">
0.632
</td>
<td style="text-align:right;">
0.246
</td>
<td style="text-align:right;">
2.567
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
0.104
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
ESR2
</td>
<td style="text-align:right;">
944.935
</td>
<td style="text-align:right;">
-0.445
</td>
<td style="text-align:right;">
0.150
</td>
<td style="text-align:right;">
-2.962
</td>
<td style="text-align:right;">
0.003
</td>
<td style="text-align:right;">
0.660
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
GNRH1
</td>
<td style="text-align:right;">
40.945
</td>
<td style="text-align:right;">
2.786
</td>
<td style="text-align:right;">
1.312
</td>
<td style="text-align:right;">
2.124
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
0.281
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113982601
</td>
<td style="text-align:right;">
10.815
</td>
<td style="text-align:right;">
-1.125
</td>
<td style="text-align:right;">
0.422
</td>
<td style="text-align:right;">
-2.667
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.588
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
1481.917
</td>
<td style="text-align:right;">
2.511
</td>
<td style="text-align:right;">
0.810
</td>
<td style="text-align:right;">
3.102
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.327
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113983511
</td>
<td style="text-align:right;">
1460.675
</td>
<td style="text-align:right;">
3.140
</td>
<td style="text-align:right;">
1.088
</td>
<td style="text-align:right;">
2.887
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.394
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/VMH%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Nothing convincingly significant after FDR correction!</p>
<div id="status-specific-gene-expression-in-the-vmh" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Status specific gene expression in the VMH</h2>
<div id="pgr" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> PGR</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Rubbish.</p>
</div>
<div id="gnrh1" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> GNRH1</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Also Rubbish.</p>
</div>
<div id="avpr1a" class="section level3" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> AVPR1A</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Also Rubbish.</p>
</div>
</div>
</div>
<div id="anterior-hypothalamus" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Anterior Hypothalamus</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Anterior Hypothalamus (AH). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model)
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
</tr>
</tbody>
</table>
<p>No candidate genes significant in the AH</p>
<p>No interaction effects</p>
</div>
<div id="paraventricular-nucleus" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Paraventricular Nucleus</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Paraventricular Nucleus (PVN). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
543.241
</td>
<td style="text-align:right;">
0.353
</td>
<td style="text-align:right;">
0.146
</td>
<td style="text-align:right;">
2.414
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.552
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
543.241
</td>
<td style="text-align:right;">
0.192
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
2.113
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:right;">
0.584
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
ESR2
</td>
<td style="text-align:right;">
818.718
</td>
<td style="text-align:right;">
-0.641
</td>
<td style="text-align:right;">
0.235
</td>
<td style="text-align:right;">
-2.725
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
0.177
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
LOC113982601
</td>
<td style="text-align:right;">
14.871
</td>
<td style="text-align:right;">
0.498
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
2.263
</td>
<td style="text-align:right;">
0.024
</td>
<td style="text-align:right;">
0.581
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
30862.339
</td>
<td style="text-align:right;">
-0.504
</td>
<td style="text-align:right;">
0.228
</td>
<td style="text-align:right;">
-2.216
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
0.581
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
LOC113983511
</td>
<td style="text-align:right;">
32779.237
</td>
<td style="text-align:right;">
-0.331
</td>
<td style="text-align:right;">
0.158
</td>
<td style="text-align:right;">
-2.096
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
0.594
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
VIPR1
</td>
<td style="text-align:right;">
380.233
</td>
<td style="text-align:right;">
0.307
</td>
<td style="text-align:right;">
0.136
</td>
<td style="text-align:right;">
2.257
</td>
<td style="text-align:right;">
0.024
</td>
<td style="text-align:right;">
0.358
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/PVN%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-paraventricular-nucleus" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Interaction in the Paraventricular Nucleus</h2>
<div id="ar" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> AR</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model comparisons in the Paraventricular Nucleus for Androgen Receptor
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AR ~ 1
</td>
<td style="text-align:right;">
157.941
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status
</td>
<td style="text-align:right;">
159.910
</td>
<td style="text-align:right;">
-1.969
</td>
<td style="text-align:right;">
0.876
</td>
<td style="text-align:right;">
0.847
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ mean T
</td>
<td style="text-align:right;">
156.546
</td>
<td style="text-align:right;">
1.394
</td>
<td style="text-align:right;">
0.101
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Strength
</td>
<td style="text-align:right;">
155.474
</td>
<td style="text-align:right;">
2.467
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.035
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status * mean T
</td>
<td style="text-align:right;">
152.126
</td>
<td style="text-align:right;">
5.815
</td>
<td style="text-align:right;">
0.039
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="medial-preoptic-area-pom" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Medial Preoptic Area (POM)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Medial Preoptic Area (POM). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
794.629
</td>
<td style="text-align:right;">
0.665
</td>
<td style="text-align:right;">
0.128
</td>
<td style="text-align:right;">
5.191
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
GNRH1
</td>
<td style="text-align:right;">
1318.753
</td>
<td style="text-align:right;">
0.399
</td>
<td style="text-align:right;">
0.144
</td>
<td style="text-align:right;">
2.769
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
0.208
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113993669
</td>
<td style="text-align:right;">
1533.720
</td>
<td style="text-align:right;">
0.546
</td>
<td style="text-align:right;">
0.227
</td>
<td style="text-align:right;">
2.405
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.370
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
PRLR
</td>
<td style="text-align:right;">
4007.589
</td>
<td style="text-align:right;">
0.254
</td>
<td style="text-align:right;">
0.076
</td>
<td style="text-align:right;">
3.360
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.054
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
828.546
</td>
<td style="text-align:right;">
0.667
</td>
<td style="text-align:right;">
0.189
</td>
<td style="text-align:right;">
3.521
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.347
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
1060.152
</td>
<td style="text-align:right;">
0.443
</td>
<td style="text-align:right;">
0.161
</td>
<td style="text-align:right;">
2.747
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
0.701
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
PRLR
</td>
<td style="text-align:right;">
4693.061
</td>
<td style="text-align:right;">
0.232
</td>
<td style="text-align:right;">
0.096
</td>
<td style="text-align:right;">
2.412
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.910
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/POM%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-medial-preoptic-area" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Interaction in the Medial Preoptic Area</h2>
<div id="vip" class="section level3" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> VIP</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the medial preoptic areaea (POM) for how VIP expression is related to mean T and Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
residual_df
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIP ~ 1
</td>
<td style="text-align:right;">
175.598
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Status
</td>
<td style="text-align:right;">
177.558
</td>
<td style="text-align:right;">
-1.960
</td>
<td style="text-align:right;">
0.861
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.656
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ mean T
</td>
<td style="text-align:right;">
177.593
</td>
<td style="text-align:right;">
-1.995
</td>
<td style="text-align:right;">
0.951
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.948
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Status * mean T
</td>
<td style="text-align:right;">
173.806
</td>
<td style="text-align:right;">
1.791
</td>
<td style="text-align:right;">
0.032
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.003
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="checking-other-multiple-hits-in-the-pom" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Checking other multiple hits in the POM</h2>
<div id="ar-1" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> AR</h3>
<img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the medial preoptic area (POM) for how AR expression is related to mean T and Status
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AR ~ 1
</td>
<td style="text-align:right;">
157.332
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status
</td>
<td style="text-align:right;">
155.312
</td>
<td style="text-align:right;">
2.020
</td>
<td style="text-align:right;">
0.077
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ mean T
</td>
<td style="text-align:right;">
144.968
</td>
<td style="text-align:right;">
12.364
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status * mean T
</td>
<td style="text-align:right;">
147.226
</td>
<td style="text-align:right;">
10.106
</td>
<td style="text-align:right;">
0.498
</td>
<td style="text-align:right;">
0.97
</td>
</tr>
</tbody>
</table>
</div>
<div id="prlr" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> PRLR</h3>
<img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PRLR ~ 1
</td>
<td style="text-align:right;">
175.706
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Status
</td>
<td style="text-align:right;">
175.772
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.221
</td>
<td style="text-align:right;">
0.016
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ mean T
</td>
<td style="text-align:right;">
168.500
</td>
<td style="text-align:right;">
7.207
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Status * mean T
</td>
<td style="text-align:right;">
169.436
</td>
<td style="text-align:right;">
6.271
</td>
<td style="text-align:right;">
0.179
</td>
<td style="text-align:right;">
0.178
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dorsomedial-intercollicular-nucleus-ico" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Dorsomedial Intercollicular Nucleus (ICo)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Dorsomedial Intercollicular Nucleus (ICo). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
405.910
</td>
<td style="text-align:right;">
-0.983
</td>
<td style="text-align:right;">
0.387
</td>
<td style="text-align:right;">
-2.542
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0.172
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
VIPR2
</td>
<td style="text-align:right;">
150.055
</td>
<td style="text-align:right;">
-0.351
</td>
<td style="text-align:right;">
0.146
</td>
<td style="text-align:right;">
-2.410
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.195
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/ICO%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-ico" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Interaction in the Ico</h2>
<div id="avpr1a-1" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> AVPR1A</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the dorsomedial intercollicular nucleus (ICo) for how AVPR1A expression, status and mean T influences social behavior
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + Year
</td>
<td style="text-align:right;">
123.327
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.946
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + Year + Status
</td>
<td style="text-align:right;">
124.515
</td>
<td style="text-align:right;">
-1.188
</td>
<td style="text-align:right;">
0.488
</td>
<td style="text-align:right;">
0.703
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~Batch + Year + mean T
</td>
<td style="text-align:right;">
125.008
</td>
<td style="text-align:right;">
-1.681
</td>
<td style="text-align:right;">
0.664
</td>
<td style="text-align:right;">
0.649
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Batch + Year + Status * mean T
</td>
<td style="text-align:right;">
121.908
</td>
<td style="text-align:right;">
1.419
</td>
<td style="text-align:right;">
0.121
</td>
<td style="text-align:right;">
0.043
</td>
</tr>
</tbody>
</table>
<p>###AVP</p>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the dorsomedial intercollicular nucleus (ICo) for how Arginine Vasotocin (AVP) expression
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AVP ~ Batch + Year
</td>
<td style="text-align:right;">
98.131
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.376
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~ Batch + Year + Status
</td>
<td style="text-align:right;">
100.045
</td>
<td style="text-align:right;">
-1.915
</td>
<td style="text-align:right;">
0.822
</td>
<td style="text-align:right;">
0.399
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~ Batch + Year + mean T
</td>
<td style="text-align:right;">
97.357
</td>
<td style="text-align:right;">
0.774
</td>
<td style="text-align:right;">
0.199
</td>
<td style="text-align:right;">
0.323
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~ Batch + Year + Status * mean T
</td>
<td style="text-align:right;">
95.611
</td>
<td style="text-align:right;">
2.520
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
0.186
</td>
</tr>
</tbody>
</table>
<p>###OXTR</p>
<img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" />
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the dorsomedial intercollicular nucleus (ICo) for how Arginine Vasotocin (AVP) expression
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
OXTR ~ Batch + Year
</td>
<td style="text-align:right;">
110.127
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.341
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
OXTR ~ Batch + Year + Status
</td>
<td style="text-align:right;">
112.094
</td>
<td style="text-align:right;">
-1.967
</td>
<td style="text-align:right;">
0.889
</td>
<td style="text-align:right;">
0.831
</td>
</tr>
<tr>
<td style="text-align:left;">
OXTR ~ Batch + Year + mean T
</td>
<td style="text-align:right;">
109.912
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.252
</td>
<td style="text-align:right;">
0.286
</td>
</tr>
<tr>
<td style="text-align:left;">
OXTR ~ Batch + Year + Status * mean T
</td>
<td style="text-align:right;">
105.092
</td>
<td style="text-align:right;">
5.035
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.028
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="midbrain-central-gray-gct" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Midbrain Central Gray (GCT)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Midbrain Central Grey (GCt). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
149.977
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:right;">
2.148
</td>
<td style="text-align:right;">
0.032
</td>
<td style="text-align:right;">
0.530
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
LOC113983511
</td>
<td style="text-align:right;">
8.100
</td>
<td style="text-align:right;">
-3.087
</td>
<td style="text-align:right;">
1.103
</td>
<td style="text-align:right;">
-2.799
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
0.175
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113993669
</td>
<td style="text-align:right;">
17.704
</td>
<td style="text-align:right;">
-1.269
</td>
<td style="text-align:right;">
0.420
</td>
<td style="text-align:right;">
-3.025
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.347
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
36.895
</td>
<td style="text-align:right;">
-0.801
</td>
<td style="text-align:right;">
0.327
</td>
<td style="text-align:right;">
-2.449
</td>
<td style="text-align:right;">
0.014
</td>
<td style="text-align:right;">
0.710
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
36.895
</td>
<td style="text-align:right;">
-0.656
</td>
<td style="text-align:right;">
0.194
</td>
<td style="text-align:right;">
-3.373
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.138
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
1047.430
</td>
<td style="text-align:right;">
-0.831
</td>
<td style="text-align:right;">
0.316
</td>
<td style="text-align:right;">
-2.629
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
0.589
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
1036.449
</td>
<td style="text-align:right;">
-0.852
</td>
<td style="text-align:right;">
0.427
</td>
<td style="text-align:right;">
-1.995
</td>
<td style="text-align:right;">
0.046
</td>
<td style="text-align:right;">
0.637
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
1047.430
</td>
<td style="text-align:right;">
-0.555
</td>
<td style="text-align:right;">
0.207
</td>
<td style="text-align:right;">
-2.684
</td>
<td style="text-align:right;">
0.007
</td>
<td style="text-align:right;">
0.340
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/GCT%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-midbrain-central-gray" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Interaction in the midbrain central gray</h2>
<div id="vip-1" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> VIP</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the midbrain central grey (GCt) for how VIP expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIP ~ Batch
</td>
<td style="text-align:right;">
192.194
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.582
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + Status
</td>
<td style="text-align:right;">
189.632
</td>
<td style="text-align:right;">
2.561
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
0.046
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + mean T
</td>
<td style="text-align:right;">
186.622
</td>
<td style="text-align:right;">
5.571
</td>
<td style="text-align:right;">
0.020
</td>
<td style="text-align:right;">
0.009
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + Strength
</td>
<td style="text-align:right;">
190.152
</td>
<td style="text-align:right;">
2.042
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.007
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
179.398
</td>
<td style="text-align:right;">
12.796
</td>
<td style="text-align:right;">
0.025
</td>
<td style="text-align:right;">
0.005
</td>
</tr>
</tbody>
</table>
</div>
<div id="pgr-1" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> PGR</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the midbrain central grey (GCt) for how Progesterone (PGR) expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PGR ~ Batch
</td>
<td style="text-align:right;">
111.711
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.883
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
PGR ~ Batch + Status
</td>
<td style="text-align:right;">
111.553
</td>
<td style="text-align:right;">
0.158
</td>
<td style="text-align:right;">
0.216
</td>
<td style="text-align:right;">
0.146
</td>
</tr>
<tr>
<td style="text-align:left;">
PGR ~ Batch + mean T
</td>
<td style="text-align:right;">
107.224
</td>
<td style="text-align:right;">
4.487
</td>
<td style="text-align:right;">
0.032
</td>
<td style="text-align:right;">
0.014
</td>
</tr>
<tr>
<td style="text-align:left;">
PGR ~ Batch + Strength
</td>
<td style="text-align:right;">
107.083
</td>
<td style="text-align:right;">
4.628
</td>
<td style="text-align:right;">
0.030
</td>
<td style="text-align:right;">
0.001
</td>
</tr>
<tr>
<td style="text-align:left;">
PGR ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
95.749
</td>
<td style="text-align:right;">
15.962
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.004
</td>
</tr>
</tbody>
</table>
</div>
<div id="esr1" class="section level3" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> ESR1</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ESR1 ~ Batch
</td>
<td style="text-align:right;">
117.237
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.681
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ Batch + Status
</td>
<td style="text-align:right;">
119.168
</td>
<td style="text-align:right;">
-1.931
</td>
<td style="text-align:right;">
0.825
</td>
<td style="text-align:right;">
0.821
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ Batch + mean T
</td>
<td style="text-align:right;">
117.949
</td>
<td style="text-align:right;">
-0.712
</td>
<td style="text-align:right;">
0.339
</td>
<td style="text-align:right;">
0.428
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
113.442
</td>
<td style="text-align:right;">
3.796
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
0.043
</td>
</tr>
</tbody>
</table>
<p>###PRLR</p>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PRLR ~ Batch
</td>
<td style="text-align:right;">
159.130
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.495
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + Status
</td>
<td style="text-align:right;">
160.715
</td>
<td style="text-align:right;">
-1.585
</td>
<td style="text-align:right;">
0.587
</td>
<td style="text-align:right;">
0.542
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + mean T
</td>
<td style="text-align:right;">
160.618
</td>
<td style="text-align:right;">
-1.489
</td>
<td style="text-align:right;">
0.547
</td>
<td style="text-align:right;">
0.535
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
155.739
</td>
<td style="text-align:right;">
3.391
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
0.010
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nucleus-taenia-tna" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Nucleus Taenia (TnA)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Nucleus Taenia (TnA). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
ESR2
</td>
<td style="text-align:right;">
182.462
</td>
<td style="text-align:right;">
0.810
</td>
<td style="text-align:right;">
0.248
</td>
<td style="text-align:right;">
3.269
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.079
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
65.301
</td>
<td style="text-align:right;">
0.835
</td>
<td style="text-align:right;">
0.426
</td>
<td style="text-align:right;">
1.962
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
0.480
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
604.216
</td>
<td style="text-align:right;">
-0.420
</td>
<td style="text-align:right;">
0.185
</td>
<td style="text-align:right;">
-2.274
</td>
<td style="text-align:right;">
0.023
</td>
<td style="text-align:right;">
0.521
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
LOC113983498
</td>
<td style="text-align:right;">
67.903
</td>
<td style="text-align:right;">
0.831
</td>
<td style="text-align:right;">
0.412
</td>
<td style="text-align:right;">
2.019
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.592
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
AVPR1A
</td>
<td style="text-align:right;">
172.360
</td>
<td style="text-align:right;">
-0.708
</td>
<td style="text-align:right;">
0.349
</td>
<td style="text-align:right;">
-2.025
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.605
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
27.859
</td>
<td style="text-align:right;">
-0.353
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:right;">
-2.043
</td>
<td style="text-align:right;">
0.041
</td>
<td style="text-align:right;">
0.595
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/TNA%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-nucleus-taenia" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Interaction in the Nucleus Taenia</h2>
<div id="cyp19a1" class="section level3" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> CYP19A1</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CYP19A1 ~ Batch
</td>
<td style="text-align:right;">
180.778
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.983
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
CYP19A1 ~ Batch + Status
</td>
<td style="text-align:right;">
180.608
</td>
<td style="text-align:right;">
0.169
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.511
</td>
</tr>
<tr>
<td style="text-align:left;">
CYP19A1 ~ Batch + mean T
</td>
<td style="text-align:right;">
182.168
</td>
<td style="text-align:right;">
-1.391
</td>
<td style="text-align:right;">
0.511
</td>
<td style="text-align:right;">
0.338
</td>
</tr>
<tr>
<td style="text-align:left;">
CYP19A1 ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
180.918
</td>
<td style="text-align:right;">
-0.140
</td>
<td style="text-align:right;">
0.156
</td>
<td style="text-align:right;">
0.048
</td>
</tr>
</tbody>
</table>
</div>
<div id="prlr-1" class="section level3" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> PRLR</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PRLR ~ Batch
</td>
<td style="text-align:right;">
181.935
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.144
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + Status
</td>
<td style="text-align:right;">
180.593
</td>
<td style="text-align:right;">
1.342
</td>
<td style="text-align:right;">
0.123
</td>
<td style="text-align:right;">
0.348
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + mean T
</td>
<td style="text-align:right;">
183.635
</td>
<td style="text-align:right;">
-1.700
</td>
<td style="text-align:right;">
0.644
</td>
<td style="text-align:right;">
0.570
</td>
</tr>
<tr>
<td style="text-align:left;">
PRLR ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
179.433
</td>
<td style="text-align:right;">
2.502
</td>
<td style="text-align:right;">
0.105
</td>
<td style="text-align:right;">
0.014
</td>
</tr>
</tbody>
</table>
</div>
<div id="vip-2" class="section level3" number="10.1.3">
<h3><span class="header-section-number">10.1.3</span> VIP</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIP ~ Batch
</td>
<td style="text-align:right;">
158.993
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.294
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + Status
</td>
<td style="text-align:right;">
160.028
</td>
<td style="text-align:right;">
-1.035
</td>
<td style="text-align:right;">
0.408
</td>
<td style="text-align:right;">
0.212
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + mean T
</td>
<td style="text-align:right;">
158.461
</td>
<td style="text-align:right;">
0.533
</td>
<td style="text-align:right;">
0.180
</td>
<td style="text-align:right;">
0.067
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
158.497
</td>
<td style="text-align:right;">
0.496
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.030
</td>
</tr>
</tbody>
</table>
<p>nothing compelling.</p>
</div>
<div id="vipr1" class="section level3" number="10.1.4">
<h3><span class="header-section-number">10.1.4</span> VIPR1</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIPR1 ~ Batch
</td>
<td style="text-align:right;">
131.796
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
0.528
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~ Batch + Status
</td>
<td style="text-align:right;">
133.719
</td>
<td style="text-align:right;">
-1.924
</td>
<td style="text-align:right;">
0.816
</td>
<td style="text-align:right;">
0.374
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~ Batch + mean T
</td>
<td style="text-align:right;">
130.893
</td>
<td style="text-align:right;">
0.902
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.130
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~ Batch + Status * mean T
</td>
<td style="text-align:right;">
129.034
</td>
<td style="text-align:right;">
2.761
</td>
<td style="text-align:right;">
0.096
</td>
<td style="text-align:right;">
0.040
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="arcopallium-intermedium-ai" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Arcopallium Intermedium (Ai)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Arcopallium Intermedium (Ai). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
28.989
</td>
<td style="text-align:right;">
-0.530
</td>
<td style="text-align:right;">
0.237
</td>
<td style="text-align:right;">
-2.237
</td>
<td style="text-align:right;">
0.025
</td>
<td style="text-align:right;">
0.499
</td>
</tr>
<tr>
<td style="text-align:left;">
mean T
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
251.548
</td>
<td style="text-align:right;">
0.373
</td>
<td style="text-align:right;">
0.158
</td>
<td style="text-align:right;">
2.355
</td>
<td style="text-align:right;">
0.019
</td>
<td style="text-align:right;">
0.427
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
PRLR
</td>
<td style="text-align:right;">
259.391
</td>
<td style="text-align:right;">
-0.423
</td>
<td style="text-align:right;">
0.199
</td>
<td style="text-align:right;">
-2.122
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
0.592
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
251.548
</td>
<td style="text-align:right;">
0.229
</td>
<td style="text-align:right;">
0.103
</td>
<td style="text-align:right;">
2.218
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
0.427
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/AI%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interaction-in-the-arcopallium-intermedium" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Interaction in the Arcopallium Intermedium</h2>
<div id="avp" class="section level3" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> AVP</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the arcopallium intermedium (ai) for how AVP expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AVP ~ 1
</td>
<td style="text-align:right;">
129.443
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~ Status
</td>
<td style="text-align:right;">
131.298
</td>
<td style="text-align:right;">
-1.855
</td>
<td style="text-align:right;">
0.738
</td>
<td style="text-align:right;">
0.843
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~ mean T
</td>
<td style="text-align:right;">
131.130
</td>
<td style="text-align:right;">
-1.687
</td>
<td style="text-align:right;">
0.623
</td>
<td style="text-align:right;">
0.644
</td>
</tr>
<tr>
<td style="text-align:left;">
AVP ~Status * mean T
</td>
<td style="text-align:right;">
131.940
</td>
<td style="text-align:right;">
-2.497
</td>
<td style="text-align:right;">
0.169
</td>
<td style="text-align:right;">
0.031
</td>
</tr>
</tbody>
</table>
<p>Looks like the parameterised model is worse than the null model.</p>
</div>
<div id="avpr1a-2" class="section level3" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> AVPR1A</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Not very convincing</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the arcopallium intermedium (ai) for how AVPR1A expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AVPR1A ~ 1
</td>
<td style="text-align:right;">
132.860
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ Status
</td>
<td style="text-align:right;">
134.692
</td>
<td style="text-align:right;">
0.719
</td>
<td style="text-align:right;">
0.404
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~ mean T
</td>
<td style="text-align:right;">
134.539
</td>
<td style="text-align:right;">
0.618
</td>
<td style="text-align:right;">
0.610
</td>
</tr>
<tr>
<td style="text-align:left;">
AVPR1A ~Status * mean T
</td>
<td style="text-align:right;">
135.096
</td>
<td style="text-align:right;">
0.216
</td>
<td style="text-align:right;">
0.027
</td>
</tr>
</tbody>
</table>
<p>And indeed this model does not appear to be bette either</p>
<p>###VIPR1</p>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the arcopallium intermedium (ai) for how VIPR1 expression is related to our interest variables
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIPR1 ~ 1
</td>
<td style="text-align:right;">
97.595
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~ Status
</td>
<td style="text-align:right;">
99.283
</td>
<td style="text-align:right;">
-1.687
</td>
<td style="text-align:right;">
0.623
</td>
<td style="text-align:right;">
0.843
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~ mean T
</td>
<td style="text-align:right;">
96.281
</td>
<td style="text-align:right;">
1.315
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:right;">
0.644
</td>
</tr>
<tr>
<td style="text-align:left;">
VIPR1 ~Status * mean T
</td>
<td style="text-align:right;">
92.123
</td>
<td style="text-align:right;">
5.473
</td>
<td style="text-align:right;">
0.031
</td>
<td style="text-align:right;">
0.038
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="checking-other-multiple-hits-in-the-ai" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Checking other multiple hits in the AI</h2>
<div id="vip-3" class="section level3" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> VIP</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Model validation in the Arcopallium Intermedium (AI) for how VIP expression, status and mean T influences social behavior
</caption>
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
VIP ~ 1
</td>
<td style="text-align:right;">
125.360
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Strength
</td>
<td style="text-align:right;">
122.290
</td>
<td style="text-align:right;">
3.070
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.027
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Status
</td>
<td style="text-align:right;">
125.416
</td>
<td style="text-align:right;">
-0.057
</td>
<td style="text-align:right;">
0.220
</td>
<td style="text-align:right;">
1.000
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ mean T
</td>
<td style="text-align:right;">
121.385
</td>
<td style="text-align:right;">
3.974
</td>
<td style="text-align:right;">
0.031
</td>
<td style="text-align:right;">
0.019
</td>
</tr>
<tr>
<td style="text-align:left;">
VIP ~ Status * mean T
</td>
<td style="text-align:right;">
123.986
</td>
<td style="text-align:right;">
1.373
</td>
<td style="text-align:right;">
0.366
</td>
<td style="text-align:right;">
0.236
</td>
</tr>
</tbody>
</table>
<p>Ok, looks like it’s tricky to distinguish between the two interest models ~ Strength and ~ Mean T</p>
<p>We have some very low absolute expression here too…</p>
</div>
</div>
</div>
<div id="lateral-septum-ls" class="section level1" number="12">
<h1><span class="header-section-number">12</span> Lateral Septum (LS)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Significant results from DESeq2 before FDR correction in the Lateral Septum (LS). Where pvalue is the raw DESeq2 p-value, padj is the DESeq2 FDR-corrected p-value (for the number of genes in the sample per model).
</caption>
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
GNRH1
</td>
<td style="text-align:right;">
22.089
</td>
<td style="text-align:right;">
1.383
</td>
<td style="text-align:right;">
0.686
</td>
<td style="text-align:right;">
2.017
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.927
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
SRD5A2
</td>
<td style="text-align:right;">
83.943
</td>
<td style="text-align:right;">
0.670
</td>
<td style="text-align:right;">
0.192
</td>
<td style="text-align:right;">
3.492
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.365
</td>
</tr>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
VIP
</td>
<td style="text-align:right;">
19.078
</td>
<td style="text-align:right;">
-1.440
</td>
<td style="text-align:right;">
0.681
</td>
<td style="text-align:right;">
-2.116
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
0.895
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
GNRH1
</td>
<td style="text-align:right;">
23.067
</td>
<td style="text-align:right;">
1.126
</td>
<td style="text-align:right;">
0.339
</td>
<td style="text-align:right;">
3.324
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/LS%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interactions-in-the-lateral-septum" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Interactions in the Lateral Septum</h2>
<div id="ar-2" class="section level3" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> AR</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
AR ~ 1
</td>
<td style="text-align:right;">
150.138
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status
</td>
<td style="text-align:right;">
151.096
</td>
<td style="text-align:right;">
-0.958
</td>
<td style="text-align:right;">
0.363
</td>
<td style="text-align:right;">
0.324
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ mean T
</td>
<td style="text-align:right;">
152.137
</td>
<td style="text-align:right;">
-1.999
</td>
<td style="text-align:right;">
0.978
</td>
<td style="text-align:right;">
0.976
</td>
</tr>
<tr>
<td style="text-align:left;">
AR ~ Status * mean T
</td>
<td style="text-align:right;">
148.336
</td>
<td style="text-align:right;">
1.802
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.004
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="bed-nucleus-of-the-stria-terminalis-bstm" class="section level1" number="13">
<h1><span class="header-section-number">13</span> Bed Nucleus of the Stria Terminalis (BSTm)</h1>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Test
</th>
<th style="text-align:left;">
gene
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Status
</td>
<td style="text-align:left;">
AR
</td>
<td style="text-align:right;">
691.964
</td>
<td style="text-align:right;">
0.567
</td>
<td style="text-align:right;">
0.168
</td>
<td style="text-align:right;">
3.379
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.210
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
ESR2
</td>
<td style="text-align:right;">
381.353
</td>
<td style="text-align:right;">
-0.299
</td>
<td style="text-align:right;">
0.150
</td>
<td style="text-align:right;">
-1.989
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.604
</td>
</tr>
<tr>
<td style="text-align:left;">
Strength
</td>
<td style="text-align:left;">
PGR
</td>
<td style="text-align:right;">
174.573
</td>
<td style="text-align:right;">
-0.334
</td>
<td style="text-align:right;">
0.096
</td>
<td style="text-align:right;">
-3.461
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.152
</td>
</tr>
</tbody>
</table>
<p><img src="06_candidate_gene_analysis_files/figure-html/BSTm%20candidate%20gene%20plots-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="interactions-in-the-bed-nucleus-of-the-stria-terminalis" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Interactions in the Bed Nucleus of the Stria Terminalis</h2>
<div id="esr1-1" class="section level3" number="13.1.1">
<h3><span class="header-section-number">13.1.1</span> ESR1</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ESR1 ~ 1
</td>
<td style="text-align:right;">
168.264
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ + Status
</td>
<td style="text-align:right;">
169.980
</td>
<td style="text-align:right;">
-1.716
</td>
<td style="text-align:right;">
0.635
</td>
<td style="text-align:right;">
0.666
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ mean T
</td>
<td style="text-align:right;">
167.759
</td>
<td style="text-align:right;">
0.505
</td>
<td style="text-align:right;">
0.159
</td>
<td style="text-align:right;">
0.167
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR1 ~ Status * mean T
</td>
<td style="text-align:right;">
167.454
</td>
<td style="text-align:right;">
0.810
</td>
<td style="text-align:right;">
0.102
</td>
<td style="text-align:right;">
0.044
</td>
</tr>
</tbody>
</table>
</div>
<div id="esr2" class="section level3" number="13.1.2">
<h3><span class="header-section-number">13.1.2</span> ESR2</h3>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
AIC
</th>
<th style="text-align:right;">
deltaAIC
</th>
<th style="text-align:right;">
lm_p
</th>
<th style="text-align:right;">
DESeq2_raw_p
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ESR2 ~ 1
</td>
<td style="text-align:right;">
158.939
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR2 ~ Status
</td>
<td style="text-align:right;">
160.599
</td>
<td style="text-align:right;">
-1.660
</td>
<td style="text-align:right;">
0.635
</td>
<td style="text-align:right;">
0.408
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR2 ~ mean T
</td>
<td style="text-align:right;">
158.141
</td>
<td style="text-align:right;">
0.798
</td>
<td style="text-align:right;">
0.159
</td>
<td style="text-align:right;">
0.161
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR2 ~ Strength
</td>
<td style="text-align:right;">
158.301
</td>
<td style="text-align:right;">
0.638
</td>
<td style="text-align:right;">
0.148
</td>
<td style="text-align:right;">
0.047
</td>
</tr>
<tr>
<td style="text-align:left;">
ESR2 ~ Status * mean T
</td>
<td style="text-align:right;">
155.896
</td>
<td style="text-align:right;">
3.042
</td>
<td style="text-align:right;">
0.102
</td>
<td style="text-align:right;">
0.008
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heat-map-of-all-candidate-genes" class="section level1" number="14">
<h1><span class="header-section-number">14</span> Heat map of all candidate genes</h1>
<p><img src="06_candidate_gene_analysis_files/figure-html/unnamed-chunk-50-1.png" width="2880" style="display: block; margin: auto;" /></p>
<pre><code>FALSE png 
FALSE   2</code></pre>
<pre><code>FALSE R version 4.0.2 (2020-06-22)
FALSE Platform: x86_64-w64-mingw32/x64 (64-bit)
FALSE Running under: Windows 10 x64 (build 19042)
FALSE 
FALSE Matrix products: default
FALSE 
FALSE locale:
FALSE [1] LC_COLLATE=English_Australia.1252  LC_CTYPE=English_Australia.1252   
FALSE [3] LC_MONETARY=English_Australia.1252 LC_NUMERIC=C                      
FALSE [5] LC_TIME=English_Australia.1252    
FALSE 
FALSE attached base packages:
FALSE  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
FALSE  [8] datasets  methods   base     
FALSE 
FALSE other attached packages:
FALSE  [1] WGCNA_1.69                  fastcluster_1.1.25         
FALSE  [3] dynamicTreeCut_1.63-1       RColorBrewer_1.1-2         
FALSE  [5] pheatmap_1.0.12             MASS_7.3-51.6              
FALSE  [7] wesanderson_0.3.6           VennDiagram_1.6.20         
FALSE  [9] futile.logger_1.4.3         kableExtra_1.3.1           
FALSE [11] DESeq2_1.28.1               SummarizedExperiment_1.18.1
FALSE [13] DelayedArray_0.14.0         matrixStats_0.56.0         
FALSE [15] Biobase_2.48.0              GenomicRanges_1.40.0       
FALSE [17] GenomeInfoDb_1.24.0         IRanges_2.22.2             
FALSE [19] S4Vectors_0.26.1            BiocGenerics_0.34.0        
FALSE [21] stringr_1.4.0               ggpubr_0.4.0               
FALSE [23] ggplot2_3.3.5               reshape2_1.4.4             
FALSE [25] plyr_1.8.6                 
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] colorspace_1.4-1       ggsignif_0.6.0         ellipsis_0.3.2        
FALSE   [4] rio_0.5.16             htmlTable_2.0.1        XVector_0.28.0        
FALSE   [7] base64enc_0.1-3        rstudioapi_0.13        farver_2.0.3          
FALSE  [10] bit64_0.9-7            AnnotationDbi_1.50.0   fansi_0.4.1           
FALSE  [13] xml2_1.3.2             codetools_0.2-16       splines_4.0.2         
FALSE  [16] doParallel_1.0.15      impute_1.62.0          geneplotter_1.66.0    
FALSE  [19] knitr_1.29             Formula_1.2-3          broom_0.7.10          
FALSE  [22] annotate_1.66.0        cluster_2.1.0          GO.db_3.11.4          
FALSE  [25] png_0.1-7              compiler_4.0.2         httr_1.4.2            
FALSE  [28] backports_1.1.7        assertthat_0.2.1       Matrix_1.2-18         
FALSE  [31] fastmap_1.1.0          formatR_1.8            acepack_1.4.1         
FALSE  [34] htmltools_0.5.2        tools_4.0.2            gtable_0.3.0          
FALSE  [37] glue_1.5.1             GenomeInfoDbData_1.2.3 dplyr_1.0.7           
FALSE  [40] Rcpp_1.0.7             carData_3.0-4          cellranger_1.1.0      
FALSE  [43] jquerylib_0.1.4        vctrs_0.3.8            nlme_3.1-148          
FALSE  [46] preprocessCore_1.50.0  iterators_1.0.12       xfun_0.29             
FALSE  [49] openxlsx_4.1.5         rvest_1.0.2            lifecycle_1.0.1       
FALSE  [52] rstatix_0.6.0          XML_3.99-0.5           zlibbioc_1.34.0       
FALSE  [55] scales_1.1.1           hms_1.1.1              lambda.r_1.2.4        
FALSE  [58] yaml_2.2.1             curl_4.3               gridExtra_2.3         
FALSE  [61] memoise_1.1.0          rpart_4.1-15           latticeExtra_0.6-29   
FALSE  [64] stringi_1.4.6          RSQLite_2.2.0          highr_0.8             
FALSE  [67] genefilter_1.70.0      foreach_1.5.0          checkmate_2.0.0       
FALSE  [70] zip_2.0.4              BiocParallel_1.22.0    rlang_0.4.12          
FALSE  [73] pkgconfig_2.0.3        bitops_1.0-6           evaluate_0.14         
FALSE  [76] lattice_0.20-41        purrr_0.3.4            labeling_0.3          
FALSE  [79] htmlwidgets_1.5.3      cowplot_1.0.0          bit_4.0.4             
FALSE  [82] tidyselect_1.1.0       magrittr_2.0.1         R6_2.4.1              
FALSE  [85] Hmisc_4.4-0            generics_0.0.2         DBI_1.1.0             
FALSE  [88] mgcv_1.8-31            pillar_1.6.4           haven_2.3.1           
FALSE  [91] foreign_0.8-80         withr_2.4.3            nnet_7.3-14           
FALSE  [94] survival_3.2-3         abind_1.4-5            RCurl_1.98-1.2        
FALSE  [97] tibble_3.1.6           crayon_1.4.2           car_3.0-8             
FALSE [100] futile.options_1.0.1   utf8_1.1.4             rmarkdown_2.11        
FALSE [103] jpeg_0.1-8.1           locfit_1.5-9.4         readxl_1.3.1          
FALSE [106] data.table_1.13.0      blob_1.2.1             forcats_0.5.1         
FALSE [109] digest_0.6.25          webshot_0.5.2          xtable_1.8-4          
FALSE [112] tidyr_1.1.4            munsell_0.5.0          viridisLite_0.4.0</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
